<!DOCTYPE html><html lang="en" class="no-js"><head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A collection of bioinformatics workflows for genomic characterization, submission preparation, and genomic epidemiology of pathogens of public health concern.">
      
      
        <meta name="author" content="Theiagen Genomics">
      
      
        <link rel="canonical" href="https://theiagen.github.io/public_health_bioinformatics/main/guides/gambit_database/">
      
      
        <link rel="prev" href="../gambit/">
      
      
        <link rel="next" href="../phylogenetics/">
      
      
      <link rel="icon" href="../../assets/logos/Theiagen-Symbol-Standard-01.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.20">
    
    
      
        <title>Guide to GAMBIT Database Creation - Public Health Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.e53b48f4.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#gambit-database-creation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version release.
  <a href="../../.."> 
    <strong>Click here to go to the latest version release.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-header__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Public Health Bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Guide to GAMBIT Database Creation
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo" aria-label="Switch to Dark Mode" type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to Dark Mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"></path></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo" aria-label="Switch to Light Mode" type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to Light Mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m3.55 19.09 1.41 1.41 1.8-1.79-1.42-1.42M12 6c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6c0-3.32-2.69-6-6-6m8 7h3v-2h-3m-2.76 7.71 1.8 1.79 1.41-1.41-1.79-1.8M20.45 5l-1.41-1.4-1.8 1.79 1.42 1.42M13 1h-2v3h2M6.76 5.39 4.96 3.6 3.55 5l1.79 1.81zM1 13h3v-2H1m12 9h-2v3h2"></path></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../.." class="md-tabs__link">
          
  
  
  Public Health Bioinformatics

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-nav__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    Public Health Bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Public Health Bioinformatics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Public Health Bioinformatics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Getting Started
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            Getting Started
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting_started/commandline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With the Command-Line
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting_started/terra/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With Terra.bio
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting_started/philosophy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Workflow Failure Philosophy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Other Resources
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_3">
            <span class="md-nav__icon md-icon"></span>
            Other Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sops/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Available SOPs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../versions/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Database and Docker Versions
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" checked>
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../gambit/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT Database Creation
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT Database Creation
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#database-creation-process" class="md-nav__link">
    <span class="md-ellipsis">
      Database Creation Process
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Database Creation Process">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#creating-a-gambit-database-from-a-pre-curated-list-of-genomes-gtdb-and-gambitdb" class="md-nav__link">
    <span class="md-ellipsis">
      Creating a GAMBIT database from a pre-curated list of genomes - GTDB and GAMBITdb
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Creating a GAMBIT database from a pre-curated list of genomes - GTDB and GAMBITdb">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#identifying-data-source" class="md-nav__link">
    <span class="md-ellipsis">
      Identifying Data Source
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#curation-of-genomic-data" class="md-nav__link">
    <span class="md-ellipsis">
      Curation of Genomic Data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-gambitdb" class="md-nav__link">
    <span class="md-ellipsis">
      Using GAMBITdb
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#manually-creating-and-curating-a-gambit-database" class="md-nav__link">
    <span class="md-ellipsis">
      Manually creating and curating a GAMBIT database
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Manually creating and curating a GAMBIT database">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sourcing-genomic-information" class="md-nav__link">
    <span class="md-ellipsis">
      Sourcing Genomic Information
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sourcing Genomic Information">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-download-genomes-to-be-included-in-the-database" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Download genomes to be included in the database
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-perform-quality-control-on-the-sourced-genomes-optional" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Perform quality control on the sourced genomes (optional)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-compress-genomes-into-the-gambit-signatures-file-and-calculate-pairwise-distances-between-all-genomes" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Compress genomes into the GAMBIT signatures file and calculate pairwise distances between all genomes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 3: Compress genomes into the GAMBIT signatures file and calculate pairwise distances between all genomes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure1" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 1
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-calculate-diameters-for-taxa-and-resolve-overlaps" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Calculate diameters for taxa and resolve overlaps
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-5-download-taxa" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Download taxa
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-6-create-the-database" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6: Create the database
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-7-finalise-the-signatures-file" class="md-nav__link">
    <span class="md-ellipsis">
      Step 7: Finalise the signatures file
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#help-available" class="md-nav__link">
    <span class="md-ellipsis">
      Help Available
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../phylogenetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Phylogenetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../custom_organisms/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Running Custom Organisms on TheiaCoV
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1">
        
          
          <label class="md-nav__link" for="__nav_1_5_1" id="__nav_1_5_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Type
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Type
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows_overview/workflows_type/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_2" id="__nav_1_5_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Import
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_2">
            <span class="md-nav__icon md-icon"></span>
            Data Import
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_3" id="__nav_1_5_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Genomic Characterization
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_3">
            <span class="md-nav__icon md-icon"></span>
            Genomic Characterization
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_4" id="__nav_1_5_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Construction
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_4">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Construction
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_5" id="__nav_1_5_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Placement
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_5">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Placement
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_6" id="__nav_1_5_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Public Data Sharing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_6">
            <span class="md-nav__icon md-icon"></span>
            Public Data Sharing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_7">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_7" id="__nav_1_5_1_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exporting Data from Terra
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_7">
            <span class="md-nav__icon md-icon"></span>
            Exporting Data from Terra
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_8">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_8" id="__nav_1_5_1_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Standalone
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_8">
            <span class="md-nav__icon md-icon"></span>
            Standalone
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2" id="__nav_1_5_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Kingdom
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Kingdom
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows_overview/workflows_kingdom/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_2" id="__nav_1_5_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Any Taxa
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_2">
            <span class="md-nav__icon md-icon"></span>
            Any Taxa
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_3" id="__nav_1_5_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Bacteria
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_3">
            <span class="md-nav__icon md-icon"></span>
            Bacteria
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_4" id="__nav_1_5_2_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Mycotics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_4">
            <span class="md-nav__icon md-icon"></span>
            Mycotics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_5" id="__nav_1_5_2_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Viral
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_5">
            <span class="md-nav__icon md-icon"></span>
            Viral
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows Alphabetically
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            Workflows Alphabetically
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows_overview/workflows_alphabetically/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Contributing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            Contributing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/doc_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Documentation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/code_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Code
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#database-creation-process" class="md-nav__link">
    <span class="md-ellipsis">
      Database Creation Process
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Database Creation Process">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#creating-a-gambit-database-from-a-pre-curated-list-of-genomes-gtdb-and-gambitdb" class="md-nav__link">
    <span class="md-ellipsis">
      Creating a GAMBIT database from a pre-curated list of genomes - GTDB and GAMBITdb
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Creating a GAMBIT database from a pre-curated list of genomes - GTDB and GAMBITdb">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#identifying-data-source" class="md-nav__link">
    <span class="md-ellipsis">
      Identifying Data Source
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#curation-of-genomic-data" class="md-nav__link">
    <span class="md-ellipsis">
      Curation of Genomic Data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-gambitdb" class="md-nav__link">
    <span class="md-ellipsis">
      Using GAMBITdb
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#manually-creating-and-curating-a-gambit-database" class="md-nav__link">
    <span class="md-ellipsis">
      Manually creating and curating a GAMBIT database
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Manually creating and curating a GAMBIT database">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sourcing-genomic-information" class="md-nav__link">
    <span class="md-ellipsis">
      Sourcing Genomic Information
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sourcing Genomic Information">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-download-genomes-to-be-included-in-the-database" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Download genomes to be included in the database
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-perform-quality-control-on-the-sourced-genomes-optional" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Perform quality control on the sourced genomes (optional)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-compress-genomes-into-the-gambit-signatures-file-and-calculate-pairwise-distances-between-all-genomes" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Compress genomes into the GAMBIT signatures file and calculate pairwise distances between all genomes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 3: Compress genomes into the GAMBIT signatures file and calculate pairwise distances between all genomes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure1" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 1
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-calculate-diameters-for-taxa-and-resolve-overlaps" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Calculate diameters for taxa and resolve overlaps
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-5-download-taxa" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Download taxa
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-6-create-the-database" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6: Create the database
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-7-finalise-the-signatures-file" class="md-nav__link">
    <span class="md-ellipsis">
      Step 7: Finalise the signatures file
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#help-available" class="md-nav__link">
    <span class="md-ellipsis">
      Help Available
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="gambit-database-creation">GAMBIT Database Creation<a class="headerlink" href="#gambit-database-creation" title="Permanent link">¶</a></h1>
<p>GAMBIT (Genomic Approximation Method for Bacterial Identification and Tracking) incorporates k-mer based strategy for the identification of taxonomic information with a highly curated searchable database.</p>
<p>A GAMBIT databases consist of two files:</p>
<ol>
<li>A <strong>signatures file</strong> containing the GAMBIT signatures (compressed representations) of all genomes represented in the database; </li>
<li>A <strong>metadata file</strong> relating the represented genomes to their genome accessions, taxonomic identifications, and species thresholds.</li>
</ol>
<p>The goal in creating a database for GAMBIT is two-fold:</p>
<ul>
<li>Genomes representing each species in the database must contain enough distinction from other species so that similarity thresholds can be extracted;</li>
<li>The highest level of confidence is obtained that a genome in the database is actually from the species that matched its labelled identification.</li>
</ul>
<div class="grid cards">
<ul>
<li>
<p><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16h-.58l-.81-.81A7.07 7.07 0 0 0 18 11c0-3.87-3.13-7-7-7-1.5 0-3 .5-4.21 1.4-3.09 2.32-3.72 6.71-1.4 9.8s6.71 3.72 9.8 1.4l.81.81V18l5 5 2-2zm-7 0c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5M3 6 1 8V1h7L6 3H3zm18-5v7l-2-2V3h-3l-2-2zM6 19l2 2H1v-7l2 2v3z"></path></svg></span> <strong>GAMBIT allows for rapid taxonomic identification of microbial pathogens.</strong></p>
<hr>
<p>The GAMBIT database is designed to facilitate the exploration and analysis of genomic data from a wide variety of prokaryotic or fungal isolates. It integrates and annotates genomic information based on information from curated sources. Several sources of complete and draft genomic sequences exist, with the most common being <a href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a>, <a href="https://www.ncbi.nlm.nih.gov/genbank/">GenBank</a> and <a href="https://gtdb.ecogenomic.org/">GTDB</a>. Alternative sources, such as <a href="https://bakrep.computational.bio/">BakRep</a>, can also be used as long as taxonomic information is available.</p>
</li>
<li>
<p><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 3c4.42 0 8 1.79 8 4s-3.58 4-8 4-8-1.79-8-4 3.58-4 8-4M4 9c0 2.21 3.58 4 8 4s8-1.79 8-4v3.08L19 12c-2.59 0-4.8 1.64-5.64 3.94L12 16c-4.42 0-8-1.79-8-4zm0 5c0 2.21 3.58 4 8 4h1c0 1.05.27 2.04.75 2.9L12 21c-4.42 0-8-1.79-8-4zm14 7.08-2.75-3 1.16-1.16L18 18.5l3.59-3.58 1.16 1.41z"></path></svg></span> <strong>Several pre-computed GAMBIT databases are available for prokaryotic and fungal genomes.</strong></p>
<hr>
<p>The full list of databases available can be consulted <a href="../gambit/#gambit-databases">here</a>, with information on what curation steps were followed and what genomes are included.</p>
</li>
</ul>
</div>
<hr>
<h2 id="database-creation-process">Database Creation Process<a class="headerlink" href="#database-creation-process" title="Permanent link">¶</a></h2>
<h3 id="creating-a-gambit-database-from-a-pre-curated-list-of-genomes-gtdb-and-gambitdb">Creating a GAMBIT database from a pre-curated list of genomes - GTDB and GAMBITdb<a class="headerlink" href="#creating-a-gambit-database-from-a-pre-curated-list-of-genomes-gtdb-and-gambitdb" title="Permanent link">¶</a></h3>
<p>The creation and curation of a GAMBIT database is a laborious process which involves sourcing genomic information, setting minimum quality thresholds, and finally building a signature and a database file.</p>
<h4 id="identifying-data-source">Identifying Data Source<a class="headerlink" href="#identifying-data-source" title="Permanent link">¶</a></h4>
<p>As of <a href="../gambit/#gambit-gtdb-database-v200">v2.0.0 of the GAMBIT Prokaryotic database</a>, <a href="https://gtdb.ecogenomic.org/">GTDB</a> is the selected source of taxonomic information for genomes sourced from <a href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a> and <a href="https://www.ncbi.nlm.nih.gov/genbank/">GenBank</a>. Automation is available through the <a href="https://github.com/gambit-suite/gambitdb">GAMBITdb</a> which takes as input a GTDB release metadata spreadsheet including information on taxonomy and assembly quality metrics. GTDB’s release spreadsheets are available at <a href="https://data.gtdb.ecogenomic.org/releases/">https://data.gtdb.ecogenomic.org/releases</a>.</p>
<div style="float: left; width: 50%; padding: 20px;">
<h4 id="curation-of-genomic-data">Curation of Genomic Data<a class="headerlink" href="#curation-of-genomic-data" title="Permanent link">¶</a></h4>
<p>By default, genomes with <a href="../gambit/#technical-details">less than 0.1 diameter</a> between them are not included if compression is enabled, allowing the database to remain at a fairly low size (less than 3GB), instead being represented by a single centroid genome. This is because many genomes are identical or nearly identical to others in the public archives and there is no benefit to having multiple examples, for example, in an outbreak dozens of identical genomes could be sequenced and after the first, there is no new information for classification.</p>
<p>The inclusion criteria for genome assemblies for addition to the GAMBIT database are as follows:</p>
<ul>
<li><strong>Completeness</strong> computed by <a href="https://github.com/Ecogenomics/CheckM/wiki">CheckM</a> being equal or over <strong>97%</strong>;</li>
<li><strong>Contamination</strong>  computed by <a href="https://github.com/Ecogenomics/CheckM/wiki">CheckM</a> being less than <strong>3%</strong>;</li>
<li>Having less than <strong>300 contigs</strong>.</li>
</ul>
<p>Species missing from previous iterations of the database were attempted to be added with looser quality criteria:</p>
<ul>
<li><strong>Completeness</strong> computed by <a href="https://github.com/Ecogenomics/CheckM/wiki">CheckM</a> being equal or over <strong>90%;</strong></li>
<li><strong>Contamination</strong>  computed by <a href="https://github.com/Ecogenomics/CheckM/wiki">CheckM</a> being less than <strong>5%</strong>;</li>
<li>Having less than <strong>600 contigs</strong>.</li>
</ul>
<p>Curation is included in the post-processing steps of <a href="https://github.com/gambit-suite/gambitdb">GAMBITdb</a>. This is primarily because taxonomy has evolved over more than a century in a pre-genomics world and has many edge cases. Examples include:</p>
<ul>
<li>The GTDB candidates for an existing genus were collapsed (e.g. genus_A, genus_B becomes genus). GTDB proposes new genus/species based on ANI however, these are not currently formally accepted by the community.</li>
<li>The following species were updated:<ul>
<li><em>Shigella sp.</em><ul>
<li>This genus is not present in GTDB  as it is collapsed under <em>Escherichia coli</em>; This is a controversial decision within the community, particularly as Shigella has specific clinical presentations.</li>
<li>All Shigella genomes in RefSeq were added to the database with no clustering using default quality criteria.</li>
</ul>
</li>
<li><em>Mycolicibacterium/Mycolicibacter/Mycolicibacillus/Mycobacteroides/Mycobacterium</em> sp.<ul>
<li>All genomes available were used.</li>
</ul>
</li>
<li><em>Tropheryma whipplei</em><ul>
<li>This species has a low completeness score of 75%;</li>
<li>The CheckM completeness score was lowered to 70% for genomes belonging to this species.  CheckM was subsequently updated to a new version (not used in this instance) and completeness for this species is rectified, so this fix will be deprecated in the next release.</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div style="float: right; width: 50%; padding: 20px;">
<h4 id="using-gambitdb">Using GAMBITdb<a class="headerlink" href="#using-gambitdb" title="Permanent link">¶</a></h4>
<p>The <a href="https://github.com/gambit-suite">GAMBIT-suite</a> organization provides multiple repositories for tools for&nbsp;<a href="https://github.com/jlumpe/gambit">GAMBIT</a>. It includes the <a href="https://github.com/gambit-suite/gambitdb">GAMBITdb</a> repository which features several tools to generate a GAMBIT database from a GTDB’s release spreadsheet.</p>
<p>To run <a href="https://github.com/gambit-suite/gambitdb">GAMBITdb</a>, the following dependencies are required:</p>
<ul>
<li><a href="https://github.com/jlumpe/gambit">GAMBIT</a></li>
<li><a href="https://github.com/kblin/ncbi-genome-download">ncbi-genome-download</a></li>
<li><a href="https://github.com/gambit-suite/gambittools">GAMBITtools</a></li>
</ul>
<p>A Bash wrapper is available for creating a database from a GTDB spreadsheet (one for bacteria, one for archaea). This script will download the data, create a GAMBIT database and signatures files, and then check the recall against the downloaded files for QC purposes. It is the easiest way to create a database.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>run_gambitdb_bacteria.sh<span class="w"> </span>&lt;/path/to/working_directory&gt;<span class="w"> </span>&lt;/path/to/gtdb_metadata_spreadsheet.tsv&gt;<span class="w"> </span>&lt;num_cores&gt;
</span></code></pre></div>
<p>A collection of scripts is called by the Bash wrapper to:</p>
<ol>
<li>Parse a GTDB spreadsheet and output a list of accessions to download, a species taxon file and a genome metadata file;</li>
<li>Download the genomes from NCBI (GenBank and RefSeq)</li>
<li>Generate a GAMBIT database from a directory containing assemblies in FASTA format and a CSV file containing the assembly file and path, and the species taxon ID;</li>
</ol>
<div class="admonition tip">
<p class="admonition-title">Automatic Database Curation</p>
<p>These steps allow you to obtain a working GAMBIT database without much manual intervention, given that all dependencies are installed correctly in your system. As of v2.0.0, the <a href="https://github.com/gambit-suite/gambitdb/blob/main/scripts/gambitdb-manual-curation">curation steps have been automated</a> but require functionality deep access within the GAMBITdb software and are mostly for advanced usage. More detailed information is <a href="https://github.com/gambit-suite/gambitdb?tab=readme-ov-file#gambitdb">available within the GAMBITdb repository</a>.</p>
</div>
<p>Sometimes repairs are required as even with curation of the GTDB spreadsheet errors can make their way to the final database. A utility is available to perform maintenance and repair operations on a GAMBIT database:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>gambitdb-repair-db<span class="w"> </span>&lt;/path/to/metadata_file&gt;
</span></code></pre></div>
<p>Manual curation efforts have been automated and are also available:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>gambitdb-manual-curation<span class="w"> </span>&lt;/path/to/metadata_file&gt;<span class="w"> </span>&lt;/path/to/signatures_file&gt;<span class="w"> </span>&lt;/path/to/gtdb_metadata_spreadsheet.tsv&gt;
</span></code></pre></div>
</div>
<div style="clear: both;"></div>
<h2 id="manually-creating-and-curating-a-gambit-database">Manually creating and curating a GAMBIT database<a class="headerlink" href="#manually-creating-and-curating-a-gambit-database" title="Permanent link">¶</a></h2>
<p>The manual creation and curation of a GAMBIT database is a complex and sometimes difficult process and is not advised without guidance. Please feel free to reach out to <a href="mailto:support@theiagen.com">support@theiagen.com</a> with any questions you might have.</p>
<h3 id="sourcing-genomic-information">Sourcing Genomic Information<a class="headerlink" href="#sourcing-genomic-information" title="Permanent link">¶</a></h3>
<p>A GAMBIT database can be created from genomes retrieved from a wide variety of sources, not just <a href="https://gtdb.ecogenomic.org/">GTDB</a>. To do so, all the automated steps in the <a href="https://github.com/gambit-suite/gambitdb">GAMBITdb</a> need to be followed individually. Some officially supported databases, such as the <a href="../gambit/#gambit-fungal-database-v020">GAMBIT Fungal Database v0.2.0</a>, were created following these steps.</p>
<h4 id="step-1-download-genomes-to-be-included-in-the-database">Step 1: Download genomes to be included in the database<a class="headerlink" href="#step-1-download-genomes-to-be-included-in-the-database" title="Permanent link">¶</a></h4>
<p>Genomes may be downloaded from public repositories like NCBI and GTDB. Genomes must be in <strong>FASTA file format.</strong> It is also necessary to create a file listing each genome with its species identity. The <a href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/">RefSeq FTP site</a> is a useful resource for both tasks.</p>
<p>Example: <a href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/fungi/assembly_summary.txt">https://ftp.ncbi.nlm.nih.gov/genomes/refseq/fungi/assembly_summary.txt</a></p>
<h4 id="step-2-perform-quality-control-on-the-sourced-genomes-optional">Step 2: Perform quality control on the sourced genomes (optional)<a class="headerlink" href="#step-2-perform-quality-control-on-the-sourced-genomes-optional" title="Permanent link">¶</a></h4>
<p>It is advised to perform some quality assessment and control on the downloaded genomes to reduce the possibility of including low-quality and/or erroneous sequences in the database. <a href="https://github.com/ablab/quast">QUAST</a> is a popular software for obtaining useful reference-free metrics such as the number of contigs, number of basepairs, length of largest contig and N50. Tools like <a href="https://busco.ezlab.org/">BUSCO</a> and <a href="https://github.com/Ecogenomics/CheckM">CheckM</a> provide valuable information regarding the completeness and contamination of a genome</p>
<h4 id="step-3-compress-genomes-into-the-gambit-signatures-file-and-calculate-pairwise-distances-between-all-genomes">Step 3: Compress genomes into the GAMBIT signatures file and calculate pairwise distances between all genomes<a class="headerlink" href="#step-3-compress-genomes-into-the-gambit-signatures-file-and-calculate-pairwise-distances-between-all-genomes" title="Permanent link">¶</a></h4>
<p>Compressing genomes into a GAMBIT signatures file can be accomplished using <a href="https://github.com/jlumpe/gambit">GAMBIT</a>’s <code>gambit signatures</code> command:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>gambit<span class="w"> </span>signatures<span class="w"> </span>create<span class="w"> </span>-k<span class="w"> </span><span class="m">11</span><span class="w"> </span>-p<span class="w"> </span>ATGAC<span class="w"> </span>-o<span class="w"> </span>signatures.gs<span class="w"> </span>fasta_dir/*.fna.gz
</span></code></pre></div>
<div class="admonition tip">
<p class="admonition-title">K-mer Length</p>
<p>The length of the k-mer, controlled by the <code>-k</code> parameter, and the k-mer prefix, controlled by the <code>-p</code> parameter, need to match between the <strong>GAMBIT database</strong> and the way <strong>GAMBIT query</strong> is run.  <strong>We strongly advise that these parameters not to be changed!</strong></p>
</div>
<p>After the GAMBIT signatures file is created, a pairwise distance matrix for every genome in the signatures file can be created using the <code>gambit dist</code> command:</p>
<div class="language-jsx highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="nx">gambit</span><span class="w"> </span><span class="nx">dist</span><span class="w"> </span><span class="o">--</span><span class="nx">qs</span><span class="w"> </span><span class="nx">signatures</span><span class="p">.</span><span class="nx">gs</span><span class="w"> </span><span class="o">--</span><span class="nx">square</span><span class="w"> </span><span class="o">-</span><span class="nx">o</span><span class="w"> </span><span class="nx">pw_dists</span><span class="p">.</span><span class="nx">csv</span>
</span></code></pre></div>
<div class="admonition caption narrow">
<p class="admonition-title">GAMBIT Metadata Database Scheme</p>
<h5 id="figure1">Figure 1<a class="headerlink" href="#figure1" title="Permanent link">¶</a></h5>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-metadata-database.png" data-desc-position="bottom"><img alt="**Figure 1:** Scheme for GAMBIT’s metadata database." src="../../assets/figures/GAMBIT-metadata-database.png"></a></p>
<p><strong>Figure 1:</strong> Scheme for GAMBIT’s metadata database</p>
</div>
<h4 id="step-4-calculate-diameters-for-taxa-and-resolve-overlaps">Step 4: Calculate diameters for taxa and resolve overlaps<a class="headerlink" href="#step-4-calculate-diameters-for-taxa-and-resolve-overlaps" title="Permanent link">¶</a></h4>
<p>The defining feature of GAMBIT is the ability to determine if a query genome is genetically similar enough to the closest genome in the GAMBIT database to assign it to that species. For GAMBIT to work as intended, the database must be curated to ensure that no genomes are mislabeled and that species classifications are well-defined.</p>
<p>Most GAMBIT databases are curated by examining the pairwise GAMBIT distance matrix generated in step 3 and resolving any of the following issues:</p>
<ol>
<li><strong>Distantly related subspecies</strong><ol>
<li><strong>Manifestation:</strong> In the pairwise-distance matrix, there are two or more groups of genomes within the species that are closely related, but are highly distant from each other.</li>
<li><strong>Solution: C</strong>reate a subspecies taxon. </li>
</ol>
</li>
<li><strong>Outlier genomes that may be labelled with the incorrect species</strong><ol>
<li><strong>Manifestation:</strong> The genome will be distant from other genomes within its species, but highly similar to genomes of another species.</li>
<li><strong>Solution:</strong> Remove this genome. Assigning a new taxa is also possible.</li>
</ol>
</li>
<li><strong>Outlier genomes that are the result of poor assemblies</strong><ol>
<li><strong>Manifestation:</strong> The genome will be distant from all other genomes in the distance matrix.</li>
<li><strong>Solution:</strong> Remove this genome.</li>
</ol>
</li>
</ol>
<p>Any remaining species overlaps following removal of outlier genomes and separation of subspecies is likely the result of species groupings reflecting historical practices that do not align well with genetic distinctions between species (i.e. <em>E. coli</em> and <em>Shigella</em>).</p>
<h4 id="step-5-download-taxa">Step 5: Download taxa<a class="headerlink" href="#step-5-download-taxa" title="Permanent link">¶</a></h4>
<p><a href="https://www.ncbi.nlm.nih.gov/search/">Entrez</a>, available through <a href="https://biopython.org/docs/1.75/api/Bio.Entrez.html">Biopython</a>, is a useful resource for downloading data directly from NCBI including Taxonomy, a database of organism names and classifications.</p>
<p>For each entry in the database, the most up-to-date taxonomy needs to be retrieved. In the GAMBIT database, the taxon IDs of the genomes need to be mapped to their species-level ancestors.</p>
<p>We provide a Python script to accomplish this. Please add the missing information marked with <code>&lt;&gt;</code> to make the utility function and make sure you have all the required dependencies installed in your system, including <a href="https://pandas.pydata.org/">Pandas</a> and <a href="https://biopython.org/">Biopython</a>.</p>
<div class="language-python highlight"><table class="highlighttable"><tbody><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-5-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-5-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-5-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-5-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-5-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-5-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-5-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-5-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-5-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-5-10">10</a></span>
<span class="normal"><a href="#__codelineno-5-11">11</a></span>
<span class="normal"><a href="#__codelineno-5-12">12</a></span>
<span class="normal"><a href="#__codelineno-5-13">13</a></span>
<span class="normal"><a href="#__codelineno-5-14">14</a></span>
<span class="normal"><a href="#__codelineno-5-15">15</a></span>
<span class="normal"><a href="#__codelineno-5-16">16</a></span>
<span class="normal"><a href="#__codelineno-5-17">17</a></span>
<span class="normal"><a href="#__codelineno-5-18">18</a></span>
<span class="normal"><a href="#__codelineno-5-19">19</a></span>
<span class="normal"><a href="#__codelineno-5-20">20</a></span>
<span class="normal"><a href="#__codelineno-5-21">21</a></span>
<span class="normal"><a href="#__codelineno-5-22">22</a></span>
<span class="normal"><a href="#__codelineno-5-23">23</a></span>
<span class="normal"><a href="#__codelineno-5-24">24</a></span>
<span class="normal"><a href="#__codelineno-5-25">25</a></span>
<span class="normal"><a href="#__codelineno-5-26">26</a></span>
<span class="normal"><a href="#__codelineno-5-27">27</a></span>
<span class="normal"><a href="#__codelineno-5-28">28</a></span>
<span class="normal"><a href="#__codelineno-5-29">29</a></span>
<span class="normal"><a href="#__codelineno-5-30">30</a></span>
<span class="normal"><a href="#__codelineno-5-31">31</a></span>
<span class="normal"><a href="#__codelineno-5-32">32</a></span>
<span class="normal"><a href="#__codelineno-5-33">33</a></span>
<span class="normal"><a href="#__codelineno-5-34">34</a></span>
<span class="normal"><a href="#__codelineno-5-35">35</a></span>
<span class="normal"><a href="#__codelineno-5-36">36</a></span>
<span class="normal"><a href="#__codelineno-5-37">37</a></span>
<span class="normal"><a href="#__codelineno-5-38">38</a></span>
<span class="normal"><a href="#__codelineno-5-39">39</a></span>
<span class="normal"><a href="#__codelineno-5-40">40</a></span>
<span class="normal"><a href="#__codelineno-5-41">41</a></span>
<span class="normal"><a href="#__codelineno-5-42">42</a></span>
<span class="normal"><a href="#__codelineno-5-43">43</a></span>
<span class="normal"><a href="#__codelineno-5-44">44</a></span>
<span class="normal"><a href="#__codelineno-5-45">45</a></span>
<span class="normal"><a href="#__codelineno-5-46">46</a></span>
<span class="normal"><a href="#__codelineno-5-47">47</a></span>
<span class="normal"><a href="#__codelineno-5-48">48</a></span>
<span class="normal"><a href="#__codelineno-5-49">49</a></span>
<span class="normal"><a href="#__codelineno-5-50">50</a></span>
<span class="normal"><a href="#__codelineno-5-51">51</a></span>
<span class="normal"><a href="#__codelineno-5-52">52</a></span>
<span class="normal"><a href="#__codelineno-5-53">53</a></span>
<span class="normal"><a href="#__codelineno-5-54">54</a></span>
<span class="normal"><a href="#__codelineno-5-55">55</a></span>
<span class="normal"><a href="#__codelineno-5-56">56</a></span>
<span class="normal"><a href="#__codelineno-5-57">57</a></span>
<span class="normal"><a href="#__codelineno-5-58">58</a></span>
<span class="normal"><a href="#__codelineno-5-59">59</a></span>
<span class="normal"><a href="#__codelineno-5-60">60</a></span>
<span class="normal"><a href="#__codelineno-5-61">61</a></span>
<span class="normal"><a href="#__codelineno-5-62">62</a></span>
<span class="normal"><a href="#__codelineno-5-63">63</a></span>
<span class="normal"><a href="#__codelineno-5-64">64</a></span>
<span class="normal"><a href="#__codelineno-5-65">65</a></span>
<span class="normal"><a href="#__codelineno-5-66">66</a></span>
<span class="normal"><a href="#__codelineno-5-67">67</a></span>
<span class="normal"><a href="#__codelineno-5-68">68</a></span>
<span class="normal"><a href="#__codelineno-5-69">69</a></span>
<span class="normal"><a href="#__codelineno-5-70">70</a></span>
<span class="normal"><a href="#__codelineno-5-71">71</a></span>
<span class="normal"><a href="#__codelineno-5-72">72</a></span>
<span class="normal"><a href="#__codelineno-5-73">73</a></span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1"></a><span class="ch">#!/usr/bin/env python</span>
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2"></a><span class="c1"># coding: utf-8</span>
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3"></a>
</span><span id="__span-5-4"><a id="__codelineno-5-4" name="__codelineno-5-4"></a><span class="c1"># Download ESummary data for taxonomy tree.</span>
</span><span id="__span-5-5"><a id="__codelineno-5-5" name="__codelineno-5-5"></a>
</span><span id="__span-5-6"><a id="__codelineno-5-6" name="__codelineno-5-6"></a><span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>
</span><span id="__span-5-7"><a id="__codelineno-5-7" name="__codelineno-5-7"></a><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
</span><span id="__span-5-8"><a id="__codelineno-5-8" name="__codelineno-5-8"></a><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
</span><span id="__span-5-9"><a id="__codelineno-5-9" name="__codelineno-5-9"></a><span class="kn">from</span><span class="w"> </span><span class="nn">Bio</span><span class="w"> </span><span class="kn">import</span> <span class="n">Entrez</span>
</span><span id="__span-5-10"><a id="__codelineno-5-10" name="__codelineno-5-10"></a><span class="kn">from</span><span class="w"> </span><span class="nn">tqdm</span><span class="w"> </span><span class="kn">import</span> <span class="n">tqdm</span>
</span><span id="__span-5-11"><a id="__codelineno-5-11" name="__codelineno-5-11"></a>
</span><span id="__span-5-12"><a id="__codelineno-5-12" name="__codelineno-5-12"></a><span class="c1"># ## Setup</span>
</span><span id="__span-5-13"><a id="__codelineno-5-13" name="__codelineno-5-13"></a>
</span><span id="__span-5-14"><a id="__codelineno-5-14" name="__codelineno-5-14"></a><span class="n">Entrez</span><span class="o">.</span><span class="n">email</span> <span class="o">=</span> <span class="s1">'&lt;your_email_here&gt;'</span> <span class="c1">#TO BE REPLACED</span>
</span><span id="__span-5-15"><a id="__codelineno-5-15" name="__codelineno-5-15"></a>
</span><span id="__span-5-16"><a id="__codelineno-5-16" name="__codelineno-5-16"></a><span class="n">DATESTR</span> <span class="o">=</span> <span class="s1">'&lt;YYMMDD&gt;'</span> <span class="c1">#TO BE REPLACED</span>
</span><span id="__span-5-17"><a id="__codelineno-5-17" name="__codelineno-5-17"></a><span class="n">NBNAME</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">DATESTR</span><span class="si">}</span><span class="s1">-download-taxa'</span>
</span><span id="__span-5-18"><a id="__codelineno-5-18" name="__codelineno-5-18"></a>
</span><span id="__span-5-19"><a id="__codelineno-5-19" name="__codelineno-5-19"></a><span class="n">infiles</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span>
</span><span id="__span-5-20"><a id="__codelineno-5-20" name="__codelineno-5-20"></a>    <span class="n">genomes</span><span class="o">=</span><span class="s1">'&lt;genomes_CSV_file&gt;'</span><span class="p">,</span>  <span class="c1">#TO BE REPLACED</span>
</span><span id="__span-5-21"><a id="__codelineno-5-21" name="__codelineno-5-21"></a><span class="p">)</span>
</span><span id="__span-5-22"><a id="__codelineno-5-22" name="__codelineno-5-22"></a>
</span><span id="__span-5-23"><a id="__codelineno-5-23" name="__codelineno-5-23"></a><span class="n">intermediate_data</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">'data-intermediate'</span><span class="p">)</span> <span class="o">/</span> <span class="n">NBNAME</span>
</span><span id="__span-5-24"><a id="__codelineno-5-24" name="__codelineno-5-24"></a><span class="n">intermediate_data</span><span class="o">.</span><span class="n">mkdir</span><span class="p">(</span><span class="n">exist_ok</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</span><span id="__span-5-25"><a id="__codelineno-5-25" name="__codelineno-5-25"></a>
</span><span id="__span-5-26"><a id="__codelineno-5-26" name="__codelineno-5-26"></a><span class="n">outfiles</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span>
</span><span id="__span-5-27"><a id="__codelineno-5-27" name="__codelineno-5-27"></a>    <span class="n">taxa</span><span class="o">=</span><span class="n">intermediate_data</span> <span class="o">/</span> <span class="s1">'taxa'</span><span class="p">,</span>
</span><span id="__span-5-28"><a id="__codelineno-5-28" name="__codelineno-5-28"></a><span class="p">)</span>
</span><span id="__span-5-29"><a id="__codelineno-5-29" name="__codelineno-5-29"></a>
</span><span id="__span-5-30"><a id="__codelineno-5-30" name="__codelineno-5-30"></a><span class="n">outfiles</span><span class="p">[</span><span class="s1">'taxa'</span><span class="p">]</span><span class="o">.</span><span class="n">mkdir</span><span class="p">(</span><span class="n">exist_ok</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</span><span id="__span-5-31"><a id="__codelineno-5-31" name="__codelineno-5-31"></a>
</span><span id="__span-5-32"><a id="__codelineno-5-32" name="__codelineno-5-32"></a><span class="c1"># ## Load data</span>
</span><span id="__span-5-33"><a id="__codelineno-5-33" name="__codelineno-5-33"></a>
</span><span id="__span-5-34"><a id="__codelineno-5-34" name="__codelineno-5-34"></a><span class="n">genomes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">infiles</span><span class="p">[</span><span class="s1">'genomes'</span><span class="p">])</span>
</span><span id="__span-5-35"><a id="__codelineno-5-35" name="__codelineno-5-35"></a>
</span><span id="__span-5-36"><a id="__codelineno-5-36" name="__codelineno-5-36"></a><span class="c1"># ## Download taxonomy data</span>
</span><span id="__span-5-37"><a id="__codelineno-5-37" name="__codelineno-5-37"></a>
</span><span id="__span-5-38"><a id="__codelineno-5-38" name="__codelineno-5-38"></a><span class="n">to_download</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">genomes</span><span class="p">[</span><span class="s1">'ncbi_taxid'</span><span class="p">])</span>
</span><span id="__span-5-39"><a id="__codelineno-5-39" name="__codelineno-5-39"></a><span class="n">taxa</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
</span><span id="__span-5-40"><a id="__codelineno-5-40" name="__codelineno-5-40"></a>
</span><span id="__span-5-41"><a id="__codelineno-5-41" name="__codelineno-5-41"></a><span class="k">while</span> <span class="n">to_download</span><span class="p">:</span>
</span><span id="__span-5-42"><a id="__codelineno-5-42" name="__codelineno-5-42"></a>    <span class="c1"># Next to download</span>
</span><span id="__span-5-43"><a id="__codelineno-5-43" name="__codelineno-5-43"></a>    <span class="n">taxid</span> <span class="o">=</span> <span class="nb">next</span><span class="p">(</span><span class="nb">iter</span><span class="p">(</span><span class="n">to_download</span><span class="p">))</span>
</span><span id="__span-5-44"><a id="__codelineno-5-44" name="__codelineno-5-44"></a>    <span class="nb">print</span><span class="p">(</span><span class="n">taxid</span><span class="p">)</span>
</span><span id="__span-5-45"><a id="__codelineno-5-45" name="__codelineno-5-45"></a>    <span class="n">file</span> <span class="o">=</span> <span class="n">outfiles</span><span class="p">[</span><span class="s1">'taxa'</span><span class="p">]</span> <span class="o">/</span> <span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">taxid</span><span class="si">}</span><span class="s1">.json'</span>
</span><span id="__span-5-46"><a id="__codelineno-5-46" name="__codelineno-5-46"></a>
</span><span id="__span-5-47"><a id="__codelineno-5-47" name="__codelineno-5-47"></a>    <span class="k">if</span> <span class="ow">not</span> <span class="n">file</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
</span><span id="__span-5-48"><a id="__codelineno-5-48" name="__codelineno-5-48"></a>        <span class="c1"># Fetch data from NCBI</span>
</span><span id="__span-5-49"><a id="__codelineno-5-49" name="__codelineno-5-49"></a>        <span class="n">result</span> <span class="o">=</span> <span class="n">Entrez</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">Entrez</span><span class="o">.</span><span class="n">efetch</span><span class="p">(</span><span class="n">db</span><span class="o">=</span><span class="s1">'taxonomy'</span><span class="p">,</span> <span class="nb">id</span><span class="o">=</span><span class="n">taxid</span><span class="p">))</span>
</span><span id="__span-5-50"><a id="__codelineno-5-50" name="__codelineno-5-50"></a>        <span class="p">(</span><span class="n">taxon</span><span class="p">,)</span> <span class="o">=</span> <span class="n">result</span>
</span><span id="__span-5-51"><a id="__codelineno-5-51" name="__codelineno-5-51"></a>
</span><span id="__span-5-52"><a id="__codelineno-5-52" name="__codelineno-5-52"></a>        <span class="c1"># Save to local file</span>
</span><span id="__span-5-53"><a id="__codelineno-5-53" name="__codelineno-5-53"></a>        <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">file</span><span class="p">,</span> <span class="s1">'w'</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
</span><span id="__span-5-54"><a id="__codelineno-5-54" name="__codelineno-5-54"></a>            <span class="n">json</span><span class="o">.</span><span class="n">dump</span><span class="p">(</span><span class="n">taxon</span><span class="p">,</span> <span class="n">f</span><span class="p">)</span>
</span><span id="__span-5-55"><a id="__codelineno-5-55" name="__codelineno-5-55"></a>
</span><span id="__span-5-56"><a id="__codelineno-5-56" name="__codelineno-5-56"></a>    <span class="k">else</span><span class="p">:</span>
</span><span id="__span-5-57"><a id="__codelineno-5-57" name="__codelineno-5-57"></a>        <span class="c1"># Already downloaded, read from the existing file</span>
</span><span id="__span-5-58"><a id="__codelineno-5-58" name="__codelineno-5-58"></a>        <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">file</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
</span><span id="__span-5-59"><a id="__codelineno-5-59" name="__codelineno-5-59"></a>            <span class="n">taxon</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
</span><span id="__span-5-60"><a id="__codelineno-5-60" name="__codelineno-5-60"></a>
</span><span id="__span-5-61"><a id="__codelineno-5-61" name="__codelineno-5-61"></a>    <span class="n">taxa</span><span class="p">[</span><span class="n">taxid</span><span class="p">]</span> <span class="o">=</span> <span class="n">taxon</span>
</span><span id="__span-5-62"><a id="__codelineno-5-62" name="__codelineno-5-62"></a>    <span class="n">to_download</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="n">taxid</span><span class="p">)</span>
</span><span id="__span-5-63"><a id="__codelineno-5-63" name="__codelineno-5-63"></a>
</span><span id="__span-5-64"><a id="__codelineno-5-64" name="__codelineno-5-64"></a>    <span class="c1"># If not a species, find the species ancestor and add to the download list</span>
</span><span id="__span-5-65"><a id="__codelineno-5-65" name="__codelineno-5-65"></a>    <span class="k">if</span> <span class="n">taxon</span><span class="p">[</span><span class="s1">'Rank'</span><span class="p">]</span> <span class="o">!=</span> <span class="s1">'species'</span><span class="p">:</span>
</span><span id="__span-5-66"><a id="__codelineno-5-66" name="__codelineno-5-66"></a>        <span class="k">for</span> <span class="n">ancestor</span> <span class="ow">in</span> <span class="n">taxon</span><span class="p">[</span><span class="s1">'LineageEx'</span><span class="p">]:</span>
</span><span id="__span-5-67"><a id="__codelineno-5-67" name="__codelineno-5-67"></a>            <span class="k">if</span> <span class="n">ancestor</span><span class="p">[</span><span class="s1">'Rank'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'species'</span><span class="p">:</span>
</span><span id="__span-5-68"><a id="__codelineno-5-68" name="__codelineno-5-68"></a>                <span class="n">tid</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">ancestor</span><span class="p">[</span><span class="s1">'TaxId'</span><span class="p">])</span>
</span><span id="__span-5-69"><a id="__codelineno-5-69" name="__codelineno-5-69"></a>                <span class="k">if</span> <span class="n">tid</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">taxa</span><span class="p">:</span>
</span><span id="__span-5-70"><a id="__codelineno-5-70" name="__codelineno-5-70"></a>                    <span class="n">to_download</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">tid</span><span class="p">)</span>
</span><span id="__span-5-71"><a id="__codelineno-5-71" name="__codelineno-5-71"></a>                <span class="k">break</span>
</span><span id="__span-5-72"><a id="__codelineno-5-72" name="__codelineno-5-72"></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="__span-5-73"><a id="__codelineno-5-73" name="__codelineno-5-73"></a>            <span class="k">raise</span> <span class="ne">RuntimeError</span><span class="p">(</span><span class="s1">'No species ancestor found'</span><span class="p">)</span>
</span></code></pre></div></td></tr></tbody></table></div>
<h4 id="step-6-create-the-database">Step 6: Create the database<a class="headerlink" href="#step-6-create-the-database" title="Permanent link">¶</a></h4>
<p>The next step is to create the SQLite database file for the GAMBIT database. <a href="#figure1">Figure 1</a> depicts the database structure expected by GAMBIT. </p>
<p>We provide a Python script to accomplish this. Please add the missing information marked with <code>&lt;&gt;</code> to make the utility function and make sure you have all the required dependencies installed in your system, including <a href="../gambit/">GAMBIT</a> and <a href="https://www.sqlalchemy.org/">SQLalchemy</a>.</p>
<div class="language-python highlight"><table class="highlighttable"><tbody><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-6-1">  1</a></span>
<span class="normal"><a href="#__codelineno-6-2">  2</a></span>
<span class="normal"><a href="#__codelineno-6-3">  3</a></span>
<span class="normal"><a href="#__codelineno-6-4">  4</a></span>
<span class="normal"><a href="#__codelineno-6-5">  5</a></span>
<span class="normal"><a href="#__codelineno-6-6">  6</a></span>
<span class="normal"><a href="#__codelineno-6-7">  7</a></span>
<span class="normal"><a href="#__codelineno-6-8">  8</a></span>
<span class="normal"><a href="#__codelineno-6-9">  9</a></span>
<span class="normal"><a href="#__codelineno-6-10"> 10</a></span>
<span class="normal"><a href="#__codelineno-6-11"> 11</a></span>
<span class="normal"><a href="#__codelineno-6-12"> 12</a></span>
<span class="normal"><a href="#__codelineno-6-13"> 13</a></span>
<span class="normal"><a href="#__codelineno-6-14"> 14</a></span>
<span class="normal"><a href="#__codelineno-6-15"> 15</a></span>
<span class="normal"><a href="#__codelineno-6-16"> 16</a></span>
<span class="normal"><a href="#__codelineno-6-17"> 17</a></span>
<span class="normal"><a href="#__codelineno-6-18"> 18</a></span>
<span class="normal"><a href="#__codelineno-6-19"> 19</a></span>
<span class="normal"><a href="#__codelineno-6-20"> 20</a></span>
<span class="normal"><a href="#__codelineno-6-21"> 21</a></span>
<span class="normal"><a href="#__codelineno-6-22"> 22</a></span>
<span class="normal"><a href="#__codelineno-6-23"> 23</a></span>
<span class="normal"><a href="#__codelineno-6-24"> 24</a></span>
<span class="normal"><a href="#__codelineno-6-25"> 25</a></span>
<span class="normal"><a href="#__codelineno-6-26"> 26</a></span>
<span class="normal"><a href="#__codelineno-6-27"> 27</a></span>
<span class="normal"><a href="#__codelineno-6-28"> 28</a></span>
<span class="normal"><a href="#__codelineno-6-29"> 29</a></span>
<span class="normal"><a href="#__codelineno-6-30"> 30</a></span>
<span class="normal"><a href="#__codelineno-6-31"> 31</a></span>
<span class="normal"><a href="#__codelineno-6-32"> 32</a></span>
<span class="normal"><a href="#__codelineno-6-33"> 33</a></span>
<span class="normal"><a href="#__codelineno-6-34"> 34</a></span>
<span class="normal"><a href="#__codelineno-6-35"> 35</a></span>
<span class="normal"><a href="#__codelineno-6-36"> 36</a></span>
<span class="normal"><a href="#__codelineno-6-37"> 37</a></span>
<span class="normal"><a href="#__codelineno-6-38"> 38</a></span>
<span class="normal"><a href="#__codelineno-6-39"> 39</a></span>
<span class="normal"><a href="#__codelineno-6-40"> 40</a></span>
<span class="normal"><a href="#__codelineno-6-41"> 41</a></span>
<span class="normal"><a href="#__codelineno-6-42"> 42</a></span>
<span class="normal"><a href="#__codelineno-6-43"> 43</a></span>
<span class="normal"><a href="#__codelineno-6-44"> 44</a></span>
<span class="normal"><a href="#__codelineno-6-45"> 45</a></span>
<span class="normal"><a href="#__codelineno-6-46"> 46</a></span>
<span class="normal"><a href="#__codelineno-6-47"> 47</a></span>
<span class="normal"><a href="#__codelineno-6-48"> 48</a></span>
<span class="normal"><a href="#__codelineno-6-49"> 49</a></span>
<span class="normal"><a href="#__codelineno-6-50"> 50</a></span>
<span class="normal"><a href="#__codelineno-6-51"> 51</a></span>
<span class="normal"><a href="#__codelineno-6-52"> 52</a></span>
<span class="normal"><a href="#__codelineno-6-53"> 53</a></span>
<span class="normal"><a href="#__codelineno-6-54"> 54</a></span>
<span class="normal"><a href="#__codelineno-6-55"> 55</a></span>
<span class="normal"><a href="#__codelineno-6-56"> 56</a></span>
<span class="normal"><a href="#__codelineno-6-57"> 57</a></span>
<span class="normal"><a href="#__codelineno-6-58"> 58</a></span>
<span class="normal"><a href="#__codelineno-6-59"> 59</a></span>
<span class="normal"><a href="#__codelineno-6-60"> 60</a></span>
<span class="normal"><a href="#__codelineno-6-61"> 61</a></span>
<span class="normal"><a href="#__codelineno-6-62"> 62</a></span>
<span class="normal"><a href="#__codelineno-6-63"> 63</a></span>
<span class="normal"><a href="#__codelineno-6-64"> 64</a></span>
<span class="normal"><a href="#__codelineno-6-65"> 65</a></span>
<span class="normal"><a href="#__codelineno-6-66"> 66</a></span>
<span class="normal"><a href="#__codelineno-6-67"> 67</a></span>
<span class="normal"><a href="#__codelineno-6-68"> 68</a></span>
<span class="normal"><a href="#__codelineno-6-69"> 69</a></span>
<span class="normal"><a href="#__codelineno-6-70"> 70</a></span>
<span class="normal"><a href="#__codelineno-6-71"> 71</a></span>
<span class="normal"><a href="#__codelineno-6-72"> 72</a></span>
<span class="normal"><a href="#__codelineno-6-73"> 73</a></span>
<span class="normal"><a href="#__codelineno-6-74"> 74</a></span>
<span class="normal"><a href="#__codelineno-6-75"> 75</a></span>
<span class="normal"><a href="#__codelineno-6-76"> 76</a></span>
<span class="normal"><a href="#__codelineno-6-77"> 77</a></span>
<span class="normal"><a href="#__codelineno-6-78"> 78</a></span>
<span class="normal"><a href="#__codelineno-6-79"> 79</a></span>
<span class="normal"><a href="#__codelineno-6-80"> 80</a></span>
<span class="normal"><a href="#__codelineno-6-81"> 81</a></span>
<span class="normal"><a href="#__codelineno-6-82"> 82</a></span>
<span class="normal"><a href="#__codelineno-6-83"> 83</a></span>
<span class="normal"><a href="#__codelineno-6-84"> 84</a></span>
<span class="normal"><a href="#__codelineno-6-85"> 85</a></span>
<span class="normal"><a href="#__codelineno-6-86"> 86</a></span>
<span class="normal"><a href="#__codelineno-6-87"> 87</a></span>
<span class="normal"><a href="#__codelineno-6-88"> 88</a></span>
<span class="normal"><a href="#__codelineno-6-89"> 89</a></span>
<span class="normal"><a href="#__codelineno-6-90"> 90</a></span>
<span class="normal"><a href="#__codelineno-6-91"> 91</a></span>
<span class="normal"><a href="#__codelineno-6-92"> 92</a></span>
<span class="normal"><a href="#__codelineno-6-93"> 93</a></span>
<span class="normal"><a href="#__codelineno-6-94"> 94</a></span>
<span class="normal"><a href="#__codelineno-6-95"> 95</a></span>
<span class="normal"><a href="#__codelineno-6-96"> 96</a></span>
<span class="normal"><a href="#__codelineno-6-97"> 97</a></span>
<span class="normal"><a href="#__codelineno-6-98"> 98</a></span>
<span class="normal"><a href="#__codelineno-6-99"> 99</a></span>
<span class="normal"><a href="#__codelineno-6-100">100</a></span>
<span class="normal"><a href="#__codelineno-6-101">101</a></span>
<span class="normal"><a href="#__codelineno-6-102">102</a></span>
<span class="normal"><a href="#__codelineno-6-103">103</a></span>
<span class="normal"><a href="#__codelineno-6-104">104</a></span>
<span class="normal"><a href="#__codelineno-6-105">105</a></span>
<span class="normal"><a href="#__codelineno-6-106">106</a></span>
<span class="normal"><a href="#__codelineno-6-107">107</a></span>
<span class="normal"><a href="#__codelineno-6-108">108</a></span>
<span class="normal"><a href="#__codelineno-6-109">109</a></span>
<span class="normal"><a href="#__codelineno-6-110">110</a></span>
<span class="normal"><a href="#__codelineno-6-111">111</a></span>
<span class="normal"><a href="#__codelineno-6-112">112</a></span>
<span class="normal"><a href="#__codelineno-6-113">113</a></span>
<span class="normal"><a href="#__codelineno-6-114">114</a></span>
<span class="normal"><a href="#__codelineno-6-115">115</a></span>
<span class="normal"><a href="#__codelineno-6-116">116</a></span>
<span class="normal"><a href="#__codelineno-6-117">117</a></span>
<span class="normal"><a href="#__codelineno-6-118">118</a></span>
<span class="normal"><a href="#__codelineno-6-119">119</a></span>
<span class="normal"><a href="#__codelineno-6-120">120</a></span>
<span class="normal"><a href="#__codelineno-6-121">121</a></span>
<span class="normal"><a href="#__codelineno-6-122">122</a></span>
<span class="normal"><a href="#__codelineno-6-123">123</a></span>
<span class="normal"><a href="#__codelineno-6-124">124</a></span>
<span class="normal"><a href="#__codelineno-6-125">125</a></span>
<span class="normal"><a href="#__codelineno-6-126">126</a></span>
<span class="normal"><a href="#__codelineno-6-127">127</a></span>
<span class="normal"><a href="#__codelineno-6-128">128</a></span>
<span class="normal"><a href="#__codelineno-6-129">129</a></span>
<span class="normal"><a href="#__codelineno-6-130">130</a></span>
<span class="normal"><a href="#__codelineno-6-131">131</a></span>
<span class="normal"><a href="#__codelineno-6-132">132</a></span>
<span class="normal"><a href="#__codelineno-6-133">133</a></span>
<span class="normal"><a href="#__codelineno-6-134">134</a></span>
<span class="normal"><a href="#__codelineno-6-135">135</a></span>
<span class="normal"><a href="#__codelineno-6-136">136</a></span>
<span class="normal"><a href="#__codelineno-6-137">137</a></span>
<span class="normal"><a href="#__codelineno-6-138">138</a></span>
<span class="normal"><a href="#__codelineno-6-139">139</a></span>
<span class="normal"><a href="#__codelineno-6-140">140</a></span>
<span class="normal"><a href="#__codelineno-6-141">141</a></span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1"></a><span class="ch">#!/usr/bin/env python</span>
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2"></a><span class="c1"># coding: utf-8</span>
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3"></a>
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4"></a><span class="c1"># This creates the sqlite database file for the Candida DB.</span>
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5"></a>
</span><span id="__span-6-6"><a id="__codelineno-6-6" name="__codelineno-6-6"></a><span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>
</span><span id="__span-6-7"><a id="__codelineno-6-7" name="__codelineno-6-7"></a><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
</span><span id="__span-6-8"><a id="__codelineno-6-8" name="__codelineno-6-8"></a><span class="kn">from</span><span class="w"> </span><span class="nn">datetime</span><span class="w"> </span><span class="kn">import</span> <span class="n">date</span>
</span><span id="__span-6-9"><a id="__codelineno-6-9" name="__codelineno-6-9"></a><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
</span><span id="__span-6-10"><a id="__codelineno-6-10" name="__codelineno-6-10"></a><span class="kn">from</span><span class="w"> </span><span class="nn">sqlalchemy</span><span class="w"> </span><span class="kn">import</span> <span class="n">create_engine</span>
</span><span id="__span-6-11"><a id="__codelineno-6-11" name="__codelineno-6-11"></a><span class="kn">from</span><span class="w"> </span><span class="nn">sqlalchemy.orm</span><span class="w"> </span><span class="kn">import</span> <span class="n">sessionmaker</span>
</span><span id="__span-6-12"><a id="__codelineno-6-12" name="__codelineno-6-12"></a><span class="kn">from</span><span class="w"> </span><span class="nn">gambit.db</span><span class="w"> </span><span class="kn">import</span> <span class="n">ReferenceGenomeSet</span><span class="p">,</span> <span class="n">Genome</span><span class="p">,</span> <span class="n">AnnotatedGenome</span><span class="p">,</span> <span class="n">Taxon</span>
</span><span id="__span-6-13"><a id="__codelineno-6-13" name="__codelineno-6-13"></a><span class="kn">from</span><span class="w"> </span><span class="nn">gambit.db.migrate</span><span class="w"> </span><span class="kn">import</span> <span class="n">init_db</span>
</span><span id="__span-6-14"><a id="__codelineno-6-14" name="__codelineno-6-14"></a><span class="kn">from</span><span class="w"> </span><span class="nn">entrez_tools.db.assembly</span><span class="w"> </span><span class="kn">import</span> <span class="n">format_summary_meta</span>
</span><span id="__span-6-15"><a id="__codelineno-6-15" name="__codelineno-6-15"></a>
</span><span id="__span-6-16"><a id="__codelineno-6-16" name="__codelineno-6-16"></a><span class="c1"># ## Setup</span>
</span><span id="__span-6-17"><a id="__codelineno-6-17" name="__codelineno-6-17"></a>
</span><span id="__span-6-18"><a id="__codelineno-6-18" name="__codelineno-6-18"></a><span class="n">DATESTR</span> <span class="o">=</span> <span class="s1">'&lt;YYMMDD&gt;'</span> <span class="c1">#TO BE REPLACED</span>
</span><span id="__span-6-19"><a id="__codelineno-6-19" name="__codelineno-6-19"></a><span class="n">NBNAME</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">DATESTR</span><span class="si">}</span><span class="s1">-create-db'</span>
</span><span id="__span-6-20"><a id="__codelineno-6-20" name="__codelineno-6-20"></a>
</span><span id="__span-6-21"><a id="__codelineno-6-21" name="__codelineno-6-21"></a><span class="n">infiles</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span>
</span><span id="__span-6-22"><a id="__codelineno-6-22" name="__codelineno-6-22"></a>    <span class="n">genomes_table</span><span class="o">=</span><span class="n">Path</span><span class="p">(</span><span class="s1">'&lt;genomes_CSV_file&gt;'</span><span class="p">),</span> <span class="c1">#TO BE REPLACED</span>
</span><span id="__span-6-23"><a id="__codelineno-6-23" name="__codelineno-6-23"></a>    <span class="n">taxa_table</span><span class="o">=</span><span class="n">Path</span><span class="p">(</span><span class="sa">f</span><span class="s1">'data-intermediate/</span><span class="si">{</span><span class="n">DATESTR</span><span class="si">}</span><span class="s1">_3-find-diameters/species-data.csv'</span><span class="p">),</span>
</span><span id="__span-6-24"><a id="__codelineno-6-24" name="__codelineno-6-24"></a>    <span class="n">fasta_dir</span><span class="o">=</span><span class="n">Path</span><span class="p">(</span><span class="s1">'&lt;path_to_directory_with_fastas&gt;'</span><span class="p">),</span> <span class="c1">#TO BE REPLACED</span>
</span><span id="__span-6-25"><a id="__codelineno-6-25" name="__codelineno-6-25"></a>    <span class="n">genome_esummary_dir</span><span class="o">=</span><span class="n">Path</span><span class="p">(</span><span class="sa">f</span><span class="s1">'data-intermediate/</span><span class="si">{</span><span class="n">DATESTR</span><span class="si">}</span><span class="s1">_1-download-genomes/esummary'</span><span class="p">),</span>
</span><span id="__span-6-26"><a id="__codelineno-6-26" name="__codelineno-6-26"></a>    <span class="n">taxon_esummary_dir</span><span class="o">=</span><span class="n">Path</span><span class="p">(</span><span class="sa">f</span><span class="s1">'data-intermediate/</span><span class="si">{</span><span class="n">DATESTR</span><span class="si">}</span><span class="s1">_4-download-taxa/taxa'</span><span class="p">),</span>
</span><span id="__span-6-27"><a id="__codelineno-6-27" name="__codelineno-6-27"></a><span class="p">)</span>
</span><span id="__span-6-28"><a id="__codelineno-6-28" name="__codelineno-6-28"></a>
</span><span id="__span-6-29"><a id="__codelineno-6-29" name="__codelineno-6-29"></a><span class="n">processed_data</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">'data-processed'</span><span class="p">)</span> <span class="o">/</span> <span class="n">NBNAME</span>
</span><span id="__span-6-30"><a id="__codelineno-6-30" name="__codelineno-6-30"></a><span class="n">processed_data</span><span class="o">.</span><span class="n">mkdir</span><span class="p">(</span><span class="n">exist_ok</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">parents</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</span><span id="__span-6-31"><a id="__codelineno-6-31" name="__codelineno-6-31"></a>
</span><span id="__span-6-32"><a id="__codelineno-6-32" name="__codelineno-6-32"></a><span class="n">outfiles</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span>
</span><span id="__span-6-33"><a id="__codelineno-6-33" name="__codelineno-6-33"></a>    <span class="n">db</span><span class="o">=</span><span class="n">processed_data</span> <span class="o">/</span> <span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">DATESTR</span><span class="si">}</span><span class="s1">-&lt;database_name&gt;.db'</span><span class="p">,</span> <span class="c1">#TO BE REPLACED</span>
</span><span id="__span-6-34"><a id="__codelineno-6-34" name="__codelineno-6-34"></a><span class="p">)</span>
</span><span id="__span-6-35"><a id="__codelineno-6-35" name="__codelineno-6-35"></a>
</span><span id="__span-6-36"><a id="__codelineno-6-36" name="__codelineno-6-36"></a><span class="c1"># ## Load data</span>
</span><span id="__span-6-37"><a id="__codelineno-6-37" name="__codelineno-6-37"></a>
</span><span id="__span-6-38"><a id="__codelineno-6-38" name="__codelineno-6-38"></a><span class="n">genomes_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">infiles</span><span class="p">[</span><span class="s1">'genomes_table'</span><span class="p">])</span>
</span><span id="__span-6-39"><a id="__codelineno-6-39" name="__codelineno-6-39"></a><span class="n">taxa_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">infiles</span><span class="p">[</span><span class="s1">'taxa_table'</span><span class="p">])</span>
</span><span id="__span-6-40"><a id="__codelineno-6-40" name="__codelineno-6-40"></a>
</span><span id="__span-6-41"><a id="__codelineno-6-41" name="__codelineno-6-41"></a><span class="n">genome_summaries</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
</span><span id="__span-6-42"><a id="__codelineno-6-42" name="__codelineno-6-42"></a>
</span><span id="__span-6-43"><a id="__codelineno-6-43" name="__codelineno-6-43"></a><span class="k">for</span> <span class="n">uid</span> <span class="ow">in</span> <span class="n">genomes_df</span><span class="p">[</span><span class="s1">'uid'</span><span class="p">]:</span>
</span><span id="__span-6-44"><a id="__codelineno-6-44" name="__codelineno-6-44"></a>    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">infiles</span><span class="p">[</span><span class="s1">'genome_esummary_dir'</span><span class="p">]</span> <span class="o">/</span> <span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">uid</span><span class="si">}</span><span class="s1">.json'</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
</span><span id="__span-6-45"><a id="__codelineno-6-45" name="__codelineno-6-45"></a>        <span class="n">genome_summaries</span><span class="p">[</span><span class="n">uid</span><span class="p">]</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
</span><span id="__span-6-46"><a id="__codelineno-6-46" name="__codelineno-6-46"></a>
</span><span id="__span-6-47"><a id="__codelineno-6-47" name="__codelineno-6-47"></a><span class="n">taxon_summaries</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
</span><span id="__span-6-48"><a id="__codelineno-6-48" name="__codelineno-6-48"></a>
</span><span id="__span-6-49"><a id="__codelineno-6-49" name="__codelineno-6-49"></a><span class="k">for</span> <span class="n">taxid</span> <span class="ow">in</span> <span class="nb">set</span><span class="p">(</span><span class="n">genomes_df</span><span class="p">[</span><span class="s1">'ncbi_taxid'</span><span class="p">]):</span>
</span><span id="__span-6-50"><a id="__codelineno-6-50" name="__codelineno-6-50"></a>    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">infiles</span><span class="p">[</span><span class="s1">'taxon_esummary_dir'</span><span class="p">]</span> <span class="o">/</span> <span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">taxid</span><span class="si">}</span><span class="s1">.json'</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
</span><span id="__span-6-51"><a id="__codelineno-6-51" name="__codelineno-6-51"></a>        <span class="n">taxon_summaries</span><span class="p">[</span><span class="n">uid</span><span class="p">]</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
</span><span id="__span-6-52"><a id="__codelineno-6-52" name="__codelineno-6-52"></a>
</span><span id="__span-6-53"><a id="__codelineno-6-53" name="__codelineno-6-53"></a><span class="c1"># ## Create database</span>
</span><span id="__span-6-54"><a id="__codelineno-6-54" name="__codelineno-6-54"></a>
</span><span id="__span-6-55"><a id="__codelineno-6-55" name="__codelineno-6-55"></a><span class="k">if</span> <span class="n">outfiles</span><span class="p">[</span><span class="s1">'db'</span><span class="p">]</span><span class="o">.</span><span class="n">is_file</span><span class="p">():</span>
</span><span id="__span-6-56"><a id="__codelineno-6-56" name="__codelineno-6-56"></a>    <span class="n">outfiles</span><span class="p">[</span><span class="s1">'db'</span><span class="p">]</span><span class="o">.</span><span class="n">unlink</span><span class="p">()</span>
</span><span id="__span-6-57"><a id="__codelineno-6-57" name="__codelineno-6-57"></a>
</span><span id="__span-6-58"><a id="__codelineno-6-58" name="__codelineno-6-58"></a><span class="n">engine</span> <span class="o">=</span> <span class="n">create_engine</span><span class="p">(</span><span class="s1">'sqlite:///'</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">outfiles</span><span class="p">[</span><span class="s1">'db'</span><span class="p">]))</span>
</span><span id="__span-6-59"><a id="__codelineno-6-59" name="__codelineno-6-59"></a>
</span><span id="__span-6-60"><a id="__codelineno-6-60" name="__codelineno-6-60"></a><span class="n">init_db</span><span class="p">(</span><span class="n">engine</span><span class="p">)</span>
</span><span id="__span-6-61"><a id="__codelineno-6-61" name="__codelineno-6-61"></a>
</span><span id="__span-6-62"><a id="__codelineno-6-62" name="__codelineno-6-62"></a><span class="n">Session</span> <span class="o">=</span> <span class="n">sessionmaker</span><span class="p">(</span><span class="n">engine</span><span class="p">)</span>
</span><span id="__span-6-63"><a id="__codelineno-6-63" name="__codelineno-6-63"></a><span class="n">session</span> <span class="o">=</span> <span class="n">Session</span><span class="p">()</span>
</span><span id="__span-6-64"><a id="__codelineno-6-64" name="__codelineno-6-64"></a>
</span><span id="__span-6-65"><a id="__codelineno-6-65" name="__codelineno-6-65"></a><span class="c1"># ### Genome set</span>
</span><span id="__span-6-66"><a id="__codelineno-6-66" name="__codelineno-6-66"></a>
</span><span id="__span-6-67"><a id="__codelineno-6-67" name="__codelineno-6-67"></a><span class="n">gset</span> <span class="o">=</span> <span class="n">ReferenceGenomeSet</span><span class="p">(</span>
</span><span id="__span-6-68"><a id="__codelineno-6-68" name="__codelineno-6-68"></a>    <span class="n">key</span><span class="o">=</span><span class="s1">'&lt;database_name&gt;'</span><span class="p">,</span> <span class="c1"># TO BE REPLACED, suggested &lt;organization/name&gt;</span>
</span><span id="__span-6-69"><a id="__codelineno-6-69" name="__codelineno-6-69"></a>    <span class="n">version</span><span class="o">=</span><span class="s1">'x.y.z'</span><span class="p">,</span> <span class="c1"># TO BE REPLACED</span>
</span><span id="__span-6-70"><a id="__codelineno-6-70" name="__codelineno-6-70"></a>    <span class="n">name</span><span class="o">=</span><span class="s1">'&lt;name&gt;'</span><span class="p">,</span> <span class="c1"># TO BE REPLACED</span>
</span><span id="__span-6-71"><a id="__codelineno-6-71" name="__codelineno-6-71"></a>    <span class="n">description</span><span class="o">=</span><span class="s1">'&lt;description&gt;'</span><span class="p">,</span> <span class="c1"># TO BE REPLACED</span>
</span><span id="__span-6-72"><a id="__codelineno-6-72" name="__codelineno-6-72"></a>    <span class="n">extra</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span>
</span><span id="__span-6-73"><a id="__codelineno-6-73" name="__codelineno-6-73"></a>        <span class="n">author</span><span class="o">=</span><span class="s1">'&lt;author&gt;'</span><span class="p">,</span> <span class="c1"># TO BE REPLACED</span>
</span><span id="__span-6-74"><a id="__codelineno-6-74" name="__codelineno-6-74"></a>        <span class="n">date</span><span class="o">=</span><span class="n">date</span><span class="o">.</span><span class="n">today</span><span class="p">()</span><span class="o">.</span><span class="n">isoformat</span><span class="p">(),</span>
</span><span id="__span-6-75"><a id="__codelineno-6-75" name="__codelineno-6-75"></a>    <span class="p">),</span>
</span><span id="__span-6-76"><a id="__codelineno-6-76" name="__codelineno-6-76"></a><span class="p">)</span>
</span><span id="__span-6-77"><a id="__codelineno-6-77" name="__codelineno-6-77"></a>
</span><span id="__span-6-78"><a id="__codelineno-6-78" name="__codelineno-6-78"></a><span class="n">session</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">gset</span><span class="p">)</span>
</span><span id="__span-6-79"><a id="__codelineno-6-79" name="__codelineno-6-79"></a><span class="n">session</span><span class="o">.</span><span class="n">commit</span><span class="p">()</span>
</span><span id="__span-6-80"><a id="__codelineno-6-80" name="__codelineno-6-80"></a>
</span><span id="__span-6-81"><a id="__codelineno-6-81" name="__codelineno-6-81"></a><span class="c1"># ### Taxa</span>
</span><span id="__span-6-82"><a id="__codelineno-6-82" name="__codelineno-6-82"></a>
</span><span id="__span-6-83"><a id="__codelineno-6-83" name="__codelineno-6-83"></a><span class="n">taxa</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
</span><span id="__span-6-84"><a id="__codelineno-6-84" name="__codelineno-6-84"></a>
</span><span id="__span-6-85"><a id="__codelineno-6-85" name="__codelineno-6-85"></a><span class="k">for</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">taxa_df</span><span class="o">.</span><span class="n">itertuples</span><span class="p">():</span>
</span><span id="__span-6-86"><a id="__codelineno-6-86" name="__codelineno-6-86"></a>    <span class="n">taxon</span> <span class="o">=</span> <span class="n">Taxon</span><span class="p">(</span>
</span><span id="__span-6-87"><a id="__codelineno-6-87" name="__codelineno-6-87"></a>        <span class="n">genome_set</span><span class="o">=</span><span class="n">gset</span><span class="p">,</span>
</span><span id="__span-6-88"><a id="__codelineno-6-88" name="__codelineno-6-88"></a>        <span class="c1"># Arbitrary unique string ID</span>
</span><span id="__span-6-89"><a id="__codelineno-6-89" name="__codelineno-6-89"></a>        <span class="c1"># These are all regular NCBI taxa</span>
</span><span id="__span-6-90"><a id="__codelineno-6-90" name="__codelineno-6-90"></a>        <span class="n">key</span><span class="o">=</span><span class="sa">f</span><span class="s1">'ncbi/</span><span class="si">{</span><span class="n">row</span><span class="o">.</span><span class="n">ncbi_taxid</span><span class="si">}</span><span class="s1">'</span><span class="p">,</span>
</span><span id="__span-6-91"><a id="__codelineno-6-91" name="__codelineno-6-91"></a>        <span class="c1"># Using names I got from you here, NCBI reassigned several to other genera</span>
</span><span id="__span-6-92"><a id="__codelineno-6-92" name="__codelineno-6-92"></a>        <span class="n">name</span><span class="o">=</span><span class="n">row</span><span class="o">.</span><span class="n">name</span><span class="p">,</span>
</span><span id="__span-6-93"><a id="__codelineno-6-93" name="__codelineno-6-93"></a>        <span class="n">rank</span><span class="o">=</span><span class="s1">'species'</span><span class="p">,</span>
</span><span id="__span-6-94"><a id="__codelineno-6-94" name="__codelineno-6-94"></a>        <span class="c1">#description</span>
</span><span id="__span-6-95"><a id="__codelineno-6-95" name="__codelineno-6-95"></a>        <span class="n">distance_threshold</span><span class="o">=</span><span class="n">row</span><span class="o">.</span><span class="n">diameter</span><span class="p">,</span>
</span><span id="__span-6-96"><a id="__codelineno-6-96" name="__codelineno-6-96"></a>        <span class="c1">#report</span>
</span><span id="__span-6-97"><a id="__codelineno-6-97" name="__codelineno-6-97"></a>        <span class="c1">#genome set id</span>
</span><span id="__span-6-98"><a id="__codelineno-6-98" name="__codelineno-6-98"></a>        <span class="c1">#parent_id</span>
</span><span id="__span-6-99"><a id="__codelineno-6-99" name="__codelineno-6-99"></a>        <span class="n">ncbi_id</span><span class="o">=</span><span class="n">row</span><span class="o">.</span><span class="n">ncbi_taxid</span><span class="p">,</span>
</span><span id="__span-6-100"><a id="__codelineno-6-100" name="__codelineno-6-100"></a>        <span class="c1"># Arbitrary JSON-formatted metadata</span>
</span><span id="__span-6-101"><a id="__codelineno-6-101" name="__codelineno-6-101"></a>        <span class="n">extra</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span>
</span><span id="__span-6-102"><a id="__codelineno-6-102" name="__codelineno-6-102"></a>            <span class="c1"># The actual name used by NCBI</span>
</span><span id="__span-6-103"><a id="__codelineno-6-103" name="__codelineno-6-103"></a>            <span class="n">ncbi_name</span><span class="o">=</span><span class="n">row</span><span class="o">.</span><span class="n">ncbi_name</span><span class="p">,</span>
</span><span id="__span-6-104"><a id="__codelineno-6-104" name="__codelineno-6-104"></a>        <span class="p">),</span>
</span><span id="__span-6-105"><a id="__codelineno-6-105" name="__codelineno-6-105"></a>    <span class="p">)</span>
</span><span id="__span-6-106"><a id="__codelineno-6-106" name="__codelineno-6-106"></a>
</span><span id="__span-6-107"><a id="__codelineno-6-107" name="__codelineno-6-107"></a>    <span class="n">taxa</span><span class="p">[</span><span class="n">row</span><span class="o">.</span><span class="n">ncbi_taxid</span><span class="p">]</span> <span class="o">=</span> <span class="n">taxon</span>
</span><span id="__span-6-108"><a id="__codelineno-6-108" name="__codelineno-6-108"></a>    <span class="n">session</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">taxon</span><span class="p">)</span>
</span><span id="__span-6-109"><a id="__codelineno-6-109" name="__codelineno-6-109"></a>
</span><span id="__span-6-110"><a id="__codelineno-6-110" name="__codelineno-6-110"></a><span class="n">session</span><span class="o">.</span><span class="n">commit</span><span class="p">()</span>
</span><span id="__span-6-111"><a id="__codelineno-6-111" name="__codelineno-6-111"></a>
</span><span id="__span-6-112"><a id="__codelineno-6-112" name="__codelineno-6-112"></a><span class="c1"># ### Genomes</span>
</span><span id="__span-6-113"><a id="__codelineno-6-113" name="__codelineno-6-113"></a>
</span><span id="__span-6-114"><a id="__codelineno-6-114" name="__codelineno-6-114"></a><span class="k">for</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">genomes_df</span><span class="o">.</span><span class="n">itertuples</span><span class="p">():</span>
</span><span id="__span-6-115"><a id="__codelineno-6-115" name="__codelineno-6-115"></a>    <span class="n">summary</span> <span class="o">=</span> <span class="n">genome_summaries</span><span class="p">[</span><span class="n">row</span><span class="o">.</span><span class="n">uid</span><span class="p">]</span>
</span><span id="__span-6-116"><a id="__codelineno-6-116" name="__codelineno-6-116"></a>    <span class="n">assembly_stats</span> <span class="o">=</span> <span class="n">format_summary_meta</span><span class="p">(</span><span class="n">summary</span><span class="p">[</span><span class="s1">'meta'</span><span class="p">])</span>
</span><span id="__span-6-117"><a id="__codelineno-6-117" name="__codelineno-6-117"></a>    <span class="n">taxon</span> <span class="o">=</span> <span class="n">taxa</span><span class="p">[</span><span class="n">row</span><span class="o">.</span><span class="n">ncbi_taxid</span><span class="p">]</span>
</span><span id="__span-6-118"><a id="__codelineno-6-118" name="__codelineno-6-118"></a>
</span><span id="__span-6-119"><a id="__codelineno-6-119" name="__codelineno-6-119"></a>    <span class="n">genome</span> <span class="o">=</span> <span class="n">Genome</span><span class="p">(</span>
</span><span id="__span-6-120"><a id="__codelineno-6-120" name="__codelineno-6-120"></a>        <span class="c1">#edited next line</span>
</span><span id="__span-6-121"><a id="__codelineno-6-121" name="__codelineno-6-121"></a>        <span class="n">key</span><span class="o">=</span><span class="s1">'/data/fasta/'</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">row</span><span class="o">.</span><span class="n">genbank_acc</span><span class="si">}</span><span class="s1">''.fna.gz'</span><span class="p">,</span>
</span><span id="__span-6-122"><a id="__codelineno-6-122" name="__codelineno-6-122"></a>        <span class="n">description</span><span class="o">=</span><span class="sa">f</span><span class="s1">'[</span><span class="si">{</span><span class="n">row</span><span class="o">.</span><span class="n">genbank_acc</span><span class="si">}</span><span class="s1">] </span><span class="si">{</span><span class="n">summary</span><span class="p">[</span><span class="s2">"organism"</span><span class="p">]</span><span class="si">}</span><span class="s1">'</span><span class="p">,</span>
</span><span id="__span-6-123"><a id="__codelineno-6-123" name="__codelineno-6-123"></a>        <span class="n">ncbi_db</span><span class="o">=</span><span class="s1">'assembly'</span><span class="p">,</span>
</span><span id="__span-6-124"><a id="__codelineno-6-124" name="__codelineno-6-124"></a>        <span class="n">ncbi_id</span><span class="o">=</span><span class="n">row</span><span class="o">.</span><span class="n">uid</span><span class="p">,</span>
</span><span id="__span-6-125"><a id="__codelineno-6-125" name="__codelineno-6-125"></a>        <span class="n">genbank_acc</span><span class="o">=</span><span class="n">row</span><span class="o">.</span><span class="n">genbank_acc</span><span class="p">,</span>
</span><span id="__span-6-126"><a id="__codelineno-6-126" name="__codelineno-6-126"></a>        <span class="c1"># More JSON metadata</span>
</span><span id="__span-6-127"><a id="__codelineno-6-127" name="__codelineno-6-127"></a>        <span class="n">extra</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span>
</span><span id="__span-6-128"><a id="__codelineno-6-128" name="__codelineno-6-128"></a>            <span class="n">length</span><span class="o">=</span><span class="n">assembly_stats</span><span class="p">[</span><span class="s1">'total_length'</span><span class="p">],</span>
</span><span id="__span-6-129"><a id="__codelineno-6-129" name="__codelineno-6-129"></a>        <span class="p">)</span>
</span><span id="__span-6-130"><a id="__codelineno-6-130" name="__codelineno-6-130"></a>    <span class="p">)</span>
</span><span id="__span-6-131"><a id="__codelineno-6-131" name="__codelineno-6-131"></a>    <span class="n">session</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">genome</span><span class="p">)</span>
</span><span id="__span-6-132"><a id="__codelineno-6-132" name="__codelineno-6-132"></a>
</span><span id="__span-6-133"><a id="__codelineno-6-133" name="__codelineno-6-133"></a>    <span class="n">ag</span> <span class="o">=</span> <span class="n">AnnotatedGenome</span><span class="p">(</span>
</span><span id="__span-6-134"><a id="__codelineno-6-134" name="__codelineno-6-134"></a>        <span class="n">genome</span><span class="o">=</span><span class="n">genome</span><span class="p">,</span>
</span><span id="__span-6-135"><a id="__codelineno-6-135" name="__codelineno-6-135"></a>        <span class="n">genome_set</span><span class="o">=</span><span class="n">gset</span><span class="p">,</span>
</span><span id="__span-6-136"><a id="__codelineno-6-136" name="__codelineno-6-136"></a>        <span class="n">taxon</span><span class="o">=</span><span class="n">taxon</span><span class="p">,</span>
</span><span id="__span-6-137"><a id="__codelineno-6-137" name="__codelineno-6-137"></a>        <span class="n">organism</span><span class="o">=</span><span class="n">taxon</span><span class="o">.</span><span class="n">name</span><span class="p">,</span>
</span><span id="__span-6-138"><a id="__codelineno-6-138" name="__codelineno-6-138"></a>    <span class="p">)</span>
</span><span id="__span-6-139"><a id="__codelineno-6-139" name="__codelineno-6-139"></a>    <span class="n">session</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">ag</span><span class="p">)</span>
</span><span id="__span-6-140"><a id="__codelineno-6-140" name="__codelineno-6-140"></a>
</span><span id="__span-6-141"><a id="__codelineno-6-141" name="__codelineno-6-141"></a><span class="n">session</span><span class="o">.</span><span class="n">commit</span><span class="p">()</span>
</span></code></pre></div></td></tr></tbody></table></div>
<h4 id="step-7-finalise-the-signatures-file">Step 7: Finalise the signatures file<a class="headerlink" href="#step-7-finalise-the-signatures-file" title="Permanent link">¶</a></h4>
<p>The Last step is to finalise the signatures file. This last Python utility adds metadata to the database signatures file. Please add the missing information marked with <code>&lt;&gt;</code> to make the utility function and make sure you have all the required dependencies installed in your system, including <a href="../gambit/">GAMBIT</a> and <a href="https://www.sqlalchemy.org/">SQLalchemy</a>.</p>
<div class="language-python highlight"><table class="highlighttable"><tbody><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-7-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-7-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-7-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-7-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-7-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-7-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-7-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-7-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-7-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-7-10">10</a></span>
<span class="normal"><a href="#__codelineno-7-11">11</a></span>
<span class="normal"><a href="#__codelineno-7-12">12</a></span>
<span class="normal"><a href="#__codelineno-7-13">13</a></span>
<span class="normal"><a href="#__codelineno-7-14">14</a></span>
<span class="normal"><a href="#__codelineno-7-15">15</a></span>
<span class="normal"><a href="#__codelineno-7-16">16</a></span>
<span class="normal"><a href="#__codelineno-7-17">17</a></span>
<span class="normal"><a href="#__codelineno-7-18">18</a></span>
<span class="normal"><a href="#__codelineno-7-19">19</a></span>
<span class="normal"><a href="#__codelineno-7-20">20</a></span>
<span class="normal"><a href="#__codelineno-7-21">21</a></span>
<span class="normal"><a href="#__codelineno-7-22">22</a></span>
<span class="normal"><a href="#__codelineno-7-23">23</a></span>
<span class="normal"><a href="#__codelineno-7-24">24</a></span>
<span class="normal"><a href="#__codelineno-7-25">25</a></span>
<span class="normal"><a href="#__codelineno-7-26">26</a></span>
<span class="normal"><a href="#__codelineno-7-27">27</a></span>
<span class="normal"><a href="#__codelineno-7-28">28</a></span>
<span class="normal"><a href="#__codelineno-7-29">29</a></span>
<span class="normal"><a href="#__codelineno-7-30">30</a></span>
<span class="normal"><a href="#__codelineno-7-31">31</a></span>
<span class="normal"><a href="#__codelineno-7-32">32</a></span>
<span class="normal"><a href="#__codelineno-7-33">33</a></span>
<span class="normal"><a href="#__codelineno-7-34">34</a></span>
<span class="normal"><a href="#__codelineno-7-35">35</a></span>
<span class="normal"><a href="#__codelineno-7-36">36</a></span>
<span class="normal"><a href="#__codelineno-7-37">37</a></span>
<span class="normal"><a href="#__codelineno-7-38">38</a></span>
<span class="normal"><a href="#__codelineno-7-39">39</a></span></pre></div></td><td class="code"><div><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1"></a><span class="ch">#!/usr/bin/env python</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2"></a><span class="c1"># coding: utf-8</span>
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3"></a>
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4"></a><span class="c1"># # Finalize signatures</span>
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5"></a>
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6"></a><span class="kn">from</span><span class="w"> </span><span class="nn">pathlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">Path</span>
</span><span id="__span-7-7"><a id="__codelineno-7-7" name="__codelineno-7-7"></a><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
</span><span id="__span-7-8"><a id="__codelineno-7-8" name="__codelineno-7-8"></a><span class="kn">from</span><span class="w"> </span><span class="nn">datetime</span><span class="w"> </span><span class="kn">import</span> <span class="n">date</span>
</span><span id="__span-7-9"><a id="__codelineno-7-9" name="__codelineno-7-9"></a><span class="kn">from</span><span class="w"> </span><span class="nn">gambit.sigs</span><span class="w"> </span><span class="kn">import</span> <span class="n">SignaturesMeta</span><span class="p">,</span> <span class="n">load_signatures</span><span class="p">,</span> <span class="n">dump_signatures</span><span class="p">,</span> <span class="n">AnnotatedSignatures</span>
</span><span id="__span-7-10"><a id="__codelineno-7-10" name="__codelineno-7-10"></a>
</span><span id="__span-7-11"><a id="__codelineno-7-11" name="__codelineno-7-11"></a><span class="c1"># ## Setup</span>
</span><span id="__span-7-12"><a id="__codelineno-7-12" name="__codelineno-7-12"></a>
</span><span id="__span-7-13"><a id="__codelineno-7-13" name="__codelineno-7-13"></a><span class="n">DATESTR</span> <span class="o">=</span> <span class="s1">'&lt;YYMMDD&gt;'</span> <span class="c1">#TO BE REPLACED</span>
</span><span id="__span-7-14"><a id="__codelineno-7-14" name="__codelineno-7-14"></a><span class="n">NBNAME</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">DATESTR</span><span class="si">}</span><span class="s1">-finalize-signatures'</span>
</span><span id="__span-7-15"><a id="__codelineno-7-15" name="__codelineno-7-15"></a>
</span><span id="__span-7-16"><a id="__codelineno-7-16" name="__codelineno-7-16"></a><span class="n">infiles</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span>
</span><span id="__span-7-17"><a id="__codelineno-7-17" name="__codelineno-7-17"></a>    <span class="n">signatures</span><span class="o">=</span><span class="n">Path</span><span class="p">(</span><span class="sa">f</span><span class="s1">'data-intermediate/</span><span class="si">{</span><span class="n">DATESTR</span><span class="si">}</span><span class="s1">-signatures-and-dists/signatures.h5'</span><span class="p">),</span>
</span><span id="__span-7-18"><a id="__codelineno-7-18" name="__codelineno-7-18"></a><span class="p">)</span>
</span><span id="__span-7-19"><a id="__codelineno-7-19" name="__codelineno-7-19"></a>
</span><span id="__span-7-20"><a id="__codelineno-7-20" name="__codelineno-7-20"></a><span class="n">processed_data</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="s1">'data-processed'</span><span class="p">)</span> <span class="o">/</span> <span class="n">NBNAME</span>
</span><span id="__span-7-21"><a id="__codelineno-7-21" name="__codelineno-7-21"></a><span class="n">processed_data</span><span class="o">.</span><span class="n">mkdir</span><span class="p">(</span><span class="n">exist_ok</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">parents</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</span><span id="__span-7-22"><a id="__codelineno-7-22" name="__codelineno-7-22"></a>
</span><span id="__span-7-23"><a id="__codelineno-7-23" name="__codelineno-7-23"></a><span class="n">outfiles</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span>
</span><span id="__span-7-24"><a id="__codelineno-7-24" name="__codelineno-7-24"></a>    <span class="n">signatures</span><span class="o">=</span><span class="n">processed_data</span> <span class="o">/</span> <span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">DATESTR</span><span class="si">}</span><span class="s1">-&lt;db_name&gt;.h5'</span><span class="p">,</span> <span class="c1">#TO BE REPLACED</span>
</span><span id="__span-7-25"><a id="__codelineno-7-25" name="__codelineno-7-25"></a><span class="p">)</span>
</span><span id="__span-7-26"><a id="__codelineno-7-26" name="__codelineno-7-26"></a>
</span><span id="__span-7-27"><a id="__codelineno-7-27" name="__codelineno-7-27"></a><span class="c1"># ## Assign metadata</span>
</span><span id="__span-7-28"><a id="__codelineno-7-28" name="__codelineno-7-28"></a>
</span><span id="__span-7-29"><a id="__codelineno-7-29" name="__codelineno-7-29"></a><span class="n">meta</span> <span class="o">=</span> <span class="n">SignaturesMeta</span><span class="p">(</span>
</span><span id="__span-7-30"><a id="__codelineno-7-30" name="__codelineno-7-30"></a>    <span class="nb">id</span><span class="o">=</span><span class="s1">'&lt;database_name&gt;'</span><span class="p">,</span> <span class="c1"># TO BE REPLACED, suggested &lt;organization/name&gt;</span>
</span><span id="__span-7-31"><a id="__codelineno-7-31" name="__codelineno-7-31"></a>    <span class="n">version</span><span class="o">=</span><span class="s1">'x.y.z'</span><span class="p">,</span> <span class="c1"># TO BE REPLACED</span>
</span><span id="__span-7-32"><a id="__codelineno-7-32" name="__codelineno-7-32"></a>    <span class="n">name</span><span class="o">=</span><span class="s1">'&lt;name&gt;'</span><span class="p">,</span> <span class="c1"># TO BE REPLACED</span>
</span><span id="__span-7-33"><a id="__codelineno-7-33" name="__codelineno-7-33"></a>    <span class="n">description</span><span class="o">=</span><span class="s1">'&lt;description&gt;'</span><span class="p">,</span> <span class="c1"># TO BE REPLACED</span>
</span><span id="__span-7-34"><a id="__codelineno-7-34" name="__codelineno-7-34"></a>    <span class="n">id_attr</span><span class="o">=</span><span class="s1">'key'</span><span class="p">,</span>
</span><span id="__span-7-35"><a id="__codelineno-7-35" name="__codelineno-7-35"></a><span class="p">)</span>
</span><span id="__span-7-36"><a id="__codelineno-7-36" name="__codelineno-7-36"></a>
</span><span id="__span-7-37"><a id="__codelineno-7-37" name="__codelineno-7-37"></a><span class="k">with</span> <span class="n">load_signatures</span><span class="p">(</span><span class="n">infiles</span><span class="p">[</span><span class="s1">'signatures'</span><span class="p">])</span> <span class="k">as</span> <span class="n">src</span><span class="p">:</span>
</span><span id="__span-7-38"><a id="__codelineno-7-38" name="__codelineno-7-38"></a>    <span class="n">out_sigs</span> <span class="o">=</span> <span class="n">AnnotatedSignatures</span><span class="p">(</span><span class="n">src</span><span class="p">,</span> <span class="n">src</span><span class="o">.</span><span class="n">ids</span><span class="p">,</span> <span class="n">meta</span><span class="p">)</span>
</span><span id="__span-7-39"><a id="__codelineno-7-39" name="__codelineno-7-39"></a>    <span class="n">dump_signatures</span><span class="p">(</span><span class="n">outfiles</span><span class="p">[</span><span class="s1">'signatures'</span><span class="p">],</span> <span class="n">out_sigs</span><span class="p">)</span>
</span></code></pre></div></td></tr></tbody></table></div>
<h2 id="help-available">Help Available<a class="headerlink" href="#help-available" title="Permanent link">¶</a></h2>
<p>Please feel free to reach out to <a href="mailto:support@theiagen.com">support@theiagen.com</a> with any questions you might have.</p>







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="May 8, 2025 21:49:10 UTC">2025-05-08</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="March 26, 2025 19:00:08 UTC">2025-03-26</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      © 2022-2025 <a href="https://www.theiagen.com" target="_blank" rel="noopener">Theiagen Genomics</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/theiagen" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://bsky.app/profile/theiagen.bsky.social" target="_blank" rel="noopener" title="bsky.app" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M407.8 294.7c-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3M288 227.1c-26.1-50.7-97.1-145.2-163.1-191.8C61.6-9.4 37.5-1.7 21.6 5.5 3.3 13.8 0 41.9 0 58.4S9.1 194 15 213.9c19.5 65.7 89.1 87.9 153.2 80.7 3.3-.5 6.6-.9 10-1.4-3.3.5-6.6 1-10 1.4-93.9 14-177.3 48.2-67.9 169.9C220.6 589.1 265.1 437.8 288 361.1c22.9 76.7 49.2 222.5 185.6 103.4 102.4-103.4 28.1-156-65.8-169.9-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3 64.1 7.1 133.6-15.1 153.2-80.7C566.9 194 576 75 576 58.4s-3.3-44.7-21.6-52.9c-15.8-7.1-40-14.9-103.2 29.8C385.1 81.9 314.1 176.4 288 227.1"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.linkedin.com/company/theiagen" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3M135.4 416H69V202.2h66.5V416zM102.2 96a38.5 38.5 0 1 1 0 77 38.5 38.5 0 1 1 0-77m282.1 320h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9z"></path></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "navigation.instant", "navigation.path", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.suggest", "toc.follow"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"alias": true, "default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../../javascripts/tablesort.js"></script>
      
        <script src="../../javascripts/table-search.js"></script>
      
    
  
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>