
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A collection of bioinformatics workflows for genomic characterization, submission preparation, and genomic epidemiology of pathogens of public health concern.">
      
      
        <meta name="author" content="Theiagen Genomics">
      
      
        <link rel="canonical" href="https://theiagen.github.io/public_health_bioinformatics/main/workflows/genomic_characterization/theiaviral/">
      
      
        <link rel="prev" href="../../standalone/theiavalidate/">
      
      
        <link rel="next" href="../../data_export/transfer_column_content/">
      
      
      <link rel="icon" href="../../../assets/logos/Theiagen-Symbol-Standard-01.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.16">
    
    
      
        <title>TheiaViral Workflow Series - Public Health Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.7e37652d.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#theiaviral-workflow-series" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version release.
  <a href="../../../.."> 
    <strong>Click here to go to the latest version release.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-header__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Public Health Bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              TheiaViral Workflow Series
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to Dark Mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to Dark Mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo"  aria-label="Switch to Light Mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to Light Mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m3.55 19.09 1.41 1.41 1.8-1.79-1.42-1.42M12 6c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6c0-3.32-2.69-6-6-6m8 7h3v-2h-3m-2.76 7.71 1.8 1.79 1.41-1.41-1.79-1.8M20.45 5l-1.41-1.4-1.8 1.79 1.42 1.42M13 1h-2v3h2M6.76 5.39 4.96 3.6 3.55 5l1.79 1.81zM1 13h3v-2H1m12 9h-2v3h2"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../.." class="md-tabs__link">
          
  
  
  Public Health Bioinformatics

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-nav__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    Public Health Bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Public Health Bioinformatics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Public Health Bioinformatics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_2" >
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Getting Started
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            Getting Started
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/commandline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With the Command-Line
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/terra/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With Terra.bio
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/philosophy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Workflow Failure Philosophy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/sops/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Available SOPs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" >
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit_database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT Database Creation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/phylogenetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Phylogenetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/custom_organisms/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Running Custom Organisms on TheiaCoV
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1" >
        
          
          <label class="md-nav__link" for="__nav_1_5_1" id="__nav_1_5_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Type
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Type
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_type/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_2" >
        
          
          <label class="md-nav__link" for="__nav_1_5_1_2" id="__nav_1_5_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Import
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_2">
            <span class="md-nav__icon md-icon"></span>
            Data Import
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_3" >
        
          
          <label class="md-nav__link" for="__nav_1_5_1_3" id="__nav_1_5_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Genomic Characterization
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_3">
            <span class="md-nav__icon md-icon"></span>
            Genomic Characterization
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaviral-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaViral Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaViral Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#workflow-diagram" class="md-nav__link">
    <span class="md-ellipsis">
      Workflow Diagram
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaviral-workflows-for-different-input-types" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaViral Workflows for Different Input Types
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaViral Workflows for Different Input Types">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      All Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="All Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#taxa-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Taxa-Specific Tasks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#acknowlegments" class="md-nav__link">
    <span class="md-ellipsis">
      Acknowlegments
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_4" >
        
          
          <label class="md-nav__link" for="__nav_1_5_1_4" id="__nav_1_5_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Construction
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_4">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Construction
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_5" >
        
          
          <label class="md-nav__link" for="__nav_1_5_1_5" id="__nav_1_5_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Placement
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_5">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Placement
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/samples_to_ref_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Samples_to_Ref_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher_PHB
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_6" >
        
          
          <label class="md-nav__link" for="__nav_1_5_1_6" id="__nav_1_5_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Public Data Sharing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_6">
            <span class="md-nav__icon md-icon"></span>
            Public Data Sharing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_7" >
        
          
          <label class="md-nav__link" for="__nav_1_5_1_7" id="__nav_1_5_1_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exporting Data from Terra
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_7">
            <span class="md-nav__icon md-icon"></span>
            Exporting Data from Terra
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_8" >
        
          
          <label class="md-nav__link" for="__nav_1_5_1_8" id="__nav_1_5_1_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Standalone
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_8">
            <span class="md-nav__icon md-icon"></span>
            Standalone
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/host_decontaminate.md" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Host_Decontaminate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2" >
        
          
          <label class="md-nav__link" for="__nav_1_5_2" id="__nav_1_5_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Kingdom
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Kingdom
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_kingdom/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_2" >
        
          
          <label class="md-nav__link" for="__nav_1_5_2_2" id="__nav_1_5_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Any Taxa
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_2">
            <span class="md-nav__icon md-icon"></span>
            Any Taxa
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/host_decontaminate.md" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Host_Decontaminate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_3" >
        
          
          <label class="md-nav__link" for="__nav_1_5_2_3" id="__nav_1_5_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Bacteria
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_3">
            <span class="md-nav__icon md-icon"></span>
            Bacteria
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_4" >
        
          
          <label class="md-nav__link" for="__nav_1_5_2_4" id="__nav_1_5_2_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Mycotics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_4">
            <span class="md-nav__icon md-icon"></span>
            Mycotics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_5" >
        
          
          <label class="md-nav__link" for="__nav_1_5_2_5" id="__nav_1_5_2_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Viral
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_5">
            <span class="md-nav__icon md-icon"></span>
            Viral
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/samples_to_ref_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Samples_to_Ref_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaviral-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaViral Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaViral Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#workflow-diagram" class="md-nav__link">
    <span class="md-ellipsis">
      Workflow Diagram
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaviral-workflows-for-different-input-types" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaViral Workflows for Different Input Types
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaViral Workflows for Different Input Types">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      All Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="All Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#taxa-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Taxa-Specific Tasks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#acknowlegments" class="md-nav__link">
    <span class="md-ellipsis">
      Acknowlegments
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher_PHB
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_3" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows Alphabetically
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            Workflows Alphabetically
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_alphabetically/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/host_decontaminate.md" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Host_Decontaminate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/samples_to_ref_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Samples_to_Ref_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaviral-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaViral Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaViral Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#workflow-diagram" class="md-nav__link">
    <span class="md-ellipsis">
      Workflow Diagram
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaviral-workflows-for-different-input-types" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaViral Workflows for Different Input Types
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaViral Workflows for Different Input Types">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      All Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="All Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#taxa-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Taxa-Specific Tasks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#acknowlegments" class="md-nav__link">
    <span class="md-ellipsis">
      Acknowlegments
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher_PHB
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6" >
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Contributing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            Contributing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/doc_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Documentation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/code_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Code
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaviral-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaViral Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaViral Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#workflow-diagram" class="md-nav__link">
    <span class="md-ellipsis">
      Workflow Diagram
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaviral-workflows-for-different-input-types" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaViral Workflows for Different Input Types
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaViral Workflows for Different Input Types">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      All Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="All Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#taxa-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Taxa-Specific Tasks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#acknowlegments" class="md-nav__link">
    <span class="md-ellipsis">
      Acknowlegments
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="theiaviral-workflow-series">TheiaViral Workflow Series<a class="headerlink" href="#theiaviral-workflow-series" title="Permanent link">&para;</a></h1>
<h2 id="quick-facts">Quick Facts<a class="headerlink" href="#quick-facts" title="Permanent link">&para;</a></h2>
<table>
<thead>
<tr>
<th><strong>Workflow Type</strong></th>
<th><strong>Applicable Kingdom</strong></th>
<th><strong>Last Known Changes</strong></th>
<th><strong>Command-line Compatibility</strong></th>
<th><strong>Workflow Level</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../../../workflows_overview/workflows_type/#genomic-characterization">Genomic Characterization</a></td>
<td><a href="../../../workflows_overview/workflows_kingdom/#viral">Viral</a></td>
<td>v3.1.0</td>
<td>No</td>
<td>Sample-level</td>
</tr>
</tbody>
</table>
<h2 id="theiaviral-workflows">TheiaViral Workflows<a class="headerlink" href="#theiaviral-workflows" title="Permanent link">&para;</a></h2>
<p><strong>TheiaViral</strong> workflows assemble, quality assess, and characterize viral genomes from diverse data sources, including metagenomic samples. TheiaViral workflows can generate consensus assemblies of recalcitrant viruses, including diverse or recombinant lineages, such as rabies virus and norovirus, through a three-step approach: 1) generating an intermediate <em>de novo</em> assembly from taxonomy-filtered reads, 2) selecting the best reference from a database of ~200,000 viral genomes using average nucleotide identity, and 3) producing a final consensus assembly through reference-based read mapping and variant calling. Reference genomes can be directly provided to TheiaViral to bypass <em>de novo</em> assembly, which enables compatibility with tiled amplicon sequencing data. Targeted viral characterization is currently ongoing and functional for <em>Lyssavirus rabies</em>.</p>
<details class="question" open="open">
<summary>What are the main differences between the TheiaViral and TheiaCov workflows?</summary>
<div class="grid cards" markdown>
<ul>
<li>
<p><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 3C7.58 3 4 4.79 4 7s3.58 4 8 4 8-1.79 8-4-3.58-4-8-4M4 9v3c0 2.21 3.58 4 8 4s8-1.79 8-4V9c0 2.21-3.58 4-8 4s-8-1.79-8-4m0 5v3c0 2.21 3.58 4 8 4s8-1.79 8-4v-3c0 2.21-3.58 4-8 4s-8-1.79-8-4"/></svg></span> <strong>TheiaCov Workflows</strong></p>
<hr />
<ul>
<li>For amplicon-derived viral sequencing methods</li>
<li>Supports a limited number of <a href="../theiacov/#supported-organisms">pathogens</a></li>
<li>Uses manually curated, static reference genomes</li>
</ul>
</li>
<li>
<p><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 3C7.58 3 4 4.79 4 7s3.58 4 8 4 8-1.79 8-4-3.58-4-8-4M4 9v3c0 2.21 3.58 4 8 4s8-1.79 8-4V9c0 2.21-3.58 4-8 4s-8-1.79-8-4m0 5v3c0 2.21 3.58 4 8 4s8-1.79 8-4v-3c0 2.21-3.58 4-8 4s-8-1.79-8-4"/></svg></span> <strong>TheiaViral Workflows</strong></p>
<hr />
<ul>
<li>Designed for a variety of sequencing methods</li>
<li>Supports relatively diverse and recombinant pathogens</li>
<li>Dynamically identifies the most similar reference genome for consensus assembly via an intermediate <em>de novo</em> assembly</li>
</ul>
</li>
</ul>
</div>
</details>
<details class="question" open="open">
<summary>Segmented viruses</summary>
<p>Segmented viruses are accounted for in TheiaViral. The reference genome database excludes segmented viral nucleotide accessions, while including RefSeq assembly accessions that include all viral segments. Consensus assembly modules are constructed to handle multi-segment references.</p>
</details>
<h3 id="workflow-diagram">Workflow Diagram<a class="headerlink" href="#workflow-diagram" title="Permanent link">&para;</a></h3>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="theiaviral_illumina_pe" name="__tabbed_1" type="radio" /><input id="theiaviral_ont" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="theiaviral_illumina_pe">TheiaViral_Illumina_PE</label><label for="theiaviral_ont">TheiaViral_ONT</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="admonition caption">
<p class="admonition-title">TheiaViral_Illumina_PE Workflow Diagram</p>
<p><a class="glightbox" href="../../../assets/figures/TheiaViral_Illumina_PE.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="TheiaViral_Illumina_PE Workflow Diagram" src="../../../assets/figures/TheiaViral_Illumina_PE.png" /></a></p>
</div>
</div>
<div class="tabbed-block">
<div class="admonition caption">
<p class="admonition-title">TheiaViral_ONT Workflow Diagram</p>
<p><a class="glightbox" href="../../../assets/figures/TheiaViral_ONT.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="TheiaViral_ONT Workflow Diagram" src="../../../assets/figures/TheiaViral_ONT.png" /></a></p>
</div>
</div>
</div>
</div>
<h2 id="theiaviral-workflows-for-different-input-types">TheiaViral Workflows for Different Input Types<a class="headerlink" href="#theiaviral-workflows-for-different-input-types" title="Permanent link">&para;</a></h2>
<div class="grid cards ">
<ul>
<li>
<p><center> <strong>TheiaViral_Illumina_PE</strong> </center></p>
<hr />
<div class="admonition dna">
<p class="admonition-title">Illumina_PE Input Read Data</p>
<p>The TheiaViral_Illumina_PE workflow inputs Illumina paired-end read data. Read file extensions should be <code>.fastq</code> or <code>.fq</code>, and can optionally include the <code>.gz</code> compression extension. Theiagen recommends compressing files with <a href="https://www.gnu.org/software/gzip/">gzip</a> before Terra uploads to minimize data upload time and storage costs.</p>
<p>Modifications to the optional parameter for <code>trim_minlen</code> may be required to appropriately trim reads shorter than 2 x 150 bp (i.e. generated using a 300-cycle sequencing kit), such as the 2 x 75bp reads generated using a 150-cycle sequencing kit.</p>
</div>
</li>
<li>
<p><center> <strong>TheiaViral_ONT</strong> </center></p>
<hr />
<div class="admonition dna_blue">
<p class="admonition-title">ONT Input Read Data</p>
<p>The TheiaViral_ONT workflow inputs base-called Oxford Nanopore Technology (ONT) read data. Read file extensions should be <code>.fastq</code> or <code>.fq</code>, and can optionally include the <code>.gz</code> compression extension. Theiagen recommends compressing files with <a href="https://www.gnu.org/software/gzip/">gzip</a> before Terra uploads to minimize data upload time and storage costs.</p>
<p>It is recommended to trim adapter sequencings via <code>dorado</code> basecalling prior to running TheiaViral_ONT, though <code>porechop</code> can optionally be called to trim adapters within the workflow.</p>
<p><strong>The ONT sequencing kit and base-calling approach can produce substantial variability in the amount and quality of read data. Genome assemblies produced by the TheiaViral_ONT workflow must be quality assessed before reporting results. We recommend using the <a href="../../standalone/dorado_basecalling/">Dorado_Basecalling_PHB</a> workflow if applicable.</strong></p>
</div>
</li>
</ul>
</div>
<h3 id="inputs">Inputs<a class="headerlink" href="#inputs" title="Permanent link">&para;</a></h3>
<details class="dna_blue" open="open">
<summary><code>taxon</code> <em>required</em> input parameter</summary>
<p><code>taxon</code> is the standardized taxonomic name (e.g. "Lyssavirus rabies") or NCBI taxon ID (e.g. "11292") of the desired virus to analyze. Inputs must be represented in the <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi">NCBI taxonomy database</a> and do not have to be species-level (see <code>read_extraction_rank</code> below).</p>
</details>
<div style="float: left; width: 50%; padding-right: 10px;">
<details class="dna">
<summary><code>host</code> optional input parameter</summary>
<p>The <code>host</code> input triggers the Host Decontaminate workflow, which removes reads that map to a reference host genome. This input needs to be an <a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy-compatible</a> taxon or an NCBI assembly accession. If using a taxon, the first retrieved genome corresponding to that taxon is retrieved. If using an accession, it must be coupled with the Host Decontaminate task <code>is_accession</code> (ONT) or Read QC Trim PE <code>host_is_accession</code> (Illumina) boolean populated as "true".</p>
</details>
<details class="dna">
<summary><code>extract_unclassified</code> optional input parameter</summary>
<p>By default, the <code>extract_unclassified</code> parameter is set to "true", which indicates that reads that are not classified by Kraken2 (Illumina) or Metabuli (ONT) will be included with reads classified as the input <code>taxon</code>. These classification software most often do not comprehensively classify reads using the default RefSeq databases, so extracting unclassified reads is desirable when host and contaminant reads have been sufficiently decontaminated. Host decontamination occurs in TheiaViral using NCBI <code>sra-human-scrubber</code>, read classification to the human genome, and/or via mapping reads to the inputted <code>host</code>. Contaminant viral reads are mostly excluded because they will be often be classified against the default RefSeq classification databases. Consider setting <code>extract_unclassified</code> to false if <em>de novo</em> assembly or Skani reference selection is failing.</p>
</details>
</div>
<div style="float: right; width: 50%; padding-left: 10px;">
<details class="dna">
<summary><code>min_allele_freq</code>, <code>min_depth</code>, and <code>min_map_quality</code> optional input parameters</summary>
<p>These parameters have a direct effect on the variants that will ultimately be reported in the consensus assembly. <code>min_allele_freq</code> determines the minimum proportion of an allelic variant to be reported in the consensus assembly. <code>min_depth</code> and <code>min_map_quality</code> affect how "N" is reported in the consensus, i.e. depth below <code>min_depth</code> is reported as "N" and reads with mapping quality below <code>min_map_quality</code> are not included in depth calculations.</p>
</details>
<details class="dna">
<summary><code>read_extraction_rank</code> optional input parameter</summary>
<p>By default, the <code>read_extraction_rank</code> parameter is set to "family", which indicates that reads will be extracted if they are classified as the taxonomic family of the input <code>taxon</code>, including all descendant taxa of the family. Read classification may not resolve to the rank of the input <code>taxon</code>, so these reads may be classified at higher ranks. For example, some <em>Lyssavirus rabies</em> (species) reads may only be resolved to <em>Lyssavirus</em> (genus), so they would not be extracted if the <code>read_extraction_rank</code> is set to "species". Setting the <code>read_extraction_rank</code> above the inputted <code>taxon</code>'s rank can therefore dramatically increase the number of reads recovered, at the potential cost of including other viruses. This likely is not a problem for scarcely represented lineages, e.g. a sample that is expected to include <em>Lyssavirus rabies</em> is unlikely to contain other viruses of the corresponding family, Rhabdoviridae, within the same sample. However, setting a <code>read_extraction_rank</code> far beyond the input <code>taxon</code> rank can be problematic when multiple representatives of the same viral family are included in similar abundance within the same sample. To further refine the desired <code>read_extraction_rank</code>, please review the corresponding classification reports of the respective classification software (kraken2 for Illumina and Metabuli for ONT)</p>
</details>
</div>
<div style="clear: both;"></div>
<div class="tabbed-set tabbed-alternate" data-tabs="2:2"><input checked="checked" id="theiaviral_illumina_pe_1" name="__tabbed_2" type="radio" /><input id="theiaviral_ont_1" name="__tabbed_2" type="radio" /><div class="tabbed-labels"><label for="theiaviral_illumina_pe_1">TheiaViral_Illumina_PE</label><label for="theiaviral_ont_1">TheiaViral_ONT</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>llumina forward read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>llumina reverse read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>Nme of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>taxon</strong></td>
<td>String</td>
<td>Taxon ID or organism name of interest</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>6</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>checkv_db</strong></td>
<td>File</td>
<td>CheckV database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/checkv/checkv-db-v1.5.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/checkv:1.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>checkv_db</strong></td>
<td>File</td>
<td>CheckV database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/checkv/checkv-db-v1.5.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/checkv:1.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>max_genome_length</strong></td>
<td>Int</td>
<td>Maximum genome length able to pass read screening</td>
<td>2673870</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>min_basepairs</strong></td>
<td>Int</td>
<td>Minimum base pairs to pass read screening</td>
<td>15000</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>min_coverage</strong></td>
<td>Int</td>
<td>Minimum coverage to pass read screening</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>min_genome_length</strong></td>
<td>Int</td>
<td>Minimum genome length to pass read screening</td>
<td>1500</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>min_proportion</strong></td>
<td>Int</td>
<td>Minimum read proportion to pass read screening</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>min_reads</strong></td>
<td>Int</td>
<td>Minimum reads to pass read screening</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>char_unknown</strong></td>
<td>String</td>
<td>Character used to represent unknown bases in the consensus sequence</td>
<td>N</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>count_orphans</strong></td>
<td>Boolean</td>
<td>True/False that determines if anomalous read pairs are NOT skipped in variant calling. Anomalous read pairs are those marked in the FLAG field as paired in sequencing but without the properly-paired flag set.</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>disable_baq</strong></td>
<td>Boolean</td>
<td>True/False that determines if base alignment quality (BAQ) computation should be disabled during samtools mpileup before consensus generation</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/artic-ncov2019-epi2me</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>max_depth</strong></td>
<td>Int</td>
<td>For a given position, read at maximum INT number of reads per input file during samtools mpileup before consensus generation</td>
<td>600000</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>min_bq</strong></td>
<td>Int</td>
<td>Minimum base quality required for a base to be considered during samtools mpileup before consensus generation</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>skip_N</strong></td>
<td>Boolean</td>
<td>True/False that determines if "N" bases should be skipped in the consensus sequence</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_variants</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_variants</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_variants</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_variants</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_variants</td>
<td><strong>reference_gff</strong></td>
<td>File</td>
<td>A GFF file in the GFF3 format can be supplied to specify coordinates of open reading frames (ORFs) so iVar can identify codons and translate variants into amino acids</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>megahit</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>megahit</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>megahit</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/megahit:1.2.9</td>
<td>Optional</td>
</tr>
<tr>
<td>megahit</td>
<td><strong>kmers</strong></td>
<td>String</td>
<td>Comma-separated list of kmer sizes to use for assembly. All must be odd, in the range 15-255, increment &lt;= 28</td>
<td>21,29,39,59,79,99,119,141</td>
<td>Optional</td>
</tr>
<tr>
<td>megahit</td>
<td><strong>megahit_opts</strong></td>
<td>String</td>
<td>Additional parameters for MEGAHIT assembler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>megahit</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>megahit</td>
<td><strong>min_contig_length</strong></td>
<td>Int</td>
<td>Minimum contig length for MEGAHIT assembler</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-insaflu-220727</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity</td>
<td>70</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>assembly_metrics_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>assembly_metrics_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>assembly_metrics_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>assembly_metrics_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>consensus_qc_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>consensus_qc_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>consensus_qc_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>consensus_qc_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_cross_reference</strong></td>
<td>File</td>
<td>An Excel file to cross-reference BLAST findings; probably useful if novel genotypes are not in the default file used by genoflu.py</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/genoflu:1.06</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/irma:1.2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_keep_ref_deletions</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if sites missed (i.e. 0 reads for a site in the reference genome) during read gathering should be deleted by ambiguation by inserting N's or deleting the sequence entirely. False sets this IRMA paramater to "DEL" and true sets it to "NNN"</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.10.2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_output_parser_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_output_parser_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_output_parser_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_output_parser_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>pangolin_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>pangolin_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>pangolin_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.10.2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>pangolin_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-datasets:16.38.1</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>include_gbff</strong></td>
<td>Boolean</td>
<td>True/False to include gbff files in the output</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>include_gff3</strong></td>
<td>Boolean</td>
<td>True/False to include gff3 files in the output</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>complete</strong></td>
<td>Boolean</td>
<td>Only query genomes labeled complete</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-datasets:16.38.1</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>refseq</strong></td>
<td>Boolean</td>
<td>Only query RefSeq genomes</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>summary_limit</strong></td>
<td>Int</td>
<td>Maximum number of genomes to return in the summary</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>use_ncbi_virus</strong></td>
<td>Boolean</td>
<td>Set to true to download from NCBI Virus Datasets</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>quast_denovo</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast_denovo</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>quast_denovo</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast_denovo</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>bases</strong></td>
<td>String</td>
<td>Explicitly set the number of bases required e.g., 4.3kb, 7Tb, 9000, 4.1MB. If this option is given, --coverage and --genome-size are ignored</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>coverage</strong></td>
<td>Float</td>
<td>The desired coverage to sub-sample the reads to. If --bases is not provided, this option and --genome-size are required</td>
<td>250</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/rasusa:2.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>frac</strong></td>
<td>Float</td>
<td>Subsample to a fraction of the reads - e.g., 0.5 samples half the reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>num</strong></td>
<td>Int</td>
<td>Subsample to a specific number of reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>seed</strong></td>
<td>Int</td>
<td>Random seed for reproducibility</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>File with adapter sequences to be removed</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Additional arguments to use with fastp</td>
<td>--detect_adapter_for_pe -g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_complete_only</strong></td>
<td>Boolean</td>
<td>Only download host reference genome labeled "complete"</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_decontaminate_mem</strong></td>
<td>Int</td>
<td>Memory allocated for minimap2 (in GB)</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_is_accession</strong></td>
<td>Boolean</td>
<td>Inputted "host" is an accession</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>A file containing the phix used during Illumina sequencing; used in the BBDuk task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>The name of the tool to perform basic read processing; options: "trimmomatic" or "fastp"</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>The tool used for quality control (QC) of reads. Options are "fastq_scan" (default) and "fastqc"</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>trim_min_length</strong></td>
<td>Int</td>
<td>Specifies minimum length of each read after trimming to be kept</td>
<td>75</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>Specifies the average quality of bases in a sliding window to be kept</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>Specifies window size for trimming (the number of bases to average the quality across)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Additional arguments to pass to trimmomatic. "-phred33" specifies the Phred Q score encoding which is almost always phred33 with modern sequence data.</td>
<td>-phred33</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_mapping_stats</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_mapping_stats</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_mapping_stats</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>read_mapping_stats</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>skani</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>skani</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>skani</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/skani:0.2.2</td>
<td>Optional</td>
</tr>
<tr>
<td>skani</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>skani</td>
<td><strong>skani_db</strong></td>
<td>File</td>
<td>Skani database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/skani/skani_db_20250606.tar</td>
<td>Optional</td>
</tr>
<tr>
<td>spades</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>spades</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>spades</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/spades:4.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>spades</td>
<td><strong>kmers</strong></td>
<td>String</td>
<td>list of k-mer sizes (must be odd and less than 128)</td>
<td>auto</td>
<td>Optional</td>
</tr>
<tr>
<td>spades</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>spades</td>
<td><strong>phred_offset</strong></td>
<td>Int</td>
<td>PHRED quality offset in the input reads (33 or 64)</td>
<td>33</td>
<td>Optional</td>
</tr>
<tr>
<td>spades</td>
<td><strong>spades_opts</strong></td>
<td>String</td>
<td>Additional parameters for Spades assembler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>call_metaviralspades</strong></td>
<td>Boolean</td>
<td>True/False to call assembly with MetaviralSPAdes and use Megahit as fallback</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>extract_unclassified</strong></td>
<td>Boolean</td>
<td>True/False that determines if unclassified reads should be extracted and combined with the taxon specific extracted reads</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Expected genome length of taxon of interest</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>host</strong></td>
<td>String</td>
<td>Host taxon/accession to dehost reads, if provided</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>Kraken2 database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>min_allele_freq</strong></td>
<td>Float</td>
<td>Minimum allele frequency required for a variant to populate the consensus sequence</td>
<td>0.6</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>min_depth</strong></td>
<td>Int</td>
<td>Minimum read depth required for a variant to populate the consensus sequence</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>min_map_quality</strong></td>
<td>Int</td>
<td>Minimum mapping quality required for read alignments</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>read_extraction_rank</strong></td>
<td>String</td>
<td>Taxonomic rank to use for read extraction - limits taxons to only those within the specified ranks.</td>
<td>family</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>reference_fasta</strong></td>
<td>File</td>
<td>Reference genome in FASTA format</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>skip_rasusa</strong></td>
<td>Boolean</td>
<td>True/False to skip read subsampling with Rasusa</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>skip_screen</strong></td>
<td>Boolean</td>
<td>True/False to skip read screening check prior to analysis</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiaviral_ont</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>Base-called ONT read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>Name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>taxon</strong></td>
<td>String</td>
<td>Taxon ID or organism name of interest</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>bcftools_consensus</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>bcftools_consensus</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>bcftools_consensus</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/bcftools:1.20</td>
<td>Optional</td>
</tr>
<tr>
<td>bcftools_consensus</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>checkv_db</strong></td>
<td>File</td>
<td>CheckV database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/checkv/checkv-db-v1.5.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/checkv:1.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>checkv_db</strong></td>
<td>File</td>
<td>CheckV database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/checkv/checkv-db-v1.5.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/checkv:1.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>clair3_model</strong></td>
<td>String</td>
<td>Model to be used by Clair3</td>
<td>r1041_e82_400bps_sup_v500</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>disable_phasing</strong></td>
<td>Boolean</td>
<td>True/False that determines if variants should be called without whatshap phasing in full alignment calling</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/clair3-extra-models:1.0.10</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>enable_gvcf</strong></td>
<td>Boolean</td>
<td>True/False that determines if an additional GVCF output should generated</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>enable_haploid_precise</strong></td>
<td>Boolean</td>
<td>True/False that determines haploid calling mode where only 1/1 is considered as a variant</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>include_all_contigs</strong></td>
<td>Boolean</td>
<td>True/False that determines if all contigs should be included in the output</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>indel_min_af</strong></td>
<td>Float</td>
<td>Minimum Indel AF required for a candidate variant</td>
<td>0.08</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>snp_min_af</strong></td>
<td>Float</td>
<td>Minimum SNP AF required for a candidate variant</td>
<td>0.08</td>
<td>Optional</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>variant_quality</strong></td>
<td>Int</td>
<td>If set, variants with &gt;$qual will be marked PASS, or LowQual otherwise</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>max_genome_length</strong></td>
<td>Int</td>
<td>Maximum genome length able to pass read screening</td>
<td>2673870</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>min_basepairs</strong></td>
<td>Int</td>
<td>Minimum base pairs to pass read screening</td>
<td>15000</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>min_coverage</strong></td>
<td>Int</td>
<td>Minimum coverage to pass read screening</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>min_genome_length</strong></td>
<td>Int</td>
<td>Minimum genome length to pass read screening</td>
<td>1500</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>min_reads</strong></td>
<td>Int</td>
<td>Minimum reads to pass read screening</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>skip_mash</strong></td>
<td>Boolean</td>
<td>If true, skips estimation of genome size and coverage using mash in read screening steps. As a result, providing true also prevents screening using these parameters.</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>fasta_utilities</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>fasta_utilities</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>fasta_utilities</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/seqkit:2.4.0--h9ee0642_0</td>
<td>Optional</td>
</tr>
<tr>
<td>fasta_utilities</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>additional_parameters</strong></td>
<td>String</td>
<td>Additional parameters for Flye assembler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>asm_coverage</strong></td>
<td>Int</td>
<td>Reduced coverage for initial disjointig assembly</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/flye:2.9.4</td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>flye_polishing_iterations</strong></td>
<td>Int</td>
<td>Number of polishing iterations</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Expected genome length for assembly - requires asm_coverage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>keep_haplotypes</strong></td>
<td>Boolean</td>
<td>True/False to prevent collapsing alternative haplotypes</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>minimum_overlap</strong></td>
<td>Int</td>
<td>Minimum overlap between reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>no_alt_contigs</strong></td>
<td>Boolean</td>
<td>True/False to disable alternative contig generation</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>read_error_rate</strong></td>
<td>Float</td>
<td>Expected error rate in reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>read_type</strong></td>
<td>String</td>
<td>Type of read data for Flye</td>
<td>--nano-hq</td>
<td>Optional</td>
</tr>
<tr>
<td>flye</td>
<td><strong>scaffold</strong></td>
<td>Boolean</td>
<td>True/False to enable scaffolding using graph</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>host_decontaminate</td>
<td><strong>complete_only</strong></td>
<td>Boolean</td>
<td>Only download genomes labeled "complete"</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>host_decontaminate</td>
<td><strong>is_accession</strong></td>
<td>Boolean</td>
<td>Inputted "host" is an accession</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>host_decontaminate</td>
<td><strong>minimap2_memory</strong></td>
<td>Int</td>
<td>Memory allocated for minimap2 (in GB)</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>host_decontaminate</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal componenet, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>host_decontaminate</td>
<td><strong>refseq</strong></td>
<td>Boolean</td>
<td>Only download RefSeq genomes</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>mask_low_coverage</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>mask_low_coverage</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>mask_low_coverage</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/bedtools:2.31.0</td>
<td>Optional</td>
</tr>
<tr>
<td>mask_low_coverage</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/metabuli:1.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>metabuli_db</strong></td>
<td>File</td>
<td>Metabuli database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/metabuli/refseq_virus-v223.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum query coverage threshold (0.0 - 1.0)</td>
<td>0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>min_score</strong></td>
<td>Float</td>
<td>Minimum sequenece similarity score (0.0 - 1.0)</td>
<td>0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>min_sp_score</strong></td>
<td>Float</td>
<td>Minimum score for species- or lower-level classification</td>
<td>0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>taxonomy_path</strong></td>
<td>File</td>
<td>Path to taxonomy file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/metabuli/new_taxdump.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>query2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-insaflu-220727</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity</td>
<td>70</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>assembly_metrics_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>assembly_metrics_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>assembly_metrics_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>assembly_metrics_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>consensus_qc_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>consensus_qc_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>consensus_qc_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>consensus_qc_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_cross_reference</strong></td>
<td>File</td>
<td>An Excel file to cross-reference BLAST findings; probably useful if novel genotypes are not in the default file used by genoflu.py</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/genoflu:1.06</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/irma:1.2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_keep_ref_deletions</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if sites missed (i.e. 0 reads for a site in the reference genome) during read gathering should be deleted by ambiguation by inserting N's or deleting the sequence entirely. False sets this IRMA paramater to "DEL" and true sets it to "NNN"</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.10.2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_output_parser_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_output_parser_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_output_parser_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_output_parser_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>pangolin_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>pangolin_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>pangolin_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.10.2</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>pangolin_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoq</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoq</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoq</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/nanoq:0.9.0--hec16e2b_1</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoq</td>
<td><strong>max_read_length</strong></td>
<td>Int</td>
<td>Maximum read length to keep</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoq</td>
<td><strong>max_read_qual</strong></td>
<td>Int</td>
<td>Maximum read quality to keep</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoq</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoq</td>
<td><strong>min_read_length</strong></td>
<td>Int</td>
<td>Minimum read length to keep</td>
<td>500</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoq</td>
<td><strong>min_read_qual</strong></td>
<td>Int</td>
<td>Minimum read quality to keep</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-datasets:16.38.1</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>include_gbff</strong></td>
<td>Boolean</td>
<td>True/False to include gbff files in the output</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>include_gff3</strong></td>
<td>Boolean</td>
<td>True/False to include gff3 files in the output</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>complete</strong></td>
<td>Boolean</td>
<td>Only query genomes labeled complete</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-datasets:16.38.1</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>refseq</strong></td>
<td>Boolean</td>
<td>Only query RefSeq genomes</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>summary_limit</strong></td>
<td>Int</td>
<td>Maximum number of genomes to return in the summary</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>use_ncbi_virus</strong></td>
<td>Boolean</td>
<td>Set to true to download from NCBI Virus Datasets</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_scrub_se</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_scrub_se</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_scrub_se</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_scrub_se</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>parse_mapping</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>parse_mapping</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>parse_mapping</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>Optional</td>
</tr>
<tr>
<td>parse_mapping</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>porechop</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>porechop</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>porechop</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/porechop:0.2.4</td>
<td>Optional</td>
</tr>
<tr>
<td>porechop</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>porechop</td>
<td><strong>trimopts</strong></td>
<td>String</td>
<td>Additional trimming options for Porechop</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>quast_denovo</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast_denovo</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>quast_denovo</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast_denovo</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>bases</strong></td>
<td>String</td>
<td>Explicitly set the number of bases required e.g., 4.3kb, 7Tb, 9000, 4.1MB. If this option is given, --coverage and --genome-size are ignored</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>coverage</strong></td>
<td>Float</td>
<td>The desired coverage to sub-sample the reads to. If --bases is not provided, this option and --genome-size are required</td>
<td>250</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/rasusa:2.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>frac</strong></td>
<td>Float</td>
<td>Subsample to a fraction of the reads - e.g., 0.5 samples half the reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>num</strong></td>
<td>Int</td>
<td>Subsample to a specific number of reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>seed</strong></td>
<td>Int</td>
<td>Random seed for reproducibility</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>raven</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>raven</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>raven</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/raven:1.8.3</td>
<td>Optional</td>
</tr>
<tr>
<td>raven</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>raven</td>
<td><strong>raven_identity</strong></td>
<td>Float</td>
<td>Threshold for overlap between two reads in order to construct an edge between them</td>
<td>0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>raven</td>
<td><strong>raven_opts</strong></td>
<td>Int</td>
<td>Additional parameters for Raven assembler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>raven</td>
<td><strong>raven_polishing_iterations</strong></td>
<td>Int</td>
<td>Number of polishing iterations</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_mapping_stats</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_mapping_stats</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_mapping_stats</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>read_mapping_stats</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>skani</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs allocated for the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>skani</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Disk size allocated for the task (in GB)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>skani</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image used for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/skani:0.2.2</td>
<td>Optional</td>
</tr>
<tr>
<td>skani</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Memory allocated for the task (in GB)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>skani</td>
<td><strong>skani_db</strong></td>
<td>File</td>
<td>Skani database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/skani/skani_db_20250606.tar</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>call_porechop</strong></td>
<td>Boolean</td>
<td>True/False to trim adapters with porechop</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>call_raven</strong></td>
<td>Boolean</td>
<td>True/False to call assembly with Raven and use Flye as fallback</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>extract_unclassified</strong></td>
<td>Boolean</td>
<td>True/False that determines if unclassified reads should be extracted and combined with the taxon specific extracted reads</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Expected genome length of taxon of interest</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>host</strong></td>
<td>String</td>
<td>Host taxon/accession to dehost reads, if provided</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>min_allele_freq</strong></td>
<td>Float</td>
<td>Minimum allele frequency required for a variant to populate the consensus sequence</td>
<td>0.6</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>min_depth</strong></td>
<td>Int</td>
<td>Minimum read depth required for a variant to populate the consensus sequence</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>min_map_quality</strong></td>
<td>Int</td>
<td>Minimum mapping quality required for read alignments</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>read_extraction_rank</strong></td>
<td>String</td>
<td>Taxonomic rank to use for read extraction - limits taxons to only those within the specified ranks.</td>
<td>family</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>reference_fasta</strong></td>
<td>File</td>
<td>Reference genome in FASTA format</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>skip_rasusa</strong></td>
<td>Boolean</td>
<td>True/False to skip read subsampling with Rasusa</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaviral_ont</td>
<td><strong>skip_screen</strong></td>
<td>Boolean</td>
<td>True/False to skip read screening check prior to analysis</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<h3 id="all-tasks">All Tasks<a class="headerlink" href="#all-tasks" title="Permanent link">&para;</a></h3>
<div class="tabbed-set tabbed-alternate" data-tabs="3:2"><input checked="checked" id="theiaviral_illumina_pe_2" name="__tabbed_3" type="radio" /><input id="theiaviral_ont_2" name="__tabbed_3" type="radio" /><div class="tabbed-labels"><label for="theiaviral_illumina_pe_2">TheiaViral_Illumina_PE</label><label for="theiaviral_ont_2">TheiaViral_ONT</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<details class="toggle">
<summary>Versioning</summary>
<details class="task">
<summary><code>versioning</code>: Version Capture</summary>
<p>The <code>versioning</code> task captures the workflow version from the GitHub (code repository) version.</p>
<div class="admonition techdetails">
<p class="admonition-title">Version Capture Technical details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/task_versioning.wdl">task_versioning.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Taxonomic Identification</summary>
<details class="task">
<summary><code>ncbi_identify</code></summary>
<p>The <code>ncbi_identify</code> task uses <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/taxonomy/"><code>NCBI Datasets</code></a> to search the NCBI Viral Genome Database and acquire taxonomic metadata from a user's inputted taxonomy and desired taxonomic rank. This task will always return a taxon ID, name, and rank, and it facilitates multiple downstream functions, including <strong>read classification and targeted read extraction</strong>. This task also generates a comprehensive summary file of all successful hits to the input <code>taxon</code>, which includes each taxon's accession number, completeness status, genome length, source, and other relevant metadata. Based on this summary, the task also calculates the average expected genome size for the input <code>taxon</code>.</p>
<details class="dna">
<summary><code>taxon</code> input parameter</summary>
<p>This parameter accepts either a NCBI taxon ID (e.g. <code>11292</code>) or an organism name (e.g. <code>Lyssavirus rabies</code>).</p>
</details>
<details class="dna">
<summary><code>rank</code> a.k.a <code>read_extraction_rank</code> input parameter</summary>
<p>Valid options include: <code>"species"</code>, <code>"genus"</code>, <code>"family"</code>, <code>"order"</code>, <code>"class"</code>, <code>"phylum"</code>, <code>"kingdom"</code>, or <code>"domain"</code>. By default it is set to <code>"family"</code>. This parameter filters metadata to report information only at the taxonomic <code>rank</code> specified by the user, regardless of the taxonomic rank implied by the original input <code>taxon</code>.</p>
</details>
<details class="warning" open="open">
<summary>Important</summary>
<ul>
<li>The <code>rank</code> parameter <strong>must</strong> specify a taxonomic rank that is <mark>equal to or above</mark> the input taxon's taxonomic rank.</li>
</ul>
<p><strong>Examples:</strong></p>
<ul>
<li>If your input <code>taxon</code> is <code>Lyssavirus rabies</code> (species level) with <code>rank</code> set to <code>family</code>, the task will return information for the family of <code>Lyssavirus rabies</code>: taxon ID for Rhabdoviridae (11270), name "Rhabdoviridae", and rank "family".</li>
<li>If your input <code>taxon</code> is <code>Lyssavirus</code> (genus level) with <code>rank</code> set to <code>species</code>, the task will fail because it cannot determine species information from an inputted genus.</li>
</ul>
</details>
<div class="admonition techdetails">
<p class="admonition-title">NCBI Datasets Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_identify_taxon_id.wdl">task_identify_taxon_id.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/datasets">NCBI Datasets on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/">NCBI Datasets Documentation on NCBI</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41597-024-03571-y">Exploring and retrieving sequence and metadata for species across the tree of life with NCBI Datasets</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Read Quality Control, Trimming, Filtering, Identification and Extraction</summary>
<details class="task">
<summary><code>read_QC_trim</code></summary>
<p><code>read_QC_trim</code> is a sub-workflow that removes low-quality reads, low-quality regions of reads, and sequencing adapters to improve data quality. It uses a number of tasks, described below. The differences between the PE and SE versions of the <code>read_QC_trim</code> sub-workflow lie in the default parameters, the use of two or one input read file(s), and the different output files.</p>
<details class="toggle">
<summary><code>HRRT</code>: Human Host Sequence Removal</summary>
<p>All reads of human origin <strong>are removed</strong>, including their mates, by using NCBI's <a href="https://github.com/ncbi/sra-human-scrubber"><strong>human read removal tool (HRRT)</strong></a>. </p>
<p>HRRT is based on the <a href="https://doi.org/10.1186/s13059-021-02490-0">SRA Taxonomy Analysis Tool</a> and employs a k-mer database constructed of k-mers from Eukaryota derived from all human RefSeq records with any k-mers found in non-Eukaryota RefSeq records subtracted from the database.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI-Scrub Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ncbi_scrub.wdl">task_ncbi_scrub.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/sra-human-scrubber">HRRT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2023/02/02/scrubbing-human-sequences-sra-submissions/">HRRT on NCBI</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Read quality trimming</summary>
<p>Either <code>trimmomatic</code> or <code>fastp</code> can be used for read-quality trimming. Trimmomatic is used by default. Both tools trim low-quality regions of reads with a sliding window (with a window size of <code>trim_window_size</code>), cutting once the average quality within the window falls below <code>trim_quality_trim_score</code>. They will both discard the read if it is trimmed below <code>trim_minlen</code>. </p>
<details class="dna">
<summary><code>read_processing</code> input parameter</summary>
<p>This input parameter accepts either <code>trimmomatic</code> or <code>fastp</code> as an input to determine which tool should be used for read quality trimming. This is set to <code>trimmomatic</code> by default.</p>
<p>If the <code>fastp</code> option is selected, see below for table of default parameters.</p>
<details class="toggle">
<summary><code>fastp</code> default read-trimming parameters</summary>
<table>
<thead>
<tr>
<th><strong>Parameter</strong></th>
<th><strong>Explanation</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>-g</td>
<td>enables polyG tail trimming</td>
</tr>
<tr>
<td>-5 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>-3 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>--detect_adapter_for_pe</td>
<td>enables adapter-trimming <strong>only for paired-end reads</strong></td>
</tr>
</tbody>
</table>
<p>Additional arguments can be passed using the <code>fastp_args</code> optional parameter.</p>
</details>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Trimmomatic and fastp Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_trimmomatic.wdl">task_trimmomatic.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_fastp.wdl">task_fastp.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/usadellab/Trimmomatic">Trimmomatic</a><br><a href="https://github.com/OpenGene/fastp">fastp on Github</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a><br><a href="https://github.com/OpenGene/fastp">fastp</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590/">Trimmomatic: a flexible trimmer for Illumina sequence data</a><br><a href="https://academic.oup.com/bioinformatics/article/34/17/i884/5093234?login=false">fastp: an ultra-fast all-in-one FASTQ preprocessor</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Adapter removal</summary>
<p>The <code>BBDuk</code> task removes adapters from sequence reads. To do this:</p>
<ul>
<li><a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/repair-guide/">Repair</a> from the <a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/">BBTools</a> package reorders reads in paired fastq files to ensure the forward and reverse reads of a pair are in the same position in the two fastq files.</li>
<li><a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk</a>  (<em>"Bestus Bioinformaticus" Decontamination Using Kmers</em>) is then used to trim the adapters and filter out all reads that have a 31-mer match to <a href="https://emea.illumina.com/products/by-type/sequencing-kits/cluster-gen-sequencing-reagents/phix-control-v3.html">PhiX</a>, which is commonly added to Illumina sequencing runs to monitor and/or improve overall run quality.</li>
</ul>
<details class="question">
<summary>What are adapters and why do they need to be removed?</summary>
<p>Adapters are manufactured oligonucleotide sequences attached to DNA fragments during the library preparation process. In Illumina sequencing, these adapter sequences are required for attaching reads to flow cells. You can read more about Illumina adapters <a href="https://emea.support.illumina.com/bulletins/2020/06/illumina-adapter-portfolio.html">here</a>. For genome analysis, it's important to remove these sequences since they're not actually from your sample. If you don't remove them, the downstream analysis may be affected.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">BBDuk Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_bbduk.wdl">task_bbduk.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/">BBTools</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Read Quantification</summary>
<p>There are two methods for read quantification to choose from: <a href="https://github.com/rpetit3/fastq-scan"><code>fastq-scan</code></a> (default) or <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"><code>fastqc</code></a>. Both quantify the forward and reverse reads in FASTQ files. For paired-end data, they also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads. <code>fastqc</code> also provides a graphical visualization of the read quality.</p>
<details class="dna">
<summary><code>read_qc</code> input parameter</summary>
<p>This input parameter accepts either <code>"fastq_scan"</code> or <code>"fastqc"</code> as an input to determine which tool should be used for read quantification. This is set to <code>"fastq-scan"</code> by default.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">fastq-scan and FastQC Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastq_scan.wdl">task_fastq_scan.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastqc.wdl&quot;">task_fastqc.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan on Github</a><br><a href="https://github.com/s-andrews/FastQC">fastqc on Github</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/fastq-scan/blob/master/README.md">fastq-scan</a><br><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>host_decontaminate</code>: Host read decontamination</summary>
<p>Host genetic data is frequently incidentally sequenced alongside pathogens, which can negatively affect the quality of downstream analysis. Host Decontaminate attempts to remove host reads by aligning to a reference host genome acquired on-the-fly. The reference host genome can be acquired via <a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy-compatible</a> taxon input or assembly accession. Host Decontaminate maps inputted reads to the host genome using <code>minimap2</code>, reports mapping statistics to this host genome, and outputs the unaligned dehosted reads. </p>
<p>The detailed steps and tasks are as follows:</p>
<details class="toggle">
<summary>Taxonomic Identification</summary>
<p>The <code>ncbi_identify</code> task uses <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/taxonomy/"><code>NCBI Datasets</code></a> to search the NCBI Viral Genome Database and acquire taxonomic metadata from a user's inputted taxonomy and desired taxonomic rank. This task will always return a taxon ID, name, and rank, and it facilitates multiple downstream functions, including <strong>read classification and targeted read extraction</strong>. This task also generates a comprehensive summary file of all successful hits to the input <code>taxon</code>, which includes each taxon's accession number, completeness status, genome length, source, and other relevant metadata. Based on this summary, the task also calculates the average expected genome size for the input <code>taxon</code>.</p>
<details class="dna">
<summary><code>taxon</code> input parameter</summary>
<p>This parameter accepts either a NCBI taxon ID (e.g. <code>11292</code>) or an organism name (e.g. <code>Lyssavirus rabies</code>).</p>
</details>
<details class="dna">
<summary><code>rank</code> a.k.a <code>read_extraction_rank</code> input parameter</summary>
<p>Valid options include: <code>"species"</code>, <code>"genus"</code>, <code>"family"</code>, <code>"order"</code>, <code>"class"</code>, <code>"phylum"</code>, <code>"kingdom"</code>, or <code>"domain"</code>. By default it is set to <code>"family"</code>. This parameter filters metadata to report information only at the taxonomic <code>rank</code> specified by the user, regardless of the taxonomic rank implied by the original input <code>taxon</code>.</p>
</details>
<details class="warning" open="open">
<summary>Important</summary>
<ul>
<li>The <code>rank</code> parameter <strong>must</strong> specify a taxonomic rank that is <mark>equal to or above</mark> the input taxon's taxonomic rank.</li>
</ul>
<p><strong>Examples:</strong></p>
<ul>
<li>If your input <code>taxon</code> is <code>Lyssavirus rabies</code> (species level) with <code>rank</code> set to <code>family</code>, the task will return information for the family of <code>Lyssavirus rabies</code>: taxon ID for Rhabdoviridae (11270), name "Rhabdoviridae", and rank "family".</li>
<li>If your input <code>taxon</code> is <code>Lyssavirus</code> (genus level) with <code>rank</code> set to <code>species</code>, the task will fail because it cannot determine species information from an inputted genus.</li>
</ul>
</details>
<div class="admonition techdetails">
<p class="admonition-title">NCBI Datasets Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_identify_taxon_id.wdl">task_identify_taxon_id.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/datasets">NCBI Datasets on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/">NCBI Datasets Documentation on NCBI</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41597-024-03571-y">Exploring and retrieving sequence and metadata for species across the tree of life with NCBI Datasets</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Download Accession</summary>
<p>The <a href="https://www.ncbi.nlm.nih.gov/datasets/"><code>NCBI Datasets</code></a> task downloads specified assemblies from NCBI using either the <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/virus-genome/">virus</a> or <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/genome/">genome</a> (for all other genome types) package as appropriate.</p>
<p>This task uses the accession ID output from the <code>skani</code> task to download the the most closely related reference genome to the input assembly. The downloaded reference is then used for downstream analysis, including variant calling and consensus generation.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI Datasets Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_import/task_ncbi_datasets.wdl">task_ncbi_datasets.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/datasets">NCBI Datasets on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/">NCBI Datasets Documentation on NCBI</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41597-024-03571-y">Exploring and retrieving sequence and metadata for species across the tree of life with NCBI Datasets</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Map Reads to Host</summary>
<p><code>minimap2</code> is a popular aligner that is used to align reads (or assemblies) to an assembly file. In minimap2, "modes" are a group of preset options.</p>
<p>The mode used in this task is <code>map-ont</code> which is the default mode for long reads and indicates that long reads of ~10% error rates should be aligned to the reference genome. The output file is in SAM format.</p>
<p>For more information regarding modes and the available options for <code>minimap2</code>, please see the <a href="https://lh3.github.io/minimap2/minimap2.html">minimap2 manpage</a></p>
<div class="admonition techdetails">
<p class="admonition-title">minimap2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_minimap2.wdl">task_minimap2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/minimap2">minimap2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://lh3.github.io/minimap2">minimap2</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/18/3094/4994778">Minimap2: pairwise alignment for nucleotide sequences</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Extract Unaligned Reads</summary>
<p>The <code>bam_to_unaligned_fastq</code> task will extract a FASTQ file of reads that failed to align, while removing unpaired reads. </p>
<div class="admonition techdetails">
<p class="admonition-title"><code>parse_mapping</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_handling/task_parse_mapping.wdl">task_parse_mapping.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Host Read Mapping Statistics</summary>
<p>The <code>assembly_metrics</code> task generates mapping statistics from a BAM file. It uses samtools to generate a summary of the mapping statistics, which includes coverage, depth, average base quality, average mapping quality, and other relevant metrics.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>assembly_metrics</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Host Decontaminate Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow File</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_host_decontaminate.wdl">wf_host_decontaminate.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Read Identification</summary>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<p>This task runs on cleaned reads passed from the <code>read_QC_trim</code> subworkflow and outputs a Kraken2 report detailing taxonomic classifications. It also separates classified reads from unclassified ones.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Read Extraction</summary>
<p>The <code>task_krakentools.wdl</code> task extracts reads from the Kraken2 output file. It uses the <a href="https://github.com/jenniferlu717/KrakenTools">KrakenTools</a> package to extract reads classified at any user-specified taxon ID.</p>
<details class="dna">
<summary><code>extract_unclassified</code> input parameter</summary>
<p>This parameter determines whether unclassified reads should also be extracted and combined with the <code>taxon</code>-specific extracted reads. By default, this is set to <code>false</code>, meaning that only reads classified to the specified input <code>taxon</code> will be extracted.</p>
</details>
<details class="warning" open="open">
<summary>Important</summary>
<p>This task will extract reads classified to the input <code>taxon</code> and <strong>all of its descendant taxa</strong>. The <code>rank</code> input parameter controls the extraction of reads classified at the specified <code>rank</code> and all suboridante taxonomic levels. See task <code>ncbi_identify</code> under the <strong>Taxonomic Identification</strong> section for more details on the <code>rank</code> input parameter.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">KrakenTools Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_krakentools.wdl">task_krakentools.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/jenniferlu717/KrakenTools">KrakenTools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/jenniferlu717/KrakenTools/blob/master/README.md">KrakenTools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1126/scitranslmed.aap9489">Metagenome analysis using the Kraken software suite</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="task">
<summary><code>rasusa</code></summary>
<p>The <code>rasusa</code> task performs subsampling on the input raw reads. By default, it subsamples reads to a target depth of 250X, using the estimated genome length either generated by the <code>ncbi_identify</code> task or provided directly by the user. Disabled by default, users can enable it by setting the <code>skip_rasusa</code> variable to <code>false</code>. The target subsampling depth can also be adjusted by modifying the <code>coverage</code> variable.</p>
<details class="dna">
<summary><code>coverage</code> input parameter</summary>
<p>This parameter specifies the target coverage for subsampling. The default value is <code>250</code>, but users can adjust it as needed.</p>
</details>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Rasusa Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/task_rasusa.wdl">task_rasusa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/mbhall88/rasusa">Rasusa on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/mbhall88/rasusa">Rasusa on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03941">Rasusa: Randomly subsample sequencing reads to a specified coverage</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>clean_check_reads</code></summary>
<p>The <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl"><code>screen</code></a> task ensures the quantity of sequence data is sufficient to undertake genomic analysis. It uses <a href="https://github.com/rpetit3/fastq-scan"><code>fastq-scan</code></a> and bash commands for quantification of reads and base pairs, and <a href="https://mash.readthedocs.io/en/latest/index.html">mash</a> sketching to estimate the genome size and its coverage. At each step, the results are assessed relative to pass/fail criteria and thresholds that may be defined by optional user inputs. Samples are run through all threshold checks, regardless of failures, and the workflow will terminate after the <code>screen</code> task if any thresholds are not met:</p>
<ol>
<li>Total number of reads: A sample will fail the read screening task if its total number of reads is less than or equal to <code>min_reads</code>.</li>
<li>The proportion of basepairs reads in the forward and reverse read files: A sample will fail the read screening if fewer than <code>min_proportion</code> basepairs are in either the reads1 or read2 files.</li>
<li>Number of basepairs: A sample will fail the read screening if there are fewer than <code>min_basepairs</code> basepairs</li>
<li>Estimated genome size:  A sample will fail the read screening if the estimated genome size is smaller than <code>min_genome_size</code> or bigger than <code>max_genome_size</code>.</li>
<li>Estimated genome coverage: A sample will fail the read screening if the estimated genome coverage is less than the <code>min_coverage</code>.</li>
</ol>
<p>Read screening is performed only on the cleaned reads. The task may be skipped by setting the <code>skip_screen</code> variable to <code>true</code>. Default values vary between the ONT and PE workflow. The rationale for these default values can be found below:</p>
<details class="toggle">
<summary>Default Thresholds and Rationales</summary>
<table>
<thead>
<tr>
<th>Variable</th>
<th>Description</th>
<th>Default Value</th>
<th>Rationale</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>min_reads</code></td>
<td>A sample will fail the read screening task if its total number of reads is less than or equal to <code>min_reads</code></td>
<td>50</td>
<td>Minimum number of base pairs for 10x coverage of the Hepatitis delta (of the <em>Deltavirus</em> genus) virus divided by 300 (longest Illumina read length)</td>
</tr>
<tr>
<td><code>min_basepairs</code></td>
<td>A sample will fail the read screening if there are fewer than <code>min_basepairs</code> basepairs</td>
<td>15000</td>
<td>Greater than 10x coverage of the Hepatitis delta (of the <em>Deltavirus</em> genus) virus</td>
</tr>
<tr>
<td><code>min_genome_size</code></td>
<td>A sample will fail the read screening if the estimated genome size is smaller than <code>min_genome_size</code></td>
<td>1500</td>
<td>Based on the Hepatitis delta (of the <em>Deltavirus</em> genus) genome- the smallest viral genome as of 2024-04-11 (1,700 bp)</td>
</tr>
<tr>
<td><code>max_genome_size</code></td>
<td>A sample will fail the read screening if the estimated genome size is smaller than <code>max_genome_size</code></td>
<td>2673870</td>
<td>Based on the <em>Pandoravirus salinus</em> genome, the biggest viral genome, (2,673,870 bp) with 2 Mbp added</td>
</tr>
<tr>
<td><code>min_coverage</code></td>
<td>A sample will fail the read screening if the estimated genome coverage is less than the <code>min_coverage</code></td>
<td>10</td>
<td>A bare-minimum coverage for genome characterization. Higher coverage would be required for high-quality phylogenetics.</td>
</tr>
<tr>
<td><code>min_proportion</code></td>
<td>A sample will fail the read screening if fewer than <code>min_proportion</code> basepairs are in either the reads1 or read2 files</td>
<td>40</td>
<td>Greater than 50% reads are in the read1 file; others are in the read2 file. (PE workflow only)</td>
</tr>
</tbody>
</table>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Screen Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L3">task_screen.wdl (PE sub-task)</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L147">task_screen.wdl (SE sub-task)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>De novo</em> Assembly and Reference Selection</summary>
<details class="warning" open="open">
<summary>These tasks are only performed if no reference genome is provided</summary>
<p>In this workflow, <em>de novo</em> assembly is primarily used to facilitate the selection of a closely related reference genome, though high quality <em>de novo</em> assemblies can be used for downstream analysis. If the user provides an input <code>reference_fasta</code>, the following assembly generation, assembly evaluation, and reference selections tasks will be <strong>skipped</strong>:</p>
<ul>
<li><code>spades</code></li>
<li><code>megahit</code></li>
<li><code>checkv_denovo</code></li>
<li><code>quast_denovo</code></li>
<li><code>skani</code></li>
<li><code>ncbi_datasets</code></li>
</ul>
</details>
<details class="task">
<summary><code>spades</code></summary>
<p>The <code>spades</code> task is a wrapper for the <a href="https://github.com/ablab/spades">SPAdes assembler</a>, which is used for <em>de novo</em> assembly of the cleaned reads. It is run with the <code>--metaviral</code> option, which is recommended for viral genomes. MetaviralSPAdes pipeline consists of three independent steps, <code>ViralAssembly</code> for finding putative viral subgraphs in a metagenomic assembly graph and generating contigs in these graphs, <code>ViralVerify</code> for checking whether the resulting contigs have viral origin and <code>ViralComplete</code> for checking whether these contigs represent complete viral genomes. For more details, please see the original publication.</p>
<p>MetaviralSPAdes was selected as the default assembler because it produces the most complete viral genomes within TheiaViral, determined by CheckV quality assessment (see task <code>checkv</code> for technical details).</p>
<details class="dna">
<summary><code>call_metaviralspades</code> input parameter</summary>
<p>This parameter controls whether or not the <code>spades</code> task is called by the workflow. By default, <code>call_metaviralspades</code> is set to <code>true</code> because MetaviralSPAdes is used as the primary assembler. MetaviralSPAdes is generally recommended for most users, but it might not perform optimally on all datasets. If users encounter issues with MetaviralSPAdes, they can set the <code>call_metaviralspades</code> variable to <code>false</code> to bypass the <code>spades</code> task and instead <em>de novo</em> assemble using <a href="https://github.com/voutcn/megahit">MEGAHIT</a> (see task <code>megahit</code> for details). Additionally, if the <code>spades</code> task fails during execution, the workflow will automatically fall back to using MEGAHIT for <em>de novo</em> assembly.</p>
</details>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">MetaviralSPAdes Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_spades.wdl">task_spades.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ablab/spades">SPAdes on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ablab.github.io/spades/index.html">SPAdes Manual</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btaa490">MetaviralSPAdes: assembly of viruses from metagenomic data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>megahit</code></summary>
<p>The <code>megahit</code> task is a wrapper for the <a href="https://github.com/voutcn/megahit">MEGAHIT assembler</a>, which is used for <em>de novo</em> metagenomic assembly of the cleaned reads. MEGAHIT is a fast and memory-efficient <em>de novo</em> assembler that can handle large datasets. This task is optional, turned off by default, and will only be called if MetaviralSPAdes fails. It can be enabled by setting the <code>skip_metaviralspades</code> parameter to <code>true</code>. The <code>megahit</code> task is used as a fallback option if the <code>spades</code> task fails during execution (see task <code>spades</code> for more details).</p>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">MEGAHIT Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_megahit.wdl">task_megahit.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/voutcn/megahit">MEGAHIT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/voutcn/megahit/blob/master/README.md">MEGAHIT</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btv033">MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>skani</code></summary>
<p>The <code>skani</code> task is used to identify and select the most closely related reference genome to the <em>de novo</em> assembly. Skani uses an approximate mapping method without base-level alignment to calculate average nucleotide identity (ANI). It is magnitudes faster than BLAST-based methods and almost as accurate.</p>
<p>By default, the reference genome is selected from a database of approximately 200,000 viral genomes. This database was constructed with the following methodology:</p>
<ol>
<li>
<p>Extracting all <a href="https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/AllNuclMetadata/">complete NCBI viral genomes</a>, excluding RefSeq accessions (redundancy), SARS-CoV-2 accessions, and segmented families (Orthomyxoviridae, Hantaviridae, Arenaviridae, and Phenuiviridae). Some complete gene accessions, and not complete genomes, are included because NCBI <code>datasets</code> completeness parameters are susceptible to metadata errors.</p>
</li>
<li>
<p>Adding complete RefSeq segmented viral assembly accessions, which represent segments as individual contigs within the FASTA</p>
</li>
<li>
<p>Adding one SARS-CoV-2 genome for each <a href="https://github.com/cov-lineages/lineages-website/raw/refs/heads/master/_data/lineage_data.full.json">major pangolin lineage</a></p>
</li>
</ol>
<div class="admonition techdetails">
<p class="admonition-title">Skani Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_skani.wdl">task_skani.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/bluenote-1577/skani">Skani on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/bluenote-1577/skani/blob/main/README.md">Skani Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41592-023-02018-3">Fast and robust metagenomic sequence comparison through sparse chaining with skani</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ncbi_datasets</code></summary>
<p>The <a href="https://www.ncbi.nlm.nih.gov/datasets/"><code>NCBI Datasets</code></a> task downloads specified assemblies from NCBI using either the <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/virus-genome/">virus</a> or <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/genome/">genome</a> (for all other genome types) package as appropriate.</p>
<p>This task uses the accession ID output from the <code>skani</code> task to download the the most closely related reference genome to the input assembly. The downloaded reference is then used for downstream analysis, including variant calling and consensus generation.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI Datasets Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_import/task_ncbi_datasets.wdl">task_ncbi_datasets.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/datasets">NCBI Datasets on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/">NCBI Datasets Documentation on NCBI</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41597-024-03571-y">Exploring and retrieving sequence and metadata for species across the tree of life with NCBI Datasets</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Reference Mapping</summary>
<details class="task">
<summary><code>bwa</code></summary>
<p>The <code>bwa</code> task is a wrapper for the BWA alignment tool. It utilizes the BWA-MEM algorithm to map cleaned reads to the reference genome, either selected by the <code>skani</code> task or provided by the user input <code>reference_fasta</code>. This creates a BAM file which is then sorted using the command <code>samtools sort</code>.</p>
<div class="admonition techdetails">
<p class="admonition-title">BWA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_bwa.wdl">task_bwa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/bwa">https://github.com/lh3/bwa</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bio-bwa.sourceforge.net/">https://bio-bwa.sourceforge.net/</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and accurate short read alignment with Burrows-Wheeler transform</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>read_mapping_stats</code></summary>
<p>The <code>read_mapping_stats</code> task generates mapping statistics from a BAM file. It uses samtools to generate a summary of the mapping statistics, which includes coverage, depth, average base quality, average mapping quality, and other relevant metrics.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>read_mapping_stats</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Variant Calling and Consensus Generation</summary>
<details class="task">
<summary><code>ivar_variants</code></summary>
<p>The <code>ivar_variants</code> task wraps the <a href="https://andersen-lab.github.io/ivar/html/index.html">iVar</a> tool to call variants from the sorted BAM file produced by the <code>bwa</code> task. It uses the <code>ivar variants</code> command to identify and report variants based on the aligned reads. The <code>ivar_variants</code> task will filter all variant calls based on user-defined parameters, including <code>min_map_quality</code>, <code>min_depth</code>, and <code>min_allele_freq</code>. This task will return a VCF file containing the variant calls, along with the total number of variants, and the proportion of intermediate variant calls.</p>
<details class="dna">
<summary><code>min_depth</code> input parameter</summary>
<p>This parameter accepts an integer value to set the minimum read depth for variant calling and subsequent consensus sequence generation. The default value is <code>10</code>.</p>
</details>
<details class="dna">
<summary><code>min_map_quality</code> input parameter</summary>
<p>This parameter accepts an integer value to set the minimum mapping quality for variant calling and subsequent consensus sequence generation. The default value is <code>20</code>.</p>
</details>
<details class="dna">
<summary><code>min_allele_freq</code> input parameter</summary>
<p>This parameter accepts a float value to set the minimum allele frequency for variant calling and subsequent consensus sequence generation. The default value is <code>0.6</code>.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">iVar Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_ivar_variant_call.wdl">task_ivar_variant_call.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://andersen-lab.github.io/ivar/html/">Ivar on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://andersen-lab.github.io/ivar/html/manualpage.html">Ivar Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="http://dx.doi.org/10.1186/s13059-018-1618-7">An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ivar consensus</code></summary>
<p>The <code>consensus</code> task wraps the <a href="https://andersen-lab.github.io/ivar/html/index.html">iVar</a> tool to generate a reference-based consensus assembly from the sorted BAM file produced by the <code>bwa</code> task. It uses the <code>ivar consensus</code> command to call variants and generate a consensus sequence based on those mapped reads. The <code>consensus</code> task will filter all variant calls based on user-defined parameters, including <code>min_map_quality</code>, <code>min_depth</code>, and <code>min_allele_freq</code>. This task will return a consensus sequence in FASTA format and the samtools mpileup output.</p>
<p>This task is functional for segmented viruses by iteratively executing iVar on a contig-by-contig basis and concantenating resulting consensus contigs.</p>
<details class="dna">
<summary><code>min_depth</code> input parameter</summary>
<p>This parameter accepts an integer value to set the minimum read depth for variant calling and subsequent consensus sequence generation. The default value is <code>10</code>.</p>
</details>
<details class="dna">
<summary><code>min_map_quality</code> input parameter</summary>
<p>This parameter accepts an integer value to set the minimum mapping quality for variant calling and subsequent consensus sequence generation. The default value is <code>20</code>.</p>
</details>
<details class="dna">
<summary><code>min_allele_freq</code> input parameter</summary>
<p>This parameter accepts a float value to set the minimum allele frequency for variant calling and subsequent consensus sequence generation. The default value is <code>0.6</code>.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">iVar Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_ivar_consensus.wdl">task_ivar_consensus.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://andersen-lab.github.io/ivar/html/">Ivar on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://andersen-lab.github.io/ivar/html/manualpage.html">Ivar Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="http://dx.doi.org/10.1186/s13059-018-1618-7">An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Assembly Evaluation and Consensus Quality Control</summary>
<details class="task">
<summary><code>quast_denovo</code></summary>
<p>QUAST stands for QUality ASsessment Tool. It evaluates genome/metagenome assemblies by computing various metrics without a reference being necessary. It includes useful metrics such as number of contigs, length of the largest contig and N50. </p>
<div class="admonition techdetails">
<p class="admonition-title">QUAST Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_quast.wdl">task_quast.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ablab/quast">QUAST on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://quast.sourceforge.net/">https://quast.sourceforge.net/</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/29/8/1072/228832">QUAST: quality assessment tool for genome assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>checkv_denovo</code> &amp; <code>checkv_consensus</code></summary>
<p>CheckV is a fully automated command-line pipeline for assessing the quality of viral genomes, including identification of host contamination for integrated proviruses, estimating completeness for genome fragments, and identification of closed genomes.</p>
<p>By default, CheckV reports results on a contig-by-contig basis. The <code>checkv</code> task additionally reports both "weighted_contamination" and "weighted_completeness", which are average percents calculated across the total assembly that are weighted by contig length.</p>
<div class="admonition techdetails">
<p class="admonition-title">CheckV Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_checkv.wdl">task_checkv.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://bitbucket.org/berkeleylab/checkv/src/master/">CheckV on Bitbucket</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bitbucket.org/berkeleylab/checkv/src/master/README.md">CheckV Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41587-020-00774-7">CheckV assesses the quality and completeness of metagenome-assembled viral genomes</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>consensus_qc</code></summary>
<p>The consensus_qc task generates a summary of genomic statistics from a consensus genome. This includes the total number of bases, "N" bases, degenerate bases, and an estimate of the percent coverage to the reference genome.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>consensus_qc</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_consensus_qc.wdl">task_consensus_qc.wdl</a></td>
</tr>
<tr>
<td>Software Source Docker Image</td>
<td><a href="https://github.com/theiagen/theiagen_docker_builds/blob/main/utility/1.1/Dockerfile">Theiagen Docker Builds: utility:1.1</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
</div>
<div class="tabbed-block">
<details class="toggle">
<summary>Versioning</summary>
<details class="task">
<summary><code>versioning</code>: Version Capture</summary>
<p>The <code>versioning</code> task captures the workflow version from the GitHub (code repository) version.</p>
<div class="admonition techdetails">
<p class="admonition-title">Version Capture Technical details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/task_versioning.wdl">task_versioning.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Taxonomic Identification</summary>
<details class="task">
<summary><code>ncbi_identify</code></summary>
<p>The <code>ncbi_identify</code> task uses <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/taxonomy/"><code>NCBI Datasets</code></a> to search the NCBI Viral Genome Database and acquire taxonomic metadata from a user's inputted taxonomy and desired taxonomic rank. This task will always return a taxon ID, name, and rank, and it facilitates multiple downstream functions, including <strong>read classification and targeted read extraction</strong>. This task also generates a comprehensive summary file of all successful hits to the input <code>taxon</code>, which includes each taxon's accession number, completeness status, genome length, source, and other relevant metadata. Based on this summary, the task also calculates the average expected genome size for the input <code>taxon</code>.</p>
<details class="dna">
<summary><code>taxon</code> input parameter</summary>
<p>This parameter accepts either a NCBI taxon ID (e.g. <code>11292</code>) or an organism name (e.g. <code>Lyssavirus rabies</code>).</p>
</details>
<details class="dna">
<summary><code>rank</code> a.k.a <code>read_extraction_rank</code> input parameter</summary>
<p>Valid options include: <code>"species"</code>, <code>"genus"</code>, <code>"family"</code>, <code>"order"</code>, <code>"class"</code>, <code>"phylum"</code>, <code>"kingdom"</code>, or <code>"domain"</code>. By default it is set to <code>"family"</code>. This parameter filters metadata to report information only at the taxonomic <code>rank</code> specified by the user, regardless of the taxonomic rank implied by the original input <code>taxon</code>.</p>
</details>
<details class="warning" open="open">
<summary>Important</summary>
<ul>
<li>The <code>rank</code> parameter <strong>must</strong> specify a taxonomic rank that is <mark>equal to or above</mark> the input taxon's taxonomic rank.</li>
</ul>
<p><strong>Examples:</strong></p>
<ul>
<li>If your input <code>taxon</code> is <code>Lyssavirus rabies</code> (species level) with <code>rank</code> set to <code>family</code>, the task will return information for the family of <code>Lyssavirus rabies</code>: taxon ID for Rhabdoviridae (11270), name "Rhabdoviridae", and rank "family".</li>
<li>If your input <code>taxon</code> is <code>Lyssavirus</code> (genus level) with <code>rank</code> set to <code>species</code>, the task will fail because it cannot determine species information from an inputted genus.</li>
</ul>
</details>
<div class="admonition techdetails">
<p class="admonition-title">NCBI Datasets Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_identify_taxon_id.wdl">task_identify_taxon_id.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/datasets">NCBI Datasets on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/">NCBI Datasets Documentation on NCBI</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41597-024-03571-y">Exploring and retrieving sequence and metadata for species across the tree of life with NCBI Datasets</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Read Quality Control, Trimming, and Filtering</summary>
<details class="task">
<summary><code>nanoplot_raw</code> &amp; <code>nanoplot_clean</code></summary>
<p>Nanoplot is used for the determination of mean quality scores, read lengths, and number of reads. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nanoplot Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_nanoplot.wdl">task_nanoplot.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/wdecoster/NanoPlot">NanoPlot</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/wdecoster/NanoPlot/blob/master/README.md">NanoPlot Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/39/5/btad311/7160911">NanoPack2: population-scale evaluation of long-read sequencing data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>porechop</code></summary>
<p>Porechop is a tool for finding and removing adapters from ONT data. Adapters on the ends of reads are trimmed, and when a read has an adapter in the middle, the read is split into two.</p>
<p>The <code>porechop</code> task is optional and is turned off by default. It can be enabled by setting the <code>call_porechop</code> parameter to <code>true</code>.</p>
<div class="admonition techdetails">
<p class="admonition-title">Porechop Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_porechop.wdl">task_porechop.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rrwick/Porechop">Porechop on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rrwick/Porechop#porechop">https://github.com/rrwick/Porechop#porechop</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>nanoq</code></summary>
<p>Reads are filtered by length and quality using <code>nanoq</code>. By default, sequences with less than 500 basepairs and quality score lower than 10 are filtered out to improve assembly accuracy.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nanoq Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_nanoq.wdl">task_nanoq.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/esteinig/nanoq">Nanoq</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/esteinig/nanoq/blob/master/README.md">Nanoq Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.02991">Nanoq: ultra-fast quality control for nanopore reads</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ncbi_scrub_se</code></summary>
<p>All reads of human origin <strong>are removed</strong>, including their mates, by using NCBI's <a href="https://github.com/ncbi/sra-human-scrubber"><strong>human read removal tool (HRRT)</strong></a>. </p>
<p>HRRT is based on the <a href="https://doi.org/10.1186/s13059-021-02490-0">SRA Taxonomy Analysis Tool</a> and employs a k-mer database constructed of k-mers from Eukaryota derived from all human RefSeq records with any k-mers found in non-Eukaryota RefSeq records subtracted from the database.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI-Scrub Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ncbi_scrub.wdl">task_ncbi_scrub.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/sra-human-scrubber">HRRT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2023/02/02/scrubbing-human-sequences-sra-submissions/">HRRT on NCBI</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>host_decontaminate</code></summary>
<p>Host genetic data is frequently incidentally sequenced alongside pathogens, which can negatively affect the quality of downstream analysis. Host Decontaminate attempts to remove host reads by aligning to a reference host genome acquired on-the-fly. The reference host genome can be acquired via <a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy-compatible</a> taxon input or assembly accession. Host Decontaminate maps inputted reads to the host genome using <code>minimap2</code>, reports mapping statistics to this host genome, and outputs the unaligned dehosted reads. </p>
<p>The detailed steps and tasks are as follows:</p>
<details class="toggle">
<summary>Taxonomic Identification</summary>
<p>The <code>ncbi_identify</code> task uses <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/taxonomy/"><code>NCBI Datasets</code></a> to search the NCBI Viral Genome Database and acquire taxonomic metadata from a user's inputted taxonomy and desired taxonomic rank. This task will always return a taxon ID, name, and rank, and it facilitates multiple downstream functions, including <strong>read classification and targeted read extraction</strong>. This task also generates a comprehensive summary file of all successful hits to the input <code>taxon</code>, which includes each taxon's accession number, completeness status, genome length, source, and other relevant metadata. Based on this summary, the task also calculates the average expected genome size for the input <code>taxon</code>.</p>
<details class="dna">
<summary><code>taxon</code> input parameter</summary>
<p>This parameter accepts either a NCBI taxon ID (e.g. <code>11292</code>) or an organism name (e.g. <code>Lyssavirus rabies</code>).</p>
</details>
<details class="dna">
<summary><code>rank</code> a.k.a <code>read_extraction_rank</code> input parameter</summary>
<p>Valid options include: <code>"species"</code>, <code>"genus"</code>, <code>"family"</code>, <code>"order"</code>, <code>"class"</code>, <code>"phylum"</code>, <code>"kingdom"</code>, or <code>"domain"</code>. By default it is set to <code>"family"</code>. This parameter filters metadata to report information only at the taxonomic <code>rank</code> specified by the user, regardless of the taxonomic rank implied by the original input <code>taxon</code>.</p>
</details>
<details class="warning" open="open">
<summary>Important</summary>
<ul>
<li>The <code>rank</code> parameter <strong>must</strong> specify a taxonomic rank that is <mark>equal to or above</mark> the input taxon's taxonomic rank.</li>
</ul>
<p><strong>Examples:</strong></p>
<ul>
<li>If your input <code>taxon</code> is <code>Lyssavirus rabies</code> (species level) with <code>rank</code> set to <code>family</code>, the task will return information for the family of <code>Lyssavirus rabies</code>: taxon ID for Rhabdoviridae (11270), name "Rhabdoviridae", and rank "family".</li>
<li>If your input <code>taxon</code> is <code>Lyssavirus</code> (genus level) with <code>rank</code> set to <code>species</code>, the task will fail because it cannot determine species information from an inputted genus.</li>
</ul>
</details>
<div class="admonition techdetails">
<p class="admonition-title">NCBI Datasets Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_identify_taxon_id.wdl">task_identify_taxon_id.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/datasets">NCBI Datasets on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/">NCBI Datasets Documentation on NCBI</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41597-024-03571-y">Exploring and retrieving sequence and metadata for species across the tree of life with NCBI Datasets</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Download Accession</summary>
<p>The <a href="https://www.ncbi.nlm.nih.gov/datasets/"><code>NCBI Datasets</code></a> task downloads specified assemblies from NCBI using either the <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/virus-genome/">virus</a> or <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/genome/">genome</a> (for all other genome types) package as appropriate.</p>
<p>This task uses the accession ID output from the <code>skani</code> task to download the the most closely related reference genome to the input assembly. The downloaded reference is then used for downstream analysis, including variant calling and consensus generation.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI Datasets Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_import/task_ncbi_datasets.wdl">task_ncbi_datasets.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/datasets">NCBI Datasets on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/">NCBI Datasets Documentation on NCBI</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41597-024-03571-y">Exploring and retrieving sequence and metadata for species across the tree of life with NCBI Datasets</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Map Reads to Host</summary>
<p><code>minimap2</code> is a popular aligner that is used to align reads (or assemblies) to an assembly file. In minimap2, "modes" are a group of preset options.</p>
<p>The mode used in this task is <code>map-ont</code> which is the default mode for long reads and indicates that long reads of ~10% error rates should be aligned to the reference genome. The output file is in SAM format.</p>
<p>For more information regarding modes and the available options for <code>minimap2</code>, please see the <a href="https://lh3.github.io/minimap2/minimap2.html">minimap2 manpage</a></p>
<div class="admonition techdetails">
<p class="admonition-title">minimap2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_minimap2.wdl">task_minimap2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/minimap2">minimap2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://lh3.github.io/minimap2">minimap2</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/18/3094/4994778">Minimap2: pairwise alignment for nucleotide sequences</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Extract Unaligned Reads</summary>
<p>The <code>bam_to_unaligned_fastq</code> task will extract a FASTQ file of reads that failed to align, while removing unpaired reads. </p>
<div class="admonition techdetails">
<p class="admonition-title"><code>parse_mapping</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_handling/task_parse_mapping.wdl">task_parse_mapping.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary>Host Read Mapping Statistics</summary>
<p>The <code>assembly_metrics</code> task generates mapping statistics from a BAM file. It uses samtools to generate a summary of the mapping statistics, which includes coverage, depth, average base quality, average mapping quality, and other relevant metrics.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>assembly_metrics</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Host Decontaminate Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow File</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_host_decontaminate.wdl">wf_host_decontaminate.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>rasusa</code></summary>
<p>The <code>rasusa</code> task performs subsampling on the input raw reads. By default, it subsamples reads to a target depth of 250X, using the estimated genome length either generated by the <code>ncbi_identify</code> task or provided directly by the user. Disabled by default, users can enable it by setting the <code>skip_rasusa</code> variable to <code>false</code>. The target subsampling depth can also be adjusted by modifying the <code>coverage</code> variable.</p>
<details class="dna">
<summary><code>coverage</code> input parameter</summary>
<p>This parameter specifies the target coverage for subsampling. The default value is <code>250</code>, but users can adjust it as needed.</p>
</details>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Rasusa Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/task_rasusa.wdl">task_rasusa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/mbhall88/rasusa">Rasusa on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/mbhall88/rasusa">Rasusa on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03941">Rasusa: Randomly subsample sequencing reads to a specified coverage</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>clean_check_reads</code></summary>
<p>The <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl"><code>screen</code></a> task ensures the quantity of sequence data is sufficient to undertake genomic analysis. It uses <a href="https://github.com/rpetit3/fastq-scan"><code>fastq-scan</code></a> and bash commands for quantification of reads and base pairs, and <a href="https://mash.readthedocs.io/en/latest/index.html">mash</a> sketching to estimate the genome size and its coverage. At each step, the results are assessed relative to pass/fail criteria and thresholds that may be defined by optional user inputs. Samples are run through all threshold checks, regardless of failures, and the workflow will terminate after the <code>screen</code> task if any thresholds are not met:</p>
<ol>
<li>Total number of reads: A sample will fail the read screening task if its total number of reads is less than or equal to <code>min_reads</code>.</li>
<li>The proportion of basepairs reads in the forward and reverse read files: A sample will fail the read screening if fewer than <code>min_proportion</code> basepairs are in either the reads1 or read2 files.</li>
<li>Number of basepairs: A sample will fail the read screening if there are fewer than <code>min_basepairs</code> basepairs</li>
<li>Estimated genome size:  A sample will fail the read screening if the estimated genome size is smaller than <code>min_genome_size</code> or bigger than <code>max_genome_size</code>.</li>
<li>Estimated genome coverage: A sample will fail the read screening if the estimated genome coverage is less than the <code>min_coverage</code>.</li>
</ol>
<p>Read screening is performed only on the cleaned reads. The task may be skipped by setting the <code>skip_screen</code> variable to <code>true</code>. Default values vary between the ONT and PE workflow. The rationale for these default values can be found below:</p>
<details class="toggle">
<summary>Default Thresholds and Rationales</summary>
<table>
<thead>
<tr>
<th>Variable</th>
<th>Description</th>
<th>Default Value</th>
<th>Rationale</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>min_reads</code></td>
<td>A sample will fail the read screening task if its total number of reads is less than or equal to <code>min_reads</code></td>
<td>50</td>
<td>Minimum number of base pairs for 10x coverage of the Hepatitis delta (of the <em>Deltavirus</em> genus) virus divided by 300 (longest Illumina read length)</td>
</tr>
<tr>
<td><code>min_basepairs</code></td>
<td>A sample will fail the read screening if there are fewer than <code>min_basepairs</code> basepairs</td>
<td>15000</td>
<td>Greater than 10x coverage of the Hepatitis delta (of the <em>Deltavirus</em> genus) virus</td>
</tr>
<tr>
<td><code>min_genome_size</code></td>
<td>A sample will fail the read screening if the estimated genome size is smaller than <code>min_genome_size</code></td>
<td>1500</td>
<td>Based on the Hepatitis delta (of the <em>Deltavirus</em> genus) genome- the smallest viral genome as of 2024-04-11 (1,700 bp)</td>
</tr>
<tr>
<td><code>max_genome_size</code></td>
<td>A sample will fail the read screening if the estimated genome size is smaller than <code>max_genome_size</code></td>
<td>2673870</td>
<td>Based on the <em>Pandoravirus salinus</em> genome, the biggest viral genome, (2,673,870 bp) with 2 Mbp added</td>
</tr>
<tr>
<td><code>min_coverage</code></td>
<td>A sample will fail the read screening if the estimated genome coverage is less than the <code>min_coverage</code></td>
<td>10</td>
<td>A bare-minimum coverage for genome characterization. Higher coverage would be required for high-quality phylogenetics.</td>
</tr>
<tr>
<td><code>min_proportion</code></td>
<td>A sample will fail the read screening if fewer than <code>min_proportion</code> basepairs are in either the reads1 or read2 files</td>
<td>40</td>
<td>Greater than 50% reads are in the read1 file; others are in the read2 file. (PE workflow only)</td>
</tr>
</tbody>
</table>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Screen Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L3">task_screen.wdl (PE sub-task)</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L147">task_screen.wdl (SE sub-task)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Read Classification and Extraction</summary>
<details class="task">
<summary><code>metabuli</code></summary>
<p>The <code>metabuli</code> task is used to classify and extract reads against a reference database. Metabuli uses a novel k-mer structure, called metamer, to analyze both amino acid (AA) and DNA sequences. It leverages AA conservation for sensitive homology detection and DNA mutations for specific differentiation between closely related taxa.</p>
<details class="dna">
<summary><code>cpu</code> / <code>memory</code> input parameters</summary>
<p>Increasing the memory and cpus allocated to Metabuli can substantially increase throughput.</p>
</details>
<details class="dna">
<summary><code>extract_unclassified</code> input parameter</summary>
<p>This parameter determines whether unclassified reads should also be extracted and combined with the <code>taxon</code>-specific extracted reads. By default, this is set to <code>false</code>, meaning that only reads classified to the specified input <code>taxon</code> will be extracted.</p>
</details>
<details class="warning" open="open">
<summary>Descendant taxa reads are extracted</summary>
<p>This task will extract reads classified to the input <code>taxon</code> and <strong>all of its descendant taxa</strong>. The <code>rank</code> input parameter controls the extraction of reads classified at the specified <code>rank</code> and all subordiante taxonomic levels. See task <code>ncbi_identify</code> under the <strong>Taxonomic Identification</strong> section above for more details on the <code>rank</code> input parameter.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Metabuli Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_metabuli.wdl">task_metabuli.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/steineggerlab/Metabuli">Metabuli on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/steineggerlab/Metabuli/blob/master/README.md">Metabuli Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41592-024-02273-y">Metabuli: sensitive and specific metagenomic classification via joint analysis of amino acid and DNA</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>De novo</em> Assembly and Reference Selection</summary>
<details class="warning" open="open">
<summary>These tasks are only performed if no reference genome is provided</summary>
<p>In this workflow, <em>de novo</em> assembly is used solely to facilitate the selection of a closely related reference genome. If the user provides an input <code>reference_fasta</code>, the following assembly generation, assembly evaluation, and reference selections tasks will be <strong>skipped</strong>:</p>
<ul>
<li><code>raven</code></li>
<li><code>flye</code></li>
<li><code>checkv_denovo</code></li>
<li><code>quast_denovo</code></li>
<li><code>skani</code></li>
<li><code>ncbi_datasets</code></li>
</ul>
</details>
<details class="task">
<summary><code>raven</code></summary>
<p>The <code>raven</code> task is used to create a <em>de novo</em> assembly from cleaned reads. Raven is an overlap-layout-consensus based assembler that accelerates the overlap step, constructs an assembly graph from reads pre-processed with pile-o-grams, applies a novel and robust graph simplification method based on graph drawings, and polishes unambiguous graph paths using Racon. </p>
<p>Based on internal benchmarking against Flye and results reported by <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC11092197/">Cook et al. (2024)</a>, Raven is faster, produces more contiguous assemblies, and yields more complete genomes within TheiaViral according to CheckV quality assessment (see task <code>checkv</code> for technical details).</p>
<details class="dna">
<summary><code>call_raven</code> input parameter</summary>
<p>This parameter controls whether or not the <code>raven</code> task is called by the workflow. By default, <code>call_raven</code> is set to <code>true</code> because Raven is used as the primary assembler. Raven is generally recommended for most users, but it might not perform optimally on all datasets. If users encounter issues with Raven, they can set the <code>call_raven</code> variable to <code>false</code> to bypass the <code>raven</code> task and instead <em>de novo</em> assemble using Flye (see task <code>flye</code> for details). Additionally, if the Raven task fails during execution, the workflow will automatically fall back to using Flye for <em>de novo</em> assembly.</p>
</details>
<details class="warning" open="open">
<summary>Error traceback</summary>
<p>Raven may fail with cryptic "segmentation fault" (segfault) errors or by failing to output an output file. It is difficult to traceback the source of these issues, though increasing the <code>memory</code> parameter may resolve some errors.</p>
</details>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Raven Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_raven.wdl">task_raven.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lbcb-sci/raven">Raven on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/lbcb-sci/raven/blob/master/README.md">Raven Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s43588-021-00073-4">Time- and memory-efficient genome assembly with Raven</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>flye</code></summary>
<p>Flye is a <em>de novo</em> assembler for long read data using repeat graphs. Compared to de Bruijn graphs, which require exact k-mer matches, repeat graphs can use approximate matches which better tolerates the error rate of ONT data.</p>
<p>It can be enabled by setting the <code>call_raven</code> parameter to <code>false</code>. The <code>flye</code> task is used as a fallback option if the <code>raven</code> task fails during execution (see task <code>raven</code> for more details).</p>
<details class="dna">
<summary><code>read_type</code> input parameter</summary>
<p>This input parameter specifies the type of sequencing reads being used for assembly. This parameter significantly impacts the assembly process and should match the characteristics of your input data. Below are the available options:</p>
<table>
<thead>
<tr>
<th><strong>Parameter</strong></th>
<th><strong>Explanation</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--nano-hq</code> (default)</td>
<td>Optimized for ONT high-quality reads, such as Guppy5+ SUP or Q20 (&lt;5% error). Recommended for ONT reads processed with Guppy5 or newer</td>
</tr>
<tr>
<td><code>--nano-raw</code></td>
<td>For ONT regular reads, pre-Guppy5 (&lt;20% error)</td>
</tr>
<tr>
<td><code>--nano-corr</code></td>
<td>ONT reads corrected with other methods (&lt;3% error)</td>
</tr>
<tr>
<td><code>--pacbio-raw</code></td>
<td>PacBio regular CLR reads (&lt;20% error)</td>
</tr>
<tr>
<td><code>--pacbio-corr</code></td>
<td>PacBio reads corrected with other methods (&lt;3% error)</td>
</tr>
<tr>
<td><code>--pacbio-hifi</code></td>
<td>PacBio HiFi reads (&lt;1% error)</td>
</tr>
</tbody>
</table>
<p>Refer to the Flye documentation for detailed guidance on selecting the appropriate <code>read_type</code> based on your sequencing data and additional optional paramaters.</p>
</details>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Flye Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_flye.wdl">task_flye.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/fenderglass/Flye">Flye on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/fenderglass/Flye/blob/flye/docs/USAGE.md">Flye Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.nature.com/articles/s41587-019-0072-8">Assembly of long, error-prone reads using repeat graphs</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>skani</code></summary>
<p>The <code>skani</code> task is used to identify and select the most closely related reference genome to the <em>de novo</em> assembly. Skani uses an approximate mapping method without base-level alignment to calculate average nucleotide identity (ANI). It is magnitudes faster than BLAST-based methods and almost as accurate.</p>
<p>By default, the reference genome is selected from a database of approximately 200,000 viral genomes. This database was constructed with the following methodology:</p>
<ol>
<li>
<p>Extracting all <a href="https://ftp.ncbi.nlm.nih.gov/genomes/Viruses/AllNuclMetadata/">complete NCBI viral genomes</a>, excluding RefSeq accessions (redundancy), SARS-CoV-2 accessions, and segmented families (Orthomyxoviridae, Hantaviridae, Arenaviridae, and Phenuiviridae). Some complete gene accessions, and not complete genomes, are included because NCBI <code>datasets</code> completeness parameters are susceptible to metadata errors.</p>
</li>
<li>
<p>Adding complete RefSeq segmented viral assembly accessions, which represent segments as individual contigs within the FASTA</p>
</li>
<li>
<p>Adding one SARS-CoV-2 genome for each <a href="https://github.com/cov-lineages/lineages-website/raw/refs/heads/master/_data/lineage_data.full.json">major pangolin lineage</a></p>
</li>
</ol>
<div class="admonition techdetails">
<p class="admonition-title">Skani Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_skani.wdl">task_skani.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/bluenote-1577/skani">Skani on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/bluenote-1577/skani/blob/main/README.md">Skani Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41592-023-02018-3">Fast and robust metagenomic sequence comparison through sparse chaining with skani</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ncbi_datasets</code></summary>
<p>The <a href="https://www.ncbi.nlm.nih.gov/datasets/"><code>NCBI Datasets</code></a> task downloads specified assemblies from NCBI using either the <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/virus-genome/">virus</a> or <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/data-packages/genome/">genome</a> (for all other genome types) package as appropriate.</p>
<p>This task uses the accession ID output from the <code>skani</code> task to download the the most closely related reference genome to the input assembly. The downloaded reference is then used for downstream analysis, including variant calling and consensus generation.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI Datasets Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_import/task_ncbi_datasets.wdl">task_ncbi_datasets.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/datasets">NCBI Datasets on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/">NCBI Datasets Documentation on NCBI</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41597-024-03571-y">Exploring and retrieving sequence and metadata for species across the tree of life with NCBI Datasets</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Reference Mapping</summary>
<details class="task">
<summary><code>minimap2</code></summary>
<p><code>minimap2</code> is a popular aligner that is used to align reads (or assemblies) to an assembly file. In minimap2, "modes" are a group of preset options.</p>
<p>The mode used in this task is <code>map-ont</code> with additional long-read-specific parameters (the <code>-L --cs --MD</code> flags) to align ONT reads to the reference genome. These specialized parameters are essential for proper handling of long read error profiles, generation of detailed alignment information, and improved mapping accuracy for long reads.</p>
<p><code>map-ont</code> is the default mode for long reads and it indicates that long reads of ~10% error rates should be aligned to the reference genome. The output file is in SAM format.</p>
<p>For more information regarding modes and the available options for <code>minimap2</code>, please see the <a href="https://lh3.github.io/minimap2/minimap2.html">minimap2 manpage</a></p>
<div class="admonition techdetails">
<p class="admonition-title">minimap2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_minimap2.wdl">task_minimap2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/minimap2">minimap2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://lh3.github.io/minimap2">minimap2</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/18/3094/4994778">Minimap2: pairwise alignment for nucleotide sequences</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>parse_mapping</code></summary>
<p>The <code>sam_to_sorted_bam</code> sub-task converts the output SAM file from the <code>minimap2</code> task and converts it to a BAM file. It then sorts the BAM file by coordinate, and creates a BAM index file.</p>
<details class="dna">
<summary><code>min_map_quality</code> input parameter</summary>
<p>This parameter accepts an integer value to set the minimum mapping quality for variant calling and subsequent consensus sequence generation. The default value is <code>20</code>.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title"><code>parse_mapping</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_handling/task_parse_mapping.wdl">task_parse_mapping.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>read_mapping_stats</code></summary>
<p>The <code>read_mapping_stats</code> task generates mapping statistics from a BAM file. It uses samtools to generate a summary of the mapping statistics, which includes coverage, depth, average base quality, average mapping quality, and other relevant metrics.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>read_mapping_stats</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>fasta_utilities</code></summary>
<p>The <code>fasta_utilities</code> task utilizes samtools to index a reference fasta file.
This reference is selected by the <code>skani</code> task or provided by the user input <code>reference_fasta</code>. This indexed reference genome is used for downstream variant calling and consensus generation tasks.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>fasta_utilities</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_handling/task_fasta_utilities.wdl">task_fasta_utilities.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Variant Calling and Consensus Generation</summary>
<details class="task">
<summary><code>clair3</code></summary>
<p><code>Clair3</code> performs deep learning-based variant detection using a multi-stage approach. The process begins with pileup-based calling for initial variant identification, followed by full-alignment analysis for comprehensive variant detection. Results are merged into a final high-confidence call set.</p>
<p>The variant calling pipeline employs specialized neural networks trained on ONT data to accurately identify:
- Single nucleotide variants (SNVs)
- Small insertions and deletions (indels)
- Structural variants</p>
<details class="dna">
<summary><code>clair3_model</code> input parameter</summary>
<p>This parameter specifies the clair3 model to use for variant calling. The default is set to <code>"r1041_e82_400bps_sup_v500"</code>, but users may select from other available models that <code>clair3</code> was trained on, which may yield better results depending on the basecaller and data type. The following models are available:</p>
<ul>
<li><code>"ont"</code></li>
<li><code>"ont_guppy2"</code></li>
<li><code>"ont_guppy5"</code></li>
<li><code>"r941_prom_sup_g5014"</code></li>
<li><code>"r941_prom_hac_g360+g422"</code></li>
<li><code>"r941_prom_hac_g238"</code></li>
<li><code>"r1041_e82_400bps_sup_v500"</code></li>
<li><code>"r1041_e82_400bps_hac_v500"</code></li>
<li><code>"r1041_e82_400bps_sup_v410"</code></li>
<li><code>"r1041_e82_400bps_hac_v410"</code></li>
</ul>
</details>
<details class="warning" open="open">
<summary>Default Parameters and Filtering</summary>
<p>In this workflow, <code>clair3</code> is run with nearly all default parameters. Note that the VCF file produced by the <code>clair3</code> task is <strong>unfiltered</strong> and does not represent the final set of variants that will be included in the final consensus genome. A filtered vcf file is generated by the <code>bcftools_consensus</code> task. The filtering parameters are as follows:</p>
<ul>
<li>The <code>min_map_quality</code> parameter is applied before calling variants.</li>
<li>The <code>min_depth</code> and <code>min_allele_freq</code> parameters are applied after variant calling during consensus genome construction.</li>
</ul>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Clair3 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/variant_calling/task_clair3.wdl">task_clair3.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/HKU-BAL/Clair3">Clair3 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/HKU-BAL/Clair3?tab=readme-ov-file#usage">Clair3 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1101/2021.12.29.474431">Symphonizing pileup and full-alignment for deep learning-based long-read variant calling</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>parse_mapping</code></summary>
<p>The <code>mask_low_coverage</code> sub-task is used to mask low coverage regions in the <code>reference_fasta</code> file to improve the accuracy of the final consensus genome. Coverage thresholds are defined by the <code>min_depth</code> parameter, which specifies the minimum read depth required for a base to be retained. Bases falling below this threshold are replaced with "N"s to clearly mark low confidence regions. The masked reference is then combined with variants from the <code>clair3</code> task to produce the final consensus genome.</p>
<details class="dna">
<summary><code>min_depth</code> input parameter</summary>
<p>This parameter accepts an integer value to set the minimum read depth for variant calling and subsequent consensus sequence generation. The default value is <code>10</code>.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title"><code>parse_mapping</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_handling/task_parse_mapping.wdl">task_parse_mapping.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>bcftools_consensus</code></summary>
<p>The <code>bcftools_consensus</code> task generates a consensus genome assembly by applying variants from the <code>clair3</code> task to a masked reference genome. It uses bcftools to filter variants based on the <code>min_depth</code> and <code>min_allele_freq</code> input parameter, left aligns and normalizes indels, indexes the VCF file, and generates a consensus genome in FASTA format. Reference bases are substituted with filtered variants where applicable, preserved in regions without variant calls, and replaced with "N"s in areas masked by the <code>mask_low_coverage</code> task.</p>
<details class="dna">
<summary><code>min_depth</code> input parameter</summary>
<p>This parameter accepts an integer value to set the minimum read depth for variant calling and subsequent consensus sequence generation. The default value is <code>10</code>.</p>
</details>
<details class="dna">
<summary><code>min_allele_freq</code> input parameter</summary>
<p>This parameter accepts a float value to set the minimum allele frequency for variant calling and subsequent consensus sequence generation. The default value is <code>0.6</code>.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title"><code>bcftools_consensus</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_bcftools_consensus.wdl">task_bcftools_consensus.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/bcftools">bcftools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://samtools.github.io/bcftools/bcftools.html">bcftools Manual Page</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Assembly Evaluation and Consensus Quality Control</summary>
<details class="task">
<summary><code>quast_denovo</code></summary>
<p>QUAST stands for QUality ASsessment Tool. It evaluates genome/metagenome assemblies by computing various metrics without a reference being necessary. It includes useful metrics such as number of contigs, length of the largest contig and N50. </p>
<div class="admonition techdetails">
<p class="admonition-title">QUAST Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_quast.wdl">task_quast.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ablab/quast">QUAST on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://quast.sourceforge.net/">https://quast.sourceforge.net/</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/29/8/1072/228832">QUAST: quality assessment tool for genome assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>checkv_denovo</code> &amp; <code>checkv_consensus</code></summary>
<p>CheckV is a fully automated command-line pipeline for assessing the quality of viral genomes, including identification of host contamination for integrated proviruses, estimating completeness for genome fragments, and identification of closed genomes.</p>
<p>By default, CheckV reports results on a contig-by-contig basis. The <code>checkv</code> task additionally reports both "weighted_contamination" and "weighted_completeness", which are average percents calculated across the total assembly that are weighted by contig length.</p>
<div class="admonition techdetails">
<p class="admonition-title">CheckV Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_checkv.wdl">task_checkv.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://bitbucket.org/berkeleylab/checkv/src/master/">CheckV on Bitbucket</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bitbucket.org/berkeleylab/checkv/src/master/README.md">CheckV Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41587-020-00774-7">CheckV assesses the quality and completeness of metagenome-assembled viral genomes</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>consensus_qc</code></summary>
<p>The consensus_qc task generates a summary of genomic statistics from a consensus genome. This includes the total number of bases, "N" bases, degenerate bases, and an estimate of the percent coverage to the reference genome.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>consensus_qc</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_consensus_qc.wdl">task_consensus_qc.wdl</a></td>
</tr>
<tr>
<td>Software Source Docker Image</td>
<td><a href="https://github.com/theiagen/theiagen_docker_builds/blob/main/utility/1.1/Dockerfile">Theiagen Docker Builds: utility:1.1</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
</div>
</div>
</div>
<h4 id="taxa-specific-tasks">Taxa-Specific Tasks<a class="headerlink" href="#taxa-specific-tasks" title="Permanent link">&para;</a></h4>
<p>The TheiaViral workflows automatically activate taxa-specific sub-workflows after the identification of relevant taxa using the taxon ID of the reference genome.</p>
<details class="toggle">
<summary><em>Lyssavirus rabies</em></summary>
<details class="task">
<summary><code>nextclade</code></summary>
<p><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/">"Nextclade is an open-source project for viral genome alignment, mutation calling, clade assignment, quality checks and phylogenetic placement."</a></p>
<p>Theiagen has implemented a full genome-based <a href="https://github.com/theiagen/rabies">Nextclade dataset</a> for <em>L. rabies</em> with subclade classification resolution.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nextclade Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_nextclade.wdl#L63">task_nextclade.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/nextclade">https://github.com/nextstrain/nextclade</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/">Nextclade</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03773">Nextclade: clade assignment, mutation calling and quality control for viral genomes.</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<h3 id="outputs">Outputs<a class="headerlink" href="#outputs" title="Permanent link">&para;</a></h3>
<div class="tabbed-set tabbed-alternate" data-tabs="4:2"><input checked="checked" id="theiaviral_illumina_pe_3" name="__tabbed_4" type="radio" /><input id="theiaviral_ont_3" name="__tabbed_4" type="radio" /><div class="tabbed-labels"><label for="theiaviral_illumina_pe_3">TheiaViral_Illumina_PE</label><label for="theiaviral_ont_3">TheiaViral_ONT</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>abricate_flu_database</td>
<td>String</td>
<td>ABRicate database used for analysis</td>
</tr>
<tr>
<td>abricate_flu_results</td>
<td>File</td>
<td>File containing all results from ABRicate</td>
</tr>
<tr>
<td>abricate_flu_subtype</td>
<td>String</td>
<td>Flu subtype as determined by ABRicate</td>
</tr>
<tr>
<td>abricate_flu_type</td>
<td>String</td>
<td>Flu type as determined by ABRicate</td>
</tr>
<tr>
<td>abricate_flu_version</td>
<td>String</td>
<td>Version of ABRicate</td>
</tr>
<tr>
<td>assembly_denovo_fasta</td>
<td>File</td>
<td>De novo assembly in FASTA format</td>
</tr>
<tr>
<td>auspice_json_flu_ha</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza HA segment that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_na</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza NA segment that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_mpxv</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Monkeypox virus that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_rabies</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Rabies virus that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>bbduk_docker</td>
<td>String</td>
<td>The Docker image for bbduk, which was used to remove the adapters from the sequences</td>
</tr>
<tr>
<td>bbduk_read1_clean</td>
<td>File</td>
<td>Clean forward reads after BBDuk processing</td>
</tr>
<tr>
<td>bbduk_read2_clean</td>
<td>File</td>
<td>Clean reverse reads after BBDuk processing</td>
</tr>
<tr>
<td>bwa_aligned_bai</td>
<td>File</td>
<td>BAM index file for reads aligned to reference</td>
</tr>
<tr>
<td>bwa_read1_aligned</td>
<td>File</td>
<td>Forward reads aligned to reference</td>
</tr>
<tr>
<td>bwa_read1_unaligned</td>
<td>File</td>
<td>Forward reads not aligned to reference</td>
</tr>
<tr>
<td>bwa_read2_aligned</td>
<td>File</td>
<td>Reverse reads aligned to reference</td>
</tr>
<tr>
<td>bwa_read2_unaligned</td>
<td>File</td>
<td>Reverse reads not aligned to reference</td>
</tr>
<tr>
<td>bwa_samtools_version</td>
<td>String</td>
<td>Version of samtools used by BWA</td>
</tr>
<tr>
<td>bwa_sorted_bai</td>
<td>File</td>
<td>Sorted BAM index file of reads aligned to reference</td>
</tr>
<tr>
<td>bwa_sorted_bam</td>
<td>File</td>
<td>Sorted BAM file of reads aligned to reference</td>
</tr>
<tr>
<td>bwa_sorted_bam_unaligned</td>
<td>File</td>
<td>A BAM file that only contains reads that did not align to the reference</td>
</tr>
<tr>
<td>bwa_sorted_bam_unaligned_bai</td>
<td>File</td>
<td>Index companion file to a BAM file that only contains reads that did not align to the reference</td>
</tr>
<tr>
<td>bwa_version</td>
<td>String</td>
<td>Version of BWA software used</td>
</tr>
<tr>
<td>checkv_consensus_contamination</td>
<td>Float</td>
<td>Contamination estimate for consensus assembly from CheckV</td>
</tr>
<tr>
<td>checkv_consensus_summary</td>
<td>File</td>
<td>Summary report from CheckV for consensus assembly</td>
</tr>
<tr>
<td>checkv_consensus_total_genes</td>
<td>Int</td>
<td>Number of genes detected in consensus assembly by CheckV</td>
</tr>
<tr>
<td>checkv_consensus_version</td>
<td>String</td>
<td>Version of CheckV used for consensus assembly</td>
</tr>
<tr>
<td>checkv_consensus_weighted_completeness</td>
<td>Float</td>
<td>Weighted completeness score for consensus assembly from CheckV</td>
</tr>
<tr>
<td>checkv_consensus_weighted_contamination</td>
<td>Float</td>
<td>Weighted contamination score for consensus assembly from CheckV</td>
</tr>
<tr>
<td>checkv_denovo_contamination</td>
<td>Float</td>
<td>Contamination estimate for de novo assembly from CheckV</td>
</tr>
<tr>
<td>checkv_denovo_summary</td>
<td>File</td>
<td>Summary report from CheckV for de novo assembly</td>
</tr>
<tr>
<td>checkv_denovo_total_genes</td>
<td>Int</td>
<td>Number of genes detected in de novo assembly by CheckV</td>
</tr>
<tr>
<td>checkv_denovo_version</td>
<td>String</td>
<td>Version of CheckV used for de novo assembly</td>
</tr>
<tr>
<td>checkv_denovo_weighted_completeness</td>
<td>Float</td>
<td>Weighted completeness score for de novo assembly from CheckV</td>
</tr>
<tr>
<td>checkv_denovo_weighted_contamination</td>
<td>Float</td>
<td>Weighted contamination score for de novo assembly from CheckV</td>
</tr>
<tr>
<td>consensus_n_variant_min_depth</td>
<td>Int</td>
<td>Minimum read depth to call variants for iVar consensus and iVar variants. Also represents the minimum consensus support threshold used by IRMA with Illumina Influenza data.</td>
</tr>
<tr>
<td>consensus_qc_assembly_length_unambiguous</td>
<td>Int</td>
<td>Length of consensus assembly excluding ambiguous bases</td>
</tr>
<tr>
<td>consensus_qc_number_Degenerate</td>
<td>Int</td>
<td>Number of degenerate bases in consensus assembly</td>
</tr>
<tr>
<td>consensus_qc_number_N</td>
<td>Int</td>
<td>Number of N bases in consensus assembly</td>
</tr>
<tr>
<td>consensus_qc_number_Total</td>
<td>Int</td>
<td>Total number of bases in consensus assembly</td>
</tr>
<tr>
<td>consensus_qc_percent_reference_coverage</td>
<td>Float</td>
<td>Percent of reference genome covered in consensus assembly</td>
</tr>
<tr>
<td>dehost_wf_dehost_read1</td>
<td>File</td>
<td>Reads that did not map to host</td>
</tr>
<tr>
<td>dehost_wf_dehost_read2</td>
<td>File</td>
<td>Paired-reads that did not map to host</td>
</tr>
<tr>
<td>dehost_wf_download_status</td>
<td>String</td>
<td>Status of host genome acquisition</td>
</tr>
<tr>
<td>dehost_wf_host_accession</td>
<td>String</td>
<td>Host genome accession</td>
</tr>
<tr>
<td>dehost_wf_host_fasta</td>
<td>File</td>
<td>Host genome FASTA file</td>
</tr>
<tr>
<td>dehost_wf_host_flagstat</td>
<td>File</td>
<td>Output from the SAMtools flagstat command to assess quality of the alignment file (BAM)</td>
</tr>
<tr>
<td>dehost_wf_host_mapped_bai</td>
<td>File</td>
<td>Indexed bam file of the reads aligned to the host reference</td>
</tr>
<tr>
<td>dehost_wf_host_mapped_bam</td>
<td>File</td>
<td>Sorted BAM file containing the alignments of reads to the host reference genome</td>
</tr>
<tr>
<td>dehost_wf_host_mapping_cov_hist</td>
<td>File</td>
<td>Coverage histogram from host read mapping</td>
</tr>
<tr>
<td>dehost_wf_host_mapping_coverage</td>
<td>Float</td>
<td>Average coverage from host read mapping</td>
</tr>
<tr>
<td>dehost_wf_host_mapping_mean_depth</td>
<td>Float</td>
<td>Average depth from host read mapping</td>
</tr>
<tr>
<td>dehost_wf_host_mapping_metrics</td>
<td>File</td>
<td>File of mapping metrics</td>
</tr>
<tr>
<td>dehost_wf_host_mapping_stats</td>
<td>File</td>
<td>File of mapping statistics</td>
</tr>
<tr>
<td>dehost_wf_host_percent_mapped_reads</td>
<td>Float</td>
<td>Percentage of reads mapped to host reference genome</td>
</tr>
<tr>
<td>fastp_html_report</td>
<td>File</td>
<td>The HTML report made with fastp</td>
</tr>
<tr>
<td>fastp_version</td>
<td>String</td>
<td>The version of fastp used</td>
</tr>
<tr>
<td>fastq_scan_clean1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_clean2_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean reverse read quality and length</td>
</tr>
<tr>
<td>fastq_scan_clean_pairs</td>
<td>Int</td>
<td>Number of read pairs after cleaning</td>
</tr>
<tr>
<td>fastq_scan_docker</td>
<td>String</td>
<td>The Docker image of fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean1</td>
<td>Int</td>
<td>The number of forward reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean2</td>
<td>Int</td>
<td>The number of reverse reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw1</td>
<td>Int</td>
<td>The number of input forward reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw2</td>
<td>Int</td>
<td>The number of input reserve reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_raw1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_raw2_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw reverse read quality and length</td>
</tr>
<tr>
<td>fastq_scan_raw_pairs</td>
<td>Int</td>
<td>Number of raw read pairs</td>
</tr>
<tr>
<td>fastq_scan_version</td>
<td>String</td>
<td>The version of fastq_scan</td>
</tr>
<tr>
<td>genoflu_all_segments</td>
<td>String</td>
<td>The genotypes for each individual flu segment</td>
</tr>
<tr>
<td>genoflu_genotype</td>
<td>String</td>
<td>The genotype of the whole genome, based off of the individual segments types</td>
</tr>
<tr>
<td>genoflu_output_tsv</td>
<td>File</td>
<td>The output file from GenoFLU</td>
</tr>
<tr>
<td>genoflu_version</td>
<td>String</td>
<td>The version of GenoFLU used</td>
</tr>
<tr>
<td>irma_docker</td>
<td>String</td>
<td>Docker image used to run IRMA</td>
</tr>
<tr>
<td>irma_subtype</td>
<td>String</td>
<td>Flu subtype as determined by IRMA</td>
</tr>
<tr>
<td>irma_subtype_notes</td>
<td>String</td>
<td>Helpful note to user about Flu B subtypes. Output will be blank for Flu A samples. For Flu B samples it will state: "IRMA does not differentiate Victoria and Yamagata Flu B lineages. See abricate_flu_subtype output column"</td>
</tr>
<tr>
<td>irma_type</td>
<td>String</td>
<td>Flu type as determined by IRMA</td>
</tr>
<tr>
<td>irma_version</td>
<td>String</td>
<td>Version of IRMA used</td>
</tr>
<tr>
<td>ivar_tsv</td>
<td>File</td>
<td>Variant descriptor file generated by iVar variants</td>
</tr>
<tr>
<td>ivar_variant_proportion_intermediate</td>
<td>String</td>
<td>The proportion of variants of intermediate frequency</td>
</tr>
<tr>
<td>ivar_variant_version</td>
<td>String</td>
<td>Version of iVar for running the iVar variants command</td>
</tr>
<tr>
<td>ivar_vcf</td>
<td>File</td>
<td>iVar tsv output converted to VCF format</td>
</tr>
<tr>
<td>ivar_version_consensus</td>
<td>String</td>
<td>Version of iVar for running the iVar consensus command</td>
</tr>
<tr>
<td>kraken2_extracted_read1</td>
<td>File</td>
<td>Forward reads extracted by taxonomic classification</td>
</tr>
<tr>
<td>kraken2_extracted_read2</td>
<td>File</td>
<td>Reverse reads extracted by taxonomic classification</td>
</tr>
<tr>
<td>kraken_database</td>
<td>File</td>
<td>Database used for Kraken classification</td>
</tr>
<tr>
<td>kraken_docker</td>
<td>String</td>
<td>Docker image used for Kraken</td>
</tr>
<tr>
<td>kraken_report</td>
<td>File</td>
<td>Full Kraken report</td>
</tr>
<tr>
<td>kraken_version</td>
<td>String</td>
<td>Version of Kraken software used</td>
</tr>
<tr>
<td>megahit_docker</td>
<td>String</td>
<td>Docker image used for MEGAHIT</td>
</tr>
<tr>
<td>megahit_status</td>
<td>String</td>
<td>Status of the MEGAHIT assembly</td>
</tr>
<tr>
<td>megahit_version</td>
<td>String</td>
<td>Version of MEGAHIT used</td>
</tr>
<tr>
<td>metaviralspades_docker</td>
<td>String</td>
<td>Docker image used for MetaviralSPAdes</td>
</tr>
<tr>
<td>metaviralspades_status</td>
<td>String</td>
<td>Status of MetaviralSPAdes assembly</td>
</tr>
<tr>
<td>metaviralspades_version</td>
<td>String</td>
<td>Version of MetaviralSPAdes used</td>
</tr>
<tr>
<td>ncbi_datasets_docker</td>
<td>String</td>
<td>Docker image used for NCBI datasets</td>
</tr>
<tr>
<td>ncbi_datasets_version</td>
<td>String</td>
<td>Version of NCBI datasets used</td>
</tr>
<tr>
<td>ncbi_identify_accession</td>
<td>String</td>
<td>NCBI accession ID of identified taxon</td>
</tr>
<tr>
<td>ncbi_identify_avg_genome_length</td>
<td>Int</td>
<td>Average genome length from NCBI taxon summary</td>
</tr>
<tr>
<td>ncbi_identify_genome_summary_tsv</td>
<td>File</td>
<td>TSV file with genome summary from NCBI</td>
</tr>
<tr>
<td>ncbi_identify_read_extraction_rank</td>
<td>String</td>
<td>Taxonomic rank used for read extraction</td>
</tr>
<tr>
<td>ncbi_identify_taxon_id</td>
<td>String</td>
<td>NCBI taxonomy ID of identified organism</td>
</tr>
<tr>
<td>ncbi_identify_taxon_name</td>
<td>String</td>
<td>Name of identified taxon</td>
</tr>
<tr>
<td>ncbi_identify_taxon_summary_tsv</td>
<td>File</td>
<td>TSV file with taxa specific summary from NCBI</td>
</tr>
<tr>
<td>ncbi_scrub_docker</td>
<td>String</td>
<td>The Docker image for NCBI's HRRT (human read removal tool)</td>
</tr>
<tr>
<td>ncbi_scrub_human_spots_removed</td>
<td>Int</td>
<td>Number of spots removed (or masked)</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_ha</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_na</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to Flu; it includes deletions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_mpxv</td>
<td>String</td>
<td>Amino-acid deletions as detected by Nextclade. Specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_aa_dels_rabies</td>
<td>String</td>
<td>Amino-acid deletions as detected by Nextclade. Specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_ha</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_na</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_mpxv</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_aa_subs_rabies</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_clade_mpxv</td>
<td>String</td>
<td>Nextclade clade designation, specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_clade_rabies</td>
<td>String</td>
<td>Nextclade clade designation, specific to Rabies</td>
</tr>
<tr>
<td>nextclade_docker</td>
<td>String</td>
<td>Docker image used to run Nextclade</td>
</tr>
<tr>
<td>nextclade_ds_tag</td>
<td>String</td>
<td>Dataset tag used to run Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_ds_tag_flu_ha</td>
<td>String</td>
<td>Dataset tag used to run Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_ds_tag_flu_na</td>
<td>String</td>
<td>Dataset tag used to run Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_json_flu_ha</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_json_flu_na</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_json_mpxv</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_json_rabies</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Rabies</td>
</tr>
<tr>
<td>nextclade_lineage_mpxv</td>
<td>String</td>
<td>Nextclade lineage designation, specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_lineage_rabies</td>
<td>String</td>
<td>Nextclade lineage designation, specific to Rabies</td>
</tr>
<tr>
<td>nextclade_qc_flu_ha</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_na</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_qc_mpxv</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_qc_rabies</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Rabies</td>
</tr>
<tr>
<td>nextclade_tsv_flu_ha</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_tsv_flu_na</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_tsv_mpxv</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_tsv_rabies</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Rabies</td>
</tr>
<tr>
<td>organism</td>
<td>String</td>
<td>Standardized organism name used for characterization</td>
</tr>
<tr>
<td>pango_lineage</td>
<td>String</td>
<td>Pango lineage as determined by Pangolin</td>
</tr>
<tr>
<td>pango_lineage_expanded</td>
<td>String</td>
<td>Pango lineage without use of aliases; e.g., "BA.1"  "B.1.1.529.1"</td>
</tr>
<tr>
<td>pango_lineage_report</td>
<td>File</td>
<td>Full Pango lineage report generated by Pangolin</td>
</tr>
<tr>
<td>pangolin_assignment_version</td>
<td>String</td>
<td>The version of the pangolin software (e.g. PANGO or PUSHER) used for lineage assignment</td>
</tr>
<tr>
<td>pangolin_conflicts</td>
<td>String</td>
<td>Number of lineage conflicts as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_docker</td>
<td>String</td>
<td>Docker image used to run Pangolin</td>
</tr>
<tr>
<td>pangolin_notes</td>
<td>String</td>
<td>Lineage notes as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_versions</td>
<td>String</td>
<td>All Pangolin software and database versions</td>
</tr>
<tr>
<td>quast_denovo_docker</td>
<td>String</td>
<td>Docker image used for QUAST</td>
</tr>
<tr>
<td>quast_denovo_gc_percent</td>
<td>Float</td>
<td>GC percentage of de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_genome_length</td>
<td>Int</td>
<td>Genome length of de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_largest_contig</td>
<td>Int</td>
<td>Size of largest contig in de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_n50_value</td>
<td>Int</td>
<td>N50 value of de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_number_contigs</td>
<td>Int</td>
<td>Number of contigs in de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_report</td>
<td>File</td>
<td>QUAST report for de novo assembly</td>
</tr>
<tr>
<td>quast_denovo_uncalled_bases</td>
<td>Int</td>
<td>Number of uncalled bases in de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_version</td>
<td>String</td>
<td>Version of QUAST used</td>
</tr>
<tr>
<td>read1_dehosted</td>
<td>File</td>
<td>The dehosted forward reads file; suggested read file for SRA submission</td>
</tr>
<tr>
<td>read2_dehosted</td>
<td>File</td>
<td>The dehosted reverse reads file; suggested read file for SRA submission</td>
</tr>
<tr>
<td>read_mapping_cov_hist</td>
<td>File</td>
<td>Coverage histogram from read mapping</td>
</tr>
<tr>
<td>read_mapping_cov_stats</td>
<td>File</td>
<td>Coverage statistics from read mapping</td>
</tr>
<tr>
<td>read_mapping_coverage</td>
<td>Float</td>
<td>Average coverage from read mapping</td>
</tr>
<tr>
<td>read_mapping_date</td>
<td>String</td>
<td>Date of read mapping analysis</td>
</tr>
<tr>
<td>read_mapping_depth</td>
<td>Float</td>
<td>Average depth from read mapping</td>
</tr>
<tr>
<td>read_mapping_flagstat</td>
<td>File</td>
<td>Flagstat file from read mapping</td>
</tr>
<tr>
<td>read_mapping_meanbaseq</td>
<td>Float</td>
<td>Mean base quality from read mapping</td>
</tr>
<tr>
<td>read_mapping_meanmapq</td>
<td>Float</td>
<td>Mean mapping quality from read mapping</td>
</tr>
<tr>
<td>read_mapping_percentage_mapped_reads</td>
<td>Float</td>
<td>Percentage of mapped reads</td>
</tr>
<tr>
<td>read_mapping_report</td>
<td>File</td>
<td>Report file from read mapping</td>
</tr>
<tr>
<td>read_mapping_samtools_version</td>
<td>String</td>
<td>Version of samtools used in read mapping</td>
</tr>
<tr>
<td>read_mapping_statistics</td>
<td>File</td>
<td>Statistics file from read mapping</td>
</tr>
<tr>
<td>reference_taxon_name</td>
<td>String</td>
<td>NCBI derived taxon name from best ANI hit accession</td>
</tr>
<tr>
<td>skani_database</td>
<td>File</td>
<td>Database used for Skani</td>
</tr>
<tr>
<td>skani_docker</td>
<td>String</td>
<td>Docker image used for Skani</td>
</tr>
<tr>
<td>skani_report</td>
<td>File</td>
<td>Report from Skani</td>
</tr>
<tr>
<td>skani_status</td>
<td>String</td>
<td>Status of Skani analysis</td>
</tr>
<tr>
<td>skani_top_accession</td>
<td>String</td>
<td>Top accession ID from Skani</td>
</tr>
<tr>
<td>skani_top_ani</td>
<td>Float</td>
<td>Top ANI score from Skani</td>
</tr>
<tr>
<td>skani_top_ani_fasta</td>
<td>File</td>
<td>FASTA file of top ANI match from Skani</td>
</tr>
<tr>
<td>skani_top_query_coverage</td>
<td>Float</td>
<td>Query coverage of top match from Skani</td>
</tr>
<tr>
<td>skani_top_score</td>
<td>Float</td>
<td>Top score from Skani</td>
</tr>
<tr>
<td>skani_version</td>
<td>String</td>
<td>Version of Skani used</td>
</tr>
<tr>
<td>skani_warning</td>
<td>String</td>
<td>Skani warning message</td>
</tr>
<tr>
<td>theiaviral_illumina_pe_date</td>
<td>String</td>
<td>Date of TheiaViral Illumina PE workflow run</td>
</tr>
<tr>
<td>theiaviral_illumina_pe_version</td>
<td>String</td>
<td>Version of TheiaViral Illumina PE workflow</td>
</tr>
<tr>
<td>trimmomatic_docker</td>
<td>String</td>
<td>The docker image used for the trimmomatic module in this workflow</td>
</tr>
<tr>
<td>trimmomatic_version</td>
<td>String</td>
<td>The version of Trimmomatic used</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>abricate_flu_database</td>
<td>String</td>
<td>ABRicate database used for analysis</td>
</tr>
<tr>
<td>abricate_flu_results</td>
<td>File</td>
<td>File containing all results from ABRicate</td>
</tr>
<tr>
<td>abricate_flu_subtype</td>
<td>String</td>
<td>Flu subtype as determined by ABRicate</td>
</tr>
<tr>
<td>abricate_flu_type</td>
<td>String</td>
<td>Flu type as determined by ABRicate</td>
</tr>
<tr>
<td>abricate_flu_version</td>
<td>String</td>
<td>Version of ABRicate</td>
</tr>
<tr>
<td>assembly_denovo_fasta</td>
<td>File</td>
<td>De novo assembly in FASTA format</td>
</tr>
<tr>
<td>assembly_to_ref_bai</td>
<td>File</td>
<td>BAM index file for reads aligned to reference</td>
</tr>
<tr>
<td>assembly_to_ref_bam</td>
<td>File</td>
<td>BAM file of reads aligned to reference</td>
</tr>
<tr>
<td>auspice_json_flu_ha</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza HA segment that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_na</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza NA segment that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_mpxv</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Monkeypox virus that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_rabies</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Rabies virus that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>bcftools_docker</td>
<td>String</td>
<td>Docker image used for bcftools</td>
</tr>
<tr>
<td>bcftools_filtered_vcf</td>
<td>File</td>
<td>Filtered variant calls in VCF format from bcftools</td>
</tr>
<tr>
<td>bcftools_version</td>
<td>String</td>
<td>Version of bcftools used</td>
</tr>
<tr>
<td>checkv_consensus_contamination</td>
<td>Float</td>
<td>Contamination estimate for consensus assembly from CheckV</td>
</tr>
<tr>
<td>checkv_consensus_summary</td>
<td>File</td>
<td>Summary report from CheckV for consensus assembly</td>
</tr>
<tr>
<td>checkv_consensus_total_genes</td>
<td>Int</td>
<td>Number of genes detected in consensus assembly by CheckV</td>
</tr>
<tr>
<td>checkv_consensus_version</td>
<td>String</td>
<td>Version of CheckV used for consensus assembly</td>
</tr>
<tr>
<td>checkv_consensus_weighted_completeness</td>
<td>Float</td>
<td>Weighted completeness score for consensus assembly from CheckV</td>
</tr>
<tr>
<td>checkv_consensus_weighted_contamination</td>
<td>Float</td>
<td>Weighted contamination score for consensus assembly from CheckV</td>
</tr>
<tr>
<td>checkv_denovo_contamination</td>
<td>Float</td>
<td>Contamination estimate for de novo assembly from CheckV</td>
</tr>
<tr>
<td>checkv_denovo_summary</td>
<td>File</td>
<td>Summary report from CheckV for de novo assembly</td>
</tr>
<tr>
<td>checkv_denovo_total_genes</td>
<td>Int</td>
<td>Number of genes detected in de novo assembly by CheckV</td>
</tr>
<tr>
<td>checkv_denovo_version</td>
<td>String</td>
<td>Version of CheckV used for de novo assembly</td>
</tr>
<tr>
<td>checkv_denovo_weighted_completeness</td>
<td>Float</td>
<td>Weighted completeness score for de novo assembly from CheckV</td>
</tr>
<tr>
<td>checkv_denovo_weighted_contamination</td>
<td>Float</td>
<td>Weighted contamination score for de novo assembly from CheckV</td>
</tr>
<tr>
<td>clair3_docker</td>
<td>String</td>
<td>Docker image used for Clair3</td>
</tr>
<tr>
<td>clair3_gvcf</td>
<td>File</td>
<td>Genomic VCF file from Clair3</td>
</tr>
<tr>
<td>clair3_model</td>
<td>String</td>
<td>Model used for Clair3 variant calling</td>
</tr>
<tr>
<td>clair3_vcf</td>
<td>File</td>
<td>Variant calls in VCF format from Clair3</td>
</tr>
<tr>
<td>clair3_version</td>
<td>String</td>
<td>Clair3 Version being used</td>
</tr>
<tr>
<td>consensus_qc_assembly_length_unambiguous</td>
<td>Int</td>
<td>Length of consensus assembly excluding ambiguous bases</td>
</tr>
<tr>
<td>consensus_qc_number_Degenerate</td>
<td>Int</td>
<td>Number of degenerate bases in consensus assembly</td>
</tr>
<tr>
<td>consensus_qc_number_N</td>
<td>Int</td>
<td>Number of N bases in consensus assembly</td>
</tr>
<tr>
<td>consensus_qc_number_Total</td>
<td>Int</td>
<td>Total number of bases in consensus assembly</td>
</tr>
<tr>
<td>consensus_qc_percent_reference_coverage</td>
<td>Float</td>
<td>Percent of reference genome covered in consensus assembly</td>
</tr>
<tr>
<td>dehost_wf_dehost_read1</td>
<td>File</td>
<td>Reads that did not map to host</td>
</tr>
<tr>
<td>dehost_wf_download_status</td>
<td>String</td>
<td>Status of host genome acquisition</td>
</tr>
<tr>
<td>dehost_wf_host_accession</td>
<td>String</td>
<td>Host genome accession</td>
</tr>
<tr>
<td>dehost_wf_host_fasta</td>
<td>File</td>
<td>Host genome FASTA file</td>
</tr>
<tr>
<td>dehost_wf_host_flagstat</td>
<td>File</td>
<td>Output from the SAMtools flagstat command to assess quality of the alignment file (BAM)</td>
</tr>
<tr>
<td>dehost_wf_host_mapped_bai</td>
<td>File</td>
<td>Indexed bam file of the reads aligned to the host reference</td>
</tr>
<tr>
<td>dehost_wf_host_mapped_bam</td>
<td>File</td>
<td>Sorted BAM file containing the alignments of reads to the host reference genome</td>
</tr>
<tr>
<td>dehost_wf_host_mapping_cov_hist</td>
<td>File</td>
<td>Coverage histogram from host read mapping</td>
</tr>
<tr>
<td>dehost_wf_host_mapping_coverage</td>
<td>Float</td>
<td>Average coverage from host read mapping</td>
</tr>
<tr>
<td>dehost_wf_host_mapping_mean_depth</td>
<td>Float</td>
<td>Average depth from host read mapping</td>
</tr>
<tr>
<td>dehost_wf_host_mapping_metrics</td>
<td>File</td>
<td>File of mapping metrics</td>
</tr>
<tr>
<td>dehost_wf_host_mapping_stats</td>
<td>File</td>
<td>File of mapping statistics</td>
</tr>
<tr>
<td>dehost_wf_host_percent_mapped_reads</td>
<td>Float</td>
<td>Percentage of reads mapped to host reference genome</td>
</tr>
<tr>
<td>fasta_utilities_fai</td>
<td>File</td>
<td>FASTA index file</td>
</tr>
<tr>
<td>fasta_utilities_samtools_docker</td>
<td>String</td>
<td>Docker image used for samtools in fasta utilities</td>
</tr>
<tr>
<td>fasta_utilities_samtools_version</td>
<td>String</td>
<td>Version of samtools used in fasta utilities</td>
</tr>
<tr>
<td>flye_denovo_docker</td>
<td>String</td>
<td>Docker image used for Flye</td>
</tr>
<tr>
<td>flye_denovo_info</td>
<td>File</td>
<td>Information file from Flye assembly</td>
</tr>
<tr>
<td>flye_denovo_status</td>
<td>String</td>
<td>Status of Flye assembly</td>
</tr>
<tr>
<td>flye_denovo_version</td>
<td>String</td>
<td>Version of Flye used</td>
</tr>
<tr>
<td>genoflu_all_segments</td>
<td>String</td>
<td>The genotypes for each individual flu segment</td>
</tr>
<tr>
<td>genoflu_genotype</td>
<td>String</td>
<td>The genotype of the whole genome, based off of the individual segments types</td>
</tr>
<tr>
<td>genoflu_output_tsv</td>
<td>File</td>
<td>The output file from GenoFLU</td>
</tr>
<tr>
<td>genoflu_version</td>
<td>String</td>
<td>The version of GenoFLU used</td>
</tr>
<tr>
<td>irma_docker</td>
<td>String</td>
<td>Docker image used to run IRMA</td>
</tr>
<tr>
<td>irma_subtype</td>
<td>String</td>
<td>Flu subtype as determined by IRMA</td>
</tr>
<tr>
<td>irma_subtype_notes</td>
<td>String</td>
<td>Helpful note to user about Flu B subtypes. Output will be blank for Flu A samples. For Flu B samples it will state: "IRMA does not differentiate Victoria and Yamagata Flu B lineages. See abricate_flu_subtype output column"</td>
</tr>
<tr>
<td>irma_type</td>
<td>String</td>
<td>Flu type as determined by IRMA</td>
</tr>
<tr>
<td>irma_version</td>
<td>String</td>
<td>Version of IRMA used</td>
</tr>
<tr>
<td>mask_low_coverage_all_coverage_bed</td>
<td>File</td>
<td>BED file showing all coverage regions</td>
</tr>
<tr>
<td>mask_low_coverage_bed</td>
<td>File</td>
<td>BED file showing masked low coverage regions</td>
</tr>
<tr>
<td>mask_low_coverage_bedtools_docker</td>
<td>String</td>
<td>Docker image used for bedtools in masking</td>
</tr>
<tr>
<td>mask_low_coverage_bedtools_version</td>
<td>String</td>
<td>Version of bedtools used in masking</td>
</tr>
<tr>
<td>mask_low_coverage_reference_fasta</td>
<td>File</td>
<td>Reference FASTA with low coverage regions masked</td>
</tr>
<tr>
<td>metabuli_classified</td>
<td>File</td>
<td>Classified reads from Metabuli</td>
</tr>
<tr>
<td>metabuli_database</td>
<td>File</td>
<td>Database used for Metabuli</td>
</tr>
<tr>
<td>metabuli_docker</td>
<td>String</td>
<td>Docker image used for Metabuli</td>
</tr>
<tr>
<td>metabuli_krona_report</td>
<td>File</td>
<td>Krona visualization report from Metabuli</td>
</tr>
<tr>
<td>metabuli_read1_extract</td>
<td>File</td>
<td>Extracted reads from Metabuli</td>
</tr>
<tr>
<td>metabuli_report</td>
<td>File</td>
<td>Classification report from Metabuli</td>
</tr>
<tr>
<td>metabuli_version</td>
<td>String</td>
<td>Version of Metabuli used</td>
</tr>
<tr>
<td>minimap2_docker</td>
<td>String</td>
<td>The Docker image of minimap2</td>
</tr>
<tr>
<td>minimap2_out</td>
<td>File</td>
<td>Output file from Minimap2 alignment</td>
</tr>
<tr>
<td>minimap2_version</td>
<td>String</td>
<td>The version of minimap2</td>
</tr>
<tr>
<td>nanoplot_html_clean</td>
<td>File</td>
<td>An HTML report describing the clean reads</td>
</tr>
<tr>
<td>nanoplot_html_raw</td>
<td>File</td>
<td>An HTML report describing the raw reads</td>
</tr>
<tr>
<td>nanoplot_num_reads_clean1</td>
<td>Int</td>
<td>Number of clean reads</td>
</tr>
<tr>
<td>nanoplot_num_reads_raw1</td>
<td>Int</td>
<td>Number of raw reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_q_clean</td>
<td>Float</td>
<td>Mean quality score of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_q_raw</td>
<td>Float</td>
<td>Mean quality score of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_readlength_clean</td>
<td>Float</td>
<td>Mean read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_readlength_raw</td>
<td>Float</td>
<td>Mean read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_q_clean</td>
<td>Float</td>
<td>Median quality score of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_q_raw</td>
<td>Float</td>
<td>Median quality score of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_readlength_clean</td>
<td>Float</td>
<td>Median read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_readlength_raw</td>
<td>Float</td>
<td>Median read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_n50_clean</td>
<td>Float</td>
<td>N50 of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_n50_raw</td>
<td>Float</td>
<td>N50 of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_stdev_readlength_clean</td>
<td>Float</td>
<td>Standard deviation read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_stdev_readlength_raw</td>
<td>Float</td>
<td>Standard deviation read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_tsv_clean</td>
<td>File</td>
<td>A TSV report describing the clean reads</td>
</tr>
<tr>
<td>nanoplot_tsv_raw</td>
<td>File</td>
<td>A TSV report describing the raw reads</td>
</tr>
<tr>
<td>nanoq_filtered_read1</td>
<td>File</td>
<td>Filtered reads from NanoQ</td>
</tr>
<tr>
<td>nanoq_version</td>
<td>String</td>
<td>Version of nanoq used in analysis</td>
</tr>
<tr>
<td>ncbi_datasets_docker</td>
<td>String</td>
<td>Docker image used for NCBI datasets</td>
</tr>
<tr>
<td>ncbi_datasets_version</td>
<td>String</td>
<td>Version of NCBI datasets used</td>
</tr>
<tr>
<td>ncbi_identify_accession</td>
<td>String</td>
<td>NCBI accession ID of identified taxon</td>
</tr>
<tr>
<td>ncbi_identify_avg_genome_length</td>
<td>Int</td>
<td>Average genome length from NCBI taxon summary</td>
</tr>
<tr>
<td>ncbi_identify_docker</td>
<td>String</td>
<td>Docker image used for NCBI identify</td>
</tr>
<tr>
<td>ncbi_identify_genome_summary_tsv</td>
<td>File</td>
<td>TSV file with genome summary from NCBI</td>
</tr>
<tr>
<td>ncbi_identify_read_extraction_rank</td>
<td>String</td>
<td>Taxonomic rank used for read extraction</td>
</tr>
<tr>
<td>ncbi_identify_taxon_id</td>
<td>String</td>
<td>NCBI taxonomy ID of identified organism</td>
</tr>
<tr>
<td>ncbi_identify_taxon_name</td>
<td>String</td>
<td>Name of identified taxon</td>
</tr>
<tr>
<td>ncbi_identify_taxon_summary_tsv</td>
<td>File</td>
<td>TSV file with taxa specific summary from NCBI</td>
</tr>
<tr>
<td>ncbi_identify_version</td>
<td>String</td>
<td>Version of NCBI identify tool used</td>
</tr>
<tr>
<td>ncbi_scrub_docker</td>
<td>String</td>
<td>The Docker image for NCBI's HRRT (human read removal tool)</td>
</tr>
<tr>
<td>ncbi_scrub_human_spots_removed</td>
<td>Int</td>
<td>Number of spots removed (or masked)</td>
</tr>
<tr>
<td>ncbi_scrub_read1_dehosted</td>
<td>File</td>
<td>Dehosted reads after NCBI scrub</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_ha</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_na</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to Flu; it includes deletions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_mpxv</td>
<td>String</td>
<td>Amino-acid deletions as detected by Nextclade. Specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_aa_dels_rabies</td>
<td>String</td>
<td>Amino-acid deletions as detected by Nextclade. Specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_ha</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_na</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_mpxv</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_aa_subs_rabies</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_clade_mpxv</td>
<td>String</td>
<td>Nextclade clade designation, specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_clade_rabies</td>
<td>String</td>
<td>Nextclade clade designation, specific to Rabies</td>
</tr>
<tr>
<td>nextclade_docker</td>
<td>String</td>
<td>Docker image used to run Nextclade</td>
</tr>
<tr>
<td>nextclade_ds_tag</td>
<td>String</td>
<td>Dataset tag used to run Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_ds_tag_flu_ha</td>
<td>String</td>
<td>Dataset tag used to run Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_ds_tag_flu_na</td>
<td>String</td>
<td>Dataset tag used to run Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_json_flu_ha</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_json_flu_na</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_json_mpxv</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_json_rabies</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Rabies</td>
</tr>
<tr>
<td>nextclade_lineage_mpxv</td>
<td>String</td>
<td>Nextclade lineage designation, specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_lineage_rabies</td>
<td>String</td>
<td>Nextclade lineage designation, specific to Rabies</td>
</tr>
<tr>
<td>nextclade_qc_flu_ha</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_na</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_qc_mpxv</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_qc_rabies</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Rabies</td>
</tr>
<tr>
<td>nextclade_tsv_flu_ha</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_tsv_flu_na</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_tsv_mpxv</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Monkeypox</td>
</tr>
<tr>
<td>nextclade_tsv_rabies</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Rabies</td>
</tr>
<tr>
<td>organism</td>
<td>String</td>
<td>Standardized organism name used for characterization</td>
</tr>
<tr>
<td>pango_lineage</td>
<td>String</td>
<td>Pango lineage as determined by Pangolin</td>
</tr>
<tr>
<td>pango_lineage_expanded</td>
<td>String</td>
<td>Pango lineage without use of aliases; e.g., "BA.1"  "B.1.1.529.1"</td>
</tr>
<tr>
<td>pango_lineage_report</td>
<td>File</td>
<td>Full Pango lineage report generated by Pangolin</td>
</tr>
<tr>
<td>pangolin_assignment_version</td>
<td>String</td>
<td>The version of the pangolin software (e.g. PANGO or PUSHER) used for lineage assignment</td>
</tr>
<tr>
<td>pangolin_conflicts</td>
<td>String</td>
<td>Number of lineage conflicts as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_docker</td>
<td>String</td>
<td>Docker image used to run Pangolin</td>
</tr>
<tr>
<td>pangolin_notes</td>
<td>String</td>
<td>Lineage notes as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_versions</td>
<td>String</td>
<td>All Pangolin software and database versions</td>
</tr>
<tr>
<td>parse_mapping_samtools_docker</td>
<td>String</td>
<td>Docker image used for samtools in parse mapping</td>
</tr>
<tr>
<td>parse_mapping_samtools_version</td>
<td>String</td>
<td>Version of samtools used in parse mapping</td>
</tr>
<tr>
<td>porechop_trimmed_read1</td>
<td>File</td>
<td>Trimmed reads from Porechop</td>
</tr>
<tr>
<td>porechop_version</td>
<td>String</td>
<td>Version of Porechop used</td>
</tr>
<tr>
<td>quast_denovo_docker</td>
<td>String</td>
<td>Docker image used for QUAST</td>
</tr>
<tr>
<td>quast_denovo_gc_percent</td>
<td>Float</td>
<td>GC percentage of de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_genome_length</td>
<td>Int</td>
<td>Genome length of de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_largest_contig</td>
<td>Int</td>
<td>Size of largest contig in de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_n50_value</td>
<td>Int</td>
<td>N50 value of de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_number_contigs</td>
<td>Int</td>
<td>Number of contigs in de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_report</td>
<td>File</td>
<td>QUAST report for de novo assembly</td>
</tr>
<tr>
<td>quast_denovo_uncalled_bases</td>
<td>Int</td>
<td>Number of uncalled bases in de novo assembly from QUAST</td>
</tr>
<tr>
<td>quast_denovo_version</td>
<td>String</td>
<td>Version of QUAST used</td>
</tr>
<tr>
<td>rasusa_read1_subsampled</td>
<td>File</td>
<td>Subsampled read file from Rasusa</td>
</tr>
<tr>
<td>rasusa_read2_subsampled</td>
<td>File</td>
<td>Subsampled read file from Rasusa (paired file)</td>
</tr>
<tr>
<td>rasusa_version</td>
<td>String</td>
<td>Version of RASUSA used for the analysis</td>
</tr>
<tr>
<td>raven_denovo_docker</td>
<td>String</td>
<td>Docker image used for Raven</td>
</tr>
<tr>
<td>raven_denovo_status</td>
<td>String</td>
<td>Status of Raven assembly</td>
</tr>
<tr>
<td>raven_denovo_version</td>
<td>String</td>
<td>Version of Raven used</td>
</tr>
<tr>
<td>read_mapping_cov_hist</td>
<td>File</td>
<td>Coverage histogram from read mapping</td>
</tr>
<tr>
<td>read_mapping_cov_stats</td>
<td>File</td>
<td>Coverage statistics from read mapping</td>
</tr>
<tr>
<td>read_mapping_coverage</td>
<td>Float</td>
<td>Average coverage from read mapping</td>
</tr>
<tr>
<td>read_mapping_date</td>
<td>String</td>
<td>Date of read mapping analysis</td>
</tr>
<tr>
<td>read_mapping_depth</td>
<td>Float</td>
<td>Average depth from read mapping</td>
</tr>
<tr>
<td>read_mapping_flagstat</td>
<td>File</td>
<td>Flagstat file from read mapping</td>
</tr>
<tr>
<td>read_mapping_meanbaseq</td>
<td>Float</td>
<td>Mean base quality from read mapping</td>
</tr>
<tr>
<td>read_mapping_meanmapq</td>
<td>Float</td>
<td>Mean mapping quality from read mapping</td>
</tr>
<tr>
<td>read_mapping_percentage_mapped_reads</td>
<td>Float</td>
<td>Percentage of mapped reads</td>
</tr>
<tr>
<td>read_mapping_report</td>
<td>File</td>
<td>Report file from read mapping</td>
</tr>
<tr>
<td>read_mapping_samtools_version</td>
<td>String</td>
<td>Version of samtools used in read mapping</td>
</tr>
<tr>
<td>read_mapping_statistics</td>
<td>File</td>
<td>Statistics file from read mapping</td>
</tr>
<tr>
<td>read_screen_clean</td>
<td>String</td>
<td>PASS or FAIL result from clean read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_clean_tsv</td>
<td>File</td>
<td>Clean read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>reference_taxon_name</td>
<td>String</td>
<td>NCBI derived taxon name from best ANI hit accession</td>
</tr>
<tr>
<td>skani_database</td>
<td>File</td>
<td>Database used for Skani</td>
</tr>
<tr>
<td>skani_docker</td>
<td>String</td>
<td>Docker image used for Skani</td>
</tr>
<tr>
<td>skani_report</td>
<td>File</td>
<td>Report from Skani</td>
</tr>
<tr>
<td>skani_status</td>
<td>String</td>
<td>Status of Skani analysis</td>
</tr>
<tr>
<td>skani_top_accession</td>
<td>String</td>
<td>Top accession ID from Skani</td>
</tr>
<tr>
<td>skani_top_ani</td>
<td>Float</td>
<td>Top ANI score from Skani</td>
</tr>
<tr>
<td>skani_top_ani_fasta</td>
<td>File</td>
<td>FASTA file of top ANI match from Skani</td>
</tr>
<tr>
<td>skani_top_query_coverage</td>
<td>Float</td>
<td>Query coverage of top match from Skani</td>
</tr>
<tr>
<td>skani_top_score</td>
<td>Float</td>
<td>Top score from Skani</td>
</tr>
<tr>
<td>skani_version</td>
<td>String</td>
<td>Version of Skani used</td>
</tr>
<tr>
<td>skani_warning</td>
<td>String</td>
<td>Skani warning message</td>
</tr>
<tr>
<td>theiaviral_ont_date</td>
<td>String</td>
<td>Date of TheiaViral ONT workflow run</td>
</tr>
<tr>
<td>theiaviral_ont_version</td>
<td>String</td>
<td>Version of TheiaViral ONT workflow</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<details class="question">
<summary>What are the differences between the <em>de novo</em> and consensus assemblies?</summary>
<p><em>De novo</em> genomes are generated from scratch without a reference to guide read assembly, while consensus genomes are generated by mapping reads to a reference and replacing reference positions with identified variants (structural and nucleotide). <em>De novo</em> assemblies are thus not biased by requiring reads map to the reference, though they may be more fragmented. Consensus assembly can generate more robust assemblies from lower coverage samples if the reference genome is sufficient quality and sufficiently closely related to the inputted sequence, though consensus assembly may not perform well in instances of significant structural variation. TheiaViral uses <em>de novo</em> assemblies as an intermediate to acquire the best reference genome for consensus assembly.     </p>
<p><strong>We generally recommend TheiaViral users focus on the consensus assembly as the desired assembly output</strong>. While we chose the best <em>de novo</em> assemblers for TheiaViral based on internal benchmarking, the consensus assembly will often be higher quality than the <em>de novo</em> assembly. However, the <em>de novo</em> assembly can approach or exceed consensus quality if the read inputs largely comprise one virus, have high depth of coverage, and/or are derived from a virus with high potential for recombination. TheiaViral does conduct assembly contiguity and viral completeness quality control for <em>de novo</em> assemblies, so <em>de novo</em> assembly that meets quality control standards can certainly be used for downstream analysis.</p>
</details>
<details class="question">
<summary>How is <em>de novo</em> assembly quality evaluated?</summary>
<p><em>De novo</em> assembly quality evaluation focuses on the completeness and contiguity of the genome. While a ground truth genome does not truly exist for quality comparison, reference genome selection can help contextualize quality if the reference is sufficiently similar to the <em>de novo</em> assembly. TheiaViral uses QUAST to acquire basic contiguity statistics and CheckV to assess viral genome completeness and contamination. Additionally, the reference selection software, Skani, can provide a quantitative comparison between the <em>de novo</em> assembly and the best reference genome.</p>
<p><strong>Completeness and contamination</strong>
<br> </p>
<ul>
<li><code>checkv_denovo_summary</code>: The summary file reports CheckV results on a contig-by-contig basis. Ideally completeness is 100% for a single contig, or 100% for all segments. If there are multiple extraneous contigs in the assembly, one is ideally 100%. The same principles apply to contamination, though it ideally is 0%.</li>
<li><code>checkv_denovo_total_genes</code>: The total genes is ideally the same number of genes as expected from the inputted viral taxon. Sometimes CheckV can fail to recover all the genes from a complete genome, so other statistics should be weighted more heavily in quality evaluation.</li>
<li><code>checkv_denovo_weighted_completeness</code>: The weighted completeness is ideally 100%.</li>
<li><code>checkv_denovo_weighted_contamination</code>: The weighted contamination is ideally 0%.</li>
</ul>
<p><strong>Length and contiguity</strong>
<br></p>
<ul>
<li><code>quast_denovo_genome_length</code>: The <em>de novo</em> genome length is ideally the same as the expected genome length of the focal virus.</li>
<li><code>quast_denovo_largest_contig</code>: The largest contig is ideally the size of the genome, or the size of the largest expected segment. If there are multiple contigs, and the largest contig is the ideal size, then the smaller contigs may be discarded based on the CheckV completeness for the largest contig (see CheckV outputs).</li>
<li><code>quast_denovo_n50_value</code>: The N50 is an evaluation of contiguity and is ideally as close as possible to the genome size. For segmented viruses, the N50 should be as close as possible to the size of the segment molecule that would cover at least 50% of the total genome size when segment lengths are added after sorting largest to smallest.</li>
<li><code>quast_denovo_number_contigs</code>: The number of contigs is ideally 1 or the total number of segments expected.</li>
</ul>
<p><strong>Reference genome similarity</strong>
<br></p>
<ul>
<li><code>skani_top_ani</code>: The percent average nucleotide identity (ANI) for the top Skani hit is ideally 100% if the sequenced virus is highly similar to a reference genome. However, if the virus is divergent, ANI is not a good indication of assembly quality.</li>
<li><code>skani_top_query_coverage</code>: The percent query coverage for the top Skani hit is ideally 100% if the sequenced virus has not undergone significant recombination/structural variation. </li>
<li><code>skani_top_score</code>: The score for the top Skani hit is the ANI x Query (<em>de novo</em> assembly) coverage and is ideally 100% if the sequenced virus is not substantially divergent from the reference dataset.</li>
</ul>
</details>
<details class="question">
<summary>How is consensus assembly quality evaluated?</summary>
<p>Consensus assemblies are derived from a reference genome, so quality assessment focuses on coverage and variant quality. Bases with insufficient coverage are denoted as "N". Additionally, the size and contiguity of a TheiaViral consensus assembly is expected to approximate the reference genome, so any discrepancy here is likely due to inferred structural variation.</p>
<p><strong>Completeness and contamination</strong>
<br></p>
<ul>
<li><code>checkv_consensus_weighted_completeness</code>: The weighted completeness is ideally 100%.</li>
</ul>
<p><strong>Consensus variant calls</strong>
<br></p>
<ul>
<li><code>consensus_qc_number_Degenerate</code>: The number of degenerate bases is ideally 0. While degenerate bases indicate ambiguity in the sequence, non-N degenerate bases indicate that some information about the base was obtained.</li>
<li><code>consensus_qc_number_N</code>: The number of "N" bases is ideally 0.</li>
</ul>
<p><strong>Coverage</strong>
<br></p>
<ul>
<li><code>consensus_qc_percent_reference_coverage</code>: The percent reference coverage is ideally 100%.</li>
<li><code>read_mapping_cov_hist</code>: The read mapping coverage histogram ideally depicts normally distributed coverage, which may indicate uniform coverage across the reference genome. However, uniform coverage is unlikely with repetitive regions that approach/exceed read length.</li>
<li><code>read_mapping_coverage</code>: The average read mapping coverage is ideally as high as possible.</li>
<li><code>read_mapping_meanbaseq</code>: The average mean mapping base quality is ideally as high as possible.</li>
<li><code>read_mapping_meanmapq</code>: The average mean mapping alignment quality is ideally as high as possible.</li>
<li><code>read_mapping_percentage_mapped_reads</code>: The percent of mapped reads is ideally 100% of the reads classified as the lineage of interest. Some unclassified reads may also map, which may indicate they were erroneously unclassified. Alternatively, these reads could have been erroneously mapped.</li>
</ul>
</details>
<details class="question">
<summary>Why did the workflow complete without generating a consensus?</summary>
<p>TheiaViral is designed to "soft fail" when specific steps do not succeed due to input data quality. This means the workflow will be reported as successful, with an output that delineates the step that failed. If the workflow fails, please look for the following outputs in this order (sorted by timing of failure, latest first):</p>
<ul>
<li><code>skani_status</code>: If this output is populated with something other than "PASS" and <code>skani_top_accession</code> is populated with "N/A", this indicates that Skani did not identify a sufficiently similar reference genome. The Skani database comprises a broad array of NCBI viral genomes, so a failure here likely indicates poor read quality because viral contigs are not found in the <em>de novo</em> assembly or are too small. It may be useful to BLAST whatever contigs do exist in the <em>de novo</em> to determine if there is contamination that can be removed via the <code>host</code> input parameter. Additionally, review CheckV <em>de novo</em> outputs to assess if viral contigs were retrieved. Finally, consider keeping <code>extract_unclassified</code> to "true", using a higher <code>read_extraction_rank</code> if it will not introduce contaminant viruses, and invoking a <code>host</code> input to remove host reads if host contigs are present.</li>
<li><code>megahit_status</code> / <code>flye_status</code>: If this output is populated with something other than "PASS", it indicates the fallback assembler did not successfully complete. The fallback assemblers are permissive, so failure here likely indicates poor read quality. Review read QC to check read quality, particularly following read classification. If read classification is dispensing with a significant number of reads, consider <code>extract_unclassified</code>, <code>read_extraction_rank</code>, and <code>host</code> input adjustment. Otherwise, sequencing quality may be poor.</li>
<li><code>metaviralspades_status</code> / <code>raven_denovo_status</code>: If this output is populated with something other than "PASS", it indicates the default assembler did not successfully complete or extract viral contigs (MetaviralSPAdes). On their own, these statuses do not correspond directly to workflow failure because fallback <em>de novo</em> assemblers are implemented for both TheiaViral workflows.</li>
<li><code>read_screen_clean</code>: If this output is populated with something other than "PASS", it indicates the reads did not pass the imposed thresholds. Either the reads are poor quality or the thresholds are too stringent, in which case the thresholds can be relaxed or <code>skip_screen</code> can be set to "true".</li>
<li><code>dehost_wf_download_status</code>: If this output is populated with something other than "PASS", it indicates a host genome could not be retrieved for decontamination. See the <code>host</code> input explanation for more information and review the <code>download_accession</code>/<code>download_taxonomy</code> task output logs for advanced error parsing.</li>
</ul>
</details>
<details class="warning">
<summary>Known errors associated with read quality</summary>
<ul>
<li>ONT workflows may fail at Metabuli if no reads are classified as the <code>taxon</code>. Check the Metabuli <code>classification.tsv</code> or <code>krona</code> report for the read extraction taxon ID to determine if any reads were classified. This error will report <code>out of memory (OOM)</code>, but increasing memory will not resolve it.</li>
<li>Illumina workflows may fail at CheckV (<em>de novo</em>) with <code>Error: 80 hmmsearch tasks failed. Program should be rerun</code> if no viral contigs were identified in the <em>de novo</em> assembly.</li>
</ul>
</details>
<h3 id="acknowlegments">Acknowlegments<a class="headerlink" href="#acknowlegments" title="Permanent link">&para;</a></h3>
<p>We would like to thank Danny Park at the Broad institute and Jared Johnson at the Washington State Department of Public Health for correspondence during the development of TheiaViral. TheiaViral was built referencing <a href="https://github.com/broadinstitute/viral-assemble/">viral-assemble</a>, <a href="https://github.com/DOH-JDJ0303/vaper">VAPER</a>, and <a href="https://github.com/artic-network/fieldbioinformatics">Artic</a>.</p>







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"/></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="July 17, 2025 20:41:44 UTC">2025-07-17</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"/></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="June 25, 2025 14:26:30 UTC">2025-06-25</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; 2022-2025 <a href="https://www.theiagen.com" target="_blank" rel="noopener">Theiagen Genomics</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/theiagen" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/theiagen" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M459.4 151.7c.3 4.5.3 9.1.3 13.6 0 138.7-105.6 298.6-298.6 298.6-59.5 0-114.7-17.2-161.1-47.1 8.4 1 16.6 1.3 25.3 1.3 49.1 0 94.2-16.6 130.3-44.8-46.1-1-84.8-31.2-98.1-72.8 6.5 1 13 1.6 19.8 1.6 9.4 0 18.8-1.3 27.6-3.6-48.1-9.7-84.1-52-84.1-103v-1.3c14 7.8 30.2 12.7 47.4 13.3-28.3-18.8-46.8-51-46.8-87.4 0-19.5 5.2-37.4 14.3-53C87.4 130.8 165 172.4 252.1 176.9c-1.6-7.8-2.6-15.9-2.6-24C249.5 95.1 296.3 48 354.4 48c30.2 0 57.5 12.7 76.7 33.1 23.7-4.5 46.5-13.3 66.6-25.3-7.8 24.4-24.4 44.8-46.1 57.8 21.1-2.3 41.6-8.1 60.4-16.2-14.3 20.8-32.2 39.3-52.6 54.3"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.linkedin.com/company/theiagen" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3M135.4 416H69V202.2h66.5V416zM102.2 96a38.5 38.5 0 1 1 0 77 38.5 38.5 0 1 1 0-77m282.1 320h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "navigation.instant", "navigation.path", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.suggest", "toc.follow"], "search": "../../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"alias": true, "default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.50899def.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../../../javascripts/tablesort.js"></script>
      
        <script src="../../../javascripts/table-search.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>