<!DOCTYPE html><html lang="en" class="no-js"><head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A collection of bioinformatics workflows for genomic characterization, submission preparation, and genomic epidemiology of pathogens of public health concern.">
      
      
        <meta name="author" content="Theiagen Genomics">
      
      
        <link rel="canonical" href="https://theiagen.github.io/public_health_bioinformatics/main/workflows/genomic_characterization/theiaeuk/">
      
      
        <link rel="prev" href="../theiacov/">
      
      
        <link rel="next" href="../theiameta/">
      
      
      <link rel="icon" href="../../../assets/logos/Theiagen-Symbol-Standard-01.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.21">
    
    
      
        <title>TheiaEuk Workflow Series - Public Health Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.2a3383ac.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  <link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#theiaeuk-workflow-series" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version release.
  <a href="../../../.."> 
    <strong>Click here to go to the latest version release.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-header__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Public Health Bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              TheiaEuk Workflow Series
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo" aria-label="Switch to Dark Mode" type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to Dark Mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"></path></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo" aria-label="Switch to Light Mode" type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to Light Mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m3.55 19.09 1.41 1.41 1.8-1.79-1.42-1.42M12 6c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6c0-3.32-2.69-6-6-6m8 7h3v-2h-3m-2.76 7.71 1.8 1.79 1.41-1.41-1.79-1.8M20.45 5l-1.41-1.4-1.8 1.79 1.42 1.42M13 1h-2v3h2M6.76 5.39 4.96 3.6 3.55 5l1.79 1.81zM1 13h3v-2H1m12 9h-2v3h2"></path></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../.." class="md-tabs__link">
          
  
  
  Public Health Bioinformatics

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-nav__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    Public Health Bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Public Health Bioinformatics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Public Health Bioinformatics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Getting Started
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            Getting Started
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/commandline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With the Command-Line
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/terra/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With Terra.bio
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/philosophy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Workflow Failure Philosophy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Other Resources
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_3">
            <span class="md-nav__icon md-icon"></span>
            Other Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/sops/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Available SOPs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/versions/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Database and Docker Versions
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit_database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT Database Creation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/phylogenetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Phylogenetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/custom_organisms/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Running Custom Organisms on TheiaCoV
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1">
        
          
          <label class="md-nav__link" for="__nav_1_5_1" id="__nav_1_5_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Type
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Type
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_type/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_2" id="__nav_1_5_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Import
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_2">
            <span class="md-nav__icon md-icon"></span>
            Data Import
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_3" id="__nav_1_5_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Genomic Characterization
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_3">
            <span class="md-nav__icon md-icon"></span>
            Genomic Characterization
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaeuk-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaEuk Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaEuk Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#workflow-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Workflow Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Core tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#organism-agnostic-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-agnostic characterization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#organism-specific-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-specific characterization
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_4" id="__nav_1_5_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Construction
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_4">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Construction
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_5" id="__nav_1_5_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Placement
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_5">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Placement
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_6" id="__nav_1_5_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Public Data Sharing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_6">
            <span class="md-nav__icon md-icon"></span>
            Public Data Sharing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_7">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_7" id="__nav_1_5_1_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exporting Data from Terra
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_7">
            <span class="md-nav__icon md-icon"></span>
            Exporting Data from Terra
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_8">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_8" id="__nav_1_5_1_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Standalone
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_8">
            <span class="md-nav__icon md-icon"></span>
            Standalone
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/chroquetas/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ChroQueTas
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2" id="__nav_1_5_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Kingdom
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Kingdom
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_kingdom/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_2" id="__nav_1_5_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Any Taxa
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_2">
            <span class="md-nav__icon md-icon"></span>
            Any Taxa
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_3" id="__nav_1_5_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Bacteria
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_3">
            <span class="md-nav__icon md-icon"></span>
            Bacteria
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_4" id="__nav_1_5_2_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Mycotics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_4">
            <span class="md-nav__icon md-icon"></span>
            Mycotics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/chroquetas/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ChroQueTas
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaeuk-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaEuk Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaEuk Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#workflow-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Workflow Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Core tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#organism-agnostic-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-agnostic characterization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#organism-specific-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-specific characterization
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_5" id="__nav_1_5_2_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Viral
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_5">
            <span class="md-nav__icon md-icon"></span>
            Viral
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_3" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows Alphabetically
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            Workflows Alphabetically
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_alphabetically/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/chroquetas/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ChroQueTas
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaeuk-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaEuk Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaEuk Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#workflow-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Workflow Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Core tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#organism-agnostic-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-agnostic characterization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#organism-specific-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-specific characterization
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Contributing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            Contributing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/doc_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Documentation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/code_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Code
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaeuk-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaEuk Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaEuk Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#workflow-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Workflow Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Core tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#organism-agnostic-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-agnostic characterization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#organism-specific-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-specific characterization
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="theiaeuk-workflow-series">TheiaEuk Workflow Series<a class="headerlink" href="#theiaeuk-workflow-series" title="Permanent link"></a></h1>
<h2 id="quick-facts">Quick Facts<a class="headerlink" href="#quick-facts" title="Permanent link"></a></h2>
<table>
<thead>
<tr>
<th><strong>Workflow Type</strong></th>
<th><strong>Applicable Kingdom</strong></th>
<th><strong>Last Known Changes</strong></th>
<th><strong>Command-line Compatibility</strong></th>
<th><strong>Workflow Level</strong></th>
<th><strong>Dockstore</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../../../workflows_overview/workflows_type/#genomic-characterization">Genomic Characterization</a></td>
<td><a href="../../../workflows_overview/workflows_kingdom/#mycotics">Mycotics</a></td>
<td>vX.X.X</td>
<td>Some optional features incompatible, Yes</td>
<td>Sample-level</td>
<td><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaEuk_Illumina_PE_PHB:main?tab=info">TheiaEuk_Illumina_PE_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaEuk_ONT_PHB:main?tab=info">TheiaEuk_ONT_PHB</a></td>
</tr>
</tbody>
</table>
<h2 id="theiaeuk-workflows">TheiaEuk Workflows<a class="headerlink" href="#theiaeuk-workflows" title="Permanent link"></a></h2>
<p><strong>The TheiaEuk workflows are for the assembly, quality assessment, and characterization of fungal genomes.</strong> It is designed to accept Illumina paired-end sequencing data or base-called ONT reads as the primary input. <strong>It is currently intended only for <mark>haploid</mark> fungal genomes like <em>Candidozyma auris</em>.</strong> Analyzing diploid genomes using TheiaEuk should be attempted only with expert attention to the resulting genome quality.</p>
<p>All input reads are processed through "core tasks" in each workflow. The core tasks include raw read quality assessment, read cleaning (quality trimming and adapter removal), de novo assembly, assembly quality assessment, species taxon identification, and antimicrobial resistance (AMR) <em>in silico</em> prediction. For some taxa identified, taxa-specific sub-workflows will be automatically activated, undertaking additional taxa-specific characterization steps, including clade-typing and/or antifungal resistance detection.</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="theiaeuk_illumina_pe" name="__tabbed_1" type="radio"><input id="theiaeuk_ont" name="__tabbed_1" type="radio"><div class="tabbed-labels"><label for="theiaeuk_illumina_pe">TheiaEuk_Illumina_PE</label><label for="theiaeuk_ont">TheiaEuk_ONT</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="admonition caption">
<p class="admonition-title">TheiaEuk Illumina PE Workflow Diagram</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/TheiaEuk_Illumina_PE.png" data-desc-position="bottom"><img alt="TheiaEuk Illumina PE Workflow Diagram" src="../../../assets/figures/TheiaEuk_Illumina_PE.png"></a></p>
</div>
</div>
<div class="tabbed-block">
<div class="admonition caption">
<p class="admonition-title">TheiaEuk ONT Workflow Diagram</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/TheiaEuk_ONT.png" data-desc-position="bottom"><img alt="TheiaEuk ONT Workflow Diagram" src="../../../assets/figures/TheiaEuk_ONT.png"></a></p>
</div>
</div>
</div>
</div>
<div class="admonition warning">
<p class="admonition-title">Before running TheiaEuk</p>
<p>For some taxa, TheiaEuk_Illumina_PE relies on <a href="#organism-specific-characterization">Snippy</a> to perform variant calling on the cleaned read dataset and then queries the resulting file for specific mutations that are known to confim antifugal resistance (see <a href="#organism-specific-characterization">Organism-specific characterization</a> section). This behaviour has been replicated in TheiaEuk_ONT but the variant calling is performed directly on the resulting assemblies. Therefore, the read support reported is, at the moment, non-reliable. Future improvements will include improvements on this module. </p>
</div>
<h3 id="inputs">Inputs<a class="headerlink" href="#inputs" title="Permanent link"></a></h3>
<div class="admonition dna">
<p class="admonition-title">Input Read Data</p>
<div class="tabbed-set tabbed-alternate" data-tabs="2:2"><input checked="checked" id="theiaeuk_illumina_pe_1" name="__tabbed_2" type="radio"><input id="theiaeuk_ont_1" name="__tabbed_2" type="radio"><div class="tabbed-labels"><label for="theiaeuk_illumina_pe_1">TheiaEuk_Illumina_PE</label><label for="theiaeuk_ont_1">TheiaEuk_ONT</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>The TheiaEuk_Illumina_PE workflow takes in Illumina paired-end read data. Read file names should end with <code>.fastq</code> or <code>.fq</code>, with the optional addition of <code>.gz</code>. When possible, Theiagen recommends zipping files with <a href="https://www.gnu.org/software/gzip/">gzip</a> before Terra uploads to minimize data upload time.</p>
<p>By default, the workflow anticipates&nbsp;<strong>2 x 150bp</strong>&nbsp;reads (i.e. the input reads were generated using a 300-cycle sequencing kit). Modifications to the optional parameter for <code>trim_minlen</code> may be required to accommodate shorter read data, such as the 2 x 75bp reads generated using a 150-cycle sequencing kit.</p>
</div>
<div class="tabbed-block">
<p>The TheiaEuk_ONT workflow takes in base-called ONT read data. Read file names should end with <code>.fastq</code> or <code>.fq</code>, with the optional addition of <code>.gz</code>. When possible, Theiagen recommends zipping files with <a href="https://www.gnu.org/software/gzip/">gzip</a> before uploading to Terra to minimize data upload time.</p>
<p><strong>The ONT sequencing kit and base-calling approach can produce substantial variability in the amount and quality of read data. Genome assemblies produced by the TheiaEuk_ONT workflow must be quality assessed before reporting results.</strong></p>
</div>
</div>
</div>
</div>
<div class="admonition caption">
<div class="tabbed-set tabbed-alternate" data-tabs="3:2"><input checked="checked" id="theiaeuk_illumina_pe_2" name="__tabbed_3" type="radio"><input id="theiaeuk_ont_2" name="__tabbed_3" type="radio"><div class="tabbed-labels"><label for="theiaeuk_illumina_pe_2">TheiaEuk_Illumina_PE</label><label for="theiaeuk_ont_2">TheiaEuk_ONT</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>FASTQ file containing read1 sequences</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>FASTQ file containing read2 sequences</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>Sample name for the analysis</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>busco</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>cg_pipe_opts</strong></td>
<td>String</td>
<td>Options to pass to CG-Pipeline for clean read assessment</td>
<td>--fast</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/lyveset:1.1.4f</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>cg_pipe_opts</strong></td>
<td>String</td>
<td>Options to pass to CG-Pipeline for raw read assessment</td>
<td>--fast</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/lyveset:1.1.4f</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>assembler</strong></td>
<td>String</td>
<td>Assembler to use (spades, skesa, megahit)</td>
<td>skesa</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>assembler_options</strong></td>
<td>String</td>
<td>Assembler-specific options that you might choose for the selected assembler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>6</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>call_pilon</strong></td>
<td>Boolean</td>
<td>Whether to run Pilon polishing after assembly</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/shovilter:0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_min_coverage</strong></td>
<td>Float</td>
<td>Minimum coverage threshold for contig filtering</td>
<td>2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_skip_coverage_filter</strong></td>
<td>Boolean</td>
<td>Skip filtering contigs based on coverage</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_skip_homopolymer_filter</strong></td>
<td>Boolean</td>
<td>Skip filtering contigs containing homopolymers</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_skip_length_filter</strong></td>
<td>Boolean</td>
<td>Skip filtering contigs based on length</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>kmers</strong></td>
<td>String</td>
<td>K-mer sizes for assembly (comma-separated)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/megahit:1.2.9</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/pilon:1.24--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_fix</strong></td>
<td>String</td>
<td>Potential issues with assembly to try and automatically fix (snps, indels, gaps, local, all, bases, none)</td>
<td>bases</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_min_base_quality</strong></td>
<td>Int</td>
<td>Minimum base quality to keep</td>
<td>3</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_min_depth</strong></td>
<td>Float</td>
<td>Minimum coverage threshold for variant calling: when set to a value 1, it requires that absolute depth of coverage; when set to a fraction &lt;1, it requires coverage at least that fraction of the mean coverage for the region</td>
<td>0.25</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_min_mapping_quality</strong></td>
<td>Int</td>
<td>Minimum mapping quality for a read to count in pileups</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>run_filter_contigs</strong></td>
<td>Boolean</td>
<td>Whether to run contig filtering step</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/skesa:2.4.0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/spades:4.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_type</strong></td>
<td>String</td>
<td>SPAdes assembly mode (isolate, meta, rna, etc.), more can be found <a href="https://ablab.github.io/spades/running.html">here</a></td>
<td>isolate</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_agr_typing_only</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/amrsearch:0.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>assembly_only</strong></td>
<td>Boolean</td>
<td>Set to true if only analyzing input assembly</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_poppunk</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_shigeifinder_reads_input</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_stxtyper</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_tbp_parser</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cauris_cladetyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_docker_image</strong></td>
<td>String</td>
<td>Docker image to use for the task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_memory</strong></td>
<td>Int</td>
<td>Amount of memory (in GB) to allocate to the task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum percent coverage of FungAMR reference protein</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum percent identity of FungAMR reference protein</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroqutas_translation_code</strong></td>
<td>String</td>
<td>Alternative codon translation code. Set for each species by default and not recommended to change</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_kmer_size</strong></td>
<td>Int</td>
<td>Kmer size for cladetyper</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_max_distance</strong></td>
<td>Float</td>
<td>The maximum GAMBIT distance to report a <em>C. auris</em> clade hit</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>Candidozyma auris</em> clade1</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade1_GCA_002759435.2_Cand_auris_B8441_V2_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade1</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade1_GCA_002759435_Cauris_B8441_V2_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade2</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade2_GCA_003013715.2_ASM301371v2_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade2</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade2_GCA_003013715.2_ASM301371v2_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade3</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade3_GCF_002775015.1_Cand_auris_B11221_V1_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade3</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade3_GCF_002775015.1_Cand_auris_B11221_V1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade4</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade4_GCA_003014415.1_Cand_auris_B11243_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade4</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade4_GCA_003014415.1_Cand_auris_B11243_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade5</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade5_GCA_016809505.1_ASM1680950v1_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade5</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade5_GCA_016809505.1_ASM1680950v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade6</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade6_GCA_032714025.1_ASM3271402v1_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade6</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>clockwork_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_print_alleles</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_verify</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_align_diff</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_cluster_distance</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_culling_limit</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_gap</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_max_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_good</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_perfect</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_mismatch</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_wf</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtypingtool_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>genotyphi_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_broken_gene_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_broken_gene_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_low_gene_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_start_end_margin</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_kaptive_confidence</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_kaptive</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_resistance</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>legsta_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/legsta:0.5.1--hdfd78af_2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>meningotype_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ngmaster_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ont_data</strong></td>
<td>Boolean</td>
<td>Set to true if your data is ONT FASTQ files</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>paired_end</strong></td>
<td>Boolean</td>
<td>Set to true if your data is paired-end FASTQ files</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_clusters_csv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_npy</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_pkl</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_external_clusters_csv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_npz</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_pkl</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_graph_gt</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_h5</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_qcreport_txt</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_npy</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_pkl</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_graph_gt</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_h5</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_unword_clusters_csv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>run_amr_search</strong></td>
<td>Boolean</td>
<td>If set to true AMR_Search workflow will be run if species is part of supported taxon, see AMR_Search docs.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seqsero2_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seroba_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>serotypefinder_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigatyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigeifinder_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_use_full_cgmlst_db</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_base_quality</strong></td>
<td>Int</td>
<td>Minimum quality for a nucleotide to be used in variant calling</td>
<td>13</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_gene_query_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_map_qual</strong></td>
<td>Int</td>
<td>Minimum mapping quality to accept in variant calling</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_maxsoft</strong></td>
<td>Int</td>
<td>Number of bases of alignment to soft-clip before discarding the alignment</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_coverage</strong></td>
<td>Int</td>
<td>Minimum read coverage of a position to identify a mutation</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_frac</strong></td>
<td>Float</td>
<td>Minimum fraction of bases at a given position to identify a mutation</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_quality</strong></td>
<td>Int</td>
<td>Minimum VCF variant call "quality"</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_query_gene</strong></td>
<td>String</td>
<td>Provide a gene to search for using Snippy</td>
<td>Default depend on detected organism</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_afumigatus</strong></td>
<td>File</td>
<td>Snippy reference for Aspergillus fumigatus</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/aspergillus/Aspergillus_fumigatus_GCF_000002655.1_ASM265v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_cryptoneo</strong></td>
<td>File</td>
<td>Snippy reference for Cryptococcus neoformans</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/cryptococcus/Cryptococcus_neoformans_GCF_000091045.1_ASM9104v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_variants_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snippy:4.6.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_mykrobe_opts</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_do_enrich</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_gene_max_mismatch</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>2000</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_max_divergence</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_depth</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_edge_depth</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>staphopia_sccmec_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_enable_debug</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_add_cs_lims</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_config</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_coverage_regions_bed</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_debug</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_etha237_frequency</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_expert_rule_regions_bed</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_depth</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_frequency</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_read_support</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_operator</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_output_seq_method_type</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>WGS</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rpob449_frequency</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_frequency</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_read_support</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_frequency</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_read_support</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_tngs_data</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_additional_parameters</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_custom_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_mapper</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_af</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_depth</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_cdph_db</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_custom_db</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_caller</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_calling_params</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_lineage_barcodes</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_skip_subsampling</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_subsampling_fraction</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_database</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent_bases_aligned</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_length_unambiguous</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>meanbaseq_trim</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_abundance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_Degenerate</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_N</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>percent_reference_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>vadr_num_alerts</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>min_contig_length</strong></td>
<td>Int</td>
<td>Minimum length of contig for QUAST</td>
<td>500</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa_task</td>
<td><strong>bases</strong></td>
<td>String</td>
<td>Explicitly set the number of bases required e.g., 4.3kb, 7Tb, 9000, 4.1MB. If this option is given, --coverage and --genome-size are ignored</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/rasusa:2.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa_task</td>
<td><strong>frac</strong></td>
<td>Float</td>
<td>Explicitly define the fraction of reads to keep in the subsample; when used, genome size and coverage are ignored; acceptable inputs include whole numbers and decimals, e.g. 50.0 will leave 50% of the reads in the subsample</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa_task</td>
<td><strong>num</strong></td>
<td>Int</td>
<td>Optional: explicitly define the number of reads in the subsample; when used, genome size and coverage are ignored; acceptable metric suffixes include: b, k, m, g, and t for base, kilo, mega, giga, and tera, respectively</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>rasusa_task</td>
<td><strong>seed</strong></td>
<td>Int</td>
<td>Use to assign a name to the "random seed" that is used by the subsampler; i.e. this allows the exact same subsample to be produced from the same input file/s in subsequent runs when providing the seed identifier; do not input values for random downsampling</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>File with adapter sequences to be removed</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the Kraken2 task should be called; for non-TheiaCoV workflows, the <code>kraken_db</code> variable must be provided.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>extract_unclassified</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Additional arguments to use with fastp</td>
<td>--detect_adapter_for_pe -g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_complete_only</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_decontaminate_mem</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_is_accession</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_refseq</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task. Increase this when using large (&gt;30GB kraken2 databases such as the "k2_standard" database)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-files-rp/terra/theiaprok-files/midas/midas_db_v1.2.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>A file containing the phix used during Illumina sequencing; used in the BBDuk task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>The name of the tool to perform basic read processing; options: "trimmomatic" or "fastp"</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>The tool used for quality control (QC) of reads. Options are "fastq_scan" (default) and "fastqc"</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>This string is searched for in the kraken2 outputs to extract the read percentage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>taxon_id</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Additional arguments to pass to trimmomatic. "-phred33" specifies the Phred Q score encoding which is almost always phred33 with modern sequence data.</td>
<td>-phred33</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>workflow_series</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>busco_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ezlabgva/busco:v5.3.2_cv1</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>busco_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>24</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>call_rasusa</strong></td>
<td>Boolean</td>
<td>If true, RASUSA will subsample raw reads to  a specified read depth (150X by default)</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>expected_taxon</strong></td>
<td>String</td>
<td>If provided, this input will override the taxonomic assignment made by GAMBIT and launch the relevant taxon-specific submodules. It will also modify the organism flag used by AMRFinderPlus. Example format: "Salmonella enterica"</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>gambit_db_genomes</strong></td>
<td>File</td>
<td>User-provided database of assembled query genomes; requires complementary signatures file. If not provided, uses default database, "/gambit-db"</td>
<td>gs://gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-metadata-1.0.0-20241213.gdb</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>gambit_db_signatures</strong></td>
<td>File</td>
<td>User-provided signatures file; requires complementary genomes file. If not specified, the file from the docker container will be used.</td>
<td>gs://gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-signatures-1.0.0-20241213.gs</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>max_genome_length</strong></td>
<td>Int</td>
<td>Maximum genome size able to pass read screening</td>
<td>178000000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>min_basepairs</strong></td>
<td>Int</td>
<td>Minimum number of base pairs able to pass read screening</td>
<td>45000000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>min_contig_length</strong></td>
<td>Int</td>
<td>Minimum contig length for assembler</td>
<td>1000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>min_coverage</strong></td>
<td>Int</td>
<td>Minimum genome coverage able to pass read screening</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>min_genome_length</strong></td>
<td>Int</td>
<td>Minimum genome size able to pass read screening</td>
<td>9000000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>min_proportion</strong></td>
<td>Int</td>
<td>Minimum proportion of total reads in each read file to pass read screening</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>min_reads</strong></td>
<td>Int</td>
<td>Minimum number of reads to pass read screening</td>
<td>30000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>qc_check_table</strong></td>
<td>File</td>
<td>TSV value with taxons for rows and QC values for columns; internal cells represent user-determined QC thresholds; if provided, turns on the QC Check task. See below for an example QC Check table.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>seq_method</strong></td>
<td>String</td>
<td>Sequencing method used for the samples</td>
<td>ILLUMINA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>skip_screen</strong></td>
<td>Boolean</td>
<td>Option to skip the read screening prior to analysis; if setting to true, please provide a value for the theiaeuk_pe genome_length optional input, OR set call_rasusa to false. Otherwise RASUSA will attempt to downsample to an expected genome size of 0 bp, and the workflow will fail.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>subsample_coverage</strong></td>
<td>Float</td>
<td>Read depth for RASUSA task to subsample reads to</td>
<td>150</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>trim_min_length</strong></td>
<td>Int</td>
<td>Specifies minimum length of each read after trimming to be kept</td>
<td>75</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>Specifies the minimum average quality of bases in a sliding window to be kept</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>Size of the trimming window to use</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiaeuk_ont</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>ONT read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>busco</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>workflow_series</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>theiaviral</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>auto_medaka_model</strong></td>
<td>Boolean</td>
<td>If true, medaka will automatically select the best Medaka model for assembly</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>bandage_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>bandage_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>bandage_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>dnaapler_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>dnaapler_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>dnaapler_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>dnaapler_mode</strong></td>
<td>String</td>
<td>Dnaapler-specific inputs</td>
<td>all</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>filter_contigs_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>filter_contigs_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>filter_contigs_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>filter_contigs_min_length</strong></td>
<td>Int</td>
<td>Minimum contig length to keep</td>
<td>1000</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_additional_parameters</strong></td>
<td>String</td>
<td>Any extra Flye-specific parameters</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_asm_coverage</strong></td>
<td>Int</td>
<td>Reduced coverage for initial disjointig assembly</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_keep_haplotypes</strong></td>
<td>Boolean</td>
<td>If true keep haplotypes</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_minimum_overlap</strong></td>
<td>Int</td>
<td>Minimum overlap between reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_no_alt_contigs</strong></td>
<td>Boolean</td>
<td>If true, do not generate alternative contigs</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_polishing_iterations</strong></td>
<td>Int</td>
<td>Default polishing iterations</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_read_error_rate</strong></td>
<td>Float</td>
<td>Maximum expected read error rate</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_read_type</strong></td>
<td>String</td>
<td>Specifies the type of sequencing reads. Options: --nano-hq (default), --nano-corr, --nano-raw, --pacbio-raw, --pacbio-corr, --pacbio-hifi. Refer to Flye documentation for details on each type.</td>
<td>--nano-hq</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_scaffold</strong></td>
<td>Boolean</td>
<td>If true, scaffolding is enabled using graph</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_uneven_coverage_mode</strong></td>
<td>Boolean</td>
<td>sets the --meta option in the case of uneven coverage (or metagenomics)</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>illumina_read1</strong></td>
<td>File</td>
<td>If Illumina reads are provided, flye_denovo subworkflow will perform Illumina polishing</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>illumina_read2</strong></td>
<td>File</td>
<td>If Illumina reads are provided, flye_denovo subworkflow will perform Illumina polishing</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>medaka_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>medaka_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>medaka_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>medaka_model</strong></td>
<td>String</td>
<td>In order to obtain the best results, the appropriate model must be set to match the sequencer's basecaller model; this string takes the format of {pore}{device}{caller variant}_{caller_version}. See also <a href="https://github.com/nanoporetech/medaka?tab=readme-ov-file#models">https://github.com/nanoporetech/medaka?tab=readme-ov-file#models</a>. If this is being run on legacy data it is likely to be r941_min_hac_g507.</td>
<td>r1041_e82_400bps_sup_v5.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polish_rounds</strong></td>
<td>Int</td>
<td>The number of polishing rounds to conduct for medaka or racon (without Illumina)</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polisher</strong></td>
<td>String</td>
<td>The polishing tool to use for assembly</td>
<td>medaka</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_careful</strong></td>
<td>Boolean</td>
<td>Polypolish-specific inputs</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_fraction_invalid</strong></td>
<td>Float</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_fraction_valid</strong></td>
<td>Float</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_high_percentile_threshold</strong></td>
<td>Float</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_low_percentile_threshold</strong></td>
<td>Float</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_maximum_errors</strong></td>
<td>Int</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_minimum_depth</strong></td>
<td>Int</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_pair_orientation</strong></td>
<td>String</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>porechop_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>porechop_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>porechop_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>porechop_trimopts</strong></td>
<td>String</td>
<td>Options to pass to Porechop for trimming</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>racon_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>racon_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>racon_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>run_porechop</strong></td>
<td>Boolean</td>
<td>If true, trims reads before assembly using Porechop</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>skip_polishing</strong></td>
<td>Boolean</td>
<td>If true, skips polishing</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_agr_typing_only</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/amrsearch:0.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_poppunk</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_shigeifinder_reads_input</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_stxtyper</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_tbp_parser</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cauris_cladetyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_docker_image</strong></td>
<td>String</td>
<td>Docker image to use for the task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_memory</strong></td>
<td>Int</td>
<td>Amount of memory (in GB) to allocate to the task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum percent coverage of FungAMR reference protein</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum percent identity of FungAMR reference protein</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroqutas_translation_code</strong></td>
<td>String</td>
<td>Alternative codon translation code. Set for each species by default and not recommended to change</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_kmer_size</strong></td>
<td>Int</td>
<td>Kmer size for cladetyper</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_max_distance</strong></td>
<td>Float</td>
<td>The maximum GAMBIT distance to report a <em>C. auris</em> clade hit</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>Candidozyma auris</em> clade1</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade1_GCA_002759435.2_Cand_auris_B8441_V2_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade1</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade1_GCA_002759435_Cauris_B8441_V2_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade2</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade2_GCA_003013715.2_ASM301371v2_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade2</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade2_GCA_003013715.2_ASM301371v2_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade3</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade3_GCF_002775015.1_Cand_auris_B11221_V1_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade3</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade3_GCF_002775015.1_Cand_auris_B11221_V1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade4</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade4_GCA_003014415.1_Cand_auris_B11243_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade4</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade4_GCA_003014415.1_Cand_auris_B11243_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade5</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade5_GCA_016809505.1_ASM1680950v1_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade5</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade5_GCA_016809505.1_ASM1680950v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade6</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade6_GCA_032714025.1_ASM3271402v1_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade6</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>clockwork_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_print_alleles</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_verify</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_align_diff</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_cluster_distance</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_culling_limit</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_gap</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_max_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_good</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_perfect</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_mismatch</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_wf</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtypingtool_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>genotyphi_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_broken_gene_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_broken_gene_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_low_gene_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_start_end_margin</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_kaptive_confidence</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_kaptive</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_resistance</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>legsta_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/legsta:0.5.1--hdfd78af_2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>meningotype_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ngmaster_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>paired_end</strong></td>
<td>Boolean</td>
<td>Set to true if your data is paired-end FASTQ files</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_clusters_csv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_npy</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_pkl</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_external_clusters_csv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_npz</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_pkl</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_graph_gt</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_h5</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_qcreport_txt</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_npy</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_pkl</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_graph_gt</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_h5</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_unword_clusters_csv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>run_amr_search</strong></td>
<td>Boolean</td>
<td>If set to true AMR_Search workflow will be run if species is part of supported taxon, see AMR_Search docs.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seqsero2_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seroba_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>serotypefinder_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigatyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigeifinder_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_use_full_cgmlst_db</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_base_quality</strong></td>
<td>Int</td>
<td>Minimum quality for a nucleotide to be used in variant calling</td>
<td>13</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_gene_query_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_map_qual</strong></td>
<td>Int</td>
<td>Minimum mapping quality to accept in variant calling</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_maxsoft</strong></td>
<td>Int</td>
<td>Number of bases of alignment to soft-clip before discarding the alignment</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_coverage</strong></td>
<td>Int</td>
<td>Minimum read coverage of a position to identify a mutation</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_frac</strong></td>
<td>Float</td>
<td>Minimum fraction of bases at a given position to identify a mutation</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_quality</strong></td>
<td>Int</td>
<td>Minimum VCF variant call "quality"</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_query_gene</strong></td>
<td>String</td>
<td>Provide a gene to search for using Snippy</td>
<td>Default depend on detected organism</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_afumigatus</strong></td>
<td>File</td>
<td>Snippy reference for Aspergillus fumigatus</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/aspergillus/Aspergillus_fumigatus_GCF_000002655.1_ASM265v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_cryptoneo</strong></td>
<td>File</td>
<td>Snippy reference for Cryptococcus neoformans</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/cryptococcus/Cryptococcus_neoformans_GCF_000091045.1_ASM9104v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_variants_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snippy:4.6.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_mykrobe_opts</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_do_enrich</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_gene_max_mismatch</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>2000</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_max_divergence</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_depth</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_edge_depth</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_percent_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>staphopia_sccmec_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_enable_debug</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_add_cs_lims</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_config</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_coverage_regions_bed</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_debug</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_etha237_frequency</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_expert_rule_regions_bed</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_depth</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_frequency</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_read_support</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_operator</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_output_seq_method_type</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>WGS</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rpob449_frequency</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_frequency</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_read_support</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_frequency</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_read_support</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_tngs_data</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_additional_parameters</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_custom_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_mapper</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_af</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_depth</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_cdph_db</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_custom_db</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_caller</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_calling_params</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_lineage_barcodes</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_skip_subsampling</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_subsampling_fraction</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_database</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_identity</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>min_contig_length</strong></td>
<td>Int</td>
<td>Minimum length of contig for QUAST</td>
<td>500</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>workflow_series</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>theiaviral</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_docker</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>downsampling_coverage</strong></td>
<td>Float</td>
<td>The desired coverage to sub-sample the reads to with RASUSA</td>
<td>150</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_docker</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/nanoq:0.9.0--hec16e2b_1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_max_read_length</strong></td>
<td>Int</td>
<td>The maximum read length to keep after trimming</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_max_read_qual</strong></td>
<td>Int</td>
<td>The maximum read quality to keep after trimming</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_min_read_length</strong></td>
<td>Int</td>
<td>The minimum read length to keep after trimming</td>
<td>500</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_min_read_qual</strong></td>
<td>Int</td>
<td>The minimum read quality to keep after trimming</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_docker</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_bases</strong></td>
<td>String</td>
<td>Explicitly set the number of bases required e.g., 4.3kb, 7Tb, 9000, 4.1MB. If this option is given, --coverage and --genome-size are ignored</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_docker</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_fraction_of_reads</strong></td>
<td>Float</td>
<td>Subsample to a fraction of the reads - e.g., 0.5 samples half the reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_number_of_reads</strong></td>
<td>Int</td>
<td>Subsample to a specific number of reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_seed</strong></td>
<td>Int</td>
<td>Random seed to use</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>run_prefix</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>busco_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ezlabgva/busco:v5.3.2_cv1</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>busco_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>24</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>gambit_db_genomes</strong></td>
<td>File</td>
<td>User-provided database of assembled query genomes; requires complementary signatures file. If not provided, uses default database, "/gambit-db"</td>
<td>gs://gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-metadata-1.0.0-20241213.gdb</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>gambit_db_signatures</strong></td>
<td>File</td>
<td>User-provided signatures file; requires complementary genomes file. If not specified, the file from the docker container will be used.</td>
<td>gs://gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-signatures-1.0.0-20241213.gs</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td>50000000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>max_genome_length</strong></td>
<td>Int</td>
<td>Maximum genome size able to pass read screening</td>
<td>178000000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>min_basepairs</strong></td>
<td>Int</td>
<td>Minimum number of base pairs able to pass read screening</td>
<td>45000000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>min_coverage</strong></td>
<td>Int</td>
<td>Minimum genome coverage able to pass read screening</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>min_genome_length</strong></td>
<td>Int</td>
<td>Minimum genome size able to pass read screening</td>
<td>9000000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>min_reads</strong></td>
<td>Int</td>
<td>Minimum number of reads to pass read screening</td>
<td>5000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>skip_mash</strong></td>
<td>Boolean</td>
<td>If true, skips estimation of genome size and coverage using mash in read screening steps. As a result, providing true also prevents screening using these parameters.</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>skip_screen</strong></td>
<td>Boolean</td>
<td>Option to skip the read screening prior to analysis; if setting to true, please provide a value for the theiaeuk_pe genome_length optional input, OR set call_rasusa to false. Otherwise RASUSA will attempt to downsample to an expected genome size of 0 bp, and the workflow will fail.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>workflow_series</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>theiaeuk</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<h3 id="workflow-tasks">Workflow Tasks<a class="headerlink" href="#workflow-tasks" title="Permanent link"></a></h3>
<p>All input reads are processed through "core tasks" in the TheiaEuk workflows. These undertake read trimming and assembly appropriate to the input data type. TheiaEuk workflow subsequently launch default genome characterization modules for quality assessment, and additional taxa-specific characterization steps. When setting up the workflow, users may choose to use "optional tasks" or alternatives to tasks run in the workflow by default.</p>
<h4 id="core-tasks">Core tasks<a class="headerlink" href="#core-tasks" title="Permanent link"></a></h4>
<div class="admonition dna">
<p>These tasks are performed regardless of organism. They include tasks that are performed regardless of and specific for the input data type. They perform read trimming and assembly appropriate to the input data type.</p>
</div>
<details class="task">
<summary><code>versioning</code>: Version Capture</summary>
<p>The <code>versioning</code> task captures the workflow version from the GitHub (code repository) version.</p>
<div class="admonition techdetails">
<p class="admonition-title">Version Capture Technical details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/task_versioning.wdl">task_versioning.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition caption">
<div class="tabbed-set tabbed-alternate" data-tabs="4:2"><input checked="checked" id="theiaeuk_illumina_pe_3" name="__tabbed_4" type="radio"><input id="theiaeuk_ont_3" name="__tabbed_4" type="radio"><div class="tabbed-labels"><label for="theiaeuk_illumina_pe_3">TheiaEuk_Illumina_PE</label><label for="theiaeuk_ont_3">TheiaEuk_ONT</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<details class="task">
<summary><code>screen</code>: Total Raw Read Quantification and Genome Size Estimation</summary>
<p>The <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl"><code>screen</code></a> task ensures the quantity of sequence data is sufficient to undertake genomic analysis. It uses <a href="https://github.com/rpetit3/fastq-scan"><code>fastq-scan</code></a> and bash commands for quantification of reads and base pairs, and <a href="https://mash.readthedocs.io/en/latest/index.html">mash</a> sketching to estimate the genome size and its coverage. At each step, the results are assessed relative to pass/fail criteria and thresholds that may be defined by optional user inputs. Samples are run through all threshold checks, regardless of failures, and the workflow will terminate after the <code>screen</code> task if any thresholds are not met:</p>
<ol>
<li>Total number of reads: A sample will fail the read screening task if its total number of reads is less than or equal to <code>min_reads</code>.</li>
<li>The proportion of basepairs reads in the forward and reverse read files: A sample will fail the read screening if fewer than <code>min_proportion</code> basepairs are in either the reads1 or read2 files.</li>
<li>Number of basepairs: A sample will fail the read screening if there are fewer than <code>min_basepairs</code> basepairs</li>
<li>Estimated genome size:  A sample will fail the read screening if the estimated genome size is smaller than <code>min_genome_size</code> or bigger than <code>max_genome_size</code>.</li>
<li>Estimated genome coverage: A sample will fail the read screening if the estimated genome coverage is less than the <code>min_coverage</code>.</li>
</ol>
<p>Read screening is undertaken on both the raw and cleaned reads. The task may be skipped by setting the <code>skip_screen</code> variable to true.</p>
<p>Default values vary between the PE, SE, and ONT workflows. The rationale for these default values can be found below. If two default values are shown, the first is for Illumina workflows and the second is for ONT.</p>
<p>| Variable  | Rationale |
| --- | --- | --- |
| <code>skip_screen</code> | false | Set to true to skip the read screen from running. If you set this value to true, please provide a value for the theiaeuk_illumina_pe <code>genome_length</code> optional input, OR set the theiaeuk_illumina_pe <code>call_rasusa</code> optional input to false. Otherwise RASUSA will attempt to downsample to an expected genome size of 0 bp, and the workflow will fail. |
| <code>min_reads</code> | 3000 | Calculated from the minimum number of base pairs required for 20x coverage of the <em>Hansenula polymorpha</em> genome, the smallest fungal genome as of 2015-04-02 (8.97 Mbp), divided by 300 (the longest Illumina read length) |
| <code>min_basepairs</code> | 45000000 | Should be greater than 10x coverage of <em>Hansenula polymorpha</em>, the smallest fungal genome as of 2015-04-02 (8.97 Mbp)  |
| <code>min_genome_length</code> | 9000000 | Based on the <em>Hansenula polymorpha</em>  genome - the smallest fungal genome as of 2015-04-02 (8.97 Mbp) |
| <code>max_genome_length</code> | 178000000 | Based on the <em>Cenococcum geophilum</em>  genome, the largest pathogenic fungal genome (177.57 Mbp), plus an additional 2 Mbp to cater for potential extra genomic material |
| <code>min_coverage</code> | 10 | A bare-minimum average per base coverage across the genome required for genome characterization. Higher coverage would be required for high-quality phylogenetics.|
| <code>min_proportion</code> | 40 | Neither read1 nor read2 files should have less than 40% of the total number of reads. For paired-end data only. |</p>
<div class="admonition techdetails">
<p class="admonition-title">Screen Technical Details</p>
<p>There is a single WDL task for read screening. The <code>screen</code> task is run twice, once for raw reads and once for clean reads.</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L3">task_screen.wdl (PE sub-task)</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L147">task_screen.wdl (SE sub-task)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Rasusa</code>: Read subsampling (optional, on by default)</summary>
<p>To deactivate this task, set <code>call_rasusa</code> to <code>false</code>.</p>
<p><code>Rasusa</code> is a tool to randomly subsample sequencing reads to a specified coverage without assuming that all reads are of equal length, making it especially suitable for long-read data while still being applicable to short-read data.</p>
<p>The <code>Rasusa</code> task performs subsampling on the input raw reads. By default, this task will subsample TheiaProk_ONT reads to a depth of 150X using an estimated genome length of 5 million basepairs (0.7 Mb higher than the average bacterial genome length), and TheiaEuk_ONT reads using an estimated genome length of 50 million basepairs. The estimated genome length can be changed by the user by providing a different value for the <code>genome_length</code> input parameter. The target subsampling depth can also be adjusted by modifying the <code>subsample_coverage</code> variable.</p>
<p>For TheiaEuk_Illumina_PE, the estimated genome length is determined by the <code>read_screen</code> task. Please note that the user can prevent the task from being launched by setting the <code>call_rasusa</code> variable to false.</p>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs since it performs <em>random</em> subsampling. To ensure reproducibility, set a a value for the <code>rasusa_seed</code> optional input variable.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Rasusa Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/task_rasusa.wdl">task_rasusa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/mbhall88/rasusa">Rasusa on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/mbhall88/rasusa">Rasusa on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03941">Rasusa: Randomly subsample sequencing reads to a specified coverage</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>read_QC_trim</code>: Read Quality Trimming, Adapter Removal, Quantification, and Identification</summary>
<p><code>read_QC_trim</code> is a sub-workflow that removes low-quality reads, low-quality regions of reads, and sequencing adapters to improve data quality. It uses a number of tasks, described below. The differences between the PE and SE versions of the <code>read_QC_trim</code> sub-workflow lie in the default parameters, the use of two or one input read file(s), and the different output files.</p>
<div class="admonition dna">
<p>By default, <code>read_processing</code> is set to <code>"trimmomatic"</code>. To use <code>fastp</code> instead, set <code>read_processing</code> to <code>"fastp"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>Trimmomatic</code>: Read Trimming (default)</summary>
<p>Read proccessing is available via <code>Trimmomatic</code> by default.</p>
<p>Trimmomatic trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>Trimmomatic</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_trimmomatic.wdl">task_trimmomatic.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/usadellab/Trimmomatic">Trimmomatic on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic Website</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590/">Trimmomatic: a flexible trimmer for Illumina sequence data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>fastp</code>: Read Trimming (alternative)</summary>
<p>To activate this task, set <code>read_processing</code> to <code>"fastp"</code>.</p>
<p><code>fastp</code> trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<p><code>fastp</code> also has additional default parameters and features that are not a part of <code>trimmomatic</code>'s default configuration.</p>
<details class="toggle">
<summary><code>fastp</code> default read-trimming parameters</summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-g</td>
<td>enables polyG tail trimming</td>
</tr>
<tr>
<td>-5 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>-3 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>--detect_adapter_for_pe</td>
<td>enables adapter-trimming <strong>only for paired-end reads</strong></td>
</tr>
</tbody>
</table>
<p>Additional arguments can be passed using the <code>fastp_args</code> optional parameter.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Trimmomatic and fastp Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_fastp.wdl">task_fastp.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/17/i884/5093234?login=false">fastp: an ultra-fast all-in-one FASTQ preprocessor</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>BBDuk</code>: Adapter Trimming and PhiX Removal</summary>
<p>Adapters are manufactured oligonucleotide sequences attached to DNA fragments during the library preparation process. In Illumina sequencing, these adapter sequences are required for attaching reads to flow cells. You can read more about <a href="https://emea.support.illumina.com/bulletins/2020/06/illumina-adapter-portfolio.html">Illumina adapters here</a>. For genome analysis, it's important to remove these sequences since they're not actually from your sample. If you don't remove them, the downstream analysis may be affected.</p>
<p>The <code>bbduk</code> task removes adapters from sequence reads. To do this:</p>
<ul>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/repair-guide/">Repair</a> from the <a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/">BBTools</a> package reorders reads in paired fastq files to ensure the forward and reverse reads of a pair are in the same position in the two fastq files (it <em>re-pairs</em>).</li>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk</a>  (<em>"Bestus Bioinformaticus" Decontamination Using Kmers</em>) is then used to trim the adapters and filter out all reads that have a 31-mer match to <a href="https://emea.illumina.com/products/by-type/sequencing-kits/cluster-gen-sequencing-reagents/phix-control-v3.html">PhiX</a>, which is commonly added to Illumina sequencing runs to monitor and/or improve overall run quality.</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">BBDuk Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_bbduk.wdl">task_bbduk.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://sourceforge.net/projects/bbmap/">BBMap on SourceForge</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk Guide (archived)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_qc</code> is set to <code>"fastq_scan"</code>. To use <code>fastqc</code> instead, set <code>read_qc</code> to <code>"fastqc"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>fastq-scan</code>: Read Quantification (default)</summary>
<p>Read quantification is available via <code>fastq-scan</code> by default.</p>
<p><code>fastq-scan</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>fastq-scan</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastq_scan.wdl">task_fastq_scan.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/fastq-scan/blob/master/README.md">fastq-scan on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>FastQC</code>: Read Quantification (alternative)</summary>
<p>To activate this task, set <code>read_qc</code> to <code>"fastqc"</code>.</p>
<p><code>FastQC</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>This tool also provides a graphical visualization of the read quality.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>FastQC</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastqc.wdl">task_fastqc.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/s-andrews/FastQC">FastQC on Github</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC Website</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>Kraken2</code>: Read Identification (optional)</summary>
<p>To activate this task, set <code>call_kraken</code> to <code>true</code> and provide a value for <code>kraken_db</code>.</p>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<p>As an alternative to <code>MIDAS</code> (see above), the <code>Kraken2</code> task can also be turned on through setting the <code>call_kraken</code> input variable as <code>true</code> for the identification of reads to detect contamination with non-target taxa.</p>
<p>A database must be provided if this optional module is activated, through the kraken_db optional input. A list of suggested databases can be found on <a href="../../standalone/kraken2/#databases">Kraken2 standalone documentation</a>.</p>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">Kraken2 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_pe.wdl">wf_read_QC_trim_pe.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_se.wdl">wf_read_QC_trim_se.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>qc_check</code>: Check QC Metrics Against User-Defined Thresholds (optional)</summary>
<p>To activate this task, provide a <code>qc_check_table</code> as input.</p>
<p>The <code>qc_check</code> task compares generated QC metrics against user-defined thresholds for each metric. This task will run if the user provides a <code>qc_check_table</code> TSV file. If all QC metrics meet the threshold, the <code>qc_check</code> output variable will read <code>QC_PASS</code>. Otherwise, the output will read <code>QC_NA</code> if the task could not proceed or <code>QC_ALERT</code> followed by a string indicating what metric failed.</p>
<p>The <code>qc_check</code> task applies quality thresholds according to the sample taxa. The sample taxa is taken from the <code>gambit_predicted_taxon</code> value inferred by the GAMBIT module OR can be manually provided by the user using the <code>expected_taxon</code> workflow input.</p>
<details class="toggle">
<summary>Formatting the <em>qc_check_table.tsv</em></summary>
<ul>
<li>The first column of the qc_check_table lists the <code>organism</code> that the task will assess and the header of this column must be "<strong>taxon</strong>".</li>
<li>Any genus or species can be included as a row of the qc_check_table. However, these taxa must <strong>uniquely</strong> match the sample taxa, meaning that the file can include multiple species from the same genus (Vibrio_cholerae and Vibrio_vulnificus), but not both a genus row and species within that genus (Vibrio and Vibrio cholerae). <strong>The taxa should be formatted with the first letter capitalized and underscores in lieu of spaces.</strong></li>
<li>Each subsequent column indicates a QC metric and lists a threshold for each organism that will be checked. <strong>The column names must exactly match expected values, so we highly recommend copy and pasting the header from the template file below as a starting place.</strong></li>
</ul>
</details>
<details class="toggle">
<summary>Template <em>qc_check_table.tsv</em> files</summary>
<ul>
<li>TheiaEuk_Illumina_PE_PHB: <a href="../../../assets/files/TheiaEuk_qc_check_template.tsv">theiaeuk_qc_check_template.tsv</a></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Example Purposes Only</p>
<p>The QC threshold values shown in the file above are for example purposes only and should not be presumed to be sufficient for every dataset.</p>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">qc_check Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_qc_check_phb.wdl">task_qc_check_phb.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>These tasks assemble the reads into a <em>de novo</em> assembly and assess the quality of the assembly.</p>
</div>
<details class="task">
<summary><code>digger_denovo</code>: <em>De novo</em> Assembly</summary>
<p><em>De novo</em>  assembly is the process or product of attempting to reconstruct a genome from scratch (without prior knowledge of the genome) using sequence reads. Assembly of fungal genomes from short-reads will produce multiple contigs per chromosome rather than a single contiguous sequence for each chromosome.</p>
<p>In TheiaProk and TheiaEuk Illumina workflows, <em>de novo</em> assembly is performed for samples that have sufficient read quantity and quality using <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_digger_denovo.wdl">digger_denovo</a>, a subworkflow based off of <a href="https://github.com/tseemann/shovill">Shovill</a> pipeline. The name "digger" is a nod to Shovill and SPAdes.</p>
<details class="toggle">
<summary><em>De novo</em> Assembly</summary>
<div class="admonition dna">
<p class="admonition-title"><code>assembler</code> with <code>skesa</code> (default), <code>spades</code>, or <code>megahit</code></p>
<p>To activate a particular assembler, set the <code>assembler</code> input parameter to either <code>skesa</code> (default), <code>spades</code>, or <code>megahit</code>.</p>
<p>These tasks are mutually exclusive.</p>
</div>
<details class="task">
<summary><code>SKESA</code>: <em>De novo</em> Assembly (default)</summary>
<p>This task is activated by default.</p>
<p><code>SKESA</code> (Strategic K-mer Extension for Scrupulous Assemblies) is a <em>de novo</em> assembler that is fairly conservative and introduces breaks in the genome at repeat regions. This leads to higher sequence quality but more fragmented assemblies, which, depending on the final analysis goal, can be either highly preferred or detrimental. Designed for Illumina reads and haploid genomes, SKESA is the default assembler in the <code>digger_denovo</code> subworkflow.</p>
<div class="admonition techdetails">
<p class="admonition-title">SKESA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_skesa.wdl">task_skesa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/SKESA">SKESA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/SKESA">SKESA on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1186/s13059-018-1540-z">SKESA: strategic k-mer externsion for scrupulous assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>SPAdes</code>: <em>De novo</em> Assembly (alternative)</summary>
<p>To activate this task, set <code>assembler</code> to <code>spades</code>.</p>
<p><code>SPAdes</code> (St. Petersburg genome assembler) is a <em>de novo</em> assembly tool that uses de Bruijn graphs to assemble genomes from Illumina short reads.</p>
<p>In TheiaProk, SPAdes is run in <code>--isolate</code> mode, which is the recommended flag for high-coverage isolate and multi-cell Illumina data, which is typical of most bacterial sequencing projects. This method is optimized for improving assembly quality and decreasing runtime.</p>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">MetaviralSPAdes Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_spades.wdl">task_spades.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ablab/spades">SPAdes on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ablab.github.io/spades/index.html">SPAdes Manual</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>TheiaProk</em>: <a href="https://doi.org/10.1089/cmb.2012.0021">SPAdes: A New Genome Assembly Algorithm and Its Applications to Single-Cell Sequencing</a><br><em>TheiaViral</em>: <a href="https://doi.org/10.1093/bioinformatics/btaa490">MetaviralSPAdes: assembly of viruses from metagenomic data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>MEGAHIT</code>: <em>De novo</em> Assembly (alternative)</summary>
<p>To activate this task, set <code>assembler</code> to <code>megahit</code>.</p>
<p>The <a href="https://github.com/voutcn/megahit">MEGAHIT assembler</a> is a fast and memory-efficient <em>de novo</em> assembler that can handle large datasets. While optimized for metagenomics, MEGAHIT also performs well on single-genome assemblies, making it a versatile choice for various assembly tasks.</p>
<p>MEGAHIT uses a multiple k-mer strategy that can be beneficial for assembling genomes with varying coverage levels, which is common in metagenomic samples. It constructs succinct de Bruijn graphs to efficiently represent the assembly process, allowing it to handle large and complex datasets with reduced memory usage.</p>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">MEGAHIT Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_megahit.wdl">task_megahit.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/voutcn/megahit">MEGAHIT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/voutcn/megahit/blob/master/README.md">MEGAHIT on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btv033">MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Assembly Polishing (optional)</summary>
<p>To activate assembly polishing, set <code>call_pilon</code> to <code>true</code>.</p>
<details class="task">
<summary><code>bwa</code>: Read Alignment to the Assembly</summary>
<p>BWA (Burrow-Wheeler Aligner) is used to align the cleaned read files to generated assembly file in order to generate an alignment. The resulting BAM file is directly passed to the Pilon task to polish the assembly for errors.</p>
<div class="admonition techdetails">
<p class="admonition-title">BWA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_bwa.wdl">task_bwa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/bwa">BWA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bio-bwa.sourceforge.net/">BWA Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and accurate short read alignment with Burrows-Wheeler transform</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Pilon</code>: Assembly Polishing</summary>
<p><code>Pilon</code> is a tool that uses read alignments to correct errors in an assembly.</p>
<p>The <code>bwa</code>-generated alignment of the read data to the assembly is used to identify inconsistences between the reads and the assembly in order to correct them. <code>Pilon</code> will attempt to fix individual base errors and small indels using the read data. This can improve the overall quality of the assembly, especially when the assembler has made mistakes due to sequencing errors or low coverage regions.</p>
<p>The default parameters were set to mimic the parameters used by <a href="https://github.com/tseemann/shovill">Shovill</a>: <code>--fix bases --minq 60 --minqual 3 --mindepth 0.25</code>. These can be modified by the user.</p>
<div class="admonition techdetails">
<p class="admonition-title">Pilon Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_pilon.wdl">task_pilon.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/broadinstitute/pilon">Pilon on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/broadinstitute/pilon/wiki">Pilon Wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1371/journal.pone.0112963">Pilon: An Integrated Tool for Comprehensive Microbial Variant Detection and Genome Assembly Improvement</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Contig Filtering (optional)</summary>
<details class="task">
<summary><code>Filter Contigs</code>: Contig Quality Control</summary>
<p>To <strong><em>de</em></strong>activate contig filtering, set <code>run_filter_contigs</code> to <code>false</code>.</p>
<p>This task filters the created contigs based on a default minimum length threshold of 200 bp and a minimum coverage of 2.0. It also eliminates homopolymer contigs (contigs of any length that consist of a single nucleotide).</p>
<p>Options are available to skip any of these filters by setting the respective parameters to <code>false</code>: <code>filter_contigs_skip_length_filter</code>, <code>filter_contigs_skip_coverage_filter</code>, and <code>filter_contigs_skip_homopolymer_filter</code>. The minimum length and coverage thresholds can be adjusted using the <code>filter_contigs_min_length</code> and <code>filter_contigs_min_coverage</code> parameters, respectively.</p>
<p>This ensures high-quality assemblies by retaining only contigs that meet specified criteria. Detailed metrics on contig counts and sequence lengths before and after filtering are provided in the output.</p>
<div class="admonition techdetails">
<p class="admonition-title">Filter Contigs Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_filter_contigs.wdl">task_filter_contigs.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Digger-Denovo Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_digger_denovo.wdl">wf_digger_denovo.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>quast</code>: Assembly Quality Assessment</summary>
<p>QUAST stands for QUality ASsessment Tool. It evaluates genome/metagenome assemblies by computing various metrics without a reference being necessary. It includes useful metrics such as number of contigs, length of the largest contig and N50.</p>
<div class="admonition techdetails">
<p class="admonition-title">QUAST Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_quast.wdl">task_quast.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ablab/quast">QUAST on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://quast.sourceforge.net/">QUAST Manual on SourceForge</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/29/8/1072/228832">QUAST: quality assessment tool for genome assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>CG-Pipeline</code>: Assessment of Read Quality, and Estimation of Genome Coverage</summary>
<p>The<code>cg_pipeline</code> task generates metrics about read quality and estimates the coverage of the genome using the <code>run_assembly_readMetrics.pl</code> script from <a href="https://github.com/lskatz/CG-Pipeline/">CG-Pipeline</a>. The genome coverage estimates are calculated using both using raw and cleaned reads, using either a user-provided <code>genome_size</code> or the estimated genome length generated by QUAST.</p>
<div class="admonition techdetails">
<p class="admonition-title">CG-Pipeline Technical Details</p>
<p>The <code>cg_pipeline</code> task is run twice in this workflow, once with raw reads, and once with clean reads.</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_cg_pipeline.wdl">task_cg_pipeline.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lskatz/CG-Pipeline/">CG-Pipeline on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/lskatz/CG-Pipeline/">CG-Pipeline on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/26/15/1819/188418">A computational genomics pipeline for prokaryotic sequencing projects</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>read_QC_trim_ont</code>: Read Quality Trimming, Quantification, and Identification</summary>
<p><code>read_QC_trim_ont</code> is a sub-workflow that filters low-quality reads and trims low-quality regions of reads. It uses several tasks, described below.</p>
<div class="admonition dna">
<p class="admonition-title">A note on estimated genome length</p>
<p>By default, the estimated genome length is set to 5 Mb, which is around 0.7 Mb higher than the average bacterial genome length, according to <a href="https://github.com/CDCgov/phoenix/blob/717d19c19338373fc0f89eba30757fe5cfb3e18a/assets/databases/NCBI_Assembly_stats_20240124.txt">the information of thousands of NCBI bacterial assemblies collated here</a>. This estimate can be overwritten by the user and is used by <code>Rasusa</code>.</p>
</div>
<details class="task">
<summary><code>Rasusa</code>: Read Subsampling</summary>
<p><code>Rasusa</code> is a tool to randomly subsample sequencing reads to a specified coverage without assuming that all reads are of equal length, making it especially suitable for long-read data while still being applicable to short-read data.</p>
<p>The <code>Rasusa</code> task performs subsampling on the input raw reads. By default, this task will subsample TheiaProk_ONT reads to a depth of 150X using an estimated genome length of 5 million basepairs (0.7 Mb higher than the average bacterial genome length), and TheiaEuk_ONT reads using an estimated genome length of 50 million basepairs. The estimated genome length can be changed by the user by providing a different value for the <code>genome_length</code> input parameter. The target subsampling depth can also be adjusted by modifying the <code>subsample_coverage</code> variable.</p>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs since it performs <em>random</em> subsampling. To ensure reproducibility, set a a value for the <code>rasusa_seed</code> optional input variable.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Rasusa Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/task_rasusa.wdl">task_rasusa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/mbhall88/rasusa">Rasusa on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/mbhall88/rasusa">Rasusa on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03941">Rasusa: Randomly subsample sequencing reads to a specified coverage</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Nanoq</code>: Read Filtering</summary>
<p>Reads are filtered by length and quality using <code>nanoq</code>. By default, sequences with less than 500 basepairs and quality scores lower than 10 are filtered out to improve assembly accuracy. These defaults are able to be modified by the user.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nanoq Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_nanoq.wdl">task_nanoq.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/esteinig/nanoq">Nanoq on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/esteinig/nanoq/blob/master/README.md">Nanoq Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.02991">Nanoq: ultra-fast quality control for nanopore reads</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Kraken2</code>: Read Identification (optional)</summary>
<p>To activate this task, set <code>call_kraken</code> to <code>true</code> and provide a value for <code>kraken_db</code>.</p>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<p>Kraken2 is run on the raw read data.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<p>A database must be provided if this optional module is activated, through the kraken_db optional input. A list of suggested databases can be found on <a href="../../standalone/kraken2/#databases">Kraken2 standalone documentation</a>.</p>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">Kraken2 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>NanoPlot</code>: Read Quantification</summary>
<p>NanoPlot is used for the determination of mean quality scores, read lengths, and number of reads. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>While this task currently is run <em>outside</em> of the <code>read_QC_trim_ont</code> workflow, it is being included here as it calculates statistics on the read data. This is done so that the actual assembly genome lengths can be used (if an estimated genome length is not provided by the user) to ensure the estimated coverage statistics are accurate.</p>
<div class="admonition techdetails">
<p class="admonition-title">NanoPlot Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_nanoplot.wdl">task_nanoplot.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/wdecoster/NanoPlot">NanoPlot on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/wdecoster/NanoPlot/blob/master/README.md">NanoPlot Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/39/5/btad311/7160911">NanoPack2: population-scale evaluation of long-read sequencing data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim_ont Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_ont.wdl">wf_read_QC_trim_ont.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>These tasks assemble the reads into a <em>de novo</em> assembly and assess the quality of the assembly.</p>
</div>
<details class="task">
<summary><code>Flye</code>: <em>De novo</em> Assembly</summary>
<p><code>flye_denovo</code> is a sub-workflow that performs <em>de novo</em> assembly using Flye for ONT data and supports additional polishing and visualization steps.</p>
<div class="admonition tip">
<p class="admonition-title">Ensure correct medaka model is selected if performing medaka polishing</p>
<p>In order to obtain the best results, the appropriate model must be set to match the sequencer's basecaller model; this string takes the format of {pore}_{device}_{caller variant}_{caller_version}. See also <a href="https://github.com/nanoporetech/medaka?tab=readme-ov-file#models">https://github.com/nanoporetech/medaka?tab=readme-ov-file#models</a>. If <code>flye</code> is being run on legacy data the medaka model will likely be <code>r941_min_hac_g507</code>. Recently generated data will likely be suited by the default model of <code>r1041_e82_400bps_sup_v5.0.0</code>.</p>
</div>
<p>The detailed steps and tasks are as follows:</p>
<details class="toggle">
<summary><code>Porechop</code>: Read Trimming (optional; off by default)</summary>
<p>Read trimming is optional and can be enabled by setting the <code>run_porchop</code> input variable to true.</p>
<p>Porechop is a tool for finding and removing adapters from ONT data. Adapters on the ends of reads are trimmed, and when a read has an adapter in the middle, the read is split into two.</p>
<div class="admonition techdetails">
<p class="admonition-title">Porechop Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_porechop.wdl">task_porechop.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rrwick/Porechop">Porechop on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rrwick/Porechop#porechop">https://github.com/rrwick/Porechop#porechop</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Flye</code>: <em>De novo</em> Assembly</summary>
<p>Flye is a <em>de novo</em> assembler for long read data using repeat graphs. Compared to de Bruijn graphs, which require exact k-mer matches, repeat graphs can use approximate matches which better tolerates the error rate of ONT data.</p>
<details class="dna">
<summary><code>flye_read_type</code> input parameter</summary>
<p>This input parameter specifies the type of sequencing reads being used for assembly. This parameter significantly impacts the assembly process and should match the characteristics of your input data. Below are the available options:</p>
<table>
<thead>
<tr>
<th><strong>Parameter</strong></th>
<th><strong>Explanation</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--nano-hq</code> (default)</td>
<td>Optimized for ONT high-quality reads, such as Guppy5+ SUP or Q20 (&lt;5% error). Recommended for ONT reads processed with Guppy5 or newer</td>
</tr>
<tr>
<td><code>--nano-raw</code></td>
<td>For ONT regular reads, pre-Guppy5 (&lt;20% error)</td>
</tr>
<tr>
<td><code>--nano-corr</code></td>
<td>ONT reads corrected with other methods (&lt;3% error)</td>
</tr>
<tr>
<td><code>--pacbio-raw</code></td>
<td>PacBio regular CLR reads (&lt;20% error)</td>
</tr>
<tr>
<td><code>--pacbio-corr</code></td>
<td>PacBio reads corrected with other methods (&lt;3% error)</td>
</tr>
<tr>
<td><code>--pacbio-hifi</code></td>
<td>PacBio HiFi reads (&lt;1% error)</td>
</tr>
</tbody>
</table>
<p>Refer to the Flye documentation for detailed guidance on selecting the appropriate <code>flye_read_type</code> based on your sequencing data and additional optional paramaters.</p>
</details>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Flye Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_flye.wdl">task_flye.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/fenderglass/Flye">Flye on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/fenderglass/Flye/blob/flye/docs/USAGE.md">Flye Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.nature.com/articles/s41587-019-0072-8">Assembly of long, error-prone reads using repeat graphs</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Bandage</code>: Graph Visualization</summary>
<p>Bandage creates <em>de novo</em> assembly graphs containing the assembled contigs and the connections between those contigs. These graphs are useful for visualizing the assembly structure, identifying potential misassemblies, and understanding the relationships between contigs.</p>
<div class="admonition techdetails">
<p class="admonition-title">Bandage Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_bandage_plot.wdl">task_bandage_plot.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rrwick/Bandage">Bandage on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rrwick/Bandage#bandage">Bandage Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/31/20/3350/196114">Bandage: interactive visualization of <em>de novo</em> genome assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Polypolish</code>: Hybrid Assembly Polishing <mark><em>for ONT and Illumina data</em></mark></summary>
<p>If short reads are provided with the optional <code>illumina_read1</code> and <code>illumina_read2</code> inputs, Polypolish will use those short-reads to correct errors in the long-read assemblies. Uniquely, Polypolish uses the short-read alignments where each read is aligned to <em>all</em> possible locations, meaning that even repeat regions will have error correction.</p>
<div class="admonition techdetails">
<p class="admonition-title">Polypolish Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/polishing/task_polypolish.wdl">task_polypolish.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rrwick/Polypolish">Polypolish on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rrwick/Polypolish#polypolish">Polypolish Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1371/journal.pcbi.1009802">Polypolish: short-read polishing of long-read bacterial genome assemblies</a><br><a href="https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.001254">How low can you go? Short-read polishing of Oxford Nanopore bacterial genome assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Medaka</code>: Polishing of Flye assembly (default; optional)</summary>
<p>Polishing is optional and can be skipped by setting the <code>skip_polishing</code> variable to true. If polishing is skipped, then neither Medaka or Racon will run.</p>
<p>Medaka is the default assembly polisher used in TheiaProk. Racon may be used alternatively, and if so, Medaka will not run. Medaka uses the raw reads to polish the assembly and generate a consensus sequence. </p>
<p>Importantly, Medaka requires knowing the model that was used to generate the read data. There are several ways to provide this information:</p>
<ul>
<li>Automatic Model Selection: Automatically determines the most appropriate Medaka model based on the input data, ensuring optimal polishing results without manual intervention. </li>
<li>User-Specified Model Override: Allows users to specify a particular <code>Medaka model</code> if automatic selection does not yield the desired outcome or for specialized use cases.</li>
<li>Default Model: If both automatic model selection fails and no user-specified model is provided, Medaka defaults to the predefined fallback model <code>r1041_e82_400bps_sup_v5.0.0</code>. </li>
</ul>
<div class="admonition info">
<p class="admonition-title">Medaka Model Resolution Process</p>
<p>Medaka's automatic model selection uses the <code>medaka tools resolve_model</code> command to identify the appropriate model for polishing. This process relies on metadata embedded in the input file, which is typically generated by the basecaller. If the automatic selection fails to identify a suitable model, Medaka gracefully falls back to the default model to maintain workflow continuity. <strong>Users should verify the chosen model and consider specifying a model override if necessary.</strong></p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Medaka Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/polishing/task_medaka.wdl">task_medaka.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nanoporetech/medaka">Medaka on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/nanoporetech/medaka#medaka">Medaka Documentation</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Racon</code>: Polishing of Flye assembly (alternative; optional)</summary>
<p>Polishing is optional and can be skipped by setting the <code>skip_polishing</code> variable to true. If polishing is skipped, then neither Medaka or Racon will run.</p>
<p><code>Racon</code> is an alternative to using <code>medaka</code> for assembly polishing, and can be run by setting the <code>polisher</code> input to "racon".  Racon is a consensus algorithm designed for refining raw de novo DNA assemblies generated from long, uncorrected sequencing reads.</p>
<div class="admonition techdetails">
<p class="admonition-title">Racon Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/polishing/task_racon.wdl">task_racon.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lbcb-sci/racon">Racon on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/lbcb-sci/racon#racon">Racon Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://genome.cshlp.org/content/27/5/737">Fast and accurate de novo genome assembly from long uncorrected reads</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Filter Contigs</code>: Filter contigs below a threshold length and remove homopolymer contigs</summary>
<p>This task filters the created contigs based on a user-defined minimum length threshold (default of 1000) and eliminates homopolymer contigs (contigs of any length that consist of a single nucleotide).</p>
<p>This ensures high-quality assemblies by retaining only contigs that meet specified criteria. Detailed metrics on contig counts and sequence lengths before and after filtering are provided in the output.</p>
<div class="admonition techdetails">
<p class="admonition-title">Filter Contigs Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_filter_contigs.wdl">task_filter_contigs.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Dnaapler</code>: Final Assembly Orientation</summary>
<p>Dnaapler reorients contigs to start at specific reference points. Dnaapler supports the following modes, which can be indicated by filling the <code>dnaapler_mode</code> input variable with the desired mode. The default is <code>all</code>, which reorients contigs to start with <code>dnaA</code>, <code>terL</code>, <code>repA</code>, or <code>COG1474</code>.</p>
<ul>
<li><strong>all</strong>: Reorients contigs to start with <code>dnaA</code>, <code>terL</code>, <code>repA</code>, or <code>COG1474</code> (<em>Default</em>)</li>
<li><strong>chromosome</strong>: Reorients to begin with the <code>dnaA</code> chromosomal replication initiator gene, commonly used for bacterial chromosome assemblies.</li>
<li><strong>plasmid</strong>: Reorients to start with the <code>repA</code> plasmid replication initiation gene, ideal for plasmid assemblie</li>
<li><strong>phage</strong>: Reorients to start with the <code>terL</code> large terminase subunit gene, used for bacteriophage assemblies</li>
<li><strong>archaea</strong>: Reorients to start with the <code>COG1474</code> archaeal Orc1/cdc6 gene, relevant for archaeal assemblies</li>
<li><strong>custom</strong>: Reorients based on a user-specified gene in amino acid FASTA format for experimental or unique workflows</li>
<li><strong>mystery</strong>: Reorients to start with a random CDS for exploratory purposes</li>
<li><strong>largest</strong>: Reorients to start with the largest CDS in the assembly, often useful for poorly annotated genomes</li>
<li><strong>nearest</strong>: Reorients to start with the first CDS nearest to the sequence start, resolving CDS breakpoints</li>
<li><strong>bulk</strong>: Processes multiple contigs to start with the desired start gene (<code>dnaA</code>, <code>terL</code>, <code>repA</code>, or custom)</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">Dnaapler Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_dnaapler.wdl">task_dnaapler.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/gbouras13/dnaapler">Dnaapler on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/gbouras13/dnaapler?tab=readme-ov-file#dnaapler">Dnaapler Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://joss.theoj.org/papers/10.21105/joss.05968">Dnaapler: a tool to reorient circular microbial genomes</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Flye-Denovo Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_flye_denovo.wdl">wf_flye_denovo.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
</div>
</div>
</div>
<h4 id="organism-agnostic-characterization">Organism-agnostic characterization<a class="headerlink" href="#organism-agnostic-characterization" title="Permanent link"></a></h4>
<div class="admonition tip">
<p>These tasks are performed regardless of the organism and provide quality control and taxonomic assignment.</p>
</div>
<details class="task">
<summary><code>BUSCO</code>: Assembly Quality Assessment</summary>
<p>BUSCO (<strong>B</strong>enchmarking <strong>U</strong>niversal <strong>S</strong>ingle-<strong>C</strong>opy <strong>O</strong>rthologue) attempts to quantify the completeness and contamination of an assembly to generate quality assessment metrics. It uses taxa-specific databases containing genes that are all expected to occur in the given taxa, each in a single copy. BUSCO examines the presence or absence of these genes, whether they are fragmented, and whether they are duplicated (suggestive that additional copies came from contaminants).</p>
<p><strong>BUSCO notation</strong> </p>
<p>Here is an example of BUSCO notation: <code>C:99.1%[S:98.9%,D:0.2%],F:0.0%,M:0.9%,n:440</code>. There are several abbreviations used in this output:</p>
<ul>
<li>Complete (C) - genes are considered "complete" when their lengths are within two standard deviations of the BUSCO group mean length.</li>
<li>Single-copy (S) - genes that are complete and have only one copy.</li>
<li>Duplicated (D) - genes that are complete and have more than one copy.</li>
<li>Fragmented (F) - genes that are only partially recovered.</li>
<li>Missing (M) - genes that were not recovered at all.</li>
<li>Number of genes examined (n) - the number of genes examined.</li>
</ul>
<p>A high equity assembly will use the appropriate database for the taxa, have high complete (C) and single-copy (S) percentages, and low duplicated (D), fragmented (F) and missing (M) percentages. </p>
<div class="admonition techdetails">
<p class="admonition-title">BUSCO Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_busco.wdl">task_busco.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://gitlab.com/ezlab/busco">BUSCO on GitLab</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://busco.ezlab.org/">https://busco.ezlab.org/</a></td>
</tr>
<tr>
<td>Orginal publication</td>
<td><a href="https://academic.oup.com/bioinformatics/article/31/19/3210/211866">BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>GAMBIT</code>: <strong>Taxon Assignment</strong></summary>
<p><a href="https://github.com/jlumpe/gambit"><code>GAMBIT</code></a> determines the taxon of the genome assembly using a k-mer based approach to match the assembly sequence to the closest complete genome in a database, thereby predicting its identity. Sometimes, GAMBIT can confidently designate the organism to the species level. Other times, it is more conservative and assigns it to a higher taxonomic rank.</p>
<p>For additional details regarding the GAMBIT tool and a list of available GAMBIT databases for analysis, please consult the <a href="../../../guides/gambit/">GAMBIT</a> tool documentation.</p>
<div class="admonition techdetails">
<p class="admonition-title">GAMBIT Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_gambit.wdl">task_gambit.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/jlumpe/gambit">GAMBIT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://gambit-genomics.readthedocs.io/en/latest/">GAMBIT ReadTheDocs</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0277575">GAMBIT (Genomic Approximation Method for Bacterial Identification and Tracking): A methodology to rapidly leverage whole genome sequencing of bacterial isolates for clinical identification</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ChroQueTas</code>: Antimicrobial Resistance Profiling</summary>
<p>This task performs <em>in silico</em> antimicrobial resistance (AMR) mutation detection using <a href="https://github.com/nmquijada/ChroQueTas">ChroQueTas</a> in conjunction with the <a href="https://card.mcmaster.ca/fungamrhome">FungAMR</a> database.</p>
<p>ChroQueTas extracts the putative expressed protein where point mutations are known to cause AMR, discards poorly aligned sequences via sequence similarity to known AMR genes via BLASTp, attempts to account for intron splicing and structural variation, and finally uses MAFFT to align and evaluate amino acid changes that may correspond to resistance relative to FungAMR genes.</p>
<p>ChroQueTas is automatically called if a <strong>compatible GAMBIT-predicted taxon or provided <code>expected_taxon</code> is identified</strong>. Currently, 58 different species are supported.</p>
<p><strong>Outputs:</strong></p>
<ul>
<li><strong>AMR Stats</strong>: Depicts the number of FungAMR and non-FungAMR mutations on a gene-by-gene (protein sequence) basis.</li>
<li><strong>AMR Summary</strong>: Depicts the mutation resistance and support for potential drug resistance on a gene-by-gene (protein sequence) basis.</li>
<li><strong>Fungicide Resistance String</strong>: Depicts detected AMR mutations AND the affected fungicides as a comma-delimited string in the following format: <code>"&lt;PROTEIN&gt;_&lt;REFERENCE_AA&gt;&lt;POSITION&gt;&lt;MUTATION_AA&gt;[&lt;FUNGICIDE1&gt;&lt;CONFIDENCE_SCORE&gt;;...],..."</code></li>
</ul>
<details class="toggle">
<summary>Supported Species and Genes</summary>
<p>The following table shows the species name and screened genes as of ChroQueTas v1.0.0. Species will be automatically determined based on the GAMBIT predicted taxon or provided <code>expected_taxon</code> name.</p>
<table>
<thead>
<tr>
<th>Species</th>
<th>Antimicrobial Resistance Genes</th>
</tr>
</thead>
<tbody>
<tr>
<td><em>Arthroderma vanbreuseghemii</em></td>
<td>Squalene epoxidase</td>
</tr>
<tr>
<td><em>Aspergillus awamori</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Aspergillus fumigatus</em></td>
<td>Beta-tubulin, Cytochrome b, Fks, HapE, Hmg1, Hmg2, Sdh</td>
</tr>
<tr>
<td><em>Aspergillus niger</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Aspergillus terreus</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Aspergillus tubingensis</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Beauveria bassiana</em></td>
<td>Beta-tubulin</td>
</tr>
<tr>
<td><em>Blumeria graminis</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Blumeria graminis hordei</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Blumeria graminis tritici</em></td>
<td>Cytochrome b</td>
</tr>
<tr>
<td><em>Botrytis cinerea</em></td>
<td>Beta-tubulin, Bos1, Cytochrome b, OS1</td>
</tr>
<tr>
<td><em>Candida albicans</em></td>
<td>Cap1, Cdr1, Cyp51, Erg3, Fur1, Mrr1, Mrr2, Rta2, Squalene epoxidase, Tac1, Upc2</td>
</tr>
<tr>
<td><em>Candida dubliniensis</em></td>
<td>Cyp51, Erg3, Fcy1, Fks</td>
</tr>
<tr>
<td><em>Candida metapsilosis</em></td>
<td>Fks</td>
</tr>
<tr>
<td><em>Candida orthopsilosis</em></td>
<td>Cyp51, Fks</td>
</tr>
<tr>
<td><em>Candida parapsilosis</em></td>
<td>Cyp51, Erg3, Fks, Mrr1</td>
</tr>
<tr>
<td><em>Candida tropicalis</em></td>
<td>Ben1, Cyp51, Erg3, Fks</td>
</tr>
<tr>
<td><em>Candidozyma auris</em></td>
<td>Cyp51, Erg10, Erg12, Erg3, Erg6, Fks, Fur1, Hmg1, Ncp1, Tac1</td>
</tr>
<tr>
<td><em>Cercospora beticola</em></td>
<td>Beta-tubulin, Cytochrome b</td>
</tr>
<tr>
<td><em>Clavispora lusitaniae</em></td>
<td>Fcy1, Fcy2, Fks</td>
</tr>
<tr>
<td><em>Colletotrichum acutatum</em></td>
<td>Cyp51, Cytochrome b</td>
</tr>
<tr>
<td><em>Colletotrichum fructicola</em></td>
<td>Beta-tubulin, Cytochrome b</td>
</tr>
<tr>
<td><em>Colletotrichum gloeosporioides</em></td>
<td>Beta-tubulin</td>
</tr>
<tr>
<td><em>Colletotrichum graminicola</em></td>
<td>Cytochrome b</td>
</tr>
<tr>
<td><em>Colletotrichum siamense</em></td>
<td>Cytochrome b</td>
</tr>
<tr>
<td><em>Corynespora cassiicola</em></td>
<td>Cytochrome b</td>
</tr>
<tr>
<td><em>Cryptococcus deuterogattii</em></td>
<td>Uxs1</td>
</tr>
<tr>
<td><em>Cryptococcus neoformans</em></td>
<td>Bck1, CNAG 04784, CNBG 2198, Cyp51, Dpb2, Fcy1, Fcy2, Fur1, Ran1, Ura6</td>
</tr>
<tr>
<td><em>Erysiphe necator</em></td>
<td>Sdh</td>
</tr>
<tr>
<td><em>Fusarium fujikuroi</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Fusarium graminearum</em></td>
<td>Beta-tubulin, Myosin-1, Myosin-5</td>
</tr>
<tr>
<td><em>Fusarium incarnatum</em></td>
<td>Myosin-5</td>
</tr>
<tr>
<td><em>Fusarium pseudograminearum</em></td>
<td>Sdh</td>
</tr>
<tr>
<td><em>Histoplasma capsulatum</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Kluyveromyces marxianus</em></td>
<td>Fks</td>
</tr>
<tr>
<td><em>Magnaporthe grisea</em></td>
<td>Cytochrome b, Sdh</td>
</tr>
<tr>
<td><em>Monilinia fructicola</em></td>
<td>Beta-tubulin</td>
</tr>
<tr>
<td><em>Mycosphaerella fijiensis</em></td>
<td>Cyp51, Cytochrome b</td>
</tr>
<tr>
<td><em>Nakaseomyces glabratus</em></td>
<td>Ben1, Cdc55, Cdc6, Cdr1, Cyp51, Dot6, Erg2, Erg3, Erg4, Erg6, Erg7, Erg8, Fcy1, Fen1, Fur1, Gph1, Mohi, Mrpl11, Msh2, Pdr1, Pdr15, Qdr2, Sui2, Tcb</td>
</tr>
<tr>
<td><em>Passalora fulva</em></td>
<td>Beta-tubulin</td>
</tr>
<tr>
<td><em>Phaeosphaeria nodorum</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Phakopsora pachyrhizi</em></td>
<td>Cyp51, Cytochrome b</td>
</tr>
<tr>
<td><em>Pichia kudriavzevii</em></td>
<td>Cyp51, Fks</td>
</tr>
<tr>
<td><em>Pneumocystis jirovecii</em></td>
<td>Dhfr</td>
</tr>
<tr>
<td><em>Puccinia triticina</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Pyrenopeziza brassicae</em></td>
<td>Beta-tubulin</td>
</tr>
<tr>
<td><em>Saccharomyces cerevisiae</em></td>
<td>Act1, Aft1, Any1, Arg2, Atp1, Atp2, Bck1, Bcy1, Beta-tubulin, Bul1, Cbk1, Cdc43, Cdc60, Cdr1, Csf1, Csg2, Cyp51, Cyr1, Cytochrome b, Dcp2, Dhh1, Elo2, Erg12, Erg20, Erg25, Erg3, Erg6, Erg7, Erg9, Fcy1, Fcy21, Fpr1, Fur1, Gfa1, Hap1, Hem1, Hrd3, Hym1, Inp53, Lem3, Pde2, Pdr11, Pma1, Pms1, Pol2, Rav2, Rox1, Rpl28, Rsp5, Scd5, Sip3, Sit4, Sur1, Swh1, Tao3, Tfc1, Top1, Top2, Tsc11, Tup1, Ura6, Utp18, Vma1, Vma16, Vma9, Yrr1</td>
</tr>
<tr>
<td><em>Saccharomyces paradoxus</em></td>
<td>Cdr1, Erg2, Erg6, Pdr10, Upc2, Yrr1</td>
</tr>
<tr>
<td><em>Tapesia acuformis</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Tapesia yallundae</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Trichophyton indotineae</em></td>
<td>Squalene epoxidase</td>
</tr>
<tr>
<td><em>Trichophyton interdigitale</em></td>
<td>Squalene epoxidase</td>
</tr>
<tr>
<td><em>Trichophyton mentagrophytes</em></td>
<td>Squalene epoxidase</td>
</tr>
<tr>
<td><em>Trichophyton rubrum</em></td>
<td>Squalene epoxidase</td>
</tr>
<tr>
<td><em>Ustilaginoidea virens</em></td>
<td>Cyp51</td>
</tr>
<tr>
<td><em>Venturia inaequalis</em></td>
<td>Beta-tubulin, Cytochrome b</td>
</tr>
<tr>
<td><em>Zymoseptoria tritici</em></td>
<td>Cyp51, Cytochrome b</td>
</tr>
</tbody>
</table>
</details>
<details class="toggle">
<summary>Confidence score for antimicrobial resistance</summary>
<p>For each mutation there can be one or several antifungals for which the resistance has been associated with resistance in the past. For each antigungal, e.g. <code>Clotrimazole(3/NA)</code>, it appears two numbers (or an <code>NA</code>) between brackets separated by a <code>/</code>. These values represent the <strong>confidence score</strong>, that assesses the degree of evidence that supports their role in drug resistance. The number in the left side of the / is the <strong>highest positive confidence score</strong> while the one on the right side is the <strong>highest negative confidence score</strong>.</p>
<ul>
<li>A <strong>positive confidence score</strong> denotes mutations reported to confer resistance</li>
<li>A <strong>negative confidence score</strong> relates to mutations reported in susceptible strains</li>
</ul>
<p>The following table shows the confidence scores and their rationale.</p>
<table>
<thead>
<tr>
<th>Confidence score</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>The mutation was created in the same susceptible species background and effect on resistance was confirmed by measurements.</td>
</tr>
<tr>
<td>2</td>
<td>The mutation was created in the same gene but expressed in another species with potentially non endogenous level of expression and effect on resistance was confirmed by experimental measurements.</td>
</tr>
<tr>
<td>3</td>
<td>The effect of the mutation was quantified by bulk competition assays such as Deep Mutational Scanning.</td>
</tr>
<tr>
<td>4</td>
<td>The mutation was identified by experimental evolution where its high frequency in independent replicates suggests it causes resistance.</td>
</tr>
<tr>
<td>5</td>
<td>The deletion of the gene causes resistance.</td>
</tr>
<tr>
<td>6</td>
<td>The overexpression or duplication of the gene causes resistance.</td>
</tr>
<tr>
<td>7</td>
<td>Significant association between the mutation and resistance in a population as determined by GWAS (or other population-based association such as QTL or classical genetics).</td>
</tr>
<tr>
<td>8</td>
<td>The mutation was identified in a 'natural' strain (e.g. a clinical isolate) that is resistant but without any further validations.</td>
</tr>
<tr>
<td>NA</td>
<td>Absence of a positive confidence score report</td>
</tr>
<tr>
<td>Negative scores (-)</td>
<td>Any confidence score where the mutation was found not to cause resistance using the approach of the corresponding positive score.</td>
</tr>
</tbody>
</table>
<p>More information can be found in <a href="https://github.com/nmquijada/ChroQueTas/wiki/Confidence-score-for-antimicrobial-resistance">ChroQueTas' documentation</a>.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">ChroQueTas Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/drug_resistance/task_chroquetas.wdl">task_chroquetas.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nmquijada/ChroQueTas">ChroQueTas on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/nmquijada/ChroQueTas">ChroQueTas on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41564-025-02084-7">FungAMR: a comprehensive database for investigating fungal mutations associated with antimicrobial resistance</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h4 id="organism-specific-characterization">Organism-specific characterization<a class="headerlink" href="#organism-specific-characterization" title="Permanent link"></a></h4>
<div class="admonition tip">
<p>The TheiaEuk workflow automatically activates taxa-specific tasks after identification of the relevant taxa using <code>GAMBIT</code>. Many of these taxa-specific tasks do not require any additional inputs from the user.</p>
</div>
<details class="toggle">
<summary><em>Candidozyma auris</em> (also known as <em>Candida auris</em>)</summary>
<p>Three tools can be deployed when <em>Candidozyma auris</em>/<em>Candida auris</em> is  identified.</p>
<details class="task">
<summary>Cladetyping: clade determination</summary>
<p>A custom GAMBIT database is created using six clade-specific <em>Candidozyma auris</em> reference genomes. Sequences undergo genomic signature comparison against this database, which then enables assignment to one of the six <em>Candidozyma auris</em> clades (Clade I to Clade VI) based on sequence similarity and phylogenetic relationships. This integrated approach ensures precise clade assignments, crucial for understanding the genetic diversity and epidemiology of <em>Candidozyma auris</em>.</p>
<p>See more information on the reference information for the six clades below:</p>
<table>
<thead>
<tr>
<th>Clade</th>
<th>Genome Accession</th>
<th>Assembly Name</th>
<th>Strain</th>
<th>BioSample Accession</th>
</tr>
</thead>
<tbody>
<tr>
<td>Clade I</td>
<td>GCA_002759435.3</td>
<td>Cand_auris_B8441_V3</td>
<td>B8441</td>
<td>SAMN05379624</td>
</tr>
<tr>
<td>Clade II</td>
<td>GCA_003013715.2</td>
<td>ASM301371v2</td>
<td>B11220</td>
<td>SAMN05379608</td>
</tr>
<tr>
<td>Clade III</td>
<td>GCA_002775015.1</td>
<td>Cand_auris_B11221_V1</td>
<td>B11221</td>
<td>SAMN05379609</td>
</tr>
<tr>
<td>Clade IV</td>
<td>GCA_003014415.1</td>
<td>Cand_auris_B11243</td>
<td>B11243</td>
<td>SAMN05379619</td>
</tr>
<tr>
<td>Clade V</td>
<td>GCA_016809505.1</td>
<td>ASM1680950v1</td>
<td>IFRC2087</td>
<td>SAMN11570381</td>
</tr>
<tr>
<td>Clade VI</td>
<td>GCA_032714025.1</td>
<td>ASM3271402v1</td>
<td>F1580</td>
<td>SAMN36753179</td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="admonition-title">Clade VI annotation</p>
<p>Clade VI does not have an available reference genome annotation at the time of adding the reference genome into Cladetyping. While Clade VI assignment is functional, downstream variant calling is not currently possible without an annotation. Users may provide a close relative annotation, such as Clade IV, though it is unknown if Clade VI variants can reliably be called with respect to such a reference. </p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Cauris_Cladetyper Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/candidozyma/task_cauris_cladetyper.wdl">task_cauris_cladetyper.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/jlumpe/gambit">GAMBIT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://theiagen.notion.site/GAMBIT-7c1376b861d0486abfbc316480046bdc?pvs=4">GAMBIT Overview</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0277575">GAMBIT (Genomic Approximation Method for Bacterial Identification and Tracking): A methodology to rapidly leverage whole genome sequencing of bacterial isolates for clinical identification</a> <br> <a href="https://doi.org/10.3389/fpubh.2023.1198213">TheiaEuk: a species-agnostic bioinformatics workflow for fungal genomic characterization</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>amr_search</code>: Antimicrobial Resistance Profiling (optional)</summary>
<p>To activate this task, set <code>run_amr_search</code> to be <code>true</code>.</p>
<p>This task performs <em>in silico</em> antimicrobial resistance (AMR) profiling for supported species using AMRsearch, the primary tool used by <a href="https://pathogen.watch/">Pathogenwatch</a> to genotype and infer antimicrobial resistance (AMR) phenotypes from assembled microbial genomes.</p>
<p>AMRsearch screens against Pathogenwatch's library of curated genotypes and inferred phenotypes, developed in collaboration with community experts. Resistance phenotypes are determined based on both <em>resistance genes</em> and <em>mutations</em>, and the system accounts for interactions between multiple SNPs, genes, and suppressors. Predictions follow <strong>S/I/R classification</strong> (<em>Sensitive</em>, <em>Intermediate</em>, <em>Resistant</em>).</p>
<p>Currently, only a subset of species are supported by this task.</p>
<details class="toggle">
<summary>Supported Species</summary>
<p>The following table shows the species name and the associated NCBI Code. If you are running AMR Search as part of TheiaProk and TheiaEuk, these codes will be automatically determined based on the GAMBIT predicted taxon, or the user-provided <code>expected_taxon</code> input.</p>
<table>
<thead>
<tr>
<th>Species</th>
<th>NCBI Code</th>
</tr>
</thead>
<tbody>
<tr>
<td><em>Neisseria gonorrhoeae</em></td>
<td>485</td>
</tr>
<tr>
<td><em>Staphylococcus aureus</em></td>
<td>1280</td>
</tr>
<tr>
<td><em>Salmonella</em> Typhi</td>
<td>90370</td>
</tr>
<tr>
<td><em>Streptococcus pneumoniae</em></td>
<td>1313</td>
</tr>
<tr>
<td><em>Klebisiella</em></td>
<td>570</td>
</tr>
<tr>
<td><em>Escherichia</em></td>
<td>561</td>
</tr>
<tr>
<td><em>Mycobacterium tuberculosis</em></td>
<td>1773</td>
</tr>
<tr>
<td><em>Candida auris</em></td>
<td>498019</td>
</tr>
<tr>
<td><em>Vibrio cholerae</em></td>
<td>666</td>
</tr>
<tr>
<td><em>Campylobacter</em></td>
<td>194</td>
</tr>
</tbody>
</table>
</details>
<p><strong>Outputs:</strong></p>
<ul>
<li><strong>JSON Output</strong>: Contains the complete AMR profile, including detailed resistance state, detected resistance genes/mutations, and supporting BLAST results.</li>
<li><strong>CSV &amp; PDF Tables</strong>: An incorporated Python script, <code>parse_amr_json.py</code>, extracts and formats results into a CSV file and PDF summary table for easier visualization.</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">amr_search Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/drug_resistance/task_amr_search.wdl">task_amr_search.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/pathogenwatch-oss/amr-search">AMRsearch on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/pathogenwatch-oss/amr-search">AMRsearch on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7545138/">PAARSNP: rapid genotypic resistance prediction for <em>Neisseria gonorrhoeae</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Snippy_Variants</code>: Antifungal Resistance Detection</summary>
<p>To detect mutations that may confer antifungal resistance, <code>Snippy</code> is used to find all variants relative to the clade-specific reference, then these variants are queried for product names associated with resistance. It's important to note that unlike <code>amr_search</code>, this task reports all variants found in the searched targets.</p>
<ul>
<li>FKS1</li>
<li>ERG11 (lanosterol 14-alpha demethylase)</li>
<li>FUR1 (uracil phosphoribosyltransferase)</li>
</ul>
<p>We query <code>Snippy</code> results to see if any mutations were identified in those genes. By default, we automatically check for the following loci (which can be overwritten by the user). You will find the mutations next to the locus tag in the <code>theiaeuk_snippy_variants_hits</code> column corresponding gene name (see below):</p>
<table>
<thead>
<tr>
<th><strong>TheiaEuk Search Term</strong></th>
<th><strong>Corresponding Gene Name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>B9J08_005340</td>
<td>ERG6</td>
</tr>
<tr>
<td>B9J08_000401</td>
<td>FLO8</td>
</tr>
<tr>
<td>B9J08_005343</td>
<td>Hypothetical protein (PSK74852)</td>
</tr>
<tr>
<td>B9J08_003102</td>
<td>MEC3</td>
</tr>
<tr>
<td>B9J08_003737</td>
<td>ERG3</td>
</tr>
<tr>
<td>lanosterol.14-alpha.demethylase</td>
<td>ERG11</td>
</tr>
<tr>
<td>uracil.phosphoribosyltransferase</td>
<td>FUR1</td>
</tr>
<tr>
<td>FKS1</td>
<td>FKS1</td>
</tr>
</tbody>
</table>
<details class="toggle">
<summary>Known resistance-conferring mutations for <em>Candidozyma auris</em></summary>
<p>Mutations in these genes that are known to confer resistance are shown below</p>
<table>
<thead>
<tr>
<th><strong>Organism</strong></th>
<th><strong>Found in</strong></th>
<th><strong>Gene name</strong></th>
<th><strong>Gene locus</strong></th>
<th><strong>AA mutation</strong></th>
<th><strong>Drug</strong></th>
<th><strong>Reference</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>Candidozyma auris</td>
<td>Human</td>
<td>ERG11</td>
<td></td>
<td>Y132F</td>
<td>Fluconazole</td>
<td><a href="https://academic.oup.com/cid/article/64/2/134/2706620/Simultaneous-Emergence-of-Multidrug-Resistant">Simultaneous Emergence of Multidrug-Resistant <em>Candida auris</em> on 3 Continents Confirmed by Whole-Genome Sequencing and Epidemiological Analyses</a></td>
</tr>
<tr>
<td>Candidozyma auris</td>
<td>Human</td>
<td>ERG11</td>
<td></td>
<td>K143R</td>
<td>Fluconazole</td>
<td><a href="https://academic.oup.com/cid/article/64/2/134/2706620/Simultaneous-Emergence-of-Multidrug-Resistant">Simultaneous Emergence of Multidrug-Resistant <em>Candida auris</em> on 3 Continents Confirmed by Whole-Genome Sequencing and Epidemiological Analyses</a></td>
</tr>
<tr>
<td>Candidozyma auris</td>
<td>Human</td>
<td>ERG11</td>
<td></td>
<td>F126T</td>
<td>Fluconazole</td>
<td><a href="https://academic.oup.com/cid/article/64/2/134/2706620/Simultaneous-Emergence-of-Multidrug-Resistant">Simultaneous Emergence of Multidrug-Resistant <em>Candida auris</em> on 3 Continents Confirmed by Whole-Genome Sequencing and Epidemiological Analyses</a></td>
</tr>
<tr>
<td>Candidozyma auris</td>
<td>Human</td>
<td>FKS1</td>
<td></td>
<td>S639P</td>
<td>Micafungin</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S0732889317303498">Activity of CD101, a long-acting echinocandin, against clinical isolates of Candida auris</a></td>
</tr>
<tr>
<td>Candidozyma auris</td>
<td>Human</td>
<td>FKS1</td>
<td></td>
<td>S639P</td>
<td>Caspofungin</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S0732889317303498">Activity of CD101, a long-acting echinocandin, against clinical isolates of Candida auris</a></td>
</tr>
<tr>
<td>Candidozyma auris</td>
<td>Human</td>
<td>FKS1</td>
<td></td>
<td>S639P</td>
<td>Anidulafungin</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S0732889317303498">Activity of CD101, a long-acting echinocandin, against clinical isolates of Candida auris</a></td>
</tr>
<tr>
<td>Candidozyma auris</td>
<td>Human</td>
<td>FKS1</td>
<td></td>
<td>S639F</td>
<td>Micafungin</td>
<td><a href="https://academic.oup.com/jac/advance-article/doi/10.1093/jac/dkx480/4794718">A multicentre study of antifungal susceptibility patterns among 350 <em>Candida auris</em> isolates (200917) in India: role of the ERG11 and FKS1 genes in azole and echinocandin resistance</a></td>
</tr>
<tr>
<td>Candidozyma auris</td>
<td>Human</td>
<td>FKS1</td>
<td></td>
<td>S639F</td>
<td>Caspofungin</td>
<td><a href="https://academic.oup.com/jac/advance-article/doi/10.1093/jac/dkx480/4794718">A multicentre study of antifungal susceptibility patterns among 350 <em>Candida auris</em> isolates (200917) in India: role of the ERG11 and FKS1 genes in azole and echinocandin resistance</a></td>
</tr>
<tr>
<td>Candidozyma auris</td>
<td>Human</td>
<td>FKS1</td>
<td></td>
<td>S639F</td>
<td>Anidulafungin</td>
<td><a href="https://academic.oup.com/jac/advance-article/doi/10.1093/jac/dkx480/4794718">A multicentre study of antifungal susceptibility patterns among 350 <em>Candida auris</em> isolates (200917) in India: role of the ERG11 and FKS1 genes in azole and echinocandin resistance</a></td>
</tr>
<tr>
<td>Candidozyma auris</td>
<td>Human</td>
<td>FUR1</td>
<td>CAMJ_004922</td>
<td>F211I</td>
<td>5-flucytosine</td>
<td><a href="https://doi.org/10.1038/s41426-018-0045-x">Genomic epidemiology of the UK outbreak of the emerging human fungal pathogen Candida auris</a></td>
</tr>
</tbody>
</table>
</details>
<details class="toggle">
<summary>Example Output Interpretation</summary>
<p>For example, one sample may have the following output for the <code>theiaeuk_snippy_variants_hits</code> column:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>lanosterol.14-alpha.demethylase: lanosterol 14-alpha demethylase (missense_variant c.428A&gt;G p.Lys143Arg; C:266 T:0),B9J08_000401: hypothetical protein (stop_gained c.424C&gt;T p.Gln142*; A:70 G:0)
</span></code></pre></div>
<p>Based on this, we can tell that ERG11 has a missense variant at position 143 (Lysine to Arginine) and B9J08_000401 (which is FLO8) has a stop-gained variant at position 142 (Glutamine to Stop).</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Snippy Variants Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_snippy_variants.wdl">task_snippy_variants.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_snippy_gene_query.wdl">task_snippy_gene_query.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/tseemann/snippy">Snippy on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/tseemann/snippy">Snippy on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Candida albicans</em></summary>
<p>When this species is detected by the taxon ID tool, an antifungal resistance detection task is deployed.</p>
<details class="task">
<summary><code>Snippy_Variants</code>: Antifungal Resistance Detection</summary>
<p>To detect mutations that may confer antifungal resistance, <code>Snippy</code> is used to find all variants relative to the clade-specific reference, then these variants are queried for product names associated with resistance. It's important to note that unlike <code>amr_search</code>, this task reports all variants found in the searched targets.</p>
<ul>
<li>ERG11</li>
<li>GCS1 (FKS1)</li>
<li>FUR1</li>
<li>RTA2</li>
</ul>
<p>We query <code>Snippy</code> results to see if any mutations were identified in those genes. By default, we automatically check for the following loci (which can be overwritten by the user). You will find the mutations next to the locus tag in the <code>theiaeuk_snippy_variants_hits</code> column corresponding gene name (see below):</p>
<table>
<thead>
<tr>
<th><strong>TheiaEuk Search Term</strong></th>
<th><strong>Corresponding Gene Name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>ERG11</td>
<td>ERG11</td>
</tr>
<tr>
<td>GCS1</td>
<td>FKS1</td>
</tr>
<tr>
<td>FUR1</td>
<td>FUR1</td>
</tr>
<tr>
<td>RTA2</td>
<td>RTA2</td>
</tr>
</tbody>
</table>
<details class="toggle">
<summary>Example Output Interpretation</summary>
<p>For example, one sample may have the following output for the <code>theiaeuk_snippy_variants_hits</code> column:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>lanosterol.14-alpha.demethylase: lanosterol 14-alpha demethylase (missense_variant c.428A&gt;G p.Lys143Arg; C:266 T:0),B9J08_000401: hypothetical protein (stop_gained c.424C&gt;T p.Gln142*; A:70 G:0)
</span></code></pre></div>
<p>Based on this, we can tell that ERG11 has a missense variant at position 143 (Lysine to Arginine) and B9J08_000401 (which is FLO8) has a stop-gained variant at position 142 (Glutamine to Stop).</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Snippy Variants Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_snippy_variants.wdl">task_snippy_variants.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_snippy_gene_query.wdl">task_snippy_gene_query.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/tseemann/snippy">Snippy on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/tseemann/snippy">Snippy on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Aspergillus fumigatus</em></summary>
<p>When this species is detected by the taxon ID tool an antifungal resistance detection task is deployed.</p>
<details class="task">
<summary><code>Snippy_Variants</code>: Antifungal Resistance Detection</summary>
<p>To detect mutations that may confer antifungal resistance, <code>Snippy</code> is used to find all variants relative to the clade-specific reference, then these variants are queried for product names associated with resistance. It's important to note that unlike <code>amr_search</code>, this task reports all variants found in the searched targets.</p>
<ul>
<li>Cyp51A</li>
<li>HapE</li>
<li>COX10 (AFUA_4G08340)</li>
</ul>
<p>We query <code>Snippy</code> results to see if any mutations were identified in those genes. By default, we automatically check for the following loci (which can be overwritten by the user). You will find the mutations next to the locus tag in the <code>theiaeuk_snippy_variants_hits</code> column corresponding gene name (see below):</p>
<table>
<thead>
<tr>
<th><strong>TheiaEuk Search Term</strong></th>
<th><strong>Corresponding Gene Name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>Cyp51A</td>
<td>Cyp51A</td>
</tr>
<tr>
<td>HapE</td>
<td>HapE</td>
</tr>
<tr>
<td>AFUA_4G08340</td>
<td>COX10</td>
</tr>
</tbody>
</table>
<details class="toggle">
<summary>Example Output Interpretation</summary>
<p>For example, one sample may have the following output for the <code>theiaeuk_snippy_variants_hits</code> column:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>lanosterol.14-alpha.demethylase: lanosterol 14-alpha demethylase (missense_variant c.428A&gt;G p.Lys143Arg; C:266 T:0),B9J08_000401: hypothetical protein (stop_gained c.424C&gt;T p.Gln142*; A:70 G:0)
</span></code></pre></div>
<p>Based on this, we can tell that ERG11 has a missense variant at position 143 (Lysine to Arginine) and B9J08_000401 (which is FLO8) has a stop-gained variant at position 142 (Glutamine to Stop).</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Snippy Variants Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_snippy_variants.wdl">task_snippy_variants.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_snippy_gene_query.wdl">task_snippy_gene_query.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/tseemann/snippy">Snippy on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/tseemann/snippy">Snippy on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Cryptococcus neoformans</em></summary>
<p>When this species is detected by the taxon ID tool an antifungal resistance detection task is deployed.</p>
<details class="task">
<summary><code>Snippy_Variants</code>: Antifungal Resistance Detection</summary>
<p>To detect mutations that may confer antifungal resistance, <code>Snippy</code> is used to find all variants relative to the clade-specific reference, then these variants are queried for product names associated with resistance. It's important to note that unlike <code>amr_search</code>, this task reports all variants found in the searched targets.</p>
<ul>
<li>ERG11 (CNA00300)</li>
</ul>
<p>We query <code>Snippy</code> results to see if any mutations were identified in those genes. By default, we automatically check for the following loci (which can be overwritten by the user). You will find the mutations next to the locus tag in the <code>theiaeuk_snippy_variants_hits</code> column corresponding gene name (see below):</p>
<table>
<thead>
<tr>
<th><strong>TheiaEuk Search Term</strong></th>
<th><strong>Corresponding Gene Name</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>CNA00300</td>
<td>ERG11</td>
</tr>
</tbody>
</table>
<details class="toggle">
<summary>Example Output Interpretation</summary>
<p>For example, one sample may have the following output for the <code>theiaeuk_snippy_variants_hits</code> column:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>lanosterol.14-alpha.demethylase: lanosterol 14-alpha demethylase (missense_variant c.428A&gt;G p.Lys143Arg; C:266 T:0),B9J08_000401: hypothetical protein (stop_gained c.424C&gt;T p.Gln142*; A:70 G:0)
</span></code></pre></div>
<p>Based on this, we can tell that ERG11 has a missense variant at position 143 (Lysine to Arginine) and B9J08_000401 (which is FLO8) has a stop-gained variant at position 142 (Glutamine to Stop).</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Snippy Variants Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_snippy_variants.wdl">task_snippy_variants.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_snippy_gene_query.wdl">task_snippy_gene_query.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/tseemann/snippy">Snippy on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/tseemann/snippy">Snippy on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<h3 id="outputs">Outputs<a class="headerlink" href="#outputs" title="Permanent link"></a></h3>
<div class="admonition caption">
<div class="tabbed-set tabbed-alternate" data-tabs="5:2"><input checked="checked" id="theiaeuk_illumina_pe_4" name="__tabbed_5" type="radio"><input id="theiaeuk_ont_4" name="__tabbed_5" type="radio"><div class="tabbed-labels"><label for="theiaeuk_illumina_pe_4">TheiaEuk_Illumina_PE</label><label for="theiaeuk_ont_4">TheiaEuk_ONT</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>amr_search_csv</td>
<td><strong>File</strong></td>
<td>CSV formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_docker</td>
<td><strong>String</strong></td>
<td>Docker image used to run AMR_Search</td>
</tr>
<tr>
<td>amr_search_results</td>
<td><strong>File</strong></td>
<td>JSON formatted AMR profile including BLAST results</td>
</tr>
<tr>
<td>amr_search_results_pdf</td>
<td><strong>File</strong></td>
<td>PDF formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_version</td>
<td><strong>String</strong></td>
<td>Version of AMR_Search libraries used</td>
</tr>
<tr>
<td>assembler</td>
<td><strong>String</strong></td>
<td>Assembler used in digger_denovo subworkflow</td>
</tr>
<tr>
<td>assembler_version</td>
<td><strong>String</strong></td>
<td>Version of the assembler used in digger_denovo</td>
</tr>
<tr>
<td>assembly_fasta</td>
<td><strong>File</strong></td>
<td><em>De novo</em> genome assembly in FASTA format</td>
</tr>
<tr>
<td>assembly_length</td>
<td><strong>Int</strong></td>
<td>Length of assembly (total contig length) as determined by QUAST</td>
</tr>
<tr>
<td>bbduk_docker</td>
<td><strong>String</strong></td>
<td>The Docker image for bbduk, which was used to remove the adapters from the sequences</td>
</tr>
<tr>
<td>busco_database</td>
<td><strong>String</strong></td>
<td>BUSCO database used</td>
</tr>
<tr>
<td>busco_docker</td>
<td><strong>String</strong></td>
<td>BUSCO docker image used</td>
</tr>
<tr>
<td>busco_report</td>
<td><strong>File</strong></td>
<td>A plain text summary of the results in BUSCO notation</td>
</tr>
<tr>
<td>busco_results</td>
<td><strong>String</strong></td>
<td>BUSCO results (see relevant toggle in <a href="https://theiagen.github.io/public_health_bioinformatics/latest/workflows/genomic_characterization/theiaprok/#post-assembly-tasks-performed-for-all-taxa">this block</a>)</td>
</tr>
<tr>
<td>busco_version</td>
<td><strong>String</strong></td>
<td>BUSCO software version used</td>
</tr>
<tr>
<td>cg_pipeline_docker</td>
<td><strong>String</strong></td>
<td>Docker file used for running CG-Pipeline on cleaned reads</td>
</tr>
<tr>
<td>cg_pipeline_report_clean</td>
<td><strong>File</strong></td>
<td>TSV file of read metrics from clean reads, including average read length, number of reads, and estimated genome coverage</td>
</tr>
<tr>
<td>cg_pipeline_report_raw</td>
<td><strong>File</strong></td>
<td>TSV file of read metrics from raw reads, including average read length, number of reads, and estimated genome coverage</td>
</tr>
<tr>
<td>chroquetas_amr_stats</td>
<td><strong>File</strong></td>
<td>File depicting quantity of AMR mutations detected on a gene-by-gene basis</td>
</tr>
<tr>
<td>chroquetas_amr_summary</td>
<td><strong>File</strong></td>
<td>File depicting positions, mutations, and associated fungicide resistance</td>
</tr>
<tr>
<td>chroquetas_fungicide_resistance</td>
<td><strong>String</strong></td>
<td>Comma-delimited summary of mutations and their associated fungicide resistance</td>
</tr>
<tr>
<td>chroquetas_status</td>
<td><strong>String</strong></td>
<td>Status depicting passing or specific errors encountered</td>
</tr>
<tr>
<td>chroquetas_version</td>
<td><strong>String</strong></td>
<td>Version of ChroQueTas</td>
</tr>
<tr>
<td>cladetyper_annotated_reference</td>
<td><strong>String</strong></td>
<td>The annotated reference file for the identified clade, "None" if no clade was identified/no annotation is inputted</td>
</tr>
<tr>
<td>cladetyper_clade</td>
<td><strong>String</strong></td>
<td>The clade assigned to the input assembly</td>
</tr>
<tr>
<td>cladetyper_docker_image</td>
<td><strong>String</strong></td>
<td>The Docker container used for the task</td>
</tr>
<tr>
<td>cladetyper_gambit_version</td>
<td><strong>String</strong></td>
<td>The version of GAMBIT used for the analysis</td>
</tr>
<tr>
<td>combined_mean_q_clean</td>
<td><strong>Float</strong></td>
<td>Mean quality score for the combined clean reads</td>
</tr>
<tr>
<td>combined_mean_q_raw</td>
<td><strong>Float</strong></td>
<td>Mean quality score for the combined raw reads</td>
</tr>
<tr>
<td>combined_mean_readlength_clean</td>
<td><strong>Float</strong></td>
<td>Mean read length for the combined clean reads</td>
</tr>
<tr>
<td>combined_mean_readlength_raw</td>
<td><strong>Float</strong></td>
<td>Mean read length for the combined raw reads</td>
</tr>
<tr>
<td>contigs_gfa</td>
<td><strong>File</strong></td>
<td>Assembly graph output generated by SPAdes (Illumina: PE, SE) or Flye (ONT), used to visualize and evaluate genome assembly results.</td>
</tr>
<tr>
<td>est_coverage_clean</td>
<td><strong>Float</strong></td>
<td>Estimated coverage calculated from clean reads and genome length</td>
</tr>
<tr>
<td>est_coverage_raw</td>
<td><strong>Float</strong></td>
<td>Estimated coverage calculated from raw reads and genome length</td>
</tr>
<tr>
<td>fastp_html_report</td>
<td><strong>File</strong></td>
<td>The HTML report made with fastp</td>
</tr>
<tr>
<td>fastp_version</td>
<td><strong>String</strong></td>
<td>The version of fastp used</td>
</tr>
<tr>
<td>fastq_scan_clean1_json</td>
<td><strong>File</strong></td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_clean2_json</td>
<td><strong>File</strong></td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean reverse read quality and length</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean1</td>
<td><strong>Int</strong></td>
<td>The number of forward reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean2</td>
<td><strong>Int</strong></td>
<td>The number of reverse reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean_pairs</td>
<td><strong>String</strong></td>
<td>The number of read pairs after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw1</td>
<td><strong>Int</strong></td>
<td>The number of input forward reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw2</td>
<td><strong>Int</strong></td>
<td>The number of input reserve reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw_pairs</td>
<td><strong>String</strong></td>
<td>The number of input read pairs as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_raw1_json</td>
<td><strong>File</strong></td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_raw2_json</td>
<td><strong>File</strong></td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw reverse read quality and length</td>
</tr>
<tr>
<td>fastq_scan_version</td>
<td><strong>String</strong></td>
<td>The version of fastq_scan</td>
</tr>
<tr>
<td>fastqc_clean1_html</td>
<td><strong>File</strong></td>
<td>An HTML file that provides a graphical visualization of clean forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_clean2_html</td>
<td><strong>File</strong></td>
<td>An HTML file that provides a graphical visualization of clean reverse read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_docker</td>
<td><strong>String</strong></td>
<td>The Docker container used for fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean1</td>
<td><strong>Int</strong></td>
<td>The number of forward reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean2</td>
<td><strong>Int</strong></td>
<td>The number of reverse reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean_pairs</td>
<td><strong>String</strong></td>
<td>The number of read pairs after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_raw1</td>
<td><strong>Int</strong></td>
<td>The number of input forward reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_num_reads_raw2</td>
<td><strong>Int</strong></td>
<td>The number of input reverse reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_num_reads_raw_pairs</td>
<td><strong>String</strong></td>
<td>The number of input read pairs by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_raw1_html</td>
<td><strong>File</strong></td>
<td>An HTML file that provides a graphical visualization of raw forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_raw2_html</td>
<td><strong>File</strong></td>
<td>An HTML file that provides a graphical visualization of raw reverse read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_version</td>
<td><strong>String</strong></td>
<td>Version of fastqc software used</td>
</tr>
<tr>
<td>filtered_contigs_metrics</td>
<td><strong>File</strong></td>
<td>File containing metrics of contigs filtered</td>
</tr>
<tr>
<td>gambit_closest_genomes</td>
<td><strong>File</strong></td>
<td>CSV file listing genomes in the GAMBIT database that are most similar to the query assembly</td>
</tr>
<tr>
<td>gambit_db_version</td>
<td><strong>String</strong></td>
<td>Version of the GAMBIT database used</td>
</tr>
<tr>
<td>gambit_docker</td>
<td><strong>String</strong></td>
<td>GAMBIT Docker used</td>
</tr>
<tr>
<td>gambit_predicted_taxon</td>
<td><strong>String</strong></td>
<td>Taxon predicted by GAMBIT</td>
</tr>
<tr>
<td>gambit_predicted_taxon_rank</td>
<td><strong>String</strong></td>
<td>Taxon rank of GAMBIT taxon prediction</td>
</tr>
<tr>
<td>gambit_report</td>
<td><strong>File</strong></td>
<td>GAMBIT report in a machine-readable format</td>
</tr>
<tr>
<td>gambit_version</td>
<td><strong>String</strong></td>
<td>Version of GAMBIT software used</td>
</tr>
<tr>
<td>n50_value</td>
<td><strong>Int</strong></td>
<td>N50 of assembly calculated by QUAST</td>
</tr>
<tr>
<td>number_contigs</td>
<td><strong>Int</strong></td>
<td>Total number of contigs in assembly</td>
</tr>
<tr>
<td>qc_check</td>
<td><strong>String</strong></td>
<td>A string that indicates whether or not the sample passes a set of pre-determined and user-provided QC thresholds</td>
</tr>
<tr>
<td>qc_standard</td>
<td><strong>File</strong></td>
<td>The file used in the QC Check task containing the QC thresholds.</td>
</tr>
<tr>
<td>quast_gc_percent</td>
<td><strong>Float</strong></td>
<td>The GC percent of your sample</td>
</tr>
<tr>
<td>quast_report</td>
<td><strong>File</strong></td>
<td>TSV report from QUAST</td>
</tr>
<tr>
<td>quast_version</td>
<td><strong>String</strong></td>
<td>The version of QUAST</td>
</tr>
<tr>
<td>r1_mean_q_raw</td>
<td><strong>Float</strong></td>
<td>Mean quality score of raw forward reads</td>
</tr>
<tr>
<td>r1_mean_readlength_raw</td>
<td><strong>Float</strong></td>
<td>Mean read length of raw forward reads</td>
</tr>
<tr>
<td>r2_mean_q_raw</td>
<td><strong>Float</strong></td>
<td>Mean quality score of raw reverse reads</td>
</tr>
<tr>
<td>r2_mean_readlength_raw</td>
<td><strong>Float</strong></td>
<td>Mean read length of raw reverse reads</td>
</tr>
<tr>
<td>rasusa_version</td>
<td><strong>String</strong></td>
<td>Version of RASUSA used for the analysis</td>
</tr>
<tr>
<td>read1_clean</td>
<td><strong>File</strong></td>
<td>Forward read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read1_subsampled</td>
<td><strong>File</strong></td>
<td>Read1 FASTQ files downsampled to desired coverage</td>
</tr>
<tr>
<td>read2_clean</td>
<td><strong>File</strong></td>
<td>Reverse read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read2_subsampled</td>
<td><strong>File</strong></td>
<td>Read2 FASTQ files downsampled to desired coverage</td>
</tr>
<tr>
<td>read_screen_clean</td>
<td><strong>String</strong></td>
<td>PASS or FAIL result from clean read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_clean_tsv</td>
<td><strong>File</strong></td>
<td>Clean read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>read_screen_raw</td>
<td><strong>String</strong></td>
<td>PASS or FAIL result from raw read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_raw_tsv</td>
<td><strong>File</strong></td>
<td>Raw read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>seq_platform</td>
<td><strong>String</strong></td>
<td>Description of the sequencing methodology used to generate the input read data</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe_analysis_date</td>
<td><strong>String</strong></td>
<td>Date of TheiaEuk PE workflow execution</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe_version</td>
<td><strong>String</strong></td>
<td>TheiaEuk PE workflow version used</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_bai</td>
<td><strong>String</strong></td>
<td>Indexed bam file of the reads aligned to the reference</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_bam</td>
<td><strong>String</strong></td>
<td>BAM file of reads aligned to the reference</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_coverage_tsv</td>
<td><strong>String</strong></td>
<td>Coverage statistics TSV file output by the <code>samtools coverage</code> command, providing genome-wide metrics such as the proportion of bases covered (depth  1), mean depth, and other related statistics.</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_gene_query_results</td>
<td><strong>String</strong></td>
<td>CSV file detailing results for mutations associated with the query strings specified by the user</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_hits</td>
<td><strong>String</strong></td>
<td>A summary of mutations associated with the query strings specified by the user</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_num_reads_aligned</td>
<td><strong>String</strong></td>
<td>Number of reads that aligned to the reference genome as calculated by samtools view -c command</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_num_variants</td>
<td><strong>String</strong></td>
<td>Number of variants detected between sample and reference genome</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_outdir_tarball</td>
<td><strong>String</strong></td>
<td>A compressed file containing the whole directory of snippy output files. This is used when running Snippy_Tree</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_percent_ref_coverage</td>
<td><strong>String</strong></td>
<td>Proportion of the reference genome covered by reads with a depth greater than or equal to the <code>min_coverage</code> threshold (default is 10).</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_query</td>
<td><strong>String</strong></td>
<td>Gene query strings specified by the user when running the workflow</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_query_check</td>
<td><strong>String</strong></td>
<td>Verification that query strings are found in the reference genome</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_reference_genome</td>
<td><strong>String</strong></td>
<td>The reference genome used in the alignment and variant calling</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_results</td>
<td><strong>String</strong></td>
<td>CSV file detailing results for all mutations identified relative to the reference</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_summary</td>
<td><strong>String</strong></td>
<td>A summary TXT fie showing the number of mutations identified for each mutation type</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_version</td>
<td><strong>String</strong></td>
<td>Version of Snippy used</td>
</tr>
<tr>
<td>trimmomatic_docker</td>
<td><strong>String</strong></td>
<td>The docker image used for the trimmomatic module in this workflow</td>
</tr>
<tr>
<td>trimmomatic_version</td>
<td><strong>String</strong></td>
<td>The version of Trimmomatic used</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>amr_search_csv</td>
<td><strong>File</strong></td>
<td>CSV formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_docker</td>
<td><strong>String</strong></td>
<td>Docker image used to run AMR_Search</td>
</tr>
<tr>
<td>amr_search_results</td>
<td><strong>File</strong></td>
<td>JSON formatted AMR profile including BLAST results</td>
</tr>
<tr>
<td>amr_search_results_pdf</td>
<td><strong>File</strong></td>
<td>PDF formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_version</td>
<td><strong>String</strong></td>
<td>Version of AMR_Search libraries used</td>
</tr>
<tr>
<td>assembly_fasta</td>
<td><strong>File</strong></td>
<td><em>De novo</em> genome assembly in FASTA format</td>
</tr>
<tr>
<td>assembly_length</td>
<td><strong>Int</strong></td>
<td>Length of assembly (total contig length) as determined by QUAST</td>
</tr>
<tr>
<td>bandage_plot</td>
<td><strong>File</strong></td>
<td>Image file (PNG) visualizing the Flye assembly graph generated by Bandage</td>
</tr>
<tr>
<td>bandage_version</td>
<td><strong>String</strong></td>
<td>Version of Bandage used</td>
</tr>
<tr>
<td>busco_database</td>
<td><strong>String</strong></td>
<td>BUSCO database used</td>
</tr>
<tr>
<td>busco_docker</td>
<td><strong>String</strong></td>
<td>BUSCO docker image used</td>
</tr>
<tr>
<td>busco_report</td>
<td><strong>File</strong></td>
<td>A plain text summary of the results in BUSCO notation</td>
</tr>
<tr>
<td>busco_results</td>
<td><strong>String</strong></td>
<td>BUSCO results (see relevant toggle in <a href="https://theiagen.github.io/public_health_bioinformatics/latest/workflows/genomic_characterization/theiaprok/#post-assembly-tasks-performed-for-all-taxa">this block</a>)</td>
</tr>
<tr>
<td>busco_version</td>
<td><strong>String</strong></td>
<td>BUSCO software version used</td>
</tr>
<tr>
<td>bwa_version</td>
<td><strong>String</strong></td>
<td>Version of BWA software used</td>
</tr>
<tr>
<td>chroquetas_amr_stats</td>
<td><strong>File</strong></td>
<td>File depicting quantity of AMR mutations detected on a gene-by-gene basis</td>
</tr>
<tr>
<td>chroquetas_amr_summary</td>
<td><strong>File</strong></td>
<td>File depicting positions, mutations, and associated fungicide resistance</td>
</tr>
<tr>
<td>chroquetas_fungicide_resistance</td>
<td><strong>String</strong></td>
<td>Comma-delimited summary of mutations and their associated fungicide resistance</td>
</tr>
<tr>
<td>chroquetas_status</td>
<td><strong>String</strong></td>
<td>Status depicting passing or specific errors encountered</td>
</tr>
<tr>
<td>chroquetas_version</td>
<td><strong>String</strong></td>
<td>Version of ChroQueTas</td>
</tr>
<tr>
<td>cladetype_annotated_ref</td>
<td><strong>String</strong></td>
<td>The annotated reference file for the identified clade, "None" if no clade was identified/no annotation is inputted</td>
</tr>
<tr>
<td>cladetyper_clade</td>
<td><strong>String</strong></td>
<td>The clade assigned to the input assembly</td>
</tr>
<tr>
<td>cladetyper_docker_image</td>
<td><strong>String</strong></td>
<td>The Docker container used for the task</td>
</tr>
<tr>
<td>cladetyper_version</td>
<td><strong>String</strong></td>
<td>The version of Cladetyper used for the analysis</td>
</tr>
<tr>
<td>contigs_gfa</td>
<td><strong>File</strong></td>
<td>Assembly graph output generated by SPAdes (Illumina: PE, SE) or Flye (ONT), used to visualize and evaluate genome assembly results.</td>
</tr>
<tr>
<td>dnaapler_version</td>
<td><strong>String</strong></td>
<td>Version of dnaapler used</td>
</tr>
<tr>
<td>est_coverage_clean</td>
<td><strong>Float</strong></td>
<td>Estimated coverage calculated from clean reads and genome length</td>
</tr>
<tr>
<td>est_coverage_raw</td>
<td><strong>Float</strong></td>
<td>Estimated coverage calculated from raw reads and genome length</td>
</tr>
<tr>
<td>est_genome_length</td>
<td><strong>Int</strong></td>
<td>Estimated genome length</td>
</tr>
<tr>
<td>filtered_contigs_metrics</td>
<td><strong>File</strong></td>
<td>File containing metrics of contigs filtered</td>
</tr>
<tr>
<td>flye_assembly_info</td>
<td><strong>String</strong></td>
<td>Information file from Flye assembly</td>
</tr>
<tr>
<td>flye_version</td>
<td><strong>String</strong></td>
<td>Version of Flye software used</td>
</tr>
<tr>
<td>gambit_closest_genomes_file</td>
<td><strong>File</strong></td>
<td>CSV file listing genomes in the GAMBIT database that are most similar to the query assembly</td>
</tr>
<tr>
<td>gambit_db_version</td>
<td><strong>String</strong></td>
<td>Version of the GAMBIT database used</td>
</tr>
<tr>
<td>gambit_docker</td>
<td><strong>String</strong></td>
<td>GAMBIT Docker used</td>
</tr>
<tr>
<td>gambit_next_taxon</td>
<td><strong>String</strong></td>
<td>Next taxon predicted by GAMBIT</td>
</tr>
<tr>
<td>gambit_next_taxon_rank</td>
<td><strong>String</strong></td>
<td>Next taxon rank predicted by GAMBIT</td>
</tr>
<tr>
<td>gambit_predicted_taxon</td>
<td><strong>String</strong></td>
<td>Taxon predicted by GAMBIT</td>
</tr>
<tr>
<td>gambit_predicted_taxon_rank</td>
<td><strong>String</strong></td>
<td>Taxon rank of GAMBIT taxon prediction</td>
</tr>
<tr>
<td>gambit_report_file</td>
<td><strong>File</strong></td>
<td>GAMBIT report in a machine-readable format</td>
</tr>
<tr>
<td>gambit_version</td>
<td><strong>String</strong></td>
<td>Version of GAMBIT software used</td>
</tr>
<tr>
<td>medaka_model</td>
<td><strong>String</strong></td>
<td>Model used by Medaka</td>
</tr>
<tr>
<td>medaka_version</td>
<td><strong>String</strong></td>
<td>Version of Medaka used</td>
</tr>
<tr>
<td>merlin_tag</td>
<td><strong>String</strong></td>
<td>Merlin tag for the assembly</td>
</tr>
<tr>
<td>n50_value</td>
<td><strong>Int</strong></td>
<td>N50 of assembly calculated by QUAST</td>
</tr>
<tr>
<td>nanoplot_docker</td>
<td><strong>String</strong></td>
<td>Docker image for nanoplot</td>
</tr>
<tr>
<td>nanoplot_html_clean</td>
<td><strong>File</strong></td>
<td>An HTML report describing the clean reads</td>
</tr>
<tr>
<td>nanoplot_html_raw</td>
<td><strong>File</strong></td>
<td>An HTML report describing the raw reads</td>
</tr>
<tr>
<td>nanoplot_num_reads_clean1</td>
<td><strong>Int</strong></td>
<td>Number of clean reads</td>
</tr>
<tr>
<td>nanoplot_num_reads_raw1</td>
<td><strong>Int</strong></td>
<td>Number of raw reads</td>
</tr>
<tr>
<td>nanoplot_r1_est_coverage_clean</td>
<td><strong>Float</strong></td>
<td>Estimated coverage on the clean reads by nanoplot</td>
</tr>
<tr>
<td>nanoplot_r1_est_coverage_raw</td>
<td><strong>Float</strong></td>
<td>Estimated coverage on the raw reads by nanoplot</td>
</tr>
<tr>
<td>nanoplot_r1_mean_q_clean</td>
<td><strong>Float</strong></td>
<td>Mean quality score of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_q_raw</td>
<td><strong>Float</strong></td>
<td>Mean quality score of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_readlength_clean</td>
<td><strong>Float</strong></td>
<td>Mean read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_readlength_raw</td>
<td><strong>Float</strong></td>
<td>Mean read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_q_clean</td>
<td><strong>Float</strong></td>
<td>Median quality score of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_q_raw</td>
<td><strong>Float</strong></td>
<td>Median quality score of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_readlength_clean</td>
<td><strong>Float</strong></td>
<td>Median read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_readlength_raw</td>
<td><strong>Float</strong></td>
<td>Median read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_n50_clean</td>
<td><strong>Float</strong></td>
<td>N50 of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_n50_raw</td>
<td><strong>Float</strong></td>
<td>N50 of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_stdev_readlength_clean</td>
<td><strong>Float</strong></td>
<td>Standard deviation read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_stdev_readlength_raw</td>
<td><strong>Float</strong></td>
<td>Standard deviation read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_tsv_clean</td>
<td><strong>File</strong></td>
<td>A TSV report describing the clean reads</td>
</tr>
<tr>
<td>nanoplot_tsv_raw</td>
<td><strong>File</strong></td>
<td>A TSV report describing the raw reads</td>
</tr>
<tr>
<td>nanoplot_version</td>
<td><strong>String</strong></td>
<td>Version of nanoplot used for analysis</td>
</tr>
<tr>
<td>nanoq_version</td>
<td><strong>String</strong></td>
<td>Version of nanoq used in analysis</td>
</tr>
<tr>
<td>number_contigs</td>
<td><strong>Int</strong></td>
<td>Total number of contigs in assembly</td>
</tr>
<tr>
<td>polypolish_version</td>
<td><strong>String</strong></td>
<td>Version of Polypolish used</td>
</tr>
<tr>
<td>porechop_version</td>
<td><strong>String</strong></td>
<td>Version of Porechop used</td>
</tr>
<tr>
<td>quast_gc_percent</td>
<td><strong>Float</strong></td>
<td>The GC percent of your sample</td>
</tr>
<tr>
<td>quast_report</td>
<td><strong>File</strong></td>
<td>TSV report from QUAST</td>
</tr>
<tr>
<td>quast_version</td>
<td><strong>String</strong></td>
<td>The version of QUAST</td>
</tr>
<tr>
<td>racon_version</td>
<td><strong>String</strong></td>
<td>Version of Racon used</td>
</tr>
<tr>
<td>read1_clean</td>
<td><strong>File</strong></td>
<td>Forward read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read_screen_clean</td>
<td><strong>String</strong></td>
<td>PASS or FAIL result from clean read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_clean_tsv</td>
<td><strong>File</strong></td>
<td>Clean read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>read_screen_raw</td>
<td><strong>String</strong></td>
<td>PASS or FAIL result from raw read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_raw_tsv</td>
<td><strong>File</strong></td>
<td>Raw read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>theiaeuk_ont_analysis_date</td>
<td><strong>String</strong></td>
<td>Date of TheiaEuk_ONT workflow execution</td>
</tr>
<tr>
<td>theiaeuk_ont_version</td>
<td><strong>String</strong></td>
<td>TheiaEuk_ONT workflow version used</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_bai</td>
<td><strong>String</strong></td>
<td>Indexed bam file of the reads aligned to the reference</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_bam</td>
<td><strong>String</strong></td>
<td>BAM file of reads aligned to the reference</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_coverage_tsv</td>
<td><strong>String</strong></td>
<td>Coverage statistics TSV file output by the <code>samtools coverage</code> command, providing genome-wide metrics such as the proportion of bases covered (depth  1), mean depth, and other related statistics.</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_gene_query_results</td>
<td><strong>String</strong></td>
<td>CSV file detailing results for mutations associated with the query strings specified by the user</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_hits</td>
<td><strong>String</strong></td>
<td>A summary of mutations associated with the query strings specified by the user</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_num_reads_aligned</td>
<td><strong>String</strong></td>
<td>Number of reads that aligned to the reference genome as calculated by samtools view -c command</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_num_variants</td>
<td><strong>String</strong></td>
<td>Number of variants detected between sample and reference genome</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_outdir_tarball</td>
<td><strong>String</strong></td>
<td>A compressed file containing the whole directory of snippy output files. This is used when running Snippy_Tree</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_percent_ref_coverage</td>
<td><strong>String</strong></td>
<td>Proportion of the reference genome covered by reads with a depth greater than or equal to the <code>min_coverage</code> threshold (default is 10).</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_query</td>
<td><strong>String</strong></td>
<td>Gene query strings specified by the user when running the workflow</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_query_check</td>
<td><strong>String</strong></td>
<td>Verification that query strings are found in the reference genome</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_results</td>
<td><strong>String</strong></td>
<td>CSV file detailing results for all mutations identified relative to the reference</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_summary</td>
<td><strong>String</strong></td>
<td>A summary TXT fie showing the number of mutations identified for each mutation type</td>
</tr>
<tr>
<td>theiaeuk_snippy_variants_version</td>
<td><strong>String</strong></td>
<td>Version of Snippy used</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="October 2, 2025 16:59:33 UTC">2025-10-02</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="September 11, 2024 17:48:55 UTC">2024-09-11</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
       2022-2025 <a href="https://www.theiagen.com" target="_blank" rel="noopener">Theiagen Genomics</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/theiagen" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://bsky.app/profile/theiagen.bsky.social" target="_blank" rel="noopener" title="bsky.app" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 7.0.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M407.8 294.7c-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3M288 227.1c-26.1-50.7-97.1-145.2-163.1-191.8C61.6-9.4 37.5-1.7 21.6 5.5 3.3 13.8 0 41.9 0 58.4S9.1 194 15 213.9c19.5 65.7 89.1 87.9 153.2 80.7 3.3-.5 6.6-.9 10-1.4-3.3.5-6.6 1-10 1.4-93.9 14-177.3 48.2-67.9 169.9C220.6 589.1 265.1 437.8 288 361.1c22.9 76.7 49.2 222.5 185.6 103.4 102.4-103.4 28.1-156-65.8-169.9-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3 64.1 7.1 133.6-15.1 153.2-80.7C566.9 194 576 75 576 58.4s-3.3-44.7-21.6-52.9c-15.8-7.1-40-14.9-103.2 29.8C385.1 81.9 314.1 176.4 288 227.1"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.linkedin.com/company/theiagen" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3M135.4 416H69V202.2h66.5V416zM102.2 96a38.5 38.5 0 1 1 0 77 38.5 38.5 0 1 1 0-77m282.1 320h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9z"></path></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "navigation.instant", "navigation.path", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.suggest", "toc.follow"], "search": "../../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"alias": true, "default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../../../javascripts/tablesort.js"></script>
      
        <script src="../../../javascripts/table-search.js"></script>
      
    
  
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>