<!DOCTYPE html><html lang="en" class="no-js"><head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A collection of bioinformatics workflows for genomic characterization, submission preparation, and genomic epidemiology of pathogens of public health concern.">
      
      
        <meta name="author" content="Theiagen Genomics">
      
      
        <link rel="canonical" href="https://theiagen.github.io/public_health_bioinformatics/main/workflows/genomic_characterization/theiaprok/">
      
      
        <link rel="prev" href="../theiameta/">
      
      
        <link rel="next" href="../../standalone/theiavalidate/">
      
      
        
      
      
      <link rel="icon" href="../../../assets/logos/Theiagen-Symbol-Standard-01.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.0">
    
    
      
        <title>TheiaProk Workflow Series - Public Health Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.618322db.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  <link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#theiaprok-workflow-series" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version release.
  <a href="../../../.."> 
    <strong>Click here to go to the latest version release.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-header__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Public Health Bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              TheiaProk Workflow Series
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo" aria-label="Switch to Dark Mode" type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to Dark Mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"></path></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo" aria-label="Switch to Light Mode" type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to Light Mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m3.55 19.09 1.41 1.41 1.8-1.79-1.42-1.42M12 6c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6c0-3.32-2.69-6-6-6m8 7h3v-2h-3m-2.76 7.71 1.8 1.79 1.41-1.41-1.79-1.8M20.45 5l-1.41-1.4-1.8 1.79 1.42 1.42M13 1h-2v3h2M6.76 5.39 4.96 3.6 3.55 5l1.79 1.81zM1 13h3v-2H1m12 9h-2v3h2"></path></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../.." class="md-tabs__link">
          
  
  
  Public Health Bioinformatics

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-nav__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    Public Health Bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Public Health Bioinformatics
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            
  
    Public Health Bioinformatics
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/commandline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    With the Command-Line
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/terra/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    With Terra.bio
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/philosophy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Workflow Failure Philosophy
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Other Resources
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Other Resources
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/sops/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Available SOPs
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/versions/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Database and Docker Versions
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Guides
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Guides
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guide to GAMBIT
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit_database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guide to GAMBIT Database Creation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/phylogenetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guide to Phylogenetics
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/custom_organisms/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guide to Running Custom Organisms on TheiaCoV
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Workflows
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Workflows
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1">
        
          
          <label class="md-nav__link" for="__nav_1_5_1" id="__nav_1_5_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Workflows by Type
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1">
            <span class="md-nav__icon md-icon"></span>
            
  
    Workflows by Type
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_type/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_2" id="__nav_1_5_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Data Import
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Data Import
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Assembly_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BaseSpace_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Create_Terra_Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fetch_SRR_Accession
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    ONT_Barcode_Concatenation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    SRA_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_3" id="__nav_1_5_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Genomic Characterization
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Genomic Characterization
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Freyja Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Pangolin_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaCoV Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaEuk Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaMeta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    TheiaProk Workflow Series
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaProk Workflow Series
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Facts
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaprok-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      
        TheiaProk Workflows
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaProk Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Inputs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Core Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembly Tasks
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#post-assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Post-Assembly Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Post-Assembly Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Control
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#taxonomic-assignment" class="md-nav__link">
    <span class="md-ellipsis">
      
        Taxonomic Assignment
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#amr-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      
        AMR Characterization
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sequence-type" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sequence Type
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembly-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembly Annotation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plasmid-identification" class="md-nav__link">
    <span class="md-ellipsis">
      
        Plasmid Identification
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#taxa-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Taxa-Specific Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Taxa-Specific Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#acinetobacter-baumannii" class="md-nav__link">
    <span class="md-ellipsis">
      
        Acinetobacter baumannii
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#escherichia-or-shigella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Escherichia or Shigella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#haemophilus-influenzae" class="md-nav__link">
    <span class="md-ellipsis">
      
        Haemophilus influenzae
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#klebsiella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Klebsiella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legionella-pneumophila" class="md-nav__link">
    <span class="md-ellipsis">
      
        Legionella pneumophila
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#listeria-monocytogenes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Listeria monocytogenes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mycobacterium-tuberculosis" class="md-nav__link">
    <span class="md-ellipsis">
      
        Mycobacterium tuberculosis
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#neisseria" class="md-nav__link">
    <span class="md-ellipsis">
      
        Neisseria spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pseudomonas-aeruginosa" class="md-nav__link">
    <span class="md-ellipsis">
      
        Pseudomonas aeruginosa
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#salmonella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Salmonella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphyloccocus-aureus" class="md-nav__link">
    <span class="md-ellipsis">
      
        Staphyloccocus aureus
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#streptococcus-pneumoniae" class="md-nav__link">
    <span class="md-ellipsis">
      
        Streptococcus pneumoniae
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#streptococcus-pyogenes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Streptococcus pyogenes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vibrio" class="md-nav__link">
    <span class="md-ellipsis">
      
        Vibrio spp.
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaViral Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    VADR_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_4" id="__nav_1_5_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Phylogenetic Construction
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Phylogenetic Construction
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Augur
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Core_Gene_SNP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Clair3_Variants_ONT
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    CZGenEpi_Prep
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Find_Shared_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP4
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Lyve_SET
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MashTree_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Tree
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_5" id="__nav_1_5_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Phylogenetic Placement
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Phylogenetic Placement
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Nextclade_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Usher
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_6" id="__nav_1_5_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Public Data Sharing
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Public Data Sharing
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Mercury_Prep_N_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_ENA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_GISAID
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_NCBI
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_7">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_7" id="__nav_1_5_1_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Exporting Data from Terra
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_7">
            <span class="md-nav__icon md-icon"></span>
            
  
    Exporting Data from Terra
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Transfer_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Zip_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_8">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_8" id="__nav_1_5_1_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Standalone
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_8">
            <span class="md-nav__icon md-icon"></span>
            
  
    Standalone
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    AMR_Search
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Cauris_CladeTyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Dorado_Basecalling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Illumina_Lanes
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GAMBIT_Query
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Kraken2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-AMRFinderPlus
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-Scrub
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    PhyloCompare
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    RASUSA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Rename_FASTQ
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TBProfiler_tNGS
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaValidate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2" id="__nav_1_5_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Workflows by Kingdom
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Workflows by Kingdom
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_kingdom/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_2" id="__nav_1_5_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Any Taxa
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Any Taxa
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    AMR_Search
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Assembly_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BaseSpace_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Clair3_Variants_ONT
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Illumina_Lanes
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Create_Terra_Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Dorado_Basecalling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fetch_SRR_Accession
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Kraken2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-Scrub
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    ONT_Barcode_Concatenation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    PhyloCompare
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    RASUSA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Rename_FASTQ
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    SRA_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaMeta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaValidate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Transfer_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Zip_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_3" id="__nav_1_5_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Bacteria
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Bacteria
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Core_Gene_SNP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Find_Shared_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GAMBIT_Query
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP4
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Lyve_SET
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MashTree_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-AMRFinderPlus
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Tree
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TBProfiler_tNGS
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_ENA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_NCBI
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    TheiaProk Workflow Series
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaProk Workflow Series
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Facts
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaprok-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      
        TheiaProk Workflows
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaProk Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Inputs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Core Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembly Tasks
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#post-assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Post-Assembly Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Post-Assembly Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Control
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#taxonomic-assignment" class="md-nav__link">
    <span class="md-ellipsis">
      
        Taxonomic Assignment
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#amr-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      
        AMR Characterization
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sequence-type" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sequence Type
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembly-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembly Annotation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plasmid-identification" class="md-nav__link">
    <span class="md-ellipsis">
      
        Plasmid Identification
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#taxa-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Taxa-Specific Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Taxa-Specific Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#acinetobacter-baumannii" class="md-nav__link">
    <span class="md-ellipsis">
      
        Acinetobacter baumannii
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#escherichia-or-shigella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Escherichia or Shigella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#haemophilus-influenzae" class="md-nav__link">
    <span class="md-ellipsis">
      
        Haemophilus influenzae
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#klebsiella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Klebsiella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legionella-pneumophila" class="md-nav__link">
    <span class="md-ellipsis">
      
        Legionella pneumophila
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#listeria-monocytogenes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Listeria monocytogenes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mycobacterium-tuberculosis" class="md-nav__link">
    <span class="md-ellipsis">
      
        Mycobacterium tuberculosis
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#neisseria" class="md-nav__link">
    <span class="md-ellipsis">
      
        Neisseria spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pseudomonas-aeruginosa" class="md-nav__link">
    <span class="md-ellipsis">
      
        Pseudomonas aeruginosa
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#salmonella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Salmonella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphyloccocus-aureus" class="md-nav__link">
    <span class="md-ellipsis">
      
        Staphyloccocus aureus
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#streptococcus-pneumoniae" class="md-nav__link">
    <span class="md-ellipsis">
      
        Streptococcus pneumoniae
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#streptococcus-pyogenes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Streptococcus pyogenes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vibrio" class="md-nav__link">
    <span class="md-ellipsis">
      
        Vibrio spp.
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_4" id="__nav_1_5_2_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Mycotics
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Mycotics
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Cauris_CladeTyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Find_Shared_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GAMBIT_Query
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP4
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MashTree_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-AMRFinderPlus
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_NCBI
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaEuk Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_5" id="__nav_1_5_2_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Viral
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Viral
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Augur
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    CZGenEpi_Prep
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Freyja Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP4
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MashTree_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Mercury_Prep_N_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Nextclade_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Pangolin_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_ENA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_GISAID
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_NCBI
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaCoV Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaViral Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Usher
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    VADR_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_3" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Workflows Alphabetically
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Workflows Alphabetically
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_alphabetically/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    AMR_Search
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Assembly_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Augur
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BaseSpace_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Cauris_CladeTyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Clair3_Variants_ONT
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Illumina_Lanes
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Core_Gene_SNP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Create_Terra_Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    CZGenEpi_Prep
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Dorado_Basecalling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fetch_SRR_Accession
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Find_Shared_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Freyja Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GAMBIT_Query
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Kraken2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP4
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Lyve_SET
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MashTree_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Mercury_Prep_N_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-AMRFinderPlus
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-Scrub
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Nextclade_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    ONT_Barcode_Concatenation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Pangolin_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    PhyloCompare
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    RASUSA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Rename_FASTQ
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Tree
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    SRA_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TBProfiler_tNGS
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_ENA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_GISAID
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_NCBI
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaCoV Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaEuk Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaMeta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    TheiaProk Workflow Series
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaProk Workflow Series
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Facts
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaprok-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      
        TheiaProk Workflows
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaProk Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Inputs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Core Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembly Tasks
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#post-assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Post-Assembly Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Post-Assembly Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Control
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#taxonomic-assignment" class="md-nav__link">
    <span class="md-ellipsis">
      
        Taxonomic Assignment
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#amr-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      
        AMR Characterization
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sequence-type" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sequence Type
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembly-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembly Annotation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plasmid-identification" class="md-nav__link">
    <span class="md-ellipsis">
      
        Plasmid Identification
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#taxa-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Taxa-Specific Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Taxa-Specific Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#acinetobacter-baumannii" class="md-nav__link">
    <span class="md-ellipsis">
      
        Acinetobacter baumannii
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#escherichia-or-shigella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Escherichia or Shigella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#haemophilus-influenzae" class="md-nav__link">
    <span class="md-ellipsis">
      
        Haemophilus influenzae
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#klebsiella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Klebsiella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legionella-pneumophila" class="md-nav__link">
    <span class="md-ellipsis">
      
        Legionella pneumophila
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#listeria-monocytogenes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Listeria monocytogenes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mycobacterium-tuberculosis" class="md-nav__link">
    <span class="md-ellipsis">
      
        Mycobacterium tuberculosis
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#neisseria" class="md-nav__link">
    <span class="md-ellipsis">
      
        Neisseria spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pseudomonas-aeruginosa" class="md-nav__link">
    <span class="md-ellipsis">
      
        Pseudomonas aeruginosa
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#salmonella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Salmonella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphyloccocus-aureus" class="md-nav__link">
    <span class="md-ellipsis">
      
        Staphyloccocus aureus
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#streptococcus-pneumoniae" class="md-nav__link">
    <span class="md-ellipsis">
      
        Streptococcus pneumoniae
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#streptococcus-pyogenes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Streptococcus pyogenes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vibrio" class="md-nav__link">
    <span class="md-ellipsis">
      
        Vibrio spp.
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaValidate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaViral Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Transfer_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Usher
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    VADR_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Zip_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Contributing
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Contributing
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/doc_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributing to the Documentation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/code_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributing to the Code
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Facts
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiaprok-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      
        TheiaProk Workflows
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaProk Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Inputs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Core Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembly Tasks
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#post-assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Post-Assembly Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Post-Assembly Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Control
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#taxonomic-assignment" class="md-nav__link">
    <span class="md-ellipsis">
      
        Taxonomic Assignment
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#amr-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      
        AMR Characterization
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sequence-type" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sequence Type
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembly-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembly Annotation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plasmid-identification" class="md-nav__link">
    <span class="md-ellipsis">
      
        Plasmid Identification
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#taxa-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Taxa-Specific Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Taxa-Specific Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#acinetobacter-baumannii" class="md-nav__link">
    <span class="md-ellipsis">
      
        Acinetobacter baumannii
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#escherichia-or-shigella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Escherichia or Shigella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#haemophilus-influenzae" class="md-nav__link">
    <span class="md-ellipsis">
      
        Haemophilus influenzae
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#klebsiella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Klebsiella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#legionella-pneumophila" class="md-nav__link">
    <span class="md-ellipsis">
      
        Legionella pneumophila
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#listeria-monocytogenes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Listeria monocytogenes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mycobacterium-tuberculosis" class="md-nav__link">
    <span class="md-ellipsis">
      
        Mycobacterium tuberculosis
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#neisseria" class="md-nav__link">
    <span class="md-ellipsis">
      
        Neisseria spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pseudomonas-aeruginosa" class="md-nav__link">
    <span class="md-ellipsis">
      
        Pseudomonas aeruginosa
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#salmonella" class="md-nav__link">
    <span class="md-ellipsis">
      
        Salmonella spp.
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#staphyloccocus-aureus" class="md-nav__link">
    <span class="md-ellipsis">
      
        Staphyloccocus aureus
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#streptococcus-pneumoniae" class="md-nav__link">
    <span class="md-ellipsis">
      
        Streptococcus pneumoniae
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#streptococcus-pyogenes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Streptococcus pyogenes
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vibrio" class="md-nav__link">
    <span class="md-ellipsis">
      
        Vibrio spp.
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
                




  <nav class="md-path" aria-label="Navigation">
    <ol class="md-path__list">
      
      
        
  
  
    
    
      <li class="md-path__item">
        <a href="../../.." class="md-path__link">
          
  <span class="md-ellipsis">
    Public Health Bioinformatics
  </span>

        </a>
      </li>
    
  

      
        
  
  
    
    
      
  
  
    
    
      <li class="md-path__item">
        <a href="../../../workflows_overview/workflows_type/" class="md-path__link">
          
  <span class="md-ellipsis">
    Workflows
  </span>

        </a>
      </li>
    
  

    
  

      
        
  
  
    
    
      <li class="md-path__item">
        <a href="../../../workflows_overview/workflows_alphabetically/" class="md-path__link">
          
  <span class="md-ellipsis">
    Workflows Alphabetically
  </span>

        </a>
      </li>
    
  

      
    </ol>
  </nav>

              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="theiaprok-workflow-series">TheiaProk Workflow Series<a class="headerlink" href="#theiaprok-workflow-series" title="Permanent link"></a></h1>
<h2 id="quick-facts">Quick Facts<a class="headerlink" href="#quick-facts" title="Permanent link"></a></h2>
<table>
<thead>
<tr>
<th><strong>Workflow Type</strong></th>
<th><strong>Applicable Kingdom</strong></th>
<th><strong>Last Known Changes</strong></th>
<th><strong>Command-line Compatibility</strong></th>
<th><strong>Workflow Level</strong></th>
<th><strong>Dockstore</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../../../workflows_overview/workflows_type/#genomic-characterization">Genomic Characterization</a></td>
<td><a href="../../../workflows_overview/workflows_kingdom/#bacteria">Bacteria</a></td>
<td>vX.X.X</td>
<td>Some optional features incompatible, Yes</td>
<td>Sample-level</td>
<td><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaProk_Illumina_PE_PHB:main?tab=info">TheiaProk_Illumina_PE_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaProk_Illumina_SE_PHB:main?tab=info">TheiaProk_Illumina_SE_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaProk_ONT_PHB:main?tab=info">TheiaProk_ONT_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaProk_FASTA_PHB:main?tab=info">TheiaProk_FASTA_PHB</a></td>
</tr>
</tbody>
</table>
<h2 id="theiaprok-workflows">TheiaProk Workflows<a class="headerlink" href="#theiaprok-workflows" title="Permanent link"></a></h2>
<p><strong>The TheiaProk workflows are for the assembly, quality assessment, and characterization of bacterial genomes.</strong> There are currently four TheiaProk workflows designed to accommodate different kinds of input data:</p>
<ol>
<li>Illumina paired-end sequencing <strong>(TheiaProk_Illumina_PE</strong>)</li>
<li>Illumina single-end sequencing (<strong>TheiaProk_Illumina_SE)</strong></li>
<li>ONT sequencing (<strong>TheiaProk_ONT</strong>)</li>
<li>Genome assemblies (<strong>TheiaProk_FASTA</strong>)</li>
</ol>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="theiaprok_illumina_pe" name="__tabbed_1" type="radio"><input id="theiaprok_ont" name="__tabbed_1" type="radio"><div class="tabbed-labels"><label for="theiaprok_illumina_pe">TheiaProk_Illumina_PE</label><label for="theiaprok_ont">TheiaProk_ONT</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="admonition caption">
<p class="admonition-title">TheiaProk Illumina PE Workflow Diagram</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/TheiaProk_Illumina.png" data-desc-position="bottom"><img alt="TheiaProk Illumina PE Workflow Diagram" src="../../../assets/figures/TheiaProk_Illumina.png"></a></p>
</div>
</div>
<div class="tabbed-block">
<div class="admonition caption">
<p class="admonition-title">TheiaProk ONT Workflow Diagram</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/TheiaProk_ONT.png" data-desc-position="bottom"><img alt="TheiaProk ONT Workflow Diagram" src="../../../assets/figures/TheiaProk_ONT.png"></a></p>
</div>
</div>
</div>
</div>
<p>All input reads are processed through "<a href="#core-tasks">core tasks</a>" in the TheiaProk Illumina and ONT workflows. These undertake read trimming and assembly appropriate to the input data type. TheiaProk workflows subsequently launch default genome characterization modules for quality assessment, species identification, antimicrobial resistance gene detection, sequence typing, and more. <strong>For some taxa identified, "<a href="#taxa-specific-tasks">taxa-specific sub-workflows</a>" will be automatically activated, undertaking additional taxa-specific characterization steps</strong>. When setting up each workflow, users may choose to use "optional tasks" as additions or alternatives to tasks run in the workflow by default.</p>
<h3 id="inputs">Inputs<a class="headerlink" href="#inputs" title="Permanent link"></a></h3>
<div class="admonition dna">
<p class="admonition-title">Input Data</p>
<div class="tabbed-set tabbed-alternate" data-tabs="2:4"><input checked="checked" id="theiaprok_illumina_pe_1" name="__tabbed_2" type="radio"><input id="theiaprok_illumina_se" name="__tabbed_2" type="radio"><input id="theiaprok_ont_1" name="__tabbed_2" type="radio"><input id="theiaprok_fasta" name="__tabbed_2" type="radio"><div class="tabbed-labels"><label for="theiaprok_illumina_pe_1">TheiaProk_Illumina_PE</label><label for="theiaprok_illumina_se">TheiaProk_Illumina_SE</label><label for="theiaprok_ont_1">TheiaProk_ONT</label><label for="theiaprok_fasta">TheiaProk_FASTA</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>The TheiaProk_Illumina_PE workflow takes in Illumina paired-end read data. Read file names should end with <code>.fastq</code> or <code>.fq</code>, with the optional addition of <code>.gz</code>. When possible, Theiagen recommends zipping files with <a href="https://www.gnu.org/software/gzip/">gzip</a> before Terra uploads to minimize data upload time.</p>
<p>By default, the workflow anticipates&nbsp;<strong>2 x 150bp</strong>&nbsp;reads (i.e. the input reads were generated using a 300-cycle sequencing kit). Modifications to the optional parameter for <code>trim_minlen</code> may be required to accommodate shorter read data, such as the 2 x 75bp reads generated using a 150-cycle sequencing kit.</p>
</div>
<div class="tabbed-block">
<p>TheiaProk_Illumina_SE takes in Illumina single-end reads. Read file names should end with <code>.fastq</code> or <code>.fq</code>, with the optional addition of <code>.gz</code>. Theiagen highly recommends zipping files with <a href="https://www.gnu.org/software/gzip/">gzip</a> before uploading to Terra to minimize data upload time &amp; save on storage costs.</p>
<p>By default, the workflow anticipates <strong>1 x 35 bp</strong> reads  (i.e. the input reads were generated using a 70-cycle sequencing kit). Modifications to the optional parameter for <code>trim_minlen</code> may be required to accommodate longer read data.</p>
</div>
<div class="tabbed-block">
<p>The TheiaProk_ONT workflow takes in base-called ONT read data. Read file names should end with <code>.fastq</code> or <code>.fq</code>, with the optional addition of <code>.gz</code>. When possible, Theiagen recommends zipping files with <a href="https://www.gnu.org/software/gzip/">gzip</a> before uploading to Terra to minimize data upload time.</p>
<p><strong>The ONT sequencing kit and base-calling approach can produce substantial variability in the amount and quality of read data. Genome assemblies produced by the TheiaProk_ONT workflow must be quality assessed before reporting results.</strong></p>
</div>
<div class="tabbed-block">
<p>The TheiaProk_FASTA workflow takes in assembly files in FASTA format.</p>
</div>
</div>
</div>
</div>
<div class="admonition tip">
<p class="admonition-title">Skip Characterization</p>
<p>Ever wanted to skip characterization? Now you can! Set the optional input <code>perform_characterization</code> to <strong><code>false</code></strong> to only generate an assembly and run assembly QC.</p>
</div>
<div class="admonition caption">
<div class="tabbed-set tabbed-alternate" data-tabs="3:4"><input checked="checked" id="theiaprok_illumina_pe_2" name="__tabbed_3" type="radio"><input id="theiaprok_illumina_se_1" name="__tabbed_3" type="radio"><input id="theiaprok_ont_2" name="__tabbed_3" type="radio"><input id="theiaprok_fasta_1" name="__tabbed_3" type="radio"><div class="tabbed-labels"><label for="theiaprok_illumina_pe_2">TheiaProk_Illumina_PE</label><label for="theiaprok_illumina_se_1">TheiaProk_Illumina_SE</label><label for="theiaprok_ont_2">TheiaProk_ONT</label><label for="theiaprok_fasta_1">TheiaProk_FASTA</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>Illumina forward read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Illumina reverse read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA %coverage for the Abricate task</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA %identity for the Abricate task</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>detailed_drug_class</strong></td>
<td>Boolean</td>
<td>If set to true, amrfinderplus_amr_classes and amrfinderplus_amr_subclasses outputs will be created</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-amrfinderplus:4.0.23-2025-07-16.1</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>hide_point_mutations</strong></td>
<td>Boolean</td>
<td>If set to true, point mutations are not reported</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum proportion of reference gene covered for a BLAST-based hit (Methods BLAST or PARTIAL)." Attribute should be a float ranging from 0-1, such as 0.6 (equal to 60% coverage)</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum identity for a blast-based hit hit (Methods BLAST or PARTIAL). -1 means use a curated threshold if it exists and 0.9 otherwise. Setting this value to something other than -1 will override any curated similarity cutoffs. Attribute should be a float ranging from 0-1, such as 0.95 (equal to 95% identity)</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>separate_betalactam_genes</strong></td>
<td>Boolean</td>
<td>Report beta-Lactam AMR genes separated out by all beta-lactam and the respective beta-lactam subclasses</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>ani_threshold</strong></td>
<td>Float</td>
<td>ANI value threshold must be surpassed in order to output the ani_top_species_match. If a genome does not surpass this threshold (and the percent_bases_aligned_threshold) then the ani_top_species_match output String will show a warning instead of a genus &amp; species.</td>
<td>80.0</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mummer:4.0.0-rgdv2</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>mash_filter</strong></td>
<td>Float</td>
<td>Mash distance threshold over which ANI is not calculated</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>percent_bases_aligned_threshold</strong></td>
<td>Float</td>
<td>Threshold regarding the proportion of bases aligned between the query genome and reference genome. If a genome does not surpass this threshold (and the ani_threshold) then the ani_top_species_match output String will show a warning instead of a genus &amp; species.</td>
<td>70.0</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>ref_genome</strong></td>
<td>File</td>
<td>If not set, uses all 43 genomes in RGDv2</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/arln_stats:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>bakta_opts</strong></td>
<td>String</td>
<td>Parameters to pass to bakta from <a href="https://github.com/oschwengers/bakta#usage">https://github.com/oschwengers/bakta#usage</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>compliant</strong></td>
<td>Boolean</td>
<td>If true, forces Genbank/ENA/DDJB compliant headers in Bakta output files</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>200</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/bakta:1.10.3</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>24</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>prodigal_tf</strong></td>
<td>File</td>
<td>Prodigal training file to use for CDS prediction by bakta</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>proteins</strong></td>
<td>File</td>
<td>Fasta file of trusted protein sequences for CDS annotation</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ezlabgva/busco:v5.7.1_cv1</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>eukaryote</strong></td>
<td>Boolean</td>
<td>Assesses eukaryotic organisms, rather than prokaryotic organisms</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>cg_pipe_opts</strong></td>
<td>String</td>
<td>Options to pass to CG-Pipeline for clean read assessment</td>
<td>--fast</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/lyveset:1.1.4f</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>cg_pipe_opts</strong></td>
<td>String</td>
<td>Options to pass to CG-Pipeline for raw read assessment</td>
<td>--fast</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/lyveset:1.1.4f</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>assembler</strong></td>
<td>String</td>
<td>Assembler to use (spades, skesa, megahit)</td>
<td>skesa</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>assembler_options</strong></td>
<td>String</td>
<td>Assembler-specific options that you might choose for the selected assembler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>6</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>call_pilon</strong></td>
<td>Boolean</td>
<td>Whether to run Pilon polishing after assembly</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/shovilter:0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_min_coverage</strong></td>
<td>Float</td>
<td>Minimum coverage threshold for contig filtering</td>
<td>2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_min_length</strong></td>
<td>Int</td>
<td>Minimum length threshold for contig filtering</td>
<td>200</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_skip_coverage_filter</strong></td>
<td>Boolean</td>
<td>Skip filtering contigs based on coverage</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_skip_homopolymer_filter</strong></td>
<td>Boolean</td>
<td>Skip filtering contigs containing homopolymers</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_skip_length_filter</strong></td>
<td>Boolean</td>
<td>Skip filtering contigs based on length</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>kmers</strong></td>
<td>String</td>
<td>K-mer sizes for assembly (comma-separated)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/megahit:1.2.9</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>min_contig_length</strong></td>
<td>Int</td>
<td>Minimum contig length to retain in final assembly</td>
<td>200</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/pilon:1.24--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_fix</strong></td>
<td>String</td>
<td>Potential issues with assembly to try and automatically fix (snps, indels, gaps, local, all, bases, none)</td>
<td>bases</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_min_base_quality</strong></td>
<td>Int</td>
<td>Minimum base quality to keep</td>
<td>3</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_min_depth</strong></td>
<td>Float</td>
<td>Minimum coverage threshold for variant calling: when set to a value 1, it requires that absolute depth of coverage; when set to a fraction &lt;1, it requires coverage at least that fraction of the mean coverage for the region</td>
<td>0.25</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_min_mapping_quality</strong></td>
<td>Int</td>
<td>Minimum mapping quality for a read to count in pileups</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>run_filter_contigs</strong></td>
<td>Boolean</td>
<td>Whether to run contig filtering step</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/skesa:2.4.0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/spades:4.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_type</strong></td>
<td>String</td>
<td>SPAdes assembly mode (isolate, meta, rna, etc.), more can be found <a href="https://ablab.github.io/spades/running.html">here</a></td>
<td>isolate</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>theiaviral_panel</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>gambit_db_genomes</strong></td>
<td>File</td>
<td>Database of metadata for assembled query genomes; requires complementary signatures file. If not provided, uses default database "/gambit-db"</td>
<td>gs://gambit-databases-rp/2.1.0/gambit-metadata-2.1.0-20250808.gdb</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>gambit_db_signatures</strong></td>
<td>File</td>
<td>Signatures file; requires complementary genomes file. If not specified, the file from the docker container will be used.</td>
<td>gs://gambit-databases-rp/2.1.0/gambit-signatures-2.1.0-20250808.gs</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gamma:2.2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>extended_output</strong></td>
<td>Boolean</td>
<td>GAMMA will provide the extended output when set to true</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>gamma_db</strong></td>
<td>File</td>
<td>Multifasta sequence database GAMMA is to use for gene matching</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/gamma/default_ResFinderDB_Combined_05-06-20.fsa</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>min_length_percent_gammas</strong></td>
<td>Int</td>
<td>Threshold of match length percent used by GAMMA-s</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>min_percent_identity</strong></td>
<td>Int</td>
<td>Threshold of identity for gene matches in percent</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>output_fasta</strong></td>
<td>Boolean</td>
<td>Allows GAMMA to output gene matches in FASTA format</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>output_gff</strong></td>
<td>Boolean</td>
<td>Allows GAMMA to output gene matches in GFF format</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>run_gammas</strong></td>
<td>Boolean</td>
<td>Turns on GAMMA-S which runs GAMMA on nucleotide sequences rather than translating input sequences and matches</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/kmerfinder:3.0.2--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>kmerfinder_args</strong></td>
<td>String</td>
<td>Kmerfinder additional arguments</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>kmerfinder_db</strong></td>
<td>File</td>
<td>Bacterial database for KmerFinder</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kmerfinder/kmerfinder_bacteria_20230911.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity; set to 95 because there is a strict threshold of 95% identity for typing purposes</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_agr_typing_only</strong></td>
<td>Boolean</td>
<td>Set to true to skip agr operon extraction and frameshift detection</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/agrvate:1.0.2--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/amrsearch:0.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>assembly_only</strong></td>
<td>Boolean</td>
<td>Set to true if only analyzing input assembly</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_poppunk</strong></td>
<td>Boolean</td>
<td>If true, runs PopPUNK for GPSC cluster designation for S. pneumoniae</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_shigeifinder_reads_input</strong></td>
<td>Boolean</td>
<td>If set to "true", the ShigEiFinder task will run again but using read files as input instead of the assembly file. Input is shown but not used for TheiaProk_FASTA.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_stxtyper</strong></td>
<td>Boolean</td>
<td>If set to "true", the StxTyper task will run on all samples regardless of the gambit_predicted_taxon output. Useful if you suspect a non-E.coli or non-Shigella sample contains stx genes.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_tbp_parser</strong></td>
<td>Boolean</td>
<td>If set to "true", activates the tbp_parser module and results in more outputs, including&nbsp;tbp_parser_looker_report_csv, tbp_parser_laboratorian_report_csv,  tbp_parser_lims_report_csv, tbp_parser_coverage_report, and tbp_parser_genome_percent_coverage</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cauris_cladetyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_kmer_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_max_distance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6</strong></td>
<td>File</td>
<td>The reference assembly for clade 6</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade6_GCA_032714025.1_ASM3271402v1_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6_annotated</strong></td>
<td>File</td>
<td>The path to the annotated reference for clade 6</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>clockwork_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/cdcgov/varpipe_wgs_with_refs:2bc7234074bd53d9e92a1048b0485763cd9bbf6f4d12d5a1cc82bfec8ca7d75e</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/ectyper:1.0.0--pyhdfd78af_1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minumum percent coverage required for an H antigen allele match</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_identity</strong></td>
<td>Int</td>
<td>Percent identity required for an H antigen allele match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minumum percent coverage required for an O antigen allele match</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_identity</strong></td>
<td>Int</td>
<td>Percent identity required for an O antigen allele match</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_print_alleles</strong></td>
<td>Boolean</td>
<td>Set to true to print the allele sequences as the final column</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_verify</strong></td>
<td>Boolean</td>
<td>Set to true to enable E. coli species verification</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_align_diff</strong></td>
<td>Int</td>
<td>Threshold for difference between alignment length and subject length in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_cluster_distance</strong></td>
<td>Int</td>
<td>Distance between cluster of matches to consider as different clusters.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_culling_limit</strong></td>
<td>Int</td>
<td>Total hits to return in a position.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/emmtyper:0.2.0--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_gap</strong></td>
<td>Int</td>
<td>Threshold gap to allow in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_max_size</strong></td>
<td>Int</td>
<td>Maximum size of PCR product.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_good</strong></td>
<td>Int</td>
<td>Minimum size where there must be 2 matches for each mismatch.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimal percent identity of sequence.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_perfect</strong></td>
<td>Int</td>
<td>Minimum size of perfect match at 3' primer end.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_mismatch</strong></td>
<td>Int</td>
<td>Threshold for number of mismatch to allow in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_wf</strong></td>
<td>String</td>
<td>Choose workflow [blast</td>
<td>pcr]</td>
<td></td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtypingtool_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/emmtypingtool:0.0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>genotyphi_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.11.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_broken_gene_length</strong></td>
<td>Int</td>
<td>Minimum length to consider a broken gene</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/hicap:1.0.3--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_broken_gene_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percentage identity to consider a broken gene</td>
<td>0.8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum percentage coverage to consider a single gene complete</td>
<td>0.8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percentage identity to consider a single gene complete</td>
<td>0.7</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kaptive:2.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_low_gene_percent_identity</strong></td>
<td>Float</td>
<td>Percent identity threshold for what counts as a low identity match in the gene BLAST search</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum required percent coverage for the gene BLAST search via tBLASTn</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum required percent identity for the gene BLAST search via tBLASTn</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_start_end_margin</strong></td>
<td>Int</td>
<td>Determines flexibility in identifying the start and end of a locus - if this value is 10, a locus match that is missing the first 8 base pairs will still count as capturing the start of the locus</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kleborate:2.2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_kaptive_confidence</strong></td>
<td>String</td>
<td>{None,Low,Good,High,Very_high,Perfect} Minimum Kaptive confidence to call K/O loci - confidence levels below this will be reported as unknown</td>
<td>Good</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum alignment percent coverage for main results</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum alignment percent identity for main results</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum alignment percent coverage for spurious results</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_identity</strong></td>
<td>Float</td>
<td>Minimum alignment percent identity for spurious results</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_kaptive</strong></td>
<td>Boolean</td>
<td>Equivalent to --kaptive_k --kaptive_</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_resistance</strong></td>
<td>Boolean</td>
<td>Set to true to turn on resistance genes screening (default: no resistance gene screening)</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>legsta_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/lissero:0.4.9--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum percent coverage of the gene to accept a match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percent identity to accept a match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>meningotype_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/meningotype:0.8.5--pyhdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ngmaster_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ngmaster:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ont_data</strong></td>
<td>Boolean</td>
<td>Set to true if your data is ONT FASTQ files</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>paired_end</strong></td>
<td>Boolean</td>
<td>Set to true if your data is paired-end FASTQ files</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pasty:1.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum coverage of a O-antigen to be considered for serogrouping by pasty</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum percent identity for a blast hit to be considered for serogrouping</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pbptyper:1.0.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum percent coverage to count a hit</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum percent identity to count a hit</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/poppunk:2.4.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_npy</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.dists.npy</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.dists.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_external_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_external_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_npz</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_fit.npz</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_fit.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_graph_gt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_graph.gt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_h5</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.h5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_qcreport_txt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_qcreport.txt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_npy</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.dists.npy</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.dists.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_graph_gt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6refs_graph.gt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_h5</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.h5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_unword_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_unword_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>run_amr_search</strong></td>
<td>Boolean</td>
<td>If set to true AMR_Search workflow will be run if species is part of supported taxon, see AMR_Search docs.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seqsero2s_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/seqsero2s:1.1.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seroba_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/seroba:1.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>serotypefinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/serotypefinder:2.0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigatyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/shigatyper:2.0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigeifinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/shigeifinder:1.3.5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/sistr_cmd:1.1.1--pyh864c0ab_2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_use_full_cgmlst_db</strong></td>
<td>Boolean</td>
<td>Set to true to use the full set of cgMLST alleles which can include highly similar alleles. By default the smaller "centroid" alleles or representative alleles are used for each marker</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_base_quality</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_gene_query_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_map_qual</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_maxsoft</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_frac</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_quality</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_query_gene</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_afumigatus</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/aspergillus/Aspergillus_fumigatus_GCF_000002655.1_ASM265v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_cryptoneo</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/cryptococcus/Cryptococcus_neoformans_GCF_000091045.1_ASM9104v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_variants_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_mykrobe_opts</strong></td>
<td>String</td>
<td>Additional options for mykrobe in sonneityping</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_do_enrich</strong></td>
<td>Boolean</td>
<td>Set to true to enable PCR product enrichment</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/spatyper:0.3.3--pyhdfd78af_3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/srst2:0.2.0-vcholerae</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_gene_max_mismatch</strong></td>
<td>Int</td>
<td>Maximum number of mismatches for SRST2 to call a gene as present</td>
<td>2000</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_max_divergence</strong></td>
<td>Int</td>
<td>Maximum divergence, in percentage, for SRST2 to call a gene as present</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_depth</strong></td>
<td>Int</td>
<td>Minimum depth of coverage for SRST2 to call a gene as present</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_edge_depth</strong></td>
<td>Int</td>
<td>Minimum edge depth for SRST2 to call a gene as present</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum breadth of coverage for SRST2 to call a gene as present</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>staphopia_sccmec_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/staphopia-sccmec:1.0.0--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/stxtyper:1.0.42</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_enable_debug</strong></td>
<td>Boolean</td>
<td>When enabled, additional messages are printed and files in $TMPDIR are not removed after running</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_add_cs_lims</strong></td>
<td>Boolean</td>
<td>Set to true add cycloserine results to the LIMS report</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_config</strong></td>
<td>File</td>
<td>The configuration file to use, in YAML format (overrides all other arguments except input_json and input_bam)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_coverage_regions_bed</strong></td>
<td>File</td>
<td>A bed file that lists the regions to be considered for QC</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_debug</strong></td>
<td>Boolean</td>
<td>Activate the debug mode on tbp_parser; increases logging outputs</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/tbp-parser:2.6.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_etha237_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in ethA at protein position 237 to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_expert_rule_regions_bed</strong></td>
<td>File</td>
<td>A file that contains the regions where R mutations and expert rules are applied</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_depth</strong></td>
<td>Int</td>
<td>Minimum depth for a variant to pass QC in tbp_parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_frequency</strong></td>
<td>Float</td>
<td>The minimum frequency for a mutation to pass QC</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_percent_coverage</strong></td>
<td>Float</td>
<td>The minimum coverage for a region to pass QC in tbp_parser</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_read_support</strong></td>
<td>Int</td>
<td>The minimum read support for a mutation to pass QC</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_operator</strong></td>
<td>String</td>
<td>Fills the "operator" field in the tbp_parser output files</td>
<td>Operator not provided</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_output_seq_method_type</strong></td>
<td>String</td>
<td>Fills out the "seq_method" field in the tbp_parser output files</td>
<td>WGS</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rpob449_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation at protein position 449 to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in rrl to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_read_support</strong></td>
<td>Int</td>
<td>Minimum read support for a mutation in rrl to pass QC in tbp-parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in rrs to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_read_support</strong></td>
<td>Int</td>
<td>Minimum read support for a mutation in rrs to pass QC in tbp-parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_tngs_data</strong></td>
<td>Boolean</td>
<td>Set to true to enable tNGS-specific parameters and runs in tbp-parser</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_additional_parameters</strong></td>
<td>String</td>
<td>Additional parameters for TBProfiler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_custom_db</strong></td>
<td>File</td>
<td>TBProfiler uses by default the TBDB database; if you have a custom database you wish to use, you must provide a custom database in this field and set tbprofiler_run_custom_db to true</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/tbprofiler:6.6.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_mapper</strong></td>
<td>String</td>
<td>The mapping tool used in TBProfiler to align the reads to the reference genome; see TBProfilers original documentation for available options.</td>
<td>bwa</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_af</strong></td>
<td>Float</td>
<td>The minimum allele frequency to call a variant</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_depth</strong></td>
<td>Int</td>
<td>The minimum depth for a variant to be called.</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_cdph_db</strong></td>
<td>Boolean</td>
<td>TBProfiler uses by default the TBDB database; set this value to "true" to use the WHO v2 database with customizations for CDPH</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_custom_db</strong></td>
<td>Boolean</td>
<td>TBProfiler uses by default the TBDB database; if you have a custom database you wish to use, you must set this value to true and provide a custom database in the tbprofiler_custom_db field</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_caller</strong></td>
<td>String</td>
<td>Select a different variant caller for TBProfiler to use by writing it in this block; see TBProfilers original documentation for available options.</td>
<td>GATK</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_calling_params</strong></td>
<td>String</td>
<td>Enter additional variant calling parameters in this free text input to customize how the variant caller works in TBProfiler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>theiaeuk</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>watronfire/vibecheck:2025.02.24</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_lineage_barcodes</strong></td>
<td>File</td>
<td>Feather formatted lineage barcodes to use instead of default O1 barcodes</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_skip_subsampling</strong></td>
<td>Boolean</td>
<td>When enabled, will not subsample reads prior to classification. Will increase computation time</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_subsampling_fraction</strong></td>
<td>Float</td>
<td>Fraction of reads to use in classification.</td>
<td>0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_database</strong></td>
<td>String</td>
<td>The specific database to use</td>
<td>virulence_ecoli</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/virulencefinder:2.0.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_coverage</strong></td>
<td>Float</td>
<td>The threshold for minimum coverage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_identity</strong></td>
<td>Float</td>
<td>The threshold for minimum blast identity</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>database</strong></td>
<td>String</td>
<td>User-specified database</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>database_path</strong></td>
<td>String</td>
<td>Path to user-specified database</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/plasmidfinder:2.1.6</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>method_path</strong></td>
<td>String</td>
<td>Path to files for a user-specified method to use (blast or kma)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Threshold for minimum coverage, default threshold from PlasmidFinder CLI tool is used (0.60)</td>
<td>0.6</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Threshold for mininum blast identity, default threshold from PlasmidFinder CLI tool is used (0.90). This default differs from the default of the PlasmidFinder webtool (0.95)</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>compliant</strong></td>
<td>Boolean</td>
<td>If true (default), forces Genbank/ENA/DDJB compliant headers in Prokka output files</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/prokka:1.14.5</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>prodigal_tf</strong></td>
<td>File</td>
<td><a href="https://github.com/tseemann/prokka#option---prodigaltf">https://github.com/tseemann/prokka#option---prodigaltf</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>prokka_arguments</strong></td>
<td>String</td>
<td>Any additional <a href="https://github.com/tseemann/prokka#command-line-options">https://github.com/tseemann/prokka#command-line-options</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>proteins</strong></td>
<td>Boolean</td>
<td>FASTA file of trusted proteins for Prokka to first use for annotations</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_length_unambiguous</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>meanbaseq_trim</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_Degenerate</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_N</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>percent_reference_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>vadr_num_alerts</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>min_contig_length</strong></td>
<td>Int</td>
<td>Minimum length of contig for QUAST</td>
<td>500</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>File with adapter sequences to be removed</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the Kraken2 task should be called; for non-TheiaCoV workflows, the <code>kraken_db</code> variable must be provided.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the MIDAS task should be called.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>extract_unclassified</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Additional arguments to use with fastp</td>
<td>--detect_adapter_for_pe -g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_complete_only</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_decontaminate_mem</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_is_accession</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_refseq</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task. Increase this when using large (&gt;30GB kraken2 databases such as the "k2_standard" database)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>The database used by the MIDAS task in .tar.gz format</td>
<td>gs://theiagen-public-files-rp/terra/theiaprok-files/midas/midas_db_v1.2.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>A file containing the phix used during Illumina sequencing; used in the BBDuk task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>The name of the tool to perform basic read processing; options: "trimmomatic" or "fastp"</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>The tool used for quality control (QC) of reads. Options are "fastq_scan" (default) and "fastqc"</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>This string is searched for in the kraken2 outputs to extract the read percentage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>taxon_id</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Additional arguments to pass to trimmomatic. "-phred33" specifies the Phred Q score encoding which is almost always phred33 with modern sequence data.</td>
<td>-phred33</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>acquired</strong></td>
<td>Boolean</td>
<td>Set to true to tell ResFinder to identify acquired resistance genes</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>call_pointfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable detection of point mutations.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/resfinder:4.1.11</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum coverage breadth of a gene for it to be identified</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum identity for ResFinder to identify a gene</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>abricate_db</strong></td>
<td>String</td>
<td>Database to use with the Abricate tool. Options: NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB</td>
<td>vfdb</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>amrfinder_use_gff</strong></td>
<td>Boolean</td>
<td>Whether or not to use the GFF and protein FASTA files for AMRFinderPlus</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>bakta_db</strong></td>
<td>String</td>
<td>Database selection for Bakta annotation. Options: light (smaller, faster), full (more comprehensive), or a Google Storage URI (gs://...) pointing to a custom Bakta database archive (.tar.gz). The selected database will be extracted before annotation.</td>
<td>full</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>call_abricate</strong></td>
<td>Boolean</td>
<td>Set to true to enable the Abricate task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>call_ani</strong></td>
<td>Boolean</td>
<td>Set to true to enable the ANI task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>call_arln_stats</strong></td>
<td>Boolean</td>
<td>Set to true to enable the return of ARLN required Auto PASS/FAIL outputs</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>call_gamma</strong></td>
<td>Boolean</td>
<td>Set to true in order to enable GAMMA task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>call_kmerfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the kmerfinder task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>call_plasmidfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the plasmidfinder task</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>call_resfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the ResFinder task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>city</strong></td>
<td>String</td>
<td>Will be used in the "city" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>collection_date</strong></td>
<td>String</td>
<td>Will be used in the "collection_date" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>county</strong></td>
<td>String</td>
<td>Will be used in the "county" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>expected_taxon</strong></td>
<td>String</td>
<td>If provided, this input will override the taxonomic assignment made by GAMBIT and launch the relevant taxon-specific submodules. It will also modify the organism flag used by AMRFinderPlus. Example format: "Salmonella enterica"</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>genome_annotation</strong></td>
<td>String</td>
<td>If set to "bakta", TheiaProk will use Bakta rather than Prokka to annotate the genome</td>
<td>prokka</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>max_genome_length</strong></td>
<td>Int</td>
<td>Maximum genome length able to pass read screening.</td>
<td>18040666</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>min_basepairs</strong></td>
<td>Int</td>
<td>Minimum number of base pairs able to pass read screening</td>
<td>2241820</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>min_coverage</strong></td>
<td>Int</td>
<td>Minimum genome coverage able to pass read screening</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>min_genome_length</strong></td>
<td>Int</td>
<td>Minimum genome length able to pass read screening.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>min_proportion</strong></td>
<td>Int</td>
<td>Minimum proportion of total reads in each read file to pass read screening</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>min_reads</strong></td>
<td>Int</td>
<td>Minimum number of reads to pass read screening</td>
<td>7472</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>mlst_run_secondary_scheme</strong></td>
<td>Boolean</td>
<td>If true, will run secondary scheme if primary scheme is of ecoli or abaumannii, these two have multiple schemes that are relevant.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>mlst_scheme_override</strong></td>
<td>Boolean</td>
<td>If true, will force E. coli scheme to be used when Gambit predicts Escherichia coli, otherwise will return scheme MLST predicts.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>originating_lab</strong></td>
<td>String</td>
<td>Will be used in the "originating_lab" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>perform_characterization</strong></td>
<td>Boolean</td>
<td>Set to "false" if you want to only generate an assembly and relevant QC metrics and skip all characterization tasks</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>qc_check_table</strong></td>
<td>File</td>
<td>TSV value with taxons for rows and QC values for columns; internal cells represent user-determined QC thresholds; if provided, turns on the QC Check task. See below for an example QC Check table.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>read1_lane2</strong></td>
<td>File</td>
<td>If provided, the Concatenate_Illumina_Lanes subworkflow will concatenate all files from the same lane before doing any subsequent analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>read1_lane3</strong></td>
<td>File</td>
<td>If provided, the Concatenate_Illumina_Lanes subworkflow will concatenate all files from the same lane before doing any subsequent analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>read1_lane4</strong></td>
<td>File</td>
<td>If provided, the Concatenate_Illumina_Lanes subworkflow will concatenate all files from the same lane before doing any subsequent analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>read2_lane2</strong></td>
<td>File</td>
<td>If provided, the Concatenate_Illumina_Lanes subworkflow will concatenate all files from the same lane before doing any subsequent analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>read2_lane3</strong></td>
<td>File</td>
<td>If provided, the Concatenate_Illumina_Lanes subworkflow will concatenate all files from the same lane before doing any subsequent analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>read2_lane4</strong></td>
<td>File</td>
<td>If provided, the Concatenate_Illumina_Lanes subworkflow will concatenate all files from the same lane before doing any subsequent analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>run_id</strong></td>
<td>String</td>
<td>Will be used in the "run_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>seq_method</strong></td>
<td>String</td>
<td>The sequencing methodology used to generate the input read data; for TheiaProk workflows, this input will be used in the "seq_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td>ILLUMINA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>skip_screen</strong></td>
<td>Boolean</td>
<td>Set to True to skip the read screening prior to analysis</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>taxon_tables</strong></td>
<td>File</td>
<td>File indicating data table names to copy samples of a particular taxon to</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>terra_project</strong></td>
<td>String</td>
<td>The name of the Terra Project where you want the taxon tables written to</td>
<td>NA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>terra_workspace</strong></td>
<td>String</td>
<td>The name of the Terra Workspace where you want the taxon tables written to</td>
<td>NA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>trim_min_length</strong></td>
<td>Int</td>
<td>Specifies minimum length of each read after trimming to be kept</td>
<td>75</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>Specifies the minimum average quality of bases in a sliding window to be kept</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>Specifies window size for trimming (the number of bases to average the quality across)</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>zip</strong></td>
<td>String</td>
<td>Will be used in the "zip" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mlst:2.23.0-2024-12-31</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum % breadth of coverage to report an MLST allele</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum % identity to known MLST gene to report an MLST allele</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>minscore</strong></td>
<td>Float</td>
<td>Minimum score <a href="https://github.com/tseemann/mlst#scoring-system">https://github.com/tseemann/mlst#scoring-system</a> to assign an MLST profile</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>nopath</strong></td>
<td>Boolean</td>
<td>true = use mlst --nopath. If set to false, filename paths are not stripped from FILE column in output TSV</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>scheme</strong></td>
<td>String</td>
<td>Dont autodetect the MLST scheme; force this scheme on all inputs (see <a href="https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab">https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab</a> for accepted strings)</td>
<td>None</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>Illumina forward read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA %coverage for the Abricate task</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA %identity for the Abricate task</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>detailed_drug_class</strong></td>
<td>Boolean</td>
<td>If set to true, amrfinderplus_amr_classes and amrfinderplus_amr_subclasses outputs will be created</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-amrfinderplus:4.0.23-2025-07-16.1</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>hide_point_mutations</strong></td>
<td>Boolean</td>
<td>If set to true, point mutations are not reported</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum proportion of reference gene covered for a BLAST-based hit (Methods BLAST or PARTIAL)." Attribute should be a float ranging from 0-1, such as 0.6 (equal to 60% coverage)</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum identity for a blast-based hit hit (Methods BLAST or PARTIAL). -1 means use a curated threshold if it exists and 0.9 otherwise. Setting this value to something other than -1 will override any curated similarity cutoffs. Attribute should be a float ranging from 0-1, such as 0.95 (equal to 95% identity)</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>separate_betalactam_genes</strong></td>
<td>Boolean</td>
<td>Report beta-Lactam AMR genes separated out by all beta-lactam and the respective beta-lactam subclasses</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>ani_threshold</strong></td>
<td>Float</td>
<td>ANI value threshold must be surpassed in order to output the ani_top_species_match. If a genome does not surpass this threshold (and the percent_bases_aligned_threshold) then the ani_top_species_match output String will show a warning instead of a genus &amp; species.</td>
<td>80.0</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mummer:4.0.0-rgdv2</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>mash_filter</strong></td>
<td>Float</td>
<td>Mash distance threshold over which ANI is not calculated</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>percent_bases_aligned_threshold</strong></td>
<td>Float</td>
<td>Threshold regarding the proportion of bases aligned between the query genome and reference genome. If a genome does not surpass this threshold (and the ani_threshold) then the ani_top_species_match output String will show a warning instead of a genus &amp; species.</td>
<td>70.0</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>ref_genome</strong></td>
<td>File</td>
<td>If not set, uses all 43 genomes in RGDv2</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/arln_stats:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>read2_clean</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>read2_raw</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>bakta_opts</strong></td>
<td>String</td>
<td>Parameters to pass to bakta from <a href="https://github.com/oschwengers/bakta#usage">https://github.com/oschwengers/bakta#usage</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>compliant</strong></td>
<td>Boolean</td>
<td>If true, forces Genbank/ENA/DDJB compliant headers in Bakta output files</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>200</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/bakta:1.10.3</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>24</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>prodigal_tf</strong></td>
<td>File</td>
<td>Prodigal training file to use for CDS prediction by bakta</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>proteins</strong></td>
<td>File</td>
<td>Fasta file of trusted protein sequences for CDS annotation</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ezlabgva/busco:v5.7.1_cv1</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>eukaryote</strong></td>
<td>Boolean</td>
<td>Assesses eukaryotic organisms, rather than prokaryotic organisms</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>cg_pipe_opts</strong></td>
<td>String</td>
<td>Options to pass to CG-Pipeline for clean read assessment</td>
<td>--fast</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/lyveset:1.1.4f</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>cg_pipe_opts</strong></td>
<td>String</td>
<td>Options to pass to CG-Pipeline for raw read assessment</td>
<td>--fast</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/lyveset:1.1.4f</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>concatenate_illumina_lanes</td>
<td><strong>read2_lane1</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>concatenate_illumina_lanes</td>
<td><strong>read2_lane2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>concatenate_illumina_lanes</td>
<td><strong>read2_lane3</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>concatenate_illumina_lanes</td>
<td><strong>read2_lane4</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>assembler</strong></td>
<td>String</td>
<td>Assembler to use (spades, skesa, megahit)</td>
<td>skesa</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>assembler_options</strong></td>
<td>String</td>
<td>Assembler-specific options that you might choose for the selected assembler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>6</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>call_pilon</strong></td>
<td>Boolean</td>
<td>Whether to run Pilon polishing after assembly</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/shovilter:0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_min_coverage</strong></td>
<td>Float</td>
<td>Minimum coverage threshold for contig filtering</td>
<td>2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_min_length</strong></td>
<td>Int</td>
<td>Minimum length threshold for contig filtering</td>
<td>200</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_skip_coverage_filter</strong></td>
<td>Boolean</td>
<td>Skip filtering contigs based on coverage</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_skip_homopolymer_filter</strong></td>
<td>Boolean</td>
<td>Skip filtering contigs containing homopolymers</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_skip_length_filter</strong></td>
<td>Boolean</td>
<td>Skip filtering contigs based on length</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>kmers</strong></td>
<td>String</td>
<td>K-mer sizes for assembly (comma-separated)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/megahit:1.2.9</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>min_contig_length</strong></td>
<td>Int</td>
<td>Minimum contig length to retain in final assembly</td>
<td>200</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/pilon:1.24--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_fix</strong></td>
<td>String</td>
<td>Potential issues with assembly to try and automatically fix (snps, indels, gaps, local, all, bases, none)</td>
<td>bases</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_min_base_quality</strong></td>
<td>Int</td>
<td>Minimum base quality to keep</td>
<td>3</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_min_depth</strong></td>
<td>Float</td>
<td>Minimum coverage threshold for variant calling: when set to a value 1, it requires that absolute depth of coverage; when set to a fraction &lt;1, it requires coverage at least that fraction of the mean coverage for the region</td>
<td>0.25</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_min_mapping_quality</strong></td>
<td>Int</td>
<td>Minimum mapping quality for a read to count in pileups</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>run_filter_contigs</strong></td>
<td>Boolean</td>
<td>Whether to run contig filtering step</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/skesa:2.4.0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/spades:4.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_type</strong></td>
<td>String</td>
<td>SPAdes assembly mode (isolate, meta, rna, etc.), more can be found <a href="https://ablab.github.io/spades/running.html">here</a></td>
<td>isolate</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>theiaviral_panel</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>gambit_db_genomes</strong></td>
<td>File</td>
<td>Database of metadata for assembled query genomes; requires complementary signatures file. If not provided, uses default database "/gambit-db"</td>
<td>gs://gambit-databases-rp/2.1.0/gambit-metadata-2.1.0-20250808.gdb</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>gambit_db_signatures</strong></td>
<td>File</td>
<td>Signatures file; requires complementary genomes file. If not specified, the file from the docker container will be used.</td>
<td>gs://gambit-databases-rp/2.1.0/gambit-signatures-2.1.0-20250808.gs</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gamma:2.2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>extended_output</strong></td>
<td>Boolean</td>
<td>GAMMA will provide the extended output when set to true</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>gamma_db</strong></td>
<td>File</td>
<td>Multifasta sequence database GAMMA is to use for gene matching</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/gamma/default_ResFinderDB_Combined_05-06-20.fsa</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>min_length_percent_gammas</strong></td>
<td>Int</td>
<td>Threshold of match length percent used by GAMMA-s</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>min_percent_identity</strong></td>
<td>Int</td>
<td>Threshold of identity for gene matches in percent</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>output_fasta</strong></td>
<td>Boolean</td>
<td>Allows GAMMA to output gene matches in FASTA format</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>output_gff</strong></td>
<td>Boolean</td>
<td>Allows GAMMA to output gene matches in GFF format</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>run_gammas</strong></td>
<td>Boolean</td>
<td>Turns on GAMMA-S which runs GAMMA on nucleotide sequences rather than translating input sequences and matches</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/kmerfinder:3.0.2--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>kmerfinder_args</strong></td>
<td>String</td>
<td>Kmerfinder additional arguments</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>kmerfinder_db</strong></td>
<td>File</td>
<td>Bacterial database for KmerFinder</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kmerfinder/kmerfinder_bacteria_20230911.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity; set to 95 because there is a strict threshold of 95% identity for typing purposes</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_agr_typing_only</strong></td>
<td>Boolean</td>
<td>Set to true to skip agr operon extraction and frameshift detection</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/agrvate:1.0.2--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/amrsearch:0.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>assembly_only</strong></td>
<td>Boolean</td>
<td>Set to true if only analyzing input assembly</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_poppunk</strong></td>
<td>Boolean</td>
<td>If true, runs PopPUNK for GPSC cluster designation for S. pneumoniae</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_shigeifinder_reads_input</strong></td>
<td>Boolean</td>
<td>If set to "true", the ShigEiFinder task will run again but using read files as input instead of the assembly file. Input is shown but not used for TheiaProk_FASTA.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_stxtyper</strong></td>
<td>Boolean</td>
<td>If set to "true", the StxTyper task will run on all samples regardless of the gambit_predicted_taxon output. Useful if you suspect a non-E.coli or non-Shigella sample contains stx genes.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_tbp_parser</strong></td>
<td>Boolean</td>
<td>If set to "true", activates the tbp_parser module and results in more outputs, including&nbsp;tbp_parser_looker_report_csv, tbp_parser_laboratorian_report_csv,  tbp_parser_lims_report_csv, tbp_parser_coverage_report, and tbp_parser_genome_percent_coverage</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cauris_cladetyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_kmer_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_max_distance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6</strong></td>
<td>File</td>
<td>The reference assembly for clade 6</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade6_GCA_032714025.1_ASM3271402v1_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6_annotated</strong></td>
<td>File</td>
<td>The path to the annotated reference for clade 6</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>clockwork_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/cdcgov/varpipe_wgs_with_refs:2bc7234074bd53d9e92a1048b0485763cd9bbf6f4d12d5a1cc82bfec8ca7d75e</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/ectyper:1.0.0--pyhdfd78af_1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minumum percent coverage required for an H antigen allele match</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_identity</strong></td>
<td>Int</td>
<td>Percent identity required for an H antigen allele match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minumum percent coverage required for an O antigen allele match</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_identity</strong></td>
<td>Int</td>
<td>Percent identity required for an O antigen allele match</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_print_alleles</strong></td>
<td>Boolean</td>
<td>Set to true to print the allele sequences as the final column</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_verify</strong></td>
<td>Boolean</td>
<td>Set to true to enable E. coli species verification</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_align_diff</strong></td>
<td>Int</td>
<td>Threshold for difference between alignment length and subject length in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_cluster_distance</strong></td>
<td>Int</td>
<td>Distance between cluster of matches to consider as different clusters.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_culling_limit</strong></td>
<td>Int</td>
<td>Total hits to return in a position.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/emmtyper:0.2.0--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_gap</strong></td>
<td>Int</td>
<td>Threshold gap to allow in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_max_size</strong></td>
<td>Int</td>
<td>Maximum size of PCR product.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_good</strong></td>
<td>Int</td>
<td>Minimum size where there must be 2 matches for each mismatch.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimal percent identity of sequence.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_perfect</strong></td>
<td>Int</td>
<td>Minimum size of perfect match at 3' primer end.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_mismatch</strong></td>
<td>Int</td>
<td>Threshold for number of mismatch to allow in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_wf</strong></td>
<td>String</td>
<td>Choose workflow [blast</td>
<td>pcr]</td>
<td></td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtypingtool_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/emmtypingtool:0.0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>genotyphi_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.11.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_broken_gene_length</strong></td>
<td>Int</td>
<td>Minimum length to consider a broken gene</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/hicap:1.0.3--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_broken_gene_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percentage identity to consider a broken gene</td>
<td>0.8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum percentage coverage to consider a single gene complete</td>
<td>0.8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percentage identity to consider a single gene complete</td>
<td>0.7</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kaptive:2.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_low_gene_percent_identity</strong></td>
<td>Float</td>
<td>Percent identity threshold for what counts as a low identity match in the gene BLAST search</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum required percent coverage for the gene BLAST search via tBLASTn</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum required percent identity for the gene BLAST search via tBLASTn</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_start_end_margin</strong></td>
<td>Int</td>
<td>Determines flexibility in identifying the start and end of a locus - if this value is 10, a locus match that is missing the first 8 base pairs will still count as capturing the start of the locus</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kleborate:2.2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_kaptive_confidence</strong></td>
<td>String</td>
<td>{None,Low,Good,High,Very_high,Perfect} Minimum Kaptive confidence to call K/O loci - confidence levels below this will be reported as unknown</td>
<td>Good</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum alignment percent coverage for main results</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum alignment percent identity for main results</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum alignment percent coverage for spurious results</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_identity</strong></td>
<td>Float</td>
<td>Minimum alignment percent identity for spurious results</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_kaptive</strong></td>
<td>Boolean</td>
<td>Equivalent to --kaptive_k --kaptive_</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_resistance</strong></td>
<td>Boolean</td>
<td>Set to true to turn on resistance genes screening (default: no resistance gene screening)</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>legsta_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/lissero:0.4.9--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum percent coverage of the gene to accept a match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percent identity to accept a match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>meningotype_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/meningotype:0.8.5--pyhdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ngmaster_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ngmaster:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ont_data</strong></td>
<td>Boolean</td>
<td>Set to true if your data is ONT FASTQ files</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pasty:1.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum coverage of a O-antigen to be considered for serogrouping by pasty</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum percent identity for a blast hit to be considered for serogrouping</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pbptyper:1.0.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum percent coverage to count a hit</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum percent identity to count a hit</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/poppunk:2.4.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_npy</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.dists.npy</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.dists.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_external_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_external_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_npz</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_fit.npz</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_fit.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_graph_gt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_graph.gt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_h5</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.h5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_qcreport_txt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_qcreport.txt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_npy</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.dists.npy</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.dists.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_graph_gt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6refs_graph.gt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_h5</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.h5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_unword_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_unword_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>run_amr_search</strong></td>
<td>Boolean</td>
<td>If set to true AMR_Search workflow will be run if species is part of supported taxon, see AMR_Search docs.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seqsero2s_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/seqsero2s:1.1.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seroba_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/seroba:1.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>serotypefinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/serotypefinder:2.0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigatyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/shigatyper:2.0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigeifinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/shigeifinder:1.3.5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/sistr_cmd:1.1.1--pyh864c0ab_2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_use_full_cgmlst_db</strong></td>
<td>Boolean</td>
<td>Set to true to use the full set of cgMLST alleles which can include highly similar alleles. By default the smaller "centroid" alleles or representative alleles are used for each marker</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_base_quality</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_gene_query_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_map_qual</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_maxsoft</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_frac</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_quality</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_query_gene</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_afumigatus</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/aspergillus/Aspergillus_fumigatus_GCF_000002655.1_ASM265v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_cryptoneo</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/cryptococcus/Cryptococcus_neoformans_GCF_000091045.1_ASM9104v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_variants_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_mykrobe_opts</strong></td>
<td>String</td>
<td>Additional options for mykrobe in sonneityping</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_do_enrich</strong></td>
<td>Boolean</td>
<td>Set to true to enable PCR product enrichment</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/spatyper:0.3.3--pyhdfd78af_3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/srst2:0.2.0-vcholerae</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_gene_max_mismatch</strong></td>
<td>Int</td>
<td>Maximum number of mismatches for SRST2 to call a gene as present</td>
<td>2000</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_max_divergence</strong></td>
<td>Int</td>
<td>Maximum divergence, in percentage, for SRST2 to call a gene as present</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_depth</strong></td>
<td>Int</td>
<td>Minimum depth of coverage for SRST2 to call a gene as present</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_edge_depth</strong></td>
<td>Int</td>
<td>Minimum edge depth for SRST2 to call a gene as present</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum breadth of coverage for SRST2 to call a gene as present</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>staphopia_sccmec_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/staphopia-sccmec:1.0.0--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/stxtyper:1.0.42</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_enable_debug</strong></td>
<td>Boolean</td>
<td>When enabled, additional messages are printed and files in $TMPDIR are not removed after running</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_add_cs_lims</strong></td>
<td>Boolean</td>
<td>Set to true add cycloserine results to the LIMS report</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_config</strong></td>
<td>File</td>
<td>The configuration file to use, in YAML format (overrides all other arguments except input_json and input_bam)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_coverage_regions_bed</strong></td>
<td>File</td>
<td>A bed file that lists the regions to be considered for QC</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_debug</strong></td>
<td>Boolean</td>
<td>Activate the debug mode on tbp_parser; increases logging outputs</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/tbp-parser:2.6.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_etha237_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in ethA at protein position 237 to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_expert_rule_regions_bed</strong></td>
<td>File</td>
<td>A file that contains the regions where R mutations and expert rules are applied</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_depth</strong></td>
<td>Int</td>
<td>Minimum depth for a variant to pass QC in tbp_parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_frequency</strong></td>
<td>Float</td>
<td>The minimum frequency for a mutation to pass QC</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_percent_coverage</strong></td>
<td>Float</td>
<td>The minimum coverage for a region to pass QC in tbp_parser</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_read_support</strong></td>
<td>Int</td>
<td>The minimum read support for a mutation to pass QC</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_operator</strong></td>
<td>String</td>
<td>Fills the "operator" field in the tbp_parser output files</td>
<td>Operator not provided</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_output_seq_method_type</strong></td>
<td>String</td>
<td>Fills out the "seq_method" field in the tbp_parser output files</td>
<td>WGS</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rpob449_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation at protein position 449 to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in rrl to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_read_support</strong></td>
<td>Int</td>
<td>Minimum read support for a mutation in rrl to pass QC in tbp-parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in rrs to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_read_support</strong></td>
<td>Int</td>
<td>Minimum read support for a mutation in rrs to pass QC in tbp-parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_tngs_data</strong></td>
<td>Boolean</td>
<td>Set to true to enable tNGS-specific parameters and runs in tbp-parser</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_additional_parameters</strong></td>
<td>String</td>
<td>Additional parameters for TBProfiler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_custom_db</strong></td>
<td>File</td>
<td>TBProfiler uses by default the TBDB database; if you have a custom database you wish to use, you must provide a custom database in this field and set tbprofiler_run_custom_db to true</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/tbprofiler:6.6.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_mapper</strong></td>
<td>String</td>
<td>The mapping tool used in TBProfiler to align the reads to the reference genome; see TBProfilers original documentation for available options.</td>
<td>bwa</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_af</strong></td>
<td>Float</td>
<td>The minimum allele frequency to call a variant</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_depth</strong></td>
<td>Int</td>
<td>The minimum depth for a variant to be called.</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_cdph_db</strong></td>
<td>Boolean</td>
<td>TBProfiler uses by default the TBDB database; set this value to "true" to use the WHO v2 database with customizations for CDPH</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_custom_db</strong></td>
<td>Boolean</td>
<td>TBProfiler uses by default the TBDB database; if you have a custom database you wish to use, you must set this value to true and provide a custom database in the tbprofiler_custom_db field</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_caller</strong></td>
<td>String</td>
<td>Select a different variant caller for TBProfiler to use by writing it in this block; see TBProfilers original documentation for available options.</td>
<td>GATK</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_calling_params</strong></td>
<td>String</td>
<td>Enter additional variant calling parameters in this free text input to customize how the variant caller works in TBProfiler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>theiaeuk</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_lineage_barcodes</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_skip_subsampling</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_subsampling_fraction</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_database</strong></td>
<td>String</td>
<td>The specific database to use</td>
<td>virulence_ecoli</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/virulencefinder:2.0.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_coverage</strong></td>
<td>Float</td>
<td>The threshold for minimum coverage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_identity</strong></td>
<td>Float</td>
<td>The threshold for minimum blast identity</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>database</strong></td>
<td>String</td>
<td>User-specified database</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>database_path</strong></td>
<td>String</td>
<td>Path to user-specified database</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/plasmidfinder:2.1.6</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>method_path</strong></td>
<td>String</td>
<td>Path to files for a user-specified method to use (blast or kma)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Threshold for minimum coverage, default threshold from PlasmidFinder CLI tool is used (0.60)</td>
<td>0.6</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Threshold for mininum blast identity, default threshold from PlasmidFinder CLI tool is used (0.90). This default differs from the default of the PlasmidFinder webtool (0.95)</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>compliant</strong></td>
<td>Boolean</td>
<td>If true (default), forces Genbank/ENA/DDJB compliant headers in Prokka output files</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/prokka:1.14.5</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>prodigal_tf</strong></td>
<td>File</td>
<td><a href="https://github.com/tseemann/prokka#option---prodigaltf">https://github.com/tseemann/prokka#option---prodigaltf</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>prokka_arguments</strong></td>
<td>String</td>
<td>Any additional <a href="https://github.com/tseemann/prokka#command-line-options">https://github.com/tseemann/prokka#command-line-options</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>proteins</strong></td>
<td>Boolean</td>
<td>FASTA file of trusted proteins for Prokka to first use for annotations</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_length_unambiguous</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>meanbaseq_trim</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_clean2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_raw2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_Degenerate</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_N</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>percent_reference_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>vadr_num_alerts</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>min_contig_length</strong></td>
<td>Int</td>
<td>Minimum length of contig for QUAST</td>
<td>500</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>File with adapter sequences to be removed</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the Kraken2 task should be called; for non-TheiaCoV workflows, the <code>kraken_db</code> variable must be provided.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the MIDAS task should be called.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Additional arguments to use with fastp</td>
<td>-g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task. Increase this when using large (&gt;30GB kraken2 databases such as the "k2_standard" database)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>The database used by the MIDAS task in .tar.gz format</td>
<td>gs://theiagen-public-files-rp/terra/theiaprok-files/midas/midas_db_v1.2.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>A file containing the phix used during Illumina sequencing; used in the BBDuk task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>The name of the tool to perform basic read processing; options: "trimmomatic" or "fastp"</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>The tool used for quality control (QC) of reads. Options are "fastq_scan" (default) and "fastqc"</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>This string is searched for in the kraken2 outputs to extract the read percentage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Additional arguments to pass to trimmomatic. "-phred33" specifies the Phred Q score encoding which is almost always phred33 with modern sequence data.</td>
<td>-phred33</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>acquired</strong></td>
<td>Boolean</td>
<td>Set to true to tell ResFinder to identify acquired resistance genes</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>call_pointfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable detection of point mutations.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/resfinder:4.1.11</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum coverage breadth of a gene for it to be identified</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum identity for ResFinder to identify a gene</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>abricate_db</strong></td>
<td>String</td>
<td>Database to use with the Abricate tool. Options: NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB</td>
<td>vfdb</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>amrfinder_use_gff</strong></td>
<td>Boolean</td>
<td>Whether or not to use the GFF and protein FASTA files for AMRFinderPlus</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>bakta_db</strong></td>
<td>String</td>
<td>Database selection for Bakta annotation. Options: light (smaller, faster), full (more comprehensive), or a Google Storage URI (gs://...) pointing to a custom Bakta database archive (.tar.gz). The selected database will be extracted before annotation.</td>
<td>full</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>call_abricate</strong></td>
<td>Boolean</td>
<td>Set to true to enable the Abricate task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>call_ani</strong></td>
<td>Boolean</td>
<td>Set to true to enable the ANI task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>call_arln_stats</strong></td>
<td>Boolean</td>
<td>Set to true to enable the return of ARLN required Auto PASS/FAIL outputs</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>call_gamma</strong></td>
<td>Boolean</td>
<td>Set to true in order to enable GAMMA task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>call_kmerfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the kmerfinder task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>call_plasmidfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the plasmidfinder task</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>call_resfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the ResFinder task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>city</strong></td>
<td>String</td>
<td>Will be used in the "city" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>collection_date</strong></td>
<td>String</td>
<td>Will be used in the "collection_date" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>county</strong></td>
<td>String</td>
<td>Will be used in the "county" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>expected_taxon</strong></td>
<td>String</td>
<td>If provided, this input will override the taxonomic assignment made by GAMBIT and launch the relevant taxon-specific submodules. It will also modify the organism flag used by AMRFinderPlus. Example format: "Salmonella enterica"</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>genome_annotation</strong></td>
<td>String</td>
<td>If set to "bakta", TheiaProk will use Bakta rather than Prokka to annotate the genome</td>
<td>prokka</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>max_genome_length</strong></td>
<td>Int</td>
<td>Maximum genome length able to pass read screening.</td>
<td>18040666</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>min_basepairs</strong></td>
<td>Int</td>
<td>Minimum number of base pairs able to pass read screening</td>
<td>2241820</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>min_coverage</strong></td>
<td>Int</td>
<td>Minimum genome coverage able to pass read screening</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>min_genome_length</strong></td>
<td>Int</td>
<td>Minimum genome length able to pass read screening.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>min_reads</strong></td>
<td>Int</td>
<td>Minimum number of reads to pass read screening</td>
<td>7472</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>mlst_run_secondary_scheme</strong></td>
<td>Boolean</td>
<td>If true, will run secondary scheme if primary scheme is of ecoli or abaumannii, these two have multiple schemes that are relevant.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>mlst_scheme_override</strong></td>
<td>Boolean</td>
<td>If true, will force E. coli scheme to be used when Gambit predicts Escherichia coli, otherwise will return scheme MLST predicts.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>originating_lab</strong></td>
<td>String</td>
<td>Will be used in the "originating_lab" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>perform_characterization</strong></td>
<td>Boolean</td>
<td>Set to "false" if you want to only generate an assembly and relevant QC metrics and skip all characterization tasks</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>qc_check_table</strong></td>
<td>File</td>
<td>TSV value with taxons for rows and QC values for columns; internal cells represent user-determined QC thresholds; if provided, turns on the QC Check task. See below for an example QC Check table.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>read1_lane2</strong></td>
<td>File</td>
<td>If provided, the Concatenate_Illumina_Lanes subworkflow will concatenate all files from the same lane before doing any subsequent analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>read1_lane3</strong></td>
<td>File</td>
<td>If provided, the Concatenate_Illumina_Lanes subworkflow will concatenate all files from the same lane before doing any subsequent analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>read1_lane4</strong></td>
<td>File</td>
<td>If provided, the Concatenate_Illumina_Lanes subworkflow will concatenate all files from the same lane before doing any subsequent analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>run_id</strong></td>
<td>String</td>
<td>Will be used in the "run_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>seq_method</strong></td>
<td>String</td>
<td>The sequencing methodology used to generate the input read data; for TheiaProk workflows, this input will be used in the "seq_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td>ILLUMINA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>skip_mash</strong></td>
<td>Boolean</td>
<td>If true, skips estimation of genome size and coverage using mash in read screening steps. As a result, providing true also prevents screening using these parameters.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>skip_screen</strong></td>
<td>Boolean</td>
<td>Set to True to skip the read screening prior to analysis</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>taxon_tables</strong></td>
<td>File</td>
<td>File indicating data table names to copy samples of a particular taxon to</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>terra_project</strong></td>
<td>String</td>
<td>The name of the Terra Project where you want the taxon tables written to</td>
<td>NA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>terra_workspace</strong></td>
<td>String</td>
<td>The name of the Terra Workspace where you want the taxon tables written to</td>
<td>NA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>trim_min_length</strong></td>
<td>Int</td>
<td>Specifies minimum length of each read after trimming to be kept</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>Specifies the average quality of bases in a sliding window to be kept</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>Specifies window size for trimming (the number of bases to average the quality across)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>zip</strong></td>
<td>String</td>
<td>Will be used in the "zip" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mlst:2.23.0-2024-12-31</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum % breadth of coverage to report an MLST allele</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum % identity to known MLST gene to report an MLST allele</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>minscore</strong></td>
<td>Float</td>
<td>Minimum score <a href="https://github.com/tseemann/mlst#scoring-system">https://github.com/tseemann/mlst#scoring-system</a> to assign an MLST profile</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>nopath</strong></td>
<td>Boolean</td>
<td>true = use mlst --nopath. If set to false, filename paths are not stripped from FILE column in output TSV</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>scheme</strong></td>
<td>String</td>
<td>Dont autodetect the MLST scheme; force this scheme on all inputs (see <a href="https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab">https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab</a> for accepted strings)</td>
<td>None</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiaprok_ont</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>ONT read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA %coverage for the Abricate task</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA %identity for the Abricate task</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>detailed_drug_class</strong></td>
<td>Boolean</td>
<td>If set to true, amrfinderplus_amr_classes and amrfinderplus_amr_subclasses outputs will be created</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-amrfinderplus:4.0.23-2025-07-16.1</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>hide_point_mutations</strong></td>
<td>Boolean</td>
<td>If set to true, point mutations are not reported</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum proportion of reference gene covered for a BLAST-based hit (Methods BLAST or PARTIAL)." Attribute should be a float ranging from 0-1, such as 0.6 (equal to 60% coverage)</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum identity for a blast-based hit hit (Methods BLAST or PARTIAL). -1 means use a curated threshold if it exists and 0.9 otherwise. Setting this value to something other than -1 will override any curated similarity cutoffs. Attribute should be a float ranging from 0-1, such as 0.95 (equal to 95% identity)</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>separate_betalactam_genes</strong></td>
<td>Boolean</td>
<td>Report beta-Lactam AMR genes separated out by all beta-lactam and the respective beta-lactam subclasses</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>ani_threshold</strong></td>
<td>Float</td>
<td>ANI value threshold must be surpassed in order to output the ani_top_species_match. If a genome does not surpass this threshold (and the percent_bases_aligned_threshold) then the ani_top_species_match output String will show a warning instead of a genus &amp; species.</td>
<td>80.0</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mummer:4.0.0-rgdv2</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>mash_filter</strong></td>
<td>Float</td>
<td>Mash distance threshold over which ANI is not calculated</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>percent_bases_aligned_threshold</strong></td>
<td>Float</td>
<td>Threshold regarding the proportion of bases aligned between the query genome and reference genome. If a genome does not surpass this threshold (and the ani_threshold) then the ani_top_species_match output String will show a warning instead of a genus &amp; species.</td>
<td>70.0</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>ref_genome</strong></td>
<td>File</td>
<td>If not set, uses all 43 genomes in RGDv2</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/arln_stats:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>read2_clean</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>read2_raw</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>bakta_opts</strong></td>
<td>String</td>
<td>Parameters to pass to bakta from <a href="https://github.com/oschwengers/bakta#usage">https://github.com/oschwengers/bakta#usage</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>compliant</strong></td>
<td>Boolean</td>
<td>If true, forces Genbank/ENA/DDJB compliant headers in Bakta output files</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>200</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/bakta:1.10.3</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>24</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>prodigal_tf</strong></td>
<td>File</td>
<td>Prodigal training file to use for CDS prediction by bakta</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>proteins</strong></td>
<td>File</td>
<td>Fasta file of trusted protein sequences for CDS annotation</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ezlabgva/busco:v5.7.1_cv1</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>eukaryote</strong></td>
<td>Boolean</td>
<td>Assesses eukaryotic organisms, rather than prokaryotic organisms</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>theiaviral_panel</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>auto_medaka_model</strong></td>
<td>Boolean</td>
<td>If true, medaka will automatically select the best Medaka model for assembly</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>bandage_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>bandage_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>bandage_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>dnaapler_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>dnaapler_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>dnaapler_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>dnaapler_mode</strong></td>
<td>String</td>
<td>Dnaapler-specific inputs</td>
<td>all</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>filter_contigs_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>filter_contigs_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>filter_contigs_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>filter_contigs_min_length</strong></td>
<td>Int</td>
<td>Minimum contig length to keep</td>
<td>1000</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_additional_parameters</strong></td>
<td>String</td>
<td>Any extra Flye-specific parameters</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_asm_coverage</strong></td>
<td>Int</td>
<td>Reduced coverage for initial disjointig assembly</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_keep_haplotypes</strong></td>
<td>Boolean</td>
<td>If true keep haplotypes</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_minimum_overlap</strong></td>
<td>Int</td>
<td>Minimum overlap between reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_no_alt_contigs</strong></td>
<td>Boolean</td>
<td>If true, do not generate alternative contigs</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_polishing_iterations</strong></td>
<td>Int</td>
<td>Default polishing iterations</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_read_error_rate</strong></td>
<td>Float</td>
<td>Maximum expected read error rate</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_read_type</strong></td>
<td>String</td>
<td>Specifies the type of sequencing reads. Options: --nano-hq (default), --nano-corr, --nano-raw, --pacbio-raw, --pacbio-corr, --pacbio-hifi. Refer to Flye documentation for details on each type.</td>
<td>--nano-hq</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_scaffold</strong></td>
<td>Boolean</td>
<td>If true, scaffolding is enabled using graph</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>flye_uneven_coverage_mode</strong></td>
<td>Boolean</td>
<td>sets the --meta option in the case of uneven coverage (or metagenomics)</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>illumina_read1</strong></td>
<td>File</td>
<td>If Illumina reads are provided, flye_denovo subworkflow will perform Illumina polishing</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>illumina_read2</strong></td>
<td>File</td>
<td>If Illumina reads are provided, flye_denovo subworkflow will perform Illumina polishing</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>medaka_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>medaka_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>medaka_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>medaka_model</strong></td>
<td>String</td>
<td>In order to obtain the best results, the appropriate model must be set to match the sequencer's basecaller model; this string takes the format of {pore}{device}{caller variant}_{caller_version}. See also <a href="https://github.com/nanoporetech/medaka?tab=readme-ov-file#models">https://github.com/nanoporetech/medaka?tab=readme-ov-file#models</a>. If this is being run on legacy data it is likely to be r941_min_hac_g507.</td>
<td>r1041_e82_400bps_sup_v5.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polish_rounds</strong></td>
<td>Int</td>
<td>The number of polishing rounds to conduct for medaka or racon (without Illumina)</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polisher</strong></td>
<td>String</td>
<td>The polishing tool to use for assembly</td>
<td>medaka</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_careful</strong></td>
<td>Boolean</td>
<td>Polypolish-specific inputs</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_fraction_invalid</strong></td>
<td>Float</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_fraction_valid</strong></td>
<td>Float</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_high_percentile_threshold</strong></td>
<td>Float</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_low_percentile_threshold</strong></td>
<td>Float</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_maximum_errors</strong></td>
<td>Int</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_minimum_depth</strong></td>
<td>Int</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>polypolish_pair_orientation</strong></td>
<td>String</td>
<td>Polypolish-specific inputs</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>porechop_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>porechop_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>porechop_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>porechop_trimopts</strong></td>
<td>String</td>
<td>Options to pass to Porechop for trimming</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>racon_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>racon_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>racon_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>run_porechop</strong></td>
<td>Boolean</td>
<td>If true, trims reads before assembly using Porechop</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>flye_denovo</td>
<td><strong>skip_polishing</strong></td>
<td>Boolean</td>
<td>If true, skips polishing</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>gambit_db_genomes</strong></td>
<td>File</td>
<td>Database of metadata for assembled query genomes; requires complementary signatures file. If not provided, uses default database "/gambit-db"</td>
<td>gs://gambit-databases-rp/2.1.0/gambit-metadata-2.1.0-20250808.gdb</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>gambit_db_signatures</strong></td>
<td>File</td>
<td>Signatures file; requires complementary genomes file. If not specified, the file from the docker container will be used.</td>
<td>gs://gambit-databases-rp/2.1.0/gambit-signatures-2.1.0-20250808.gs</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gamma:2.2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>extended_output</strong></td>
<td>Boolean</td>
<td>GAMMA will provide the extended output when set to true</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>gamma_db</strong></td>
<td>File</td>
<td>Multifasta sequence database GAMMA is to use for gene matching</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/gamma/default_ResFinderDB_Combined_05-06-20.fsa</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>min_length_percent_gammas</strong></td>
<td>Int</td>
<td>Threshold of match length percent used by GAMMA-s</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>min_percent_identity</strong></td>
<td>Int</td>
<td>Threshold of identity for gene matches in percent</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>output_fasta</strong></td>
<td>Boolean</td>
<td>Allows GAMMA to output gene matches in FASTA format</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>output_gff</strong></td>
<td>Boolean</td>
<td>Allows GAMMA to output gene matches in GFF format</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>run_gammas</strong></td>
<td>Boolean</td>
<td>Turns on GAMMA-S which runs GAMMA on nucleotide sequences rather than translating input sequences and matches</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/kmerfinder:3.0.2--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>kmerfinder_args</strong></td>
<td>String</td>
<td>Kmerfinder additional arguments</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>kmerfinder_db</strong></td>
<td>File</td>
<td>Bacterial database for KmerFinder</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kmerfinder/kmerfinder_bacteria_20230911.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity; set to 95 because there is a strict threshold of 95% identity for typing purposes</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_agr_typing_only</strong></td>
<td>Boolean</td>
<td>Set to true to skip agr operon extraction and frameshift detection</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/agrvate:1.0.2--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/amrsearch:0.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>assembly_only</strong></td>
<td>Boolean</td>
<td>Set to true if only analyzing input assembly</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_poppunk</strong></td>
<td>Boolean</td>
<td>If true, runs PopPUNK for GPSC cluster designation for S. pneumoniae</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_shigeifinder_reads_input</strong></td>
<td>Boolean</td>
<td>If set to "true", the ShigEiFinder task will run again but using read files as input instead of the assembly file. Input is shown but not used for TheiaProk_FASTA.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_stxtyper</strong></td>
<td>Boolean</td>
<td>If set to "true", the StxTyper task will run on all samples regardless of the gambit_predicted_taxon output. Useful if you suspect a non-E.coli or non-Shigella sample contains stx genes.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_tbp_parser</strong></td>
<td>Boolean</td>
<td>If set to "true", activates the tbp_parser module and results in more outputs, including&nbsp;tbp_parser_looker_report_csv, tbp_parser_laboratorian_report_csv,  tbp_parser_lims_report_csv, tbp_parser_coverage_report, and tbp_parser_genome_percent_coverage</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cauris_cladetyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_kmer_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_max_distance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6</strong></td>
<td>File</td>
<td>The reference assembly for clade 6</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade6_GCA_032714025.1_ASM3271402v1_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6_annotated</strong></td>
<td>File</td>
<td>The path to the annotated reference for clade 6</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>clockwork_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/cdcgov/varpipe_wgs_with_refs:2bc7234074bd53d9e92a1048b0485763cd9bbf6f4d12d5a1cc82bfec8ca7d75e</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/ectyper:1.0.0--pyhdfd78af_1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minumum percent coverage required for an H antigen allele match</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_identity</strong></td>
<td>Int</td>
<td>Percent identity required for an H antigen allele match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minumum percent coverage required for an O antigen allele match</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_identity</strong></td>
<td>Int</td>
<td>Percent identity required for an O antigen allele match</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_print_alleles</strong></td>
<td>Boolean</td>
<td>Set to true to print the allele sequences as the final column</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_verify</strong></td>
<td>Boolean</td>
<td>Set to true to enable E. coli species verification</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_align_diff</strong></td>
<td>Int</td>
<td>Threshold for difference between alignment length and subject length in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_cluster_distance</strong></td>
<td>Int</td>
<td>Distance between cluster of matches to consider as different clusters.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_culling_limit</strong></td>
<td>Int</td>
<td>Total hits to return in a position.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/emmtyper:0.2.0--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_gap</strong></td>
<td>Int</td>
<td>Threshold gap to allow in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_max_size</strong></td>
<td>Int</td>
<td>Maximum size of PCR product.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_good</strong></td>
<td>Int</td>
<td>Minimum size where there must be 2 matches for each mismatch.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimal percent identity of sequence.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_perfect</strong></td>
<td>Int</td>
<td>Minimum size of perfect match at 3' primer end.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_mismatch</strong></td>
<td>Int</td>
<td>Threshold for number of mismatch to allow in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_wf</strong></td>
<td>String</td>
<td>Choose workflow [blast</td>
<td>pcr]</td>
<td></td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtypingtool_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/emmtypingtool:0.0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>genotyphi_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.11.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_broken_gene_length</strong></td>
<td>Int</td>
<td>Minimum length to consider a broken gene</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/hicap:1.0.3--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_broken_gene_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percentage identity to consider a broken gene</td>
<td>0.8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum percentage coverage to consider a single gene complete</td>
<td>0.8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percentage identity to consider a single gene complete</td>
<td>0.7</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kaptive:2.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_low_gene_percent_identity</strong></td>
<td>Float</td>
<td>Percent identity threshold for what counts as a low identity match in the gene BLAST search</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum required percent coverage for the gene BLAST search via tBLASTn</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum required percent identity for the gene BLAST search via tBLASTn</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_start_end_margin</strong></td>
<td>Int</td>
<td>Determines flexibility in identifying the start and end of a locus - if this value is 10, a locus match that is missing the first 8 base pairs will still count as capturing the start of the locus</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kleborate:2.2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_kaptive_confidence</strong></td>
<td>String</td>
<td>{None,Low,Good,High,Very_high,Perfect} Minimum Kaptive confidence to call K/O loci - confidence levels below this will be reported as unknown</td>
<td>Good</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum alignment percent coverage for main results</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum alignment percent identity for main results</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum alignment percent coverage for spurious results</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_identity</strong></td>
<td>Float</td>
<td>Minimum alignment percent identity for spurious results</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_kaptive</strong></td>
<td>Boolean</td>
<td>Equivalent to --kaptive_k --kaptive_</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_resistance</strong></td>
<td>Boolean</td>
<td>Set to true to turn on resistance genes screening (default: no resistance gene screening)</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>legsta_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/lissero:0.4.9--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum percent coverage of the gene to accept a match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percent identity to accept a match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>meningotype_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/meningotype:0.8.5--pyhdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ngmaster_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ngmaster:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>paired_end</strong></td>
<td>Boolean</td>
<td>Set to true if your data is paired-end FASTQ files</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pasty:1.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum coverage of a O-antigen to be considered for serogrouping by pasty</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum percent identity for a blast hit to be considered for serogrouping</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pbptyper:1.0.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum percent coverage to count a hit</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum percent identity to count a hit</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/poppunk:2.4.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_npy</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.dists.npy</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.dists.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_external_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_external_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_npz</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_fit.npz</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_fit.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_graph_gt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_graph.gt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_h5</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.h5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_qcreport_txt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_qcreport.txt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_npy</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.dists.npy</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.dists.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_graph_gt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6refs_graph.gt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_h5</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.h5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_unword_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_unword_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>run_amr_search</strong></td>
<td>Boolean</td>
<td>If set to true AMR_Search workflow will be run if species is part of supported taxon, see AMR_Search docs.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seqsero2s_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/seqsero2s:1.1.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seroba_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/seroba:1.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>serotypefinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/serotypefinder:2.0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigatyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/shigatyper:2.0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigeifinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/shigeifinder:1.3.5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/sistr_cmd:1.1.1--pyh864c0ab_2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_use_full_cgmlst_db</strong></td>
<td>Boolean</td>
<td>Set to true to use the full set of cgMLST alleles which can include highly similar alleles. By default the smaller "centroid" alleles or representative alleles are used for each marker</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_base_quality</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_gene_query_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_map_qual</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_maxsoft</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_frac</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_quality</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_query_gene</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_afumigatus</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/aspergillus/Aspergillus_fumigatus_GCF_000002655.1_ASM265v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_cryptoneo</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/cryptococcus/Cryptococcus_neoformans_GCF_000091045.1_ASM9104v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_variants_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_mykrobe_opts</strong></td>
<td>String</td>
<td>Additional options for mykrobe in sonneityping</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_do_enrich</strong></td>
<td>Boolean</td>
<td>Set to true to enable PCR product enrichment</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/spatyper:0.3.3--pyhdfd78af_3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/srst2:0.2.0-vcholerae</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_gene_max_mismatch</strong></td>
<td>Int</td>
<td>Maximum number of mismatches for SRST2 to call a gene as present</td>
<td>2000</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_max_divergence</strong></td>
<td>Int</td>
<td>Maximum divergence, in percentage, for SRST2 to call a gene as present</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_depth</strong></td>
<td>Int</td>
<td>Minimum depth of coverage for SRST2 to call a gene as present</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_edge_depth</strong></td>
<td>Int</td>
<td>Minimum edge depth for SRST2 to call a gene as present</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum breadth of coverage for SRST2 to call a gene as present</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>staphopia_sccmec_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/staphopia-sccmec:1.0.0--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/stxtyper:1.0.42</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_enable_debug</strong></td>
<td>Boolean</td>
<td>When enabled, additional messages are printed and files in $TMPDIR are not removed after running</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_add_cs_lims</strong></td>
<td>Boolean</td>
<td>Set to true add cycloserine results to the LIMS report</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_config</strong></td>
<td>File</td>
<td>The configuration file to use, in YAML format (overrides all other arguments except input_json and input_bam)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_coverage_regions_bed</strong></td>
<td>File</td>
<td>A bed file that lists the regions to be considered for QC</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_debug</strong></td>
<td>Boolean</td>
<td>Activate the debug mode on tbp_parser; increases logging outputs</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/tbp-parser:2.6.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_etha237_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in ethA at protein position 237 to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_expert_rule_regions_bed</strong></td>
<td>File</td>
<td>A file that contains the regions where R mutations and expert rules are applied</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_depth</strong></td>
<td>Int</td>
<td>Minimum depth for a variant to pass QC in tbp_parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_frequency</strong></td>
<td>Float</td>
<td>The minimum frequency for a mutation to pass QC</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_percent_coverage</strong></td>
<td>Float</td>
<td>The minimum coverage for a region to pass QC in tbp_parser</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_read_support</strong></td>
<td>Int</td>
<td>The minimum read support for a mutation to pass QC</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_operator</strong></td>
<td>String</td>
<td>Fills the "operator" field in the tbp_parser output files</td>
<td>Operator not provided</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_output_seq_method_type</strong></td>
<td>String</td>
<td>Fills out the "seq_method" field in the tbp_parser output files</td>
<td>WGS</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rpob449_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation at protein position 449 to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in rrl to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_read_support</strong></td>
<td>Int</td>
<td>Minimum read support for a mutation in rrl to pass QC in tbp-parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in rrs to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_read_support</strong></td>
<td>Int</td>
<td>Minimum read support for a mutation in rrs to pass QC in tbp-parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_tngs_data</strong></td>
<td>Boolean</td>
<td>Set to true to enable tNGS-specific parameters and runs in tbp-parser</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_additional_parameters</strong></td>
<td>String</td>
<td>Additional parameters for TBProfiler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_custom_db</strong></td>
<td>File</td>
<td>TBProfiler uses by default the TBDB database; if you have a custom database you wish to use, you must provide a custom database in this field and set tbprofiler_run_custom_db to true</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/tbprofiler:6.6.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_mapper</strong></td>
<td>String</td>
<td>The mapping tool used in TBProfiler to align the reads to the reference genome; see TBProfilers original documentation for available options.</td>
<td>bwa</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_af</strong></td>
<td>Float</td>
<td>The minimum allele frequency to call a variant</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_depth</strong></td>
<td>Int</td>
<td>The minimum depth for a variant to be called.</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_cdph_db</strong></td>
<td>Boolean</td>
<td>TBProfiler uses by default the TBDB database; set this value to "true" to use the WHO v2 database with customizations for CDPH</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_custom_db</strong></td>
<td>Boolean</td>
<td>TBProfiler uses by default the TBDB database; if you have a custom database you wish to use, you must set this value to true and provide a custom database in the tbprofiler_custom_db field</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_caller</strong></td>
<td>String</td>
<td>Select a different variant caller for TBProfiler to use by writing it in this block; see TBProfilers original documentation for available options.</td>
<td>GATK</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_calling_params</strong></td>
<td>String</td>
<td>Enter additional variant calling parameters in this free text input to customize how the variant caller works in TBProfiler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>theiaeuk</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_lineage_barcodes</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_skip_subsampling</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_subsampling_fraction</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_database</strong></td>
<td>String</td>
<td>The specific database to use</td>
<td>virulence_ecoli</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/virulencefinder:2.0.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_coverage</strong></td>
<td>Float</td>
<td>The threshold for minimum coverage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_identity</strong></td>
<td>Float</td>
<td>The threshold for minimum blast identity</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>database</strong></td>
<td>String</td>
<td>User-specified database</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>database_path</strong></td>
<td>String</td>
<td>Path to user-specified database</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/plasmidfinder:2.1.6</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>method_path</strong></td>
<td>String</td>
<td>Path to files for a user-specified method to use (blast or kma)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Threshold for minimum coverage, default threshold from PlasmidFinder CLI tool is used (0.60)</td>
<td>0.6</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Threshold for mininum blast identity, default threshold from PlasmidFinder CLI tool is used (0.90). This default differs from the default of the PlasmidFinder webtool (0.95)</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>compliant</strong></td>
<td>Boolean</td>
<td>If true (default), forces Genbank/ENA/DDJB compliant headers in Prokka output files</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/prokka:1.14.5</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>prodigal_tf</strong></td>
<td>File</td>
<td><a href="https://github.com/tseemann/prokka#option---prodigaltf">https://github.com/tseemann/prokka#option---prodigaltf</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>prokka_arguments</strong></td>
<td>String</td>
<td>Any additional <a href="https://github.com/tseemann/prokka#command-line-options">https://github.com/tseemann/prokka#command-line-options</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>proteins</strong></td>
<td>Boolean</td>
<td>FASTA file of trusted proteins for Prokka to first use for annotations</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_length_unambiguous</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>meanbaseq_trim</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_abundance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_clean2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_raw2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_Degenerate</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_N</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>percent_reference_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>vadr_num_alerts</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>min_contig_length</strong></td>
<td>Int</td>
<td>Minimum length of contig for QUAST</td>
<td>500</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_docker</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the Kraken2 task should be called; for non-TheiaCoV workflows, the <code>kraken_db</code> variable must be provided.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>downsampling_coverage</strong></td>
<td>Float</td>
<td>The desired coverage to sub-sample the reads to with RASUSA</td>
<td>150</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-08-28-v4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task. Increase this when using large (&gt;30GB kraken2 databases such as the "k2_standard" database)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/nanoq:0.9.0--hec16e2b_1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_max_read_length</strong></td>
<td>Int</td>
<td>The maximum read length to keep after trimming</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_max_read_qual</strong></td>
<td>Int</td>
<td>The maximum read quality to keep after trimming</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_min_read_length</strong></td>
<td>Int</td>
<td>The minimum read length to keep after trimming</td>
<td>500</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_min_read_qual</strong></td>
<td>Int</td>
<td>The minimum read quality to keep after trimming</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_docker</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_bases</strong></td>
<td>String</td>
<td>Explicitly set the number of bases required e.g., 4.3kb, 7Tb, 9000, 4.1MB. If this option is given, --coverage and --genome-size are ignored</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/rasusa:2.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_fraction_of_reads</strong></td>
<td>Float</td>
<td>Subsample to a fraction of the reads - e.g., 0.5 samples half the reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_number_of_reads</strong></td>
<td>Int</td>
<td>Subsample to a specific number of reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_seed</strong></td>
<td>Int</td>
<td>Random seed to use</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>run_prefix</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>acquired</strong></td>
<td>Boolean</td>
<td>Set to true to tell ResFinder to identify acquired resistance genes</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>call_pointfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable detection of point mutations.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/resfinder:4.1.11</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum coverage breadth of a gene for it to be identified</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum identity for ResFinder to identify a gene</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>abricate_db</strong></td>
<td>String</td>
<td>Database to use with the Abricate tool. Options: NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB</td>
<td>vfdb</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>amrfinder_use_gff</strong></td>
<td>Boolean</td>
<td>Whether or not to use the GFF and protein FASTA files for AMRFinderPlus</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>bakta_db</strong></td>
<td>String</td>
<td>Database selection for Bakta annotation. Options: light (smaller, faster), full (more comprehensive), or a Google Storage URI (gs://...) pointing to a custom Bakta database archive (.tar.gz). The selected database will be extracted before annotation.</td>
<td>full</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>call_abricate</strong></td>
<td>Boolean</td>
<td>Set to true to enable the Abricate task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>call_ani</strong></td>
<td>Boolean</td>
<td>Set to true to enable the ANI task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>call_arln_stats</strong></td>
<td>Boolean</td>
<td>Set to true to enable the return of ARLN required Auto PASS/FAIL outputs</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>call_gamma</strong></td>
<td>Boolean</td>
<td>Set to true in order to enable GAMMA task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>call_kmerfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the kmerfinder task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>call_plasmidfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the plasmidfinder task</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>call_resfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the ResFinder task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>city</strong></td>
<td>String</td>
<td>Will be used in the "city" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>collection_date</strong></td>
<td>String</td>
<td>Will be used in the "collection_date" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>county</strong></td>
<td>String</td>
<td>Will be used in the "county" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>expected_taxon</strong></td>
<td>String</td>
<td>If provided, this input will override the taxonomic assignment made by GAMBIT and launch the relevant taxon-specific submodules. It will also modify the organism flag used by AMRFinderPlus. Example format: "Salmonella enterica"</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>genome_annotation</strong></td>
<td>String</td>
<td>If set to "bakta", TheiaProk will use Bakta rather than Prokka to annotate the genome</td>
<td>prokka</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>max_genome_length</strong></td>
<td>Int</td>
<td>Maximum genome length able to pass read screening. For TheiaProk_ONT, screening using max_genome_length is skipped by default.</td>
<td>18040666</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>min_basepairs</strong></td>
<td>Int</td>
<td>Minimum number of base pairs able to pass read screening</td>
<td>2241820</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>min_coverage</strong></td>
<td>Int</td>
<td>Minimum genome coverage able to pass read screening. Screening using min_coverage is skipped by default.</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>min_genome_length</strong></td>
<td>Int</td>
<td>Minimum genome length able to pass read screening. For TheiaProk_ONT, screening using min_genome_length is skipped by default.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>min_reads</strong></td>
<td>Int</td>
<td>Minimum number of reads to pass read screening</td>
<td>5000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>mlst_run_secondary_scheme</strong></td>
<td>Boolean</td>
<td>If true, will run secondary scheme if primary scheme is of ecoli or abaumannii, these two have multiple schemes that are relevant.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>mlst_scheme_override</strong></td>
<td>Boolean</td>
<td>If true, will force E. coli scheme to be used when Gambit predicts Escherichia coli, otherwise will return scheme MLST predicts.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>originating_lab</strong></td>
<td>String</td>
<td>Will be used in the "originating_lab" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>perform_characterization</strong></td>
<td>Boolean</td>
<td>Set to "false" if you want to only generate an assembly and relevant QC metrics and skip all characterization tasks</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>qc_check_table</strong></td>
<td>File</td>
<td>TSV value with taxons for rows and QC values for columns; internal cells represent user-determined QC thresholds; if provided, turns on the QC Check task. See below for an example QC Check table.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>run_id</strong></td>
<td>String</td>
<td>Will be used in the "run_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>seq_method</strong></td>
<td>String</td>
<td>The sequencing methodology used to generate the input read data; for TheiaProk workflows, this input will be used in the "seq_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td>ONT</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>skip_mash</strong></td>
<td>Boolean</td>
<td>If true, skips estimation of genome size and coverage using mash in read screening steps. As a result, providing true also prevents screening using these parameters.</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>skip_screen</strong></td>
<td>Boolean</td>
<td>Set to True to skip the read screening prior to analysis</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>taxon_tables</strong></td>
<td>File</td>
<td>File indicating data table names to copy samples of a particular taxon to</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>terra_project</strong></td>
<td>String</td>
<td>The name of the Terra Project where you want the taxon tables written to</td>
<td>NA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>terra_workspace</strong></td>
<td>String</td>
<td>The name of the Terra Workspace where you want the taxon tables written to</td>
<td>NA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>zip</strong></td>
<td>String</td>
<td>Will be used in the "zip" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mlst:2.23.0-2024-12-31</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum % breadth of coverage to report an MLST allele</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum % identity to known MLST gene to report an MLST allele</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>minscore</strong></td>
<td>Float</td>
<td>Minimum score <a href="https://github.com/tseemann/mlst#scoring-system">https://github.com/tseemann/mlst#scoring-system</a> to assign an MLST profile</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>nopath</strong></td>
<td>Boolean</td>
<td>true = use mlst --nopath. If set to false, filename paths are not stripped from FILE column in output TSV</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>scheme</strong></td>
<td>String</td>
<td>Dont autodetect the MLST scheme; force this scheme on all inputs (see <a href="https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab">https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab</a> for accepted strings)</td>
<td>None</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiaprok_fasta</td>
<td><strong>assembly_fasta</strong></td>
<td>File</td>
<td>The assembly file for your sample in FASTA format</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA %coverage for the Abricate task</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>abricate</td>
<td><strong>min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA %identity for the Abricate task</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>detailed_drug_class</strong></td>
<td>Boolean</td>
<td>If set to true, amrfinderplus_amr_classes and amrfinderplus_amr_subclasses outputs will be created</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-amrfinderplus:4.0.23-2025-07-16.1</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>hide_point_mutations</strong></td>
<td>Boolean</td>
<td>If set to true, point mutations are not reported</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum proportion of reference gene covered for a BLAST-based hit (Methods BLAST or PARTIAL)." Attribute should be a float ranging from 0-1, such as 0.6 (equal to 60% coverage)</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum identity for a blast-based hit hit (Methods BLAST or PARTIAL). -1 means use a curated threshold if it exists and 0.9 otherwise. Setting this value to something other than -1 will override any curated similarity cutoffs. Attribute should be a float ranging from 0-1, such as 0.95 (equal to 95% identity)</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>separate_betalactam_genes</strong></td>
<td>Boolean</td>
<td>Report beta-Lactam AMR genes separated out by all beta-lactam and the respective beta-lactam subclasses</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>ani_threshold</strong></td>
<td>Float</td>
<td>ANI value threshold must be surpassed in order to output the ani_top_species_match. If a genome does not surpass this threshold (and the percent_bases_aligned_threshold) then the ani_top_species_match output String will show a warning instead of a genus &amp; species.</td>
<td>80.0</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mummer:4.0.0-rgdv2</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>mash_filter</strong></td>
<td>Float</td>
<td>Mash distance threshold over which ANI is not calculated</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>percent_bases_aligned_threshold</strong></td>
<td>Float</td>
<td>Threshold regarding the proportion of bases aligned between the query genome and reference genome. If a genome does not surpass this threshold (and the ani_threshold) then the ani_top_species_match output String will show a warning instead of a genus &amp; species.</td>
<td>70.0</td>
<td>Optional</td>
</tr>
<tr>
<td>ani</td>
<td><strong>ref_genome</strong></td>
<td>File</td>
<td>If not set, uses all 43 genomes in RGDv2</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/arln_stats:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>read1_clean</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>read1_raw</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>read2_clean</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>read2_raw</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>bakta_opts</strong></td>
<td>String</td>
<td>Parameters to pass to bakta from <a href="https://github.com/oschwengers/bakta#usage">https://github.com/oschwengers/bakta#usage</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>compliant</strong></td>
<td>Boolean</td>
<td>If true, forces Genbank/ENA/DDJB compliant headers in Bakta output files</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>200</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/bakta:1.10.3</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>24</td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>prodigal_tf</strong></td>
<td>File</td>
<td>Prodigal training file to use for CDS prediction by bakta</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>proteins</strong></td>
<td>File</td>
<td>Fasta file of trusted protein sequences for CDS annotation</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ezlabgva/busco:v5.7.1_cv1</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>eukaryote</strong></td>
<td>Boolean</td>
<td>Assesses eukaryotic organisms, rather than prokaryotic organisms</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>busco</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>theiaviral_panel</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>gambit_db_genomes</strong></td>
<td>File</td>
<td>Database of metadata for assembled query genomes; requires complementary signatures file. If not provided, uses default database "/gambit-db"</td>
<td>gs://gambit-databases-rp/2.1.0/gambit-metadata-2.1.0-20250808.gdb</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>gambit_db_signatures</strong></td>
<td>File</td>
<td>Signatures file; requires complementary genomes file. If not specified, the file from the docker container will be used.</td>
<td>gs://gambit-databases-rp/2.1.0/gambit-signatures-2.1.0-20250808.gs</td>
<td>Optional</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gamma:2.2</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>extended_output</strong></td>
<td>Boolean</td>
<td>GAMMA will provide the extended output when set to true</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>gamma_db</strong></td>
<td>File</td>
<td>Multifasta sequence database GAMMA is to use for gene matching</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/gamma/default_ResFinderDB_Combined_05-06-20.fsa</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>min_length_percent_gammas</strong></td>
<td>Int</td>
<td>Threshold of match length percent used by GAMMA-s</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>min_percent_identity</strong></td>
<td>Int</td>
<td>Threshold of identity for gene matches in percent</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>output_fasta</strong></td>
<td>Boolean</td>
<td>Allows GAMMA to output gene matches in FASTA format</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>output_gff</strong></td>
<td>Boolean</td>
<td>Allows GAMMA to output gene matches in GFF format</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>run_gammas</strong></td>
<td>Boolean</td>
<td>Turns on GAMMA-S which runs GAMMA on nucleotide sequences rather than translating input sequences and matches</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/kmerfinder:3.0.2--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>kmerfinder_args</strong></td>
<td>String</td>
<td>Kmerfinder additional arguments</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>kmerfinder_db</strong></td>
<td>File</td>
<td>Bacterial database for KmerFinder</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kmerfinder/kmerfinder_bacteria_20230911.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity; set to 95 because there is a strict threshold of 95% identity for typing purposes</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_agr_typing_only</strong></td>
<td>Boolean</td>
<td>Set to true to skip agr operon extraction and frameshift detection</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/agrvate:1.0.2--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/amrsearch:0.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_poppunk</strong></td>
<td>Boolean</td>
<td>If true, runs PopPUNK for GPSC cluster designation for S. pneumoniae</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_shigeifinder_reads_input</strong></td>
<td>Boolean</td>
<td>If set to "true", the ShigEiFinder task will run again but using read files as input instead of the assembly file. Input is shown but not used for TheiaProk_FASTA.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_stxtyper</strong></td>
<td>Boolean</td>
<td>If set to "true", the StxTyper task will run on all samples regardless of the gambit_predicted_taxon output. Useful if you suspect a non-E.coli or non-Shigella sample contains stx genes.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>call_tbp_parser</strong></td>
<td>Boolean</td>
<td>If set to "true", activates the tbp_parser module and results in more outputs, including&nbsp;tbp_parser_looker_report_csv, tbp_parser_laboratorian_report_csv,  tbp_parser_lims_report_csv, tbp_parser_coverage_report, and tbp_parser_genome_percent_coverage</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cauris_cladetyper_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_kmer_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_max_distance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5_annotated</strong></td>
<td>File</td>
<td>*Provide an empty file if running TheiaProk on the command-line</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6</strong></td>
<td>File</td>
<td>The reference assembly for clade 6</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade6_GCA_032714025.1_ASM3271402v1_genomic.fasta</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6_annotated</strong></td>
<td>File</td>
<td>The path to the annotated reference for clade 6</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>clockwork_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/cdcgov/varpipe_wgs_with_refs:2bc7234074bd53d9e92a1048b0485763cd9bbf6f4d12d5a1cc82bfec8ca7d75e</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/ectyper:1.0.0--pyhdfd78af_1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minumum percent coverage required for an H antigen allele match</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_h_min_percent_identity</strong></td>
<td>Int</td>
<td>Percent identity required for an H antigen allele match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minumum percent coverage required for an O antigen allele match</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_o_min_percent_identity</strong></td>
<td>Int</td>
<td>Percent identity required for an O antigen allele match</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_print_alleles</strong></td>
<td>Boolean</td>
<td>Set to true to print the allele sequences as the final column</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_verify</strong></td>
<td>Boolean</td>
<td>Set to true to enable E. coli species verification</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_align_diff</strong></td>
<td>Int</td>
<td>Threshold for difference between alignment length and subject length in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_cluster_distance</strong></td>
<td>Int</td>
<td>Distance between cluster of matches to consider as different clusters.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_culling_limit</strong></td>
<td>Int</td>
<td>Total hits to return in a position.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/emmtyper:0.2.0--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_gap</strong></td>
<td>Int</td>
<td>Threshold gap to allow in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_max_size</strong></td>
<td>Int</td>
<td>Maximum size of PCR product.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_good</strong></td>
<td>Int</td>
<td>Minimum size where there must be 2 matches for each mismatch.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimal percent identity of sequence.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_min_perfect</strong></td>
<td>Int</td>
<td>Minimum size of perfect match at 3' primer end.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_mismatch</strong></td>
<td>Int</td>
<td>Threshold for number of mismatch to allow in BLAST hit.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_wf</strong></td>
<td>String</td>
<td>Choose workflow [blast</td>
<td>pcr]</td>
<td></td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtypingtool_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/emmtypingtool:0.0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>genotyphi_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.11.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_broken_gene_length</strong></td>
<td>Int</td>
<td>Minimum length to consider a broken gene</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/hicap:1.0.3--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_broken_gene_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percentage identity to consider a broken gene</td>
<td>0.8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum percentage coverage to consider a single gene complete</td>
<td>0.8</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_min_gene_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percentage identity to consider a single gene complete</td>
<td>0.7</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kaptive:2.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_low_gene_percent_identity</strong></td>
<td>Float</td>
<td>Percent identity threshold for what counts as a low identity match in the gene BLAST search</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum required percent coverage for the gene BLAST search via tBLASTn</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum required percent identity for the gene BLAST search via tBLASTn</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_start_end_margin</strong></td>
<td>Int</td>
<td>Determines flexibility in identifying the start and end of a locus - if this value is 10, a locus match that is missing the first 8 base pairs will still count as capturing the start of the locus</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kleborate:2.2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_kaptive_confidence</strong></td>
<td>String</td>
<td>{None,Low,Good,High,Very_high,Perfect} Minimum Kaptive confidence to call K/O loci - confidence levels below this will be reported as unknown</td>
<td>Good</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum alignment percent coverage for main results</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum alignment percent identity for main results</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum alignment percent coverage for spurious results</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_min_spurious_percent_identity</strong></td>
<td>Float</td>
<td>Minimum alignment percent identity for spurious results</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_kaptive</strong></td>
<td>Boolean</td>
<td>Equivalent to --kaptive_k --kaptive_</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_skip_resistance</strong></td>
<td>Boolean</td>
<td>Set to true to turn on resistance genes screening (default: no resistance gene screening)</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>legsta_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/lissero:0.4.9--py_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum percent coverage of the gene to accept a match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum percent identity to accept a match</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>meningotype_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/meningotype:0.8.5--pyhdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ngmaster_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ngmaster:1.0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ont_data</strong></td>
<td>Boolean</td>
<td>Set to true if your data is ONT FASTQ files</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pasty:1.0.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum coverage of a O-antigen to be considered for serogrouping by pasty</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum percent identity for a blast hit to be considered for serogrouping</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pbptyper:1.0.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum percent coverage to count a hit</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum percent identity to count a hit</td>
<td>90</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/poppunk:2.4.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_npy</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.dists.npy</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.dists.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_external_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_external_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_npz</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_fit.npz</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_fit.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_graph_gt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_graph.gt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_h5</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.h5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_qcreport_txt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_qcreport.txt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_npy</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.dists.npy</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.dists.pkl</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_graph_gt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6refs_graph.gt</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_h5</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.h5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_unword_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_unword_clusters.csv</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>run_amr_search</strong></td>
<td>Boolean</td>
<td>If set to true AMR_Search workflow will be run if species is part of supported taxon, see AMR_Search docs.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seqsero2s_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/seqsero2s:1.1.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seroba_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/seroba:1.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>serotypefinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/serotypefinder:2.0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigatyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/shigatyper:2.0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigeifinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/shigeifinder:1.3.5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/sistr_cmd:1.1.1--pyh864c0ab_2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_use_full_cgmlst_db</strong></td>
<td>Boolean</td>
<td>Set to true to use the full set of cgMLST alleles which can include highly similar alleles. By default the smaller "centroid" alleles or representative alleles are used for each marker</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_base_quality</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_gene_query_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_map_qual</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_maxsoft</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_coverage</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_frac</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_min_quality</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_query_gene</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_afumigatus</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/aspergillus/Aspergillus_fumigatus_GCF_000002655.1_ASM265v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_reference_cryptoneo</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/cryptococcus/Cryptococcus_neoformans_GCF_000091045.1_ASM9104v1_genomic.gbff</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_variants_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_mykrobe_opts</strong></td>
<td>String</td>
<td>Additional options for mykrobe in sonneityping</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_do_enrich</strong></td>
<td>Boolean</td>
<td>Set to true to enable PCR product enrichment</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/spatyper:0.3.3--pyhdfd78af_3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/srst2:0.2.0-vcholerae</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_gene_max_mismatch</strong></td>
<td>Int</td>
<td>Maximum number of mismatches for SRST2 to call a gene as present</td>
<td>2000</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_max_divergence</strong></td>
<td>Int</td>
<td>Maximum divergence, in percentage, for SRST2 to call a gene as present</td>
<td>20</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_depth</strong></td>
<td>Int</td>
<td>Minimum depth of coverage for SRST2 to call a gene as present</td>
<td>5</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_edge_depth</strong></td>
<td>Int</td>
<td>Minimum edge depth for SRST2 to call a gene as present</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum breadth of coverage for SRST2 to call a gene as present</td>
<td>80</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>staphopia_sccmec_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/staphopia-sccmec:1.0.0--hdfd78af_0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/stxtyper:1.0.42</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_enable_debug</strong></td>
<td>Boolean</td>
<td>When enabled, additional messages are printed and files in $TMPDIR are not removed after running</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_add_cs_lims</strong></td>
<td>Boolean</td>
<td>Set to true add cycloserine results to the LIMS report</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_config</strong></td>
<td>File</td>
<td>The configuration file to use, in YAML format (overrides all other arguments except input_json and input_bam)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_coverage_regions_bed</strong></td>
<td>File</td>
<td>A bed file that lists the regions to be considered for QC</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_debug</strong></td>
<td>Boolean</td>
<td>Activate the debug mode on tbp_parser; increases logging outputs</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/tbp-parser:2.6.0</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_etha237_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in ethA at protein position 237 to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_expert_rule_regions_bed</strong></td>
<td>File</td>
<td>A file that contains the regions where R mutations and expert rules are applied</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_depth</strong></td>
<td>Int</td>
<td>Minimum depth for a variant to pass QC in tbp_parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_frequency</strong></td>
<td>Float</td>
<td>The minimum frequency for a mutation to pass QC</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_percent_coverage</strong></td>
<td>Float</td>
<td>The minimum coverage for a region to pass QC in tbp_parser</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_min_read_support</strong></td>
<td>Int</td>
<td>The minimum read support for a mutation to pass QC</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_operator</strong></td>
<td>String</td>
<td>Fills the "operator" field in the tbp_parser output files</td>
<td>Operator not provided</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_output_seq_method_type</strong></td>
<td>String</td>
<td>Fills out the "seq_method" field in the tbp_parser output files</td>
<td>WGS</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rpob449_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation at protein position 449 to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in rrl to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrl_read_support</strong></td>
<td>Int</td>
<td>Minimum read support for a mutation in rrl to pass QC in tbp-parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_frequency</strong></td>
<td>Float</td>
<td>Minimum frequency for a mutation in rrs to pass QC in tbp-parser</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_rrs_read_support</strong></td>
<td>Int</td>
<td>Minimum read support for a mutation in rrs to pass QC in tbp-parser</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_tngs_data</strong></td>
<td>Boolean</td>
<td>Set to true to enable tNGS-specific parameters and runs in tbp-parser</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_additional_parameters</strong></td>
<td>String</td>
<td>Additional parameters for TBProfiler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_custom_db</strong></td>
<td>File</td>
<td>TBProfiler uses by default the TBDB database; if you have a custom database you wish to use, you must provide a custom database in this field and set tbprofiler_run_custom_db to true</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/tbprofiler:6.6.3</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_mapper</strong></td>
<td>String</td>
<td>The mapping tool used in TBProfiler to align the reads to the reference genome; see TBProfilers original documentation for available options.</td>
<td>bwa</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_af</strong></td>
<td>Float</td>
<td>The minimum allele frequency to call a variant</td>
<td>0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_min_depth</strong></td>
<td>Int</td>
<td>The minimum depth for a variant to be called.</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_cdph_db</strong></td>
<td>Boolean</td>
<td>TBProfiler uses by default the TBDB database; set this value to "true" to use the WHO v2 database with customizations for CDPH</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_run_custom_db</strong></td>
<td>Boolean</td>
<td>TBProfiler uses by default the TBDB database; if you have a custom database you wish to use, you must set this value to true and provide a custom database in the tbprofiler_custom_db field</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_caller</strong></td>
<td>String</td>
<td>Select a different variant caller for TBProfiler to use by writing it in this block; see TBProfilers original documentation for available options.</td>
<td>GATK</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_variant_calling_params</strong></td>
<td>String</td>
<td>Enter additional variant calling parameters in this free text input to customize how the variant caller works in TBProfiler</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>theiaeuk</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_lineage_barcodes</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_skip_subsampling</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_subsampling_fraction</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_database</strong></td>
<td>String</td>
<td>The specific database to use</td>
<td>virulence_ecoli</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/virulencefinder:2.0.4</td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_coverage</strong></td>
<td>Float</td>
<td>The threshold for minimum coverage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_min_percent_identity</strong></td>
<td>Float</td>
<td>The threshold for minimum blast identity</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>database</strong></td>
<td>String</td>
<td>User-specified database</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>database_path</strong></td>
<td>String</td>
<td>Path to user-specified database</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/plasmidfinder:2.1.6</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>method_path</strong></td>
<td>String</td>
<td>Path to files for a user-specified method to use (blast or kma)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Threshold for minimum coverage, default threshold from PlasmidFinder CLI tool is used (0.60)</td>
<td>0.6</td>
<td>Optional</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Threshold for mininum blast identity, default threshold from PlasmidFinder CLI tool is used (0.90). This default differs from the default of the PlasmidFinder webtool (0.95)</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>compliant</strong></td>
<td>Boolean</td>
<td>If true (default), forces Genbank/ENA/DDJB compliant headers in Prokka output files</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/prokka:1.14.5</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>prodigal_tf</strong></td>
<td>File</td>
<td><a href="https://github.com/tseemann/prokka#option---prodigaltf">https://github.com/tseemann/prokka#option---prodigaltf</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>prokka_arguments</strong></td>
<td>String</td>
<td>Any additional <a href="https://github.com/tseemann/prokka#command-line-options">https://github.com/tseemann/prokka#command-line-options</a></td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>proteins</strong></td>
<td>Boolean</td>
<td>FASTA file of trusted proteins for Prokka to first use for annotations</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_length_unambiguous</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>meanbaseq_trim</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_abundance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_clean1</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_clean2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_raw1</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_raw2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_Degenerate</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_N</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>percent_reference_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>vadr_num_alerts</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quast</td>
<td><strong>min_contig_length</strong></td>
<td>Int</td>
<td>Minimum length of contig for QUAST</td>
<td>500</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>acquired</strong></td>
<td>Boolean</td>
<td>Set to true to tell ResFinder to identify acquired resistance genes</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>call_pointfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable detection of point mutations.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/resfinder:4.1.11</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum coverage breadth of a gene for it to be identified</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum identity for ResFinder to identify a gene</td>
<td>0.9</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>abricate_db</strong></td>
<td>String</td>
<td>Database to use with the Abricate tool. Options: NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB</td>
<td>vfdb</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>amrfinder_use_gff</strong></td>
<td>Boolean</td>
<td>Whether or not to use the GFF and protein FASTA files for AMRFinderPlus</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>bakta_db</strong></td>
<td>String</td>
<td>Database selection for Bakta annotation. Options: light (smaller, faster), full (more comprehensive), or a Google Storage URI (gs://...) pointing to a custom Bakta database archive (.tar.gz). The selected database will be extracted before annotation.</td>
<td>full</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>call_abricate</strong></td>
<td>Boolean</td>
<td>Set to true to enable the Abricate task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>call_ani</strong></td>
<td>Boolean</td>
<td>Set to true to enable the ANI task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>call_arln_stats</strong></td>
<td>Boolean</td>
<td>Set to true to enable the return of ARLN required Auto PASS/FAIL outputs</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>call_gamma</strong></td>
<td>Boolean</td>
<td>Set to true in order to enable GAMMA task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>call_kmerfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the kmerfinder task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>call_plasmidfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the plasmidfinder task</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>call_resfinder</strong></td>
<td>Boolean</td>
<td>Set to true to enable the ResFinder task</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>city</strong></td>
<td>String</td>
<td>Will be used in the "city" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>collection_date</strong></td>
<td>String</td>
<td>Will be used in the "collection_date" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>county</strong></td>
<td>String</td>
<td>Will be used in the "county" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>expected_taxon</strong></td>
<td>String</td>
<td>If provided, this input will override the taxonomic assignment made by GAMBIT and launch the relevant taxon-specific submodules. It will also modify the organism flag used by AMRFinderPlus. Example format: "Salmonella enterica"</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>genome_annotation</strong></td>
<td>String</td>
<td>If set to "bakta", TheiaProk will use Bakta rather than Prokka to annotate the genome</td>
<td>prokka</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>mlst_run_secondary_scheme</strong></td>
<td>Boolean</td>
<td>If true, will run secondary scheme if primary scheme is of ecoli or abaumannii, these two have multiple schemes that are relevant.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>mlst_scheme_override</strong></td>
<td>Boolean</td>
<td>If true, will force E. coli scheme to be used when Gambit predicts Escherichia coli, otherwise will return scheme MLST predicts.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>originating_lab</strong></td>
<td>String</td>
<td>Will be used in the "originating_lab" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>perform_characterization</strong></td>
<td>Boolean</td>
<td>Set to "false" if you want to only generate an assembly and relevant QC metrics and skip all characterization tasks</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>qc_check_table</strong></td>
<td>File</td>
<td>TSV value with taxons for rows and QC values for columns; internal cells represent user-determined QC thresholds; if provided, turns on the QC Check task. See below for an example QC Check table.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>run_id</strong></td>
<td>String</td>
<td>Will be used in the "run_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>seq_method</strong></td>
<td>String</td>
<td>The sequencing methodology used to generate the input read data; for TheiaProk workflows, this input will be used in the "seq_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>taxon_tables</strong></td>
<td>File</td>
<td>File indicating data table names to copy samples of a particular taxon to</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>terra_project</strong></td>
<td>String</td>
<td>The name of the Terra Project where you want the taxon tables written to</td>
<td>NA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>terra_workspace</strong></td>
<td>String</td>
<td>The name of the Terra Workspace where you want the taxon tables written to</td>
<td>NA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>zip</strong></td>
<td>String</td>
<td>Will be used in the "zip" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mlst:2.23.0-2024-12-31</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>min_percent_coverage</strong></td>
<td>Float</td>
<td>Minimum % breadth of coverage to report an MLST allele</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>min_percent_identity</strong></td>
<td>Float</td>
<td>Minimum % identity to known MLST gene to report an MLST allele</td>
<td>95</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>minscore</strong></td>
<td>Float</td>
<td>Minimum score <a href="https://github.com/tseemann/mlst#scoring-system">https://github.com/tseemann/mlst#scoring-system</a> to assign an MLST profile</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>nopath</strong></td>
<td>Boolean</td>
<td>true = use mlst --nopath. If set to false, filename paths are not stripped from FILE column in output TSV</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>scheme</strong></td>
<td>String</td>
<td>Dont autodetect the MLST scheme; force this scheme on all inputs (see <a href="https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab">https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab</a> for accepted strings)</td>
<td>None</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<h3 id="core-tasks">Core Tasks<a class="headerlink" href="#core-tasks" title="Permanent link"></a></h3>
<div class="admonition dna">
<p>These tasks are performed regardless of organism. They include tasks that are performed regardless of and specific for the input data type. They perform read trimming and assembly appropriate to the input data type.</p>
</div>
<details class="task">
<summary><code>versioning</code>: Version Capture</summary>
<p>The <code>versioning</code> task captures the workflow version from the GitHub (code repository) version.</p>
<div class="admonition techdetails">
<p class="admonition-title">Version Capture Technical details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/task_versioning.wdl">task_versioning.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h4 id="assembly-tasks">Assembly Tasks<a class="headerlink" href="#assembly-tasks" title="Permanent link"></a></h4>
<div class="admonition caption">
<div class="tabbed-set tabbed-alternate" data-tabs="4:4"><input checked="checked" id="theiaprok_illumina_pe_3" name="__tabbed_4" type="radio"><input id="theiaprok_illumina_se_2" name="__tabbed_4" type="radio"><input id="theiaprok_ont_3" name="__tabbed_4" type="radio"><input id="theiaprok_fasta_2" name="__tabbed_4" type="radio"><div class="tabbed-labels"><label for="theiaprok_illumina_pe_3">TheiaProk_Illumina_PE</label><label for="theiaprok_illumina_se_2">TheiaProk_Illumina_SE</label><label for="theiaprok_ont_3">TheiaProk_ONT</label><label for="theiaprok_fasta_2">TheiaProk_FASTA</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<details class="task">
<summary><code>concatenate_illumina_lanes</code>: Concatenate Multi-Lane Illumina FASTQs</summary>
<p>The <code>concatenate_illumina_lanes</code> task concatenates Illumina FASTQ files from multiple lanes into a single file. This task only runs if the <code>read1_lane2</code> input file has been provided. All read1 lanes are concatenated together and are used in subsequent tasks, as are the read2 lanes if applicable. These concatenated files are also provided as output.</p>
<div class="admonition techdetails">
<p class="admonition-title">Concatenate Illumina Lanes Technical Details</p>
<p>The <code>concatenate_illumina_lanes</code> task is run before any downstream steps take place.</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/file_handling/wf_concatenate_illumina_lanes.wdl">wf_concatenate_illumina_lanes.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>screen</code>: Total Raw Read Quantification and Genome Size Estimation</summary>
<p>The <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl"><code>screen</code></a> task ensures the quantity of sequence data is sufficient to undertake genomic analysis. It uses <a href="https://github.com/rpetit3/fastq-scan"><code>fastq-scan</code></a> and bash commands for quantification of reads and base pairs, and <a href="https://mash.readthedocs.io/en/latest/index.html">mash</a> sketching to estimate the genome size and its coverage. At each step, the results are assessed relative to pass/fail criteria and thresholds that may be defined by optional user inputs. Samples are run through all threshold checks, regardless of failures, and the workflow will terminate after the <code>screen</code> task if any thresholds are not met:</p>
<ol>
<li>Total number of reads: A sample will fail the read screening task if its total number of reads is less than or equal to <code>min_reads</code>.</li>
<li>The proportion of basepairs reads in the forward and reverse read files: A sample will fail the read screening if fewer than <code>min_proportion</code> basepairs are in either the reads1 or read2 files.</li>
<li>Number of basepairs: A sample will fail the read screening if there are fewer than <code>min_basepairs</code> basepairs</li>
<li>Estimated genome size:  A sample will fail the read screening if the estimated genome size is smaller than <code>min_genome_size</code> or bigger than <code>max_genome_size</code>.</li>
<li>Estimated genome coverage: A sample will fail the read screening if the estimated genome coverage is less than the <code>min_coverage</code>.</li>
</ol>
<p>Read screening is undertaken on both the raw and cleaned reads. The task may be skipped by setting the <code>skip_screen</code> variable to true.</p>
<p>Default values vary between the PE, SE, and ONT workflows. The rationale for these default values can be found below. If two default values are shown, the first is for Illumina workflows and the second is for ONT.</p>
<table>
<thead>
<tr>
<th>Variable</th>
<th>Default Value</th>
<th>Rationale</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>skip_screen</code></td>
<td>false</td>
<td>Set to true to skip the read screen from running</td>
</tr>
<tr>
<td><code>min_reads</code></td>
<td>7472 or 5000</td>
<td>Calculated from the minimum number of base pairs required for 20x coverage of the Nasuia deltocephalinicola genome, the smallest known bacterial genome as of 2019-08-07 (112,091 bp), divided by 300 (the longest Illumina read length) or 5000 (estimate of ONT read length)</td>
</tr>
<tr>
<td><code>min_basepairs</code></td>
<td>2241820</td>
<td>Should be greater than 20x coverage of Nasuia deltocephalinicola, the smallest known bacterial genome (112,091 bp)</td>
</tr>
<tr>
<td><code>min_genome_length</code></td>
<td>100000</td>
<td>Based on the Nasuia deltocephalinicola genome, the smallest known bacterial genome (112,091 bp)</td>
</tr>
<tr>
<td><code>max_genome_length</code></td>
<td>18040666</td>
<td>Based on the Minicystis rosea genome, the largest known bacterial genome (16,040,666 bp), plus an additional 2 Mbp to cater for potential extra genomic material</td>
</tr>
<tr>
<td><code>min_coverage</code></td>
<td>10 or 5</td>
<td>A bare-minimum average per base coverage across the genome required for genome characterization. Higher coverage would be required for high-quality phylogenetics.</td>
</tr>
<tr>
<td><code>min_proportion</code></td>
<td>40</td>
<td>Neither read1 nor read2 files should have less than 40% of the total number of reads. For paired-end data only.</td>
</tr>
</tbody>
</table>
<div class="admonition techdetails">
<p class="admonition-title">Screen Technical Details</p>
<p>There is a single WDL task for read screening. The <code>screen</code> task is run twice, once for raw reads and once for clean reads.</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L3">task_screen.wdl (PE sub-task)</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L147">task_screen.wdl (SE sub-task)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>These tasks clean then assemble the reads into a <em>de novo</em> assembly and assess the quality of the assembly.</p>
</div>
<details class="task">
<summary><code>read_QC_trim</code>: Read Quality Trimming, Adapter Removal, Quantification, and Identification</summary>
<p><code>read_QC_trim</code> is a sub-workflow that removes low-quality reads, low-quality regions of reads, and sequencing adapters to improve data quality. It uses a number of tasks, described below. The differences between the PE and SE versions of the <code>read_QC_trim</code> sub-workflow lie in the default parameters, the use of two or one input read file(s), and the different output files.</p>
<div class="admonition dna">
<p>By default, <code>read_processing</code> is set to <code>"trimmomatic"</code>. To use <code>fastp</code> instead, set <code>read_processing</code> to <code>"fastp"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>Trimmomatic</code>: Read Trimming (default)</summary>
<p>Read proccessing is available via <code>Trimmomatic</code> by default.</p>
<p>Trimmomatic trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>Trimmomatic</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_trimmomatic.wdl">task_trimmomatic.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/usadellab/Trimmomatic">Trimmomatic on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic Website</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590/">Trimmomatic: a flexible trimmer for Illumina sequence data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>fastp</code>: Read Trimming (alternative)</summary>
<p>To activate this task, set <code>read_processing</code> to <code>"fastp"</code>.</p>
<p><code>fastp</code> trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<p><code>fastp</code> also has additional default parameters and features that are not a part of <code>trimmomatic</code>'s default configuration.</p>
<details class="toggle">
<summary><code>fastp</code> default read-trimming parameters</summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-g</td>
<td>enables polyG tail trimming</td>
</tr>
<tr>
<td>-5 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>-3 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>--detect_adapter_for_pe</td>
<td>enables adapter-trimming <strong>only for paired-end reads</strong></td>
</tr>
</tbody>
</table>
<p>Additional arguments can be passed using the <code>fastp_args</code> optional parameter.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Trimmomatic and fastp Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_fastp.wdl">task_fastp.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/17/i884/5093234?login=false">fastp: an ultra-fast all-in-one FASTQ preprocessor</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>BBDuk</code>: Adapter Trimming and PhiX Removal</summary>
<p>Adapters are manufactured oligonucleotide sequences attached to DNA fragments during the library preparation process. In Illumina sequencing, these adapter sequences are required for attaching reads to flow cells. You can read more about <a href="https://emea.support.illumina.com/bulletins/2020/06/illumina-adapter-portfolio.html">Illumina adapters here</a>. For genome analysis, it's important to remove these sequences since they're not actually from your sample. If you don't remove them, the downstream analysis may be affected.</p>
<p>The <code>bbduk</code> task removes adapters from sequence reads. To do this:</p>
<ul>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/repair-guide/">Repair</a> from the <a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/">BBTools</a> package reorders reads in paired fastq files to ensure the forward and reverse reads of a pair are in the same position in the two fastq files (it <em>re-pairs</em>).</li>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk</a>  (<em>"Bestus Bioinformaticus" Decontamination Using Kmers</em>) is then used to trim the adapters and filter out all reads that have a 31-mer match to <a href="https://emea.illumina.com/products/by-type/sequencing-kits/cluster-gen-sequencing-reagents/phix-control-v3.html">PhiX</a>, which is commonly added to Illumina sequencing runs to monitor and/or improve overall run quality.</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">BBDuk Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_bbduk.wdl">task_bbduk.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://sourceforge.net/projects/bbmap/">BBMap on SourceForge</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk Guide (archived)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_qc</code> is set to <code>"fastq_scan"</code>. To use <code>fastqc</code> instead, set <code>read_qc</code> to <code>"fastqc"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>fastq-scan</code>: Read Quantification (default)</summary>
<p>Read quantification is available via <code>fastq-scan</code> by default.</p>
<p><code>fastq-scan</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>fastq-scan</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastq_scan.wdl">task_fastq_scan.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/fastq-scan/blob/master/README.md">fastq-scan on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>FastQC</code>: Read Quantification (alternative)</summary>
<p>To activate this task, set <code>read_qc</code> to <code>"fastqc"</code>.</p>
<p><code>FastQC</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>This tool also provides a graphical visualization of the read quality.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>FastQC</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastqc.wdl">task_fastqc.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/s-andrews/FastQC">FastQC on Github</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC Website</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>MIDAS</code>: Read Identification (optional)</summary>
<p>To activate this task, set <code>call_midas</code> to <code>true</code>.</p>
<p>The <code>MIDAS</code> task is for the identification of reads to detect contamination with non-target taxa.</p>
<p>The MIDAS tool was originally designed for metagenomic sequencing data but has been co-opted for use with bacterial isolate WGS methods. It can be used to detect contamination present in raw sequencing data by estimating bacterial species abundance in bacterial isolate WGS data. If a secondary genus is detected above a relative frequency of 0.01 (1%), then the sample should fail QC and be investigated further for potential contamination.</p>
<p>This task is similar to those used in commercial software, BioNumerics, for estimating secondary species abundance.</p>
<details class="toggle">
<summary>How are the MIDAS output columns determined?</summary>
<p>Example MIDAS report in the <code>midas_report</code> column:</p>
<table>
<thead>
<tr>
<th>species_id</th>
<th>count_reads</th>
<th>coverage</th>
<th>relative_abundance</th>
</tr>
</thead>
<tbody>
<tr>
<td>Salmonella_enterica_58156</td>
<td>3309</td>
<td>89.88006645</td>
<td>0.855888033</td>
</tr>
<tr>
<td>Salmonella_enterica_58266</td>
<td>501</td>
<td>11.60606061</td>
<td>0.110519371</td>
</tr>
<tr>
<td>Salmonella_enterica_53987</td>
<td>99</td>
<td>2.232896237</td>
<td>0.021262881</td>
</tr>
<tr>
<td>Citrobacter_youngae_61659</td>
<td>46</td>
<td>0.995216227</td>
<td>0.009477003</td>
</tr>
<tr>
<td>Escherichia_coli_58110</td>
<td>5</td>
<td>0.123668877</td>
<td>0.001177644</td>
</tr>
</tbody>
</table>
<p>MIDAS report column descriptions:</p>
<ul>
<li>species_id: species identifier</li>
<li>count_reads: number of reads mapped to marker genes</li>
<li>coverage: estimated genome-coverage (i.e. read-depth) of species in metagenome</li>
<li>relative_abundance: estimated relative abundance of species in metagenome</li>
</ul>
<p>The value in the <code>midas_primary_genus</code> column is derived by ordering the rows in order of "relative_abundance" and identifying the genus of top species in the "species_id" column (Salmonella). The value in the <code>midas_secondary_genus</code> column is derived from the genus of the second-most prevalent genus in the "species_id" column (Citrobacter). The <code>midas_secondary_genus_abundance</code> column is the "relative_abundance" of the second-most prevalent genus (0.009477003). The <code>midas_secondary_genus_coverage</code> is the "coverage" of the second-most prevalent genus (0.995216227).</p>
</details>
<p><strong>MIDAS Reference Database Overview</strong></p>
<p>The <strong>MIDAS reference database</strong> is a comprehensive tool for identifying bacterial species in metagenomic and bacterial isolate WGS data. It includes several layers of genomic data, helping detect species abundance and potential contaminants.</p>
<div class="admonition dna">
<p class="admonition-title">Key Components of the MIDAS Database</p>
<ol>
<li>
<p><strong>Species Groups</strong>: </p>
<ul>
<li>MIDAS clusters bacterial genomes based on 96.5% sequence identity, forming over 5,950 species groups from 31,007 genomes. These groups align with the gold-standard species definition (95% ANI), ensuring highly accurate species identification.</li>
</ul>
</li>
<li>
<p><strong>Genomic Data Structure</strong>:</p>
<ul>
<li><em>Marker Genes</em>: Contains 15 universal single-copy genes used to estimate species abundance.</li>
<li><em>Representative Genome</em>: Each species group has a selected representative genome, which minimizes genetic variation and aids in accurate SNP identification.</li>
<li><em>Pan-genome</em>: The database includes clusters of non-redundant genes, with options for multi-level clustering (e.g., 99%, 95%, 90% identity), enabling MIDAS to identify gene content within strains at various clustering thresholds.</li>
</ul>
</li>
<li>
<p><strong>Taxonomic Annotation</strong>: </p>
<ul>
<li>Genomes are annotated based on consensus Latin names. Discrepancies in name assignments may occur due to factors like unclassified genomes or genus-level ambiguities.</li>
</ul>
</li>
</ol>
</div>
<hr>
<p><strong>Using the Default MIDAS Database</strong></p>
<p>TheiaProk and TheiaEuk use the pre-loaded MIDAS database in Terra (see input table for current version) by default for bacterial species detection in metagenomic data, requiring no additional setup.</p>
<div class="admonition tip">
<p class="admonition-title">Create a Custom MIDAS Database</p>
<p>Users can also build their own custom MIDAS database if they want to include specific genomes or configurations. This custom database can replace the default MIDAS database used in Terra. To build a custom MIDAS database, follow the <a href="https://github.com/snayfach/MIDAS/blob/master/docs/build_db.md">MIDAS GitHub guide on building a custom database</a>. Once the database is built, users can upload it to a Google Cloud Storage bucket or Terra workkspace and provide the link to the database in the <code>midas_db</code> input variable.</p>
</div>
<hr>
<div class="admonition techdetails">
<p class="admonition-title">MIDAS Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_midas.wdl">task_midas.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/snayfach/MIDAS">MIDAS on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/snayfach/MIDAS">MIDAS on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/27803195">An integrated metagenomics pipeline for strain profiling reveals novel patterns of bacterial transmission and biogeography</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Kraken2</code>: Read Identification (optional)</summary>
<p>To activate this task, set <code>call_kraken</code> to <code>true</code> and provide a value for <code>kraken_db</code>.</p>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<p>As an alternative to <code>MIDAS</code> (see above), the <code>Kraken2</code> task can also be turned on through setting the <code>call_kraken</code> input variable as <code>true</code> for the identification of reads to detect contamination with non-target taxa.</p>
<p>A database must be provided if this optional module is activated, through the kraken_db optional input. A list of suggested databases can be found on <a href="../../standalone/kraken2/#databases">Kraken2 standalone documentation</a>.</p>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">Kraken2 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_pe.wdl">wf_read_QC_trim_pe.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_se.wdl">wf_read_QC_trim_se.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>CG-Pipeline</code>: Assessment of Read Quality, and Estimation of Genome Coverage</summary>
<p>The<code>cg_pipeline</code> task generates metrics about read quality and estimates the coverage of the genome using the <code>run_assembly_readMetrics.pl</code> script from <a href="https://github.com/lskatz/CG-Pipeline/">CG-Pipeline</a>. The genome coverage estimates are calculated using both using raw and cleaned reads, using either a user-provided <code>genome_size</code> or the estimated genome length generated by QUAST.</p>
<div class="admonition techdetails">
<p class="admonition-title">CG-Pipeline Technical Details</p>
<p>The <code>cg_pipeline</code> task is run twice in this workflow, once with raw reads, and once with clean reads.</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_cg_pipeline.wdl">task_cg_pipeline.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lskatz/CG-Pipeline/">CG-Pipeline on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/lskatz/CG-Pipeline/">CG-Pipeline on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/26/15/1819/188418">A computational genomics pipeline for prokaryotic sequencing projects</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>digger_denovo</code>: <em>De novo</em> Assembly</summary>
<p><em>De novo</em>  assembly is the process or product of attempting to reconstruct a genome from scratch (without prior knowledge of the genome) using sequence reads. Assembly of fungal genomes from short-reads will produce multiple contigs per chromosome rather than a single contiguous sequence for each chromosome.</p>
<p>In TheiaProk and TheiaEuk Illumina workflows, <em>de novo</em> assembly is performed for samples that have sufficient read quantity and quality using <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_digger_denovo.wdl">digger_denovo</a>, a subworkflow based off of <a href="https://github.com/tseemann/shovill">Shovill</a> pipeline. The name "digger" is a nod to Shovill and SPAdes.</p>
<details class="toggle">
<summary><em>De novo</em> Assembly</summary>
<div class="admonition dna">
<p class="admonition-title"><code>assembler</code> with <code>skesa</code> (default), <code>spades</code>, or <code>megahit</code></p>
<p>To activate a particular assembler, set the <code>assembler</code> input parameter to either <code>skesa</code> (default), <code>spades</code>, or <code>megahit</code>.</p>
<p>These tasks are mutually exclusive.</p>
</div>
<details class="task">
<summary><code>SKESA</code>: <em>De novo</em> Assembly (default)</summary>
<p>This task is activated by default.</p>
<p><code>SKESA</code> (Strategic K-mer Extension for Scrupulous Assemblies) is a <em>de novo</em> assembler that is fairly conservative and introduces breaks in the genome at repeat regions. This leads to higher sequence quality but more fragmented assemblies, which, depending on the final analysis goal, can be either highly preferred or detrimental. Designed for Illumina reads and haploid genomes, SKESA is the default assembler in the <code>digger_denovo</code> subworkflow.</p>
<div class="admonition techdetails">
<p class="admonition-title">SKESA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_skesa.wdl">task_skesa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/SKESA">SKESA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/SKESA">SKESA on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1186/s13059-018-1540-z">SKESA: strategic k-mer externsion for scrupulous assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>SPAdes</code>: <em>De novo</em> Assembly (alternative)</summary>
<p>To activate this task, set <code>assembler</code> to <code>spades</code>.</p>
<p><code>SPAdes</code> (St. Petersburg genome assembler) is a <em>de novo</em> assembly tool that uses de Bruijn graphs to assemble genomes from Illumina short reads.</p>
<p>In TheiaProk, SPAdes is run in <code>--isolate</code> mode, which is the recommended flag for high-coverage isolate and multi-cell Illumina data, which is typical of most bacterial sequencing projects. This method is optimized for improving assembly quality and decreasing runtime.</p>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">MetaviralSPAdes Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_spades.wdl">task_spades.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ablab/spades">SPAdes on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ablab.github.io/spades/index.html">SPAdes Manual</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>TheiaProk</em>: <a href="https://doi.org/10.1089/cmb.2012.0021">SPAdes: A New Genome Assembly Algorithm and Its Applications to Single-Cell Sequencing</a><br><em>TheiaViral</em>: <a href="https://doi.org/10.1093/bioinformatics/btaa490">MetaviralSPAdes: assembly of viruses from metagenomic data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>MEGAHIT</code>: <em>De novo</em> Assembly (alternative)</summary>
<p>To activate this task, set <code>assembler</code> to <code>megahit</code>.</p>
<p>The <a href="https://github.com/voutcn/megahit">MEGAHIT assembler</a> is a fast and memory-efficient <em>de novo</em> assembler that can handle large datasets. While optimized for metagenomics, MEGAHIT also performs well on single-genome assemblies, making it a versatile choice for various assembly tasks.</p>
<p>MEGAHIT uses a multiple k-mer strategy that can be beneficial for assembling genomes with varying coverage levels, which is common in metagenomic samples. It constructs succinct de Bruijn graphs to efficiently represent the assembly process, allowing it to handle large and complex datasets with reduced memory usage.</p>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">MEGAHIT Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_megahit.wdl">task_megahit.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/voutcn/megahit">MEGAHIT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/voutcn/megahit/blob/master/README.md">MEGAHIT on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btv033">MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Assembly Polishing (optional)</summary>
<p>To activate assembly polishing, set <code>call_pilon</code> to <code>true</code>.</p>
<details class="task">
<summary><code>bwa</code>: Read Alignment to the Assembly</summary>
<p>BWA (Burrow-Wheeler Aligner) is used to align the cleaned read files to generated assembly file in order to generate an alignment. The resulting BAM file is directly passed to the Pilon task to polish the assembly for errors.</p>
<div class="admonition techdetails">
<p class="admonition-title">BWA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_bwa.wdl">task_bwa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/bwa">BWA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bio-bwa.sourceforge.net/">BWA Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and accurate short read alignment with Burrows-Wheeler transform</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Pilon</code>: Assembly Polishing</summary>
<p><code>Pilon</code> is a tool that uses read alignments to correct errors in an assembly.</p>
<p>The <code>bwa</code>-generated alignment of the read data to the assembly is used to identify inconsistences between the reads and the assembly in order to correct them. <code>Pilon</code> will attempt to fix individual base errors and small indels using the read data. This can improve the overall quality of the assembly, especially when the assembler has made mistakes due to sequencing errors or low coverage regions.</p>
<p>The default parameters were set to mimic the parameters used by <a href="https://github.com/tseemann/shovill">Shovill</a>: <code>--fix bases --minq 60 --minqual 3 --mindepth 0.25</code>. These can be modified by the user.</p>
<div class="admonition techdetails">
<p class="admonition-title">Pilon Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_pilon.wdl">task_pilon.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/broadinstitute/pilon">Pilon on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/broadinstitute/pilon/wiki">Pilon Wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1371/journal.pone.0112963">Pilon: An Integrated Tool for Comprehensive Microbial Variant Detection and Genome Assembly Improvement</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Contig Filtering (optional)</summary>
<details class="task">
<summary><code>Filter Contigs</code>: Contig Quality Control</summary>
<p>To <strong><em>de</em></strong>activate contig filtering, set <code>run_filter_contigs</code> to <code>false</code>.</p>
<p>This task filters the created contigs based on a default minimum length threshold of 200 bp and a minimum coverage of 2.0. It also eliminates homopolymer contigs (contigs of any length that consist of a single nucleotide).</p>
<p>Options are available to skip any of these filters by setting the respective parameters to <code>false</code>: <code>filter_contigs_skip_length_filter</code>, <code>filter_contigs_skip_coverage_filter</code>, and <code>filter_contigs_skip_homopolymer_filter</code>. The minimum length and coverage thresholds can be adjusted using the <code>filter_contigs_min_length</code> and <code>filter_contigs_min_coverage</code> parameters, respectively.</p>
<p>This ensures high-quality assemblies by retaining only contigs that meet specified criteria. Detailed metrics on contig counts and sequence lengths before and after filtering are provided in the output.</p>
<div class="admonition techdetails">
<p class="admonition-title">Filter Contigs Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_filter_contigs.wdl">task_filter_contigs.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Digger-Denovo Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_digger_denovo.wdl">wf_digger_denovo.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>concatenate_illumina_lanes</code>: Concatenate Multi-Lane Illumina FASTQs</summary>
<p>The <code>concatenate_illumina_lanes</code> task concatenates Illumina FASTQ files from multiple lanes into a single file. This task only runs if the <code>read1_lane2</code> input file has been provided. All read1 lanes are concatenated together and are used in subsequent tasks, as are the read2 lanes if applicable. These concatenated files are also provided as output.</p>
<div class="admonition techdetails">
<p class="admonition-title">Concatenate Illumina Lanes Technical Details</p>
<p>The <code>concatenate_illumina_lanes</code> task is run before any downstream steps take place.</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/file_handling/wf_concatenate_illumina_lanes.wdl">wf_concatenate_illumina_lanes.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>screen</code>: Total Raw Read Quantification and Genome Size Estimation</summary>
<p>The <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl"><code>screen</code></a> task ensures the quantity of sequence data is sufficient to undertake genomic analysis. It uses <a href="https://github.com/rpetit3/fastq-scan"><code>fastq-scan</code></a> and bash commands for quantification of reads and base pairs, and <a href="https://mash.readthedocs.io/en/latest/index.html">mash</a> sketching to estimate the genome size and its coverage. At each step, the results are assessed relative to pass/fail criteria and thresholds that may be defined by optional user inputs. Samples are run through all threshold checks, regardless of failures, and the workflow will terminate after the <code>screen</code> task if any thresholds are not met:</p>
<ol>
<li>Total number of reads: A sample will fail the read screening task if its total number of reads is less than or equal to <code>min_reads</code>.</li>
<li>The proportion of basepairs reads in the forward and reverse read files: A sample will fail the read screening if fewer than <code>min_proportion</code> basepairs are in either the reads1 or read2 files.</li>
<li>Number of basepairs: A sample will fail the read screening if there are fewer than <code>min_basepairs</code> basepairs</li>
<li>Estimated genome size:  A sample will fail the read screening if the estimated genome size is smaller than <code>min_genome_size</code> or bigger than <code>max_genome_size</code>.</li>
<li>Estimated genome coverage: A sample will fail the read screening if the estimated genome coverage is less than the <code>min_coverage</code>.</li>
</ol>
<p>Read screening is undertaken on both the raw and cleaned reads. The task may be skipped by setting the <code>skip_screen</code> variable to true.</p>
<p>Default values vary between the PE, SE, and ONT workflows. The rationale for these default values can be found below. If two default values are shown, the first is for Illumina workflows and the second is for ONT.</p>
<table>
<thead>
<tr>
<th>Variable</th>
<th>Default Value</th>
<th>Rationale</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>skip_screen</code></td>
<td>false</td>
<td>Set to true to skip the read screen from running</td>
</tr>
<tr>
<td><code>min_reads</code></td>
<td>7472 or 5000</td>
<td>Calculated from the minimum number of base pairs required for 20x coverage of the Nasuia deltocephalinicola genome, the smallest known bacterial genome as of 2019-08-07 (112,091 bp), divided by 300 (the longest Illumina read length) or 5000 (estimate of ONT read length)</td>
</tr>
<tr>
<td><code>min_basepairs</code></td>
<td>2241820</td>
<td>Should be greater than 20x coverage of Nasuia deltocephalinicola, the smallest known bacterial genome (112,091 bp)</td>
</tr>
<tr>
<td><code>min_genome_length</code></td>
<td>100000</td>
<td>Based on the Nasuia deltocephalinicola genome, the smallest known bacterial genome (112,091 bp)</td>
</tr>
<tr>
<td><code>max_genome_length</code></td>
<td>18040666</td>
<td>Based on the Minicystis rosea genome, the largest known bacterial genome (16,040,666 bp), plus an additional 2 Mbp to cater for potential extra genomic material</td>
</tr>
<tr>
<td><code>min_coverage</code></td>
<td>10 or 5</td>
<td>A bare-minimum average per base coverage across the genome required for genome characterization. Higher coverage would be required for high-quality phylogenetics.</td>
</tr>
<tr>
<td><code>min_proportion</code></td>
<td>40</td>
<td>Neither read1 nor read2 files should have less than 40% of the total number of reads. For paired-end data only.</td>
</tr>
</tbody>
</table>
<div class="admonition techdetails">
<p class="admonition-title">Screen Technical Details</p>
<p>There is a single WDL task for read screening. The <code>screen</code> task is run twice, once for raw reads and once for clean reads.</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L3">task_screen.wdl (PE sub-task)</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L147">task_screen.wdl (SE sub-task)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>These tasks clean then assemble the reads into a <em>de novo</em> assembly and assess the quality of the assembly.</p>
</div>
<details class="task">
<summary><code>read_QC_trim</code>: Read Quality Trimming, Adapter Removal, Quantification, and Identification</summary>
<p><code>read_QC_trim</code> is a sub-workflow that removes low-quality reads, low-quality regions of reads, and sequencing adapters to improve data quality. It uses a number of tasks, described below. The differences between the PE and SE versions of the <code>read_QC_trim</code> sub-workflow lie in the default parameters, the use of two or one input read file(s), and the different output files.</p>
<div class="admonition dna">
<p>By default, <code>read_processing</code> is set to <code>"trimmomatic"</code>. To use <code>fastp</code> instead, set <code>read_processing</code> to <code>"fastp"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>Trimmomatic</code>: Read Trimming (default)</summary>
<p>Read proccessing is available via <code>Trimmomatic</code> by default.</p>
<p>Trimmomatic trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>Trimmomatic</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_trimmomatic.wdl">task_trimmomatic.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/usadellab/Trimmomatic">Trimmomatic on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic Website</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590/">Trimmomatic: a flexible trimmer for Illumina sequence data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>fastp</code>: Read Trimming (alternative)</summary>
<p>To activate this task, set <code>read_processing</code> to <code>"fastp"</code>.</p>
<p><code>fastp</code> trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<p><code>fastp</code> also has additional default parameters and features that are not a part of <code>trimmomatic</code>'s default configuration.</p>
<details class="toggle">
<summary><code>fastp</code> default read-trimming parameters</summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-g</td>
<td>enables polyG tail trimming</td>
</tr>
<tr>
<td>-5 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>-3 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>--detect_adapter_for_pe</td>
<td>enables adapter-trimming <strong>only for paired-end reads</strong></td>
</tr>
</tbody>
</table>
<p>Additional arguments can be passed using the <code>fastp_args</code> optional parameter.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Trimmomatic and fastp Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_fastp.wdl">task_fastp.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/17/i884/5093234?login=false">fastp: an ultra-fast all-in-one FASTQ preprocessor</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>BBDuk</code>: Adapter Trimming and PhiX Removal</summary>
<p>Adapters are manufactured oligonucleotide sequences attached to DNA fragments during the library preparation process. In Illumina sequencing, these adapter sequences are required for attaching reads to flow cells. You can read more about <a href="https://emea.support.illumina.com/bulletins/2020/06/illumina-adapter-portfolio.html">Illumina adapters here</a>. For genome analysis, it's important to remove these sequences since they're not actually from your sample. If you don't remove them, the downstream analysis may be affected.</p>
<p>The <code>bbduk</code> task removes adapters from sequence reads. To do this:</p>
<ul>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/repair-guide/">Repair</a> from the <a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/">BBTools</a> package reorders reads in paired fastq files to ensure the forward and reverse reads of a pair are in the same position in the two fastq files (it <em>re-pairs</em>).</li>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk</a>  (<em>"Bestus Bioinformaticus" Decontamination Using Kmers</em>) is then used to trim the adapters and filter out all reads that have a 31-mer match to <a href="https://emea.illumina.com/products/by-type/sequencing-kits/cluster-gen-sequencing-reagents/phix-control-v3.html">PhiX</a>, which is commonly added to Illumina sequencing runs to monitor and/or improve overall run quality.</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">BBDuk Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_bbduk.wdl">task_bbduk.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://sourceforge.net/projects/bbmap/">BBMap on SourceForge</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk Guide (archived)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_qc</code> is set to <code>"fastq_scan"</code>. To use <code>fastqc</code> instead, set <code>read_qc</code> to <code>"fastqc"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>fastq-scan</code>: Read Quantification (default)</summary>
<p>Read quantification is available via <code>fastq-scan</code> by default.</p>
<p><code>fastq-scan</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>fastq-scan</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastq_scan.wdl">task_fastq_scan.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/fastq-scan/blob/master/README.md">fastq-scan on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>FastQC</code>: Read Quantification (alternative)</summary>
<p>To activate this task, set <code>read_qc</code> to <code>"fastqc"</code>.</p>
<p><code>FastQC</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>This tool also provides a graphical visualization of the read quality.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>FastQC</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastqc.wdl">task_fastqc.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/s-andrews/FastQC">FastQC on Github</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC Website</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>MIDAS</code>: Read Identification (optional)</summary>
<p>To activate this task, set <code>call_midas</code> to <code>true</code>.</p>
<p>The <code>MIDAS</code> task is for the identification of reads to detect contamination with non-target taxa.</p>
<p>The MIDAS tool was originally designed for metagenomic sequencing data but has been co-opted for use with bacterial isolate WGS methods. It can be used to detect contamination present in raw sequencing data by estimating bacterial species abundance in bacterial isolate WGS data. If a secondary genus is detected above a relative frequency of 0.01 (1%), then the sample should fail QC and be investigated further for potential contamination.</p>
<p>This task is similar to those used in commercial software, BioNumerics, for estimating secondary species abundance.</p>
<details class="toggle">
<summary>How are the MIDAS output columns determined?</summary>
<p>Example MIDAS report in the <code>midas_report</code> column:</p>
<table>
<thead>
<tr>
<th>species_id</th>
<th>count_reads</th>
<th>coverage</th>
<th>relative_abundance</th>
</tr>
</thead>
<tbody>
<tr>
<td>Salmonella_enterica_58156</td>
<td>3309</td>
<td>89.88006645</td>
<td>0.855888033</td>
</tr>
<tr>
<td>Salmonella_enterica_58266</td>
<td>501</td>
<td>11.60606061</td>
<td>0.110519371</td>
</tr>
<tr>
<td>Salmonella_enterica_53987</td>
<td>99</td>
<td>2.232896237</td>
<td>0.021262881</td>
</tr>
<tr>
<td>Citrobacter_youngae_61659</td>
<td>46</td>
<td>0.995216227</td>
<td>0.009477003</td>
</tr>
<tr>
<td>Escherichia_coli_58110</td>
<td>5</td>
<td>0.123668877</td>
<td>0.001177644</td>
</tr>
</tbody>
</table>
<p>MIDAS report column descriptions:</p>
<ul>
<li>species_id: species identifier</li>
<li>count_reads: number of reads mapped to marker genes</li>
<li>coverage: estimated genome-coverage (i.e. read-depth) of species in metagenome</li>
<li>relative_abundance: estimated relative abundance of species in metagenome</li>
</ul>
<p>The value in the <code>midas_primary_genus</code> column is derived by ordering the rows in order of "relative_abundance" and identifying the genus of top species in the "species_id" column (Salmonella). The value in the <code>midas_secondary_genus</code> column is derived from the genus of the second-most prevalent genus in the "species_id" column (Citrobacter). The <code>midas_secondary_genus_abundance</code> column is the "relative_abundance" of the second-most prevalent genus (0.009477003). The <code>midas_secondary_genus_coverage</code> is the "coverage" of the second-most prevalent genus (0.995216227).</p>
</details>
<p><strong>MIDAS Reference Database Overview</strong></p>
<p>The <strong>MIDAS reference database</strong> is a comprehensive tool for identifying bacterial species in metagenomic and bacterial isolate WGS data. It includes several layers of genomic data, helping detect species abundance and potential contaminants.</p>
<div class="admonition dna">
<p class="admonition-title">Key Components of the MIDAS Database</p>
<ol>
<li>
<p><strong>Species Groups</strong>: </p>
<ul>
<li>MIDAS clusters bacterial genomes based on 96.5% sequence identity, forming over 5,950 species groups from 31,007 genomes. These groups align with the gold-standard species definition (95% ANI), ensuring highly accurate species identification.</li>
</ul>
</li>
<li>
<p><strong>Genomic Data Structure</strong>:</p>
<ul>
<li><em>Marker Genes</em>: Contains 15 universal single-copy genes used to estimate species abundance.</li>
<li><em>Representative Genome</em>: Each species group has a selected representative genome, which minimizes genetic variation and aids in accurate SNP identification.</li>
<li><em>Pan-genome</em>: The database includes clusters of non-redundant genes, with options for multi-level clustering (e.g., 99%, 95%, 90% identity), enabling MIDAS to identify gene content within strains at various clustering thresholds.</li>
</ul>
</li>
<li>
<p><strong>Taxonomic Annotation</strong>: </p>
<ul>
<li>Genomes are annotated based on consensus Latin names. Discrepancies in name assignments may occur due to factors like unclassified genomes or genus-level ambiguities.</li>
</ul>
</li>
</ol>
</div>
<hr>
<p><strong>Using the Default MIDAS Database</strong></p>
<p>TheiaProk and TheiaEuk use the pre-loaded MIDAS database in Terra (see input table for current version) by default for bacterial species detection in metagenomic data, requiring no additional setup.</p>
<div class="admonition tip">
<p class="admonition-title">Create a Custom MIDAS Database</p>
<p>Users can also build their own custom MIDAS database if they want to include specific genomes or configurations. This custom database can replace the default MIDAS database used in Terra. To build a custom MIDAS database, follow the <a href="https://github.com/snayfach/MIDAS/blob/master/docs/build_db.md">MIDAS GitHub guide on building a custom database</a>. Once the database is built, users can upload it to a Google Cloud Storage bucket or Terra workkspace and provide the link to the database in the <code>midas_db</code> input variable.</p>
</div>
<hr>
<div class="admonition techdetails">
<p class="admonition-title">MIDAS Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_midas.wdl">task_midas.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/snayfach/MIDAS">MIDAS on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/snayfach/MIDAS">MIDAS on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/27803195">An integrated metagenomics pipeline for strain profiling reveals novel patterns of bacterial transmission and biogeography</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Kraken2</code>: Read Identification (optional)</summary>
<p>To activate this task, set <code>call_kraken</code> to <code>true</code> and provide a value for <code>kraken_db</code>.</p>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<p>As an alternative to <code>MIDAS</code> (see above), the <code>Kraken2</code> task can also be turned on through setting the <code>call_kraken</code> input variable as <code>true</code> for the identification of reads to detect contamination with non-target taxa.</p>
<p>A database must be provided if this optional module is activated, through the kraken_db optional input. A list of suggested databases can be found on <a href="../../standalone/kraken2/#databases">Kraken2 standalone documentation</a>.</p>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">Kraken2 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_pe.wdl">wf_read_QC_trim_pe.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_se.wdl">wf_read_QC_trim_se.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>CG-Pipeline</code>: Assessment of Read Quality, and Estimation of Genome Coverage</summary>
<p>The<code>cg_pipeline</code> task generates metrics about read quality and estimates the coverage of the genome using the <code>run_assembly_readMetrics.pl</code> script from <a href="https://github.com/lskatz/CG-Pipeline/">CG-Pipeline</a>. The genome coverage estimates are calculated using both using raw and cleaned reads, using either a user-provided <code>genome_size</code> or the estimated genome length generated by QUAST.</p>
<div class="admonition techdetails">
<p class="admonition-title">CG-Pipeline Technical Details</p>
<p>The <code>cg_pipeline</code> task is run twice in this workflow, once with raw reads, and once with clean reads.</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_cg_pipeline.wdl">task_cg_pipeline.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lskatz/CG-Pipeline/">CG-Pipeline on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/lskatz/CG-Pipeline/">CG-Pipeline on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/26/15/1819/188418">A computational genomics pipeline for prokaryotic sequencing projects</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>digger_denovo</code>: <em>De novo</em> Assembly</summary>
<p><em>De novo</em>  assembly is the process or product of attempting to reconstruct a genome from scratch (without prior knowledge of the genome) using sequence reads. Assembly of fungal genomes from short-reads will produce multiple contigs per chromosome rather than a single contiguous sequence for each chromosome.</p>
<p>In TheiaProk and TheiaEuk Illumina workflows, <em>de novo</em> assembly is performed for samples that have sufficient read quantity and quality using <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_digger_denovo.wdl">digger_denovo</a>, a subworkflow based off of <a href="https://github.com/tseemann/shovill">Shovill</a> pipeline. The name "digger" is a nod to Shovill and SPAdes.</p>
<details class="toggle">
<summary><em>De novo</em> Assembly</summary>
<div class="admonition dna">
<p class="admonition-title"><code>assembler</code> with <code>skesa</code> (default), <code>spades</code>, or <code>megahit</code></p>
<p>To activate a particular assembler, set the <code>assembler</code> input parameter to either <code>skesa</code> (default), <code>spades</code>, or <code>megahit</code>.</p>
<p>These tasks are mutually exclusive.</p>
</div>
<details class="task">
<summary><code>SKESA</code>: <em>De novo</em> Assembly (default)</summary>
<p>This task is activated by default.</p>
<p><code>SKESA</code> (Strategic K-mer Extension for Scrupulous Assemblies) is a <em>de novo</em> assembler that is fairly conservative and introduces breaks in the genome at repeat regions. This leads to higher sequence quality but more fragmented assemblies, which, depending on the final analysis goal, can be either highly preferred or detrimental. Designed for Illumina reads and haploid genomes, SKESA is the default assembler in the <code>digger_denovo</code> subworkflow.</p>
<div class="admonition techdetails">
<p class="admonition-title">SKESA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_skesa.wdl">task_skesa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/SKESA">SKESA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/SKESA">SKESA on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1186/s13059-018-1540-z">SKESA: strategic k-mer externsion for scrupulous assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>SPAdes</code>: <em>De novo</em> Assembly (alternative)</summary>
<p>To activate this task, set <code>assembler</code> to <code>spades</code>.</p>
<p><code>SPAdes</code> (St. Petersburg genome assembler) is a <em>de novo</em> assembly tool that uses de Bruijn graphs to assemble genomes from Illumina short reads.</p>
<p>In TheiaProk, SPAdes is run in <code>--isolate</code> mode, which is the recommended flag for high-coverage isolate and multi-cell Illumina data, which is typical of most bacterial sequencing projects. This method is optimized for improving assembly quality and decreasing runtime.</p>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">MetaviralSPAdes Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_spades.wdl">task_spades.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ablab/spades">SPAdes on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ablab.github.io/spades/index.html">SPAdes Manual</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>TheiaProk</em>: <a href="https://doi.org/10.1089/cmb.2012.0021">SPAdes: A New Genome Assembly Algorithm and Its Applications to Single-Cell Sequencing</a><br><em>TheiaViral</em>: <a href="https://doi.org/10.1093/bioinformatics/btaa490">MetaviralSPAdes: assembly of viruses from metagenomic data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>MEGAHIT</code>: <em>De novo</em> Assembly (alternative)</summary>
<p>To activate this task, set <code>assembler</code> to <code>megahit</code>.</p>
<p>The <a href="https://github.com/voutcn/megahit">MEGAHIT assembler</a> is a fast and memory-efficient <em>de novo</em> assembler that can handle large datasets. While optimized for metagenomics, MEGAHIT also performs well on single-genome assemblies, making it a versatile choice for various assembly tasks.</p>
<p>MEGAHIT uses a multiple k-mer strategy that can be beneficial for assembling genomes with varying coverage levels, which is common in metagenomic samples. It constructs succinct de Bruijn graphs to efficiently represent the assembly process, allowing it to handle large and complex datasets with reduced memory usage.</p>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">MEGAHIT Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_megahit.wdl">task_megahit.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/voutcn/megahit">MEGAHIT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/voutcn/megahit/blob/master/README.md">MEGAHIT on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btv033">MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Assembly Polishing (optional)</summary>
<p>To activate assembly polishing, set <code>call_pilon</code> to <code>true</code>.</p>
<details class="task">
<summary><code>bwa</code>: Read Alignment to the Assembly</summary>
<p>BWA (Burrow-Wheeler Aligner) is used to align the cleaned read files to generated assembly file in order to generate an alignment. The resulting BAM file is directly passed to the Pilon task to polish the assembly for errors.</p>
<div class="admonition techdetails">
<p class="admonition-title">BWA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_bwa.wdl">task_bwa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/bwa">BWA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bio-bwa.sourceforge.net/">BWA Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and accurate short read alignment with Burrows-Wheeler transform</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Pilon</code>: Assembly Polishing</summary>
<p><code>Pilon</code> is a tool that uses read alignments to correct errors in an assembly.</p>
<p>The <code>bwa</code>-generated alignment of the read data to the assembly is used to identify inconsistences between the reads and the assembly in order to correct them. <code>Pilon</code> will attempt to fix individual base errors and small indels using the read data. This can improve the overall quality of the assembly, especially when the assembler has made mistakes due to sequencing errors or low coverage regions.</p>
<p>The default parameters were set to mimic the parameters used by <a href="https://github.com/tseemann/shovill">Shovill</a>: <code>--fix bases --minq 60 --minqual 3 --mindepth 0.25</code>. These can be modified by the user.</p>
<div class="admonition techdetails">
<p class="admonition-title">Pilon Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_pilon.wdl">task_pilon.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/broadinstitute/pilon">Pilon on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/broadinstitute/pilon/wiki">Pilon Wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1371/journal.pone.0112963">Pilon: An Integrated Tool for Comprehensive Microbial Variant Detection and Genome Assembly Improvement</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary>Contig Filtering (optional)</summary>
<details class="task">
<summary><code>Filter Contigs</code>: Contig Quality Control</summary>
<p>To <strong><em>de</em></strong>activate contig filtering, set <code>run_filter_contigs</code> to <code>false</code>.</p>
<p>This task filters the created contigs based on a default minimum length threshold of 200 bp and a minimum coverage of 2.0. It also eliminates homopolymer contigs (contigs of any length that consist of a single nucleotide).</p>
<p>Options are available to skip any of these filters by setting the respective parameters to <code>false</code>: <code>filter_contigs_skip_length_filter</code>, <code>filter_contigs_skip_coverage_filter</code>, and <code>filter_contigs_skip_homopolymer_filter</code>. The minimum length and coverage thresholds can be adjusted using the <code>filter_contigs_min_length</code> and <code>filter_contigs_min_coverage</code> parameters, respectively.</p>
<p>This ensures high-quality assemblies by retaining only contigs that meet specified criteria. Detailed metrics on contig counts and sequence lengths before and after filtering are provided in the output.</p>
<div class="admonition techdetails">
<p class="admonition-title">Filter Contigs Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_filter_contigs.wdl">task_filter_contigs.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Digger-Denovo Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_digger_denovo.wdl">wf_digger_denovo.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>screen</code>: Total Raw Read Quantification and Genome Size Estimation</summary>
<p>The <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl"><code>screen</code></a> task ensures the quantity of sequence data is sufficient to undertake genomic analysis. It uses <a href="https://github.com/rpetit3/fastq-scan"><code>fastq-scan</code></a> and bash commands for quantification of reads and base pairs, and <a href="https://mash.readthedocs.io/en/latest/index.html">mash</a> sketching to estimate the genome size and its coverage. At each step, the results are assessed relative to pass/fail criteria and thresholds that may be defined by optional user inputs. Samples are run through all threshold checks, regardless of failures, and the workflow will terminate after the <code>screen</code> task if any thresholds are not met:</p>
<ol>
<li>Total number of reads: A sample will fail the read screening task if its total number of reads is less than or equal to <code>min_reads</code>.</li>
<li>The proportion of basepairs reads in the forward and reverse read files: A sample will fail the read screening if fewer than <code>min_proportion</code> basepairs are in either the reads1 or read2 files.</li>
<li>Number of basepairs: A sample will fail the read screening if there are fewer than <code>min_basepairs</code> basepairs</li>
<li>Estimated genome size:  A sample will fail the read screening if the estimated genome size is smaller than <code>min_genome_size</code> or bigger than <code>max_genome_size</code>.</li>
<li>Estimated genome coverage: A sample will fail the read screening if the estimated genome coverage is less than the <code>min_coverage</code>.</li>
</ol>
<p>Read screening is undertaken on both the raw and cleaned reads. The task may be skipped by setting the <code>skip_screen</code> variable to true.</p>
<p>Default values vary between the PE, SE, and ONT workflows. The rationale for these default values can be found below. If two default values are shown, the first is for Illumina workflows and the second is for ONT.</p>
<table>
<thead>
<tr>
<th>Variable</th>
<th>Default Value</th>
<th>Rationale</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>skip_screen</code></td>
<td>false</td>
<td>Set to true to skip the read screen from running</td>
</tr>
<tr>
<td><code>min_reads</code></td>
<td>7472 or 5000</td>
<td>Calculated from the minimum number of base pairs required for 20x coverage of the Nasuia deltocephalinicola genome, the smallest known bacterial genome as of 2019-08-07 (112,091 bp), divided by 300 (the longest Illumina read length) or 5000 (estimate of ONT read length)</td>
</tr>
<tr>
<td><code>min_basepairs</code></td>
<td>2241820</td>
<td>Should be greater than 20x coverage of Nasuia deltocephalinicola, the smallest known bacterial genome (112,091 bp)</td>
</tr>
<tr>
<td><code>min_genome_length</code></td>
<td>100000</td>
<td>Based on the Nasuia deltocephalinicola genome, the smallest known bacterial genome (112,091 bp)</td>
</tr>
<tr>
<td><code>max_genome_length</code></td>
<td>18040666</td>
<td>Based on the Minicystis rosea genome, the largest known bacterial genome (16,040,666 bp), plus an additional 2 Mbp to cater for potential extra genomic material</td>
</tr>
<tr>
<td><code>min_coverage</code></td>
<td>10 or 5</td>
<td>A bare-minimum average per base coverage across the genome required for genome characterization. Higher coverage would be required for high-quality phylogenetics.</td>
</tr>
<tr>
<td><code>min_proportion</code></td>
<td>40</td>
<td>Neither read1 nor read2 files should have less than 40% of the total number of reads. For paired-end data only.</td>
</tr>
</tbody>
</table>
<div class="admonition techdetails">
<p class="admonition-title">Screen Technical Details</p>
<p>There is a single WDL task for read screening. The <code>screen</code> task is run twice, once for raw reads and once for clean reads.</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L3">task_screen.wdl (PE sub-task)</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_screen.wdl#L147">task_screen.wdl (SE sub-task)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>These tasks clean then assemble the reads into a <em>de novo</em> assembly and assess the quality of the assembly.</p>
</div>
<details class="task">
<summary><code>read_QC_trim_ont</code>: Read Quality Trimming, Quantification, and Identification</summary>
<p><code>read_QC_trim_ont</code> is a sub-workflow that filters low-quality reads and trims low-quality regions of reads. It uses several tasks, described below.</p>
<div class="admonition dna">
<p class="admonition-title">A note on estimated genome length</p>
<p>By default, the estimated genome length is set to 5 Mb, which is around 0.7 Mb higher than the average bacterial genome length, according to <a href="https://github.com/CDCgov/phoenix/blob/717d19c19338373fc0f89eba30757fe5cfb3e18a/assets/databases/NCBI_Assembly_stats_20240124.txt">the information of thousands of NCBI bacterial assemblies collated here</a>. This estimate can be overwritten by the user and is used by <code>Rasusa</code>.</p>
</div>
<details class="task">
<summary><code>Rasusa</code>: Read Subsampling</summary>
<p><code>Rasusa</code> is a tool to randomly subsample sequencing reads to a specified coverage without assuming that all reads are of equal length, making it especially suitable for long-read data while still being applicable to short-read data.</p>
<p>The <code>Rasusa</code> task performs subsampling on the input raw reads. By default, this task will subsample TheiaProk_ONT reads to a depth of 150X using an estimated genome length of 5 million basepairs (0.7 Mb higher than the average bacterial genome length), and TheiaEuk_ONT reads using an estimated genome length of 50 million basepairs. The estimated genome length can be changed by the user by providing a different value for the <code>genome_length</code> input parameter. The target subsampling depth can also be adjusted by modifying the <code>subsample_coverage</code> variable.</p>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs since it performs <em>random</em> subsampling. To ensure reproducibility, set a a value for the <code>rasusa_seed</code> optional input variable.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Rasusa Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/task_rasusa.wdl">task_rasusa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/mbhall88/rasusa">Rasusa on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/mbhall88/rasusa">Rasusa on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03941">Rasusa: Randomly subsample sequencing reads to a specified coverage</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Nanoq</code>: Read Filtering</summary>
<p>Reads are filtered by length and quality using <code>nanoq</code>. By default, sequences with less than 500 basepairs and quality scores lower than 10 are filtered out to improve assembly accuracy. These defaults are able to be modified by the user.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nanoq Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_nanoq.wdl">task_nanoq.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/esteinig/nanoq">Nanoq on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/esteinig/nanoq/blob/master/README.md">Nanoq Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.02991">Nanoq: ultra-fast quality control for nanopore reads</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Kraken2</code>: Read Identification (optional)</summary>
<p>To activate this task, set <code>call_kraken</code> to <code>true</code> and provide a value for <code>kraken_db</code>.</p>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<p>Kraken2 is run on the raw read data.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<p>A database must be provided if this optional module is activated, through the kraken_db optional input. A list of suggested databases can be found on <a href="../../standalone/kraken2/#databases">Kraken2 standalone documentation</a>.</p>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">Kraken2 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>NanoPlot</code>: Read Quantification</summary>
<p>NanoPlot is used for the determination of mean quality scores, read lengths, and number of reads. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>While this task currently is run <em>outside</em> of the <code>read_QC_trim_ont</code> workflow, it is being included here as it calculates statistics on the read data. This is done so that the actual assembly genome lengths can be used (if an estimated genome length is not provided by the user) to ensure the estimated coverage statistics are accurate.</p>
<div class="admonition techdetails">
<p class="admonition-title">NanoPlot Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_nanoplot.wdl">task_nanoplot.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/wdecoster/NanoPlot">NanoPlot on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/wdecoster/NanoPlot/blob/master/README.md">NanoPlot Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/39/5/btad311/7160911">NanoPack2: population-scale evaluation of long-read sequencing data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim_ont Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_ont.wdl">wf_read_QC_trim_ont.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Flye</code>: <em>De novo</em> Assembly</summary>
<p><code>flye_denovo</code> is a sub-workflow that performs <em>de novo</em> assembly using Flye for ONT data and supports additional polishing and visualization steps.</p>
<div class="admonition tip">
<p class="admonition-title">Ensure correct medaka model is selected if performing medaka polishing</p>
<p>In order to obtain the best results, the appropriate model must be set to match the sequencer's basecaller model; this string takes the format of {pore}_{device}_{caller variant}_{caller_version}. See also <a href="https://github.com/nanoporetech/medaka?tab=readme-ov-file#models">https://github.com/nanoporetech/medaka?tab=readme-ov-file#models</a>. If <code>flye</code> is being run on legacy data the medaka model will likely be <code>r941_min_hac_g507</code>. Recently generated data will likely be suited by the default model of <code>r1041_e82_400bps_sup_v5.0.0</code>.</p>
</div>
<p>The detailed steps and tasks are as follows:</p>
<details class="toggle">
<summary><code>Porechop</code>: Read Trimming (optional; off by default)</summary>
<p>Read trimming is optional and can be enabled by setting the <code>run_porchop</code> input variable to true.</p>
<p>Porechop is a tool for finding and removing adapters from ONT data. Adapters on the ends of reads are trimmed, and when a read has an adapter in the middle, the read is split into two.</p>
<div class="admonition techdetails">
<p class="admonition-title">Porechop Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_porechop.wdl">task_porechop.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rrwick/Porechop">Porechop on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rrwick/Porechop#porechop">https://github.com/rrwick/Porechop#porechop</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Flye</code>: <em>De novo</em> Assembly</summary>
<p>Flye is a <em>de novo</em> assembler for long read data using repeat graphs. Compared to de Bruijn graphs, which require exact k-mer matches, repeat graphs can use approximate matches which better tolerates the error rate of ONT data.</p>
<details class="dna">
<summary><code>flye_read_type</code> input parameter</summary>
<p>This input parameter specifies the type of sequencing reads being used for assembly. This parameter significantly impacts the assembly process and should match the characteristics of your input data. Below are the available options:</p>
<table>
<thead>
<tr>
<th><strong>Parameter</strong></th>
<th><strong>Explanation</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--nano-hq</code> (default)</td>
<td>Optimized for ONT high-quality reads, such as Guppy5+ SUP or Q20 (&lt;5% error). Recommended for ONT reads processed with Guppy5 or newer</td>
</tr>
<tr>
<td><code>--nano-raw</code></td>
<td>For ONT regular reads, pre-Guppy5 (&lt;20% error)</td>
</tr>
<tr>
<td><code>--nano-corr</code></td>
<td>ONT reads corrected with other methods (&lt;3% error)</td>
</tr>
<tr>
<td><code>--pacbio-raw</code></td>
<td>PacBio regular CLR reads (&lt;20% error)</td>
</tr>
<tr>
<td><code>--pacbio-corr</code></td>
<td>PacBio reads corrected with other methods (&lt;3% error)</td>
</tr>
<tr>
<td><code>--pacbio-hifi</code></td>
<td>PacBio HiFi reads (&lt;1% error)</td>
</tr>
</tbody>
</table>
<p>Refer to the Flye documentation for detailed guidance on selecting the appropriate <code>flye_read_type</code> based on your sequencing data and additional optional paramaters.</p>
</details>
<details class="warning" open="open">
<summary>Non-deterministic output(s)</summary>
<p>This task may yield non-deterministic outputs.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Flye Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_flye.wdl">task_flye.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/fenderglass/Flye">Flye on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/fenderglass/Flye/blob/flye/docs/USAGE.md">Flye Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.nature.com/articles/s41587-019-0072-8">Assembly of long, error-prone reads using repeat graphs</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Bandage</code>: Graph Visualization</summary>
<p>Bandage creates <em>de novo</em> assembly graphs containing the assembled contigs and the connections between those contigs. These graphs are useful for visualizing the assembly structure, identifying potential misassemblies, and understanding the relationships between contigs.</p>
<div class="admonition techdetails">
<p class="admonition-title">Bandage Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_bandage_plot.wdl">task_bandage_plot.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rrwick/Bandage">Bandage on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rrwick/Bandage#bandage">Bandage Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/31/20/3350/196114">Bandage: interactive visualization of <em>de novo</em> genome assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Polypolish</code>: Hybrid Assembly Polishing <mark><em>for ONT and Illumina data</em></mark></summary>
<p>If short reads are provided with the optional <code>illumina_read1</code> and <code>illumina_read2</code> inputs, Polypolish will use those short-reads to correct errors in the long-read assemblies. Uniquely, Polypolish uses the short-read alignments where each read is aligned to <em>all</em> possible locations, meaning that even repeat regions will have error correction.</p>
<div class="admonition techdetails">
<p class="admonition-title">Polypolish Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/polishing/task_polypolish.wdl">task_polypolish.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rrwick/Polypolish">Polypolish on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rrwick/Polypolish#polypolish">Polypolish Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1371/journal.pcbi.1009802">Polypolish: short-read polishing of long-read bacterial genome assemblies</a><br><a href="https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.001254">How low can you go? Short-read polishing of Oxford Nanopore bacterial genome assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Medaka</code>: Polishing of Flye assembly (default; optional)</summary>
<p>Polishing is optional and can be skipped by setting the <code>skip_polishing</code> variable to true. If polishing is skipped, then neither Medaka or Racon will run.</p>
<p>Medaka is the default assembly polisher used in TheiaProk. Racon may be used alternatively, and if so, Medaka will not run. Medaka uses the raw reads to polish the assembly and generate a consensus sequence. </p>
<p>Importantly, Medaka requires knowing the model that was used to generate the read data. There are several ways to provide this information:</p>
<ul>
<li>Automatic Model Selection: Automatically determines the most appropriate Medaka model based on the input data, ensuring optimal polishing results without manual intervention. </li>
<li>User-Specified Model Override: Allows users to specify a particular <code>Medaka model</code> if automatic selection does not yield the desired outcome or for specialized use cases.</li>
<li>Default Model: If both automatic model selection fails and no user-specified model is provided, Medaka defaults to the predefined fallback model <code>r1041_e82_400bps_sup_v5.0.0</code>. </li>
</ul>
<div class="admonition info">
<p class="admonition-title">Medaka Model Resolution Process</p>
<p>Medaka's automatic model selection uses the <code>medaka tools resolve_model</code> command to identify the appropriate model for polishing. This process relies on metadata embedded in the input file, which is typically generated by the basecaller. If the automatic selection fails to identify a suitable model, Medaka gracefully falls back to the default model to maintain workflow continuity. <strong>Users should verify the chosen model and consider specifying a model override if necessary.</strong></p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Medaka Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/polishing/task_medaka.wdl">task_medaka.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nanoporetech/medaka">Medaka on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/nanoporetech/medaka#medaka">Medaka Documentation</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Racon</code>: Polishing of Flye assembly (alternative; optional)</summary>
<p>Polishing is optional and can be skipped by setting the <code>skip_polishing</code> variable to true. If polishing is skipped, then neither Medaka or Racon will run.</p>
<p><code>Racon</code> is an alternative to using <code>medaka</code> for assembly polishing, and can be run by setting the <code>polisher</code> input to "racon".  Racon is a consensus algorithm designed for refining raw de novo DNA assemblies generated from long, uncorrected sequencing reads.</p>
<div class="admonition techdetails">
<p class="admonition-title">Racon Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/polishing/task_racon.wdl">task_racon.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lbcb-sci/racon">Racon on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/lbcb-sci/racon#racon">Racon Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://genome.cshlp.org/content/27/5/737">Fast and accurate de novo genome assembly from long uncorrected reads</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Filter Contigs</code>: Filter contigs below a threshold length and remove homopolymer contigs</summary>
<p>This task filters the created contigs based on a user-defined minimum length threshold (default of 1000) and eliminates homopolymer contigs (contigs of any length that consist of a single nucleotide).</p>
<p>This ensures high-quality assemblies by retaining only contigs that meet specified criteria. Detailed metrics on contig counts and sequence lengths before and after filtering are provided in the output.</p>
<div class="admonition techdetails">
<p class="admonition-title">Filter Contigs Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_filter_contigs.wdl">task_filter_contigs.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><code>Dnaapler</code>: Final Assembly Orientation</summary>
<p>Dnaapler reorients contigs to start at specific reference points. Dnaapler supports the following modes, which can be indicated by filling the <code>dnaapler_mode</code> input variable with the desired mode. The default is <code>all</code>, which reorients contigs to start with <code>dnaA</code>, <code>terL</code>, <code>repA</code>, or <code>COG1474</code>.</p>
<ul>
<li><strong>all</strong>: Reorients contigs to start with <code>dnaA</code>, <code>terL</code>, <code>repA</code>, or <code>COG1474</code> (<em>Default</em>)</li>
<li><strong>chromosome</strong>: Reorients to begin with the <code>dnaA</code> chromosomal replication initiator gene, commonly used for bacterial chromosome assemblies.</li>
<li><strong>plasmid</strong>: Reorients to start with the <code>repA</code> plasmid replication initiation gene, ideal for plasmid assemblie</li>
<li><strong>phage</strong>: Reorients to start with the <code>terL</code> large terminase subunit gene, used for bacteriophage assemblies</li>
<li><strong>archaea</strong>: Reorients to start with the <code>COG1474</code> archaeal Orc1/cdc6 gene, relevant for archaeal assemblies</li>
<li><strong>custom</strong>: Reorients based on a user-specified gene in amino acid FASTA format for experimental or unique workflows</li>
<li><strong>mystery</strong>: Reorients to start with a random CDS for exploratory purposes</li>
<li><strong>largest</strong>: Reorients to start with the largest CDS in the assembly, often useful for poorly annotated genomes</li>
<li><strong>nearest</strong>: Reorients to start with the first CDS nearest to the sequence start, resolving CDS breakpoints</li>
<li><strong>bulk</strong>: Processes multiple contigs to start with the desired start gene (<code>dnaA</code>, <code>terL</code>, <code>repA</code>, or custom)</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">Dnaapler Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>WDL Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_dnaapler.wdl">task_dnaapler.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/gbouras13/dnaapler">Dnaapler on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/gbouras13/dnaapler?tab=readme-ov-file#dnaapler">Dnaapler Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://joss.theoj.org/papers/10.21105/joss.05968">Dnaapler: a tool to reorient circular microbial genomes</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Flye-Denovo Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_flye_denovo.wdl">wf_flye_denovo.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<div class="admonition dna">
<p>Since this workflow requires FASTA files as input, no assembly or read trimming is performed, and the workflow proceeds directly to the "post-assembly tasks" section below.</p>
</div>
</div>
</div>
</div>
</div>
<h4 id="post-assembly-tasks">Post-Assembly Tasks<a class="headerlink" href="#post-assembly-tasks" title="Permanent link"></a></h4>
<div class="admonition dna">
<p>The following tasks are performed for all taxa, regardless of the input data type. They include quality control, assembly characterization, and taxonomic identification.</p>
</div>
<h5 id="quality-control">Quality Control<a class="headerlink" href="#quality-control" title="Permanent link"></a></h5>
<details class="task">
<summary><code>quast</code>: Assembly Quality Assessment</summary>
<p>QUAST stands for QUality ASsessment Tool. It evaluates genome/metagenome assemblies by computing various metrics without a reference being necessary. It includes useful metrics such as number of contigs, length of the largest contig and N50.</p>
<div class="admonition techdetails">
<p class="admonition-title">QUAST Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_quast.wdl">task_quast.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ablab/quast">QUAST on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://quast.sourceforge.net/">QUAST Manual on SourceForge</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/29/8/1072/228832">QUAST: quality assessment tool for genome assemblies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>BUSCO</code>: Assembly Quality Assessment</summary>
<p>BUSCO (<strong>B</strong>enchmarking <strong>U</strong>niversal <strong>S</strong>ingle-<strong>C</strong>opy <strong>O</strong>rthologue) attempts to quantify the completeness and contamination of an assembly to generate quality assessment metrics. It uses taxa-specific databases containing genes that are all expected to occur in the given taxa, each in a single copy. BUSCO examines the presence or absence of these genes, whether they are fragmented, and whether they are duplicated (suggestive that additional copies came from contaminants).</p>
<p><strong>BUSCO notation</strong> </p>
<p>Here is an example of BUSCO notation: <code>C:99.1%[S:98.9%,D:0.2%],F:0.0%,M:0.9%,n:440</code>. There are several abbreviations used in this output:</p>
<ul>
<li>Complete (C) - genes are considered "complete" when their lengths are within two standard deviations of the BUSCO group mean length.</li>
<li>Single-copy (S) - genes that are complete and have only one copy.</li>
<li>Duplicated (D) - genes that are complete and have more than one copy.</li>
<li>Fragmented (F) - genes that are only partially recovered.</li>
<li>Missing (M) - genes that were not recovered at all.</li>
<li>Number of genes examined (n) - the number of genes examined.</li>
</ul>
<p>A high equity assembly will use the appropriate database for the taxa, have high complete (C) and single-copy (S) percentages, and low duplicated (D), fragmented (F) and missing (M) percentages. </p>
<div class="admonition techdetails">
<p class="admonition-title">BUSCO Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_busco.wdl">task_busco.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://gitlab.com/ezlab/busco">BUSCO on GitLab</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://busco.ezlab.org/">https://busco.ezlab.org/</a></td>
</tr>
<tr>
<td>Orginal publication</td>
<td><a href="https://academic.oup.com/bioinformatics/article/31/19/3210/211866">BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>qc_check</code>: Check QC Metrics Against User-Defined Thresholds (optional)</summary>
<p>To activate this task, provide a <code>qc_check_table</code> as input.</p>
<p>The <code>qc_check</code> task compares generated QC metrics against user-defined thresholds for each metric. This task will run if the user provides a <code>qc_check_table</code> TSV file. If all QC metrics meet the threshold, the <code>qc_check</code> output variable will read <code>QC_PASS</code>. Otherwise, the output will read <code>QC_NA</code> if the task could not proceed or <code>QC_ALERT</code> followed by a string indicating what metric failed.</p>
<p>The <code>qc_check</code> task applies quality thresholds according to the sample taxa. The sample taxa is taken from the <code>gambit_predicted_taxon</code> value inferred by the GAMBIT module OR can be manually provided by the user using the <code>expected_taxon</code> workflow input.</p>
<details class="toggle">
<summary>Formatting the <em>qc_check_table.tsv</em></summary>
<ul>
<li>The first column of the qc_check_table lists the <code>organism</code> that the task will assess and the header of this column must be "<strong>taxon</strong>".</li>
<li>Any genus or species can be included as a row of the qc_check_table. However, these taxa must <strong>uniquely</strong> match the sample taxa, meaning that the file can include multiple species from the same genus (Vibrio_cholerae and Vibrio_vulnificus), but not both a genus row and species within that genus (Vibrio and Vibrio cholerae). <strong>The taxa should be formatted with the first letter capitalized and underscores in lieu of spaces.</strong></li>
<li>Each subsequent column indicates a QC metric and lists a threshold for each organism that will be checked. <strong>The column names must exactly match expected values, so we highly recommend copy and pasting the header from the template file below as a starting place.</strong></li>
</ul>
</details>
<details class="toggle">
<summary>Template <em>qc_check_table.tsv</em> files</summary>
<ul>
<li>TheiaProk_Illumina_PE: <a href="../../../assets/files/TheiaProk_Illumina_PE_qc_check_template.tsv">theiaprok_illumina_pe_qc_check_template.tsv</a></li>
<li>TheiaProk_FASTA: <a href="../../../assets/files/TheiaProk_FASTA_qc_check_template.tsv">theiaprok_fasta_qc_check_template.tsv</a></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Example Purposes Only</p>
<p>The QC threshold values shown in the file above are for example purposes only and should not be presumed to be sufficient for every dataset.</p>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">qc_check Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_qc_check_phb.wdl">task_qc_check_phb.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>arln_stats</code>: Quality Assessment for ARLN (optional)</summary>
<p>To activate this task, set <code>call_arln_stats</code> to <code>true</code>.</p>
<p>The <code>arln_stats</code> task will provide the user with Antimicrobial Resistance Laboratory Network (<a href="https://www.cdc.gov/antimicrobial-resistance-laboratory-networks/php/about/domestic.html">ARLN</a>)-compliant statistics used for PASS/FAIL assessment, such as assembly ratio, percent GC statistics, and the percent Q30 of raw and cleaned reads.</p>
<p>The Q30 statistics are calculated via a Python script (<a href="https://github.com/theiagen/theiagen_docker_builds/blob/main/arln_stats/1.0.0/q30.py">q30.py</a>) that is incorporated within this task's Docker image. The assembly ratio statistic is obtained by parsing an included <a href="https://github.com/CDCgov/phoenix/blob/main/assets/databases/NCBI_Assembly_stats_20240124.txt">NCBI Assembly Stats</a> text file that is created by the CDC to aggregate NCBI prokaryotic assembly data. The file is sorted by taxon and has statistics for each taxon (see toggle below). This data is used to obtain and calculate the assembly ratio of samples passed through <code>arln_stats</code>, much the same as CDC's <a href="https://github.com/CDCgov/phoenix/wiki/Pipeline-Overview#qc-of-assembled-scaffolds--500bps">Phoenix pipeline</a>.</p>
<details class="toggle">
<summary>NCBI Assembly Stats Explained</summary>
<p>This aggregated data file is produced by the <a href="https://github.com/CDCgov/phoenix/wiki/Pipeline-Overview#qc-of-assembled-scaffolds--500bps">CDC</a> and the columns are as follows: </p>
<ul>
<li>Species</li>
<li>Assembly_Size_Min</li>
<li>Assembly_Size_Max</li>
<li>Assembly_vMedian</li>
<li>Assembly_Size_Mean</li>
<li>Assembly_Size_StDev</li>
<li>Assembly_count</li>
<li>GC_Min</li>
<li>GC_Max</li>
<li>GC_Median</li>
<li>GC_Mean</li>
<li>GC_Stdev</li>
<li>GC_count</li>
<li>CDS_Min</li>
<li>CDS_Max</li>
<li>CDS_Median</li>
<li>CDS_Mean</li>
<li>CDS_Stdev</li>
<li>CDS_count</li>
<li>Consensus_TAXID</li>
</ul>
</details>
<div class="admonition techdetails">
<p class="admonition-title">arln_stats Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_handling/task_arln_stats.wdl">task_arln_stats.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h5 id="taxonomic-assignment">Taxonomic Assignment<a class="headerlink" href="#taxonomic-assignment" title="Permanent link"></a></h5>
<details class="task">
<summary><code>GAMBIT</code>: <strong>Taxon Assignment</strong></summary>
<p><a href="https://github.com/jlumpe/gambit"><code>GAMBIT</code></a> determines the taxon of the genome assembly using a k-mer based approach to match the assembly sequence to the closest complete genome in a database, thereby predicting its identity. Sometimes, GAMBIT can confidently designate the organism to the species level. Other times, it is more conservative and assigns it to a higher taxonomic rank.</p>
<p>For additional details regarding the GAMBIT tool and a list of available GAMBIT databases for analysis, please consult the <a href="../../../guides/gambit/">GAMBIT</a> tool documentation.</p>
<div class="admonition techdetails">
<p class="admonition-title">GAMBIT Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_gambit.wdl">task_gambit.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/jlumpe/gambit">GAMBIT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://gambit-genomics.readthedocs.io/en/latest/">GAMBIT ReadTheDocs</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0277575">GAMBIT (Genomic Approximation Method for Bacterial Identification and Tracking): A methodology to rapidly leverage whole genome sequencing of bacterial isolates for clinical identification</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>KmerFinder</code>: Taxon Assignment (optional)</summary>
<p>To activate this task, set <code>call_kmerfinder</code> to <code>true</code>.</p>
<p>KmerFinder predicts prokaryotic species based on the number of overlapping (co-occurring)&nbsp;<em>k</em>-mers, i.e., 16-mers, between the query genome and genomes in a reference database. These <em>k</em>-mers are selected with the prefix of <code>ATGAC</code> in order to focus on coding regions of genomes. A prediction is made by identifying which species in the training data has the highest number of 16-mers in common with the query. This match is made regardless of position. Ties will result in an alphabetical sorting of tied species and the return of the first species in the alphabetical list. </p>
<p>This is a simpler approach than other <em>k</em>-mer based tools such as GAMBIT, which uses a 11-mer approach while implementing thresholds to its classification algorithm allowing it to re-assess at a higher level of classification if need be.</p>
<div class="admonition techdetails">
<p class="admonition-title">KmerFinder Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kmerfinder.wdl">task_kmerfinder.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://bitbucket.org/genomicepidemiology/kmerfinder">KmerFinder BitBucket</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://cge.food.dtu.dk/services/KmerFinder/instructions.php">CGE KmerFinder Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://journals.asm.org/doi/full/10.1128/jcm.02981-13?rfr_dat=cr_pub++0pubmed&amp;url_ver=Z39.88-2003&amp;rfr_id=ori%3Arid%3Acrossref.org">Benchmarking of Methods for Genomic Taxonomy</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>MUMmer_ANI</code>: Taxon Assignment using Average Nucleotide Identity (optional)</summary>
<p>To activate this task, set <code>call_ani</code> to <code>true</code>.</p>
<p>Average Nucleotide Identity (ANI) is a useful approach for taxonomic identification. The higher the percentage ANI of a query sequence to a given reference genome, the more likely the sequence is the same taxa as the reference. </p>
<p>ANI is calculated in TheiaProk using <a href="https://github.com/lskatz/ani-m">a perl script written by Lee Katz</a> (ani-m.pl). This uses <a href="http://mummer.sourceforge.net/">MUMmer</a> to rapidly align entire query assemblies to one or more reference genomes. By default, TheiaProk uses a set of 43 reference genomes in <a href="https://github.com/StaPH-B/docker-builds/blob/master/build-files/fastani/1.34-RGDV2/RGDv2-metadata.tsv">RGDv2</a>, a database containing genomes of enteric pathogens commonly sequenced by CDC EDLB &amp; PulseNet participating laboratories. The user may also provide their own reference genome. After genome alignment with MUMmer, ani-m.pl calculates the average nucleotide identity and percent bases aligned between 2 genomes (query and reference genomes)</p>
<p>The default database of reference genomes used is called "Reference Genome Database version 2" AKA "RGDv2". This database is composed of 43 enteric bacteria representing 32 species and is intended for identification of enteric pathogens and common contaminants. It contains six Campylobacter spp., three Escherichia/Shigella spp., one <em>Grimontia hollisae</em>, six <em>Listeria spp.</em>, one <em>Photobacterium damselae</em>, two <em>Salmonella spp.</em>, and thirteen <em>Vibrio spp.</em> </p>
<p>2 Thresholds are utilized to prevent false positive hits. The <code>ani_top_species_match</code> will only report a genus &amp; species match if both thresholds are surpassed. Both of these thresholds are set to match those used in BioNumerics for PulseNet organisms.</p>
<ol>
<li><code>ani_threshold</code> default value of 80.0</li>
<li><code>percent_bases_aligned_threshold</code> default value of 70.0</li>
</ol>
<p>For more information on RGDv2 database of reference genomes, please see <a href="https://doi.org/10.3389/fmicb.2023.1225207">the publication here</a>.</p>
<div class="admonition techdetails">
<p class="admonition-title">MUMmer_ANI Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_mummer_ani.wdl">task_mummer_ani.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lskatz/ani-m">ani-m on GitHub</a><br><a href="https://github.com/mummer4/mummer">MUMmer on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/lskatz/ani-m">ani-m on GitHub</a><br><a href="https://mummer.sourceforge.net/">MUMmer on SourceForge</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>MUMmer4</em>: <a href="https://doi.org/10.1371/journal.pcbi.1005944">MUMmer4: A fast and versatile genome alignment system</a><br><em>RGDv2 database</em>: <a href="https://doi.org/10.3389/fmicb.2023.1225207">Rapid identification of enteric bacteria from whole genome sequences using average nucleotide identity metrics</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h5 id="amr-characterization">AMR Characterization<a class="headerlink" href="#amr-characterization" title="Permanent link"></a></h5>
<details class="task">
<summary><code>AMRFinderPlus</code>: AMR Genotyping</summary>
<p>NCBI's <a href="https://github.com/ncbi/amr/wiki">AMRFinderPlus</a> is the default antimicrobial resistance (AMR) detection tool used in TheiaProk. ResFinder may be used alternatively and if so, AMRFinderPlus is not run. </p>
<p>AMRFinderPlus identifies acquired antimicrobial resistance (AMR) genes, virulence genes, and stress genes.  Such AMR genes confer resistance to antibiotics, metals, biocides, heat, or acid. For some taxa (see <a href="https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#--organism-option">here</a>), AMRFinderPlus will provide taxa-specific results including filtering out genes that are almost ubiquitous in the taxa (intrinsic genes) and identifying resistance-associated point mutations.  In TheiaProk, the taxon used by AMRFinderPlus is specified based on the <code>gambit_predicted_taxon</code> or a user-provided <code>expected_taxon</code>. AMRFinderPlus also has the ability to utilize a GFF and protein FASTA file which can be enabled via <code>amrfinder_use_gff</code> allowing for more accurate calls.</p>
<p>You can check if a gene or point mutation is in the AMRFinderPlus database with the <a href="https://www.ncbi.nlm.nih.gov/pathogens/refgene/#">Reference Gene Catalog</a>, find the sequences of reference genes in the <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA313047">Bacterial Antimicrobial Resistance Reference Gene Database BioProject</a>, and search the query Hidden Markov Models (HMMs) used by AMRFinderPlus to identify AMR genes and some stress and virulence proteins in the <a href="https://www.ncbi.nlm.nih.gov/pathogens/hmm/">Reference HMM Catalog</a>. The AMRFinderPlus database is updated frequently. You can ensure you are using the most up-to-date version by specifying the Docker image in the optional workflow input.</p>
<details class="toggle">
<summary>AMRFinderPlus results can be used to confirm taxonomic assignment in <em>A. baumannii</em></summary>
<p>The <em>bla</em>OXA-51-like genes, also known as <em>oxaAB</em>, are considered intrinsic to <em>Acinetobacter baumannii</em> but are not found in other <em>Acinetobacter</em> species. <strong>Identification of a <em>bla</em>OXA-51-like gene with this tool is therefore considered to confirm the species' identity as <em>A. baumannii</em>.</strong></p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">AMRFinderPlus Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/drug_resistance/task_amrfinderplus.wdl">task_amrfinderplus.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/amr">NCBI's AMRFinderPlus on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/amr/wiki">https://github.com/ncbi/amr/wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8208984/">AMRFinderPlus and the Reference Gene Catalog facilitate examination of the genomic links among antimicrobial resistance, stress response, and virulence</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ResFinder</code>: AMR Genotyping and XDR Shigella Prediction (optional)</summary>
<p>To activate this task, set <code>call_resfinder</code> to <code>true</code>.</p>
<p><strong>AMR Genotyping</strong></p>
<p>The <code>ResFinder</code> task is an optional task that can be used in conjunction with AMRFinderPlus for detection and identification of AMR genes and resistance-associated mutations. This task runs the Centre for Genomic Epidemiology (CGE) ResFinder tool to identify acquired antimicrobial resistance. The default thresholds for calling AMR genes are 90% identity and 50% coverage of the reference genes (expressed as a fraction in workflow inputs: 0.9 and 0.5). These are the same thresholds utilized in BioNumerics for calling AMR genes.</p>
<p>This task can also run the CGE PointFinder tool if the <code>call_pointfinder</code> variable is set with to <code>true</code>. The databases underlying the task are different to those used by AMRFinderPlus.</p>
<details class="toggle">
<summary>PointFinder-supported organisms</summary>
<p>The following organisms are currently supported by <em>PointFinder</em> for mutational-based predicted resistance. If GAMBIT (<a href="#taxonomic-assignment">see above</a>) predicted the sample to be an organism not on this list, PointFinder will be skipped.</p>
<ul>
<li>Campylobacter coli &amp; C. jejuni</li>
<li>Enterococcus faecalis</li>
<li>Enterococcus faecium</li>
<li>Escherichia coli &amp; Shigella spp.</li>
<li>Helicobacter pylori</li>
<li>Neisseria gonorrhoeae</li>
<li>Klebsiella</li>
<li>Mycobacterium tuberculosis</li>
<li>Salmonella spp.</li>
<li>Staphylococcus aureus</li>
</ul>
</details>
<p><strong>XDR Shigella prediction</strong></p>
<p>The <code>ResFinder</code> Task also has the ability to predict whether or not a sample meets the CDC's definition for extensively drug-resistant (XDR) Shigella. </p>
<blockquote>
<p><em>CDC defines XDR Shigella bacteria as strains that are resistant to all commonly recommended empiric and alternative antibiotics  azithromycin, ciprofloxacin, ceftriaxone, trimethoprim-sulfamethoxazole (TMP-SMX), and ampicillin</em>. <a href="https://emergency.cdc.gov/han/2023/han00486.asp">See also the <em>Increase in Extensively Drug-Resistance Shigellosis in the United State</em> CDC Health Network Alert</a> where this definition can be found.</p>
</blockquote>
<details class="toggle">
<summary>Criteria for XDR Shigella Prediction</summary>
<p>A sample is required to meet <strong>all 7 criteria</strong> in order to be predicted as <code>XDR Shigella</code> </p>
<ol>
<li>The GAMBIT task in the workflow must identify the sample as <code>Shigella</code> OR the user must input the word <code>Shigella</code> somewhere within the input String variable called <code>expected_taxon</code>. This requirement serves as the identification of a sample to be of the Shigella genus.</li>
<li>Resfinder or PointFinder predicted resistance to <strong>Ampicillin</strong></li>
<li>Resfinder or PointFinder predicted resistance to <strong>Azithromycin</strong></li>
<li>Resfinder or PointFinder predicted resistance to <strong>Ciprofloxacin</strong></li>
<li>Resfinder or PointFinder predicted resistance to <strong>Ceftriazone</strong></li>
<li>Resfinder or PointFinder predicted resistance to <strong>Trimethoprim</strong></li>
<li>Resfinder or PointFinder predicted resistance to <strong>Sulfamethoxazole</strong></li>
</ol>
</details>
<p>There are 3 potential outputs for the <strong><code>resfinder_predicted_xdr_shigella</code></strong> output string<strong>:</strong></p>
<ul>
<li><strong><code>Not Shigella based on gambit_predicted_taxon or user input</code></strong></li>
<li><strong><code>Not XDR Shigella</code></strong>&nbsp;for samples identified as Shigella by GAMBIT or user input BUT does ResFinder did not predict resistance to <strong>all 6 drugs in XDR definition</strong></li>
<li><strong><code>XDR Shigella</code></strong>&nbsp;meaning the sample was identified as Shigella and ResFinder/PointFinder did predict resistance to ceftriazone, azithromycin, ciprofloxacin, trimethoprim, sulfamethoxazole, and ampicillin.</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">ResFinder Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/drug_resistance/task_resfinder.wdl">task_resfinder.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://bitbucket.org/genomicepidemiology/resfinder/src/master/">ResFinder Tool on BitBucket</a><br><a href="https://bitbucket.org/genomicepidemiology/resfinder_db/src/master/">ResFinder Database on BitBucket</a><br><a href="https://bitbucket.org/genomicepidemiology/pointfinder_db/src/master/">PointFinder Database on BitBucket</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bitbucket.org/genomicepidemiology/resfinder/src/master/README.md">ResFinder on BitBucket</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>ResFinder tool</em>: <a href="https://academic.oup.com/jac/article/75/12/3491/5890997">ResFinder 4.0 for predictions of phenotypes from genotypes</a><br><em>ResFinder database</em>: <a href="https://doi.org/10.1093/jac/dks261">Identification of acquired antimicrobial resistance genes</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>GAMMA</code>: AMR Genotyping (optional)</summary>
<p>To activate this task, set <code>call_gamma</code> to <code>true</code>.</p>
<p><a href="https://github.com/rastanton/GAMMA/tree/main">GAMMA</a> (Gene Allele Mutation Microbial Assessment) is a protein identity based tool that identifies gene matches in microbial genomic data. GAMMA will also translate and annotate each match providing mutational and truncation information for identified matches. This is done much like AMRFinder, however, GAMMA uses BLAT as opposed to BLAST allowing for faster calls with comparable accuracy.  </p>
<p>GAMMA utilizes a multifasta database of the coding sequences of genes specified by the user. This allows for GAMMA to search for AMR, hypervirulence, plasmid markers, or any prokaryotic database. The default for <code>task_gamma.wdl</code> is the GAMMA-provided <a href="gs:/theiagen-public-resources-rp/reference_data/databases/gamma/default_ResFinderDB_Combined_05-06-20.fsa">ResFinder Database</a>. GAMMA will then return gene matches with mutation and truncation information as a <code>.gamma</code> file which can be accompanied with a GFF output utilizing the <code>--gff</code> flag.</p>
<p>GAMMA also allows for the usage of only nucleotide sequences rather than translated sequences. Using GAMMA-S, enabled with the boolean <code>run_gammas</code>, will find the best matches from a multifasta database without translating sequences. </p>
<div class="admonition techdetails">
<p class="admonition-title">GAMMA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/drug_resistance/task_gamma.wdl">task_gamma.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rastanton/GAMMA/tree/main">GAMMA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rastanton/GAMMA/tree/main">GAMMA on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/38/2/546/6355578">GAMMA: a tool for the rapid identification, classification and annotation of translated gene matches from sequencing data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ABRicate</code>: AMR Genotyping (optional)</summary>
<p>To activate this task, set <code>call_abricate</code> to <code>true</code>.</p>
<p>The <code>abricate</code> module, if enabled, will run ABRicate with the database defined in <code>abricate_db</code> to perform mass screening of contigs for antimicrobial resistance or virulence genes. It comes bundled with multiple databases: NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB. It only detects acquired resistance genes,&nbsp;<strong>NOT</strong>&nbsp;point mutations.</p>
<p>By default, the "vfdb" database is used. The <a href="https://www.mgc.ac.cn/VFs/">virulence factor database (VFDB)</a> is a comprehensive resource for bacterial pathogen virulence factors.</p>
<div class="admonition techdetails">
<p class="admonition-title">ABRicate Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/drug_resistance/task_abricate.wdl">task_abricate.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://www.mgc.ac.cn/VFs/main.htm">VFDB Database</a><br><a href="https://github.com/tseemann/abricate">ABRicate on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.mgc.ac.cn/VFs/main.htm">VFDB Database</a><br><a href="https://github.com/tseemann/abricate">ABRicate on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>VFDB database</em>: <a href="https://doi.org/10.1093/nar/gky1080">VFDB 2019: a comparative pathogenomic platform with an interactive web interface</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>amr_search</code>: Antimicrobial Resistance Profiling (optional)</summary>
<p>To activate this task, set <code>run_amr_search</code> to be <code>true</code>.</p>
<p>This task performs <em>in silico</em> antimicrobial resistance (AMR) profiling for supported species using AMRsearch, the primary tool used by <a href="https://pathogen.watch/">Pathogenwatch</a> to genotype and infer antimicrobial resistance (AMR) phenotypes from assembled microbial genomes.</p>
<p>AMRsearch screens against Pathogenwatch's library of curated genotypes and inferred phenotypes, developed in collaboration with community experts. Resistance phenotypes are determined based on both <em>resistance genes</em> and <em>mutations</em>, and the system accounts for interactions between multiple SNPs, genes, and suppressors. Predictions follow <strong>S/I/R classification</strong> (<em>Sensitive</em>, <em>Intermediate</em>, <em>Resistant</em>).</p>
<p>Currently, only a subset of species are supported by this task.</p>
<details class="toggle">
<summary>Supported Species</summary>
<p>The following table shows the species name and the associated NCBI Code. If you are running AMR Search as part of TheiaProk and TheiaEuk, these codes will be automatically determined based on the GAMBIT predicted taxon, or the user-provided <code>expected_taxon</code> input.</p>
<table>
<thead>
<tr>
<th>Species</th>
<th>NCBI Code</th>
</tr>
</thead>
<tbody>
<tr>
<td><em>Neisseria gonorrhoeae</em></td>
<td>485</td>
</tr>
<tr>
<td><em>Staphylococcus aureus</em></td>
<td>1280</td>
</tr>
<tr>
<td><em>Salmonella</em> Typhi</td>
<td>90370</td>
</tr>
<tr>
<td><em>Streptococcus pneumoniae</em></td>
<td>1313</td>
</tr>
<tr>
<td><em>Klebisiella</em></td>
<td>570</td>
</tr>
<tr>
<td><em>Escherichia</em></td>
<td>561</td>
</tr>
<tr>
<td><em>Mycobacterium tuberculosis</em></td>
<td>1773</td>
</tr>
<tr>
<td><em>Candida auris</em></td>
<td>498019</td>
</tr>
<tr>
<td><em>Vibrio cholerae</em></td>
<td>666</td>
</tr>
<tr>
<td><em>Campylobacter</em></td>
<td>194</td>
</tr>
</tbody>
</table>
</details>
<p><strong>Outputs:</strong></p>
<ul>
<li><strong>JSON Output</strong>: Contains the complete AMR profile, including detailed resistance state, detected resistance genes/mutations, and supporting BLAST results.</li>
<li><strong>CSV &amp; PDF Tables</strong>: An incorporated Python script, <code>parse_amr_json.py</code>, extracts and formats results into a CSV file and PDF summary table for easier visualization.</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">amr_search Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/drug_resistance/task_amr_search.wdl">task_amr_search.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/pathogenwatch-oss/amr-search">AMRsearch on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/pathogenwatch-oss/amr-search">AMRsearch on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7545138/">PAARSNP: rapid genotypic resistance prediction for <em>Neisseria gonorrhoeae</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h5 id="sequence-type">Sequence Type<a class="headerlink" href="#sequence-type" title="Permanent link"></a></h5>
<details class="task">
<summary><code>TS_MLST</code>: MLST Profiling</summary>
<p><a href="https://doi.org/10.1073/pnas.95.6.3140">Multilocus sequence typing (MLST)</a> is a typing method reflecting population structure. It was developed as a portable, unambiguous method for global epidemiology using PCR, but can be applied to whole-genome sequences <em>in silico</em>. MLST is commonly used for pathogen surveillance, ruling out transmission, and grouping related genomes for comparative analysis.</p>
<p>TheiaProk uses <a href="https://github.com/tseemann/mlst">the MLST tool developed by Torsten Seeman</a> to assess MLST using traditional <a href="https://pubmlst.org/">PubMLST</a> typing schemes. </p>
<p>MLST schemes are taxa-specific. Each scheme uses fragments of typically 7 housekeeping genes ("loci") and has a database associating an arbitrary number with each distinct allele of each locus. Each unique combination of alleles ("allelic profile") is assigned a numbered sequence type (ST). Significant diversification of genomes is captured by changes to the MLST loci via mutational events creating new alleles and STs, or recombinational events replacing the allele and changing the ST. Relationships between STs are based on the number of alleles they share. Clonal complexes share a scheme-specific number of alleles (usually for five of the seven loci).</p>
<div class="admonition tip">
<p class="admonition-title">MLST Limitations</p>
<p>Some taxa have multiple MLST schemes, and some MLST schemes are insufficiently robust.</p>
</div>
<details class="toggle" open="open">
<summary>Interpretation of MLST results</summary>
<p>Each MLST results file returns the ST and allele results for one sample. If the alleles and ST are correctly assigned, only a single integer value will be present for each. If an ST cannot be assigned, multiple integers or additional characters will be shown, <a href="https://github.com/tseemann/mlst/tree/v2.22.0#missing-data">representing the issues with assignment as described here</a>.</p>
</details>
<details class="toggle">
<summary>Identifying novel alleles and STs</summary>
<p>The MLST schemes used in TheiaProk are curated on the PubMLST website.If you identify novel alleles or allelic profiles in your data using TheiaProk's MLST task, you can get these assigned via PubMLST:</p>
<ol>
<li>Check that the novel allele or ST has not already been assigned a type on PubMLST. <ol>
<li>Download the assembly file from Terra for your sample with the novel allele or ST</li>
<li>Go to the <a href="https://pubmlst.org/organisms">PubMLST webpage for the organism of interest</a> </li>
<li>Navigate to the organism "Typing" page</li>
<li>Under "Query a sequence" choose "Single sequence" (e.g., <a href="https://pubmlst.org/bigsdb?db=pubmlst_hinfluenzae_seqdef&amp;page=sequenceQuery">this is the page for <em>H. influenzae</em></a>), select the MLST scheme under "Please select locus/scheme", upload the assembly fasta file, and click submit</li>
<li>Results will be returned lower on the page</li>
</ol>
</li>
<li>If the allele or ST has not been typed previously on the PubMLST website (step 1), new allele or ST numbers can be assigned using the instructions <a href="https://pubmlst.org/submit-data">provided by pubMLST</a>.</li>
</ol>
</details>
<details class="toggle" open="open">
<summary>Taxa with multiple MLST schemes</summary>
<p>By default, the MLST tool automatically detects the genome's taxa to select the MLST scheme. Users may specify a desired scheme using the <code>scheme</code> variable of the <code>ts_mlst</code> task. Available schemes are listed <a href="https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab">here</a> and the scheme name should be provided in quotation marks ("...").</p>
<p>Some taxa have multiple MLST schemes, e.g. the <em>Escherichia</em> and <em>Leptospira</em> genera,  <em>Acinetobacter baumannii</em>, <em>Clostridium difficile</em>, and <em>Streptococcus thermophilus</em>. Only one scheme will be used by default. This can be changed for <em>Escherichia</em> and <em>A. baumannii</em> by setting the <code>ts_mlst run_secondary_scheme</code> variable to <code>true</code>. This will cause MLST to run the other scheme associated with those organisms and output those results in addition to the default scheme.</p>
<p>For <em>E. coli</em>, the user may set <code>ts_mlst scheme_override</code> to be <code>true</code> to prevent running <em>E. coli</em> samples through the "aeromonas", "cfreundii", and "senterica" schemes, which can be common mischaracterizations.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">TS_MLST Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/multi/task_ts_mlst.wdl">task_ts_mlst.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/tseemann/mlst">mlst on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/tseemann/mlst">mlst on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>PubMLST database</em>: <a href="https://doi.org/10.12688/wellcomeopenres.14826.1">Open-access bacterial population genomics: BIGSdb software, the PubMLST.org website and their applications</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h5 id="assembly-annotation">Assembly Annotation<a class="headerlink" href="#assembly-annotation" title="Permanent link"></a></h5>
<div class="admonition tip">
<p>By default, <code>genome_annotation</code> is set to <code>"prokka"</code>. To use <code>bakta</code> instead, set <code>genome_annotation</code> to <code>"bakta"</code>. These tools are mutually exclusive.</p>
</div>
<details class="task">
<summary><code>Prokka</code>: Assembly Annotation (default)</summary>
<p>Assembly annotation is available via <code>Prokka</code> as default.</p>
<p><a href="https://github.com/tseemann/prokka"><code>Prokka</code></a> is a prokaryotic genome annotation tool used to identify and describe features of interest within the genome sequence. Prokka annotates the genome by querying three core databases: <a href="https://isfinder.biotoul.fr/">ISfinder</a>, <a href="https://www.ncbi.nlm.nih.gov/bioproject/313047">NCBI's Bacterial Antimicrobial Resistance Reference Gene Database</a>, and <a href="https://www.uniprot.org/uniprotkb?query=reviewed%3Atrue">UniProtKB (SwissProt)</a>. Additional databases can be used or specified, with instructions on how to do so <a href="https://github.com/tseemann/prokka#databases">located in the Prokka README</a>.</p>
<p>The most versatile output from Prokka is likely the GFF3 file, as it contains all of the generated information, though other file formats are available for the instances when a reduction of the GFF3 is useful.</p>
<div class="admonition techdetails">
<p class="admonition-title">Prokka Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/annotation/task_prokka.wdl">task_prokka.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/tseemann/prokka">Prokka on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/tseemann/prokka">Prokka on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/30/14/2068/2390517?login=false">Prokka: rapid prokaryotic genome annotation</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Bakta</code>: Assembly Annotation (alternative)</summary>
<p>To activate this task, set <code>genome_annotation</code> to <code>"bakta"</code>.</p>
<p><code>Bakta</code> is intended for the annotation of bacterial genomes and plasmids and is used to identify and describe regions of interest within the genome.</p>
<p>In addition to the standard annotation outputs, <code>Bakta</code> also provides a plot summarizing the annotation results, which can be useful for visualizing genome features.</p>
<p><strong>Bakta Database Options</strong></p>
<p>Our implementation of <code>Bakta</code> supports three database configurations:</p>
<ul>
<li><strong>Light</strong>: Optimized for faster performance and lower resource usage, with a focused set of core reference data for most bacterial genome annotations. Recommended for quick annotations or limited computational resources. Specify "light" for the <code>bakta_db</code> input.</li>
<li><strong>Full</strong> (default): Comprehensive with extensive reference annotations, suitable for detailed and accurate annotations. Specify "full" for the <code>bakta_db</code> input.</li>
<li><strong>Custom</strong>: Allows users to provide a Bakta-compatible database stored in Google Cloud Storage Bucket. This file must be a .tar.gz archive containing a properly formatted Bakta database with a valid version.json. Please see the <a href="https://github.com/oschwengers/bakta#database">Bakta database documentation</a> for detailed formatting requirements. Example: <code>"bakta_db": "gs://my-bucket/custom_bakta_db.tar.gz"</code></li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">Bakta Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/annotation/task_bakta.wdl">task_bakta.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/oschwengers/bakta">Bakta on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/oschwengers/bakta">Bakta on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000685">Bakta: rapid and standardized annotation of bacterial genomes via alignment-free sequence identification</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h5 id="plasmid-identification">Plasmid Identification<a class="headerlink" href="#plasmid-identification" title="Permanent link"></a></h5>
<details class="task">
<summary><code>PlasmidFinder</code>: Plasmid Identification (optional)</summary>
<p>To activate this task, set <code>call_plasmidfinder</code> to <code>true</code>.</p>
<p><code>PlasmidFinder</code>, from the Center for Genomic Epidemiology, detects plasmids in total or partially sequenced genomes and identifies the closest plasmid type in the database for typing purposes.</p>
<details class="toggle" open="open">
<summary>What are plasmids?</summary>
<p>Plasmids are double-stranded circular or linear DNA molecules that are capable of replication independently of the chromosome and may be transferred between different species and clones. Many plasmids contain resistance or virulence genes, though some do not clearly confer an advantage to their host bacterium.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">PlasmidFinder Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/plasmid_detection/task_plasmidfinder.wdl">task_plasmidfinder.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://bitbucket.org/genomicepidemiology/plasmidfinder/src/master/">PlasmidFinder Tool on BitBucket</a><br><a href="https://bitbucket.org/genomicepidemiology/plasmidfinder_db/src/master/">PlasmidFinder Database on BitBucket</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bitbucket.org/genomicepidemiology/plasmidfinder/src/master/">PlasmidFinder on BitBucket</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4068535/">In Silico Detection and Typing of Plasmids using PlasmidFinder and Plasmid Multilocus Sequence Typing</a><br></td>
</tr>
</tbody>
</table>
</div>
</details>
<h3 id="taxa-specific-tasks">Taxa-Specific Tasks<a class="headerlink" href="#taxa-specific-tasks" title="Permanent link"></a></h3>
<div class="admonition dna">
<p>The TheiaProk workflows automatically activate taxa-specific sub-workflows after the identification of relevant taxa using <code>GAMBIT</code>. Alternatively, the user can provide the expected taxa in the <code>expected_taxon</code> workflow input to override the taxonomic assignment made by GAMBIT. Modules are launched for all TheiaProk workflows unless otherwise indicated.</p>
<p>Please note that some modules require specific input data that may render it incompatible with every workflow. For example, ShigaTyper (a <em>Shigella</em>/EIEC serotyping tool) is not available for TheiaProk_FASTA as it requires read data as input. We have made a note next to each module to indicate which workflows are compatible with the module if there are restrictions.</p>
</div>
<details class="task">
<summary><code>Taxon Tables</code>: Copy outputs to new data tables based on taxonomic assignment (optional)</summary>
<div class="admonition tip">
<p>This task is incompatible when running TheiaProk on the command-line as it is geared specifically for Terra. Do not activate this task if you are a command-line user.</p>
</div>
<p>Activate this task by providing a value for the <code>taxon_tables</code> input variable. If provided, the user must also provide values to the <code>terra_project</code> and <code>terra_workspace</code> optional input variables.</p>
<p>The <code>taxon_tables</code> module, if enabled, will copy sample data to a different data table based on the taxonomic assignment. For example, if an <em>E. coli</em> sample is analyzed, the module will copy the sample data to a new table for <em>E. coli</em> samples or add the sample data to an existing table.</p>
<div class="admonition tip">
<p>To activate the <code>taxon_tables</code> module, provide a file indicating data table names to copy samples of each taxa to in the <code>taxon_tables</code> input variable.</p>
<p><strong>Formatting the <code>taxon_tables</code> file</strong></p>
<p>The <code>taxon_tables</code> file must be uploaded a Google storage bucket that is accessible by Terra and should be in the format below. Briefly, the bacterial genera or species should be listed in the leftmost column with the name of the data table to copy samples of that taxon to in the rightmost column.</p>
<table>
<thead>
<tr>
<th>taxon</th>
<th>taxon_table</th>
</tr>
</thead>
<tbody>
<tr>
<td>Listeria_monocytogenes</td>
<td>lmonocytogenes_specimen</td>
</tr>
<tr>
<td>Salmonella</td>
<td>salmonella_specimen</td>
</tr>
<tr>
<td>Escherichia</td>
<td>ecoli_specimen</td>
</tr>
<tr>
<td>Shigella</td>
<td>shigella_specimen</td>
</tr>
<tr>
<td>Streptococcus</td>
<td>strep_pneumo_specimen</td>
</tr>
<tr>
<td>Legionella</td>
<td>legionella_specimen</td>
</tr>
<tr>
<td>Klebsiella</td>
<td>klebsiella_specimen</td>
</tr>
<tr>
<td>Mycobacterium</td>
<td>mycobacterium_specimen</td>
</tr>
<tr>
<td>Acinetobacter</td>
<td>acinetobacter_specimen</td>
</tr>
<tr>
<td>Pseudomonas</td>
<td>pseudomonas_specimen</td>
</tr>
<tr>
<td>Staphylococcus</td>
<td>staphyloccus_specimen</td>
</tr>
<tr>
<td>Neisseria</td>
<td>neisseria_specimen</td>
</tr>
</tbody>
</table>
</div>
<div class="admonition tip">
<p>There are no output columns for the taxon table task. The only output of the task is that additional data tables will appear for in the Terra workspace for samples matching a taxa in the taxon table file.</p>
</div>
<div class="admonition techdetails">
<p class="admonition-title"><code>export_taxon_table</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/utilities/data_export/task_export_taxon_table.wdl">task_export_taxon_table.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="toggle">
<summary><em>Acinetobacter baumannii</em></summary>
<h5 id="acinetobacter-baumannii"><em>Acinetobacter baumannii</em><a class="headerlink" href="#acinetobacter-baumannii" title="Permanent link"></a></h5>
<details class="task">
<summary><code>Kaptive</code>: Capsule and Lipooligosaccharide Outer Core Typing</summary>
<p>The cell-surface capsular polysaccharide (CPS) of <em>Acinetobacter baumannii</em> can be used as an epidemiological marker. CPS varies in its composition and structure and is a key determinant in virulence and a target for non-antibiotic therapeutics. Specificity for non-antibiotic therapeutics (e.g. phage therapy) bear particular significance given the extent of antibiotic resistance found in this <a href="https://journals.asm.org/doi/10.1128/CMR.00181-19">ESKAPE</a> pathogen. </p>
<p>Biosynthesis and export of CPS is encoded by genes clustering at the K locus (KL). Additional genes associated with CPS biosynthesis and export are sometimes found in other chromosomal locations. The full combination of these genes is summarized as a "<a href="https://www.biorxiv.org/content/10.1101/2022.05.19.492579v1">K type</a>", described as a "predicted serotype associated with the best match locus". You can read more about this <a href="https://github.com/katholt/Kaptive/wiki/Databases-distributed-with-Kaptive#acinetobacter-baunannii-k-and-oc-locus-databases">in the Kaptive Wiki</a>.</p>
<p>Previously, <a href="https://journals.asm.org/doi/10.1128/jcm.27.12.2713-2716.1989">serotyping of <em>A. baumannii</em></a> focused on a major immunogenic polysaccharide which was considered the O antigen for the species. This serotyping approach appears to no longer be used and <a href="https://doi.org/10.1159/000007300">the serotyping scheme has not been updated in over 20 years</a>. Nonetheless, the O-antigen polysaccharide is attached to lipooligosaccharide, and the outer core (OC) of this lipooligosaccharide varies. Biosynthesis of the outer core lipooligosaccharide is encoded by a cluster of genes at the outer core (OC) locus.</p>
<p>Variation in the KL and OCL can be characterized with the <strong>Kaptive</strong> tool and its associated <a href="https://github.com/katholt/Kaptive/wiki/Databases-distributed-with-Kaptive#acinetobacter-baunannii-k-and-oc-locus-databases">databases</a> of numbered <em>A. baumannii</em> <a href="https://github.com/katholt/Kaptive/blob/master/extras/Acinetobacter_baumannii_KL_reference_information.pdf">K</a> and <a href="https://github.com/katholt/Kaptive/blob/master/extras/Acinetobacter_baumannii_OCL_reference_information.pdf">OC</a> locus variants. Kaptive takes in a genome assembly file (fasta), and assigns the K and OC locus to their numbered variants, provides a K type, and a description of genes in the K or OC loci and elsewhere in the chromosome, alongside metrics for quality of locus match. A description of <a href="https://github.com/katholt/Kaptive/wiki/How-does-Kaptive-work%3F">how Kaptive works</a>, <a href="https://github.com/katholt/Kaptive/wiki/How-to-run#summary-table">explanations of the full output reports</a>, and <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">resources for interpreting outputs</a> are available on the <a href="https://github.com/katholt/Kaptive/wiki/How-to-run#summary-table">Kaptive Wiki page</a>.</p>
<div class="admonition techdetails">
<p class="admonition-title">Kaptive Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/acinetobacter/task_kaptive.wdl">task_kaptive.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/katholt/Kaptive">Kaptive on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/katholt/Kaptive/wiki">Kaptive on GitHub</a></td>
</tr>
<tr>
<td>Orginal Publication(s)</td>
<td><a href="https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000339">Identification of Acinetobacter baumannii loci for capsular polysaccharide (KL) and lipooligosaccharide outer core (OCL) synthesis in genome assemblies using curated reference databases compatible with Kaptive</a><br><a href="https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000878">An update to the database for Acinetobacter baumannii capsular polysaccharide locus typing extends the extensive and diverse repertoire of genes found at and outside the K locus</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>abricate_abaum</code>: Plasmid Identification</summary>
<p><em>Acinetobacter</em> plasmids are not included in the PlasmidFinder database (see <a href="#plasmid-identification">the above section on Plasmid Identification</a>). Instead, the <a href="https://github.com/MehradHamidian/AcinetobacterPlasmidTyping">AcinetobacterPlasmidTyping</a> database contains variants of the plasmid <em>rep</em> gene for <em>A. baumannii</em> plasmid identification. When matched with &gt;/= 95 % identity, this represents a typing scheme for <em>Acinetobacter baumannii</em> plasmids.</p>
<p>The bioinformatics software for querying sample assemblies against the AcinetobacterPlasmidTyping database is <a href="https://github.com/tseemann/abricate">ABRicate</a>. By default, a 95% minimum identity threshold is set in order for successful classification.</p>
<div class="admonition techdetails">
<p class="admonition-title">abricate_abaum Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/drug_resistance/task_abricate.wdl">task_abricate.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/MehradHamidian/AcinetobacterPlasmidTyping/tree/v1.0.0">AcinetobacterPlasmidTyping Database on GitHub</a><br><a href="https://github.com/tseemann/abricate">ABRicate on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/MehradHamidian/AcinetobacterPlasmidTyping/tree/v1.0.0">AcinetobacterPlasmidTyping Database on GitHub</a><br><a href="https://github.com/tseemann/abricate">ABRicate on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>AcinetobacterPlasmidTyping database</em>: <a href="https://journals.asm.org/doi/10.1128/spectrum.02478-22?url_ver=Z39.88-2003&amp;rfr_id=ori:rid:crossref.org&amp;rfr_dat=cr_pub%20%200pubmed">Detection and Typing of Plasmids in&nbsp;<em>Acinetobacter baumannii</em>&nbsp;Using&nbsp;<em>rep</em>&nbsp;Genes Encoding Replication Initiation Proteins</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Escherichia</em> or <em>Shigella</em> spp.</summary>
<h5 id="escherichia-or-shigella"><em>Escherichia</em> or <em>Shigella</em> spp.<a class="headerlink" href="#escherichia-or-shigella" title="Permanent link"></a></h5>
<p>The <em>Escherichia</em> and <em>Shigella</em> genera are <a href="https://www.sciencedirect.com/science/article/abs/pii/S1286457902016374">difficult to differentiate as they do not comply with genomic definitions of genera and species</a>. Consequently, when either <em>Escherichia</em> or <em>Shigella</em> are identified by GAMBIT, all tools intended for these taxa are used. </p>
<div class="admonition tip">
<p><code>SerotypeFinder</code> and <code>ECTyper</code> are intended for analysis of <em>E. coli</em>. Both tools are used as there are occasional discrepancies between the serotypes predicted. This primarily arises due to differences in the databases used by each tool.</p>
</div>
<details class="task">
<summary><code>SerotypeFinder</code>: Serotyping</summary>
<p><a href="https://bitbucket.org/genomicepidemiology/serotypefinder/src/master/">SerotypeFinder</a>, from the Centre for Genomic Epidemiology (CGE), identifies the serotype of total or partially-sequenced isolates of <em>E. coli</em>. By using BLAST and KMA, the SerotypeFinder database of specific O-antigen processing system genes (for O typing) and flagellin genes (for H typing) is queried and compared against the target sequence to identify the closest serotype.</p>
<div class="admonition techdetails">
<p class="admonition-title">SerotypeFinder Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/escherichia_shigella/task_serotypefinder.wdl">task_serotypefinder.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://bitbucket.org/genomicepidemiology/serotypefinder/src/master/">SerotypeFinder on BitBucket</a><br><a href="https://bitbucket.org/genomicepidemiology/serotypefinder_db/src/master/">SerotypeFinder Database on BitBucket</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bitbucket.org/genomicepidemiology/serotypefinder/src/master/">SerotypeFinder on BitBucket</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://journals.asm.org/doi/10.1128/JCM.00008-15">Rapid and Easy In Silico Serotyping of Escherichia coli Isolates by Use of Whole-Genome Sequencing Data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ECTyper</code>: Serotyping</summary>
<p><a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper</a> is a serotyping module for <em>E. coli</em>. ECTyper provides species identification and quality control for <em>E. coli</em>, allowing for complete reports on serotyping, Shiga toxin typing, and pathotyping. Pathotype is identified using an ECTyper internal typing database that looks at toxin and pathotype signature marker sequences. Pathotypes are distinct categories defined by specific virulence factors used to group <em>E. coli</em> specimens based on pathogenicity. </p>
<div class="admonition techdetails">
<p class="admonition-title">ECTyper Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/escherichia_shigella/task_ectyper.wdl">task_ectyper.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/phac-nml/ecoli_serotyping">ECTyper on GitHub</a></td>
</tr>
<tr>
<td>Orginal publication</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8767331/">ECTyper: in silico Escherichia coli serotype and species prediction from raw and assembled whole-genome sequence data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition tip">
<p><code>VirulenceFinder</code> identifies virulence genes in total or partial sequenced isolates of bacteria. Currently, only <em>E. coli</em> is supported in TheiaProk workflows. </p>
</div>
<details class="task">
<summary><code>VirulenceFinder</code>: Virulence Gene Identification</summary>
<p>VirulenceFinder, from the Center for Genomic Epidemiology (GCE) in TheiaProk is only run on assembly files due to issues regarding discordant results when using read files on the web application versus the command-line. VirulenceFinder uses BLAST and KMA to query against a database of virulence genes in <em>E. coli</em> to identify any virulence factors in the sample.</p>
<div class="admonition techdetails">
<p class="admonition-title">VirulenceFinder Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/escherichia_shigella/task_virulencefinder.wdl">task_virulencefinder.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://bitbucket.org/genomicepidemiology/virulencefinder/src/master/">VirulenceFinder on BitBucket</a><br><a href="https://bitbucket.org/genomicepidemiology/virulencefinder_db/src/master/">VirulenceFinder Database on BitBucket</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bitbucket.org/genomicepidemiology/virulencefinder/src/master/">VirulenceFinder on BitBucket</a><br><a href="https://bitbucket.org/genomicepidemiology/virulencefinder_db/src/master/">VirulenceFinder Database on BitBucket</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/24574290/">Real-time whole-genome sequencing for routine typing, surveillance, and outbreak detection of verotoxigenic Escherichia coli</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition tip">
<p><code>ShigaTyper</code> and <code>ShigEiFinder</code> are intended for differentiation and serotype prediction for any <em>Shigella</em> species and Enteroinvasive <em>Escherichia coli</em> (EIEC). You can read about differences between these <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC517479/">here</a> and <a href="https://www.microbiologyresearch.org/content/journal/micro/10.1099/00221287-144-9-2667">here</a>. ShigEiFinder can be run using either the assembly (default) or reads. These tasks will report if the samples are neither <em>Shigella</em> nor EIEC.</p>
</div>
<details class="task">
<summary><code>ShigaTyper</code>: <em>Shigella</em>/EIEC differentiation and serotyping <mark><em>for Illumina and ONT only</em></mark></summary>
<p>ShigaTyper predicts <em>Shigella</em> spp. serotypes from Illumina or ONT read data. If the genome is not <em>Shigella</em> or enteroinvasive <em>E. coli</em> (EIEC), the results from this tool will state this. In the notes it provides, it also reports on the presence of <em>ipaB</em>  which is suggestive of the presence of the "virulent invasion plasmid".</p>
<p>ShigaTyper works by mapping the sample sequence to a <em>Shigella</em> reference sequence database using minimap2. Specifically, serotype prediction is made through the serotype-specific <em>wzx</em> gene as O-antigen expression is dependent on this gene. Additional criteria are applied if the serotype could not be solely predicted by this gene; please see the publication for more details.</p>
<div class="admonition techdetails">
<p class="admonition-title">ShigaTyper Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/escherichia_shigella/task_shigatyper.wdl">task_shigatyper.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/CFSAN-Biostatistics/shigatyper">ShigaTyper on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/CFSAN-Biostatistics/shigatyper">ShigaTyper on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1128/AEM.00165-19">In Silico Serotyping Based on Whole-Genome Sequencing Improves the Accuracy of Shigella Identification</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ShigEiFinder</code>: <em>Shigella</em>/EIEC Differentiation and Serotyping <mark><em>using the assembly file as input</em></mark></summary>
<p>ShigEiFinder differentiates&nbsp;<em>Shigella</em> and enteroinvasive <em>E. coli</em> (EIEC) using cluster-specific genes, identifies some serotypes based on the presence of O-antigen and H-antigen genes (<em>wzx</em> and <em>wzy</em>), and predicts the number of virulence plasmids. It can serotype over 59 <em>Shigella</em> and 22 EIEC serotypes using BLAST and BWA.</p>
<div class="admonition techdetails">
<p class="admonition-title">ShigEiFinder Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/escherichia_shigella/task_shigeifinder.wdl">task_shigeifinder.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/LanLab/ShigEiFinder">ShigEiFinder on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/LanLab/ShigEiFinder">ShigEiFinder on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/34889728/">Cluster-specific gene markers enhance Shigella and enteroinvasive Escherichia coli in silico serotyping</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ShigEiFinder_reads</code>: <em>Shigella</em>/EIEC Differentiation and Serotyping <mark><em>using Illumina read files as input</em></mark> (optional)</summary>
<p>To activate the <code>shigeifinder_reads</code> task, set the <code>call_shigeifinder_reads_input</code> to be <code>true</code>. If set to <code>true</code>, <code>shigeifinder_reads</code> will run <strong>in addition to</strong> the assembly-based <code>shigeifinder</code> task.</p>
<p>ShigEiFinder differentiates&nbsp;<em>Shigella</em> and enteroinvasive <em>E. coli</em> (EIEC) using cluster-specific genes, identifies some serotypes based on the presence of O-antigen and H-antigen genes (<em>wzx</em> and <em>wzy</em>), and predicts the number of virulence plasmids. It can serotype over 59 <em>Shigella</em> and 22 EIEC serotypes using BLAST and BWA.</p>
<div class="admonition techdetails">
<p class="admonition-title">ShigEiFinder Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/escherichia_shigella/task_shigeifinder.wdl">task_shigeifinder.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/LanLab/ShigEiFinder">ShigEiFinder on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/LanLab/ShigEiFinder">ShigEiFinder on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/34889728/">Cluster-specific gene markers enhance Shigella and enteroinvasive Escherichia coli in silico serotyping</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition tip">
<p><code>SonneiTyper</code> is run only when GAMBIT predicts the <em>S. sonnei</em> species. This is the most common <em>Shigella</em> species in the United States.</p>
</div>
<details class="task">
<summary><code>SonneiTyper</code>: <em>Shigella sonnei</em> identification, genotyping, and resistance mutation identification <mark><em>for Illumina and ONT data only</em></mark></summary>
<p>SonneiTyper identifies <em>Shigella sonnei</em>, and uses <strong>single-nucleotide variants</strong> for genotyping and prediction of quinolone resistance in <em>gyrA</em> (S83L, D87G, D87Y) and <em>parC</em> (S80I). Outputs are provided in <a href="https://github.com/katholt/sonneityping#example-output">a TSV format described here</a>.</p>
<p>SonneiTyper is a wrapper script around another tool, Mykrobe, that analyses the <em>S. sonnei</em> genomes using the <em>S. sonnei</em>-specific genotyping scheme. SonneiTyper parses the Mykrobe predict results and tabulates the results.</p>
<div class="admonition techdetails">
<p class="admonition-title">SonneiTyper Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/escherichia_shigella/task_sonneityping.wdl">task_sonneityping.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/Mykrobe-tools/mykrobe">Mykrobe on GitHub</a><br><a href="https://github.com/katholt/sonneityping">sonneityping on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/Mykrobe-tools/mykrobe/wiki">Mykrobe Wiki</a><br><a href="https://github.com/katholt/sonneityping">sonneityping on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>Mykrobe tool</em>: <a href="https://doi.org/10.12688/wellcomeopenres.15603.1">Antibiotic resistance prediction for Mycobacterium tuberculosis from genome sequence data with Mykrobe</a><br><em>S. sonnei genotyping scheme</em>: <a href="https://www.nature.com/articles/s41467-021-22700-4">Global population structure and genotyping framework for genomic surveillance of the major dysentery pathogen,&nbsp;<em>Shigella sonnei</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition tip">
<p><strong>Shigella XDR prediction.</strong> Please see <a href="#amr-characterization">the documentation section above for ResFinder</a> for details regarding this taxa-specific analysis. </p>
</div>
<details class="task">
<summary><code>StxTyper</code>: Identification and Typing of Shiga toxin (Stx) Genes <mark><em>using the assembly file as input</em></mark></summary>
<p>StxTyper screens bacterial genome assemblies for shiga toxin genes and subtypes them into known subtypes and also looks for novel subtypes in cases where the detected sequences diverge from the reference sequences.</p>
<p>Shiga toxin is the main virulence factor of Shiga-toxin-producing E. coli (STEC), though these genes are also found in Shigella species as well as some other genera more rarely, such as Klebsiella. <a href="https://doi.org/10.3390/microorganisms12040687">Please see this review paper that describes shiga toxins in great detail</a>.</p>
<div class="admonition tip">
<p class="admonition-title">Running StxTyper via the TheiaProk workflows</p>
<p>The TheiaProk workflow will automatically run <code>stxtyper</code> on all E. coli and Shigella spp. samples, but <em>the user can opt-in to running the tool on any sample by setting the optional input variable <code>call_stxtyper</code> to <code>true</code> when configuring the workflow.</em></p>
</div>
<p>Generally, <code>stxtyper</code> looks for <em>stxA</em> and <em>stxB</em> subunits that compose a complete operon. The A subunit is longer (in amino acid length) than the B subunit. Stxtyper attempts to detect these, compare them to a database of known sequences, and type them based on amino acid composition.  There typing algorithm and rules defining how to type these genes &amp; operons will be described more completely in a publication that will be available in the future.</p>
<p>The <code>stxtyper_report</code> output TSV is provided in <a href="https://github.com/ncbi/stxtyper/tree/v1.0.24?tab=readme-ov-file#output">this output format.</a></p>
<p>This tool has been incorporated into v4.0.3 of AMRFinderPlus and runs behind-the-scenes when the user (or in this case, the TheiaProk workflow) provides the <code>amrfinder --organism Escherichia --plus</code> options.</p>
<div class="admonition techdetails">
<p class="admonition-title">StxTyper Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/escherichia_shigella/task_stxtyper.wdl">task_stxtyper.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/stxtyper">ncbi/stxtyper GitHub repository</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/stxtyper">ncbi/stxtyper GitHub repository</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Haemophilus influenzae</em></summary>
<h5 id="haemophilus-influenzae"><em>Haemophilus influenzae</em><a class="headerlink" href="#haemophilus-influenzae" title="Permanent link"></a></h5>
<details class="task">
<summary><code>hicap</code>: Serotyping</summary>
<p><code>hicap</code> identifies the&nbsp;<em>cap</em>&nbsp;locus serotype in&nbsp;<em>Haemophilus influenzae</em>&nbsp;assemblies. As described in the <code>hicap</code> documentation:</p>
<blockquote>
<p>The&nbsp;<em>cap</em>&nbsp;locus of&nbsp;<em>H. influenzae</em>&nbsp;is categorised into 6 different groups based on serology (a-f). There are three functionally distinct regions of the&nbsp;<em>cap</em>&nbsp;locus, designated&nbsp;<code>region I</code>,&nbsp;<code>region II</code>, and&nbsp;<code>region III</code>. Genes within&nbsp;<code>region I</code>&nbsp;(<code>bexABCD</code>) and&nbsp;<code>region III</code>&nbsp;(<code>hcsAB</code>) are associated with transport and post-translation modification. The&nbsp;<code>region II</code>&nbsp;genes encode serotype-specific proteins, with each serotype (a-f) having a distinct set of genes.&nbsp;<em>cap</em>&nbsp;loci are often subject to structural changes (e.g. duplication, deletion) making the process of&nbsp;<em>in silico</em>&nbsp;typing and characterisation of loci difficult.</p>
</blockquote>
<div class="admonition techdetails">
<p class="admonition-title">hicap Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/haemophilus/task_hicap.wdl">task_hicap.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/scwatts/hicap">hicap on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/scwatts/hicap">hicap on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1128/jcm.00190-19">hicap: In Silico Serotyping of the Haemophilus influenzae Capsule Locus</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Klebsiella</em> spp.</summary>
<h5 id="klebsiella"><em>Klebsiella</em> spp.<a class="headerlink" href="#klebsiella" title="Permanent link"></a></h5>
<details class="task">
<summary><code>Kleborate</code>: Species Identification, MLST, Serotyping, AMR and Virulence Characterization</summary>
<p><a href="https://github.com/katholt/Kleborate">Kleborate</a> is a tool to identify the <em>Klebsiella</em> species, MLST sequence type, serotype, virulence factors (ICE_Kp_ and plasmid associated), and AMR genes and mutations. Serotyping is based on the capsular (K antigen) and lipopolysaccharide (LPS) (O antigen) genes. The acquired resistance genes identified by Kleborate can be found in <a href="https://kleborate.readthedocs.io/en/latest/kpsc_modules.html#acquired-amr-genes">the Kleborate documentation here</a>, along with other useful information regarding all of Kleborate's modules.</p>
<p><code>Kaptive</code> can be run as well by setting the optional input variable <code>kleborate_skip_kaptive</code> to <code>false</code> in order to recieve the K antigen and O antigen locus typing via <em>wzi</em> alleles.</p>
<div class="admonition techdetails">
<p class="admonition-title">Kleborate Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/klebsiella/task_kleborate.wdl">task_kleborate.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/klebgenomics/Kleborate">Kleborate on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://kleborate.readthedocs.io/en/latest/">Kleborate Documentation on ReadTheDocs</a></td>
</tr>
<tr>
<td>Original publication</td>
<td><a href="https://www.nature.com/articles/s41467-021-24448-3">A genomic surveillance framework and genotyping tool for Klebsiella pneumoniae and its related species complex</a><br><a href="https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000102">Identification of Klebsiella capsule synthesis loci from whole genome data</a><br></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Legionella pneumophila</em></summary>
<h5 id="legionella-pneumophila"><em>Legionella pneumophila</em><a class="headerlink" href="#legionella-pneumophila" title="Permanent link"></a></h5>
<details class="task">
<summary><code>Legsta</code>: Sequence-based typing</summary>
<p><a href="https://github.com/tseemann/legsta">Legsta</a> performs a sequence-based typing (SBT) of <em>Legionella pneumophila</em>, with the intention of being used for outbreak investigations. This tool outputs the allele number for seven genes (<em>flaA</em>, <em>pilE</em>, <em>asd</em>, <em>mip</em>, <em>mompS</em>, <em>proA</em>, and <em>neuA</em>), and combines the identified alleles to determine the overall SBT of the sample. Over 2,794 SBTs are able to be identified using this tool. Alleles with no <em>in silico</em> products are denoted with <code>-</code>s and novel alleles are listed with <code>?</code>s.</p>
<div class="admonition techdetails">
<p class="admonition-title">Legsta Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/legionella/task_legsta.wdl">task_legsta.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/tseemann/legsta">Legsta on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/tseemann/legsta">Legsta on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Listeria monocytogenes</em></summary>
<h5 id="listeria-monocytogenes"><em>Listeria monocytogenes</em><a class="headerlink" href="#listeria-monocytogenes" title="Permanent link"></a></h5>
<details class="task">
<summary><code>LisSero</code>: Serogroup Prediction</summary>
<p><a href="https://github.com/MDU-PHL/LisSero">LisSero</a> performs serogroup prediction for <em>Listeria monocytogenes</em> based on the presence or absence of five genes, <em>lmo1118</em>, <em>lmo0737</em>, ORF2110, ORF2819, and <em>Prs</em>. LisSero does not predict somatic (O) or flagellar (H) biosynthesis.</p>
<p>LisSero uses BLAST to query the sample against a built-in database to determine the most likely serogroup. </p>
<details class="toggle">
<summary>Serogroup Decision Tree</summary>
<p>The following decision tree is used to determine the serogroup for the sample.
</p><div class="language-text highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>if not Prs:
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>    stop
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>elif lmo0737 and not (ORF2819 or ORF2110):
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>    if lmo1118:
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>        Serogroup 1/2c, 3c
</span><span id="__span-0-6"><a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>    else:
</span><span id="__span-0-7"><a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>        Serogroup 1/2a, 3a
</span><span id="__span-0-8"><a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>elif ORF2819 and not (lmo0737 or lmo1118):
</span><span id="__span-0-9"><a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>    if ORF2110:
</span><span id="__span-0-10"><a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a>        Serogroup 4b, 4d, 4e
</span><span id="__span-0-11"><a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a>    else:
</span><span id="__span-0-12"><a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a>        Serogroup 1/2b, 3b, 7
</span><span id="__span-0-13"><a id="__codelineno-0-13" name="__codelineno-0-13" href="#__codelineno-0-13"></a>elif lmo0737 and ORF2819 and ORF2110:
</span><span id="__span-0-14"><a id="__codelineno-0-14" name="__codelineno-0-14" href="#__codelineno-0-14"></a>    Serogroup 4b, 4d, 4e*
</span><span id="__span-0-15"><a id="__codelineno-0-15" name="__codelineno-0-15" href="#__codelineno-0-15"></a>else:
</span><span id="__span-0-16"><a id="__codelineno-0-16" name="__codelineno-0-16" href="#__codelineno-0-16"></a>    Nontypable
</span></code></pre></div><p></p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">LisSero Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/listeria/task_lissero.wdl">task_lissero.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/MDU-PHL/LisSero">LisSero on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/MDU-PHL/LisSero">LisSero on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Mycobacterium tuberculosis</em></summary>
<h5 id="mycobacterium-tuberculosis"><em>Mycobacterium tuberculosis</em><a class="headerlink" href="#mycobacterium-tuberculosis" title="Permanent link"></a></h5>
<details class="task">
<summary><code>Clockwork</code>: Read Decontamination <mark><em>for Illumina PE only</em></mark></summary>
<p><a href="https://github.com/iqbal-lab-org/clockwork/wiki">Clockwork</a> decontaminates paired-end <em>Mycobacterium tuberculosis</em> data by removing all non-TB reads. At a high level, the sample is processed by aligning the reads with BWA to the H37Rv reference genome, and retaining only reads that have been mapped. This greatly improves the quality of any called variants and ensures that any variants called by TBProfiler are of suitable reliability.</p>
<div class="admonition techdetails">
<p class="admonition-title">Clockwork Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/mycobacterium/task_clockwork.wdl">task_clockwork.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/iqbal-lab-org/clockwork">Clockwork on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/iqbal-lab-org/clockwork/wiki">Clockwork Wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>Clockwork tool</em>: <a href="https://doi.org/10.1186/s13059-022-02714-x">Minos: variant adjudication and joint genotyping of cohorts of bacterial genome</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>TBProfiler</code>: Lineage and Drug Susceptibility Prediction <mark><em>for Illumina and ONT only</em></mark></summary>
<p><a href="https://github.com/jodyphelan/TBProfiler">TBProfiler</a> identifies <em>Mycobacterium tuberculosis</em> complex species, lineages, sub-lineages and drug resistance-associated mutations.</p>
<p>TBProfiler aligns the input reads to the H37Rv (NC_000962.3/AL123456.3) reference genome with <code>bwa mem</code> (or <code>minimap2</code>) and then calls variants using <code>gatk</code> (default), though other options are available (<code>bcftools</code>, <code>freebayes</code>, <code>lofreq</code>, <code>pilon</code>). After mutations are called and filtered, they are compared against <a href="https://github.com/jodyphelan/tbdb">TBProfiler's database (TBDB)</a>.</p>
<p>A number of outputs are made available from TBProfiler, all of which can be found in the summary results JSON file. Although the JSON file contains the most information, it is not very human readable, which is why the results CSV and TXT files have been made available to the user as outputs. TBProfiler is able to detect the identified lineage and any sublineages, determine the predicted type of drug resistance, a lists the genes that are associated with resistance, along with several other useful outputs.     </p>
<div class="admonition techdetails">
<p class="admonition-title">TBProfiler Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/mycobacterium/task_tbprofiler.wdl">task_tbprofiler.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/jodyphelan/TBProfiler">TBProfiler on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://jodyphelan.github.io/tb-profiler-docs/en/">TBProfiler Docs on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-019-0650-x">Integrating informatics tools and portable sequencing technology for rapid detection of resistance to anti-tuberculous drugs</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>tbp-parser</code>: Interpretation and Parsing of TBProfiler JSON outputs (optional)</summary>
<p>To activate this task, set <code>call_tbp_parser</code> to <code>true</code></p>
<p><a href="https://github.com/theiagen/tbp-parser/">tbp-parser</a> was developed by Theiagen in partnership with the California Department of Public Health. This tool adds useful drug resistance interpretation by applying expert rules and organizing the outputs from TBProfiler. To understand this module and its functions, <a href="https://theiagen.github.io/tbp-parser/latest/">please examine the documentation for this tool</a>. </p>
<p>This tool generates reports that can be automatically imported into your local LIMS system and is highly customizable with additional quality control metrics and fine-tuning. It is highly recommended to read the associated documentation to understand the full capabilities and configuration options available.</p>
<p><em>Please note that this tool has <strong>not</strong> been tested on ONT data and although it is available, result accuracy should be considered carefully.</em></p>
<div class="admonition techdetails">
<p class="admonition-title">tbp-parser Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/mycobacterium/task_tbp_parser.wdl">task_tbp_parser.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/theiagen/tbp-parser/">tbp-parser on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://theiagen.github.io/tbp-parser">tbp-parser Documentation</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Neisseria</em> spp.</summary>
<h5 id="neisseria"><em>Neisseria</em> spp.<a class="headerlink" href="#neisseria" title="Permanent link"></a></h5>
<details class="task">
<summary><code>ngmaster</code>: <em>Neisseria gonorrhoeae</em> Sequence Typing</summary>
<p>NG-MAST is currently the most widely used method for epidemiological surveillance of&nbsp;<em>Neisseria gonorrhoea</em>. This tool is targeted at clinical and research microbiology laboratories that have performed WGS of&nbsp;<em>N. gonorrhoeae</em> isolates and wish to understand the molecular context of their data in comparison to previously published epidemiological studies. As WGS becomes more routinely performed,&nbsp;NGMASTER&nbsp;has been developed to completely replace PCR-based NG-MAST, reducing time and labor costs.</p>
<p>NG-STAR offers a standardized method of classifying seven well-characterized genes associated antimicrobial resistance in <em>N. gonorrhoeae</em> (<em>penA</em>, <em>mtrR</em>, <em>porB</em>, <em>ponA</em>, <em>gyrA</em>, <em>parC</em> and 23S rRNA) to three classes of antibiotics (cephalosporins, macrolides and fluoroquinolones).</p>
<p>ngmaster combines two tools: NG-MAST (<em>in silico</em> multi-antigen sequencing typing) and NG-STAR (sequencing typing for antimicrobial resistance) and returns the results from both tools.</p>
<div class="admonition techdetails">
<p class="admonition-title">ngmaster Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/neisseria/task_ngmaster.wdl">task_ngmaster.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/MDU-PHL/ngmaster">ngmaster on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/MDU-PHL/ngmaster">ngmaster on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000076">NGMASTER: <em>in silico</em> multi-antigen sequence typing for <em>Neisseria gonorrhoeae</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>meningotype</code>: <em>Neisseria meningitidis</em> Serotyping</summary>
<p>This tool performs <em>in silico</em> typing of <em>N. meningitidis</em>. It performs the following functions: serogrouping, finetyping of <em>porA</em> and <em>fetA</em>, <em>porB</em> sequencing typing, and Bexsero Antigen Sequencing Typing (BAST) (<em>fHbp</em>, <em>NHBA</em>, <em>NadA</em>, and <em>PorA</em>). These results are all parsed and provided to the user as outputs.</p>
<p>The allele database is extracted from PubMLST's <em>N. meningitidis</em> database. This tool works by using BLAST to compare the sample sequence against the database entries.</p>
<div class="admonition techdetails">
<p class="admonition-title">meningotype Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/neisseria/task_meningotype.wdl">task_meningotype.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/MDU-PHL/meningotype">meningotype on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/MDU-PHL/meningotype">meningotype on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Pseudomonas aeruginosa</em></summary>
<h5 id="pseudomonas-aeruginosa"><em>Pseudomonas aeruginosa</em><a class="headerlink" href="#pseudomonas-aeruginosa" title="Permanent link"></a></h5>
<details class="task">
<summary><code>pasty</code>: Serotyping</summary>
<p><code>pasty</code> is a tool for <em>in silico</em> serogrouping of <em>Pseudomonas aeruginosa</em> isolates. <code>pasty</code> was developed by Robert Petit, based on the <a href="https://github.com/Sandramses/PAst">PAst</a> tool from the Centre for Genomic Epidemiology.</p>
<p><code>pasty</code> uses the <a href="https://github.com/rpetit3/camlhmp"><code>camlhmp</code></a> tool to identify the serogroup. It uses BLAST to compare an input asssembly against a set of O-antigens. The serogroup can be predicted based off of those results.</p>
<div class="admonition techdetails">
<p class="admonition-title">pasty Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/pseudomonas/task_pasty.wdl">task_pasty.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/pasty">pasty on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/pasty">pasty on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://journals.asm.org/doi/10.1128/JCM.00349-16">Application of Whole-Genome Sequencing Data for O-Specific Antigen Analysis and In Silico Serotyping of Pseudomonas aeruginosa Isolates.</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Salmonella</em> spp.</summary>
<h5 id="salmonella"><em>Salmonella</em> spp.<a class="headerlink" href="#salmonella" title="Permanent link"></a></h5>
<div class="admonition tip">
<p>Both SISTR and SeqSero2 are used for serotyping all <em>Salmonella</em> spp. Occasionally, the predicted serotypes may differ between SISTR and SeqSero2. When this occurs, differences are typically small and analogous, and are likely as a result of differing source databases. More information about Salmonella serovar nomenclature can be found <a href="https://www.happykhan.com/posts/binfie-guide-serovar/">here</a>.</p>
</div>
<details class="task">
<summary><code>SISTR</code>: <em>Salmonella</em> Serovar Prediction</summary>
<p><a href="https://github.com/phac-nml/sistr_cmd">SISTR</a> performs <em>Salmonella spp</em>. serotype prediction using antigen gene and cgMLST gene alleles. In TheiaProk, SISTR is run on genome assemblies and uses the default database setting (which contains smaller "centroid" alleles or representative alleles instead of the full set of cgMLST alleles). The full set of cgMLST alleles can be activated by setting the <code>sistr_use_full_cgmlst_db</code> optional input variable to <code>true</code>. It also runs a QC module to determine the level of confidence in the serovar prediction (please see <a href="https://github.com/phac-nml/sistr_cmd#qc-by-sistr_cmd---qc">the section on QC in the SISTR documentation here</a>).</p>
<p>SISTR uses a database of <em>Salmonella</em> serovar determination antigens, cgMLST profiles, and MASH sketches of appropriate reference genomes. BLAST is used to compare input assemblies to this database for serotyping, and the <a href="https://mash.readthedocs.io/en/latest/">Mash MinHash</a> algorithm is used for serovar prediction.</p>
<div class="admonition techdetails">
<p class="admonition-title">SISTR Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/salmonella/task_sistr.wdl">task_sistr.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/phac-nml/sistr_cmd">SISTR on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/phac-nml/sistr_cmd">SISTR on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0147101">The <em>Salmonella In Silico</em> Typing Resource (SISTR): an open web-accessible tool for rapidly typing and subtyping draft <em>Salmonella</em> genome assemblies.</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>SeqSero2S</code>: Serotyping</summary>
<p><a href="https://github.com/LSTUGA/SeqSero2S">SeqSero2S</a> is a tool for <em>Salmonella</em> serotype prediction. SeqSero2S is a software package that determines serotype antigens by finding the genes responsible for the flagellar filament (H antigen) and the outermost oligosaccharides of LPS (O antigen) in <em>Salmonella</em> and matches them to known representative alleles for those genes in a curated database. An antigenic formula describes all serotypes; <em>Salmonella enterica</em> subspecies <em>enterica</em> serotypes are also assigned a serotype name by the genotyping tool.</p>
<p>In the TheiaProkIllumina workflows, SeqSero2S takes in raw sequencing reads and performs targeted assembly of serotype determinant alleles, which can be used to predict serotypes including contamination between serotypes. For the TheiaProk ONT and FASTA workflows, SeqSero2S uses the genome assembly as input.</p>
<p>If reads are provided, SeqSero2S performs allele micro-assembly by default. This occurs through targeted assembly of serotype determinant alleles, and any assembled alleles are used to predict the sample's serotype, and can predict potential contamination. If the <code>seqsero2s_mode</code> optional variable is changed to <code>"k"</code> (for k-mer mode), SeqSero2S will perform serotyping based on unique k-mers of serotype determinants. If the input data is an assembly FASTA, the k-mer mode <em>must</em> be used, and the genome assembly is used to generate the search k-mers instead of the raw reads. </p>
<details class="dna">
<summary>What is the difference between SeqSero2 and SeqSero2S?</summary>
<p>Recently, the SeqSero2S software was released and adopted by the PulseNet 2.0 (PN2.0) platform for subspecies identification and serotype determination of <em>Salmonella</em> spp. Genetic determination of rarer serotypes can be problematic due to a lack of sequences for rare antigen types and alleles, a lack of understanding of the genetic basis for some antigens, or some inconsistencies in the White-Kauffmann-Le Minor (WKL) Scheme for <em>Salmonella</em> serotype designation. As such, SeqSero2S predicts serotypes using a simplified interpretation based on the most commonly seen serotypes. SeqSero2S also includes additional functionalities to improve genomic prediction of serotypes in general and mitigate potential drawbacks of the simplified scheme.</p>
<p>The simplification of SeqSero2S can be summarized as such:</p>
<ul>
<li>178 serotypes were provisionally removed<ul>
<li>84 serotypes, of which the exact antigenic formula could not be determined</li>
<li>66 serotypes had no alleles, or probes were not available</li>
<li>10 serotypes had no probes available for O:9,46,27</li>
<li>14 serotypes were removed because O:54 is encoded on a plasmid</li>
<li>4 serotypes could not be differentiated from other serotypes</li>
</ul>
</li>
<li>57 serotypes were merged into other serotypes</li>
<li>The antigenic formulae of 890 serotypes were simplified, but the simplifications did not result in merging</li>
</ul>
</details>
<div class="admonition techdetails">
<p class="admonition-title">SeqSero2S Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/salmonella/task_seqsero2s.wdl">task_seqsero2s.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/LSTUGA/SeqSero2S">SeqSero2S on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/LSTUGA/SeqSero2S">SeqSero2S on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://journals.asm.org/doi/10.1128/JCM.00323-15">Salmonella serotype determination utilizing high-throughput genome sequencing data.</a><br><a href="https://journals.asm.org/doi/10.1128/AEM.01746-19">SeqSero2: rapid and improved Salmonella serotype determination using whole genome sequencing data.</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>genotyphi</code>: <em>Salmonella</em> Typhi Characterization <mark><em>for Illumina and ONT only</em></mark></summary>
<p><a href="https://github.com/typhoidgenomics/genotyphi"><code>genotyphi</code></a> is activated upon the identification of the "Typhi" serotype by SISTR or SeqSero2S (via either the <code>seqsero2s_predicted_serotype</code>, or the <code>sistr_predicted_serotype</code>). <code>genotyphi</code> divides the <em>Salmonella enterica</em> serovar Typhi population into detailed lineages, clades, and subclades. It also detects mutations in the quinolone-resistance determining regions, acquired antimicrobial resistance genes, plasmid replicons, and subtypes of the IncHI1 plasmid which is associated with multidrug resistance.</p>
<p>This task uses <a href="https://github.com/typhoidgenomics/genotyphi/blob/main/README.md#typhi-mykrobe">Mykrobe</a> in order to perform k-mer based genotyping with a genotyping scheme specific to <em>Salmonella</em> Typhi, and then parses that output using the <code>genotyphi</code> tool. This scheme divides the <em>Salmonella</em> Typhi population into genotypes based on unique SNV markers.</p>
<div class="admonition techdetails">
<p class="admonition-title">genotyphi Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/salmonella/task_genotyphi.wdl">task_genotyphi.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/typhoidgenomics/genotyphi">genotyphi on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/typhoidgenomics/genotyphi/master/README.md">genotyphi on GitHub</a></td>
</tr>
<tr>
<td>Orginal publication(s)</td>
<td><a href="https://www.nature.com/articles/ncomms12827/">An extended genotyping framework for Salmonella enterica serovar Typhi, the cause of human typhoid</a><br><a href="https://academic.oup.com/jid/article/224/Supplement_7/S775/6358992?login=false">Five Years of GenoTyphi: Updates to the Global Salmonella Typhi Genotyping Framework</a><br><a href="https://www.biorxiv.org/content/10.1101/2024.09.30.613582v1">Typhi Mykrobe: fast and accurate lineage identification and antimicrobial resistance genotyping directly from sequence reads for the typhoid fever agent <em>Salmonella</em> Typhi</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Staphyloccocus aureus</em></summary>
<h5 id="staphyloccocus-aureus"><em>Staphyloccocus aureus</em><a class="headerlink" href="#staphyloccocus-aureus" title="Permanent link"></a></h5>
<details class="task">
<summary><code>spaTyper</code>: Sequence Typing</summary>
<p>spaTyper works by identifying the number and order of repeats in the <em>Staphylococcus</em> protein A gene (also known as spa) of a <em>S. aureus</em> assembly. The repeats are assigned a numerical code that is used to assign a spa-type. The tool uses the <a href="http://spaserver2.ridom.de/">Ridom SpaServer Database</a> in order to assign the appropriate spa-type (please note that this link is typically considered unsafe by modern web browsers).</p>
<p>Spa-types can be used for accurate and reliable typing of MRSA (methicillin-resistant <em>Staphylococcus aureus</em>) strains, and are often used in hospital infection control and epidemiological studies.</p>
<div class="admonition techdetails">
<p class="admonition-title">spatyper Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/staphylococcus/task_spatyper.wdl">task_spatyper.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/HCGB-IGTP/spaTyper">spaTyper on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/HCGB-IGTP/spaTyper">spaTyper on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>Ridom SpaServer database</em>: <a href="https://doi.org/10.1128/jcm.41.12.5442-5448.2003">Typing of methicillin-resistant <em>Staphylococcus aureus</em> in a university hospital setting by using novel software for spa repeat determination and database management</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>staphopia-sccmec</code>: Sequence Typing</summary>
<p>SCC_mec_ (staphylococcal cassette chromosome <em>mec</em>) is a mobile genetic element of <em>Staphylococcus</em> species. It includes a methicilin-resistant <em>mecA</em> gene that is shared between <em>Staphylococcus</em> strains via horizontal gene transfer, which leads to MRSA strains. SCC_mec_ has also been found to confer resistance to non-beta-lactam drugs as well, making it an important target for identifying antimicrobial resistance in <em>Staphylococcus aureus</em>.</p>
<p>This tool assigns a SCC_mec_ type by using BLAST to compare the SCC_mec_ primers against the provided <em>S. aureus</em> assembly. <code>staphopia-sccmec</code> reports&nbsp;<code>True</code> for exact primer matches and&nbsp;<code>False</code> all others. The <a href="https://en.wikipedia.org/wiki/Hamming_distance">Hamming Distance</a> is also used in TheiaProk's implementation of the tool to include the number of mismatches so that the <code>False</code> results can be examined in more detail.</p>
<div class="admonition techdetails">
<p class="admonition-title">staphopia-sccmec Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/staphylococcus/task_staphopiasccmec.wdl">task_staphopiasccmec.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/staphopia/staphopia-sccmec">staphopia-sccmec on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/staphopia/staphopia-sccmec">staphopia-sccmec on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.7717/peerj.5261"><em>Staphylococcus aureus</em> viewed from the perspective of 40,000+ genomes</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>AgrVATE</code>: Sequence Typing</summary>
<p>This tool identifies the accessory gene regulator (<em>agr</em>) locus type and reports possible variants in the <em>agr</em> operon. The <em>agr</em> system is a major regulator of virulence phenotypes in <em>Staphylococcus aureus</em>. Many <em>S. aureus</em> strains often have nonfunctional <em>agr</em> activity due to various loss-of-function mutations in the <em>agr</em> operon. This may be associated with increased disease severity, making its detection clinically significant.</p>
<p>AgrVATE workings by using BLAST to compare a database of unique <em>agr</em>-group-specific k-mers against the provided assembly. The <em>agr</em> operon is then extracted using <em>in-silico</em> PCR computational methods. If an <em>agr</em>-group is assigned, variants are called using an Agr-group specific reference operon in order to detect possible non-functional <em>agr</em>. If the <em>agr</em>-group was untypeable, the sequence is searched for potential non-canonical <em>agrD</em>.</p>
<div class="admonition techdetails">
<p class="admonition-title">AgrVATE Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/staphylococcus/task_agrvate.wdl">task_agrvate.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/VishnuRaghuram94/AgrVATE">AgrVATE on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/VishnuRaghuram94/AgrVATE">AgrVATE on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1128/spectrum.01334-21">Species-Wide Phylogenomics of the <em>Staphylococcus aureus</em> Agr Operon Revealed Convergent Evolution of Frameshift Mutations</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Streptococcus pneumoniae</em></summary>
<h5 id="streptococcus-pneumoniae"><em>Streptococcus pneumoniae</em><a class="headerlink" href="#streptococcus-pneumoniae" title="Permanent link"></a></h5>
<details class="task">
<summary><code>PopPUNK</code>: Global Pneumococcal Sequence Cluster Typing</summary>
<p>Global Pneumococcal Sequence Clusters (GPSC) define and name pneumococcal strains. Each GPSC is an <a href="https://doi.org/10.1016/j.ebiom.2019.04.021">international definition of a pneumococcal lineage</a> and can capture all variations across the entire genome, leading to better vaccine development due to increased disease surveillance. GPSC designation is undertaken using the PopPUNK (Population Partitioning Using Nucleotide K-mers) software and the GPSC database.</p>
<p>PopPUNK works by using variable-length k-mers to distinguish between sample divergences in shared genomic content. Clusters are assigned based on the resulting pairwise distance distributions. </p>
<div class="admonition tip">
<p class="admonition-title">Interpreting GPSC results</p>
<ul>
<li>In the&nbsp;<code>*_external_clusters.csv</code> novel clusters are assigned NA. For isolates that are assigned a novel cluster and pass QC, you can email&nbsp;<a href="mailto:globalpneumoseq@gmail.com">globalpneumoseq@gmail.com</a>&nbsp;to have these novel clusters added to the database.</li>
<li>Unsampled diversity in the pneumococcal population may represent missing variation that links two GPS clusters. When this is discovered, GPSCs are merged and the merge history is indicated. For example, if GPSC23 and GPSC362 merged, the GPSC would be reported as GPSC23, with a merge history of GPSC23;362.</li>
</ul>
</div>
<div class="admonition techdetails">
<p class="admonition-title">PopPUNK Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/streptococcus/task_poppunk_streppneumo.wdl">task_poppunk_streppneumo.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/bacpop/PopPUNK">PopPUNK on GitHub</a><br><a href="https://www.pneumogen.net/gps/#/">Global Pneumococcal Sequencing Project</a><br><a href="https://www.pneumogen.net/gps/#/gpsc#lineages">GPSC Database</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://poppunk-docs.bacpop.org/">PopPUNK Documentation</a><br><a href="https://www.pneumogen.net/gps/#/training">GPS Training Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>PopPUNK tool</em>: <a href="https://genome.cshlp.org/content/29/2/304">Fast and flexible bacterial genomic epidemiology with PopPUNK</a><br><em>GPSC database</em>: <a href="https://doi.org/10.1016/j.ebiom.2019.04.021">International genomic definition of pneumococcal lineages, to contextualise disease, antibiotic resistance and vaccine impact</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>SeroBA</code>: Serotyping <mark><em>for Illumina_PE only</em></mark></summary>
<p>SeroBA is a k-mer based method for serotyping using the capsular polysaccharide biosynthesis (<em>cps</em>) locus of <em>Streptococcus pneumoniae</em> from paired-end sequencing data. This locus encodes the serotype and is a major virulence factor for the species. Identifying circulating serotypes is important to determine the epidemiological trends and vaccine impact.</p>
<p>By adapting a database from <a href="https://github.com/ukhsa-collaboration/PneumoCaT">PneumoCaT (Pneumococcal Capsular Typing)</a>, SeroBA uses <a href="https://github.com/refresh-bio/KMC">KMC</a> to generate a k-mer database and then uses a capsular type variant database and an <a href="https://github.com/sanger-pathogens/ariba">ARIBA</a>-compatible database that clusters all serotypes by serogroups. A k-mer analysis is performed and the serotype with the highest normalized sequence converage is selected. ARIBA then is used to build an assembly to confirm the selected serotype from the read data and aligns the <em>cps</em> sequence against a reference to identify variants.</p>
<div class="admonition techdetails">
<p class="admonition-title">SeroBA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/streptococcus/task_seroba.wdl">task_seroba.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/sanger-pathogens/seroba">SeroBA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/sanger-pathogens/seroba">SeroBA on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000186">SeroBA: rapid high-throughput serotyping of Streptococcus pneumoniae from whole genome sequence data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>PBPtyper</code>: Penicillin-Binding Protein Genotyping</summary>
<p>The Penicillin-binding proteins (PBP) are responsible for the minimum inhibitory concentration (MIC) phenotypes for beta-lactam antibiotics, which are the drugs of choice for treating pneumococcal infections. In <em>Streptococcus pneumoniae</em>, these PBP genes (PBP1a, PBP2b, and PBP2x) can be identified and typed with PBPTyper to help predict beta-lactam resistance levels in <em>S. pneumoniae</em>, which can be invaluable in disease surveillance.</p>
<p>pbptyper uses BLAST and average nucleotide identity (ANI) to identify the closest matching PBP types from a database of known PBP types and uses the PBP typing scheme described by the linked publication below.</p>
<div class="admonition techdetails">
<p class="admonition-title">pbptyper Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/streptococcus/task_pbptyper.wdl">task_pbptyper.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/pbptyper">pbptyper on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/pbptyper">pbptyper on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>PBP typing method</em>: <a href="https://journals.asm.org/doi/full/10.1128/mBio.00756-16">Penicillin-Binding Protein Transpeptidase Signatures for Tracking and Predicting -Lactam Resistance Levels in <em>Streptococcus pneumoniae</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Streptococcus pyogenes</em></summary>
<h5 id="streptococcus-pyogenes"><em>Streptococcus pyogenes</em><a class="headerlink" href="#streptococcus-pyogenes" title="Permanent link"></a></h5>
<details class="task">
<summary><code>emm-typing-tool</code>: Sequence Typing <mark><em>for Illumina_PE only</em></mark></summary>
<p>The <em>Streptococcus pyogenes</em> M protein (encoded by <em>emm</em>) is used for sequencing typing and disease surveillence as it is a major virulence factor. There are over 275 <em>emm</em> types.</p>
<p><code>emm-typing-tool</code> maps the reads to a CDC-curated <em>emm</em> type database using bowtie2 to identify any <em>emm</em> genes. Alleles with 100% coverage and over 90% identity are selected, and the allele with the highest percent identity is generally reported (see the <a href="https://github.com/ukhsa-collaboration/emm-typing-tool/blob/master/decision_algorithm.png">decision tree</a> for the nuances).</p>
<div class="admonition techdetails">
<p class="admonition-title">emm-typing-tool Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/streptococcus/task_emmtypingtool.wdl">task_emmtypingtool.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ukhsa-collaboration/emm-typing-tool">emm-typing-tool on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ukhsa-collaboration/emm-typing-tool">emm-typing-tool on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>emm typing scheme</em>: <a href="https://doi.org/10.1093/infdis/jiu260">A systematic and functional classification of Streptococcus pyogenes that serves as a new tool for molecular typing and vaccine development</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>emmtyper</code>: Sequence Typing</summary>
<p>The <em>Streptococcus pyogenes</em> M protein (encoded by <em>emm</em>) is used for sequencing typing and disease surveillence as it is a major virulence factor. There are over 275 <em>emm</em> types.</p>
<p><code>emmtyper</code> uses BLAST to compare the genome assembly to the CDC-curated trimmed <em>emm</em> type database (by default). An <em>in silico</em> PCR method is also available. The BLAST results are then processed to distinguish between <em>emm</em> and <em>emm</em>-like alleles to derive the isolates' M-type. The predicted <em>emm</em>-type is reported in addition to any possible <em>emm</em>-like alleles and the functional <em>emm</em> cluster.</p>
<div class="admonition techdetails">
<p class="admonition-title">emm-typing-tool Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/streptococcus/task_emmtyper.wdl">task_emmtyper.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/MDU-PHL/emmtyper/tree/master">emmtyper on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/MDU-PHL/emmtyper/tree/master">emmtyper on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>emm typing scheme</em>: <a href="https://doi.org/10.1093/infdis/jiu260">A systematic and functional classification of Streptococcus pyogenes that serves as a new tool for molecular typing and vaccine development</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<details class="toggle">
<summary><em>Vibrio</em> spp.</summary>
<h5 id="vibrio"><em>Vibrio</em> spp.<a class="headerlink" href="#vibrio" title="Permanent link"></a></h5>
<details class="task">
<summary><code>abricate_vibrio</code>: <em>Vibrio</em> Characterization</summary>
<p>The <code>abricate_vibrio</code> task is used to perform general characterization of <em>Vibrio</em> genomes using a database of target sequences that are traditionally used in PCR methods. The sequences included in the database are as follows:</p>
<details class="toggle">
<summary>Resistence Gene Database</summary>
<table>
<thead>
<tr>
<th>Sequence Name</th>
<th>Sequence Role</th>
<th>Purpose in Database</th>
</tr>
</thead>
<tbody>
<tr>
<td><em>toxR</em></td>
<td>Transcriptional activator</td>
<td>Species marker where presence identifies <em>V. cholerae</em></td>
</tr>
<tr>
<td><em>ompW</em></td>
<td>Outer Membrane Protein</td>
<td>Species marker where presence identifies <em>V. cholerae</em></td>
</tr>
<tr>
<td><em>ctxA</em></td>
<td>Cholera toxin</td>
<td>Indicates cholera toxin production</td>
</tr>
<tr>
<td><em>tcpA_classical</em></td>
<td>Toxin co-pilus A allele, associated with the Classical biotype</td>
<td>Used to infer identity as Classical biotype</td>
</tr>
<tr>
<td><em>tcpA_ElTor</em></td>
<td>Toxin co-pilus A allele, associated with the El Tor biotype</td>
<td>Used to infer identity as El Tor biotype</td>
</tr>
<tr>
<td><em>wbeN</em></td>
<td>O antigen encoding region</td>
<td>Used to infer identity as O1 serogroup</td>
</tr>
<tr>
<td><em>wbfR</em></td>
<td>O antigen encoding region</td>
<td>Used to infer identity as O139 serogroup</td>
</tr>
</tbody>
</table>
<p>This database was developed via communication with Dr. Christine Lee, of the National Listeria, Yersinia, Vibrio and Enterobacterales Reference Laboratory within the Enteric Diseases Laboratory Branch at CDC. It is identical to the database used in the <code>srst2</code> task except it is formatted for ABRicate.</p>
</details>
<p>This task works by using the ABRicate tool, which uses BLAST to compare the assembled genome to the target sequences in the database and then reporting the details of the genes that pass quality thresholds. The presence or absence of specific genes are used to verify the species, identify cholera toxin production, and designation both the biotype and serogroup of the sample. See the table above for the genes used for each of these purposes.</p>
<div class="admonition techdetails">
<p class="admonition-title">abricate_vibrio Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/vibrio/task_abricate_vibrio.wdl">task_abricate.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/tseemann/abricate">ABRicate on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/tseemann/abricate">ABRicate on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>SRST2</code>: <em>Vibrio</em> Characterization <mark><em>for Illumina only</em></mark></summary>
<p>SRST2 is used to perform general characterization of <em>Vibrio</em> genomes using a database of target sequences that are traditionally used in PCR methods. The sequences included in the database are as follows:</p>
<details class="toggle">
<summary>Resistance Gene Database</summary>
<table>
<thead>
<tr>
<th>Sequence Name</th>
<th>Sequence Role</th>
<th>Purpose in Database</th>
</tr>
</thead>
<tbody>
<tr>
<td><em>toxR</em></td>
<td>Transcriptional activator</td>
<td>Species marker where presence identifies <em>V. cholerae</em></td>
</tr>
<tr>
<td><em>ompW</em></td>
<td>Outer Membrane Protein</td>
<td>Species marker where presence identifies <em>V. cholerae</em></td>
</tr>
<tr>
<td><em>ctxA</em></td>
<td>Cholera toxin</td>
<td>Indicates cholera toxin production</td>
</tr>
<tr>
<td><em>tcpA_classical</em></td>
<td>Toxin co-pilus A allele, associated with the Classical biotype</td>
<td>Used to infer identity as Classical biotype</td>
</tr>
<tr>
<td><em>tcpA_ElTor</em></td>
<td>Toxin co-pilus A allele, associated with the El Tor biotype</td>
<td>Used to infer identity as El Tor biotype</td>
</tr>
<tr>
<td><em>wbeN</em></td>
<td>O antigen encoding region</td>
<td>Used to infer identity as O1 serogroup</td>
</tr>
<tr>
<td><em>wbfR</em></td>
<td>O antigen encoding region</td>
<td>Used to infer identity as O139 serogroup</td>
</tr>
</tbody>
</table>
<p>This database was developed via communication with Dr. Christine Lee, of the National Listeria, Yersinia, Vibrio and Enterobacterales Reference Laboratory within the Enteric Diseases Laboratory Branch at CDC. It is identical to the database used in the <code>abricate_vibrio</code> task except it is formatted for SRST2.</p>
</details>
<p>SRST2 works by mapping reads to the target sequences in the database and then reporting the details of all those genes that pass the quality thresholds. The presence or absence of specific genes are used to verify the species, identify cholera toxin production, and designation both the biotype and serogroup of the sample. See the table above for the genes used for each of these purposes.</p>
<div class="admonition techdetails">
<p class="admonition-title">SRST2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/vibrio/task_srst2_vibrio.wdl">task_srst2_vibrio.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/katholt/srst2">srst2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/katholt/srst2">srst2 on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1186/s13073-014-0090-6">SRST2: Rapid genomic surveillance for public health and hospital microbiology labs</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Vibecheck</code>: O1 <em>Vibrio  cholerae</em> Lineage Classification <mark><em>for Illumina PE only</em></mark></summary>
<p>The <code>Vibecheck</code> task classifies O1 <em>V. cholerae</em> sequences into canonical lineages (T1-T17) using variant frequency demixing. The O1 designation is determined through the use of the SRST2 task.</p>
<p>Vibecheck works by aligning the reads to an O1 <em>V. cholerae</em> reference genome, calling variants from the alignment, and estimating lineage abundances using <a href="../freyja/">Freyja</a> by using a database built from canonical SNPs that define the known lineages.</p>
<div class="admonition techdetails">
<p class="admonition-title">Vibecheck Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/vibrio/task_vibecheck_vibrio.wdl">task_vibecheck_vibrio.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/CholGen/Vibecheck">Vibecheck on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/CholGen/Vibecheck">Vibecheck on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</details>
<h3 id="outputs">Outputs<a class="headerlink" href="#outputs" title="Permanent link"></a></h3>
<div class="tabbed-set tabbed-alternate" data-tabs="5:4"><input checked="checked" id="theiaprok_illumina_pe_4" name="__tabbed_5" type="radio"><input id="theiaprok_illumina_se_3" name="__tabbed_5" type="radio"><input id="theiaprok_ont_4" name="__tabbed_5" type="radio"><input id="theiaprok_fasta_3" name="__tabbed_5" type="radio"><div class="tabbed-labels"><label for="theiaprok_illumina_pe_4">TheiaProk_Illumina_PE</label><label for="theiaprok_illumina_se_3">TheiaProk_Illumina_SE</label><label for="theiaprok_ont_4">TheiaProk_ONT</label><label for="theiaprok_fasta_3">TheiaProk_FASTA</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>abricate_abaum_database</td>
<td>String</td>
<td>Database of reference A. baumannii plasmid typing genes used for plasmid typing</td>
</tr>
<tr>
<td>abricate_abaum_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_abaum_plasmid_tsv</td>
<td>File</td>
<td><a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a> containing a row for each A. baumannii plasmid type gene found in the sample</td>
</tr>
<tr>
<td>abricate_abaum_plasmid_type_genes</td>
<td>String</td>
<td>A. baumannii Plasmid typing genes found in the sample; from GENE column in <a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a></td>
</tr>
<tr>
<td>abricate_abaum_version</td>
<td>String</td>
<td>Version of abricate used for A. baumannii plasmid typing</td>
</tr>
<tr>
<td>abricate_database</td>
<td>String</td>
<td>Database of reference used with Abricate</td>
</tr>
<tr>
<td>abricate_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_genes</td>
<td>String</td>
<td>Genes found in the sample; from GENE column in <a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a></td>
</tr>
<tr>
<td>abricate_results_tsv</td>
<td>File</td>
<td><a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a> containing a row for each gene found in the sample</td>
</tr>
<tr>
<td>abricate_version</td>
<td>String</td>
<td>Version of abricate used for A. baumannii plasmid typing</td>
</tr>
<tr>
<td>abricate_vibrio_biotype</td>
<td>String</td>
<td>Biotype classification according to tcpA gene sequence (Classical or ElTor)</td>
</tr>
<tr>
<td>abricate_vibrio_ctxA</td>
<td>String</td>
<td>Presence or absence of the ctxA gene</td>
</tr>
<tr>
<td>abricate_vibrio_database</td>
<td>String</td>
<td>Database to use while running abricate</td>
</tr>
<tr>
<td>abricate_vibrio_detailed_tsv</td>
<td>File</td>
<td>Detailed ABRicate output file</td>
</tr>
<tr>
<td>abricate_vibrio_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_vibrio_ompW</td>
<td>String</td>
<td>Presence or absence of the ompW gene</td>
</tr>
<tr>
<td>abricate_vibrio_serogroup</td>
<td>String</td>
<td>Serotype classification as O1 (wbeN gene), O139 (wbfR gene) or not detected.</td>
</tr>
<tr>
<td>abricate_vibrio_toxR</td>
<td>String</td>
<td>Presence or absence of the toxR gene</td>
</tr>
<tr>
<td>abricate_vibrio_version</td>
<td>String</td>
<td>The abricate version run</td>
</tr>
<tr>
<td>agrvate_agr_canonical</td>
<td>String</td>
<td>Canonical or non-canonical agrD</td>
</tr>
<tr>
<td>agrvate_agr_group</td>
<td>String</td>
<td>Agr group</td>
</tr>
<tr>
<td>agrvate_agr_match_score</td>
<td>String</td>
<td>Match score for agr group</td>
</tr>
<tr>
<td>agrvate_agr_multiple</td>
<td>String</td>
<td>If multiple agr groups were found</td>
</tr>
<tr>
<td>agrvate_agr_num_frameshifts</td>
<td>String</td>
<td>Number of frameshifts found in CDS of extracted agr operon</td>
</tr>
<tr>
<td>agrvate_docker</td>
<td>String</td>
<td>The docker used for AgrVATE</td>
</tr>
<tr>
<td>agrvate_results</td>
<td>File</td>
<td>A gzipped tarball of all results</td>
</tr>
<tr>
<td>agrvate_summary</td>
<td>File</td>
<td>The summary file produced</td>
</tr>
<tr>
<td>agrvate_version</td>
<td>String</td>
<td>The version of AgrVATE used</td>
</tr>
<tr>
<td>amr_search_csv</td>
<td>File</td>
<td>CSV formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_docker</td>
<td>String</td>
<td>Docker image used to run AMR_Search</td>
</tr>
<tr>
<td>amr_search_results</td>
<td>File</td>
<td>JSON formatted AMR profile including BLAST results</td>
</tr>
<tr>
<td>amr_search_results_pdf</td>
<td>File</td>
<td>PDF formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_version</td>
<td>String</td>
<td>Version of AMR_Search libraries used</td>
</tr>
<tr>
<td>amrfinderplus_all_report</td>
<td>File</td>
<td>Output TSV file from AMRFinderPlus (described <a href="https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#fields">https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#fields</a>)</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_betalactam_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to beta-lactams</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_carbapenem_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to carbapenem</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_cephalosporin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to cephalosporin</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_cephalothin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to cephalothin</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_methicillin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to methicilin</td>
</tr>
<tr>
<td>amrfinderplus_amr_classes</td>
<td>String</td>
<td>AMRFinderPlus predictions for classes of drugs that genes found in the reads are known to confer resistance to</td>
</tr>
<tr>
<td>amrfinderplus_amr_core_genes</td>
<td>String</td>
<td>AMR genes identified by AMRFinderPlus where the scope is "core"</td>
</tr>
<tr>
<td>amrfinderplus_amr_plus_genes</td>
<td>String</td>
<td>AMR genes identified by AMRFinderPlus where the scope is "plus"</td>
</tr>
<tr>
<td>amrfinderplus_amr_report</td>
<td>File</td>
<td>TSV file detailing AMR genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>amrfinderplus_amr_subclasses</td>
<td>String</td>
<td>More specificity about the drugs that genes identified in the reads confer resistance to</td>
</tr>
<tr>
<td>amrfinderplus_db_version</td>
<td>String</td>
<td>AMRFinderPlus database version used</td>
</tr>
<tr>
<td>amrfinderplus_stress_genes</td>
<td>String</td>
<td>Stress genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_stress_report</td>
<td>File</td>
<td>TSV file detailing stress genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>amrfinderplus_version</td>
<td>String</td>
<td>AMRFinderPlus version used</td>
</tr>
<tr>
<td>amrfinderplus_virulence_genes</td>
<td>String</td>
<td>Virulence genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_virulence_report</td>
<td>File</td>
<td>TSV file detailing virulence genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>ani_highest_percent</td>
<td>Float</td>
<td>Highest ANI between query and any given reference genome (top species match)</td>
</tr>
<tr>
<td>ani_highest_percent_bases_aligned</td>
<td>Float</td>
<td>Percentage of bases aligned between query genome and top species match</td>
</tr>
<tr>
<td>ani_mummer_docker</td>
<td>String</td>
<td>Docker image used to run the ANI_mummer task</td>
</tr>
<tr>
<td>ani_mummer_version</td>
<td>String</td>
<td>Version of MUMmer used</td>
</tr>
<tr>
<td>ani_output_tsv</td>
<td>File</td>
<td>Full output TSV from ani-m</td>
</tr>
<tr>
<td>ani_top_species_match</td>
<td>String</td>
<td>Species of genome with highest ANI to query FASTA</td>
</tr>
<tr>
<td>arln_assembly_ratio</td>
<td>String</td>
<td>Assembly ratio of the final assembly compared to the mean assembly length of the taxon. These stats are gathered from a statistics file created by the CDC aggregating NCBI assembly data by taxon.</td>
</tr>
<tr>
<td>arln_assembly_zscore</td>
<td>String</td>
<td>ZScore of the samples assembly length to assess the uniformity of the assembly when compared to taxon means.</td>
</tr>
<tr>
<td>arln_r1_q30_clean</td>
<td>String</td>
<td>Q30% of the cleaned R1 FASTQ</td>
</tr>
<tr>
<td>arln_r1_q30_raw</td>
<td>String</td>
<td>Q30% of the raw R1 FASTQ</td>
</tr>
<tr>
<td>arln_r2_q30_clean</td>
<td>String</td>
<td>Q30% of the cleaned R2 FASTQ</td>
</tr>
<tr>
<td>arln_r2_q30_raw</td>
<td>String</td>
<td>Q30% of the raw R2 FASTQ</td>
</tr>
<tr>
<td>arln_stats_docker_version</td>
<td>String</td>
<td>Docker version for task_arln_stats.wdl</td>
</tr>
<tr>
<td>arln_taxon_assembly_ratio_stdev</td>
<td>String</td>
<td>Reported assembely length standard deviation of the given taxonomy pulled from NCBI aggregated data.</td>
</tr>
<tr>
<td>arln_taxon_gc_mean</td>
<td>String</td>
<td>Mean GC percent of the taxon.</td>
</tr>
<tr>
<td>arln_taxon_gc_percent_stdev</td>
<td>String</td>
<td>Reported GC percent standard deviation of the given taxonomy pulled from NCBI aggregated data.</td>
</tr>
<tr>
<td>assembler</td>
<td>String</td>
<td>Assembler used in digger_denovo subworkflow</td>
</tr>
<tr>
<td>assembler_version</td>
<td>String</td>
<td>Version of the assembler used in digger_denovo</td>
</tr>
<tr>
<td>assembly_fasta</td>
<td>File</td>
<td><em>De novo</em> genome assembly in FASTA format</td>
</tr>
<tr>
<td>assembly_length</td>
<td>Int</td>
<td>Length of assembly (total contig length) as determined by QUAST</td>
</tr>
<tr>
<td>bakta_gbff</td>
<td>File</td>
<td>Genomic GenBank format annotation file</td>
</tr>
<tr>
<td>bakta_gff3</td>
<td>File</td>
<td>Generic Feature Format Version 3 file</td>
</tr>
<tr>
<td>bakta_plot</td>
<td>File</td>
<td>Bakta plot output PNG file summarizing annotated genome features such as coding sequences, RNA genes, and hypothetical proteins.</td>
</tr>
<tr>
<td>bakta_summary</td>
<td>File</td>
<td>Bakta summary output TXT file</td>
</tr>
<tr>
<td>bakta_tsv</td>
<td>File</td>
<td>Annotations as simple human readable TSV</td>
</tr>
<tr>
<td>bakta_version</td>
<td>String</td>
<td>Bakta version used</td>
</tr>
<tr>
<td>bbduk_docker</td>
<td>String</td>
<td>The Docker image for bbduk, which was used to remove the adapters from the sequences</td>
</tr>
<tr>
<td>busco_database</td>
<td>String</td>
<td>BUSCO database used</td>
</tr>
<tr>
<td>busco_docker</td>
<td>String</td>
<td>BUSCO docker image used</td>
</tr>
<tr>
<td>busco_report</td>
<td>File</td>
<td>A plain text summary of the results in BUSCO notation</td>
</tr>
<tr>
<td>busco_results</td>
<td>String</td>
<td>BUSCO results (see relevant toggle in <a href="https://theiagen.github.io/public_health_bioinformatics/latest/workflows/genomic_characterization/theiaprok/#post-assembly-tasks-performed-for-all-taxa">this block</a>)</td>
</tr>
<tr>
<td>busco_version</td>
<td>String</td>
<td>BUSCO software version used</td>
</tr>
<tr>
<td>cg_pipeline_docker</td>
<td>String</td>
<td>Docker file used for running CG-Pipeline on cleaned reads</td>
</tr>
<tr>
<td>cg_pipeline_report_clean</td>
<td>File</td>
<td>TSV file of read metrics from clean reads, including average read length, number of reads, and estimated genome coverage</td>
</tr>
<tr>
<td>cg_pipeline_report_raw</td>
<td>File</td>
<td>TSV file of read metrics from raw reads, including average read length, number of reads, and estimated genome coverage</td>
</tr>
<tr>
<td>clockwork_decontaminated_read1</td>
<td>File</td>
<td>Decontaminated forward reads by Clockwork</td>
</tr>
<tr>
<td>clockwork_decontaminated_read2</td>
<td>File</td>
<td>Decontaminated reverse reads by Clockwork</td>
</tr>
<tr>
<td>combined_mean_q_clean</td>
<td>Float</td>
<td>Mean quality score for the combined clean reads</td>
</tr>
<tr>
<td>combined_mean_q_raw</td>
<td>Float</td>
<td>Mean quality score for the combined raw reads</td>
</tr>
<tr>
<td>combined_mean_readlength_clean</td>
<td>Float</td>
<td>Mean read length for the combined clean reads</td>
</tr>
<tr>
<td>combined_mean_readlength_raw</td>
<td>Float</td>
<td>Mean read length for the combined raw reads</td>
</tr>
<tr>
<td>contigs_gfa</td>
<td>File</td>
<td>Assembly graph output generated by SPAdes (Illumina: PE, SE) or Flye (ONT), used to visualize and evaluate genome assembly results.</td>
</tr>
<tr>
<td>ectyper_database_version</td>
<td>String</td>
<td>Version of the ECTyper database used</td>
</tr>
<tr>
<td>ectyper_docker</td>
<td>String</td>
<td>Docker image used to run ECTyper</td>
</tr>
<tr>
<td>ectyper_pathodb_version</td>
<td>String</td>
<td>Version of the ECTyper pathotype database used</td>
</tr>
<tr>
<td>ectyper_pathotype</td>
<td>String</td>
<td>Samples pathotype predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_pathotype_count</td>
<td>String</td>
<td>Number of pathotypes predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_pathotype_genes</td>
<td>String</td>
<td>Pathotype genes observed by ECTyper</td>
</tr>
<tr>
<td>ectyper_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_qc_result</td>
<td>String</td>
<td>QC pass or fail for the sample analyzed</td>
</tr>
<tr>
<td>ectyper_results</td>
<td>File</td>
<td>TSV file of evidence for ECTyper predicted serotype (see <a href="https://github.com/phac-nml/ecoli_serotyping#report-format">https://github.com/phac-nml/ecoli_serotyping#report-format</a>)</td>
</tr>
<tr>
<td>ectyper_stx_subtypes</td>
<td>String</td>
<td>Shiga toxin subtypes predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_version</td>
<td>String</td>
<td>Version of ECTyper used</td>
</tr>
<tr>
<td>ectyper_warnings</td>
<td>File</td>
<td>ECTyper warning messages</td>
</tr>
<tr>
<td>emmtyper_docker</td>
<td>String</td>
<td>Docker image for emmtyper</td>
</tr>
<tr>
<td>emmtyper_emm_type</td>
<td>String</td>
<td>The emm-type of a Streptococcus pyogenes assembly</td>
</tr>
<tr>
<td>emmtyper_results_tsv</td>
<td>File</td>
<td>TSV file with emmtyper results</td>
</tr>
<tr>
<td>emmtyper_version</td>
<td>String</td>
<td>Version of emmtyper used</td>
</tr>
<tr>
<td>emmtypingtool_docker</td>
<td>String</td>
<td>Docker image for emm-typing-tool</td>
</tr>
<tr>
<td>emmtypingtool_emm_type</td>
<td>String</td>
<td>emm-type predicted</td>
</tr>
<tr>
<td>emmtypingtool_results_xml</td>
<td>File</td>
<td>XML file with emm-typing-tool resuls</td>
</tr>
<tr>
<td>emmtypingtool_version</td>
<td>String</td>
<td>Version of emm-typing-tool used</td>
</tr>
<tr>
<td>est_coverage_clean</td>
<td>Float</td>
<td>Estimated coverage calculated from clean reads and genome length</td>
</tr>
<tr>
<td>est_coverage_raw</td>
<td>Float</td>
<td>Estimated coverage calculated from raw reads and genome length</td>
</tr>
<tr>
<td>fastp_html_report</td>
<td>File</td>
<td>The HTML report made with fastp</td>
</tr>
<tr>
<td>fastp_version</td>
<td>String</td>
<td>The version of fastp used</td>
</tr>
<tr>
<td>fastq_scan_clean1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_clean2_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean reverse read quality and length</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean1</td>
<td>Int</td>
<td>The number of forward reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean2</td>
<td>Int</td>
<td>The number of reverse reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean_pairs</td>
<td>String</td>
<td>The number of read pairs after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw1</td>
<td>Int</td>
<td>The number of input forward reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw2</td>
<td>Int</td>
<td>The number of input reserve reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw_pairs</td>
<td>String</td>
<td>The number of input read pairs as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_raw1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_raw2_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw reverse read quality and length</td>
</tr>
<tr>
<td>fastq_scan_version</td>
<td>String</td>
<td>The version of fastq_scan</td>
</tr>
<tr>
<td>fastqc_clean1_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of clean forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_clean2_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of clean reverse read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_docker</td>
<td>String</td>
<td>The Docker container used for fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean1</td>
<td>Int</td>
<td>The number of forward reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean2</td>
<td>Int</td>
<td>The number of reverse reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean_pairs</td>
<td>String</td>
<td>The number of read pairs after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_raw1</td>
<td>Int</td>
<td>The number of input forward reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_num_reads_raw2</td>
<td>Int</td>
<td>The number of input reverse reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_num_reads_raw_pairs</td>
<td>String</td>
<td>The number of input read pairs by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_raw1_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of raw forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_raw2_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of raw reverse read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_version</td>
<td>String</td>
<td>Version of fastqc software used</td>
</tr>
<tr>
<td>filtered_contigs_metrics</td>
<td>File</td>
<td>File containing metrics of contigs filtered</td>
</tr>
<tr>
<td>gambit_closest_genomes</td>
<td>File</td>
<td>CSV file listing genomes in the GAMBIT database that are most similar to the query assembly</td>
</tr>
<tr>
<td>gambit_db_version</td>
<td>String</td>
<td>Version of the GAMBIT database used</td>
</tr>
<tr>
<td>gambit_docker</td>
<td>String</td>
<td>GAMBIT Docker used</td>
</tr>
<tr>
<td>gambit_predicted_taxon</td>
<td>String</td>
<td>Taxon predicted by GAMBIT</td>
</tr>
<tr>
<td>gambit_predicted_taxon_rank</td>
<td>String</td>
<td>Taxon rank of GAMBIT taxon prediction</td>
</tr>
<tr>
<td>gambit_report</td>
<td>File</td>
<td>GAMBIT report in a machine-readable format</td>
</tr>
<tr>
<td>gambit_version</td>
<td>String</td>
<td>Version of GAMBIT software used</td>
</tr>
<tr>
<td>gamma_docker</td>
<td>String</td>
<td>The Docker container used</td>
</tr>
<tr>
<td>gamma_fasta</td>
<td>File</td>
<td>The FASTA file reporting gene matches if boolean was set to true</td>
</tr>
<tr>
<td>gamma_gff</td>
<td>File</td>
<td>The GFF file reporting gene matches if boolean was set to true</td>
</tr>
<tr>
<td>gamma_results</td>
<td>File</td>
<td>The .gamma file reporting gene matches</td>
</tr>
<tr>
<td>gamma_version</td>
<td>String</td>
<td>The version of GAMMA used</td>
</tr>
<tr>
<td>genotyphi_final_genotype</td>
<td>String</td>
<td>Final genotype call from GenoTyphi</td>
</tr>
<tr>
<td>genotyphi_genotype_confidence</td>
<td>String</td>
<td>Confidence in the final genotype call made by GenoTyphi</td>
</tr>
<tr>
<td>genotyphi_mykrobe_json</td>
<td>File</td>
<td>JSON file of GenoTyphi output, described <a href="https://github.com/katholt/genotyphi#explanation-of-columns-in-the-output">https://github.com/katholt/genotyphi#explanation-of-columns-in-the-output</a></td>
</tr>
<tr>
<td>genotyphi_report_tsv</td>
<td>File</td>
<td>TSV file of GenoTyphi output, described <a href="https://github.com/katholt/genotyphi#explanation-of-columns-in-the-output">https://github.com/katholt/genotyphi#explanation-of-columns-in-the-output</a></td>
</tr>
<tr>
<td>genotyphi_species</td>
<td>String</td>
<td>Species call from Mykrobe, used to run GenoTyphi</td>
</tr>
<tr>
<td>genotyphi_st_probes_percent_coverage</td>
<td>Float</td>
<td>Percentage coverage to the Typhi MLST probes</td>
</tr>
<tr>
<td>genotyphi_version</td>
<td>String</td>
<td>Version of GenoTyphi used</td>
</tr>
<tr>
<td>hicap_docker</td>
<td>String</td>
<td>Docker image used for hicap</td>
</tr>
<tr>
<td>hicap_genes</td>
<td>String</td>
<td>cap genes identified. genes on different contigs delimited by;. truncation shown by trailing *</td>
</tr>
<tr>
<td>hicap_results_tsv</td>
<td>File</td>
<td>TSV file of hicap output</td>
</tr>
<tr>
<td>hicap_serotype</td>
<td>String</td>
<td>hicap serotype</td>
</tr>
<tr>
<td>hicap_version</td>
<td>String</td>
<td>hicap version used</td>
</tr>
<tr>
<td>kaptive_k_locus</td>
<td>String</td>
<td>Best matching K locus identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_k_type</td>
<td>String</td>
<td>Best matching K type identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_kl_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kaptive_oc_locus</td>
<td>String</td>
<td>Best matching K locus identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_ocl_confidence</td>
<td>String</td>
<td>Kaptives confidence in the OCL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kaptive_output_file_k</td>
<td>File</td>
<td>TSV <a href="https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom">https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom</a> the K locus from Kaptive</td>
</tr>
<tr>
<td>kaptive_output_file_oc</td>
<td>File</td>
<td>TSV <a href="https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom">https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom</a> the OC locus from Kaptive</td>
</tr>
<tr>
<td>kaptive_version</td>
<td>String</td>
<td>Version of Kaptive used</td>
</tr>
<tr>
<td>kleborate_docker</td>
<td>String</td>
<td>Kleborate docker image used</td>
</tr>
<tr>
<td>kleborate_genomic_resistance_mutations</td>
<td>String</td>
<td>Genomic resistance mutations identifies by Kleborate</td>
</tr>
<tr>
<td>kleborate_key_resistance_genes</td>
<td>String</td>
<td>Key resistance genes identified by Kleborate</td>
</tr>
<tr>
<td>kleborate_klocus</td>
<td>String</td>
<td>Best matching K locus identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_klocus_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kleborate_ktype</td>
<td>String</td>
<td>Best matching K type identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_mlst_sequence_type</td>
<td>String</td>
<td><a href="https://github.com/katholt/Kleborate/wiki/MLST#multi-locus-sequence-typing-mlst">https://github.com/katholt/Kleborate/wiki/MLST#multi-locus-sequence-typing-mlst</a> call by Kleborate</td>
</tr>
<tr>
<td>kleborate_olocus</td>
<td>String</td>
<td>Best matching OC locus identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_olocus_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kleborate_otype</td>
<td>String</td>
<td>Best matching OC type identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_output_file</td>
<td>File</td>
<td>See also <a href="https://github.com/katholt/Kleborate/wiki/Scores-and-counts">https://github.com/katholt/Kleborate/wiki/Scores-and-counts</a></td>
</tr>
<tr>
<td>kleborate_resistance_score</td>
<td>String</td>
<td>Resistance score as given by kleborate</td>
</tr>
<tr>
<td>kleborate_version</td>
<td>String</td>
<td>Version of Kleborate used</td>
</tr>
<tr>
<td>kleborate_virulence_score</td>
<td>String</td>
<td>Virulence score as given by kleborate</td>
</tr>
<tr>
<td>kmerfinder_database</td>
<td>String</td>
<td>Database used to run KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_docker</td>
<td>String</td>
<td>Docker image used to run KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_query_coverage</td>
<td>String</td>
<td>KmerFinders query coverage of the top hit result</td>
</tr>
<tr>
<td>kmerfinder_results_tsv</td>
<td>File</td>
<td>Output TSV file created by KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_template_coverage</td>
<td>String</td>
<td>Percent of kmer coverage pertaining to the template, or reference selected</td>
</tr>
<tr>
<td>kmerfinder_top_hit</td>
<td>String</td>
<td>Top hit species of KmerFinder</td>
</tr>
<tr>
<td>kraken2_database</td>
<td>String</td>
<td>Kraken2 database used for the taxonomic assignment</td>
</tr>
<tr>
<td>kraken2_report</td>
<td>String</td>
<td>TXT document describing taxonomic prediction of every FASTQ record. This file is usually very large and cumbersome to open and view</td>
</tr>
<tr>
<td>kraken2_version</td>
<td>String</td>
<td>The version of kraken2 used</td>
</tr>
<tr>
<td>kraken_docker</td>
<td>String</td>
<td>Docker image used for Kraken</td>
</tr>
<tr>
<td>legsta_predicted_sbt</td>
<td>String</td>
<td>Sequence based type predicted by Legsta</td>
</tr>
<tr>
<td>legsta_results</td>
<td>File</td>
<td>TSV file of legsta results (see <a href="https://github.com/tseemann/legsta#output">https://github.com/tseemann/legsta#output</a>)</td>
</tr>
<tr>
<td>legsta_version</td>
<td>String</td>
<td>Version of legsta used</td>
</tr>
<tr>
<td>lissero_results</td>
<td>File</td>
<td>TSV results file from LisSero (see <a href="https://github.com/MDU-PHL/LisSero#example-output">https://github.com/MDU-PHL/LisSero#example-output</a>)</td>
</tr>
<tr>
<td>lissero_serotype</td>
<td>String</td>
<td>Serotype predicted by LisSero</td>
</tr>
<tr>
<td>lissero_version</td>
<td>String</td>
<td>Version of LisSero used</td>
</tr>
<tr>
<td>meningotype_BAST</td>
<td>String</td>
<td>BAST type</td>
</tr>
<tr>
<td>meningotype_FetA</td>
<td>String</td>
<td>FetA type</td>
</tr>
<tr>
<td>meningotype_NHBA</td>
<td>String</td>
<td>NHBA type</td>
</tr>
<tr>
<td>meningotype_NadA</td>
<td>String</td>
<td>NBA type</td>
</tr>
<tr>
<td>meningotype_PorA</td>
<td>String</td>
<td>PorA type</td>
</tr>
<tr>
<td>meningotype_PorB</td>
<td>String</td>
<td>PorB type</td>
</tr>
<tr>
<td>meningotype_fHbp</td>
<td>String</td>
<td>fHbp type</td>
</tr>
<tr>
<td>meningotype_serogroup</td>
<td>String</td>
<td>Serogroup</td>
</tr>
<tr>
<td>meningotype_tsv</td>
<td>File</td>
<td>Full result file</td>
</tr>
<tr>
<td>meningotype_version</td>
<td>String</td>
<td>Version of meningotype used</td>
</tr>
<tr>
<td>midas_docker</td>
<td>String</td>
<td>MIDAS docker image used</td>
</tr>
<tr>
<td>midas_primary_genus</td>
<td>String</td>
<td>The primary genus detected by MIDAS</td>
</tr>
<tr>
<td>midas_report</td>
<td>File</td>
<td>TSV report of full MIDAS results</td>
</tr>
<tr>
<td>midas_secondary_genus</td>
<td>String</td>
<td>Genus of the next most abundant species after removing all species of the most abundant genus</td>
</tr>
<tr>
<td>midas_secondary_genus_abundance</td>
<td>Float</td>
<td>Relative abundance of secondary genus</td>
</tr>
<tr>
<td>midas_secondary_genus_coverage</td>
<td>Float</td>
<td>Absolute coverage of secondary genus</td>
</tr>
<tr>
<td>n50_value</td>
<td>Int</td>
<td>N50 of assembly calculated by QUAST</td>
</tr>
<tr>
<td>ngmaster_ngmast_porB_allele</td>
<td>String</td>
<td>porB allele number</td>
</tr>
<tr>
<td>ngmaster_ngmast_sequence_type</td>
<td>String</td>
<td>NG-MAST sequence type</td>
</tr>
<tr>
<td>ngmaster_ngmast_tbpB_allele</td>
<td>String</td>
<td>tbpB allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_23S_allele</td>
<td>String</td>
<td>23S rRNA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_gyrA_allele</td>
<td>String</td>
<td>gyrA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_mtrR_allele</td>
<td>String</td>
<td>mtrR allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_parC_allele</td>
<td>String</td>
<td>parC allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_penA_allele</td>
<td>String</td>
<td>penA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_ponA_allele</td>
<td>String</td>
<td>ponA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_porB_allele</td>
<td>String</td>
<td>porB allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_sequence_type</td>
<td>String</td>
<td>NG-STAR sequence type</td>
</tr>
<tr>
<td>ngmaster_tsv</td>
<td>File</td>
<td>TSV file with NG-MAST/NG-STAR typing</td>
</tr>
<tr>
<td>ngmaster_version</td>
<td>String</td>
<td>ngmaster version</td>
</tr>
<tr>
<td>number_contigs</td>
<td>Int</td>
<td>Total number of contigs in assembly</td>
</tr>
<tr>
<td>pasty_all_serogroups</td>
<td>File</td>
<td>TSV file with details of each serogroup from pasty (see <a href="https://github.com/rpetit3/pasty#example-prefixdetailstsv">https://github.com/rpetit3/pasty#example-prefixdetailstsv</a>)</td>
</tr>
<tr>
<td>pasty_blast_hits</td>
<td>File</td>
<td>TSV file of BLAST hits from pasty (see <a href="https://github.com/rpetit3/pasty#example-prefixblastntsv">https://github.com/rpetit3/pasty#example-prefixblastntsv</a>)</td>
</tr>
<tr>
<td>pasty_comment</td>
<td>String</td>
<td></td>
</tr>
<tr>
<td>pasty_docker</td>
<td>String</td>
<td>pasty docker image used</td>
</tr>
<tr>
<td>pasty_serogroup</td>
<td>String</td>
<td>Serogroup predicted by pasty</td>
</tr>
<tr>
<td>pasty_serogroup_coverage</td>
<td>Float</td>
<td>The breadth of coverage of the O-antigen by pasty</td>
</tr>
<tr>
<td>pasty_serogroup_fragments</td>
<td>Int</td>
<td>Number of BLAST hits included in the prediction (fewer is better)</td>
</tr>
<tr>
<td>pasty_summary_tsv</td>
<td>File</td>
<td>TSV summary file of pasty outputs (see <a href="https://github.com/rpetit3/pasty#example-prefixtsv">https://github.com/rpetit3/pasty#example-prefixtsv</a>)</td>
</tr>
<tr>
<td>pasty_version</td>
<td>String</td>
<td>Version of pasty used</td>
</tr>
<tr>
<td>pbptyper_docker</td>
<td>String</td>
<td>pbptyper docker image used</td>
</tr>
<tr>
<td>pbptyper_pbptype_predicted_tsv</td>
<td>File</td>
<td>TSV file of pbptyper results (see <a href="https://github.com/rpetit3/pbptyper#example-prefixtsv">https://github.com/rpetit3/pbptyper#example-prefixtsv</a>)</td>
</tr>
<tr>
<td>pbptyper_predicted_1A_2B_2X</td>
<td>String</td>
<td>PBP type predicted by pbptyper</td>
</tr>
<tr>
<td>pbptyper_version</td>
<td>String</td>
<td>Version of pbptyper used</td>
</tr>
<tr>
<td>plasmidfinder_db_version</td>
<td>String</td>
<td>Version of PlasmidFnder used</td>
</tr>
<tr>
<td>plasmidfinder_docker</td>
<td>String</td>
<td>PlasmidFinder docker image used</td>
</tr>
<tr>
<td>plasmidfinder_plasmids</td>
<td>String</td>
<td>Names of plasmids identified by PlasmidFinder</td>
</tr>
<tr>
<td>plasmidfinder_results</td>
<td>File</td>
<td>Output file from PlasmidFinder in TSV format</td>
</tr>
<tr>
<td>plasmidfinder_seqs</td>
<td>File</td>
<td>Hit_in_genome_seq.fsa file produced by PlasmidFinder</td>
</tr>
<tr>
<td>poppunk_GPS_db_version</td>
<td>String</td>
<td>Version of GPSC database used</td>
</tr>
<tr>
<td>poppunk_docker</td>
<td>String</td>
<td>PopPUNK docker image with GPSC database used</td>
</tr>
<tr>
<td>poppunk_gps_cluster</td>
<td>String</td>
<td>GPS cluster predicted by PopPUNK</td>
</tr>
<tr>
<td>poppunk_gps_external_cluster_csv</td>
<td>File</td>
<td>GPSC v6 scheme designations</td>
</tr>
<tr>
<td>poppunk_version</td>
<td>String</td>
<td>Version of PopPUNK used</td>
</tr>
<tr>
<td>prokka_gbk</td>
<td>File</td>
<td>GenBank file produced from Prokka annotation of input FASTA</td>
</tr>
<tr>
<td>prokka_gff</td>
<td>File</td>
<td>Prokka output GFF3 file containing sequence and annotation (you can view this in IGV)</td>
</tr>
<tr>
<td>prokka_sqn</td>
<td>File</td>
<td>A Sequin file for GenBank submission</td>
</tr>
<tr>
<td>qc_check</td>
<td>String</td>
<td>A string that indicates whether or not the sample passes a set of pre-determined and user-provided QC thresholds</td>
</tr>
<tr>
<td>qc_standard</td>
<td>File</td>
<td>The file used in the QC Check task containing the QC thresholds.</td>
</tr>
<tr>
<td>quast_gc_percent</td>
<td>Float</td>
<td>The GC percent of your sample</td>
</tr>
<tr>
<td>quast_report</td>
<td>File</td>
<td>TSV report from QUAST</td>
</tr>
<tr>
<td>quast_version</td>
<td>String</td>
<td>The version of QUAST</td>
</tr>
<tr>
<td>r1_mean_q_clean</td>
<td>Float</td>
<td>Mean quality score of clean forward reads</td>
</tr>
<tr>
<td>r1_mean_q_raw</td>
<td>Float</td>
<td>Mean quality score of raw forward reads</td>
</tr>
<tr>
<td>r1_mean_readlength_clean</td>
<td>Float</td>
<td>Mean read length of clean forward reads</td>
</tr>
<tr>
<td>r1_mean_readlength_raw</td>
<td>Float</td>
<td>Mean read length of raw forward reads</td>
</tr>
<tr>
<td>r2_mean_q_clean</td>
<td>Float</td>
<td>Mean quality score of clean reverse reads</td>
</tr>
<tr>
<td>r2_mean_q_raw</td>
<td>Float</td>
<td>Mean quality score of raw reverse reads</td>
</tr>
<tr>
<td>r2_mean_readlength_clean</td>
<td>Float</td>
<td>Mean read length of clean reverse reads</td>
</tr>
<tr>
<td>r2_mean_readlength_raw</td>
<td>Float</td>
<td>Mean read length of raw reverse reads</td>
</tr>
<tr>
<td>read1_clean</td>
<td>File</td>
<td>Forward read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read1_concatenated</td>
<td>File</td>
<td>Concatenated forward reads</td>
</tr>
<tr>
<td>read2_clean</td>
<td>File</td>
<td>Reverse read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read2_concatenated</td>
<td>File</td>
<td>Concatenated reverse reads</td>
</tr>
<tr>
<td>read_screen_clean</td>
<td>String</td>
<td>PASS or FAIL result from clean read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_clean_tsv</td>
<td>File</td>
<td>Clean read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>read_screen_raw</td>
<td>String</td>
<td>PASS or FAIL result from raw read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_raw_tsv</td>
<td>File</td>
<td>Raw read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>resfinder_db_version</td>
<td>String</td>
<td>Version of ResFinder database</td>
</tr>
<tr>
<td>resfinder_docker</td>
<td>String</td>
<td>ResFinder docker image used</td>
</tr>
<tr>
<td>resfinder_pheno_table</td>
<td>File</td>
<td>Table containing al AMR phenotypes</td>
</tr>
<tr>
<td>resfinder_pheno_table_species</td>
<td>File</td>
<td>Table with species-specific AMR phenotypes</td>
</tr>
<tr>
<td>resfinder_pointfinder_pheno_table</td>
<td>File</td>
<td>TSV showing presence(1)/absence(0) of predicted resistance against an antibiotic class</td>
</tr>
<tr>
<td>resfinder_pointfinder_results</td>
<td>File</td>
<td>Predicted point mutations, grouped by the gene they occur in</td>
</tr>
<tr>
<td>resfinder_predicted_pheno_resistance</td>
<td>String</td>
<td>Semicolon delimited list of antimicrobial drugs and associated genes and/or point mutations.&nbsp;<drug1>: <gene1>, <gene1>, <point_mutation1>; <drug2>: <gene3>, <gene4>;</gene4></gene3></drug2></point_mutation1></gene1></gene1></drug1></td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Amp</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ampicillin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Axo</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ceftriaxone based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Azm</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Azithromycin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Cip</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ciprofloxacin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Smx</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Sulfamethoxazole based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Tmp</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Trimothoprim based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_xdr_shigella</td>
<td>String</td>
<td>Final prediction of XDR Shigella status based on CDC definition. Explanation can be found in the description above this table.</td>
</tr>
<tr>
<td>resfinder_results</td>
<td>File</td>
<td>Predicted resistance genes grouped by antibiotic class</td>
</tr>
<tr>
<td>resfinder_seqs</td>
<td>File</td>
<td>FASTA of resistance gene sequences from users input sequence</td>
</tr>
<tr>
<td>seq_platform</td>
<td>String</td>
<td>Description of the sequencing methodology used to generate the input read data</td>
</tr>
<tr>
<td>seqsero2s_note</td>
<td>String</td>
<td>Additional notes produced by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_predicted_antigenic_profile</td>
<td>String</td>
<td>Antigenic profile predicted for Salmonella spp. by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_predicted_contamination</td>
<td>String</td>
<td>Indicates whether contamination between Salmonella with different serotypes was predicted by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_report</td>
<td>String</td>
<td>TSV report produced by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_version</td>
<td>String</td>
<td>Version of SeqSero2S used</td>
</tr>
<tr>
<td>seroba_ariba_identity</td>
<td>String</td>
<td>Percentage identity between the query sequence and ARIBA-predicted serotype</td>
</tr>
<tr>
<td>seroba_ariba_serotype</td>
<td>String</td>
<td>Serotype predicted by ARIBA, via SeroBA</td>
</tr>
<tr>
<td>seroba_details</td>
<td>File</td>
<td>Detailed TSV file from SeroBA</td>
</tr>
<tr>
<td>seroba_docker</td>
<td>String</td>
<td>SeroBA docker image used</td>
</tr>
<tr>
<td>seroba_serotype</td>
<td>String</td>
<td>Serotype predicted by SeroBA</td>
</tr>
<tr>
<td>seroba_version</td>
<td>String</td>
<td>SeroBA version used</td>
</tr>
<tr>
<td>serotypefinder_docker</td>
<td>String</td>
<td>SerotypeFinder docker image used</td>
</tr>
<tr>
<td>serotypefinder_report</td>
<td>File</td>
<td>TSV report produced by SerotypeFinder</td>
</tr>
<tr>
<td>serotypefinder_serotype</td>
<td>String</td>
<td>Serotype predicted by SerotypeFinder</td>
</tr>
<tr>
<td>shigatyper_docker</td>
<td>String</td>
<td>ShigaTyper docker image used</td>
</tr>
<tr>
<td>shigatyper_hits_tsv</td>
<td>File</td>
<td>Detailed TSV report from ShigaTyper (see <a href="https://github.com/CFSAN-Biostatistics/shigatyper#example-prefix-hitstsv">https://github.com/CFSAN-Biostatistics/shigatyper#example-prefix-hitstsv</a>)</td>
</tr>
<tr>
<td>shigatyper_ipaB_presence_absence</td>
<td>String</td>
<td>Presence (+) or absence (-) of ipaB identified by ShigaTyper</td>
</tr>
<tr>
<td>shigatyper_notes</td>
<td>String</td>
<td>Any notes output from ShigaTyper</td>
</tr>
<tr>
<td>shigatyper_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by ShigaTyper</td>
</tr>
<tr>
<td>shigatyper_summary_tsv</td>
<td>File</td>
<td>TSV summary report from ShigaTyper (see <a href="https://github.com/CFSAN-Biostatistics/shigatyper#example-prefixtsv">https://github.com/CFSAN-Biostatistics/shigatyper#example-prefixtsv</a>)</td>
</tr>
<tr>
<td>shigatyper_version</td>
<td>String</td>
<td>Version of ShigaTyper used</td>
</tr>
<tr>
<td>shigeifinder_H_antigen</td>
<td>String</td>
<td>H-antigen gene identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_H_antigen_reads</td>
<td>String</td>
<td>H-antigen gene identified by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_O_antigen</td>
<td>String</td>
<td>O-antigen gene identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_O_antigen_reads</td>
<td>String</td>
<td>O-antigen gene identified by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_cluster</td>
<td>String</td>
<td>Shigella/EIEC cluster identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_cluster_reads</td>
<td>String</td>
<td>Shigella/EIEC cluster identified by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_docker</td>
<td>String</td>
<td>ShigEiFinder docker image used</td>
</tr>
<tr>
<td>shigeifinder_docker_reads</td>
<td>String</td>
<td>ShigEiFinder docker image used using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_ipaH_presence_absence</td>
<td>String</td>
<td>Presence (+) or absence (-) of ipaH identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_ipaH_presence_absence_reads</td>
<td>String</td>
<td>Presence (+) or absence (-) of ipaH identified by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_notes</td>
<td>String</td>
<td>Any notes output from ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_notes_reads</td>
<td>String</td>
<td>Any notes output from ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_num_virulence_plasmid_genes</td>
<td>String</td>
<td>Number of virulence plasmid genes identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_num_virulence_plasmid_genes_reads</td>
<td>String</td>
<td>Number of virulence plasmid genes identified by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_report</td>
<td>File</td>
<td>TSV report from ShigEiFinder (see <a href="https://github.com/LanLab/ShigEiFinder#shigeifinder">https://github.com/LanLab/ShigEiFinder#shigeifinder</a>)</td>
</tr>
<tr>
<td>shigeifinder_report_reads</td>
<td>File</td>
<td>TSV report from ShigEiFinder (see <a href="https://github.com/LanLab/ShigEiFinder#shigeifinder">https://github.com/LanLab/ShigEiFinder#shigeifinder</a>) using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_serotype</td>
<td>String</td>
<td>Serotype predicted by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_serotype_reads</td>
<td>String</td>
<td>Serotype predicted by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_version</td>
<td>String</td>
<td>ShigEiFinder version used</td>
</tr>
<tr>
<td>shigeifinder_version_reads</td>
<td>String</td>
<td>ShigEiFinder version used using read files as inputs</td>
</tr>
<tr>
<td>sistr_allele_fasta</td>
<td>File</td>
<td>FASTA file of novel cgMLST alleles from SISTR</td>
</tr>
<tr>
<td>sistr_allele_json</td>
<td>File</td>
<td>JSON file of cgMLST allele sequences and information (see <a href="https://github.com/phac-nml/sistr_cmd#cgmlst-allele-search-results">https://github.com/phac-nml/sistr_cmd#cgmlst-allele-search-results</a>)</td>
</tr>
<tr>
<td>sistr_antigenic_formula</td>
<td>String</td>
<td>A field that aggregates the O, H1, and H2, antigen values in a single location for convenience</td>
</tr>
<tr>
<td>sistr_cgmlst</td>
<td>File</td>
<td>CSV file of the cgMLST allelic profile from SISTR (see <a href="https://github.com/phac-nml/sistr_cmd#cgmlst-allelic-profiles-output---cgmlst-profiles-cgmlst-profilescsv">https://github.com/phac-nml/sistr_cmd#cgmlst-allelic-profiles-output---cgmlst-profiles-cgmlst-profilescsv</a>)</td>
</tr>
<tr>
<td>sistr_h1_antigens</td>
<td>String</td>
<td>The predicted H1 antigen</td>
</tr>
<tr>
<td>sistr_h2_antigens</td>
<td>String</td>
<td>The predicted H2 antigen</td>
</tr>
<tr>
<td>sistr_o_antigens</td>
<td>String</td>
<td>The predicted O antigen</td>
</tr>
<tr>
<td>sistr_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by SISTR</td>
</tr>
<tr>
<td>sistr_results</td>
<td>File</td>
<td>TSV results file produced by SISTR (see <a href="https://github.com/phac-nml/sistr_cmd#primary-results-output--o-sistr-results">https://github.com/phac-nml/sistr_cmd#primary-results-output--o-sistr-results</a>)</td>
</tr>
<tr>
<td>sistr_serogroup</td>
<td>String</td>
<td>Serogroup predicted by SISTR</td>
</tr>
<tr>
<td>sistr_serotype_cgmlst</td>
<td>String</td>
<td>cgMLST of the serogroup prediicted by SISTR</td>
</tr>
<tr>
<td>sistr_version</td>
<td>String</td>
<td>Version of SISTR used</td>
</tr>
<tr>
<td>sonneityping_final_genotype</td>
<td>String</td>
<td>Final genotype call from Mykrobe, via sonneityper</td>
</tr>
<tr>
<td>sonneityping_final_report_tsv</td>
<td>File</td>
<td>Detailed TSV report from mykrobe, via sonneityper (see <a href="https://github.com/katholt/sonneityping#example-output">https://github.com/katholt/sonneityping#example-output</a>)</td>
</tr>
<tr>
<td>sonneityping_genotype_confidence</td>
<td>String</td>
<td>Confidence in the final genotype call from sonneityper</td>
</tr>
<tr>
<td>sonneityping_genotype_name</td>
<td>String</td>
<td>Human readable alias for genotype, where available provided by sonneityper</td>
</tr>
<tr>
<td>sonneityping_mykrobe_docker</td>
<td>String</td>
<td>sonneityping docker image used</td>
</tr>
<tr>
<td>sonneityping_mykrobe_report_csv</td>
<td>File</td>
<td>CSV report from mykrobe via sonneityper (see <a href="https://github.com/Mykrobe-tools/mykrobe/wiki/AMR-prediction-output#csv-file">https://github.com/Mykrobe-tools/mykrobe/wiki/AMR-prediction-output#csv-file</a>)</td>
</tr>
<tr>
<td>sonneityping_mykrobe_report_json</td>
<td>File</td>
<td>JSON report from mykrobe via sonneityper (see <a href="https://github.com/Mykrobe-tools/mykrobe/wiki/AMR-prediction-output#json-file">https://github.com/Mykrobe-tools/mykrobe/wiki/AMR-prediction-output#json-file</a>)</td>
</tr>
<tr>
<td>sonneityping_mykrobe_version</td>
<td>String</td>
<td>Version of sonneityping used</td>
</tr>
<tr>
<td>sonneityping_species</td>
<td>String</td>
<td>Species call from Mykrobe via sonneityping</td>
</tr>
<tr>
<td>spatyper_docker</td>
<td>String</td>
<td>spatyper docker image used</td>
</tr>
<tr>
<td>spatyper_repeats</td>
<td>String</td>
<td>order of identified repeats</td>
</tr>
<tr>
<td>spatyper_tsv</td>
<td>File</td>
<td>TSV report with spatyper results</td>
</tr>
<tr>
<td>spatyper_type</td>
<td>String</td>
<td>spa type</td>
</tr>
<tr>
<td>spatyper_version</td>
<td>String</td>
<td>spatyper version used</td>
</tr>
<tr>
<td>srst2_vibrio_biotype</td>
<td>String</td>
<td>Biotype classification according to tcpA gene sequence (Classical or ElTor)</td>
</tr>
<tr>
<td>srst2_vibrio_ctxA</td>
<td>String</td>
<td>Presence or absence of the ctxA gene</td>
</tr>
<tr>
<td>srst2_vibrio_database</td>
<td>String</td>
<td>Database used for srst2 analysis</td>
</tr>
<tr>
<td>srst2_vibrio_detailed_tsv</td>
<td>File</td>
<td>Detailed <a href="https://github.com/katholt/srst2">https://github.com/katholt/srst2</a> output file</td>
</tr>
<tr>
<td>srst2_vibrio_docker</td>
<td>String</td>
<td>Docker image used for srst2 analysis</td>
</tr>
<tr>
<td>srst2_vibrio_ompW</td>
<td>String</td>
<td>Presence or absence of the ompW gene</td>
</tr>
<tr>
<td>srst2_vibrio_serogroup</td>
<td>String</td>
<td>Serotype classification as O1 (wbeN gene), O139 (wbfR gene) or not detected.</td>
</tr>
<tr>
<td>srst2_vibrio_toxR</td>
<td>String</td>
<td>Presence or absence of the toxR gene</td>
</tr>
<tr>
<td>srst2_vibrio_version</td>
<td>String</td>
<td>The SRST2 version run</td>
</tr>
<tr>
<td>staphopiasccmec_docker</td>
<td>String</td>
<td>staphopia-sccmec docker image used</td>
</tr>
<tr>
<td>staphopiasccmec_hamming_distance_tsv</td>
<td>File</td>
<td>staphopia-sccmec version</td>
</tr>
<tr>
<td>staphopiasccmec_results_tsv</td>
<td>File</td>
<td>sccmec types and mecA presence</td>
</tr>
<tr>
<td>staphopiasccmec_types_and_mecA_presence</td>
<td>String</td>
<td>staphopia-sccmec Hamming distance file</td>
</tr>
<tr>
<td>staphopiasccmec_version</td>
<td>String</td>
<td>staphopia-sccmec presence and absence TSV file</td>
</tr>
<tr>
<td>stxtyper_all_hits</td>
<td>String</td>
<td>Comma-separated list of matches of all types. Includes complete, partial, frameshift, internal stop, and novel hits. List is de-duplicated so multiple identical hits are only listed once. For example if 5 partial stx2 hits are detected in the genome, only 1 "stx2" will be listed in this field. To view the potential subtype for each partial hit, the user will need to view the stxtyper_report TSV file.</td>
</tr>
<tr>
<td>stxtyper_ambiguous_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "AMBIGUOUS". Ambiguous bases found in the query sequence (e.g., N)</td>
</tr>
<tr>
<td>stxtyper_complete_operons</td>
<td>String</td>
<td>Comma-separated list of all COMPLETE operons detected by StxTyper. Show multiple hits if present in results.</td>
</tr>
<tr>
<td>stxtyper_docker</td>
<td>String</td>
<td>Name of docker image used by the stxtyper task.</td>
</tr>
<tr>
<td>stxtyper_extended_operons</td>
<td>String</td>
<td>Comma-separated list of all EXTENDED operons detected by StxTyper if coding sequence extends beyond the reference stop codon for one or both of the reference proteins.</td>
</tr>
<tr>
<td>stxtyper_novel_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "COMPLETE_NOVEL". Possible outputs "stx1", "stx2", or "stx1,stx2"</td>
</tr>
<tr>
<td>stxtyper_num_hits</td>
<td>Int</td>
<td>Number of "hits" or rows present in the <code>stxtyper_report</code> TSV file</td>
</tr>
<tr>
<td>stxtyper_partial_hits</td>
<td>String</td>
<td>Possible outputs "stx1", "stx2", or "stx1,stx2". Tells the user that there was a partial hit to either the A or B subunit, but does not describe which subunit, only the possible types from the PARTIAL matches.</td>
</tr>
<tr>
<td>stxtyper_report</td>
<td>File</td>
<td>Raw results TSV file produced by StxTyper</td>
</tr>
<tr>
<td>stxtyper_stx_frameshifts_or_internal_stop_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "FRAMESHIFT" or "INTERNAL_STOP". Possible outputs "stx1", "stx2", or "stx1,stx2"</td>
</tr>
<tr>
<td>stxtyper_version</td>
<td>String</td>
<td>Version of StxTyper used</td>
</tr>
<tr>
<td>taxon_table_status</td>
<td>String</td>
<td>Status of the taxon table upload</td>
</tr>
<tr>
<td>tbp_parser_average_genome_depth</td>
<td>Float</td>
<td>The mean depth of coverage across all target regions included in the analysis</td>
</tr>
<tr>
<td>tbp_parser_coverage_report</td>
<td>File</td>
<td>A file containing the breadth of coverage across each target loci</td>
</tr>
<tr>
<td>tbp_parser_docker</td>
<td>String</td>
<td>The docker image and version tag for the tbp_parser tool</td>
</tr>
<tr>
<td>tbp_parser_genome_percent_coverage</td>
<td>Float</td>
<td>The percent breadth of coverage across the entire genome</td>
</tr>
<tr>
<td>tbp_parser_laboratorian_report_csv</td>
<td>File</td>
<td>An output file containing information regarding each mutation and its associated drug resistance profile in a CSV file. This file also contains two interpretation fields -- "Looker" and "MDL" which are generated using the CDC's expert rules for interpreting the severity of potential drug resistance mutations.</td>
</tr>
<tr>
<td>tbp_parser_lims_report_csv</td>
<td>File</td>
<td>An output file formatted specifically for STAR LIMS. This CSV report summarizes the highest severity mutations for each antimicrobial and lists the relevant mutations for each gene.</td>
</tr>
<tr>
<td>tbp_parser_looker_report_csv</td>
<td>File</td>
<td>An output file that contains condensed information suitable for generating a dashboard in Google's Looker studio.</td>
</tr>
<tr>
<td>tbp_parser_version</td>
<td>String</td>
<td>The version number of tbp_parser</td>
</tr>
<tr>
<td>tbprofiler_dr_type</td>
<td>String</td>
<td>The drug resistance category as determined by TBProfiler (sensitive, Pre-MDR, MDR, Pre-XDR, XDR)</td>
</tr>
<tr>
<td>tbprofiler_main_lineage</td>
<td>String</td>
<td>The Mycobacterium tuberculosis lineage assignment as made by TBProfiler</td>
</tr>
<tr>
<td>tbprofiler_median_depth</td>
<td>Float</td>
<td>The median depth of the H37Rv TB reference genome covered by the sample</td>
</tr>
<tr>
<td>tbprofiler_output_bai</td>
<td>File</td>
<td>Index BAM file generated by mapping sequencing reads to reference genome by TBProfiler</td>
</tr>
<tr>
<td>tbprofiler_output_bam</td>
<td>File</td>
<td>BAM alignment file produced by TBProfiler</td>
</tr>
<tr>
<td>tbprofiler_output_file</td>
<td>File</td>
<td>CSV report from TBProfiler</td>
</tr>
<tr>
<td>tbprofiler_output_vcf</td>
<td>File</td>
<td>VCF file output from TBProfiler; the concatenation of all of the different VCF files produced during TBProfiler analysis</td>
</tr>
<tr>
<td>tbprofiler_pct_reads_mapped</td>
<td>Float</td>
<td>The percentage of reads that successfully mapped to the H37Rv genome</td>
</tr>
<tr>
<td>tbprofiler_resistance_genes</td>
<td>String</td>
<td>The genes in which a mutation was detected that may be resistance conferring, in the format of <code>&lt;gene target&gt; &lt;variant detected&gt; (&lt;estimated fraction of reads that support the variant&gt;)</code></td>
</tr>
<tr>
<td>tbprofiler_sub_lineage</td>
<td>String</td>
<td>The Mycobacterium tuberculosis sub-lineage assignment as made by TBProfiler</td>
</tr>
<tr>
<td>tbprofiler_version</td>
<td>String</td>
<td>The version of TBProfiler used for this analysis</td>
</tr>
<tr>
<td>theiaprok_illumina_pe_analysis_date</td>
<td>String</td>
<td>Date of TheiaProk PE workflow execution</td>
</tr>
<tr>
<td>theiaprok_illumina_pe_version</td>
<td>String</td>
<td>Version of TheiaProk PE workflow execution</td>
</tr>
<tr>
<td>trimmomatic_docker</td>
<td>String</td>
<td>The docker image used for the trimmomatic module in this workflow</td>
</tr>
<tr>
<td>trimmomatic_version</td>
<td>String</td>
<td>The version of Trimmomatic used</td>
</tr>
<tr>
<td>ts_mlst_allelic_profile</td>
<td>String</td>
<td>Profile of MLST loci and allele numbers predicted by MLST</td>
</tr>
<tr>
<td>ts_mlst_docker</td>
<td>String</td>
<td>Docker image used for MLST</td>
</tr>
<tr>
<td>ts_mlst_novel_alleles</td>
<td>File</td>
<td>FASTA file containing nucleotide sequence of any alleles that are not in the MLST database used by TheiaProk</td>
</tr>
<tr>
<td>ts_mlst_predicted_secondary_st</td>
<td>String</td>
<td>ST predicted by secondary MLST run</td>
</tr>
<tr>
<td>ts_mlst_predicted_st</td>
<td>String</td>
<td>ST predicted by MLST</td>
</tr>
<tr>
<td>ts_mlst_pubmlst_scheme</td>
<td>String</td>
<td>PubMLST scheme used byMLST</td>
</tr>
<tr>
<td>ts_mlst_pubmlst_secondary_scheme</td>
<td>String</td>
<td>PubMLST secondary scheme used by MLST</td>
</tr>
<tr>
<td>ts_mlst_results</td>
<td>File</td>
<td>TSV report with detailed MLST profile, including <a href="https://github.com/tseemann/mlst#missing-data">https://github.com/tseemann/mlst#missing-data</a></td>
</tr>
<tr>
<td>ts_mlst_secondary_allelic_profile</td>
<td>String</td>
<td>Profile of alleles predicted by secondary MLST run</td>
</tr>
<tr>
<td>ts_mlst_secondary_novel_alleles</td>
<td>File</td>
<td>FASTA file containing nucleotide sequence of any alleles that are not in the MLST database used by TheiaProk, from the secondary scheme run</td>
</tr>
<tr>
<td>ts_mlst_version</td>
<td>String</td>
<td>Version of Torsten Seemans MLST tool used</td>
</tr>
<tr>
<td>vibecheck_classification_notes</td>
<td>String</td>
<td>Additional information provided by Vibecheck during classification</td>
</tr>
<tr>
<td>vibecheck_confidence</td>
<td>Float</td>
<td>How confidence lineage assignment is. 0 - uncertain; 100 - Very certain</td>
</tr>
<tr>
<td>vibecheck_docker</td>
<td>String</td>
<td>Docker image used by Vibecheck task</td>
</tr>
<tr>
<td>vibecheck_lineage_report</td>
<td>File</td>
<td>Output CSV file created by Vibecheck</td>
</tr>
<tr>
<td>vibecheck_top_lineage</td>
<td>String</td>
<td>Most likely lineage assigned by Vibecheck</td>
</tr>
<tr>
<td>vibecheck_version</td>
<td>String</td>
<td>Version of Vibecheck used</td>
</tr>
<tr>
<td>virulencefinder_docker</td>
<td>String</td>
<td>VirulenceFinder docker image used</td>
</tr>
<tr>
<td>virulencefinder_hits</td>
<td>String</td>
<td>Virulence genes detected by VirulenceFinder</td>
</tr>
<tr>
<td>virulencefinder_report_tsv</td>
<td>File</td>
<td>Output TSV file created by VirulenceFinder</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>abricate_abaum_database</td>
<td>String</td>
<td>Database of reference A. baumannii plasmid typing genes used for plasmid typing</td>
</tr>
<tr>
<td>abricate_abaum_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_abaum_plasmid_tsv</td>
<td>File</td>
<td><a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a> containing a row for each A. baumannii plasmid type gene found in the sample</td>
</tr>
<tr>
<td>abricate_abaum_plasmid_type_genes</td>
<td>String</td>
<td>A. baumannii Plasmid typing genes found in the sample; from GENE column in <a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a></td>
</tr>
<tr>
<td>abricate_abaum_version</td>
<td>String</td>
<td>Version of abricate used for A. baumannii plasmid typing</td>
</tr>
<tr>
<td>abricate_database</td>
<td>String</td>
<td>Database of reference used with Abricate</td>
</tr>
<tr>
<td>abricate_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_genes</td>
<td>String</td>
<td>Genes found in the sample; from GENE column in <a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a></td>
</tr>
<tr>
<td>abricate_results_tsv</td>
<td>File</td>
<td><a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a> containing a row for each gene found in the sample</td>
</tr>
<tr>
<td>abricate_version</td>
<td>String</td>
<td>Version of abricate used for A. baumannii plasmid typing</td>
</tr>
<tr>
<td>abricate_vibrio_biotype</td>
<td>String</td>
<td>Biotype classification according to tcpA gene sequence (Classical or ElTor)</td>
</tr>
<tr>
<td>abricate_vibrio_ctxA</td>
<td>String</td>
<td>Presence or absence of the ctxA gene</td>
</tr>
<tr>
<td>abricate_vibrio_database</td>
<td>String</td>
<td>Database to use while running abricate</td>
</tr>
<tr>
<td>abricate_vibrio_detailed_tsv</td>
<td>File</td>
<td>Detailed ABRicate output file</td>
</tr>
<tr>
<td>abricate_vibrio_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_vibrio_ompW</td>
<td>String</td>
<td>Presence or absence of the ompW gene</td>
</tr>
<tr>
<td>abricate_vibrio_serogroup</td>
<td>String</td>
<td>Serotype classification as O1 (wbeN gene), O139 (wbfR gene) or not detected.</td>
</tr>
<tr>
<td>abricate_vibrio_toxR</td>
<td>String</td>
<td>Presence or absence of the toxR gene</td>
</tr>
<tr>
<td>abricate_vibrio_version</td>
<td>String</td>
<td>The abricate version run</td>
</tr>
<tr>
<td>agrvate_agr_canonical</td>
<td>String</td>
<td>Canonical or non-canonical agrD</td>
</tr>
<tr>
<td>agrvate_agr_group</td>
<td>String</td>
<td>Agr group</td>
</tr>
<tr>
<td>agrvate_agr_match_score</td>
<td>String</td>
<td>Match score for agr group</td>
</tr>
<tr>
<td>agrvate_agr_multiple</td>
<td>String</td>
<td>If multiple agr groups were found</td>
</tr>
<tr>
<td>agrvate_agr_num_frameshifts</td>
<td>String</td>
<td>Number of frameshifts found in CDS of extracted agr operon</td>
</tr>
<tr>
<td>agrvate_docker</td>
<td>String</td>
<td>The docker used for AgrVATE</td>
</tr>
<tr>
<td>agrvate_results</td>
<td>File</td>
<td>A gzipped tarball of all results</td>
</tr>
<tr>
<td>agrvate_summary</td>
<td>File</td>
<td>The summary file produced</td>
</tr>
<tr>
<td>agrvate_version</td>
<td>String</td>
<td>The version of AgrVATE used</td>
</tr>
<tr>
<td>amr_search_csv</td>
<td>File</td>
<td>CSV formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_docker</td>
<td>String</td>
<td>Docker image used to run AMR_Search</td>
</tr>
<tr>
<td>amr_search_results</td>
<td>File</td>
<td>JSON formatted AMR profile including BLAST results</td>
</tr>
<tr>
<td>amr_search_results_pdf</td>
<td>File</td>
<td>PDF formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_version</td>
<td>String</td>
<td>Version of AMR_Search libraries used</td>
</tr>
<tr>
<td>amrfinderplus_all_report</td>
<td>File</td>
<td>Output TSV file from AMRFinderPlus (described <a href="https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#fields">https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#fields</a>)</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_betalactam_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to beta-lactams</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_carbapenem_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to carbapenem</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_cephalosporin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to cephalosporin</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_cephalothin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to cephalothin</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_methicillin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to methicilin</td>
</tr>
<tr>
<td>amrfinderplus_amr_classes</td>
<td>String</td>
<td>AMRFinderPlus predictions for classes of drugs that genes found in the reads are known to confer resistance to</td>
</tr>
<tr>
<td>amrfinderplus_amr_core_genes</td>
<td>String</td>
<td>AMR genes identified by AMRFinderPlus where the scope is "core"</td>
</tr>
<tr>
<td>amrfinderplus_amr_plus_genes</td>
<td>String</td>
<td>AMR genes identified by AMRFinderPlus where the scope is "plus"</td>
</tr>
<tr>
<td>amrfinderplus_amr_report</td>
<td>File</td>
<td>TSV file detailing AMR genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>amrfinderplus_amr_subclasses</td>
<td>String</td>
<td>More specificity about the drugs that genes identified in the reads confer resistance to</td>
</tr>
<tr>
<td>amrfinderplus_db_version</td>
<td>String</td>
<td>AMRFinderPlus database version used</td>
</tr>
<tr>
<td>amrfinderplus_stress_genes</td>
<td>String</td>
<td>Stress genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_stress_report</td>
<td>File</td>
<td>TSV file detailing stress genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>amrfinderplus_version</td>
<td>String</td>
<td>AMRFinderPlus version used</td>
</tr>
<tr>
<td>amrfinderplus_virulence_genes</td>
<td>String</td>
<td>Virulence genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_virulence_report</td>
<td>File</td>
<td>TSV file detailing virulence genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>ani_highest_percent</td>
<td>Float</td>
<td>Highest ANI between query and any given reference genome (top species match)</td>
</tr>
<tr>
<td>ani_highest_percent_bases_aligned</td>
<td>Float</td>
<td>Percentage of bases aligned between query genome and top species match</td>
</tr>
<tr>
<td>ani_mummer_docker</td>
<td>String</td>
<td>Docker image used to run the ANI_mummer task</td>
</tr>
<tr>
<td>ani_mummer_version</td>
<td>String</td>
<td>Version of MUMmer used</td>
</tr>
<tr>
<td>ani_output_tsv</td>
<td>File</td>
<td>Full output TSV from ani-m</td>
</tr>
<tr>
<td>ani_top_species_match</td>
<td>String</td>
<td>Species of genome with highest ANI to query FASTA</td>
</tr>
<tr>
<td>arln_assembly_ratio</td>
<td>String</td>
<td>Assembly ratio of the final assembly compared to the mean assembly length of the taxon. These stats are gathered from a statistics file created by the CDC aggregating NCBI assembly data by taxon.</td>
</tr>
<tr>
<td>arln_assembly_zscore</td>
<td>String</td>
<td>ZScore of the samples assembly length to assess the uniformity of the assembly when compared to taxon means.</td>
</tr>
<tr>
<td>arln_r1_q30_clean</td>
<td>String</td>
<td>Q30% of the cleaned R1 FASTQ</td>
</tr>
<tr>
<td>arln_r1_q30_raw</td>
<td>String</td>
<td>Q30% of the raw R1 FASTQ</td>
</tr>
<tr>
<td>arln_stats_docker_version</td>
<td>String</td>
<td>Docker version for task_arln_stats.wdl</td>
</tr>
<tr>
<td>arln_taxon_assembly_ratio_stdev</td>
<td>String</td>
<td>Reported assembely length standard deviation of the given taxonomy pulled from NCBI aggregated data.</td>
</tr>
<tr>
<td>arln_taxon_gc_mean</td>
<td>String</td>
<td>Mean GC percent of the taxon.</td>
</tr>
<tr>
<td>arln_taxon_gc_percent_stdev</td>
<td>String</td>
<td>Reported GC percent standard deviation of the given taxonomy pulled from NCBI aggregated data.</td>
</tr>
<tr>
<td>assembler</td>
<td>String</td>
<td>Assembler used in digger_denovo subworkflow</td>
</tr>
<tr>
<td>assembler_version</td>
<td>String</td>
<td>Version of the assembler used in digger_denovo</td>
</tr>
<tr>
<td>assembly_fasta</td>
<td>File</td>
<td><em>De novo</em> genome assembly in FASTA format</td>
</tr>
<tr>
<td>assembly_length</td>
<td>Int</td>
<td>Length of assembly (total contig length) as determined by QUAST</td>
</tr>
<tr>
<td>bakta_gbff</td>
<td>File</td>
<td>Genomic GenBank format annotation file</td>
</tr>
<tr>
<td>bakta_gff3</td>
<td>File</td>
<td>Generic Feature Format Version 3 file</td>
</tr>
<tr>
<td>bakta_plot</td>
<td>File</td>
<td>Bakta plot output PNG file summarizing annotated genome features such as coding sequences, RNA genes, and hypothetical proteins.</td>
</tr>
<tr>
<td>bakta_summary</td>
<td>File</td>
<td>Bakta summary output TXT file</td>
</tr>
<tr>
<td>bakta_tsv</td>
<td>File</td>
<td>Annotations as simple human readable TSV</td>
</tr>
<tr>
<td>bakta_version</td>
<td>String</td>
<td>Bakta version used</td>
</tr>
<tr>
<td>bbduk_docker</td>
<td>String</td>
<td>The Docker image for bbduk, which was used to remove the adapters from the sequences</td>
</tr>
<tr>
<td>busco_database</td>
<td>String</td>
<td>BUSCO database used</td>
</tr>
<tr>
<td>busco_docker</td>
<td>String</td>
<td>BUSCO docker image used</td>
</tr>
<tr>
<td>busco_report</td>
<td>File</td>
<td>A plain text summary of the results in BUSCO notation</td>
</tr>
<tr>
<td>busco_results</td>
<td>String</td>
<td>BUSCO results (see relevant toggle in <a href="https://theiagen.github.io/public_health_bioinformatics/latest/workflows/genomic_characterization/theiaprok/#post-assembly-tasks-performed-for-all-taxa">this block</a>)</td>
</tr>
<tr>
<td>busco_version</td>
<td>String</td>
<td>BUSCO software version used</td>
</tr>
<tr>
<td>cg_pipeline_docker</td>
<td>String</td>
<td>Docker file used for running CG-Pipeline on cleaned reads</td>
</tr>
<tr>
<td>cg_pipeline_report_clean</td>
<td>File</td>
<td>TSV file of read metrics from clean reads, including average read length, number of reads, and estimated genome coverage</td>
</tr>
<tr>
<td>cg_pipeline_report_raw</td>
<td>File</td>
<td>TSV file of read metrics from raw reads, including average read length, number of reads, and estimated genome coverage</td>
</tr>
<tr>
<td>contigs_gfa</td>
<td>File</td>
<td>Assembly graph output generated by SPAdes (Illumina: PE, SE) or Flye (ONT), used to visualize and evaluate genome assembly results.</td>
</tr>
<tr>
<td>ectyper_database_version</td>
<td>String</td>
<td>Version of the ECTyper database used</td>
</tr>
<tr>
<td>ectyper_docker</td>
<td>String</td>
<td>Docker image used to run ECTyper</td>
</tr>
<tr>
<td>ectyper_pathodb_version</td>
<td>String</td>
<td>Version of the ECTyper pathotype database used</td>
</tr>
<tr>
<td>ectyper_pathotype</td>
<td>String</td>
<td>Samples pathotype predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_pathotype_count</td>
<td>String</td>
<td>Number of pathotypes predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_pathotype_genes</td>
<td>String</td>
<td>Pathotype genes observed by ECTyper</td>
</tr>
<tr>
<td>ectyper_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_qc_result</td>
<td>String</td>
<td>QC pass or fail for the sample analyzed</td>
</tr>
<tr>
<td>ectyper_results</td>
<td>File</td>
<td>TSV file of evidence for ECTyper predicted serotype (see <a href="https://github.com/phac-nml/ecoli_serotyping#report-format">https://github.com/phac-nml/ecoli_serotyping#report-format</a>)</td>
</tr>
<tr>
<td>ectyper_stx_subtypes</td>
<td>String</td>
<td>Shiga toxin subtypes predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_version</td>
<td>String</td>
<td>Version of ECTyper used</td>
</tr>
<tr>
<td>ectyper_warnings</td>
<td>File</td>
<td>ECTyper warning messages</td>
</tr>
<tr>
<td>emmtyper_docker</td>
<td>String</td>
<td>Docker image for emmtyper</td>
</tr>
<tr>
<td>emmtyper_emm_type</td>
<td>String</td>
<td>The emm-type of a Streptococcus pyogenes assembly</td>
</tr>
<tr>
<td>emmtyper_results_tsv</td>
<td>File</td>
<td>TSV file with emmtyper results</td>
</tr>
<tr>
<td>emmtyper_version</td>
<td>String</td>
<td>Version of emmtyper used</td>
</tr>
<tr>
<td>est_coverage_clean</td>
<td>Float</td>
<td>Estimated coverage calculated from clean reads and genome length</td>
</tr>
<tr>
<td>est_coverage_raw</td>
<td>Float</td>
<td>Estimated coverage calculated from raw reads and genome length</td>
</tr>
<tr>
<td>fastp_html_report</td>
<td>File</td>
<td>The HTML report made with fastp</td>
</tr>
<tr>
<td>fastp_version</td>
<td>String</td>
<td>The version of fastp used</td>
</tr>
<tr>
<td>fastq_scan_clean1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean1</td>
<td>Int</td>
<td>The number of forward reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw1</td>
<td>Int</td>
<td>The number of input forward reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_raw1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_version</td>
<td>String</td>
<td>The version of fastq_scan</td>
</tr>
<tr>
<td>fastqc_clean1_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of clean forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_docker</td>
<td>String</td>
<td>The Docker container used for fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean1</td>
<td>Int</td>
<td>The number of forward reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_raw1</td>
<td>Int</td>
<td>The number of input forward reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_raw1_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of raw forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_version</td>
<td>String</td>
<td>Version of fastqc software used</td>
</tr>
<tr>
<td>filtered_contigs_metrics</td>
<td>File</td>
<td>File containing metrics of contigs filtered</td>
</tr>
<tr>
<td>gambit_closest_genomes</td>
<td>File</td>
<td>CSV file listing genomes in the GAMBIT database that are most similar to the query assembly</td>
</tr>
<tr>
<td>gambit_db_version</td>
<td>String</td>
<td>Version of the GAMBIT database used</td>
</tr>
<tr>
<td>gambit_docker</td>
<td>String</td>
<td>GAMBIT Docker used</td>
</tr>
<tr>
<td>gambit_predicted_taxon</td>
<td>String</td>
<td>Taxon predicted by GAMBIT</td>
</tr>
<tr>
<td>gambit_predicted_taxon_rank</td>
<td>String</td>
<td>Taxon rank of GAMBIT taxon prediction</td>
</tr>
<tr>
<td>gambit_report</td>
<td>File</td>
<td>GAMBIT report in a machine-readable format</td>
</tr>
<tr>
<td>gambit_version</td>
<td>String</td>
<td>Version of GAMBIT software used</td>
</tr>
<tr>
<td>gamma_docker</td>
<td>String</td>
<td>The Docker container used</td>
</tr>
<tr>
<td>gamma_fasta</td>
<td>File</td>
<td>The FASTA file reporting gene matches if boolean was set to true</td>
</tr>
<tr>
<td>gamma_gff</td>
<td>File</td>
<td>The GFF file reporting gene matches if boolean was set to true</td>
</tr>
<tr>
<td>gamma_results</td>
<td>File</td>
<td>The .gamma file reporting gene matches</td>
</tr>
<tr>
<td>gamma_version</td>
<td>String</td>
<td>The version of GAMMA used</td>
</tr>
<tr>
<td>genotyphi_final_genotype</td>
<td>String</td>
<td>Final genotype call from GenoTyphi</td>
</tr>
<tr>
<td>genotyphi_genotype_confidence</td>
<td>String</td>
<td>Confidence in the final genotype call made by GenoTyphi</td>
</tr>
<tr>
<td>genotyphi_mykrobe_json</td>
<td>File</td>
<td>JSON file of GenoTyphi output, described <a href="https://github.com/katholt/genotyphi#explanation-of-columns-in-the-output">https://github.com/katholt/genotyphi#explanation-of-columns-in-the-output</a></td>
</tr>
<tr>
<td>genotyphi_report_tsv</td>
<td>File</td>
<td>TSV file of GenoTyphi output, described <a href="https://github.com/katholt/genotyphi#explanation-of-columns-in-the-output">https://github.com/katholt/genotyphi#explanation-of-columns-in-the-output</a></td>
</tr>
<tr>
<td>genotyphi_species</td>
<td>String</td>
<td>Species call from Mykrobe, used to run GenoTyphi</td>
</tr>
<tr>
<td>genotyphi_st_probes_percent_coverage</td>
<td>Float</td>
<td>Percentage coverage to the Typhi MLST probes</td>
</tr>
<tr>
<td>genotyphi_version</td>
<td>String</td>
<td>Version of GenoTyphi used</td>
</tr>
<tr>
<td>hicap_docker</td>
<td>String</td>
<td>Docker image used for hicap</td>
</tr>
<tr>
<td>hicap_genes</td>
<td>String</td>
<td>cap genes identified. genes on different contigs delimited by;. truncation shown by trailing *</td>
</tr>
<tr>
<td>hicap_results_tsv</td>
<td>File</td>
<td>TSV file of hicap output</td>
</tr>
<tr>
<td>hicap_serotype</td>
<td>String</td>
<td>hicap serotype</td>
</tr>
<tr>
<td>hicap_version</td>
<td>String</td>
<td>hicap version used</td>
</tr>
<tr>
<td>kaptive_k_locus</td>
<td>String</td>
<td>Best matching K locus identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_k_type</td>
<td>String</td>
<td>Best matching K type identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_kl_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kaptive_oc_locus</td>
<td>String</td>
<td>Best matching K locus identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_ocl_confidence</td>
<td>String</td>
<td>Kaptives confidence in the OCL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kaptive_output_file_k</td>
<td>File</td>
<td>TSV <a href="https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom">https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom</a> the K locus from Kaptive</td>
</tr>
<tr>
<td>kaptive_output_file_oc</td>
<td>File</td>
<td>TSV <a href="https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom">https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom</a> the OC locus from Kaptive</td>
</tr>
<tr>
<td>kaptive_version</td>
<td>String</td>
<td>Version of Kaptive used</td>
</tr>
<tr>
<td>kleborate_docker</td>
<td>String</td>
<td>Kleborate docker image used</td>
</tr>
<tr>
<td>kleborate_genomic_resistance_mutations</td>
<td>String</td>
<td>Genomic resistance mutations identifies by Kleborate</td>
</tr>
<tr>
<td>kleborate_key_resistance_genes</td>
<td>String</td>
<td>Key resistance genes identified by Kleborate</td>
</tr>
<tr>
<td>kleborate_klocus</td>
<td>String</td>
<td>Best matching K locus identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_klocus_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kleborate_ktype</td>
<td>String</td>
<td>Best matching K type identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_mlst_sequence_type</td>
<td>String</td>
<td><a href="https://github.com/katholt/Kleborate/wiki/MLST#multi-locus-sequence-typing-mlst">https://github.com/katholt/Kleborate/wiki/MLST#multi-locus-sequence-typing-mlst</a> call by Kleborate</td>
</tr>
<tr>
<td>kleborate_olocus</td>
<td>String</td>
<td>Best matching OC locus identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_olocus_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kleborate_otype</td>
<td>String</td>
<td>Best matching OC type identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_output_file</td>
<td>File</td>
<td>See also <a href="https://github.com/katholt/Kleborate/wiki/Scores-and-counts">https://github.com/katholt/Kleborate/wiki/Scores-and-counts</a></td>
</tr>
<tr>
<td>kleborate_resistance_score</td>
<td>String</td>
<td>Resistance score as given by kleborate</td>
</tr>
<tr>
<td>kleborate_version</td>
<td>String</td>
<td>Version of Kleborate used</td>
</tr>
<tr>
<td>kleborate_virulence_score</td>
<td>String</td>
<td>Virulence score as given by kleborate</td>
</tr>
<tr>
<td>kmerfinder_database</td>
<td>String</td>
<td>Database used to run KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_docker</td>
<td>String</td>
<td>Docker image used to run KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_query_coverage</td>
<td>String</td>
<td>KmerFinders query coverage of the top hit result</td>
</tr>
<tr>
<td>kmerfinder_results_tsv</td>
<td>File</td>
<td>Output TSV file created by KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_template_coverage</td>
<td>String</td>
<td>Percent of kmer coverage pertaining to the template, or reference selected</td>
</tr>
<tr>
<td>kmerfinder_top_hit</td>
<td>String</td>
<td>Top hit species of KmerFinder</td>
</tr>
<tr>
<td>kraken2_database</td>
<td>String</td>
<td>Kraken2 database used for the taxonomic assignment</td>
</tr>
<tr>
<td>kraken2_docker</td>
<td>String</td>
<td>Docker image used to run kraken2</td>
</tr>
<tr>
<td>kraken2_report</td>
<td>String</td>
<td>TXT document describing taxonomic prediction of every FASTQ record. This file is usually very large and cumbersome to open and view</td>
</tr>
<tr>
<td>kraken2_version</td>
<td>String</td>
<td>The version of kraken2 used</td>
</tr>
<tr>
<td>legsta_predicted_sbt</td>
<td>String</td>
<td>Sequence based type predicted by Legsta</td>
</tr>
<tr>
<td>legsta_results</td>
<td>File</td>
<td>TSV file of legsta results (see <a href="https://github.com/tseemann/legsta#output">https://github.com/tseemann/legsta#output</a>)</td>
</tr>
<tr>
<td>legsta_version</td>
<td>String</td>
<td>Version of legsta used</td>
</tr>
<tr>
<td>lissero_results</td>
<td>File</td>
<td>TSV results file from LisSero (see <a href="https://github.com/MDU-PHL/LisSero#example-output">https://github.com/MDU-PHL/LisSero#example-output</a>)</td>
</tr>
<tr>
<td>lissero_serotype</td>
<td>String</td>
<td>Serotype predicted by LisSero</td>
</tr>
<tr>
<td>lissero_version</td>
<td>String</td>
<td>Version of LisSero used</td>
</tr>
<tr>
<td>meningotype_BAST</td>
<td>String</td>
<td>BAST type</td>
</tr>
<tr>
<td>meningotype_FetA</td>
<td>String</td>
<td>FetA type</td>
</tr>
<tr>
<td>meningotype_NHBA</td>
<td>String</td>
<td>NHBA type</td>
</tr>
<tr>
<td>meningotype_NadA</td>
<td>String</td>
<td>NBA type</td>
</tr>
<tr>
<td>meningotype_PorA</td>
<td>String</td>
<td>PorA type</td>
</tr>
<tr>
<td>meningotype_PorB</td>
<td>String</td>
<td>PorB type</td>
</tr>
<tr>
<td>meningotype_fHbp</td>
<td>String</td>
<td>fHbp type</td>
</tr>
<tr>
<td>meningotype_serogroup</td>
<td>String</td>
<td>Serogroup</td>
</tr>
<tr>
<td>meningotype_tsv</td>
<td>File</td>
<td>Full result file</td>
</tr>
<tr>
<td>meningotype_version</td>
<td>String</td>
<td>Version of meningotype used</td>
</tr>
<tr>
<td>midas_docker</td>
<td>String</td>
<td>MIDAS docker image used</td>
</tr>
<tr>
<td>midas_primary_genus</td>
<td>String</td>
<td>The primary genus detected by MIDAS</td>
</tr>
<tr>
<td>midas_report</td>
<td>File</td>
<td>TSV report of full MIDAS results</td>
</tr>
<tr>
<td>midas_secondary_genus</td>
<td>String</td>
<td>Genus of the next most abundant species after removing all species of the most abundant genus</td>
</tr>
<tr>
<td>midas_secondary_genus_abundance</td>
<td>Float</td>
<td>Relative abundance of secondary genus</td>
</tr>
<tr>
<td>midas_secondary_genus_coverage</td>
<td>Float</td>
<td>Absolute coverage of secondary genus</td>
</tr>
<tr>
<td>n50_value</td>
<td>Int</td>
<td>N50 of assembly calculated by QUAST</td>
</tr>
<tr>
<td>ngmaster_ngmast_porB_allele</td>
<td>String</td>
<td>porB allele number</td>
</tr>
<tr>
<td>ngmaster_ngmast_sequence_type</td>
<td>String</td>
<td>NG-MAST sequence type</td>
</tr>
<tr>
<td>ngmaster_ngmast_tbpB_allele</td>
<td>String</td>
<td>tbpB allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_23S_allele</td>
<td>String</td>
<td>23S rRNA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_gyrA_allele</td>
<td>String</td>
<td>gyrA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_mtrR_allele</td>
<td>String</td>
<td>mtrR allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_parC_allele</td>
<td>String</td>
<td>parC allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_penA_allele</td>
<td>String</td>
<td>penA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_ponA_allele</td>
<td>String</td>
<td>ponA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_porB_allele</td>
<td>String</td>
<td>porB allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_sequence_type</td>
<td>String</td>
<td>NG-STAR sequence type</td>
</tr>
<tr>
<td>ngmaster_tsv</td>
<td>File</td>
<td>TSV file with NG-MAST/NG-STAR typing</td>
</tr>
<tr>
<td>ngmaster_version</td>
<td>String</td>
<td>ngmaster version</td>
</tr>
<tr>
<td>number_contigs</td>
<td>Int</td>
<td>Total number of contigs in assembly</td>
</tr>
<tr>
<td>pasty_all_serogroups</td>
<td>File</td>
<td>TSV file with details of each serogroup from pasty (see <a href="https://github.com/rpetit3/pasty#example-prefixdetailstsv">https://github.com/rpetit3/pasty#example-prefixdetailstsv</a>)</td>
</tr>
<tr>
<td>pasty_blast_hits</td>
<td>File</td>
<td>TSV file of BLAST hits from pasty (see <a href="https://github.com/rpetit3/pasty#example-prefixblastntsv">https://github.com/rpetit3/pasty#example-prefixblastntsv</a>)</td>
</tr>
<tr>
<td>pasty_comment</td>
<td>String</td>
<td></td>
</tr>
<tr>
<td>pasty_docker</td>
<td>String</td>
<td>pasty docker image used</td>
</tr>
<tr>
<td>pasty_serogroup</td>
<td>String</td>
<td>Serogroup predicted by pasty</td>
</tr>
<tr>
<td>pasty_serogroup_coverage</td>
<td>Float</td>
<td>The breadth of coverage of the O-antigen by pasty</td>
</tr>
<tr>
<td>pasty_serogroup_fragments</td>
<td>Int</td>
<td>Number of BLAST hits included in the prediction (fewer is better)</td>
</tr>
<tr>
<td>pasty_summary_tsv</td>
<td>File</td>
<td>TSV summary file of pasty outputs (see <a href="https://github.com/rpetit3/pasty#example-prefixtsv">https://github.com/rpetit3/pasty#example-prefixtsv</a>)</td>
</tr>
<tr>
<td>pasty_version</td>
<td>String</td>
<td>Version of pasty used</td>
</tr>
<tr>
<td>pbptyper_docker</td>
<td>String</td>
<td>pbptyper docker image used</td>
</tr>
<tr>
<td>pbptyper_pbptype_predicted_tsv</td>
<td>File</td>
<td>TSV file of pbptyper results (see <a href="https://github.com/rpetit3/pbptyper#example-prefixtsv">https://github.com/rpetit3/pbptyper#example-prefixtsv</a>)</td>
</tr>
<tr>
<td>pbptyper_predicted_1A_2B_2X</td>
<td>String</td>
<td>PBP type predicted by pbptyper</td>
</tr>
<tr>
<td>pbptyper_version</td>
<td>String</td>
<td>Version of pbptyper used</td>
</tr>
<tr>
<td>plasmidfinder_db_version</td>
<td>String</td>
<td>Version of PlasmidFnder used</td>
</tr>
<tr>
<td>plasmidfinder_docker</td>
<td>String</td>
<td>PlasmidFinder docker image used</td>
</tr>
<tr>
<td>plasmidfinder_plasmids</td>
<td>String</td>
<td>Names of plasmids identified by PlasmidFinder</td>
</tr>
<tr>
<td>plasmidfinder_results</td>
<td>File</td>
<td>Output file from PlasmidFinder in TSV format</td>
</tr>
<tr>
<td>plasmidfinder_seqs</td>
<td>File</td>
<td>Hit_in_genome_seq.fsa file produced by PlasmidFinder</td>
</tr>
<tr>
<td>poppunk_GPS_db_version</td>
<td>String</td>
<td>Version of GPSC database used</td>
</tr>
<tr>
<td>poppunk_docker</td>
<td>String</td>
<td>PopPUNK docker image with GPSC database used</td>
</tr>
<tr>
<td>poppunk_gps_cluster</td>
<td>String</td>
<td>GPS cluster predicted by PopPUNK</td>
</tr>
<tr>
<td>poppunk_gps_external_cluster_csv</td>
<td>File</td>
<td>GPSC v6 scheme designations</td>
</tr>
<tr>
<td>poppunk_version</td>
<td>String</td>
<td>Version of PopPUNK used</td>
</tr>
<tr>
<td>prokka_gbk</td>
<td>File</td>
<td>GenBank file produced from Prokka annotation of input FASTA</td>
</tr>
<tr>
<td>prokka_gff</td>
<td>File</td>
<td>Prokka output GFF3 file containing sequence and annotation (you can view this in IGV)</td>
</tr>
<tr>
<td>prokka_sqn</td>
<td>File</td>
<td>A Sequin file for GenBank submission</td>
</tr>
<tr>
<td>qc_check</td>
<td>String</td>
<td>A string that indicates whether or not the sample passes a set of pre-determined and user-provided QC thresholds</td>
</tr>
<tr>
<td>qc_standard</td>
<td>File</td>
<td>The file used in the QC Check task containing the QC thresholds.</td>
</tr>
<tr>
<td>quast_gc_percent</td>
<td>Float</td>
<td>The GC percent of your sample</td>
</tr>
<tr>
<td>quast_report</td>
<td>File</td>
<td>TSV report from QUAST</td>
</tr>
<tr>
<td>quast_version</td>
<td>String</td>
<td>The version of QUAST</td>
</tr>
<tr>
<td>r1_mean_q_clean</td>
<td>Float</td>
<td>Mean quality score of clean forward reads</td>
</tr>
<tr>
<td>r1_mean_q_raw</td>
<td>Float</td>
<td>Mean quality score of raw forward reads</td>
</tr>
<tr>
<td>r1_mean_readlength_clean</td>
<td>Float</td>
<td>Mean read length of clean forward reads</td>
</tr>
<tr>
<td>r1_mean_readlength_raw</td>
<td>Float</td>
<td>Mean read length of raw forward reads</td>
</tr>
<tr>
<td>read1_clean</td>
<td>File</td>
<td>Forward read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read1_concatenated</td>
<td>File</td>
<td>Concatenated forward reads</td>
</tr>
<tr>
<td>read_screen_clean</td>
<td>String</td>
<td>PASS or FAIL result from clean read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_clean_tsv</td>
<td>File</td>
<td>Clean read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>read_screen_raw</td>
<td>String</td>
<td>PASS or FAIL result from raw read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_raw_tsv</td>
<td>File</td>
<td>Raw read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>resfinder_db_version</td>
<td>String</td>
<td>Version of ResFinder database</td>
</tr>
<tr>
<td>resfinder_docker</td>
<td>String</td>
<td>ResFinder docker image used</td>
</tr>
<tr>
<td>resfinder_pheno_table</td>
<td>File</td>
<td>Table containing al AMR phenotypes</td>
</tr>
<tr>
<td>resfinder_pheno_table_species</td>
<td>File</td>
<td>Table with species-specific AMR phenotypes</td>
</tr>
<tr>
<td>resfinder_pointfinder_pheno_table</td>
<td>File</td>
<td>TSV showing presence(1)/absence(0) of predicted resistance against an antibiotic class</td>
</tr>
<tr>
<td>resfinder_pointfinder_results</td>
<td>File</td>
<td>Predicted point mutations, grouped by the gene they occur in</td>
</tr>
<tr>
<td>resfinder_predicted_pheno_resistance</td>
<td>String</td>
<td>Semicolon delimited list of antimicrobial drugs and associated genes and/or point mutations.&nbsp;<drug1>: <gene1>, <gene1>, <point_mutation1>; <drug2>: <gene3>, <gene4>;</gene4></gene3></drug2></point_mutation1></gene1></gene1></drug1></td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Amp</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ampicillin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Axo</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ceftriaxone based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Azm</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Azithromycin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Cip</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ciprofloxacin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Smx</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Sulfamethoxazole based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Tmp</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Trimothoprim based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_xdr_shigella</td>
<td>String</td>
<td>Final prediction of XDR Shigella status based on CDC definition. Explanation can be found in the description above this table.</td>
</tr>
<tr>
<td>resfinder_results</td>
<td>File</td>
<td>Predicted resistance genes grouped by antibiotic class</td>
</tr>
<tr>
<td>resfinder_seqs</td>
<td>File</td>
<td>FASTA of resistance gene sequences from users input sequence</td>
</tr>
<tr>
<td>seq_platform</td>
<td>String</td>
<td>Description of the sequencing methodology used to generate the input read data</td>
</tr>
<tr>
<td>seqsero2s_note</td>
<td>String</td>
<td>Additional notes produced by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_predicted_antigenic_profile</td>
<td>String</td>
<td>Antigenic profile predicted for Salmonella spp. by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_predicted_contamination</td>
<td>String</td>
<td>Indicates whether contamination between Salmonella with different serotypes was predicted by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_report</td>
<td>String</td>
<td>TSV report produced by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_version</td>
<td>String</td>
<td>Version of SeqSero2S used</td>
</tr>
<tr>
<td>serotypefinder_docker</td>
<td>String</td>
<td>SerotypeFinder docker image used</td>
</tr>
<tr>
<td>serotypefinder_report</td>
<td>File</td>
<td>TSV report produced by SerotypeFinder</td>
</tr>
<tr>
<td>serotypefinder_serotype</td>
<td>String</td>
<td>Serotype predicted by SerotypeFinder</td>
</tr>
<tr>
<td>shigatyper_docker</td>
<td>String</td>
<td>ShigaTyper docker image used</td>
</tr>
<tr>
<td>shigatyper_hits_tsv</td>
<td>File</td>
<td>Detailed TSV report from ShigaTyper (see <a href="https://github.com/CFSAN-Biostatistics/shigatyper#example-prefix-hitstsv">https://github.com/CFSAN-Biostatistics/shigatyper#example-prefix-hitstsv</a>)</td>
</tr>
<tr>
<td>shigatyper_ipaB_presence_absence</td>
<td>String</td>
<td>Presence (+) or absence (-) of ipaB identified by ShigaTyper</td>
</tr>
<tr>
<td>shigatyper_notes</td>
<td>String</td>
<td>Any notes output from ShigaTyper</td>
</tr>
<tr>
<td>shigatyper_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by ShigaTyper</td>
</tr>
<tr>
<td>shigatyper_summary_tsv</td>
<td>File</td>
<td>TSV summary report from ShigaTyper (see <a href="https://github.com/CFSAN-Biostatistics/shigatyper#example-prefixtsv">https://github.com/CFSAN-Biostatistics/shigatyper#example-prefixtsv</a>)</td>
</tr>
<tr>
<td>shigatyper_version</td>
<td>String</td>
<td>Version of ShigaTyper used</td>
</tr>
<tr>
<td>shigeifinder_H_antigen</td>
<td>String</td>
<td>H-antigen gene identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_H_antigen_reads</td>
<td>String</td>
<td>H-antigen gene identified by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_O_antigen</td>
<td>String</td>
<td>O-antigen gene identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_O_antigen_reads</td>
<td>String</td>
<td>O-antigen gene identified by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_cluster</td>
<td>String</td>
<td>Shigella/EIEC cluster identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_cluster_reads</td>
<td>String</td>
<td>Shigella/EIEC cluster identified by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_docker</td>
<td>String</td>
<td>ShigEiFinder docker image used</td>
</tr>
<tr>
<td>shigeifinder_docker_reads</td>
<td>String</td>
<td>ShigEiFinder docker image used using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_ipaH_presence_absence</td>
<td>String</td>
<td>Presence (+) or absence (-) of ipaH identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_ipaH_presence_absence_reads</td>
<td>String</td>
<td>Presence (+) or absence (-) of ipaH identified by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_notes</td>
<td>String</td>
<td>Any notes output from ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_notes_reads</td>
<td>String</td>
<td>Any notes output from ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_num_virulence_plasmid_genes</td>
<td>String</td>
<td>Number of virulence plasmid genes identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_num_virulence_plasmid_genes_reads</td>
<td>String</td>
<td>Number of virulence plasmid genes identified by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_report</td>
<td>File</td>
<td>TSV report from ShigEiFinder (see <a href="https://github.com/LanLab/ShigEiFinder#shigeifinder">https://github.com/LanLab/ShigEiFinder#shigeifinder</a>)</td>
</tr>
<tr>
<td>shigeifinder_report_reads</td>
<td>File</td>
<td>TSV report from ShigEiFinder (see <a href="https://github.com/LanLab/ShigEiFinder#shigeifinder">https://github.com/LanLab/ShigEiFinder#shigeifinder</a>) using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_serotype</td>
<td>String</td>
<td>Serotype predicted by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_serotype_reads</td>
<td>String</td>
<td>Serotype predicted by ShigEiFinder using read files as inputs</td>
</tr>
<tr>
<td>shigeifinder_version</td>
<td>String</td>
<td>ShigEiFinder version used</td>
</tr>
<tr>
<td>shigeifinder_version_reads</td>
<td>String</td>
<td>ShigEiFinder version used using read files as inputs</td>
</tr>
<tr>
<td>sistr_allele_fasta</td>
<td>File</td>
<td>FASTA file of novel cgMLST alleles from SISTR</td>
</tr>
<tr>
<td>sistr_allele_json</td>
<td>File</td>
<td>JSON file of cgMLST allele sequences and information (see <a href="https://github.com/phac-nml/sistr_cmd#cgmlst-allele-search-results">https://github.com/phac-nml/sistr_cmd#cgmlst-allele-search-results</a>)</td>
</tr>
<tr>
<td>sistr_antigenic_formula</td>
<td>String</td>
<td>A field that aggregates the O, H1, and H2, antigen values in a single location for convenience</td>
</tr>
<tr>
<td>sistr_cgmlst</td>
<td>File</td>
<td>CSV file of the cgMLST allelic profile from SISTR (see <a href="https://github.com/phac-nml/sistr_cmd#cgmlst-allelic-profiles-output---cgmlst-profiles-cgmlst-profilescsv">https://github.com/phac-nml/sistr_cmd#cgmlst-allelic-profiles-output---cgmlst-profiles-cgmlst-profilescsv</a>)</td>
</tr>
<tr>
<td>sistr_h1_antigens</td>
<td>String</td>
<td>The predicted H1 antigen</td>
</tr>
<tr>
<td>sistr_h2_antigens</td>
<td>String</td>
<td>The predicted H2 antigen</td>
</tr>
<tr>
<td>sistr_o_antigens</td>
<td>String</td>
<td>The predicted O antigen</td>
</tr>
<tr>
<td>sistr_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by SISTR</td>
</tr>
<tr>
<td>sistr_results</td>
<td>File</td>
<td>TSV results file produced by SISTR (see <a href="https://github.com/phac-nml/sistr_cmd#primary-results-output--o-sistr-results">https://github.com/phac-nml/sistr_cmd#primary-results-output--o-sistr-results</a>)</td>
</tr>
<tr>
<td>sistr_serogroup</td>
<td>String</td>
<td>Serogroup predicted by SISTR</td>
</tr>
<tr>
<td>sistr_serotype_cgmlst</td>
<td>String</td>
<td>cgMLST of the serogroup prediicted by SISTR</td>
</tr>
<tr>
<td>sistr_version</td>
<td>String</td>
<td>Version of SISTR used</td>
</tr>
<tr>
<td>sonneityping_final_genotype</td>
<td>String</td>
<td>Final genotype call from Mykrobe, via sonneityper</td>
</tr>
<tr>
<td>sonneityping_final_report_tsv</td>
<td>File</td>
<td>Detailed TSV report from mykrobe, via sonneityper (see <a href="https://github.com/katholt/sonneityping#example-output">https://github.com/katholt/sonneityping#example-output</a>)</td>
</tr>
<tr>
<td>sonneityping_genotype_confidence</td>
<td>String</td>
<td>Confidence in the final genotype call from sonneityper</td>
</tr>
<tr>
<td>sonneityping_genotype_name</td>
<td>String</td>
<td>Human readable alias for genotype, where available provided by sonneityper</td>
</tr>
<tr>
<td>sonneityping_mykrobe_docker</td>
<td>String</td>
<td>sonneityping docker image used</td>
</tr>
<tr>
<td>sonneityping_mykrobe_report_csv</td>
<td>File</td>
<td>CSV report from mykrobe via sonneityper (see <a href="https://github.com/Mykrobe-tools/mykrobe/wiki/AMR-prediction-output#csv-file">https://github.com/Mykrobe-tools/mykrobe/wiki/AMR-prediction-output#csv-file</a>)</td>
</tr>
<tr>
<td>sonneityping_mykrobe_report_json</td>
<td>File</td>
<td>JSON report from mykrobe via sonneityper (see <a href="https://github.com/Mykrobe-tools/mykrobe/wiki/AMR-prediction-output#json-file">https://github.com/Mykrobe-tools/mykrobe/wiki/AMR-prediction-output#json-file</a>)</td>
</tr>
<tr>
<td>sonneityping_mykrobe_version</td>
<td>String</td>
<td>Version of sonneityping used</td>
</tr>
<tr>
<td>sonneityping_species</td>
<td>String</td>
<td>Species call from Mykrobe via sonneityping</td>
</tr>
<tr>
<td>spatyper_docker</td>
<td>String</td>
<td>spatyper docker image used</td>
</tr>
<tr>
<td>spatyper_repeats</td>
<td>String</td>
<td>order of identified repeats</td>
</tr>
<tr>
<td>spatyper_tsv</td>
<td>File</td>
<td>TSV report with spatyper results</td>
</tr>
<tr>
<td>spatyper_type</td>
<td>String</td>
<td>spa type</td>
</tr>
<tr>
<td>spatyper_version</td>
<td>String</td>
<td>spatyper version used</td>
</tr>
<tr>
<td>srst2_vibrio_biotype</td>
<td>String</td>
<td>Biotype classification according to tcpA gene sequence (Classical or ElTor)</td>
</tr>
<tr>
<td>srst2_vibrio_ctxA</td>
<td>String</td>
<td>Presence or absence of the ctxA gene</td>
</tr>
<tr>
<td>srst2_vibrio_database</td>
<td>String</td>
<td>Database used for srst2 analysis</td>
</tr>
<tr>
<td>srst2_vibrio_detailed_tsv</td>
<td>File</td>
<td>Detailed <a href="https://github.com/katholt/srst2">https://github.com/katholt/srst2</a> output file</td>
</tr>
<tr>
<td>srst2_vibrio_docker</td>
<td>String</td>
<td>Docker image used for srst2 analysis</td>
</tr>
<tr>
<td>srst2_vibrio_ompW</td>
<td>String</td>
<td>Presence or absence of the ompW gene</td>
</tr>
<tr>
<td>srst2_vibrio_serogroup</td>
<td>String</td>
<td>Serotype classification as O1 (wbeN gene), O139 (wbfR gene) or not detected.</td>
</tr>
<tr>
<td>srst2_vibrio_toxR</td>
<td>String</td>
<td>Presence or absence of the toxR gene</td>
</tr>
<tr>
<td>srst2_vibrio_version</td>
<td>String</td>
<td>The SRST2 version run</td>
</tr>
<tr>
<td>staphopiasccmec_docker</td>
<td>String</td>
<td>staphopia-sccmec docker image used</td>
</tr>
<tr>
<td>staphopiasccmec_hamming_distance_tsv</td>
<td>File</td>
<td>staphopia-sccmec version</td>
</tr>
<tr>
<td>staphopiasccmec_results_tsv</td>
<td>File</td>
<td>sccmec types and mecA presence</td>
</tr>
<tr>
<td>staphopiasccmec_types_and_mecA_presence</td>
<td>String</td>
<td>staphopia-sccmec Hamming distance file</td>
</tr>
<tr>
<td>staphopiasccmec_version</td>
<td>String</td>
<td>staphopia-sccmec presence and absence TSV file</td>
</tr>
<tr>
<td>stxtyper_all_hits</td>
<td>String</td>
<td>Comma-separated list of matches of all types. Includes complete, partial, frameshift, internal stop, and novel hits. List is de-duplicated so multiple identical hits are only listed once. For example if 5 partial stx2 hits are detected in the genome, only 1 "stx2" will be listed in this field. To view the potential subtype for each partial hit, the user will need to view the stxtyper_report TSV file.</td>
</tr>
<tr>
<td>stxtyper_ambiguous_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "AMBIGUOUS". Ambiguous bases found in the query sequence (e.g., N)</td>
</tr>
<tr>
<td>stxtyper_complete_operons</td>
<td>String</td>
<td>Comma-separated list of all COMPLETE operons detected by StxTyper. Show multiple hits if present in results.</td>
</tr>
<tr>
<td>stxtyper_docker</td>
<td>String</td>
<td>Name of docker image used by the stxtyper task.</td>
</tr>
<tr>
<td>stxtyper_extended_operons</td>
<td>String</td>
<td>Comma-separated list of all EXTENDED operons detected by StxTyper if coding sequence extends beyond the reference stop codon for one or both of the reference proteins.</td>
</tr>
<tr>
<td>stxtyper_novel_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "COMPLETE_NOVEL". Possible outputs "stx1", "stx2", or "stx1,stx2"</td>
</tr>
<tr>
<td>stxtyper_num_hits</td>
<td>Int</td>
<td>Number of "hits" or rows present in the <code>stxtyper_report</code> TSV file</td>
</tr>
<tr>
<td>stxtyper_partial_hits</td>
<td>String</td>
<td>Possible outputs "stx1", "stx2", or "stx1,stx2". Tells the user that there was a partial hit to either the A or B subunit, but does not describe which subunit, only the possible types from the PARTIAL matches.</td>
</tr>
<tr>
<td>stxtyper_report</td>
<td>File</td>
<td>Raw results TSV file produced by StxTyper</td>
</tr>
<tr>
<td>stxtyper_stx_frameshifts_or_internal_stop_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "FRAMESHIFT" or "INTERNAL_STOP". Possible outputs "stx1", "stx2", or "stx1,stx2"</td>
</tr>
<tr>
<td>stxtyper_version</td>
<td>String</td>
<td>Version of StxTyper used</td>
</tr>
<tr>
<td>taxon_table_status</td>
<td>String</td>
<td>Status of the taxon table upload</td>
</tr>
<tr>
<td>tbp_parser_average_genome_depth</td>
<td>Float</td>
<td>The mean depth of coverage across all target regions included in the analysis</td>
</tr>
<tr>
<td>tbp_parser_coverage_report</td>
<td>File</td>
<td>A file containing the breadth of coverage across each target loci</td>
</tr>
<tr>
<td>tbp_parser_genome_percent_coverage</td>
<td>Float</td>
<td>The percent breadth of coverage across the entire genome</td>
</tr>
<tr>
<td>tbp_parser_laboratorian_report_csv</td>
<td>File</td>
<td>An output file containing information regarding each mutation and its associated drug resistance profile in a CSV file. This file also contains two interpretation fields -- "Looker" and "MDL" which are generated using the CDC's expert rules for interpreting the severity of potential drug resistance mutations.</td>
</tr>
<tr>
<td>tbp_parser_lims_report_csv</td>
<td>File</td>
<td>An output file formatted specifically for STAR LIMS. This CSV report summarizes the highest severity mutations for each antimicrobial and lists the relevant mutations for each gene.</td>
</tr>
<tr>
<td>tbp_parser_looker_report_csv</td>
<td>File</td>
<td>An output file that contains condensed information suitable for generating a dashboard in Google's Looker studio.</td>
</tr>
<tr>
<td>tbprofiler_dr_type</td>
<td>String</td>
<td>The drug resistance category as determined by TBProfiler (sensitive, Pre-MDR, MDR, Pre-XDR, XDR)</td>
</tr>
<tr>
<td>tbprofiler_main_lineage</td>
<td>String</td>
<td>The Mycobacterium tuberculosis lineage assignment as made by TBProfiler</td>
</tr>
<tr>
<td>tbprofiler_output_bai</td>
<td>File</td>
<td>Index BAM file generated by mapping sequencing reads to reference genome by TBProfiler</td>
</tr>
<tr>
<td>tbprofiler_output_bam</td>
<td>File</td>
<td>BAM alignment file produced by TBProfiler</td>
</tr>
<tr>
<td>tbprofiler_output_file</td>
<td>File</td>
<td>CSV report from TBProfiler</td>
</tr>
<tr>
<td>tbprofiler_output_vcf</td>
<td>File</td>
<td>VCF file output from TBProfiler; the concatenation of all of the different VCF files produced during TBProfiler analysis</td>
</tr>
<tr>
<td>tbprofiler_resistance_genes</td>
<td>String</td>
<td>The genes in which a mutation was detected that may be resistance conferring, in the format of <code>&lt;gene target&gt; &lt;variant detected&gt; (&lt;estimated fraction of reads that support the variant&gt;)</code></td>
</tr>
<tr>
<td>tbprofiler_sub_lineage</td>
<td>String</td>
<td>The Mycobacterium tuberculosis sub-lineage assignment as made by TBProfiler</td>
</tr>
<tr>
<td>tbprofiler_version</td>
<td>String</td>
<td>The version of TBProfiler used for this analysis</td>
</tr>
<tr>
<td>theiaprok_illumina_se_analysis_date</td>
<td>String</td>
<td>Date of TheiaProk SE workflow execution</td>
</tr>
<tr>
<td>theiaprok_illumina_se_version</td>
<td>String</td>
<td>Version of TheiaProk SE workflow execution</td>
</tr>
<tr>
<td>trimmomatic_docker</td>
<td>String</td>
<td>The docker image used for the trimmomatic module in this workflow</td>
</tr>
<tr>
<td>trimmomatic_version</td>
<td>String</td>
<td>The version of Trimmomatic used</td>
</tr>
<tr>
<td>ts_mlst_allelic_profile</td>
<td>String</td>
<td>Profile of MLST loci and allele numbers predicted by MLST</td>
</tr>
<tr>
<td>ts_mlst_docker</td>
<td>String</td>
<td>Docker image used for MLST</td>
</tr>
<tr>
<td>ts_mlst_novel_alleles</td>
<td>File</td>
<td>FASTA file containing nucleotide sequence of any alleles that are not in the MLST database used by TheiaProk</td>
</tr>
<tr>
<td>ts_mlst_predicted_secondary_st</td>
<td>String</td>
<td>ST predicted by secondary MLST run</td>
</tr>
<tr>
<td>ts_mlst_predicted_st</td>
<td>String</td>
<td>ST predicted by MLST</td>
</tr>
<tr>
<td>ts_mlst_pubmlst_scheme</td>
<td>String</td>
<td>PubMLST scheme used byMLST</td>
</tr>
<tr>
<td>ts_mlst_pubmlst_secondary_scheme</td>
<td>String</td>
<td>PubMLST secondary scheme used by MLST</td>
</tr>
<tr>
<td>ts_mlst_results</td>
<td>File</td>
<td>TSV report with detailed MLST profile, including <a href="https://github.com/tseemann/mlst#missing-data">https://github.com/tseemann/mlst#missing-data</a></td>
</tr>
<tr>
<td>ts_mlst_secondary_allelic_profile</td>
<td>String</td>
<td>Profile of alleles predicted by secondary MLST run</td>
</tr>
<tr>
<td>ts_mlst_secondary_novel_alleles</td>
<td>File</td>
<td>FASTA file containing nucleotide sequence of any alleles that are not in the MLST database used by TheiaProk, from the secondary scheme run</td>
</tr>
<tr>
<td>ts_mlst_version</td>
<td>String</td>
<td>Version of Torsten Seemans MLST tool used</td>
</tr>
<tr>
<td>virulencefinder_docker</td>
<td>String</td>
<td>VirulenceFinder docker image used</td>
</tr>
<tr>
<td>virulencefinder_hits</td>
<td>String</td>
<td>Virulence genes detected by VirulenceFinder</td>
</tr>
<tr>
<td>virulencefinder_report_tsv</td>
<td>File</td>
<td>Output TSV file created by VirulenceFinder</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>abricate_abaum_database</td>
<td>String</td>
<td>Database of reference A. baumannii plasmid typing genes used for plasmid typing</td>
</tr>
<tr>
<td>abricate_abaum_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_abaum_plasmid_tsv</td>
<td>File</td>
<td><a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a> containing a row for each A. baumannii plasmid type gene found in the sample</td>
</tr>
<tr>
<td>abricate_abaum_plasmid_type_genes</td>
<td>String</td>
<td>A. baumannii Plasmid typing genes found in the sample; from GENE column in <a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a></td>
</tr>
<tr>
<td>abricate_abaum_version</td>
<td>String</td>
<td>Version of abricate used for A. baumannii plasmid typing</td>
</tr>
<tr>
<td>abricate_database</td>
<td>String</td>
<td>Database of reference used with Abricate</td>
</tr>
<tr>
<td>abricate_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_genes</td>
<td>String</td>
<td>Genes found in the sample; from GENE column in <a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a></td>
</tr>
<tr>
<td>abricate_results_tsv</td>
<td>File</td>
<td><a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a> containing a row for each gene found in the sample</td>
</tr>
<tr>
<td>abricate_version</td>
<td>String</td>
<td>Version of abricate used for A. baumannii plasmid typing</td>
</tr>
<tr>
<td>abricate_vibrio_biotype</td>
<td>String</td>
<td>Biotype classification according to tcpA gene sequence (Classical or ElTor)</td>
</tr>
<tr>
<td>abricate_vibrio_ctxA</td>
<td>String</td>
<td>Presence or absence of the ctxA gene</td>
</tr>
<tr>
<td>abricate_vibrio_database</td>
<td>String</td>
<td>Database to use while running abricate</td>
</tr>
<tr>
<td>abricate_vibrio_detailed_tsv</td>
<td>File</td>
<td>Detailed ABRicate output file</td>
</tr>
<tr>
<td>abricate_vibrio_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_vibrio_ompW</td>
<td>String</td>
<td>Presence or absence of the ompW gene</td>
</tr>
<tr>
<td>abricate_vibrio_serogroup</td>
<td>String</td>
<td>Serotype classification as O1 (wbeN gene), O139 (wbfR gene) or not detected.</td>
</tr>
<tr>
<td>abricate_vibrio_toxR</td>
<td>String</td>
<td>Presence or absence of the toxR gene</td>
</tr>
<tr>
<td>abricate_vibrio_version</td>
<td>String</td>
<td>The abricate version run</td>
</tr>
<tr>
<td>agrvate_agr_canonical</td>
<td>String</td>
<td>Canonical or non-canonical agrD</td>
</tr>
<tr>
<td>agrvate_agr_group</td>
<td>String</td>
<td>Agr group</td>
</tr>
<tr>
<td>agrvate_agr_match_score</td>
<td>String</td>
<td>Match score for agr group</td>
</tr>
<tr>
<td>agrvate_agr_multiple</td>
<td>String</td>
<td>If multiple agr groups were found</td>
</tr>
<tr>
<td>agrvate_agr_num_frameshifts</td>
<td>String</td>
<td>Number of frameshifts found in CDS of extracted agr operon</td>
</tr>
<tr>
<td>agrvate_docker</td>
<td>String</td>
<td>The docker used for AgrVATE</td>
</tr>
<tr>
<td>agrvate_results</td>
<td>File</td>
<td>A gzipped tarball of all results</td>
</tr>
<tr>
<td>agrvate_summary</td>
<td>File</td>
<td>The summary file produced</td>
</tr>
<tr>
<td>agrvate_version</td>
<td>String</td>
<td>The version of AgrVATE used</td>
</tr>
<tr>
<td>amr_search_csv</td>
<td>File</td>
<td>CSV formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_docker</td>
<td>String</td>
<td>Docker image used to run AMR_Search</td>
</tr>
<tr>
<td>amr_search_results</td>
<td>File</td>
<td>JSON formatted AMR profile including BLAST results</td>
</tr>
<tr>
<td>amr_search_results_pdf</td>
<td>File</td>
<td>PDF formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_version</td>
<td>String</td>
<td>Version of AMR_Search libraries used</td>
</tr>
<tr>
<td>amrfinderplus_all_report</td>
<td>File</td>
<td>Output TSV file from AMRFinderPlus (described <a href="https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#fields">https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#fields</a>)</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_betalactam_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to beta-lactams</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_carbapenem_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to carbapenem</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_cephalosporin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to cephalosporin</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_cephalothin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to cephalothin</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_methicillin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to methicilin</td>
</tr>
<tr>
<td>amrfinderplus_amr_classes</td>
<td>String</td>
<td>AMRFinderPlus predictions for classes of drugs that genes found in the reads are known to confer resistance to</td>
</tr>
<tr>
<td>amrfinderplus_amr_core_genes</td>
<td>String</td>
<td>AMR genes identified by AMRFinderPlus where the scope is "core"</td>
</tr>
<tr>
<td>amrfinderplus_amr_plus_genes</td>
<td>String</td>
<td>AMR genes identified by AMRFinderPlus where the scope is "plus"</td>
</tr>
<tr>
<td>amrfinderplus_amr_report</td>
<td>File</td>
<td>TSV file detailing AMR genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>amrfinderplus_amr_subclasses</td>
<td>String</td>
<td>More specificity about the drugs that genes identified in the reads confer resistance to</td>
</tr>
<tr>
<td>amrfinderplus_db_version</td>
<td>String</td>
<td>AMRFinderPlus database version used</td>
</tr>
<tr>
<td>amrfinderplus_stress_genes</td>
<td>String</td>
<td>Stress genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_stress_report</td>
<td>File</td>
<td>TSV file detailing stress genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>amrfinderplus_version</td>
<td>String</td>
<td>AMRFinderPlus version used</td>
</tr>
<tr>
<td>amrfinderplus_virulence_genes</td>
<td>String</td>
<td>Virulence genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_virulence_report</td>
<td>File</td>
<td>TSV file detailing virulence genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>ani_highest_percent</td>
<td>Float</td>
<td>Highest ANI between query and any given reference genome (top species match)</td>
</tr>
<tr>
<td>ani_highest_percent_bases_aligned</td>
<td>Float</td>
<td>Percentage of bases aligned between query genome and top species match</td>
</tr>
<tr>
<td>ani_mummer_docker</td>
<td>String</td>
<td>Docker image used to run the ANI_mummer task</td>
</tr>
<tr>
<td>ani_mummer_version</td>
<td>String</td>
<td>Version of MUMmer used</td>
</tr>
<tr>
<td>ani_output_tsv</td>
<td>File</td>
<td>Full output TSV from ani-m</td>
</tr>
<tr>
<td>ani_top_species_match</td>
<td>String</td>
<td>Species of genome with highest ANI to query FASTA</td>
</tr>
<tr>
<td>arln_assembly_ratio</td>
<td>String</td>
<td>Assembly ratio of the final assembly compared to the mean assembly length of the taxon. These stats are gathered from a statistics file created by the CDC aggregating NCBI assembly data by taxon.</td>
</tr>
<tr>
<td>arln_assembly_zscore</td>
<td>String</td>
<td>ZScore of the samples assembly length to assess the uniformity of the assembly when compared to taxon means.</td>
</tr>
<tr>
<td>arln_stats_docker_version</td>
<td>String</td>
<td>Docker version for task_arln_stats.wdl</td>
</tr>
<tr>
<td>arln_taxon_assembly_ratio_stdev</td>
<td>String</td>
<td>Reported assembely length standard deviation of the given taxonomy pulled from NCBI aggregated data.</td>
</tr>
<tr>
<td>arln_taxon_gc_mean</td>
<td>String</td>
<td>Mean GC percent of the taxon.</td>
</tr>
<tr>
<td>arln_taxon_gc_percent_stdev</td>
<td>String</td>
<td>Reported GC percent standard deviation of the given taxonomy pulled from NCBI aggregated data.</td>
</tr>
<tr>
<td>assembly_length</td>
<td>Int</td>
<td>Length of assembly (total contig length) as determined by QUAST</td>
</tr>
<tr>
<td>bakta_gbff</td>
<td>File</td>
<td>Genomic GenBank format annotation file</td>
</tr>
<tr>
<td>bakta_gff3</td>
<td>File</td>
<td>Generic Feature Format Version 3 file</td>
</tr>
<tr>
<td>bakta_plot</td>
<td>File</td>
<td>Bakta plot output PNG file summarizing annotated genome features such as coding sequences, RNA genes, and hypothetical proteins.</td>
</tr>
<tr>
<td>bakta_summary</td>
<td>File</td>
<td>Bakta summary output TXT file</td>
</tr>
<tr>
<td>bakta_tsv</td>
<td>File</td>
<td>Annotations as simple human readable TSV</td>
</tr>
<tr>
<td>bakta_version</td>
<td>String</td>
<td>Bakta version used</td>
</tr>
<tr>
<td>busco_database</td>
<td>String</td>
<td>BUSCO database used</td>
</tr>
<tr>
<td>busco_docker</td>
<td>String</td>
<td>BUSCO docker image used</td>
</tr>
<tr>
<td>busco_report</td>
<td>File</td>
<td>A plain text summary of the results in BUSCO notation</td>
</tr>
<tr>
<td>busco_results</td>
<td>String</td>
<td>BUSCO results (see relevant toggle in <a href="https://theiagen.github.io/public_health_bioinformatics/latest/workflows/genomic_characterization/theiaprok/#post-assembly-tasks-performed-for-all-taxa">this block</a>)</td>
</tr>
<tr>
<td>busco_version</td>
<td>String</td>
<td>BUSCO software version used</td>
</tr>
<tr>
<td>ectyper_database_version</td>
<td>String</td>
<td>Version of the ECTyper database used</td>
</tr>
<tr>
<td>ectyper_docker</td>
<td>String</td>
<td>Docker image used to run ECTyper</td>
</tr>
<tr>
<td>ectyper_pathodb_version</td>
<td>String</td>
<td>Version of the ECTyper pathotype database used</td>
</tr>
<tr>
<td>ectyper_pathotype</td>
<td>String</td>
<td>Samples pathotype predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_pathotype_count</td>
<td>String</td>
<td>Number of pathotypes predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_pathotype_genes</td>
<td>String</td>
<td>Pathotype genes observed by ECTyper</td>
</tr>
<tr>
<td>ectyper_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_qc_result</td>
<td>String</td>
<td>QC pass or fail for the sample analyzed</td>
</tr>
<tr>
<td>ectyper_results</td>
<td>File</td>
<td>TSV file of evidence for ECTyper predicted serotype (see <a href="https://github.com/phac-nml/ecoli_serotyping#report-format">https://github.com/phac-nml/ecoli_serotyping#report-format</a>)</td>
</tr>
<tr>
<td>ectyper_stx_subtypes</td>
<td>String</td>
<td>Shiga toxin subtypes predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_version</td>
<td>String</td>
<td>Version of ECTyper used</td>
</tr>
<tr>
<td>ectyper_warnings</td>
<td>File</td>
<td>ECTyper warning messages</td>
</tr>
<tr>
<td>emmtyper_docker</td>
<td>String</td>
<td>Docker image for emmtyper</td>
</tr>
<tr>
<td>emmtyper_emm_type</td>
<td>String</td>
<td>The emm-type of a Streptococcus pyogenes assembly</td>
</tr>
<tr>
<td>emmtyper_results_tsv</td>
<td>File</td>
<td>TSV file with emmtyper results</td>
</tr>
<tr>
<td>emmtyper_version</td>
<td>String</td>
<td>Version of emmtyper used</td>
</tr>
<tr>
<td>gambit_closest_genomes</td>
<td>File</td>
<td>CSV file listing genomes in the GAMBIT database that are most similar to the query assembly</td>
</tr>
<tr>
<td>gambit_db_version</td>
<td>String</td>
<td>Version of the GAMBIT database used</td>
</tr>
<tr>
<td>gambit_docker</td>
<td>String</td>
<td>GAMBIT Docker used</td>
</tr>
<tr>
<td>gambit_predicted_taxon</td>
<td>String</td>
<td>Taxon predicted by GAMBIT</td>
</tr>
<tr>
<td>gambit_predicted_taxon_rank</td>
<td>String</td>
<td>Taxon rank of GAMBIT taxon prediction</td>
</tr>
<tr>
<td>gambit_report</td>
<td>File</td>
<td>GAMBIT report in a machine-readable format</td>
</tr>
<tr>
<td>gambit_version</td>
<td>String</td>
<td>Version of GAMBIT software used</td>
</tr>
<tr>
<td>gamma_docker</td>
<td>String</td>
<td>The Docker container used</td>
</tr>
<tr>
<td>gamma_fasta</td>
<td>File</td>
<td>The FASTA file reporting gene matches if boolean was set to true</td>
</tr>
<tr>
<td>gamma_gff</td>
<td>File</td>
<td>The GFF file reporting gene matches if boolean was set to true</td>
</tr>
<tr>
<td>gamma_results</td>
<td>File</td>
<td>The .gamma file reporting gene matches</td>
</tr>
<tr>
<td>gamma_version</td>
<td>String</td>
<td>The version of GAMMA used</td>
</tr>
<tr>
<td>hicap_docker</td>
<td>String</td>
<td>Docker image used for hicap</td>
</tr>
<tr>
<td>hicap_genes</td>
<td>String</td>
<td>cap genes identified. genes on different contigs delimited by;. truncation shown by trailing *</td>
</tr>
<tr>
<td>hicap_results_tsv</td>
<td>File</td>
<td>TSV file of hicap output</td>
</tr>
<tr>
<td>hicap_serotype</td>
<td>String</td>
<td>hicap serotype</td>
</tr>
<tr>
<td>hicap_version</td>
<td>String</td>
<td>hicap version used</td>
</tr>
<tr>
<td>kaptive_k_locus</td>
<td>String</td>
<td>Best matching K locus identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_k_type</td>
<td>String</td>
<td>Best matching K type identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_kl_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kaptive_oc_locus</td>
<td>String</td>
<td>Best matching K locus identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_ocl_confidence</td>
<td>String</td>
<td>Kaptives confidence in the OCL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kaptive_output_file_k</td>
<td>File</td>
<td>TSV <a href="https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom">https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom</a> the K locus from Kaptive</td>
</tr>
<tr>
<td>kaptive_output_file_oc</td>
<td>File</td>
<td>TSV <a href="https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom">https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom</a> the OC locus from Kaptive</td>
</tr>
<tr>
<td>kaptive_version</td>
<td>String</td>
<td>Version of Kaptive used</td>
</tr>
<tr>
<td>kleborate_docker</td>
<td>String</td>
<td>Kleborate docker image used</td>
</tr>
<tr>
<td>kleborate_genomic_resistance_mutations</td>
<td>String</td>
<td>Genomic resistance mutations identifies by Kleborate</td>
</tr>
<tr>
<td>kleborate_key_resistance_genes</td>
<td>String</td>
<td>Key resistance genes identified by Kleborate</td>
</tr>
<tr>
<td>kleborate_klocus</td>
<td>String</td>
<td>Best matching K locus identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_klocus_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kleborate_ktype</td>
<td>String</td>
<td>Best matching K type identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_mlst_sequence_type</td>
<td>String</td>
<td><a href="https://github.com/katholt/Kleborate/wiki/MLST#multi-locus-sequence-typing-mlst">https://github.com/katholt/Kleborate/wiki/MLST#multi-locus-sequence-typing-mlst</a> call by Kleborate</td>
</tr>
<tr>
<td>kleborate_olocus</td>
<td>String</td>
<td>Best matching OC locus identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_olocus_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kleborate_otype</td>
<td>String</td>
<td>Best matching OC type identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_output_file</td>
<td>File</td>
<td>See also <a href="https://github.com/katholt/Kleborate/wiki/Scores-and-counts">https://github.com/katholt/Kleborate/wiki/Scores-and-counts</a></td>
</tr>
<tr>
<td>kleborate_resistance_score</td>
<td>String</td>
<td>Resistance score as given by kleborate</td>
</tr>
<tr>
<td>kleborate_version</td>
<td>String</td>
<td>Version of Kleborate used</td>
</tr>
<tr>
<td>kleborate_virulence_score</td>
<td>String</td>
<td>Virulence score as given by kleborate</td>
</tr>
<tr>
<td>kmerfinder_database</td>
<td>String</td>
<td>Database used to run KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_docker</td>
<td>String</td>
<td>Docker image used to run KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_query_coverage</td>
<td>String</td>
<td>KmerFinders query coverage of the top hit result</td>
</tr>
<tr>
<td>kmerfinder_results_tsv</td>
<td>File</td>
<td>Output TSV file created by KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_template_coverage</td>
<td>String</td>
<td>Percent of kmer coverage pertaining to the template, or reference selected</td>
</tr>
<tr>
<td>kmerfinder_top_hit</td>
<td>String</td>
<td>Top hit species of KmerFinder</td>
</tr>
<tr>
<td>legsta_predicted_sbt</td>
<td>String</td>
<td>Sequence based type predicted by Legsta</td>
</tr>
<tr>
<td>legsta_results</td>
<td>File</td>
<td>TSV file of legsta results (see <a href="https://github.com/tseemann/legsta#output">https://github.com/tseemann/legsta#output</a>)</td>
</tr>
<tr>
<td>legsta_version</td>
<td>String</td>
<td>Version of legsta used</td>
</tr>
<tr>
<td>lissero_results</td>
<td>File</td>
<td>TSV results file from LisSero (see <a href="https://github.com/MDU-PHL/LisSero#example-output">https://github.com/MDU-PHL/LisSero#example-output</a>)</td>
</tr>
<tr>
<td>lissero_serotype</td>
<td>String</td>
<td>Serotype predicted by LisSero</td>
</tr>
<tr>
<td>lissero_version</td>
<td>String</td>
<td>Version of LisSero used</td>
</tr>
<tr>
<td>meningotype_BAST</td>
<td>String</td>
<td>BAST type</td>
</tr>
<tr>
<td>meningotype_FetA</td>
<td>String</td>
<td>FetA type</td>
</tr>
<tr>
<td>meningotype_NHBA</td>
<td>String</td>
<td>NHBA type</td>
</tr>
<tr>
<td>meningotype_NadA</td>
<td>String</td>
<td>NBA type</td>
</tr>
<tr>
<td>meningotype_PorA</td>
<td>String</td>
<td>PorA type</td>
</tr>
<tr>
<td>meningotype_PorB</td>
<td>String</td>
<td>PorB type</td>
</tr>
<tr>
<td>meningotype_fHbp</td>
<td>String</td>
<td>fHbp type</td>
</tr>
<tr>
<td>meningotype_serogroup</td>
<td>String</td>
<td>Serogroup</td>
</tr>
<tr>
<td>meningotype_tsv</td>
<td>File</td>
<td>Full result file</td>
</tr>
<tr>
<td>meningotype_version</td>
<td>String</td>
<td>Version of meningotype used</td>
</tr>
<tr>
<td>n50_value</td>
<td>Int</td>
<td>N50 of assembly calculated by QUAST</td>
</tr>
<tr>
<td>ngmaster_ngmast_porB_allele</td>
<td>String</td>
<td>porB allele number</td>
</tr>
<tr>
<td>ngmaster_ngmast_sequence_type</td>
<td>String</td>
<td>NG-MAST sequence type</td>
</tr>
<tr>
<td>ngmaster_ngmast_tbpB_allele</td>
<td>String</td>
<td>tbpB allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_23S_allele</td>
<td>String</td>
<td>23S rRNA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_gyrA_allele</td>
<td>String</td>
<td>gyrA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_mtrR_allele</td>
<td>String</td>
<td>mtrR allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_parC_allele</td>
<td>String</td>
<td>parC allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_penA_allele</td>
<td>String</td>
<td>penA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_ponA_allele</td>
<td>String</td>
<td>ponA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_porB_allele</td>
<td>String</td>
<td>porB allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_sequence_type</td>
<td>String</td>
<td>NG-STAR sequence type</td>
</tr>
<tr>
<td>ngmaster_tsv</td>
<td>File</td>
<td>TSV file with NG-MAST/NG-STAR typing</td>
</tr>
<tr>
<td>ngmaster_version</td>
<td>String</td>
<td>ngmaster version</td>
</tr>
<tr>
<td>number_contigs</td>
<td>Int</td>
<td>Total number of contigs in assembly</td>
</tr>
<tr>
<td>pasty_all_serogroups</td>
<td>File</td>
<td>TSV file with details of each serogroup from pasty (see <a href="https://github.com/rpetit3/pasty#example-prefixdetailstsv">https://github.com/rpetit3/pasty#example-prefixdetailstsv</a>)</td>
</tr>
<tr>
<td>pasty_blast_hits</td>
<td>File</td>
<td>TSV file of BLAST hits from pasty (see <a href="https://github.com/rpetit3/pasty#example-prefixblastntsv">https://github.com/rpetit3/pasty#example-prefixblastntsv</a>)</td>
</tr>
<tr>
<td>pasty_comment</td>
<td>String</td>
<td></td>
</tr>
<tr>
<td>pasty_docker</td>
<td>String</td>
<td>pasty docker image used</td>
</tr>
<tr>
<td>pasty_serogroup</td>
<td>String</td>
<td>Serogroup predicted by pasty</td>
</tr>
<tr>
<td>pasty_serogroup_coverage</td>
<td>Float</td>
<td>The breadth of coverage of the O-antigen by pasty</td>
</tr>
<tr>
<td>pasty_serogroup_fragments</td>
<td>Int</td>
<td>Number of BLAST hits included in the prediction (fewer is better)</td>
</tr>
<tr>
<td>pasty_summary_tsv</td>
<td>File</td>
<td>TSV summary file of pasty outputs (see <a href="https://github.com/rpetit3/pasty#example-prefixtsv">https://github.com/rpetit3/pasty#example-prefixtsv</a>)</td>
</tr>
<tr>
<td>pasty_version</td>
<td>String</td>
<td>Version of pasty used</td>
</tr>
<tr>
<td>pbptyper_docker</td>
<td>String</td>
<td>pbptyper docker image used</td>
</tr>
<tr>
<td>pbptyper_pbptype_predicted_tsv</td>
<td>File</td>
<td>TSV file of pbptyper results (see <a href="https://github.com/rpetit3/pbptyper#example-prefixtsv">https://github.com/rpetit3/pbptyper#example-prefixtsv</a>)</td>
</tr>
<tr>
<td>pbptyper_predicted_1A_2B_2X</td>
<td>String</td>
<td>PBP type predicted by pbptyper</td>
</tr>
<tr>
<td>pbptyper_version</td>
<td>String</td>
<td>Version of pbptyper used</td>
</tr>
<tr>
<td>plasmidfinder_db_version</td>
<td>String</td>
<td>Version of PlasmidFnder used</td>
</tr>
<tr>
<td>plasmidfinder_docker</td>
<td>String</td>
<td>PlasmidFinder docker image used</td>
</tr>
<tr>
<td>plasmidfinder_plasmids</td>
<td>String</td>
<td>Names of plasmids identified by PlasmidFinder</td>
</tr>
<tr>
<td>plasmidfinder_results</td>
<td>File</td>
<td>Output file from PlasmidFinder in TSV format</td>
</tr>
<tr>
<td>plasmidfinder_seqs</td>
<td>File</td>
<td>Hit_in_genome_seq.fsa file produced by PlasmidFinder</td>
</tr>
<tr>
<td>poppunk_GPS_db_version</td>
<td>String</td>
<td>Version of GPSC database used</td>
</tr>
<tr>
<td>poppunk_docker</td>
<td>String</td>
<td>PopPUNK docker image with GPSC database used</td>
</tr>
<tr>
<td>poppunk_gps_cluster</td>
<td>String</td>
<td>GPS cluster predicted by PopPUNK</td>
</tr>
<tr>
<td>poppunk_gps_external_cluster_csv</td>
<td>File</td>
<td>GPSC v6 scheme designations</td>
</tr>
<tr>
<td>poppunk_version</td>
<td>String</td>
<td>Version of PopPUNK used</td>
</tr>
<tr>
<td>prokka_gbk</td>
<td>File</td>
<td>GenBank file produced from Prokka annotation of input FASTA</td>
</tr>
<tr>
<td>prokka_gff</td>
<td>File</td>
<td>Prokka output GFF3 file containing sequence and annotation (you can view this in IGV)</td>
</tr>
<tr>
<td>prokka_sqn</td>
<td>File</td>
<td>A Sequin file for GenBank submission</td>
</tr>
<tr>
<td>qc_check</td>
<td>String</td>
<td>A string that indicates whether or not the sample passes a set of pre-determined and user-provided QC thresholds</td>
</tr>
<tr>
<td>qc_standard</td>
<td>File</td>
<td>The file used in the QC Check task containing the QC thresholds.</td>
</tr>
<tr>
<td>quast_gc_percent</td>
<td>Float</td>
<td>The GC percent of your sample</td>
</tr>
<tr>
<td>quast_report</td>
<td>File</td>
<td>TSV report from QUAST</td>
</tr>
<tr>
<td>quast_version</td>
<td>String</td>
<td>The version of QUAST</td>
</tr>
<tr>
<td>resfinder_db_version</td>
<td>String</td>
<td>Version of ResFinder database</td>
</tr>
<tr>
<td>resfinder_docker</td>
<td>String</td>
<td>ResFinder docker image used</td>
</tr>
<tr>
<td>resfinder_pheno_table</td>
<td>File</td>
<td>Table containing al AMR phenotypes</td>
</tr>
<tr>
<td>resfinder_pheno_table_species</td>
<td>File</td>
<td>Table with species-specific AMR phenotypes</td>
</tr>
<tr>
<td>resfinder_pointfinder_pheno_table</td>
<td>File</td>
<td>TSV showing presence(1)/absence(0) of predicted resistance against an antibiotic class</td>
</tr>
<tr>
<td>resfinder_pointfinder_results</td>
<td>File</td>
<td>Predicted point mutations, grouped by the gene they occur in</td>
</tr>
<tr>
<td>resfinder_predicted_pheno_resistance</td>
<td>String</td>
<td>Semicolon delimited list of antimicrobial drugs and associated genes and/or point mutations.&nbsp;<drug1>: <gene1>, <gene1>, <point_mutation1>; <drug2>: <gene3>, <gene4>;</gene4></gene3></drug2></point_mutation1></gene1></gene1></drug1></td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Amp</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ampicillin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Axo</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ceftriaxone based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Azm</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Azithromycin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Cip</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ciprofloxacin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Smx</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Sulfamethoxazole based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Tmp</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Trimothoprim based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_xdr_shigella</td>
<td>String</td>
<td>Final prediction of XDR Shigella status based on CDC definition. Explanation can be found in the description above this table.</td>
</tr>
<tr>
<td>resfinder_results</td>
<td>File</td>
<td>Predicted resistance genes grouped by antibiotic class</td>
</tr>
<tr>
<td>resfinder_seqs</td>
<td>File</td>
<td>FASTA of resistance gene sequences from users input sequence</td>
</tr>
<tr>
<td>seq_platform</td>
<td>String</td>
<td>Description of the sequencing methodology used to generate the input read data</td>
</tr>
<tr>
<td>seqsero2s_note</td>
<td>String</td>
<td>Additional notes produced by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_predicted_antigenic_profile</td>
<td>String</td>
<td>Antigenic profile predicted for Salmonella spp. by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_report</td>
<td>String</td>
<td>TSV report produced by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_version</td>
<td>String</td>
<td>Version of SeqSero2S used</td>
</tr>
<tr>
<td>serotypefinder_docker</td>
<td>String</td>
<td>SerotypeFinder docker image used</td>
</tr>
<tr>
<td>serotypefinder_report</td>
<td>File</td>
<td>TSV report produced by SerotypeFinder</td>
</tr>
<tr>
<td>serotypefinder_serotype</td>
<td>String</td>
<td>Serotype predicted by SerotypeFinder</td>
</tr>
<tr>
<td>shigeifinder_H_antigen</td>
<td>String</td>
<td>H-antigen gene identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_O_antigen</td>
<td>String</td>
<td>O-antigen gene identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_cluster</td>
<td>String</td>
<td>Shigella/EIEC cluster identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_docker</td>
<td>String</td>
<td>ShigEiFinder docker image used</td>
</tr>
<tr>
<td>shigeifinder_ipaH_presence_absence</td>
<td>String</td>
<td>Presence (+) or absence (-) of ipaH identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_notes</td>
<td>String</td>
<td>Any notes output from ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_num_virulence_plasmid_genes</td>
<td>String</td>
<td>Number of virulence plasmid genes identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_report</td>
<td>File</td>
<td>TSV report from ShigEiFinder (see <a href="https://github.com/LanLab/ShigEiFinder#shigeifinder">https://github.com/LanLab/ShigEiFinder#shigeifinder</a>)</td>
</tr>
<tr>
<td>shigeifinder_serotype</td>
<td>String</td>
<td>Serotype predicted by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_version</td>
<td>String</td>
<td>ShigEiFinder version used</td>
</tr>
<tr>
<td>sistr_allele_fasta</td>
<td>File</td>
<td>FASTA file of novel cgMLST alleles from SISTR</td>
</tr>
<tr>
<td>sistr_allele_json</td>
<td>File</td>
<td>JSON file of cgMLST allele sequences and information (see <a href="https://github.com/phac-nml/sistr_cmd#cgmlst-allele-search-results">https://github.com/phac-nml/sistr_cmd#cgmlst-allele-search-results</a>)</td>
</tr>
<tr>
<td>sistr_antigenic_formula</td>
<td>String</td>
<td>A field that aggregates the O, H1, and H2, antigen values in a single location for convenience</td>
</tr>
<tr>
<td>sistr_cgmlst</td>
<td>File</td>
<td>CSV file of the cgMLST allelic profile from SISTR (see <a href="https://github.com/phac-nml/sistr_cmd#cgmlst-allelic-profiles-output---cgmlst-profiles-cgmlst-profilescsv">https://github.com/phac-nml/sistr_cmd#cgmlst-allelic-profiles-output---cgmlst-profiles-cgmlst-profilescsv</a>)</td>
</tr>
<tr>
<td>sistr_h1_antigens</td>
<td>String</td>
<td>The predicted H1 antigen</td>
</tr>
<tr>
<td>sistr_h2_antigens</td>
<td>String</td>
<td>The predicted H2 antigen</td>
</tr>
<tr>
<td>sistr_o_antigens</td>
<td>String</td>
<td>The predicted O antigen</td>
</tr>
<tr>
<td>sistr_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by SISTR</td>
</tr>
<tr>
<td>sistr_results</td>
<td>File</td>
<td>TSV results file produced by SISTR (see <a href="https://github.com/phac-nml/sistr_cmd#primary-results-output--o-sistr-results">https://github.com/phac-nml/sistr_cmd#primary-results-output--o-sistr-results</a>)</td>
</tr>
<tr>
<td>sistr_serogroup</td>
<td>String</td>
<td>Serogroup predicted by SISTR</td>
</tr>
<tr>
<td>sistr_serotype_cgmlst</td>
<td>String</td>
<td>cgMLST of the serogroup prediicted by SISTR</td>
</tr>
<tr>
<td>sistr_version</td>
<td>String</td>
<td>Version of SISTR used</td>
</tr>
<tr>
<td>spatyper_docker</td>
<td>String</td>
<td>spatyper docker image used</td>
</tr>
<tr>
<td>spatyper_repeats</td>
<td>String</td>
<td>order of identified repeats</td>
</tr>
<tr>
<td>spatyper_tsv</td>
<td>File</td>
<td>TSV report with spatyper results</td>
</tr>
<tr>
<td>spatyper_type</td>
<td>String</td>
<td>spa type</td>
</tr>
<tr>
<td>spatyper_version</td>
<td>String</td>
<td>spatyper version used</td>
</tr>
<tr>
<td>staphopiasccmec_docker</td>
<td>String</td>
<td>staphopia-sccmec docker image used</td>
</tr>
<tr>
<td>staphopiasccmec_hamming_distance_tsv</td>
<td>File</td>
<td>staphopia-sccmec version</td>
</tr>
<tr>
<td>staphopiasccmec_results_tsv</td>
<td>File</td>
<td>sccmec types and mecA presence</td>
</tr>
<tr>
<td>staphopiasccmec_types_and_mecA_presence</td>
<td>String</td>
<td>staphopia-sccmec Hamming distance file</td>
</tr>
<tr>
<td>staphopiasccmec_version</td>
<td>String</td>
<td>staphopia-sccmec presence and absence TSV file</td>
</tr>
<tr>
<td>stxtyper_all_hits</td>
<td>String</td>
<td>Comma-separated list of matches of all types. Includes complete, partial, frameshift, internal stop, and novel hits. List is de-duplicated so multiple identical hits are only listed once. For example if 5 partial stx2 hits are detected in the genome, only 1 "stx2" will be listed in this field. To view the potential subtype for each partial hit, the user will need to view the stxtyper_report TSV file.</td>
</tr>
<tr>
<td>stxtyper_ambiguous_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "AMBIGUOUS". Ambiguous bases found in the query sequence (e.g., N)</td>
</tr>
<tr>
<td>stxtyper_complete_operons</td>
<td>String</td>
<td>Comma-separated list of all COMPLETE operons detected by StxTyper. Show multiple hits if present in results.</td>
</tr>
<tr>
<td>stxtyper_docker</td>
<td>String</td>
<td>Name of docker image used by the stxtyper task.</td>
</tr>
<tr>
<td>stxtyper_extended_operons</td>
<td>String</td>
<td>Comma-separated list of all EXTENDED operons detected by StxTyper if coding sequence extends beyond the reference stop codon for one or both of the reference proteins.</td>
</tr>
<tr>
<td>stxtyper_novel_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "COMPLETE_NOVEL". Possible outputs "stx1", "stx2", or "stx1,stx2"</td>
</tr>
<tr>
<td>stxtyper_num_hits</td>
<td>Int</td>
<td>Number of "hits" or rows present in the <code>stxtyper_report</code> TSV file</td>
</tr>
<tr>
<td>stxtyper_partial_hits</td>
<td>String</td>
<td>Possible outputs "stx1", "stx2", or "stx1,stx2". Tells the user that there was a partial hit to either the A or B subunit, but does not describe which subunit, only the possible types from the PARTIAL matches.</td>
</tr>
<tr>
<td>stxtyper_report</td>
<td>File</td>
<td>Raw results TSV file produced by StxTyper</td>
</tr>
<tr>
<td>stxtyper_stx_frameshifts_or_internal_stop_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "FRAMESHIFT" or "INTERNAL_STOP". Possible outputs "stx1", "stx2", or "stx1,stx2"</td>
</tr>
<tr>
<td>stxtyper_version</td>
<td>String</td>
<td>Version of StxTyper used</td>
</tr>
<tr>
<td>taxon_table_status</td>
<td>String</td>
<td>Status of the taxon table upload</td>
</tr>
<tr>
<td>theiaprok_fasta_analysis_date</td>
<td>String</td>
<td>Date of TheiaProk FASTA workflow execution</td>
</tr>
<tr>
<td>theiaprok_fasta_version</td>
<td>String</td>
<td>Version of TheiaProk FASTA workflow execution</td>
</tr>
<tr>
<td>ts_mlst_allelic_profile</td>
<td>String</td>
<td>Profile of MLST loci and allele numbers predicted by MLST</td>
</tr>
<tr>
<td>ts_mlst_docker</td>
<td>String</td>
<td>Docker image used for MLST</td>
</tr>
<tr>
<td>ts_mlst_novel_alleles</td>
<td>File</td>
<td>FASTA file containing nucleotide sequence of any alleles that are not in the MLST database used by TheiaProk</td>
</tr>
<tr>
<td>ts_mlst_predicted_secondary_st</td>
<td>String</td>
<td>ST predicted by secondary MLST run</td>
</tr>
<tr>
<td>ts_mlst_predicted_st</td>
<td>String</td>
<td>ST predicted by MLST</td>
</tr>
<tr>
<td>ts_mlst_pubmlst_scheme</td>
<td>String</td>
<td>PubMLST scheme used byMLST</td>
</tr>
<tr>
<td>ts_mlst_pubmlst_secondary_scheme</td>
<td>String</td>
<td>PubMLST secondary scheme used by MLST</td>
</tr>
<tr>
<td>ts_mlst_results</td>
<td>File</td>
<td>TSV report with detailed MLST profile, including <a href="https://github.com/tseemann/mlst#missing-data">https://github.com/tseemann/mlst#missing-data</a></td>
</tr>
<tr>
<td>ts_mlst_secondary_allelic_profile</td>
<td>String</td>
<td>Profile of alleles predicted by secondary MLST run</td>
</tr>
<tr>
<td>ts_mlst_secondary_novel_alleles</td>
<td>File</td>
<td>FASTA file containing nucleotide sequence of any alleles that are not in the MLST database used by TheiaProk, from the secondary scheme run</td>
</tr>
<tr>
<td>ts_mlst_version</td>
<td>String</td>
<td>Version of Torsten Seemans MLST tool used</td>
</tr>
<tr>
<td>virulencefinder_docker</td>
<td>String</td>
<td>VirulenceFinder docker image used</td>
</tr>
<tr>
<td>virulencefinder_hits</td>
<td>String</td>
<td>Virulence genes detected by VirulenceFinder</td>
</tr>
<tr>
<td>virulencefinder_report_tsv</td>
<td>File</td>
<td>Output TSV file created by VirulenceFinder</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>abricate_abaum_database</td>
<td>String</td>
<td>Database of reference A. baumannii plasmid typing genes used for plasmid typing</td>
</tr>
<tr>
<td>abricate_abaum_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_abaum_plasmid_tsv</td>
<td>File</td>
<td><a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a> containing a row for each A. baumannii plasmid type gene found in the sample</td>
</tr>
<tr>
<td>abricate_abaum_plasmid_type_genes</td>
<td>String</td>
<td>A. baumannii Plasmid typing genes found in the sample; from GENE column in <a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a></td>
</tr>
<tr>
<td>abricate_abaum_version</td>
<td>String</td>
<td>Version of abricate used for A. baumannii plasmid typing</td>
</tr>
<tr>
<td>abricate_database</td>
<td>String</td>
<td>Database of reference used with Abricate</td>
</tr>
<tr>
<td>abricate_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_genes</td>
<td>String</td>
<td>Genes found in the sample; from GENE column in <a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a></td>
</tr>
<tr>
<td>abricate_results_tsv</td>
<td>File</td>
<td><a href="https://github.com/tseemann/abricate#output">https://github.com/tseemann/abricate#output</a> containing a row for each gene found in the sample</td>
</tr>
<tr>
<td>abricate_version</td>
<td>String</td>
<td>Version of abricate used for A. baumannii plasmid typing</td>
</tr>
<tr>
<td>abricate_vibrio_biotype</td>
<td>String</td>
<td>Biotype classification according to tcpA gene sequence (Classical or ElTor)</td>
</tr>
<tr>
<td>abricate_vibrio_ctxA</td>
<td>String</td>
<td>Presence or absence of the ctxA gene</td>
</tr>
<tr>
<td>abricate_vibrio_database</td>
<td>String</td>
<td>Database to use while running abricate</td>
</tr>
<tr>
<td>abricate_vibrio_detailed_tsv</td>
<td>File</td>
<td>Detailed ABRicate output file</td>
</tr>
<tr>
<td>abricate_vibrio_docker</td>
<td>String</td>
<td>Docker file used for running abricate</td>
</tr>
<tr>
<td>abricate_vibrio_ompW</td>
<td>String</td>
<td>Presence or absence of the ompW gene</td>
</tr>
<tr>
<td>abricate_vibrio_serogroup</td>
<td>String</td>
<td>Serotype classification as O1 (wbeN gene), O139 (wbfR gene) or not detected.</td>
</tr>
<tr>
<td>abricate_vibrio_toxR</td>
<td>String</td>
<td>Presence or absence of the toxR gene</td>
</tr>
<tr>
<td>abricate_vibrio_version</td>
<td>String</td>
<td>The abricate version run</td>
</tr>
<tr>
<td>agrvate_agr_canonical</td>
<td>String</td>
<td>Canonical or non-canonical agrD</td>
</tr>
<tr>
<td>agrvate_agr_group</td>
<td>String</td>
<td>Agr group</td>
</tr>
<tr>
<td>agrvate_agr_match_score</td>
<td>String</td>
<td>Match score for agr group</td>
</tr>
<tr>
<td>agrvate_agr_multiple</td>
<td>String</td>
<td>If multiple agr groups were found</td>
</tr>
<tr>
<td>agrvate_agr_num_frameshifts</td>
<td>String</td>
<td>Number of frameshifts found in CDS of extracted agr operon</td>
</tr>
<tr>
<td>agrvate_docker</td>
<td>String</td>
<td>The docker used for AgrVATE</td>
</tr>
<tr>
<td>agrvate_results</td>
<td>File</td>
<td>A gzipped tarball of all results</td>
</tr>
<tr>
<td>agrvate_summary</td>
<td>File</td>
<td>The summary file produced</td>
</tr>
<tr>
<td>agrvate_version</td>
<td>String</td>
<td>The version of AgrVATE used</td>
</tr>
<tr>
<td>amr_search_csv</td>
<td>File</td>
<td>CSV formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_docker</td>
<td>String</td>
<td>Docker image used to run AMR_Search</td>
</tr>
<tr>
<td>amr_search_results</td>
<td>File</td>
<td>JSON formatted AMR profile including BLAST results</td>
</tr>
<tr>
<td>amr_search_results_pdf</td>
<td>File</td>
<td>PDF formatted AMR profile</td>
</tr>
<tr>
<td>amr_search_version</td>
<td>String</td>
<td>Version of AMR_Search libraries used</td>
</tr>
<tr>
<td>amrfinderplus_all_report</td>
<td>File</td>
<td>Output TSV file from AMRFinderPlus (described <a href="https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#fields">https://github.com/ncbi/amr/wiki/Running-AMRFinderPlus#fields</a>)</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_betalactam_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to beta-lactams</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_carbapenem_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to carbapenem</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_cephalosporin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to cephalosporin</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_cephalothin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to cephalothin</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_amr_betalactam_methicillin_genes</td>
<td>String</td>
<td>Beta-lactam AMR genes identified by AMRFinderPlus that are known to confer resistance to methicilin</td>
</tr>
<tr>
<td>amrfinderplus_amr_classes</td>
<td>String</td>
<td>AMRFinderPlus predictions for classes of drugs that genes found in the reads are known to confer resistance to</td>
</tr>
<tr>
<td>amrfinderplus_amr_core_genes</td>
<td>String</td>
<td>AMR genes identified by AMRFinderPlus where the scope is "core"</td>
</tr>
<tr>
<td>amrfinderplus_amr_plus_genes</td>
<td>String</td>
<td>AMR genes identified by AMRFinderPlus where the scope is "plus"</td>
</tr>
<tr>
<td>amrfinderplus_amr_report</td>
<td>File</td>
<td>TSV file detailing AMR genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>amrfinderplus_amr_subclasses</td>
<td>String</td>
<td>More specificity about the drugs that genes identified in the reads confer resistance to</td>
</tr>
<tr>
<td>amrfinderplus_db_version</td>
<td>String</td>
<td>AMRFinderPlus database version used</td>
</tr>
<tr>
<td>amrfinderplus_stress_genes</td>
<td>String</td>
<td>Stress genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_stress_report</td>
<td>File</td>
<td>TSV file detailing stress genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>amrfinderplus_version</td>
<td>String</td>
<td>AMRFinderPlus version used</td>
</tr>
<tr>
<td>amrfinderplus_virulence_genes</td>
<td>String</td>
<td>Virulence genes identified by AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_virulence_report</td>
<td>File</td>
<td>TSV file detailing virulence genes only, from the amrfinderplus_all_report</td>
</tr>
<tr>
<td>ani_highest_percent</td>
<td>Float</td>
<td>Highest ANI between query and any given reference genome (top species match)</td>
</tr>
<tr>
<td>ani_highest_percent_bases_aligned</td>
<td>Float</td>
<td>Percentage of bases aligned between query genome and top species match</td>
</tr>
<tr>
<td>ani_mummer_docker</td>
<td>String</td>
<td>Docker image used to run the ANI_mummer task</td>
</tr>
<tr>
<td>ani_mummer_version</td>
<td>String</td>
<td>Version of MUMmer used</td>
</tr>
<tr>
<td>ani_output_tsv</td>
<td>File</td>
<td>Full output TSV from ani-m</td>
</tr>
<tr>
<td>ani_top_species_match</td>
<td>String</td>
<td>Species of genome with highest ANI to query FASTA</td>
</tr>
<tr>
<td>arln_assembly_ratio</td>
<td>String</td>
<td>Assembly ratio of the final assembly compared to the mean assembly length of the taxon. These stats are gathered from a statistics file created by the CDC aggregating NCBI assembly data by taxon.</td>
</tr>
<tr>
<td>arln_assembly_zscore</td>
<td>String</td>
<td>ZScore of the samples assembly length to assess the uniformity of the assembly when compared to taxon means.</td>
</tr>
<tr>
<td>arln_stats_docker_version</td>
<td>String</td>
<td>Docker version for task_arln_stats.wdl</td>
</tr>
<tr>
<td>arln_taxon_assembly_ratio_stdev</td>
<td>String</td>
<td>Reported assembely length standard deviation of the given taxonomy pulled from NCBI aggregated data.</td>
</tr>
<tr>
<td>arln_taxon_gc_mean</td>
<td>String</td>
<td>Mean GC percent of the taxon.</td>
</tr>
<tr>
<td>arln_taxon_gc_percent_stdev</td>
<td>String</td>
<td>Reported GC percent standard deviation of the given taxonomy pulled from NCBI aggregated data.</td>
</tr>
<tr>
<td>assembly_length</td>
<td>Int</td>
<td>Length of assembly (total contig length) as determined by QUAST</td>
</tr>
<tr>
<td>bakta_gbff</td>
<td>File</td>
<td>Genomic GenBank format annotation file</td>
</tr>
<tr>
<td>bakta_gff3</td>
<td>File</td>
<td>Generic Feature Format Version 3 file</td>
</tr>
<tr>
<td>bakta_plot</td>
<td>File</td>
<td>Bakta plot output PNG file summarizing annotated genome features such as coding sequences, RNA genes, and hypothetical proteins.</td>
</tr>
<tr>
<td>bakta_summary</td>
<td>File</td>
<td>Bakta summary output TXT file</td>
</tr>
<tr>
<td>bakta_tsv</td>
<td>File</td>
<td>Annotations as simple human readable TSV</td>
</tr>
<tr>
<td>bakta_version</td>
<td>String</td>
<td>Bakta version used</td>
</tr>
<tr>
<td>busco_database</td>
<td>String</td>
<td>BUSCO database used</td>
</tr>
<tr>
<td>busco_docker</td>
<td>String</td>
<td>BUSCO docker image used</td>
</tr>
<tr>
<td>busco_report</td>
<td>File</td>
<td>A plain text summary of the results in BUSCO notation</td>
</tr>
<tr>
<td>busco_results</td>
<td>String</td>
<td>BUSCO results (see relevant toggle in <a href="https://theiagen.github.io/public_health_bioinformatics/latest/workflows/genomic_characterization/theiaprok/#post-assembly-tasks-performed-for-all-taxa">this block</a>)</td>
</tr>
<tr>
<td>busco_version</td>
<td>String</td>
<td>BUSCO software version used</td>
</tr>
<tr>
<td>ectyper_database_version</td>
<td>String</td>
<td>Version of the ECTyper database used</td>
</tr>
<tr>
<td>ectyper_docker</td>
<td>String</td>
<td>Docker image used to run ECTyper</td>
</tr>
<tr>
<td>ectyper_pathodb_version</td>
<td>String</td>
<td>Version of the ECTyper pathotype database used</td>
</tr>
<tr>
<td>ectyper_pathotype</td>
<td>String</td>
<td>Samples pathotype predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_pathotype_count</td>
<td>String</td>
<td>Number of pathotypes predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_pathotype_genes</td>
<td>String</td>
<td>Pathotype genes observed by ECTyper</td>
</tr>
<tr>
<td>ectyper_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_qc_result</td>
<td>String</td>
<td>QC pass or fail for the sample analyzed</td>
</tr>
<tr>
<td>ectyper_results</td>
<td>File</td>
<td>TSV file of evidence for ECTyper predicted serotype (see <a href="https://github.com/phac-nml/ecoli_serotyping#report-format">https://github.com/phac-nml/ecoli_serotyping#report-format</a>)</td>
</tr>
<tr>
<td>ectyper_stx_subtypes</td>
<td>String</td>
<td>Shiga toxin subtypes predicted by ECTyper</td>
</tr>
<tr>
<td>ectyper_version</td>
<td>String</td>
<td>Version of ECTyper used</td>
</tr>
<tr>
<td>ectyper_warnings</td>
<td>File</td>
<td>ECTyper warning messages</td>
</tr>
<tr>
<td>emmtyper_docker</td>
<td>String</td>
<td>Docker image for emmtyper</td>
</tr>
<tr>
<td>emmtyper_emm_type</td>
<td>String</td>
<td>The emm-type of a Streptococcus pyogenes assembly</td>
</tr>
<tr>
<td>emmtyper_results_tsv</td>
<td>File</td>
<td>TSV file with emmtyper results</td>
</tr>
<tr>
<td>emmtyper_version</td>
<td>String</td>
<td>Version of emmtyper used</td>
</tr>
<tr>
<td>gambit_closest_genomes</td>
<td>File</td>
<td>CSV file listing genomes in the GAMBIT database that are most similar to the query assembly</td>
</tr>
<tr>
<td>gambit_db_version</td>
<td>String</td>
<td>Version of the GAMBIT database used</td>
</tr>
<tr>
<td>gambit_docker</td>
<td>String</td>
<td>GAMBIT Docker used</td>
</tr>
<tr>
<td>gambit_predicted_taxon</td>
<td>String</td>
<td>Taxon predicted by GAMBIT</td>
</tr>
<tr>
<td>gambit_predicted_taxon_rank</td>
<td>String</td>
<td>Taxon rank of GAMBIT taxon prediction</td>
</tr>
<tr>
<td>gambit_report</td>
<td>File</td>
<td>GAMBIT report in a machine-readable format</td>
</tr>
<tr>
<td>gambit_version</td>
<td>String</td>
<td>Version of GAMBIT software used</td>
</tr>
<tr>
<td>gamma_docker</td>
<td>String</td>
<td>The Docker container used</td>
</tr>
<tr>
<td>gamma_fasta</td>
<td>File</td>
<td>The FASTA file reporting gene matches if boolean was set to true</td>
</tr>
<tr>
<td>gamma_gff</td>
<td>File</td>
<td>The GFF file reporting gene matches if boolean was set to true</td>
</tr>
<tr>
<td>gamma_results</td>
<td>File</td>
<td>The .gamma file reporting gene matches</td>
</tr>
<tr>
<td>gamma_version</td>
<td>String</td>
<td>The version of GAMMA used</td>
</tr>
<tr>
<td>hicap_docker</td>
<td>String</td>
<td>Docker image used for hicap</td>
</tr>
<tr>
<td>hicap_genes</td>
<td>String</td>
<td>cap genes identified. genes on different contigs delimited by;. truncation shown by trailing *</td>
</tr>
<tr>
<td>hicap_results_tsv</td>
<td>File</td>
<td>TSV file of hicap output</td>
</tr>
<tr>
<td>hicap_serotype</td>
<td>String</td>
<td>hicap serotype</td>
</tr>
<tr>
<td>hicap_version</td>
<td>String</td>
<td>hicap version used</td>
</tr>
<tr>
<td>kaptive_k_locus</td>
<td>String</td>
<td>Best matching K locus identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_k_type</td>
<td>String</td>
<td>Best matching K type identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_kl_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kaptive_oc_locus</td>
<td>String</td>
<td>Best matching K locus identified by Kaptive</td>
</tr>
<tr>
<td>kaptive_ocl_confidence</td>
<td>String</td>
<td>Kaptives confidence in the OCL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kaptive_output_file_k</td>
<td>File</td>
<td>TSV <a href="https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom">https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom</a> the K locus from Kaptive</td>
</tr>
<tr>
<td>kaptive_output_file_oc</td>
<td>File</td>
<td>TSV <a href="https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom">https://github.com/katholt/Kaptive/wiki/How-to-run#output-filesfrom</a> the OC locus from Kaptive</td>
</tr>
<tr>
<td>kaptive_version</td>
<td>String</td>
<td>Version of Kaptive used</td>
</tr>
<tr>
<td>kleborate_docker</td>
<td>String</td>
<td>Kleborate docker image used</td>
</tr>
<tr>
<td>kleborate_genomic_resistance_mutations</td>
<td>String</td>
<td>Genomic resistance mutations identifies by Kleborate</td>
</tr>
<tr>
<td>kleborate_key_resistance_genes</td>
<td>String</td>
<td>Key resistance genes identified by Kleborate</td>
</tr>
<tr>
<td>kleborate_klocus</td>
<td>String</td>
<td>Best matching K locus identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_klocus_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kleborate_ktype</td>
<td>String</td>
<td>Best matching K type identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_mlst_sequence_type</td>
<td>String</td>
<td><a href="https://github.com/katholt/Kleborate/wiki/MLST#multi-locus-sequence-typing-mlst">https://github.com/katholt/Kleborate/wiki/MLST#multi-locus-sequence-typing-mlst</a> call by Kleborate</td>
</tr>
<tr>
<td>kleborate_olocus</td>
<td>String</td>
<td>Best matching OC locus identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_olocus_confidence</td>
<td>String</td>
<td>Kaptives confidence in the KL match (see <a href="https://github.com/katholt/Kaptive/wiki/Interpreting-the-results">https://github.com/katholt/Kaptive/wiki/Interpreting-the-results</a>)</td>
</tr>
<tr>
<td>kleborate_otype</td>
<td>String</td>
<td>Best matching OC type identified by  Kleborate via Kaptive</td>
</tr>
<tr>
<td>kleborate_output_file</td>
<td>File</td>
<td>See also <a href="https://github.com/katholt/Kleborate/wiki/Scores-and-counts">https://github.com/katholt/Kleborate/wiki/Scores-and-counts</a></td>
</tr>
<tr>
<td>kleborate_resistance_score</td>
<td>String</td>
<td>Resistance score as given by kleborate</td>
</tr>
<tr>
<td>kleborate_version</td>
<td>String</td>
<td>Version of Kleborate used</td>
</tr>
<tr>
<td>kleborate_virulence_score</td>
<td>String</td>
<td>Virulence score as given by kleborate</td>
</tr>
<tr>
<td>kmerfinder_database</td>
<td>String</td>
<td>Database used to run KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_docker</td>
<td>String</td>
<td>Docker image used to run KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_query_coverage</td>
<td>String</td>
<td>KmerFinders query coverage of the top hit result</td>
</tr>
<tr>
<td>kmerfinder_results_tsv</td>
<td>File</td>
<td>Output TSV file created by KmerFinder</td>
</tr>
<tr>
<td>kmerfinder_template_coverage</td>
<td>String</td>
<td>Percent of kmer coverage pertaining to the template, or reference selected</td>
</tr>
<tr>
<td>kmerfinder_top_hit</td>
<td>String</td>
<td>Top hit species of KmerFinder</td>
</tr>
<tr>
<td>legsta_predicted_sbt</td>
<td>String</td>
<td>Sequence based type predicted by Legsta</td>
</tr>
<tr>
<td>legsta_results</td>
<td>File</td>
<td>TSV file of legsta results (see <a href="https://github.com/tseemann/legsta#output">https://github.com/tseemann/legsta#output</a>)</td>
</tr>
<tr>
<td>legsta_version</td>
<td>String</td>
<td>Version of legsta used</td>
</tr>
<tr>
<td>lissero_results</td>
<td>File</td>
<td>TSV results file from LisSero (see <a href="https://github.com/MDU-PHL/LisSero#example-output">https://github.com/MDU-PHL/LisSero#example-output</a>)</td>
</tr>
<tr>
<td>lissero_serotype</td>
<td>String</td>
<td>Serotype predicted by LisSero</td>
</tr>
<tr>
<td>lissero_version</td>
<td>String</td>
<td>Version of LisSero used</td>
</tr>
<tr>
<td>meningotype_BAST</td>
<td>String</td>
<td>BAST type</td>
</tr>
<tr>
<td>meningotype_FetA</td>
<td>String</td>
<td>FetA type</td>
</tr>
<tr>
<td>meningotype_NHBA</td>
<td>String</td>
<td>NHBA type</td>
</tr>
<tr>
<td>meningotype_NadA</td>
<td>String</td>
<td>NBA type</td>
</tr>
<tr>
<td>meningotype_PorA</td>
<td>String</td>
<td>PorA type</td>
</tr>
<tr>
<td>meningotype_PorB</td>
<td>String</td>
<td>PorB type</td>
</tr>
<tr>
<td>meningotype_fHbp</td>
<td>String</td>
<td>fHbp type</td>
</tr>
<tr>
<td>meningotype_serogroup</td>
<td>String</td>
<td>Serogroup</td>
</tr>
<tr>
<td>meningotype_tsv</td>
<td>File</td>
<td>Full result file</td>
</tr>
<tr>
<td>meningotype_version</td>
<td>String</td>
<td>Version of meningotype used</td>
</tr>
<tr>
<td>n50_value</td>
<td>Int</td>
<td>N50 of assembly calculated by QUAST</td>
</tr>
<tr>
<td>ngmaster_ngmast_porB_allele</td>
<td>String</td>
<td>porB allele number</td>
</tr>
<tr>
<td>ngmaster_ngmast_sequence_type</td>
<td>String</td>
<td>NG-MAST sequence type</td>
</tr>
<tr>
<td>ngmaster_ngmast_tbpB_allele</td>
<td>String</td>
<td>tbpB allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_23S_allele</td>
<td>String</td>
<td>23S rRNA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_gyrA_allele</td>
<td>String</td>
<td>gyrA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_mtrR_allele</td>
<td>String</td>
<td>mtrR allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_parC_allele</td>
<td>String</td>
<td>parC allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_penA_allele</td>
<td>String</td>
<td>penA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_ponA_allele</td>
<td>String</td>
<td>ponA allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_porB_allele</td>
<td>String</td>
<td>porB allele number</td>
</tr>
<tr>
<td>ngmaster_ngstar_sequence_type</td>
<td>String</td>
<td>NG-STAR sequence type</td>
</tr>
<tr>
<td>ngmaster_tsv</td>
<td>File</td>
<td>TSV file with NG-MAST/NG-STAR typing</td>
</tr>
<tr>
<td>ngmaster_version</td>
<td>String</td>
<td>ngmaster version</td>
</tr>
<tr>
<td>number_contigs</td>
<td>Int</td>
<td>Total number of contigs in assembly</td>
</tr>
<tr>
<td>pasty_all_serogroups</td>
<td>File</td>
<td>TSV file with details of each serogroup from pasty (see <a href="https://github.com/rpetit3/pasty#example-prefixdetailstsv">https://github.com/rpetit3/pasty#example-prefixdetailstsv</a>)</td>
</tr>
<tr>
<td>pasty_blast_hits</td>
<td>File</td>
<td>TSV file of BLAST hits from pasty (see <a href="https://github.com/rpetit3/pasty#example-prefixblastntsv">https://github.com/rpetit3/pasty#example-prefixblastntsv</a>)</td>
</tr>
<tr>
<td>pasty_comment</td>
<td>String</td>
<td></td>
</tr>
<tr>
<td>pasty_docker</td>
<td>String</td>
<td>pasty docker image used</td>
</tr>
<tr>
<td>pasty_serogroup</td>
<td>String</td>
<td>Serogroup predicted by pasty</td>
</tr>
<tr>
<td>pasty_serogroup_coverage</td>
<td>Float</td>
<td>The breadth of coverage of the O-antigen by pasty</td>
</tr>
<tr>
<td>pasty_serogroup_fragments</td>
<td>Int</td>
<td>Number of BLAST hits included in the prediction (fewer is better)</td>
</tr>
<tr>
<td>pasty_summary_tsv</td>
<td>File</td>
<td>TSV summary file of pasty outputs (see <a href="https://github.com/rpetit3/pasty#example-prefixtsv">https://github.com/rpetit3/pasty#example-prefixtsv</a>)</td>
</tr>
<tr>
<td>pasty_version</td>
<td>String</td>
<td>Version of pasty used</td>
</tr>
<tr>
<td>pbptyper_docker</td>
<td>String</td>
<td>pbptyper docker image used</td>
</tr>
<tr>
<td>pbptyper_pbptype_predicted_tsv</td>
<td>File</td>
<td>TSV file of pbptyper results (see <a href="https://github.com/rpetit3/pbptyper#example-prefixtsv">https://github.com/rpetit3/pbptyper#example-prefixtsv</a>)</td>
</tr>
<tr>
<td>pbptyper_predicted_1A_2B_2X</td>
<td>String</td>
<td>PBP type predicted by pbptyper</td>
</tr>
<tr>
<td>pbptyper_version</td>
<td>String</td>
<td>Version of pbptyper used</td>
</tr>
<tr>
<td>plasmidfinder_db_version</td>
<td>String</td>
<td>Version of PlasmidFnder used</td>
</tr>
<tr>
<td>plasmidfinder_docker</td>
<td>String</td>
<td>PlasmidFinder docker image used</td>
</tr>
<tr>
<td>plasmidfinder_plasmids</td>
<td>String</td>
<td>Names of plasmids identified by PlasmidFinder</td>
</tr>
<tr>
<td>plasmidfinder_results</td>
<td>File</td>
<td>Output file from PlasmidFinder in TSV format</td>
</tr>
<tr>
<td>plasmidfinder_seqs</td>
<td>File</td>
<td>Hit_in_genome_seq.fsa file produced by PlasmidFinder</td>
</tr>
<tr>
<td>poppunk_GPS_db_version</td>
<td>String</td>
<td>Version of GPSC database used</td>
</tr>
<tr>
<td>poppunk_docker</td>
<td>String</td>
<td>PopPUNK docker image with GPSC database used</td>
</tr>
<tr>
<td>poppunk_gps_cluster</td>
<td>String</td>
<td>GPS cluster predicted by PopPUNK</td>
</tr>
<tr>
<td>poppunk_gps_external_cluster_csv</td>
<td>File</td>
<td>GPSC v6 scheme designations</td>
</tr>
<tr>
<td>poppunk_version</td>
<td>String</td>
<td>Version of PopPUNK used</td>
</tr>
<tr>
<td>prokka_gbk</td>
<td>File</td>
<td>GenBank file produced from Prokka annotation of input FASTA</td>
</tr>
<tr>
<td>prokka_gff</td>
<td>File</td>
<td>Prokka output GFF3 file containing sequence and annotation (you can view this in IGV)</td>
</tr>
<tr>
<td>prokka_sqn</td>
<td>File</td>
<td>A Sequin file for GenBank submission</td>
</tr>
<tr>
<td>qc_check</td>
<td>String</td>
<td>A string that indicates whether or not the sample passes a set of pre-determined and user-provided QC thresholds</td>
</tr>
<tr>
<td>qc_standard</td>
<td>File</td>
<td>The file used in the QC Check task containing the QC thresholds.</td>
</tr>
<tr>
<td>quast_gc_percent</td>
<td>Float</td>
<td>The GC percent of your sample</td>
</tr>
<tr>
<td>quast_report</td>
<td>File</td>
<td>TSV report from QUAST</td>
</tr>
<tr>
<td>quast_version</td>
<td>String</td>
<td>The version of QUAST</td>
</tr>
<tr>
<td>resfinder_db_version</td>
<td>String</td>
<td>Version of ResFinder database</td>
</tr>
<tr>
<td>resfinder_docker</td>
<td>String</td>
<td>ResFinder docker image used</td>
</tr>
<tr>
<td>resfinder_pheno_table</td>
<td>File</td>
<td>Table containing al AMR phenotypes</td>
</tr>
<tr>
<td>resfinder_pheno_table_species</td>
<td>File</td>
<td>Table with species-specific AMR phenotypes</td>
</tr>
<tr>
<td>resfinder_pointfinder_pheno_table</td>
<td>File</td>
<td>TSV showing presence(1)/absence(0) of predicted resistance against an antibiotic class</td>
</tr>
<tr>
<td>resfinder_pointfinder_results</td>
<td>File</td>
<td>Predicted point mutations, grouped by the gene they occur in</td>
</tr>
<tr>
<td>resfinder_predicted_pheno_resistance</td>
<td>String</td>
<td>Semicolon delimited list of antimicrobial drugs and associated genes and/or point mutations.&nbsp;<drug1>: <gene1>, <gene1>, <point_mutation1>; <drug2>: <gene3>, <gene4>;</gene4></gene3></drug2></point_mutation1></gene1></gene1></drug1></td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Amp</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ampicillin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Axo</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ceftriaxone based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Azm</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Azithromycin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Cip</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Ciprofloxacin based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Smx</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Sulfamethoxazole based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_resistance_Tmp</td>
<td>String</td>
<td>States either&nbsp;Resistance&nbsp;or&nbsp;No Resistance predicted&nbsp;to Trimothoprim based on resfinder phenotypic predictions</td>
</tr>
<tr>
<td>resfinder_predicted_xdr_shigella</td>
<td>String</td>
<td>Final prediction of XDR Shigella status based on CDC definition. Explanation can be found in the description above this table.</td>
</tr>
<tr>
<td>resfinder_results</td>
<td>File</td>
<td>Predicted resistance genes grouped by antibiotic class</td>
</tr>
<tr>
<td>resfinder_seqs</td>
<td>File</td>
<td>FASTA of resistance gene sequences from users input sequence</td>
</tr>
<tr>
<td>seq_platform</td>
<td>String</td>
<td>Description of the sequencing methodology used to generate the input read data</td>
</tr>
<tr>
<td>seqsero2s_note</td>
<td>String</td>
<td>Additional notes produced by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_predicted_antigenic_profile</td>
<td>String</td>
<td>Antigenic profile predicted for Salmonella spp. by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_report</td>
<td>String</td>
<td>TSV report produced by SeqSero2S</td>
</tr>
<tr>
<td>seqsero2s_version</td>
<td>String</td>
<td>Version of SeqSero2S used</td>
</tr>
<tr>
<td>serotypefinder_docker</td>
<td>String</td>
<td>SerotypeFinder docker image used</td>
</tr>
<tr>
<td>serotypefinder_report</td>
<td>File</td>
<td>TSV report produced by SerotypeFinder</td>
</tr>
<tr>
<td>serotypefinder_serotype</td>
<td>String</td>
<td>Serotype predicted by SerotypeFinder</td>
</tr>
<tr>
<td>shigeifinder_H_antigen</td>
<td>String</td>
<td>H-antigen gene identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_O_antigen</td>
<td>String</td>
<td>O-antigen gene identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_cluster</td>
<td>String</td>
<td>Shigella/EIEC cluster identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_docker</td>
<td>String</td>
<td>ShigEiFinder docker image used</td>
</tr>
<tr>
<td>shigeifinder_ipaH_presence_absence</td>
<td>String</td>
<td>Presence (+) or absence (-) of ipaH identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_notes</td>
<td>String</td>
<td>Any notes output from ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_num_virulence_plasmid_genes</td>
<td>String</td>
<td>Number of virulence plasmid genes identified by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_report</td>
<td>File</td>
<td>TSV report from ShigEiFinder (see <a href="https://github.com/LanLab/ShigEiFinder#shigeifinder">https://github.com/LanLab/ShigEiFinder#shigeifinder</a>)</td>
</tr>
<tr>
<td>shigeifinder_serotype</td>
<td>String</td>
<td>Serotype predicted by ShigEiFinder</td>
</tr>
<tr>
<td>shigeifinder_version</td>
<td>String</td>
<td>ShigEiFinder version used</td>
</tr>
<tr>
<td>sistr_allele_fasta</td>
<td>File</td>
<td>FASTA file of novel cgMLST alleles from SISTR</td>
</tr>
<tr>
<td>sistr_allele_json</td>
<td>File</td>
<td>JSON file of cgMLST allele sequences and information (see <a href="https://github.com/phac-nml/sistr_cmd#cgmlst-allele-search-results">https://github.com/phac-nml/sistr_cmd#cgmlst-allele-search-results</a>)</td>
</tr>
<tr>
<td>sistr_antigenic_formula</td>
<td>String</td>
<td>A field that aggregates the O, H1, and H2, antigen values in a single location for convenience</td>
</tr>
<tr>
<td>sistr_cgmlst</td>
<td>File</td>
<td>CSV file of the cgMLST allelic profile from SISTR (see <a href="https://github.com/phac-nml/sistr_cmd#cgmlst-allelic-profiles-output---cgmlst-profiles-cgmlst-profilescsv">https://github.com/phac-nml/sistr_cmd#cgmlst-allelic-profiles-output---cgmlst-profiles-cgmlst-profilescsv</a>)</td>
</tr>
<tr>
<td>sistr_h1_antigens</td>
<td>String</td>
<td>The predicted H1 antigen</td>
</tr>
<tr>
<td>sistr_h2_antigens</td>
<td>String</td>
<td>The predicted H2 antigen</td>
</tr>
<tr>
<td>sistr_o_antigens</td>
<td>String</td>
<td>The predicted O antigen</td>
</tr>
<tr>
<td>sistr_predicted_serotype</td>
<td>String</td>
<td>Serotype predicted by SISTR</td>
</tr>
<tr>
<td>sistr_results</td>
<td>File</td>
<td>TSV results file produced by SISTR (see <a href="https://github.com/phac-nml/sistr_cmd#primary-results-output--o-sistr-results">https://github.com/phac-nml/sistr_cmd#primary-results-output--o-sistr-results</a>)</td>
</tr>
<tr>
<td>sistr_serogroup</td>
<td>String</td>
<td>Serogroup predicted by SISTR</td>
</tr>
<tr>
<td>sistr_serotype_cgmlst</td>
<td>String</td>
<td>cgMLST of the serogroup prediicted by SISTR</td>
</tr>
<tr>
<td>sistr_version</td>
<td>String</td>
<td>Version of SISTR used</td>
</tr>
<tr>
<td>spatyper_docker</td>
<td>String</td>
<td>spatyper docker image used</td>
</tr>
<tr>
<td>spatyper_repeats</td>
<td>String</td>
<td>order of identified repeats</td>
</tr>
<tr>
<td>spatyper_tsv</td>
<td>File</td>
<td>TSV report with spatyper results</td>
</tr>
<tr>
<td>spatyper_type</td>
<td>String</td>
<td>spa type</td>
</tr>
<tr>
<td>spatyper_version</td>
<td>String</td>
<td>spatyper version used</td>
</tr>
<tr>
<td>staphopiasccmec_docker</td>
<td>String</td>
<td>staphopia-sccmec docker image used</td>
</tr>
<tr>
<td>staphopiasccmec_hamming_distance_tsv</td>
<td>File</td>
<td>staphopia-sccmec version</td>
</tr>
<tr>
<td>staphopiasccmec_results_tsv</td>
<td>File</td>
<td>sccmec types and mecA presence</td>
</tr>
<tr>
<td>staphopiasccmec_types_and_mecA_presence</td>
<td>String</td>
<td>staphopia-sccmec Hamming distance file</td>
</tr>
<tr>
<td>staphopiasccmec_version</td>
<td>String</td>
<td>staphopia-sccmec presence and absence TSV file</td>
</tr>
<tr>
<td>stxtyper_all_hits</td>
<td>String</td>
<td>Comma-separated list of matches of all types. Includes complete, partial, frameshift, internal stop, and novel hits. List is de-duplicated so multiple identical hits are only listed once. For example if 5 partial stx2 hits are detected in the genome, only 1 "stx2" will be listed in this field. To view the potential subtype for each partial hit, the user will need to view the stxtyper_report TSV file.</td>
</tr>
<tr>
<td>stxtyper_ambiguous_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "AMBIGUOUS". Ambiguous bases found in the query sequence (e.g., N)</td>
</tr>
<tr>
<td>stxtyper_complete_operons</td>
<td>String</td>
<td>Comma-separated list of all COMPLETE operons detected by StxTyper. Show multiple hits if present in results.</td>
</tr>
<tr>
<td>stxtyper_docker</td>
<td>String</td>
<td>Name of docker image used by the stxtyper task.</td>
</tr>
<tr>
<td>stxtyper_extended_operons</td>
<td>String</td>
<td>Comma-separated list of all EXTENDED operons detected by StxTyper if coding sequence extends beyond the reference stop codon for one or both of the reference proteins.</td>
</tr>
<tr>
<td>stxtyper_novel_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "COMPLETE_NOVEL". Possible outputs "stx1", "stx2", or "stx1,stx2"</td>
</tr>
<tr>
<td>stxtyper_num_hits</td>
<td>Int</td>
<td>Number of "hits" or rows present in the <code>stxtyper_report</code> TSV file</td>
</tr>
<tr>
<td>stxtyper_partial_hits</td>
<td>String</td>
<td>Possible outputs "stx1", "stx2", or "stx1,stx2". Tells the user that there was a partial hit to either the A or B subunit, but does not describe which subunit, only the possible types from the PARTIAL matches.</td>
</tr>
<tr>
<td>stxtyper_report</td>
<td>File</td>
<td>Raw results TSV file produced by StxTyper</td>
</tr>
<tr>
<td>stxtyper_stx_frameshifts_or_internal_stop_hits</td>
<td>String</td>
<td>Comma-separated list of matches that have the OPERON output of "FRAMESHIFT" or "INTERNAL_STOP". Possible outputs "stx1", "stx2", or "stx1,stx2"</td>
</tr>
<tr>
<td>stxtyper_version</td>
<td>String</td>
<td>Version of StxTyper used</td>
</tr>
<tr>
<td>taxon_table_status</td>
<td>String</td>
<td>Status of the taxon table upload</td>
</tr>
<tr>
<td>theiaprok_fasta_analysis_date</td>
<td>String</td>
<td>Date of TheiaProk FASTA workflow execution</td>
</tr>
<tr>
<td>theiaprok_fasta_version</td>
<td>String</td>
<td>Version of TheiaProk FASTA workflow execution</td>
</tr>
<tr>
<td>ts_mlst_allelic_profile</td>
<td>String</td>
<td>Profile of MLST loci and allele numbers predicted by MLST</td>
</tr>
<tr>
<td>ts_mlst_docker</td>
<td>String</td>
<td>Docker image used for MLST</td>
</tr>
<tr>
<td>ts_mlst_novel_alleles</td>
<td>File</td>
<td>FASTA file containing nucleotide sequence of any alleles that are not in the MLST database used by TheiaProk</td>
</tr>
<tr>
<td>ts_mlst_predicted_secondary_st</td>
<td>String</td>
<td>ST predicted by secondary MLST run</td>
</tr>
<tr>
<td>ts_mlst_predicted_st</td>
<td>String</td>
<td>ST predicted by MLST</td>
</tr>
<tr>
<td>ts_mlst_pubmlst_scheme</td>
<td>String</td>
<td>PubMLST scheme used byMLST</td>
</tr>
<tr>
<td>ts_mlst_pubmlst_secondary_scheme</td>
<td>String</td>
<td>PubMLST secondary scheme used by MLST</td>
</tr>
<tr>
<td>ts_mlst_results</td>
<td>File</td>
<td>TSV report with detailed MLST profile, including <a href="https://github.com/tseemann/mlst#missing-data">https://github.com/tseemann/mlst#missing-data</a></td>
</tr>
<tr>
<td>ts_mlst_secondary_allelic_profile</td>
<td>String</td>
<td>Profile of alleles predicted by secondary MLST run</td>
</tr>
<tr>
<td>ts_mlst_secondary_novel_alleles</td>
<td>File</td>
<td>FASTA file containing nucleotide sequence of any alleles that are not in the MLST database used by TheiaProk, from the secondary scheme run</td>
</tr>
<tr>
<td>ts_mlst_version</td>
<td>String</td>
<td>Version of Torsten Seemans MLST tool used</td>
</tr>
<tr>
<td>virulencefinder_docker</td>
<td>String</td>
<td>VirulenceFinder docker image used</td>
</tr>
<tr>
<td>virulencefinder_hits</td>
<td>String</td>
<td>Virulence genes detected by VirulenceFinder</td>
</tr>
<tr>
<td>virulencefinder_report_tsv</td>
<td>File</td>
<td>Output TSV file created by VirulenceFinder</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="November 12, 2025 20:00:11 UTC">2025-11-12</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="September 11, 2024 17:48:55 UTC">2024-09-11</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
       2022-2025 <a href="https://www.theiagen.com" target="_blank" rel="noopener">Theiagen Genomics</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/theiagen" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://bsky.app/profile/theiagen.bsky.social" target="_blank" rel="noopener" title="bsky.app" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M407.8 294.7c-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3M288 227.1c-26.1-50.7-97.1-145.2-163.1-191.8C61.6-9.4 37.5-1.7 21.6 5.5 3.3 13.8 0 41.9 0 58.4S9.1 194 15 213.9c19.5 65.7 89.1 87.9 153.2 80.7 3.3-.5 6.6-.9 10-1.4-3.3.5-6.6 1-10 1.4-93.9 14-177.3 48.2-67.9 169.9C220.6 589.1 265.1 437.8 288 361.1c22.9 76.7 49.2 222.5 185.6 103.4 102.4-103.4 28.1-156-65.8-169.9-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3 64.1 7.1 133.6-15.1 153.2-80.7C566.9 194 576 75 576 58.4s-3.3-44.7-21.6-52.9c-15.8-7.1-40-14.9-103.2 29.8C385.1 81.9 314.1 176.4 288 227.1"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.linkedin.com/company/theiagen" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3M135.4 416H69V202.2h66.5V416zM102.2 96a38.5 38.5 0 1 1 0 77 38.5 38.5 0 1 1 0-77m282.1 320h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9z"></path></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "../../..", "features": ["content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "navigation.instant", "navigation.path", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.suggest", "toc.follow"], "search": "../../../assets/javascripts/workers/search.7a47a382.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"alias": true, "default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.e71a0d61.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../../../javascripts/tablesort.js"></script>
      
        <script src="../../../javascripts/table-search.js"></script>
      
    
  
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>