<!DOCTYPE html><html lang="en" class="no-js"><head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A collection of bioinformatics workflows for genomic characterization, submission preparation, and genomic epidemiology of pathogens of public health concern.">
      
      
        <meta name="author" content="Theiagen Genomics">
      
      
        <link rel="canonical" href="https://theiagen.github.io/public_health_bioinformatics/v4.1.0/workflows/phylogenetic_construction/augur/">
      
      
        <link rel="prev" href="../../data_import/assembly_fetch/">
      
      
        <link rel="next" href="../../data_import/basespace_fetch/">
      
      
        
      
      
      <link rel="icon" href="../../../assets/logos/Theiagen-Symbol-Standard-01.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.1">
    
    
      
        <title>Augur - Public Health Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  <link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#augur" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version release.
  <a href="../../../.."> 
    <strong>Click here to go to the latest version release.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-header__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Public Health Bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Augur
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo" aria-label="Switch to Dark Mode" type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to Dark Mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"></path></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo" aria-label="Switch to Light Mode" type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to Light Mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m3.55 19.09 1.41 1.41 1.8-1.79-1.42-1.42M12 6c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6c0-3.32-2.69-6-6-6m8 7h3v-2h-3m-2.76 7.71 1.8 1.79 1.41-1.41-1.79-1.8M20.45 5l-1.41-1.4-1.8 1.79 1.42 1.42M13 1h-2v3h2M6.76 5.39 4.96 3.6 3.55 5l1.79 1.81zM1 13h3v-2H1m12 9h-2v3h2"></path></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../.." class="md-tabs__link">
          
  
  
  Public Health Bioinformatics

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-nav__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    Public Health Bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Public Health Bioinformatics
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            
  
    Public Health Bioinformatics
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/commandline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    With the Command-Line
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/terra/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    With Terra.bio
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/philosophy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Workflow Failure Philosophy
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Other Resources
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Other Resources
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/sops/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Available SOPs
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/versions/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Database and Docker Versions
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Guides
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Guides
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guide to GAMBIT
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit_database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guide to GAMBIT Database Creation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/phylogenetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guide to Phylogenetics
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/custom_organisms/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guide to Running Custom Organisms on TheiaCoV
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Workflows
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Workflows
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1">
        
          
          <label class="md-nav__link" for="__nav_1_5_1" id="__nav_1_5_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Workflows by Type
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1">
            <span class="md-nav__icon md-icon"></span>
            
  
    Workflows by Type
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_type/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_2" id="__nav_1_5_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Data Import
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Data Import
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Assembly_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BaseSpace_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Create_Terra_Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fetch_SRR_Accession
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    ONT_Barcode_Concatenation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    SRA_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_3" id="__nav_1_5_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Genomic Characterization
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Genomic Characterization
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Freyja Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Pangolin_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaCoV Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaEuk Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaMeta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaProk Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaViral Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    VADR_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_4" id="__nav_1_5_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Phylogenetic Construction
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Phylogenetic Construction
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Augur
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Augur
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Facts
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#augur-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Workflows
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur_prep_phb" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep_PHB
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur_Prep_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur_prep-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep Inputs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#augur_prep-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#augur_phb" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_PHB
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Inputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur Inputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#a-note-on-optional-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        A Note on Optional Inputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#workflow-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Workflow Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      
        References
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Core_Gene_SNP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Clair3_Variants_ONT
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    CZGenEpi_Prep
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Find_Shared_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP4
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Lyve_SET
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MashTree_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Tree
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_5" id="__nav_1_5_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Phylogenetic Placement
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Phylogenetic Placement
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Nextclade_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Usher
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_6" id="__nav_1_5_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Public Data Sharing
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Public Data Sharing
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Mercury_Prep_N_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_ENA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_GISAID
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_NCBI
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_7">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_7" id="__nav_1_5_1_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Exporting Data from Terra
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_7">
            <span class="md-nav__icon md-icon"></span>
            
  
    Exporting Data from Terra
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Transfer_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Zip_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_8">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_8" id="__nav_1_5_1_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Standalone
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_8">
            <span class="md-nav__icon md-icon"></span>
            
  
    Standalone
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    AMR_Search
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Cauris_CladeTyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Dorado_Basecalling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Illumina_Lanes
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GAMBIT_Query
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Kraken2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-AMRFinderPlus
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-Scrub
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    PhyloCompare
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    RASUSA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Rename_FASTQ
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TBProfiler_tNGS
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaValidate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2" id="__nav_1_5_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Workflows by Kingdom
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Workflows by Kingdom
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_kingdom/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_2" id="__nav_1_5_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Any Taxa
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Any Taxa
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    AMR_Search
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Assembly_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BaseSpace_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Clair3_Variants_ONT
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Illumina_Lanes
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Create_Terra_Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Dorado_Basecalling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fetch_SRR_Accession
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Kraken2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-Scrub
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    ONT_Barcode_Concatenation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    PhyloCompare
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    RASUSA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Rename_FASTQ
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    SRA_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaMeta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaValidate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Transfer_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Zip_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_3" id="__nav_1_5_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Bacteria
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Bacteria
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Core_Gene_SNP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Find_Shared_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GAMBIT_Query
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP4
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Lyve_SET
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MashTree_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-AMRFinderPlus
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Tree
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TBProfiler_tNGS
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_ENA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_NCBI
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaProk Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_4" id="__nav_1_5_2_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Mycotics
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Mycotics
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Cauris_CladeTyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Find_Shared_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GAMBIT_Query
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP4
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MashTree_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-AMRFinderPlus
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_NCBI
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaEuk Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_5" id="__nav_1_5_2_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Viral
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Viral
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Augur
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Augur
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Facts
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#augur-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Workflows
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur_prep_phb" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep_PHB
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur_Prep_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur_prep-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep Inputs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#augur_prep-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#augur_phb" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_PHB
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Inputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur Inputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#a-note-on-optional-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        A Note on Optional Inputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#workflow-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Workflow Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      
        References
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    CZGenEpi_Prep
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Freyja Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP4
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MashTree_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Mercury_Prep_N_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Nextclade_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Pangolin_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_ENA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_GISAID
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_NCBI
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaCoV Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaViral Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Usher
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    VADR_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_3" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Workflows Alphabetically
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Workflows Alphabetically
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_alphabetically/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    AMR_Search
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Assembly_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Augur
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Augur
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Facts
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#augur-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Workflows
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur_prep_phb" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep_PHB
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur_Prep_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur_prep-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep Inputs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#augur_prep-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#augur_phb" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_PHB
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Inputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur Inputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#a-note-on-optional-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        A Note on Optional Inputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#workflow-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Workflow Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      
        References
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BaseSpace_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Cauris_CladeTyper
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Clair3_Variants_ONT
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concatenate_Illumina_Lanes
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Core_Gene_SNP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Create_Terra_Table
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    CZGenEpi_Prep
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Dorado_Basecalling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fetch_SRR_Accession
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Find_Shared_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Freyja Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GAMBIT_Query
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Kraken2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP3
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    kSNP4
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Lyve_SET
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MashTree_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Mercury_Prep_N_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-AMRFinderPlus
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    NCBI-Scrub
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Nextclade_Batch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    ONT_Barcode_Concatenation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Pangolin_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    PhyloCompare
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    RASUSA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Rename_FASTQ
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Streamline_FASTA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Tree
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Snippy_Variants
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    SRA_Fetch
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TBProfiler_tNGS
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_ENA
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_GISAID
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Terra_2_NCBI
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaCoV Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaEuk Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaMeta
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaProk Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaValidate
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    TheiaViral Workflow Series
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Transfer_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Usher
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    VADR_Update
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Zip_Column_Content
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Contributing
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Contributing
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/doc_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributing to the Documentation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/code_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributing to the Code
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Facts
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#augur-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Workflows
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur_prep_phb" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep_PHB
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur_Prep_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur_prep-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep Inputs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#augur_prep-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_Prep Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#augur_phb" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur_PHB
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Inputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Augur Inputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#a-note-on-optional-inputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        A Note on Optional Inputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#workflow-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Workflow Tasks
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#augur-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Augur Outputs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      
        References
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
                




  <nav class="md-path" aria-label="Navigation">
    <ol class="md-path__list">
      
      
        
  
  
    
    
      <li class="md-path__item">
        <a href="../../.." class="md-path__link">
          
  <span class="md-ellipsis">
    Public Health Bioinformatics
  </span>

        </a>
      </li>
    
  

      
        
  
  
    
    
      
  
  
    
    
      <li class="md-path__item">
        <a href="../../../workflows_overview/workflows_type/" class="md-path__link">
          
  <span class="md-ellipsis">
    Workflows
  </span>

        </a>
      </li>
    
  

    
  

      
        
  
  
    
    
      <li class="md-path__item">
        <a href="../../../workflows_overview/workflows_alphabetically/" class="md-path__link">
          
  <span class="md-ellipsis">
    Workflows Alphabetically
  </span>

        </a>
      </li>
    
  

      
    </ol>
  </nav>

              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="augur">Augur<a class="headerlink" href="#augur" title="Permanent link"></a></h1>
<h2 id="quick-facts">Quick Facts<a class="headerlink" href="#quick-facts" title="Permanent link"></a></h2>
<table>
<thead>
<tr>
<th><strong>Workflow Type</strong></th>
<th><strong>Applicable Kingdom</strong></th>
<th><strong>Last Known Changes</strong></th>
<th><strong>Command-line Compatibility</strong></th>
<th><strong>Workflow Level</strong></th>
<th><strong>Dockstore</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../../../workflows_overview/workflows_type/#phylogenetic-construction">Phylogenetic Construction</a></td>
<td><a href="../../../workflows_overview/workflows_kingdom/#viral">Viral</a></td>
<td>v4.1.0</td>
<td>Yes</td>
<td>Sample-level, Set-level</td>
<td><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/Augur_Prep_PHB:main?tab=info">Augur_Prep_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/Augur_PHB:main?tab=info">Augur_PHB</a></td>
</tr>
</tbody>
</table>
<h2 id="augur-workflows">Augur Workflows<a class="headerlink" href="#augur-workflows" title="Permanent link"></a></h2>
<div class="admonition caption">
<p class="admonition-title">Augur Workflow Diagram</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/Augur_MAIN.png" data-desc-position="bottom"><img alt="Augur Workflow Diagram" src="../../../assets/figures/Augur_MAIN.png"></a></p>
</div>
<div class="admonition dna inline end">
<p class="admonition-title"><strong>Helpful resources for epidemiological interpretation</strong></p>
<p><a href="https://www.cdc.gov/advanced-molecular-detection/php/training/">CDC's COVID-19 Epidemiology Toolkit</a> is a useful resource for learning more about genomic epidemiology. Some particularly relevant modules to the Augur workflows include:</p>
<ul>
<li><a href="https://www.cdc.gov/advanced-molecular-detection/php/training/module-1-3.html">Module 1.3: How to read a phylogenetic tree</a></li>
<li><a href="https://www.cdc.gov/advanced-molecular-detection/php/training/module-3-1.html">Module 3.1: Getting started with Nextstrain</a>&nbsp;(which includes Auspice)</li>
<li><a href="https://www.cdc.gov/advanced-molecular-detection/php/training/module-3-4.html">Module 3.4: Walking through Nextstrain trees</a></li>
</ul>
</div>
<p>Genomic epidemiology is an important approach to understand and mitigate disease transmission. A critical step in viral genomic epidemiology is generating phylogenetic trees to explore the genetic relationship between viruses on a local, regional, national, or global scale. The Augur workflows enable <strong>viral phylogenetic analysis</strong> by generating phylogenetic trees from genome assemblies and incorporating metadata into a intuitive visual platform via <a href="https://auspice.us">Auspice</a>.</p>
<p>Two workflows are offered: <strong>Augur_Prep_PHB</strong> and <strong>Augur_PHB</strong>. <strong>Augur_Prep_PHB</strong> prepares sample metadata to be visualized alongside the phylogenetic tree produced by <strong>Augur_PHB</strong>. <strong>Augur_PHB</strong> can be run without metadata to produce only a phylogenetic tree. If you want metadata incorporated in your final tree, these workflows must be run sequentially. The final outputs from <strong>Augur_PHB</strong> can be visualized in&nbsp;<a href="https://auspice.us">Auspice</a>, which is the recommended platform. Alternative tree visualization platforms can also be used, though these may not support all metadata features.</p>
<h3 id="augur_prep_phb">Augur_Prep_PHB<a class="headerlink" href="#augur_prep_phb" title="Permanent link"></a></h3>
<p>The <strong>Augur_Prep_PHB</strong> workflow was written to prepare individual sample assemblies and their metadata for inclusion in Augur_PHB analysis. The optional metadata inputs include collection date information (in <code>YYYY-MM-DD</code> format), clade information (like <code>nextclade clade</code> and/or <code>pango lineage</code>), and geographical information.</p>
<p>This workflow runs on the sample level, and takes assembly FASTA files and associated metadata formatted in a data table. FASTA files may be generated with one of the TheiaCoV/TheiaViral Characterization workflows and should adhere to quality control guidelines, (e.g.&nbsp;<a href="https://github.com/pha4ge/pipeline-resources/blob/main/docs/qc-solutions.md">QC guidelines produced by PHA4GE</a>).</p>
<div class="admonition dna">
<p class="admonition-title"><strong>How to prepare metadata</strong></p>
<p>If you are running this workflow on Terra, we recommend carefully preparing metadata in a TSV file that will be uploaded to the same Terra datatable that contains the sample genetic information. An example of a correctly formatted TSV file can be found in <a href="../../../assets/files/example-augur-prep-metadata.tsv">this&nbsp;example</a>. A few important considerations are:</p>
<ul>
<li>Please always include the <strong>date information</strong> in <code>YYYY-MM-DD</code> format. Other date formats are incompatible with Augur. You can specify unknown dates or month by replacing the respective values by <code>XX</code> (e.g. <code>2013-01-XX</code> or <code>2011-XX-XX</code>), while completely unknown dates can be shown with <code>20XX-XX-XX</code> (which does not restrict the sequence to being in the 21st century - they could be earlier). Alternatively, reduced precision format can also be used (e.g. <code>2018</code>, <code>2018-03</code>).<ul>
<li>Because Excel will automatically change the date formatting, we recommend not opening or preparing your meta data file in Excel. If the metadata is already in Excel, or you decide to prepare it in Excel, we recommend using another program to correct the dates afterwards (and be caution if you open it in Excel again!).</li>
</ul>
</li>
<li>Different levels of <strong>geographical information</strong> can be passed to Augur. A latitude and longitude file is provided by default by Theiagen, which mirrors what can be found <a href="https://github.com/nextstrain/augur/blob/master/augur/data/lat_longs.tsv">here</a>. Just ensure that your spelling matches what is in the file exactly or alternatively provide your own. Augur_Prep supports the following levels:<ul>
<li><code>region</code> - Lowest-level resolution, used often for continents (e.g.: <code>europe</code>, <code>asia</code>, <code>north america</code>)</li>
<li><code>country</code> - Denotes the country where the sample originated (e.g.: <code>Argentina</code>, <code>Japan</code>, <code>USA</code>)</li>
<li><code>divisions</code> - Denotes divisions, or states, or sometimes cities, within the country (e.g.: <code>California</code>, <code>Colorado</code>, <code>Cork</code>)</li>
<li><code>location</code> - Highest-level resolution, often used for custom latitude and longitude for futher detail on divisions, like cities within states. Just ensure that this level is provided in either the default latitude and longitude file or in a custom one. </li>
</ul>
</li>
<li>Optional <strong>clade</strong> information, such the one assigned by <em>Nextclade</em>.</li>
<li>Optional <strong>Pangolin lineage</strong> information for SARS-CoV-2 samples.</li>
</ul>
</div>
<h4 id="augur_prep-inputs">Augur_Prep Inputs<a class="headerlink" href="#augur_prep-inputs" title="Permanent link"></a></h4>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>augur_prep</td>
<td><strong>assembly</strong></td>
<td>File</td>
<td>The assembly file for your sample in FASTA format</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>augur_prep</td>
<td><strong>clade</strong></td>
<td>String</td>
<td>Clade membership metadata</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_prep</td>
<td><strong>collection_date</strong></td>
<td>String</td>
<td>Collection date of the sample</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_prep</td>
<td><strong>country</strong></td>
<td>String</td>
<td>Country where sample was collected</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_prep</td>
<td><strong>division</strong></td>
<td>String</td>
<td>Sub-national location (e.g. state/province) where sample was collected</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_prep</td>
<td><strong>location</strong></td>
<td>String</td>
<td>Sub-divisional location (e.g. county) where sample was collected</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_prep</td>
<td><strong>pango_lineage</strong></td>
<td>String</td>
<td>The Pangolin lineage of the sample</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_prep</td>
<td><strong>region</strong></td>
<td>String</td>
<td>Continental/regional location where sample was collected</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>prep_augur_metadata</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>prep_augur_metadata</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>prep_augur_metadata</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>prep_augur_metadata</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>3</td>
<td>Optional</td>
</tr>
<tr>
<td>prep_augur_metadata</td>
<td><strong>organism</strong></td>
<td>String</td>
<td>The organism to be analyzed in Augur; options: "sars-cov-2", "flu", "MPXV", "rsv-a", "rsv-b"</td>
<td>sars-cov-2</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
<h4 id="augur_prep-outputs">Augur_Prep Outputs<a class="headerlink" href="#augur_prep-outputs" title="Permanent link"></a></h4>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>augur_metadata</td>
<td>File</td>
<td>TSV file of the metadata provided as input to the workflow in the proper format for Augur analysis</td>
</tr>
<tr>
<td>augur_prep_phb_analysis_date</td>
<td>String</td>
<td>Date of analysis</td>
</tr>
<tr>
<td>augur_prep_phb_version</td>
<td>String</td>
<td>Version of the Public Health Bioinformatics (PHB) repository used</td>
</tr>
</tbody>
</table>
</div>
<h3 id="augur_phb">Augur_PHB<a class="headerlink" href="#augur_phb" title="Permanent link"></a></h3>
<p><a href="https://docs.nextstrain.org/projects/augur/en/stable/">Augur</a> is a bioinformatics toolkit to track evolution from sequence data, ingesting sequences and metadata such as dates and sampling locations, filtering the data, aligning the sequences, infering a tree, and export the results in a format that can be visualized by <a href="https://auspice.us/">Auspice</a>. This is the tool behind <a href="https://nextstrain.org/">Nextrain's builds</a> available for a large collection of viral organisms.</p>
<div class="admonition info">
<p class="admonition-title">Before getting started</p>
<p><strong>Phylogenetic inference requires careful planning, quality control of sequences, and metadata curation</strong>. You may have to generate phylogenies multiple times by running the Augur_PHB workflow, with several iterations of assessing results and amending inputs, in order to generate a final tree with sufficient diversity and high-quality data of interest. Theiagen's <a href="../../../guides/phylogenetics/">Introduction to Phylogenetics</a> is one resource to give you the necessary information on the considerations you'll need to have before performing this type of analysis.</p>
</div>
<p><strong>Augur_PHB</strong> takes as input a <strong>set of assembly/consensus files</strong> (FASTA format), an <strong>optional viral organism designation</strong>, and an <strong>optional sample metadata files</strong> (TSV format) that have been formatted via the&nbsp;Augur_Prep_PHB workflow. Augur_PHB runs <a href="https://docs.nextstrain.org/projects/augur/en/stable/">Augur</a> to generate a phylogenetic tree following the construction of a SNP distance matrix and alignment. Provided metadata will be used to refine the final tree and incorporated into the <a href="https://auspice.us/">Auspice</a>-formatted tree visual.</p>
<h4 id="augur-inputs">Augur Inputs<a class="headerlink" href="#augur-inputs" title="Permanent link"></a></h4>
<div class="admonition warning">
<p class="admonition-title">Sample diversity and tree building</p>
<p>Before attempting a phylogenetic tree, you must ensure that the input FASTAs meet quality-control metrics. Sets of FASTAs with highly discordant quality metrics may result in the inaccurate inference of genetic relatedness.</p>
<p>There must be some sequence diversity among the set of input assemblies. If insufficient diversity is present, it may be necessary to add a more divergent sequence to the set of samples to be analyzed.</p>
</div>
<p>Some inputs will automatically bypass or trigger modules, such as populating <code>alignment_fasta</code>, which bypasses alignment. Clade-defining mutations can be automatically extracted if the "clade_membership" metadata field is provided and the <code>extract_clade_mutations</code> optional input is set to <code>true</code>.</p>
<p>Any metadata present in the final JSON file for Auspice visualization is determined by what metadata was provided by the user. If the <code>sample_metadata_tsvs</code> optional input parameter is <strong>not</strong> provided, the final tree visual will only include the distance tree. If metadata was provided, different metadata fields will trigger different steps: date information will trigger the refinement of the distance tree into a time tree; clade information will be assigned to the tree nodes; geographical information will be represented in the Auspice visual within a map. The following figure illustrates this logic.</p>
<div class="admonition caption">
<p class="admonition-title">Augur Metadata Conditionals</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/Augur_Metadata_Conditionals.png" data-description="The metadata and type of tree in the output JSON for Auspice will depend on the metadata that is present in input metadata file. If no metadata file is provided, the output JSON will only contain a distance tree. If date information is present, the distance tree will be replaced by a tree refined by time (time tree). If clade and/or pango lineage (for SARS-CoV-2) information is provided, the tree will display an option to color by lineage. If geographical information is present, a map will load in Auspice using the information provided." data-desc-position="bottom"><img alt="Augur Metadata Conditionals" data-description="The metadata and type of tree in the output JSON for Auspice will depend on the metadata that is present in input metadata file. If no metadata file is provided, the output JSON will only contain a distance tree. If date information is present, the distance tree will be replaced by a tree refined by time (time tree). If clade and/or pango lineage (for SARS-CoV-2) information is provided, the tree will display an option to color by lineage. If geographical information is present, a map will load in Auspice using the information provided." src="../../../assets/figures/Augur_Metadata_Conditionals.png"></a></p>
</div>
<h5 id="a-note-on-optional-inputs"><strong>A Note on Optional Inputs</strong><a class="headerlink" href="#a-note-on-optional-inputs" title="Permanent link"></a></h5>
<div class="admonition warning">
<p class="admonition-title">Defaults change based on the specified organism input</p>
<p>Default values that mimic the Nextstrain builds for the following organisms have been preselected:</p>
<ul>
<li>Flu (<code>"flu"</code>), which also requires the following two inputs:<ul>
<li><code>flu_segment</code> (<code>"HA"</code> or <code>"NA"</code>)</li>
<li><code>flu_subtype</code> (<code>"H1N1"</code>, <code>"H3N2"</code>, <code>"Victoria"</code>, <code>"Yamagata"</code>, or <code>"H5N1"</code>)</li>
</ul>
</li>
<li>RSV-A (<code>"rsv-a"</code>)</li>
<li>RSV-B (<code>"rsv-b"</code>)</li>
<li>Mpox (<code>"mpox"</code>)</li>
<li>SARS-CoV-2 (default; <code>"sars-cov-2"</code>) </li>
</ul>
<p>View these default parameters in the relevant toggle block below.</p>
<details class="task">
<summary><code>organism_parameters</code>: Setting default values for specific organisms</summary>
<p>Organism Parameters acquires and propagates default files and variable values for specific organisms.</p>
<!--if: virus -->
<details class="toggle">
<summary>Default values for SARS-CoV-2</summary>
<ul>
<li>min_num_unambig = 27000</li>
<li>clades_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/sc2_clades_20251008.tsv"</code></li>
<li>lat_longs_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/sc2_lat_longs_20251008.tsv"</code></li>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/MN908947.fasta"</code></li>
<li>reference_genbank = <code>"gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/sc2_reference_seq_20251008.gb"</code></li>
<li>auspice_config = <code>"gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/sc2_auspice_config_20251008.json"</code></li>
<li>min_date = 2020.0</li>
<li>pivot_interval = 1</li>
<li>pivot_interval_units = "weeks"</li>
<li>narrow_bandwidth = 0.05</li>
<li>proportion_wide = 0.0</li>
</ul>
</details>
<details class="toggle">
<summary>Default values for Flu</summary>
<ul>
<li>lat_longs_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/lat_longs.tsv"</code></li>
<li>min_num_unambig = 900</li>
<li>min_date = 2020.0</li>
<li>pivot_interval = 1</li>
<li>narrow_bandwidth = 0.1666667</li>
<li>proportion_wide = 0.0</li>
</ul>
<details class="toggle">
<summary>H1N1</summary>
<ul>
<li>auspice_config = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/auspice_config_h1n1pdm.json"</code></li>
<li>HA<ul>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_h1n1pdm_ha.gb"</code></li>
<li>clades_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/clades_h1n1pdm_ha.tsv"</code></li>
</ul>
</li>
<li>NA<ul>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_h1n1pdm_na.gb"</code></li>
</ul>
</li>
</ul>
</details>
<details class="toggle">
<summary>H3N2</summary>
<ul>
<li>auspice_config = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/auspice_config_h3n2.json"</code></li>
<li>HA<ul>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_h3n2_ha.gb"</code></li>
<li>clades_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/clades_h3n2_ha.tsv"</code></li>
</ul>
</li>
<li>NA<ul>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_h3n2_na.gb"</code></li>
</ul>
</li>
</ul>
</details>
<details class="toggle">
<summary>Victoria</summary>
<ul>
<li>auspice_config = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/auspice_config_vic.json"</code></li>
<li>HA<ul>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_vic_ha.gb"</code></li>
<li>clades_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/clades_vic_ha.tsv"</code></li>
</ul>
</li>
<li>NA<ul>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_vic_na.gb"</code></li>
</ul>
</li>
</ul>
</details>
<details class="toggle">
<summary>Yamagata</summary>
<ul>
<li>auspice_config = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/auspice_config_yam.json"</code></li>
<li>HA<ul>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_yam_ha.gb"</code></li>
<li>clades_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/clades_yam_ha.tsv"</code></li>
</ul>
</li>
<li>NA<ul>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_yam_na.gb"</code></li>
</ul>
</li>
</ul>
</details>
<details class="toggle">
<summary>H5N1</summary>
<ul>
<li>auspice_config = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/auspice_config_h5n1.json"</code></li>
<li>HA<ul>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_h5n1_ha.gb"</code></li>
<li>clades_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/h5nx-clades.tsv"</code></li>
</ul>
</li>
</ul>
</details>
</details>
<details class="toggle">
<summary>Default values for MPXV</summary>
<ul>
<li>min_num_unambig = 150000</li>
<li>clades_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/mpox/mpox_clades.tsv"</code></li>
<li>lat_longs_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/lat_longs.tsv"</code></li>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/mpox/NC_063383.1.reference.fasta"</code></li>
<li>reference_genbank = <code>"gs://theiagen-public-resources-rp/reference_data/viral/mpox/NC_063383.1_reference.gb"</code></li>
<li>auspice_config = <code>"gs://theiagen-public-resources-rp/reference_data/viral/mpox/mpox_auspice_config_mpxv.json"</code></li>
<li>min_date = 2020.0</li>
<li>pivot_interval = 1</li>
<li>narrow_bandwidth = 0.1666667</li>
<li>proportion_wide = 0.0</li>
</ul>
</details>
<details class="toggle">
<summary>Default values for RSV-A</summary>
<ul>
<li>min_num_unambig = 10850</li>
<li>clades_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/rsv/rsv_a_clades.tsv"</code></li>
<li>lat_longs_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/lat_longs.tsv"</code></li>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/rsv/reference_rsv_a.EPI_ISL_412866.fasta"</code></li>
<li>reference_genbank = <code>""gs://theiagen-public-resources-rp/reference_data/viral/rsv/reference_rsv_a.gb"</code></li>
<li>auspice_config = <code>""gs://theiagen-public-resources-rp/reference_data/viral/rsv/rsv_auspice_config.json"</code></li>
<li>min_date = 2020.0</li>
<li>pivot_interval = 1</li>
<li>narrow_bandwidth = 0.1666667</li>
<li>proportion_wide = 0.0</li>
</ul>
</details>
<details class="toggle">
<summary>Default values for RSV-B</summary>
<ul>
<li>min_num_unambig = 10850</li>
<li>clades_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/rsv/rsv_b_clades.tsv"</code></li>
<li>lat_longs_tsv = <code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/lat_longs.tsv"</code></li>
<li>reference_fasta = <code>"gs://theiagen-public-resources-rp/reference_data/viral/rsv/reference_rsv_b.EPI_ISL_1653999.fasta"</code></li>
<li>reference_genbank = <code>""gs://theiagen-public-resources-rp/reference_data/viral/rsv/reference_rsv_b.gb"</code></li>
<li>auspice_config = <code>""gs://theiagen-public-resources-rp/reference_data/viral/rsv/rsv_auspice_config.json"</code></li>
<li>min_date = 2020.0</li>
<li>pivot_interval = 1</li>
<li>narrow_bandwidth = 0.1666667</li>
<li>proportion_wide = 0.0</li>
</ul>
</details>
<!-- endif -->
<div class="admonition techdetails">
<p class="admonition-title">Organism Parameters Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">wf_organism_parameters.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="info" open="open">
<summary>Running Augur_PHB on custom organsisms</summary>
<p>For non-default organisms (listed above), several optional inputs are required to guarantee workflow functionality</p>
<table>
<thead>
<tr>
<th><strong>Task</strong></th>
<th><strong>Input</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>augur</td>
<td><em>reference_fasta</em></td>
<td>Reference sequence in FASTA format.</td>
</tr>
<tr>
<td>augur</td>
<td><em>reference_genbank</em></td>
<td>Reference sequence in GenBank format.</td>
</tr>
<tr>
<td>augur</td>
<td><em>organism</em></td>
<td>Name of expected organism.</td>
</tr>
<tr>
<td>augur</td>
<td><em>min_num_unambig</em></td>
<td>Minimum number of called bases in genome to pass prefilter.</td>
</tr>
<tr>
<td>augur</td>
<td><em>lat_longs_tsv</em></td>
<td>Tab-delimited file of geographic location names with corresponding latitude and longitude values. Only necessary if geographical information is in the metadata; must follow <a href="https://github.com/nextstrain/augur/blob/master/augur/data/lat_longs.tsv">this</a> format</td>
</tr>
<tr>
<td>augur</td>
<td><em>clades_tsv</em></td>
<td>TSV file containing clade mutation positions in four columns. Only necessary if clade information is in the metadata.</td>
</tr>
</tbody>
</table>
<p>In the inputs table below, these fields have both the "Required" and "Optional" tags.</p>
</details>
<p>There are <strong>many</strong> optional user inputs. For more information regarding these optional inputs, please view <a href="https://docs.nextstrain.org/projects/augur/en/stable/usage/usage.html">Nextstrain's detailed documentation on Augur</a></p>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>augur</td>
<td><strong>assembly_fastas</strong></td>
<td>Array[File]+</td>
<td>The assembly files for your samples in FASTA format</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>augur</td>
<td><strong>build_name</strong></td>
<td>String</td>
<td>Name to give to the Augur build</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>augur</td>
<td><strong>clades_tsv</strong></td>
<td>File</td>
<td>TSV file containing clade mutation positions in four columns</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, an empty clades file is provided to prevent workflow failure, "gs://theiagen-public-resources-rp/empty_files/minimal-clades.tsv", but will not be as useful as an organism specific clades file.</td>
<td>Optional, Required</td>
</tr>
<tr>
<td>augur</td>
<td><strong>flu_subtype</strong></td>
<td>String</td>
<td>Required if organism = "flu". The subtype of the flu samples being analyzed; options: "H1N1", "H3N2", "Victoria", "Yamagata", "H5N1"</td>
<td>N/A</td>
<td>Optional, Required</td>
</tr>
<tr>
<td>augur</td>
<td><strong>reference_fasta</strong></td>
<td>File</td>
<td>The reference FASTA file used to align the genomes and build the trees</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, a reference fasta file must be provided otherwise the workflow fails.</td>
<td>Optional, Required</td>
</tr>
<tr>
<td>augur</td>
<td><strong>reference_genbank</strong></td>
<td>File</td>
<td>The GenBank .gb file for the same reference genome used for the reference_fasta</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>.</td>
<td>Optional, Required</td>
</tr>
<tr>
<td>augur</td>
<td><strong>alignment_fasta</strong></td>
<td>File</td>
<td>The alignment fasta file in Augur</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>augur_id_column</strong></td>
<td>String</td>
<td>Column name of sequence IDs</td>
<td>strain</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>augur_trait_columns</strong></td>
<td>String</td>
<td>Comma-separated list of columns to use for trait analysis in Augur</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>auspice_config</strong></td>
<td>File</td>
<td>Auspice config file for customizing visualizations; takes priority over the other customization values available for augur_export</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, a minimal auspice config file is provided to prevent workflow failure, "gs://theiagen-public-resources-rp/empty_files/minimal-auspice-config.json", but will not be as useful as an organism specific config file.</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>extract_clade_mutations</strong></td>
<td>Boolean</td>
<td>Generate a "clades.tsv" using "clade_membership" column metadata (overrides clades_tsv)</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>flu_segment</strong></td>
<td>String</td>
<td>Required if organism = "flu". The name of the segment to be analyzed; options: "HA" or "NA"</td>
<td>HA</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>lat_longs_tsv</strong></td>
<td>File</td>
<td>Tab-delimited file of geographic location names with corresponding latitude and longitude values</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, a minimal lat-long file is provided to prevent workflow failure, "gs://theiagen-public-resources-rp/empty_files/minimal-lat-longs.tsv", but will not be as useful as a detailed lat-longs file covering all the locations for the samples to be visualized.</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>midpoint_root_tree</strong></td>
<td>Boolean</td>
<td>Boolean variable that will instruct the workflow to reroot the tree at the midpoint</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>min_date</strong></td>
<td>Float</td>
<td>Minimum date to begin filtering or frequencies calculations</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default value is 0.0</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>min_num_unambig</strong></td>
<td>Int</td>
<td>Minimum number of called bases in genome to pass prefilter</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default value is 0</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>narrow_bandwidth</strong></td>
<td>Float</td>
<td>The bandwidth for the narrow KDE</td>
<td>0.08333</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>organism</strong></td>
<td>String</td>
<td>Organism used to preselect default values; options: "sars-cov-2", "flu", "mpxv", "rsv-a", "rsv-b"</td>
<td>sars-cov-2</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>outgroup_root</strong></td>
<td>String</td>
<td>Tip name to root phylogenetic tree upon (overrides midpoint root)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>pivot_interval</strong></td>
<td>Int</td>
<td>Number of units between pivots</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default value is 3</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>proportion_wide</strong></td>
<td>Float</td>
<td>The proportion of the wide bandwidth to use in the KDE mixture model</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default value is 0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>remove_reference</strong></td>
<td>Boolean</td>
<td>Whether or not to remove the reference in Augur</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>augur</td>
<td><strong>sample_metadata_tsvs</strong></td>
<td>Array[File]</td>
<td>An array of the metadata files produced in Augur_Prep_PHB</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_align</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>64</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_align</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>750</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_align</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/augur:31.5.0</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_align</td>
<td><strong>fill_gaps</strong></td>
<td>Boolean</td>
<td>If true, gaps represent missing data rather than true indels and so are replaced by N after aligning.</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_align</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>128</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_ancestral</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_ancestral</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_ancestral</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/augur:31.5.0</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_ancestral</td>
<td><strong>infer_ambiguous</strong></td>
<td>Boolean</td>
<td>If true, infer nucleotides and ambiguous sites and replace with most likely</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_ancestral</td>
<td><strong>inference</strong></td>
<td>String</td>
<td>Calculate joint or marginal maximum likelihood ancestral sequence states; options: "joint", "marginal"</td>
<td>joint</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_ancestral</td>
<td><strong>keep_ambiguous</strong></td>
<td>Boolean</td>
<td>If true, do not infer nucleotides at ambiguous (N) sides</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_ancestral</td>
<td><strong>keep_overhangs</strong></td>
<td>Boolean</td>
<td>If true, do not infer nucleotides for gaps on either side of the alignment</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_ancestral</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_clades</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_clades</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_clades</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/augur:31.5.0</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_clades</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_export</td>
<td><strong>colors_tsv</strong></td>
<td>File</td>
<td>Custom color defintiions, one per line in the format TRAIT_TYPE\tTRAIT_VALUE\tHEX_CODE</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_export</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_export</td>
<td><strong>description_md</strong></td>
<td>File</td>
<td>Markdown file with description of build and/or acknowledgements</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_export</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_export</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/augur:31.5.0</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_export</td>
<td><strong>include_root_sequence</strong></td>
<td>Boolean</td>
<td>Export an additional JSON containing the root sequence used to identify mutations</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_export</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>64</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_export</td>
<td><strong>title</strong></td>
<td>String</td>
<td>The title to be displayed by auspice</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>branch_length_inference</strong></td>
<td>String</td>
<td>Branch length mode of timetree to use; options: "auto", "joint", "marginal", "input"</td>
<td>auto</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>clock_filter_iqd</strong></td>
<td>Int</td>
<td>Remove tips that deviate more than n_iqd interquartile ranges from the root-to-tip vs time regression</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>clock_rate</strong></td>
<td>Float</td>
<td>Fixed clock rate to use for time tree calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>clock_std_dev</strong></td>
<td>Float</td>
<td>Standard deviation of the fixed clock_rate estimate</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>coalescent</strong></td>
<td>String</td>
<td>Coalescent time scale in units of inverse clock rate (float), optimize as scalar ("opt") or skyline ("skyline")</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>covariance</strong></td>
<td>Boolean</td>
<td>If true, account for covariation when estimating rates and/or rerooting</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>date_confidence</strong></td>
<td>Boolean</td>
<td>If true, calculate confidence intervals for node dates</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>date_inference</strong></td>
<td>String</td>
<td>Assign internal nodes to their marginally most likely dates; options: "joint", "marginal"</td>
<td>marginal</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>divergence_units</strong></td>
<td>String</td>
<td>Units in which sequence divergences is exported; options: "mutations" or "mutations-per-site"</td>
<td>mutations</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/augur:31.5.0</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>gen_per_year</strong></td>
<td>Int</td>
<td>Number of generations per year</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>keep_polytomies</strong></td>
<td>Boolean</td>
<td>If true, don't attempt to resolve polytomies</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>keep_root</strong></td>
<td>Boolean</td>
<td>If true, do not reroot the tree; use it as-is (overrides anything specified by root)</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>precision</strong></td>
<td>Int</td>
<td>Precision used by TreeTime to determine the number of grid points that are used for the evaluation of the branch length interpolation objects. Values range from 0 (rough) to 3 (ultra fine) and default to 'auto'</td>
<td>'auto'</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>root</strong></td>
<td>String</td>
<td>Rooting mechanism; options: "best", "least-squares", "min_dev", "oldest", etc.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_traits</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_traits</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_traits</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/augur:31.5.0</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_traits</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_traits</td>
<td><strong>metadata_id_columns</strong></td>
<td>String</td>
<td>The names of possible metadata columns containing identifier information, ordered by priority</td>
<td>'strain' 'name'</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_traits</td>
<td><strong>weights</strong></td>
<td>File</td>
<td>a dictionary of key/value mappings in JSON format used to weight KDE tip frequencies</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_translate</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_translate</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_translate</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/augur:31.5.0</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_translate</td>
<td><strong>genes</strong></td>
<td>File</td>
<td>A file containing a list of genes to translate (from nucleotides to amino acids)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_translate</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_tree</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>64</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_tree</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>750</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_tree</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/augur:31.5.0</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_tree</td>
<td><strong>exclude_sites</strong></td>
<td>File</td>
<td>File of one-based sites to exclude for raw tree building (BED format in .bed files, DRM format in tab-delimited files, or one position per line)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>augur_tree</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_tree</td>
<td><strong>method</strong></td>
<td>String</td>
<td>The method used to build the tree. Options: "fasttree", "raxml", "iqtree" (default)</td>
<td>iqtree</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_tree</td>
<td><strong>override_default_args</strong></td>
<td>Boolean</td>
<td>If true, override default tree builder arguments instead of augmenting them</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_tree</td>
<td><strong>substitution_model</strong></td>
<td>String</td>
<td>The substitution model to use; only available for iqtree. Specify "auto" to run ModelTest; model options can be found here</td>
<td>GTR</td>
<td>Optional</td>
</tr>
<tr>
<td>augur_tree</td>
<td><strong>tree_builder_args</strong></td>
<td>String</td>
<td>Additional tree builder arguments either augmenting or overriding the default arguments. FastTree defaults: "-nt -nosupport". RAxML defaults: "-f d -m GTRCAT -c 25 -p 235813". IQ-TREE defaults: "-ninit 2 -n 2 -me 0.05 -nt AUTO -redo"</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>cat_files</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>cat_files</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>cat_files</td>
<td><strong>docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>cat_files</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>clade_extraction_task</td>
<td><strong>allow_missing_tips</strong></td>
<td>Boolean</td>
<td>Generate clades if metadata is missing for some tips</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>clade_extraction_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clade_extraction_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>clade_extraction_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/extract_clade_mutations:0.0.1</td>
<td>Optional</td>
</tr>
<tr>
<td>clade_extraction_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>clade_extraction_task</td>
<td><strong>skip_singleton_clades</strong></td>
<td>Boolean</td>
<td>Skip extracting sequence signatures for clades with a single tip</td>
<td>True</td>
<td>Optional</td>
</tr>
<tr>
<td>fasta_to_ids</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>fasta_to_ids</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>375</td>
<td>Optional</td>
</tr>
<tr>
<td>fasta_to_ids</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>ubuntu</td>
<td>Optional</td>
</tr>
<tr>
<td>fasta_to_ids</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>filter_sequences_by_length</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>filter_sequences_by_length</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>300</td>
<td>Optional</td>
</tr>
<tr>
<td>filter_sequences_by_length</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/broadinstitute/viral-core:2.1.33</td>
<td>Optional</td>
</tr>
<tr>
<td>filter_sequences_by_length</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>mutation_context</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>mutation_context</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>mutation_context</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/nextstrain-mpox-mutation-context:2024-06-27</td>
<td>Optional</td>
</tr>
<tr>
<td>mutation_context</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_genoflu_genotype</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>gene_locations_bed_file</strong></td>
<td>File</td>
<td>Use to provide locations of interest where average coverage will be calculated</td>
<td>Default provided for SARS-CoV-2 ("gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/sc2_gene_locations.bed") and mpox ("gs://theiagen-public-resources-rp/reference_data/viral/mpox/mpox_gene_locations.bed")</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>genome_length_input</strong></td>
<td>Int</td>
<td>Use to specify the expected genome length; provided by default for all supported organisms</td>
<td>Default provided for SARS-CoV-2 (29903), mpox (197200), WNV (11000), flu (13000), RSV-A (16000), RSV-B (16000), HIV (primer versions 1 [9181] and 2 [9840])</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>hiv_primer_version</strong></td>
<td>String</td>
<td>The version of HIV primers used. Options are <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156</a> and <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164</a>. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>v1</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>kraken_target_organism_input</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td>Default provided for mpox (Monkeypox virus), WNV (West Nile virus), and HIV (Human immunodeficiency virus 1)</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>nextclade_dataset_name_input</strong></td>
<td>String</td>
<td>NextClade organism dataset name</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default is "NA".</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>nextclade_dataset_tag_input</strong></td>
<td>String</td>
<td>NextClade organism dataset tag</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default is "NA".</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>pangolin_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>primer_bed_file</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td>REQUIRED FOR SARS-CoV-2, MPOX, WNV, RSV-A &amp; RSV-B. Provided by default only for HIV primer versions 1 ("gs://theiagen-public-resources-rp/reference_data/viral/hiv/HIV-1_v1.0.primer.hyphen.bed" and 2 ("gs://theiagen-public-resources-rp/reference_data/viral/hiv/HIV-1_v2.0.primer.hyphen400.1.bed")</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_gff_file</strong></td>
<td>File</td>
<td>Reference GFF file for the organism being analyzed</td>
<td>Default provided for mpox ("gs://theiagen-public-resources-rp/reference_data/viral/mpox/Mpox-MT903345.1.reference.gff3") and HIV (primer versions 1 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/NC_001802.1.gff3"] and 2 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/AY228557.1.gff3"])</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_max_length</strong></td>
<td>Int</td>
<td>Maximum length for the fasta-trim-terminal-ambigs.pl VADR script</td>
<td>Default provided for SARS-CoV-2 (30000), mpox (210000), WNV (11000), flu (0), RSV-A (15500) and RSV-B (15500).</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_mem</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32 (RSV-A, RSV-B, WNV) and 16 (all other TheiaCoV organisms)</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_model</strong></td>
<td>File</td>
<td>Path to the a tar + gzipped VADR model file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-sarscov2-1.3-2.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_options</strong></td>
<td>String</td>
<td>Options for the v-annotate.pl VADR script</td>
<td>--mkey sarscov2 --glsearch -s -r --nomisc --lowsim5seq 6 --lowsim3seq 6 --alt_fail lowscore,insertnn,deletinn --noseqnamemax --out_allfasta</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_skip_length</strong></td>
<td>Int</td>
<td>Minimum assembly length (unambiguous) to run VADR</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>reorder_matrix</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>reorder_matrix</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>reorder_matrix</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>Optional</td>
</tr>
<tr>
<td>reorder_matrix</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>reorder_matrix</td>
<td><strong>phandango_coloring</strong></td>
<td>Boolean</td>
<td>Whether or not Phandango coloring will color all items from the same column the same</td>
<td>False</td>
<td>Optional</td>
</tr>
<tr>
<td>snp_dists</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>snp_dists</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>snp_dists</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snp-dists:0.8.2</td>
<td>Optional</td>
</tr>
<tr>
<td>snp_dists</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>tsv_join</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>tsv_join</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>tsv_join</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/broadinstitute/viral-core:2.1.33</td>
<td>Optional</td>
</tr>
<tr>
<td>tsv_join</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>7</td>
<td>Optional</td>
</tr>
<tr>
<td>tsv_join</td>
<td><strong>out_suffix</strong></td>
<td>String</td>
<td>Suffix of merged tsv files</td>
<td>.tsv</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
<h3 id="workflow-tasks">Workflow Tasks<a class="headerlink" href="#workflow-tasks" title="Permanent link"></a></h3>
<p>For the Augur subcommands, please view the <a href="https://docs.nextstrain.org/projects/augur/en/stable/usage/usage.html">Nextstrain Augur documentation</a> for more details and explanations.</p>
<details class="task">
<summary><code>versioning</code>: Version Capture</summary>
<p>The <code>versioning</code> task captures the workflow version from the GitHub (code repository) version.</p>
<div class="admonition techdetails">
<p class="admonition-title">Version Capture Technical details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/task_versioning.wdl">task_versioning.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>augur align</code>: Genome Assembly Alignment</summary>
<p><code>augur align</code> aligns multiple nucleotide sequences and strips insertions relative to a reference sequence by using the MAFFT aligner. </p>
<div class="admonition techdetails">
<p class="admonition-title"><code>augur align</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/augur/task_augur_align.wdl">task_augur_align.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/augur">Augur on GitHub</a><br><a href="https://github.com/GSLBiotech/mafft">MAFFT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/augur/en/stable/usage/cli/align.html">augur align on Nexstrain</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/bty407">Nextstrain: real-time tracking of pathogen evolution</a><br><a href="https://doi.org/10.1093/bib/bbn013">Recent developments in the MAFFT multiple sequence alignment program</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>snp-dists</code>: Pairwise SNP Distance Calculation</summary>
<p><code>SNP-dists</code> computes pairwise SNP distances between genomes. It takes the same alignment of genomes used to generate your phylogenetic tree and produces a matrix of pairwise SNP distances between sequences. This means that if you generated pairwise core-genome phylogeny, the output will consist of pairwise core-genome SNP (cgSNP) distances. Otherwise, these will be whole-genome SNP distances. Regardless of whether core-genome or whole-genome SNPs, this SNP distance matrix will exclude all SNPs in masked regions (i.e. masked with a bed file or gubbins). </p>
<p>The SNP-distance output can be visualized using software such as <a href="http://jameshadfield.github.io/phandango/#/main">Phandango</a> to explore the relationships between the genomic sequences. The task can optionally add a Phandango coloring tag (:c1) to the column names in the output matrix to ensure that all columns are colored with the same color scheme throughout by setting <code>phandango_coloring</code> to <code>true</code>.</p>
<div class="admonition techdetails">
<p class="admonition-title">SNP-dists Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/utilities/task_snp_dists.wdl">task_snp_dists.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/tseemann/snp-dists">SNP-dists on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/tseemann/snp-dists">SNP-dists on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>augur tree</code>: Phylogenetic Tree Reconstruction</summary>
<p><code>augur tree</code> generates a phylogenetic tree using either <code>iqtree</code>, <code>fasttree</code>, or <code>raxml</code>. By default, the tree will be generated by IQ-TREE 2. FastTree can result in significant performance improvements at the cost of how phylogenetic resolution if the alignment contains 100s of sequences and computational throughput is a priority. Additionally, the task defaults to using a GTR evolutionary model, which can be modified via the <code>substitution_model</code> input.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>augur tree</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/augur/task_augur_tree.wdl">task_augur_tree.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/augur">Augur on GitHub</a><br><a href="https://github.com/morgannprice/fasttree">FastTree on GitHub</a><br><a href="https://github.com/iqtree/iqtree2">IQ-TREE on GitHub</a><br><a href="https://github.com/stamatak/standard-RAxML">RAxML on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/augur/en/stable/usage/cli/tree.html">augur tree on Nexstrain</a><br><a href="https://morgannprice.github.io/fasttree/">FastTree Documentation</a><br><a href="https://iqtree.github.io/doc/">IQ-TREE on GitHub</a><br><a href="https://cme.h-its.org/exelixis/web/software/raxml/">RAxML Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/bty407">Nextstrain: real-time tracking of pathogen evolution</a><br><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0009490">FastTree 2 - Approximately Maximum-Likelihood Trees for Large Alignments</a><br><a href="https://doi.org/10.1093/molbev/msaa015">New models and efficient methods for phylogenetic inference in the genomic era</a><br><a href="https://doi.org/10.1093/bib/bbn013">RAxML Version 8: A tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>reorder_matrix</code>: Matrix Reorder and Phylogenetic Tree Rooting</summary>
<p>Reorder Matrix will reorder a TSV matrix file's entries in the same order as the tips of an inputted phylogenetic tree. The phylogenetic tree can be rooted at the midpoint via the <code>midpoint_root_tree</code> Boolean, or via an outgroup tip via the <code>outgroup_root</code> String input. The provided outgroup tip must be an exact match for a sequence input into the phylogenetic tree. Please note that phylogenetic tree tip names are derived from the alignment FASTA file headers in PHB, and may deviate from <code>samplename</code> inputs.</p>
<div class="admonition techdetails">
<p class="admonition-title">Reorder Matrix Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/utilities/task_reorder_matrix.wdl">task_reorder_matrix.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>augur refine</code>: Phylogenetic Tree Refinement and Time-Calibration</summary>
<p><code>augur refine</code> adds metadata to a phylogenetic tree and can optionally calibrate branch lengths with respect to time using maximum-likelihood via TimeTree. Time-Calibration is triggered by incorporating <code>collection_date</code> in the provided metadata generated by Augur_Prep_PHB.</p>
<div class="admonition techdetails">
<p class="admonition-title">Augur Refine Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/augur/task_augur_refine.wdl">task_augur_refine.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/augur">Augur on GitHub</a><br><a href="https://github.com/neherlab/treetime">TreeTime on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/augur/en/stable/usage/cli/refine.html">augur refine on Nextstrain</a><br><a href="https://treetime.readthedocs.io/en/latest/">TreeTime ReadTheDocs</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/bty407">Nextstrain: real-time tracking of pathogen evolution</a><br><a href="https://academic.oup.com/ve/article/4/1/vex042/4794731?login=false">TreeTime: Maximum-likelihood phylodynamic analysis</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>augur ancestral</code>: Ancestral Nucleotide Sequence Reconstruction</summary>
<p><code>augur ancestral</code> infers ancestral nucleotide sequences based on phylogenetic relatedness using maximum-likelihood via TreeTime. A "joint" maximum likelihood model is used by default, though "marginal" input is permitted. </p>
<p>NOTE: <code>keep-ambiguous</code> and <code>infer-ambiguous</code> are mutually-exclusive, incompatible options that will raise an error if selected together.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>augur ancestral</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/augur/task_augur_ancestral.wdl">task_augur_ancestral.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/augur">Augur on GitHub</a><br><a href="https://github.com/neherlab/treetime">TreeTime on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/augur/en/stable/usage/cli/ancestral.html">augur ancestral on Nexstrain</a><br><a href="https://treetime.readthedocs.io/en/latest/">TreeTime ReadTheDocs</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/bty407">Nextstrain: real-time tracking of pathogen evolution</a><br><a href="https://academic.oup.com/ve/article/4/1/vex042/4794731?login=false">TreeTime: Maximum-likelihood phylodynamic analysis</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>augur translate</code>: Translate Nucleotide Sequences</summary>
<p><code>augur translate</code> translates nucleotide sequences of nodes in a phylogeny to amino acids based on annotated features in a provided <code>reference_genbank</code>. The task will not run without a <code>reference_genbank</code> input.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>augur_translate</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/augur/task_augur_translate.wdl">task_augur_translate.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/augur">Augur on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/augur/en/stable/usage/cli/translate.html">augur translate on Nextstrain</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/bty407">Nextstrain: real-time tracking of pathogen evolution</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>extract mutation context</code>: Contextualizing Mpox Mutation <mark>(only for mpox)</mark></summary>
<p>This task quantifies the number of G/A, C/T, and dinucleotide conversions for mpox samples. These mutations have been shown to be a characteristic of APOBEC3-type editing, which indicate adaptation of the virus to circulation among humans.</p>
<p>The results are formatted into a JSON that is used to contextualize mutation metadata on the Augur phylogeny.</p>
<p>When visualizing the output <code>auspice_input_json</code> file, there will be 2 new choices in the drop-down menu for "Color By":</p>
<ul>
<li>GA or CT fraction</li>
<li>NGA/TCN context of GA or CT mutations.</li>
</ul>
<p>An example Mpox tree with these "Color By" options can be viewed <a href="https://nextstrain.org/mpox/clade-IIb?c=GA_CT_fraction">here</a></p>
<div class="admonition techdetails">
<p class="admonition-title">Extract Mutation Context Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/augur/task_augur_mutation_context.wdl">task_augur_mutation_context.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>augur traits</code>: Ancestral Trait Reconstruction</summary>
<p><code>augur traits</code> will reconstruct the ancestral traits of provided metadata. By default, only the "pango_lineage" and "clade_membership" columns are included, though the <code>augur_traits_columns</code> String inputted can be populated with a comma-delimited string to determine what trait metadata to use.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>augur traits</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/augur/task_augur_traits.wdl">task_augur_traits.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/augur">Augur on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/augur/en/stable/usage/cli/traits.html">augur traits on Nextstrain</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/bty407">Nextstrain: real-time tracking of pathogen evolution</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>extract clade mutations</code>: Extract Clade-Defining Signature Sequences</summary>
<p>Extract Clade Mutations will create an Augur-compatible "clades.tsv" by extracting signature clade-defining sequences. A nucleotide JSON outputted by Augur Ancestral is required, and an optional amino acid JSON outputted by Augur Translate can be used to infer specific amino acid mutations.</p>
<p>Clade-defining signatures can only be extracted from monophyletic clades with unique mutation signatures. If no clade-defining mutations are reported, an error is raised. If the clade metadata column does not exist, then an error is raised as well.</p>
<div class="admonition techdetails">
<p class="admonition-title">Extract Clade Mutations Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/augur/task_extract_clade_mutations.wdl">task_extract_clade_mutations.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/theiagen/utilities">Theiagen Utilities on GitHub</a><br><a href="https://github.com/theiagen/theiaphylo">TheiaPhylo on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>augur clades</code>: Assigning Clades based on Sequence Signatures</summary>
<p><code>augur clades</code> assigns clades to nodes in a tree based on amino-acid or nucleotide signatures. A <code>clades_tsv</code> is required to delineate clade-defining mutations in a tab-delimitted file with the header "clade\tgene\tsite\alt", where the site is an integer site and the alternative sequence character is delineated in the "alt" column. Clade designation preferentially references amino acid sequences if they are provided.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>augur clades</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/augur/task_augur_clades.wdl">task_augur_clades.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/augur">Augur on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/augur/en/stable/usage/cli/clades.html">augur clades on Nexstrain</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/bty407">Nextstrain: real-time tracking of pathogen evolution</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>augur export</code>: Exporting Auspice-formatted Phylogenetic Trees</summary>
<p><code>augur export</code> outputs an Auspice-formatted JSON from an Augur-refined phylogenetic tree and its included metadata. The resulting JSON can be inputted directly into the web-based tree viewer, <a href="https://auspice.us">auspice.us</a>.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>augur export</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/augur/task_augur_export.wdl">task_augur_export.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/augur">Augur on GitHub</a><br><a href="https://github.com/nextstrain/auspice">Auspice on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/augur/en/stable/usage/cli/export.html">augur export on Nextstrain</a><br><a href="https://docs.nextstrain.org/projects/auspice/en/stable/">Auspice on Nextstrain</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/bty407">Nextstrain: real-time tracking of pathogen evolution</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h4 id="augur-outputs">Augur Outputs<a class="headerlink" href="#augur-outputs" title="Permanent link"></a></h4>
<p>The <code>auspice_input_json</code> is intended to be uploaded to&nbsp;<a href="https://auspice.us/">Auspice</a>&nbsp;to view the resulting phylogenetic tree with the provided metadata. Alternatively, a phylogenetic tree in Newick fromat is also available for visualization in other platforms. The <code>metadata_merged</code> output can also be uploaded to either Auspice or a different visualization platform to add further context to the phylogenetic visualization. The <code>combined_assemblies</code> output can be uploaded to&nbsp;<a href="https://genome.ucsc.edu/cgi-bin/hgPhyloPlace">UShER</a>&nbsp;to view the samples on a global tree of representative sequences from public repositories.</p>
<p>The Nextstrain team hosts documentation surrounding the Augur workflow to Auspice visualization here, which details the various components of the Auspice interface: <a href="https://docs.nextstrain.org/en/latest/learn/augur-to-auspice.html">How data is exported by Augur for visualisation in Auspice</a>.</p>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>augur_aligned_fastas</td>
<td>File</td>
<td>Alignment of inputted FASTAs</td>
</tr>
<tr>
<td>augur_auspice_input_json</td>
<td>File</td>
<td>Auspice-formatted JSON of tree and associated metadata</td>
</tr>
<tr>
<td>augur_clade_mutations</td>
<td>File</td>
<td>Clade-specific mutations identified in the analysis</td>
</tr>
<tr>
<td>augur_combined_assemblies</td>
<td>File</td>
<td>Concatenated FASTAs</td>
</tr>
<tr>
<td>augur_distance_tree</td>
<td>File</td>
<td>Phylogenetic tree without time-calibration</td>
</tr>
<tr>
<td>augur_fasttree_version</td>
<td>String</td>
<td>The fasttree version used, blank if other tree method used</td>
</tr>
<tr>
<td>augur_iqtree_model_used</td>
<td>String</td>
<td>The iqtree model used during augur tree, blank if iqtree not used</td>
</tr>
<tr>
<td>augur_iqtree_version</td>
<td>String</td>
<td>The iqtree version used during augur tree (defualt), blank if other tree method used</td>
</tr>
<tr>
<td>augur_keep_list</td>
<td>File</td>
<td>List of samples that were kept and met length filters</td>
</tr>
<tr>
<td>augur_mafft_version</td>
<td>String</td>
<td>The mafft version used in augur align</td>
</tr>
<tr>
<td>augur_metadata_merged</td>
<td>File</td>
<td>TSV file of merged metadata</td>
</tr>
<tr>
<td>augur_phb_analysis_date</td>
<td>String</td>
<td>The date the analysis was run</td>
</tr>
<tr>
<td>augur_phb_version</td>
<td>String</td>
<td>The version of the Public Health Bioinformatics (PHB) repository used</td>
</tr>
<tr>
<td>augur_raxml_version</td>
<td>String</td>
<td>The version of raxml used during augur tree, blank if other tree method used</td>
</tr>
<tr>
<td>augur_snp_matrix</td>
<td>File</td>
<td>SNP matrix generated from alignment</td>
</tr>
<tr>
<td>augur_time_tree</td>
<td>File</td>
<td>Time-calibrated phylogenetic tree</td>
</tr>
<tr>
<td>augur_traits_json</td>
<td>File</td>
<td>JSON of traits used for applying metadata to the phylogenetic tree</td>
</tr>
<tr>
<td>augur_version</td>
<td>String</td>
<td>Version of Augur used</td>
</tr>
</tbody>
</table>
</div>
<h3 id="references">References<a class="headerlink" href="#references" title="Permanent link"></a></h3>
<p>When publishing work using the Augur_PHB workflow, please reference the following:</p>
<blockquote>
<p>Nextstrain:&nbsp;Hadfield J, Megill C, Bell SM, Huddleston J, Potter B, Callender C, Sagulenko P, Bedford T, Neher RA. Nextstrain: real-time tracking of pathogen evolution. Bioinformatics. 2018 Dec 1;34(23):4121-3.</p>
</blockquote>







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="November 11, 2025 18:51:28 UTC">2025-11-11</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="September 11, 2024 17:48:55 UTC">2024-09-11</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
       2022-2025 <a href="https://www.theiagen.com" target="_blank" rel="noopener">Theiagen Genomics</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/theiagen" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://bsky.app/profile/theiagen.bsky.social" target="_blank" rel="noopener" title="bsky.app" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M407.8 294.7c-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3M288 227.1c-26.1-50.7-97.1-145.2-163.1-191.8C61.6-9.4 37.5-1.7 21.6 5.5 3.3 13.8 0 41.9 0 58.4S9.1 194 15 213.9c19.5 65.7 89.1 87.9 153.2 80.7 3.3-.5 6.6-.9 10-1.4-3.3.5-6.6 1-10 1.4-93.9 14-177.3 48.2-67.9 169.9C220.6 589.1 265.1 437.8 288 361.1c22.9 76.7 49.2 222.5 185.6 103.4 102.4-103.4 28.1-156-65.8-169.9-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3 64.1 7.1 133.6-15.1 153.2-80.7C566.9 194 576 75 576 58.4s-3.3-44.7-21.6-52.9c-15.8-7.1-40-14.9-103.2 29.8C385.1 81.9 314.1 176.4 288 227.1"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.linkedin.com/company/theiagen" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3M135.4 416H69V202.2h66.5V416zM102.2 96a38.5 38.5 0 1 1 0 77 38.5 38.5 0 1 1 0-77m282.1 320h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9z"></path></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "../../..", "features": ["content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "navigation.instant", "navigation.path", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.suggest", "toc.follow"], "search": "../../../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"alias": true, "default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.79ae519e.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../../../javascripts/tablesort.js"></script>
      
        <script src="../../../javascripts/table-search.js"></script>
      
    
  
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>