<!DOCTYPE html><html lang="en" class="no-js"><head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A collection of bioinformatics workflows for genomic characterization, submission preparation, and genomic epidemiology of pathogens of public health concern.">
      
      
        <meta name="author" content="Theiagen Genomics">
      
      
        <link rel="canonical" href="https://theiagen.github.io/public_health_bioinformatics/main/guides/versions/">
      
      
        <link rel="prev" href="../sops/">
      
      
        <link rel="next" href="../gambit/">
      
      
      <link rel="icon" href="../../assets/logos/Theiagen-Symbol-Standard-01.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.21">
    
    
      
        <title>Database and Docker Versions - Public Health Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.2a3383ac.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#database-and-docker-versions" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version release.
  <a href="../../.."> 
    <strong>Click here to go to the latest version release.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-header__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Public Health Bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Database and Docker Versions
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo" aria-label="Switch to Dark Mode" type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to Dark Mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"></path></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo" aria-label="Switch to Light Mode" type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to Light Mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m3.55 19.09 1.41 1.41 1.8-1.79-1.42-1.42M12 6c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6c0-3.32-2.69-6-6-6m8 7h3v-2h-3m-2.76 7.71 1.8 1.79 1.41-1.41-1.79-1.8M20.45 5l-1.41-1.4-1.8 1.79 1.42 1.42M13 1h-2v3h2M6.76 5.39 4.96 3.6 3.55 5l1.79 1.81zM1 13h3v-2H1m12 9h-2v3h2"></path></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../.." class="md-tabs__link">
          
  
  
  Public Health Bioinformatics

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-nav__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    Public Health Bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Public Health Bioinformatics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Public Health Bioinformatics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Getting Started
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            Getting Started
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting_started/commandline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With the Command-Line
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting_started/terra/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With Terra.bio
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting_started/philosophy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Workflow Failure Philosophy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3" checked>
        
          
          <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Other Resources
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_3">
            <span class="md-nav__icon md-icon"></span>
            Other Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sops/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Available SOPs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Database and Docker Versions
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Database and Docker Versions
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#default-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Default Databases
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#default-docker-images" class="md-nav__link">
    <span class="md-ellipsis">
      Default Docker Images
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#default-references" class="md-nav__link">
    <span class="md-ellipsis">
      Default References
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../gambit/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../gambit_database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT Database Creation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../phylogenetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Phylogenetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../custom_organisms/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Running Custom Organisms on TheiaCoV
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1">
        
          
          <label class="md-nav__link" for="__nav_1_5_1" id="__nav_1_5_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Type
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Type
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows_overview/workflows_type/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_2" id="__nav_1_5_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Import
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_2">
            <span class="md-nav__icon md-icon"></span>
            Data Import
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_3" id="__nav_1_5_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Genomic Characterization
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_3">
            <span class="md-nav__icon md-icon"></span>
            Genomic Characterization
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_4" id="__nav_1_5_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Construction
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_4">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Construction
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_5" id="__nav_1_5_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Placement
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_5">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Placement
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_6" id="__nav_1_5_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Public Data Sharing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_6">
            <span class="md-nav__icon md-icon"></span>
            Public Data Sharing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_7">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_7" id="__nav_1_5_1_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exporting Data from Terra
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_7">
            <span class="md-nav__icon md-icon"></span>
            Exporting Data from Terra
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_8">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_8" id="__nav_1_5_1_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Standalone
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_8">
            <span class="md-nav__icon md-icon"></span>
            Standalone
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/chroquetas/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ChroQueTas
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2" id="__nav_1_5_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Kingdom
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Kingdom
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows_overview/workflows_kingdom/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_2" id="__nav_1_5_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Any Taxa
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_2">
            <span class="md-nav__icon md-icon"></span>
            Any Taxa
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_3" id="__nav_1_5_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Bacteria
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_3">
            <span class="md-nav__icon md-icon"></span>
            Bacteria
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_4" id="__nav_1_5_2_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Mycotics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_4">
            <span class="md-nav__icon md-icon"></span>
            Mycotics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/chroquetas/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ChroQueTas
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_5" id="__nav_1_5_2_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Viral
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_5">
            <span class="md-nav__icon md-icon"></span>
            Viral
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows Alphabetically
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            Workflows Alphabetically
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows_overview/workflows_alphabetically/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/chroquetas/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ChroQueTas
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Contributing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            Contributing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/doc_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Documentation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/code_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Code
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#default-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Default Databases
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#default-docker-images" class="md-nav__link">
    <span class="md-ellipsis">
      Default Docker Images
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#default-references" class="md-nav__link">
    <span class="md-ellipsis">
      Default References
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="database-and-docker-versions">Database and Docker Versions<a class="headerlink" href="#database-and-docker-versions" title="Permanent link"></a></h1>
<p>Throughout this repository, we have carefully selected a number of default databases and Docker images that are intended to be sufficient starting points for most analyses. The following tables describe the default values for all databases, Docker images, and reference files used throughout the suite of PHB workflows.</p>
<h2 id="default-databases">Default Databases<a class="headerlink" href="#default-databases" title="Permanent link"></a></h2>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Workflow</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>checkv_consensus</td>
<td><strong>checkv_db</strong></td>
<td>File</td>
<td>CheckV database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/checkv/checkv-db-v1.5.tar.gz</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>checkv_db</strong></td>
<td>File</td>
<td>CheckV database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/checkv/checkv-db-v1.5.tar.gz</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>gambit_db_genomes</strong></td>
<td>File</td>
<td>Database of metadata for assembled query genomes; requires complementary signatures file. If not provided, uses default database "/gambit-db"</td>
<td>gs://gambit-databases-rp/2.1.0/gambit-metadata-2.1.0-20250808.gdb</td>
<td>Gambit_Query, TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>gambit_db_signatures</strong></td>
<td>File</td>
<td>Signatures file; requires complementary genomes file. If not specified, the file from the docker container will be used.</td>
<td>gs://gambit-databases-rp/2.1.0/gambit-signatures-2.1.0-20250808.gs</td>
<td>Gambit_Query, TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>gamma_db</strong></td>
<td>File</td>
<td>Multifasta sequence database GAMMA is to use for gene matching</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/gamma/default_ResFinderDB_Combined_05-06-20.fsa</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>kmerfinder_db</strong></td>
<td>File</td>
<td>Bacterial database for KmerFinder</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kmerfinder/kmerfinder_bacteria_20230911.tar.gz</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>kraken2</td>
<td><strong>kraken2_db</strong></td>
<td>File</td>
<td>The database used to run Kraken2</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>NCBI_Scrub_PE, NCBI_Scrub_SE</td>
</tr>
<tr>
<td>kraken2_dehosted</td>
<td><strong>kraken2_db</strong></td>
<td>File</td>
<td>The database used to run Kraken2. Must contain viral and human sequences.</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>TheiaCoV_ClearLabs</td>
</tr>
<tr>
<td>kraken2_raw</td>
<td><strong>kraken2_db</strong></td>
<td>File</td>
<td>The database used to run Kraken2. Must contain viral and human sequences.</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>TheiaCoV_ClearLabs</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_clusters.csv</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_npy</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.dists.npy</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_dists_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.dists.pkl</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_external_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_external_clusters.csv</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_npz</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_fit.npz</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_fit_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_fit.pkl</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_graph_gt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_graph.gt</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_h5</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.h5</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_qcreport_txt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_qcreport.txt</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_npy</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.dists.npy</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_dists_pkl</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.dists.pkl</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_graph_gt</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6refs_graph.gt</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_refs_h5</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6.refs.h5</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_gps_unword_clusters_csv</strong></td>
<td>File</td>
<td>Poppunk database file *Provide an empty or local file if running TheiaProk on the command-line</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/poppunk/GPS_v6/GPS_v6_unword_clusters.csv</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_database</strong></td>
<td>String</td>
<td>The specific database to use</td>
<td>virulence_ecoli</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>metabuli_db</strong></td>
<td>File</td>
<td>Metabuli database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/metabuli/refseq_virus-v223.tar.gz</td>
<td>TheiaViral_ONT</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database. Must contain human and viral sequences</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>The database used by the MIDAS task in .tar.gz format</td>
<td>gs://theiagen-public-files-rp/terra/theiaprok-files/midas/midas_db_v1.2.tar.gz</td>
<td>TheiaMeta_Illumina_PE, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>referenceseeker</td>
<td><strong>referenceseeker_db</strong></td>
<td>File</td>
<td>Database used by the referenceseeker tool that contains bacterial genomes from RefSeq release 205. Downloaded from the referenceseeker GitHub repository.</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/referenceseeker/referenceseeker-bacteria-refseq-205.v20210406.tar.gz</td>
<td>Assembly_Fetch, Snippy_Streamline, Snippy_Streamline_FASTA</td>
</tr>
<tr>
<td>skani</td>
<td><strong>skani_db</strong></td>
<td>File</td>
<td>Skani database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/skani/skani_db_20250613.tar</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>gambit_db_genomes</strong></td>
<td>File</td>
<td>User-provided database of assembled query genomes; requires complementary signatures file. If not provided, uses default database, "/gambit-db"</td>
<td>gs://gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-metadata-1.0.0-20241213.gdb</td>
<td>TheiaEuk_Illumina_PE</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>gambit_db_signatures</strong></td>
<td>File</td>
<td>User-provided signatures file; requires complementary genomes file. If not specified, the file from the docker container will be used.</td>
<td>gs://gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-signatures-1.0.0-20241213.gs</td>
<td>TheiaEuk_Illumina_PE</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>gambit_db_genomes</strong></td>
<td>File</td>
<td>User-provided database of assembled query genomes; requires complementary signatures file. If not provided, uses default database, "/gambit-db"</td>
<td>gs://gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-metadata-1.0.0-20241213.gdb</td>
<td>TheiaEuk_ONT</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>gambit_db_signatures</strong></td>
<td>File</td>
<td>User-provided signatures file; requires complementary genomes file. If not specified, the file from the docker container will be used.</td>
<td>gs://gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-signatures-1.0.0-20241213.gs</td>
<td>TheiaEuk_ONT</td>
</tr>
<tr>
<td>theiameta_illumina_pe</td>
<td><strong>kraken2_db</strong></td>
<td>File</td>
<td>A Kraken2 database in .tar.gz format</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/k2_standard_08gb_20230605.tar.gz</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>abricate_db</strong></td>
<td>String</td>
<td>Database to use with the Abricate tool. Options: NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB</td>
<td>vfdb</td>
<td>TheiaProk_FASTA</td>
</tr>
<tr>
<td>theiaprok_fasta</td>
<td><strong>bakta_db</strong></td>
<td>String</td>
<td>Database selection for Bakta annotation. Options: light (smaller, faster), full (more comprehensive), or a Google Storage URI (gs://...) pointing to a custom Bakta database archive (.tar.gz). The selected database will be extracted before annotation.</td>
<td>full</td>
<td>TheiaProk_FASTA</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>abricate_db</strong></td>
<td>String</td>
<td>Database to use with the Abricate tool. Options: NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB</td>
<td>vfdb</td>
<td>TheiaProk_Illumina_PE</td>
</tr>
<tr>
<td>theiaprok_illumina_pe</td>
<td><strong>bakta_db</strong></td>
<td>String</td>
<td>Database selection for Bakta annotation. Options: light (smaller, faster), full (more comprehensive), or a Google Storage URI (gs://...) pointing to a custom Bakta database archive (.tar.gz). The selected database will be extracted before annotation.</td>
<td>full</td>
<td>TheiaProk_Illumina_PE</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>abricate_db</strong></td>
<td>String</td>
<td>Database to use with the Abricate tool. Options: NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB</td>
<td>vfdb</td>
<td>TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>theiaprok_illumina_se</td>
<td><strong>bakta_db</strong></td>
<td>String</td>
<td>Database selection for Bakta annotation. Options: light (smaller, faster), full (more comprehensive), or a Google Storage URI (gs://...) pointing to a custom Bakta database archive (.tar.gz). The selected database will be extracted before annotation.</td>
<td>full</td>
<td>TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>abricate_db</strong></td>
<td>String</td>
<td>Database to use with the Abricate tool. Options: NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB</td>
<td>vfdb</td>
<td>TheiaProk_ONT</td>
</tr>
<tr>
<td>theiaprok_ont</td>
<td><strong>bakta_db</strong></td>
<td>String</td>
<td>Database selection for Bakta annotation. Options: light (smaller, faster), full (more comprehensive), or a Google Storage URI (gs://...) pointing to a custom Bakta database archive (.tar.gz). The selected database will be extracted before annotation.</td>
<td>full</td>
<td>TheiaProk_ONT</td>
</tr>
<tr>
<td>theiaviral_illumina_pe</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>Kraken2 database file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>TheiaViral_Illumina_PE</td>
</tr>
<tr>
<td>vadr_update</td>
<td><strong>vadr_model_file</strong></td>
<td>File</td>
<td>Path to the a tar + gzipped VADR model file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-sarscov2-1.3-2.tar.gz</td>
<td>VADR_Update</td>
</tr>
</tbody>
</table>
</div>
<h2 id="default-docker-images">Default Docker Images<a class="headerlink" href="#default-docker-images" title="Permanent link"></a></h2>
<p>All Docker variables are <strong>String</strong> types with the description: "The Docker container to use for the task".</p>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Workflow</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>abricate</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>add_biosample_accessions</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Terra_2_NCBI</td>
</tr>
<tr>
<td>amr_search</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/amrsearch:0.2.1</td>
<td>AMR_Search</td>
</tr>
<tr>
<td>amrfinderplus_nuc</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-amrfinderplus:4.0.23-2025-07-16.1</td>
<td>AMRFinderPlus</td>
</tr>
<tr>
<td>amrfinderplus_task</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-amrfinderplus:4.0.23-2025-07-16.1</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>ani</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mummer:4.0.0-rgdv2</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>arln_stats</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/arln_stats:1.0.0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>assembled_reads_percent</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>augur_align</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/augur:22.0.2--pyhdfd78af_0</td>
<td>Augur</td>
</tr>
<tr>
<td>augur_ancestral</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/augur:22.0.2--pyhdfd78af_0</td>
<td>Augur</td>
</tr>
<tr>
<td>augur_clades</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/augur:22.0.2--pyhdfd78af_0</td>
<td>Augur</td>
</tr>
<tr>
<td>augur_export</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/augur:22.0.2--pyhdfd78af_0</td>
<td>Augur</td>
</tr>
<tr>
<td>augur_refine</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/augur:22.0.2--pyhdfd78af_0</td>
<td>Augur</td>
</tr>
<tr>
<td>augur_traits</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/augur:22.0.2--pyhdfd78af_0</td>
<td>Augur</td>
</tr>
<tr>
<td>augur_translate</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/augur:22.0.2--pyhdfd78af_0</td>
<td>Augur</td>
</tr>
<tr>
<td>augur_tree</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/augur:22.0.2--pyhdfd78af_0</td>
<td>Augur</td>
</tr>
<tr>
<td>bakta</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/bakta:1.10.3</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>bam_to_unaligned_fastq</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>Host_Decontaminate</td>
</tr>
<tr>
<td>bcftools_consensus</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/bcftools:1.20</td>
<td>TheiaViral_ONT</td>
</tr>
<tr>
<td>biosample_submit_tsv_ftp_upload</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/broadinstitute/ncbi-tools:2.10.7.10</td>
<td>Terra_2_NCBI</td>
</tr>
<tr>
<td>busco</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/ezlabgva/busco:v5.7.1_cv1</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Freyja_FASTQ, TheiaMeta_Illumina_PE, TheiaViral_Illumina_PE</td>
</tr>
<tr>
<td>calculate_coverage</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/bedtools:2.31.0</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>calculate_coverage_paf</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/quay/ubuntu:latest</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>cat_files</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Augur, Concatenate_Column_Content</td>
</tr>
<tr>
<td>cat_files_fasta</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>TheiaCoV_FASTA_Batch</td>
</tr>
<tr>
<td>cat_lanes</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.2</td>
<td>Concatenate_Illumina_Lanes</td>
</tr>
<tr>
<td>cat_ont_barcodes</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/ont-barcodes:0.0.2</td>
<td>ONT_Barcode_Concatenation</td>
</tr>
<tr>
<td>cat_variants</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Find_Shared_Variants</td>
</tr>
<tr>
<td>centroid</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/centroid:0.1.0</td>
<td>Snippy_Streamline, Snippy_Streamline_FASTA</td>
</tr>
<tr>
<td>cg_pipeline_clean</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/lyveset:1.1.4f</td>
<td>TheiaEuk_Illumina_PE, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>cg_pipeline_raw</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/lyveset:1.1.4f</td>
<td>TheiaEuk_Illumina_PE, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>cg_reorder_matrix</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>Snippy_Tree</td>
</tr>
<tr>
<td>checkv_consensus</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/checkv:1.0.3</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>checkv_denovo</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/checkv:1.0.3</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>chroquetas</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/chroquetas:1.0.0</td>
<td>ChroQueTas</td>
</tr>
<tr>
<td>chunk_files</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Dorado_Basecalling</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>clair3</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/clair3-extra-models:1.0.10</td>
<td>TheiaViral_ONT</td>
</tr>
<tr>
<td>clair3_variants_ont</td>
<td><strong>clair3_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/clair3:1.0.10</td>
<td>Clair3_Variants_ONT</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT, TheiaEuk_Illumina_PE, TheiaEuk_ONT, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT, TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>compare_two_tsvs</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/theiavalidate:1.1.2</td>
<td>TheiaValidate</td>
</tr>
<tr>
<td>concatenate_qc_metrics</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Snippy_Tree</td>
</tr>
<tr>
<td>concatenate_variants</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Snippy_Tree</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/artic-ncov2019-epi2me</td>
<td>TheiaCoV_ONT</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>TheiaViral_Illumina_PE</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>TheiaCoV_ClearLabs, TheiaCoV_FASTA, TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT, TheiaViral_Illumina_PE, TheiaViral_ONT, VADR_Update</td>
</tr>
<tr>
<td>core_iqtree</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/iqtree:1.6.7</td>
<td>Core_Gene_SNP</td>
</tr>
<tr>
<td>core_reorder_matrix</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>Core_Gene_SNP, kSNP3, kSNP4</td>
</tr>
<tr>
<td>core_snp_dists</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snp-dists:0.8.2</td>
<td>Core_Gene_SNP, kSNP3, kSNP4</td>
</tr>
<tr>
<td>core_snp_sites</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snp-sites:2.5.1</td>
<td>Core_Gene_SNP</td>
</tr>
<tr>
<td>create_table_from_array</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>Dorado_Basecalling</td>
</tr>
<tr>
<td>create_terra_table</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>ONT_Barcode_Concatenation</td>
</tr>
<tr>
<td>czgenepi_wrangling</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-08-08-2</td>
<td>CZGenEpi_Prep</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>bwa_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>TheiaEuk_Illumina_PE, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>filter_contigs_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/shovilter:0.2</td>
<td>TheiaEuk_Illumina_PE, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>megahit_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/megahit:1.2.9</td>
<td>TheiaEuk_Illumina_PE, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>pilon_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/pilon:1.24--hdfd78af_0</td>
<td>TheiaEuk_Illumina_PE, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>skesa_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/skesa:2.4.0</td>
<td>TheiaEuk_Illumina_PE, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>digger_denovo</td>
<td><strong>spades_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/spades:4.1.0</td>
<td>TheiaEuk_Illumina_PE, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE</td>
</tr>
<tr>
<td>dorado_basecall</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/dorado:0.9.0-cuda12.2.0</td>
<td>Dorado_Basecalling</td>
</tr>
<tr>
<td>dorado_demux</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/dorado:0.9.0-cuda12.2.0</td>
<td>Dorado_Basecalling</td>
</tr>
<tr>
<td>dorado_trim</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/dorado:0.8.3</td>
<td>Dorado_Basecalling</td>
</tr>
<tr>
<td>download_accession</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-datasets:16.38.1</td>
<td>Host_Decontaminate</td>
</tr>
<tr>
<td>download_terra_table</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>CZGenEpi_Prep, Mercury_Prep_N_Batch, Terra_2_ENA</td>
</tr>
<tr>
<td>export_taxon_table</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>export_two_tsvs</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>TheiaValidate</td>
</tr>
<tr>
<td>fasta_to_ids</td>
<td><strong>docker</strong></td>
<td>ubuntu</td>
<td>Augur</td>
</tr>
<tr>
<td>fasta_utilities</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>TheiaViral_ONT</td>
</tr>
<tr>
<td>fastq_scan_clean_reads</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/fastq-scan:1.0.1--h4ac6f70_3</td>
<td>TheiaCoV_ClearLabs</td>
</tr>
<tr>
<td>fastq_scan_raw_reads</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/fastq-scan:1.0.1--h4ac6f70_3</td>
<td>TheiaCoV_ClearLabs</td>
</tr>
<tr>
<td>fetch_bs</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/basespace_cli:1.2.1</td>
<td>BaseSpace_Fetch</td>
</tr>
<tr>
<td>fetch_sra_to_fastq</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/fastq-dl:2.0.4--pyhdfd78af_0</td>
<td>SRA_Fetch</td>
</tr>
<tr>
<td>fetch_srr</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/fastq-dl:2.0.4--pyhdfd78af_0</td>
<td>Fetch_SRR_Accession</td>
</tr>
<tr>
<td>filter_sequences_by_length</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/broadinstitute/viral-core:2.1.33</td>
<td>Augur</td>
</tr>
<tr>
<td>find_files</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/cloudsdktool/google-cloud-cli:427.0.0-alpine</td>
<td>Dorado_Basecalling</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-insaflu-220727</td>
<td>TheiaCoV_FASTA, TheiaCoV_Illumina_PE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/genoflu:1.06</td>
<td>TheiaCoV_FASTA, TheiaCoV_Illumina_PE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/irma:1.2.0</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.14.5</td>
<td>TheiaCoV_FASTA, TheiaCoV_Illumina_PE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>TheiaCoV_FASTA, TheiaCoV_Illumina_PE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>flye</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/flye:2.9.4</td>
<td>TheiaViral_ONT</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/freyja:1.5.3</td>
<td>Freyja_FASTQ</td>
</tr>
<tr>
<td>freyja_dashboard_task</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/freyja:1.5.3</td>
<td>Freyja_Dashboard</td>
</tr>
<tr>
<td>freyja_plot_task</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/freyja:1.5.3</td>
<td>Freyja_Plot</td>
</tr>
<tr>
<td>freyja_update_refs</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/freyja:1.5.3</td>
<td>Freyja_Update</td>
</tr>
<tr>
<td>gambit</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0</td>
<td>Gambit_Query, TheiaEuk_Illumina_PE, TheiaEuk_ONT, TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>gamma</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/gamma:2.2</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>TheiaCoV_ClearLabs, TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/seqkit:2.4.0--h9ee0642_0</td>
<td>Freyja_FASTQ</td>
</tr>
<tr>
<td>gisaid_upload</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/broadinstitute/gisaid-cli:3.0</td>
<td>Terra_2_GISAID</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_bwa_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_consensus_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_trim_primers_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_variant_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_primtrim_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE</td>
</tr>
<tr>
<td>ivar_variants</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>TheiaViral_Illumina_PE</td>
</tr>
<tr>
<td>kmerfinder</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/kmerfinder:3.0.2--hdfd78af_0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>kraken2</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>NCBI_Scrub_PE, NCBI_Scrub_SE</td>
</tr>
<tr>
<td>kraken2_clean</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>kraken2_dehosted</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>TheiaCoV_ClearLabs</td>
</tr>
<tr>
<td>kraken2_pe</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>Kraken_PE</td>
</tr>
<tr>
<td>kraken2_raw</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>kraken2_raw</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>TheiaCoV_ClearLabs</td>
</tr>
<tr>
<td>kraken2_recalculate_abundances</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-08-28-v4</td>
<td>Kraken_ONT</td>
</tr>
<tr>
<td>kraken2_se</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>Kraken_ONT, Kraken_SE</td>
</tr>
<tr>
<td>krona</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/krona:2.8.1</td>
<td>Kraken_PE, Kraken_SE</td>
</tr>
<tr>
<td>krona_clean</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/krona:2.8.1</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>krona_raw</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/krona:2.8.1</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>ksnp3_task</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ksnp3:3.1</td>
<td>kSNP3</td>
</tr>
<tr>
<td>ksnp4_task</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ksnp4:4.1</td>
<td>kSNP4</td>
</tr>
<tr>
<td>lyveset</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/lyveset:1.1.4f</td>
<td>Lyve_SET</td>
</tr>
<tr>
<td>make_table</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>Create_Terra_Table</td>
</tr>
<tr>
<td>mashtree_task</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mashtree:1.2.0</td>
<td>MashTree_FASTA</td>
</tr>
<tr>
<td>mask_low_coverage</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/bedtools:2.31.0</td>
<td>TheiaViral_ONT</td>
</tr>
<tr>
<td>megahit</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/megahit:1.2.9</td>
<td>TheiaViral_Illumina_PE</td>
</tr>
<tr>
<td>mercury</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/mercury:1.1.3</td>
<td>Mercury_Prep_N_Batch</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_abaum_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>abricate_vibrio_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-abaum-plasmid</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>agrvate_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/agrvate:1.0.2--hdfd78af_0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>amr_search_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/amrsearch:0.2.1</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT, TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>chroquetas_docker_image</strong></td>
<td></td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT, TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>clockwork_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/cdcgov/varpipe_wgs_with_refs:2bc7234074bd53d9e92a1048b0485763cd9bbf6f4d12d5a1cc82bfec8ca7d75e</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ectyper_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/ectyper:1.0.0--pyhdfd78af_1</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtyper_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/emmtyper:0.2.0--py_0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>emmtypingtool_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/emmtypingtool:0.0.1</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>genotyphi_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.11.0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>hicap_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/hicap:1.0.3--py_0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kaptive_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kaptive:2.0.3</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>kleborate_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kleborate:2.2.0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>legsta_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/legsta:0.5.1--hdfd78af_2</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>lissero_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/lissero:0.4.9--py_0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>meningotype_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/meningotype:0.8.5--pyhdfd78af_0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>ngmaster_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ngmaster:1.0.0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pasty_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pasty:1.0.3</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>pbptyper_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pbptyper:1.0.4</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>poppunk_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/poppunk:2.4.0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seqsero2_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/seqsero2:1.2.1</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>seroba_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/seroba:1.0.2</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>serotypefinder_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/serotypefinder:2.0.1</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigatyper_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/shigatyper:2.0.5</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>shigeifinder_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/shigeifinder:1.3.5</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sistr_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/sistr_cmd:1.1.1--pyh864c0ab_2</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_gene_query_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>snippy_variants_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snippy:4.6.0</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>sonneityping_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>spatyper_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/spatyper:0.3.3--pyhdfd78af_3</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>srst2_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/srst2:0.2.0-vcholerae</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>staphopia_sccmec_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/staphopia-sccmec:1.0.0--hdfd78af_0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>stxtyper_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/stxtyper:1.0.42</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbp_parser_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/tbp-parser:2.6.0</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>tbprofiler_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/tbprofiler:6.6.3</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>vibecheck_docker_image</strong></td>
<td>watronfire/vibecheck:2025.02.24</td>
<td>TheiaProk_Illumina_PE</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>virulencefinder_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/virulencefinder:2.0.4</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>metabuli</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/metabuli:1.1.0</td>
<td>TheiaViral_ONT</td>
</tr>
<tr>
<td>metaspades_pe</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/spades:3.12.0--h9ee0642_3</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22</td>
<td>Clair3_Variants_ONT, Freyja_FASTQ, TheiaViral_ONT</td>
</tr>
<tr>
<td>minimap2_assembly</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>minimap2_assembly_correction</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>minimap2_ont</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22</td>
<td>Host_Decontaminate</td>
</tr>
<tr>
<td>minimap2_pe</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22</td>
<td>Host_Decontaminate</td>
</tr>
<tr>
<td>minimap2_reads</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>abricate_flu_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-insaflu-220727</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>assembly_metrics_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>genoflu_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/genoflu:1.06</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>irma_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/irma:1.2.0</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.10.2</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>nextclade_output_parser_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>morgana_magic</td>
<td><strong>pangolin_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.10.2</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>mutation_context</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/nextstrain-mpox-mutation-context:2024-06-27</td>
<td>Augur</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Freyja_FASTQ, TheiaCoV_ONT, TheiaEuk_ONT, TheiaProk_ONT, TheiaViral_ONT</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Freyja_FASTQ, TheiaCoV_ONT, TheiaEuk_ONT, TheiaProk_ONT, TheiaViral_ONT</td>
</tr>
<tr>
<td>nanoq</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/nanoq:0.9.0--hec16e2b_1</td>
<td>TheiaViral_ONT</td>
</tr>
<tr>
<td>ncbi_datasets</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-datasets:16.38.1</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>ncbi_datasets_download_genome_accession</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-datasets:16.38.1</td>
<td>Assembly_Fetch, Snippy_Streamline, Snippy_Streamline_FASTA</td>
</tr>
<tr>
<td>ncbi_identify</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-datasets:16.38.1</td>
<td>Host_Decontaminate, TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>ncbi_scrub_pe</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>NCBI_Scrub_PE</td>
</tr>
<tr>
<td>ncbi_scrub_se</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>NCBI_Scrub_SE, TheiaCoV_ClearLabs, TheiaViral_ONT</td>
</tr>
<tr>
<td>ncbi_sftp_upload</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/broadinstitute/ncbi-tools:2.10.7.10</td>
<td>Terra_2_NCBI</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>TheiaCoV_ClearLabs, TheiaCoV_FASTA, TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.16.0</td>
<td>TheiaCoV_ClearLabs, TheiaCoV_FASTA, TheiaCoV_FASTA_Batch, TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>nextclade_v3_set</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.14.5</td>
<td>Nextclade_Batch</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>pangolin_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>Augur, TheiaCoV_ClearLabs, TheiaCoV_FASTA</td>
</tr>
<tr>
<td>pan_iqtree</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/iqtree:1.6.7</td>
<td>Core_Gene_SNP</td>
</tr>
<tr>
<td>pan_reorder_matrix</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>Core_Gene_SNP, kSNP3, kSNP4</td>
</tr>
<tr>
<td>pan_snp_dists</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snp-dists:0.8.2</td>
<td>Core_Gene_SNP, kSNP3, kSNP4</td>
</tr>
<tr>
<td>pangolin_update_log</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Pangolin_Update</td>
</tr>
<tr>
<td>parse_mapping</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>Host_Decontaminate, TheiaViral_ONT</td>
</tr>
<tr>
<td>phylovalidate_task</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/theiaphylo:0.1.8</td>
<td>PhyloCompare</td>
</tr>
<tr>
<td>pilon</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/pilon:1.24--hdfd78af_0</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>pirate</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/pirate:1.0.5--hdfd78af_0</td>
<td>Core_Gene_SNP</td>
</tr>
<tr>
<td>plasmidfinder</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/plasmidfinder:2.1.6</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>porechop</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/porechop:0.2.4</td>
<td>TheiaViral_ONT</td>
</tr>
<tr>
<td>prep_augur_metadata</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Augur_Prep</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Freyja_FASTQ</td>
</tr>
<tr>
<td>prokka</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/prokka:1.14.5</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>prune_table</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Terra_2_NCBI</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>TheiaCoV_ClearLabs, TheiaCoV_FASTA, TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT, TheiaEuk_Illumina_PE, TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>quasitools_illumina_pe</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/quasitools:0.7.0--pyh864c0ab_1</td>
<td>TheiaCoV_Illumina_PE</td>
</tr>
<tr>
<td>quasitools_ont</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/quasitools:0.7.0--pyh864c0ab_1</td>
<td>TheiaCoV_ONT</td>
</tr>
<tr>
<td>quast</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT, TheiaMeta_Illumina_PE, TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>quast_denovo</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>rasusa</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/rasusa:2.1.0</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>rasusa_task</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/rasusa:2.1.0</td>
<td>RASUSA, TheiaEuk_Illumina_PE</td>
</tr>
<tr>
<td>raven</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/raven:1.8.3</td>
<td>TheiaViral_ONT</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT, TheiaEuk_Illumina_PE, TheiaEuk_ONT, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/artic-ncov2019:1.3.0-medaka-1.4.3</td>
<td>TheiaCoV_ONT</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-08-28-v4</td>
<td>TheiaCoV_ONT, TheiaProk_ONT</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>TheiaCoV_ONT, TheiaProk_ONT</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/nanoq:0.9.0--hec16e2b_1</td>
<td>TheiaCoV_ONT, TheiaEuk_ONT, TheiaProk_ONT</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>TheiaCoV_ONT</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/rasusa:2.1.0</td>
<td>TheiaProk_ONT</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_docker</strong></td>
<td></td>
<td>Freyja_FASTQ</td>
</tr>
<tr>
<td>read_mapping_stats</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Host_Decontaminate, TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>referenceseeker</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/referenceseeker:1.8.0--pyhdfd78af_0</td>
<td>Assembly_Fetch, Snippy_Streamline, Snippy_Streamline_FASTA</td>
</tr>
<tr>
<td>register_ena_samples</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra_to_ena:0.6</td>
<td>Terra_2_ENA</td>
</tr>
<tr>
<td>rename_PE_files</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/ubuntu/ubuntu:jammy-20230816</td>
<td>Rename_FASTQ</td>
</tr>
<tr>
<td>rename_SE_files</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/ubuntu/ubuntu:jammy-20230816</td>
<td>Rename_FASTQ</td>
</tr>
<tr>
<td>reorder_matrix</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>Augur, MashTree_FASTA</td>
</tr>
<tr>
<td>resfinder_task</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/resfinder:4.1.11</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>retrieve_aligned_contig_paf</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/seqkit:2.4.0--h9ee0642_0</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>retrieve_aligned_pe_reads_sam</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>retrieve_unaligned_pe_reads_sam</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>root_tree1_task</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/theiaphylo:0.1.8</td>
<td>PhyloCompare</td>
</tr>
<tr>
<td>root_tree2_task</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/theiaphylo:0.1.8</td>
<td>PhyloCompare</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>Clair3_Variants_ONT, Freyja_FASTQ, TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>samtools_faidx</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>Clair3_Variants_ONT</td>
</tr>
<tr>
<td>sc2_defaults</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/augur:22.0.2--pyhdfd78af_0</td>
<td>Augur</td>
</tr>
<tr>
<td>semibin</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/semibin:2.0.2--pyhdfd78af_0</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>shared_variants</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Find_Shared_Variants, Snippy_Tree</td>
</tr>
<tr>
<td>skani</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/skani:0.2.2</td>
<td>TheiaViral_Illumina_PE, TheiaViral_ONT</td>
</tr>
<tr>
<td>sm_theiacov_fasta_wrangling</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-08-28-v4</td>
<td>TheiaCoV_FASTA_Batch</td>
</tr>
<tr>
<td>snippy_gene_query</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-06-21</td>
<td>Snippy_Variants</td>
</tr>
<tr>
<td>snippy_tree_wf</td>
<td><strong>gubbins_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/gubbins:3.3--py310pl5321h8472f5a_0</td>
<td>Snippy_Streamline, Snippy_Streamline_FASTA, Snippy_Tree</td>
</tr>
<tr>
<td>snippy_tree_wf</td>
<td><strong>iqtree2_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/iqtree2:2.1.2</td>
<td>Snippy_Streamline, Snippy_Streamline_FASTA, Snippy_Tree</td>
</tr>
<tr>
<td>snippy_tree_wf</td>
<td><strong>snippy_core_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snippy:4.6.0</td>
<td>Snippy_Streamline, Snippy_Streamline_FASTA, Snippy_Tree</td>
</tr>
<tr>
<td>snippy_tree_wf</td>
<td><strong>snp_dists_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snp-dists:0.8.2</td>
<td>Snippy_Streamline, Snippy_Streamline_FASTA, Snippy_Tree</td>
</tr>
<tr>
<td>snippy_tree_wf</td>
<td><strong>snp_sites_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snp-sites:2.5.1</td>
<td>Snippy_Streamline, Snippy_Streamline_FASTA, Snippy_Tree</td>
</tr>
<tr>
<td>snippy_variants_wf</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snippy:4.6.0</td>
<td>Snippy_Streamline, Snippy_Streamline_FASTA, Snippy_Variants</td>
</tr>
<tr>
<td>snp_dists</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/snp-dists:0.8.2</td>
<td>Augur</td>
</tr>
<tr>
<td>sort_bam_assembly_correction</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>TheiaMeta_Illumina_PE</td>
</tr>
<tr>
<td>spades</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/spades:4.1.0</td>
<td>TheiaViral_Illumina_PE</td>
</tr>
<tr>
<td>sra_tsv_to_xml</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/broadinstitute/ncbi-tools:2.10.7.10</td>
<td>Terra_2_NCBI</td>
</tr>
<tr>
<td>stats_n_coverage</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>TheiaCoV_ClearLabs, TheiaCoV_ONT</td>
</tr>
<tr>
<td>stats_n_coverage_primtrim</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>TheiaCoV_ClearLabs, TheiaCoV_ONT</td>
</tr>
<tr>
<td>submit_ena_data</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra_to_ena:0.6</td>
<td>Terra_2_ENA</td>
</tr>
<tr>
<td>summarize_data</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Core_Gene_SNP, MashTree_FASTA, Snippy_Tree, kSNP3, kSNP4</td>
</tr>
<tr>
<td>table2asn</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ncbi-table2asn:1.26.678</td>
<td>Mercury_Prep_N_Batch</td>
</tr>
<tr>
<td>tbp_parser</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/tbp-parser:2.6.0</td>
<td>TBProfiler_tNGS</td>
</tr>
<tr>
<td>tbprofiler</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/tbprofiler:6.6.3</td>
<td>TBProfiler_tNGS</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>medaka_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/artic-ncov2019:1.3.0-medaka-1.4.3</td>
<td>TheiaCoV_ClearLabs</td>
</tr>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>pangolin_docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>TheiaCoV_FASTA_Batch</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>pangolin_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>TheiaCoV_Illumina_PE</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>pangolin_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>TheiaCoV_Illumina_SE</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>pangolin_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>TheiaCoV_ONT</td>
</tr>
<tr>
<td>theiaeuk_illumina_pe</td>
<td><strong>busco_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/ezlabgva/busco:v5.3.2_cv1</td>
<td>TheiaEuk_Illumina_PE</td>
</tr>
<tr>
<td>theiaeuk_ont</td>
<td><strong>busco_docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/ezlabgva/busco:v5.3.2_cv1</td>
<td>TheiaEuk_ONT</td>
</tr>
<tr>
<td>transfer_files</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/cloudsdktool/google-cloud-cli:427.0.0-alpine</td>
<td>Freyja_Update</td>
</tr>
<tr>
<td>transfer_files</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/cloudsdktool/google-cloud-cli:427.0.0-alpine</td>
<td>Transfer_Column_Content</td>
</tr>
<tr>
<td>trim_genbank_fastas</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/vadr:1.3</td>
<td>Mercury_Prep_N_Batch</td>
</tr>
<tr>
<td>trimmomatic_pe</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/trimmomatic:0.39</td>
<td>TBProfiler_tNGS</td>
</tr>
<tr>
<td>ts_mlst</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mlst:2.23.0-2024-12-31</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>tsv_join</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/broadinstitute/viral-core:2.1.33</td>
<td>Augur</td>
</tr>
<tr>
<td>usher</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/pathogengenomics/usher:0.6.2</td>
<td>Usher</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/vadr:1.6.4</td>
<td>TheiaCoV_ClearLabs, TheiaCoV_FASTA, TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT, VADR_Update</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>AMRFinderPlus, AMR_Search, Assembly_Fetch, Augur, Augur_Prep, BaseSpace_Fetch, CZGenEpi_Prep, Cauris_CladeTyper, ChroQueTas, Clair3_Variants_ONT, Concatenate_Column_Content, Concatenate_Illumina_Lanes, Core_Gene_SNP, Dorado_Basecalling, Fetch_SRR_Accession, Find_Shared_Variants, Freyja_Dashboard, Freyja_FASTQ, Freyja_Plot, Gambit_Query, Kraken_ONT, Kraken_PE, Kraken_SE, Lyve_SET, MashTree_FASTA, Mercury_Prep_N_Batch, NCBI_Scrub_PE, NCBI_Scrub_SE, Nextclade_Batch, Pangolin_Update, PhyloCompare, RASUSA, Rename_FASTQ, SRA_Fetch, Snippy_Streamline, Snippy_Streamline_FASTA, Snippy_Tree, Snippy_Variants, TBProfiler_tNGS, Terra_2_ENA, Terra_2_GISAID, Terra_2_NCBI, TheiaCoV_ClearLabs, TheiaCoV_FASTA, TheiaCoV_FASTA_Batch, TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT, TheiaEuk_Illumina_PE, TheiaEuk_ONT, TheiaMeta_Illumina_PE, TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT, TheiaValidate, TheiaViral_Illumina_PE, TheiaViral_ONT, Transfer_Column_Content, Usher, VADR_Update, Zip_Column_Content, kSNP3, kSNP4</td>
</tr>
<tr>
<td>wg_reorder_matrix</td>
<td><strong>docker</strong></td>
<td>us-docker.pkg.dev/general-theiagen/staphb/mykrobe:0.12.1</td>
<td>Snippy_Tree</td>
</tr>
<tr>
<td>zip_files</td>
<td><strong>docker_image</strong></td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Zip_Column_Content</td>
</tr>
</tbody>
</table>
</div>
<h2 id="default-references">Default References<a class="headerlink" href="#default-references" title="Permanent link"></a></h2>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Workflow</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>augur</td>
<td><strong>reference_fasta</strong></td>
<td>File</td>
<td>The reference FASTA file used to align the genomes and build the trees</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, a reference fasta file must be provided otherwise the workflow fails.</td>
<td>Augur</td>
</tr>
<tr>
<td>augur</td>
<td><strong>reference_genbank</strong></td>
<td>File</td>
<td>The GenBank .gb file for the same reference genome used for the reference_fasta</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, a reference genbank file must be provided otherwise the workflow fails.</td>
<td>Augur</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade1</strong></td>
<td>File</td>
<td>The reference assembly for clade 1</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade1_GCA_002759435.3_Cand_auris_B8441_V3_genomic.fasta</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade1_annotated</strong></td>
<td>String</td>
<td>The path to the annotated reference for clade 1</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade1_GCA_002759435.3_Cand_auris_B8441_V3_genomic.gbff</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade2</strong></td>
<td>File</td>
<td>The reference assembly for clade 2</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade2_GCA_003013715.2_ASM301371v2_genomic.fasta</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade2_annotated</strong></td>
<td>String</td>
<td>The path to the annotated reference for clade 2</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade2_GCA_003013715.2_ASM301371v2_genomic.gbff</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade3</strong></td>
<td>File</td>
<td>The reference assembly for clade 3</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade3_GCF_002775015.1_Cand_auris_B11221_V1_genomic.fasta</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade3_annotated</strong></td>
<td>String</td>
<td>The path to the annotated reference for clade 3</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade3_GCF_002775015.1_Cand_auris_B11221_V1_genomic.gbff</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade4</strong></td>
<td>File</td>
<td>The reference assembly for clade 4</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade4_GCA_003014415.1_Cand_auris_B11243_genomic.fasta</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade4_annotated</strong></td>
<td>String</td>
<td>The path to the annotated reference for clade 4</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade4_GCA_003014415.1_Cand_auris_B11243_genomic.gbff</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade5</strong></td>
<td>File</td>
<td>The reference assembly for clade 5</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade5_GCA_016809505.1_ASM1680950v1_genomic.fasta</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade5_annotated</strong></td>
<td>String</td>
<td>The path to the annotated reference for clade 5</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade5_GCA_016809505.1_ASM1680950v1_genomic.gbff</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>cladetyper</td>
<td><strong>ref_clade6</strong></td>
<td>File</td>
<td>The reference assembly for clade 6</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade6_GCA_032714025.1_ASM3271402v1_genomic.fasta</td>
<td>Cauris_CladeTyper</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>Candidozyma auris</em> clade1</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade1_GCA_002759435.2_Cand_auris_B8441_V2_genomic.fasta</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade1_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade1</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade1_GCA_002759435_Cauris_B8441_V2_genomic.gbff</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade2</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade2_GCA_003013715.2_ASM301371v2_genomic.fasta</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade2_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade2</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade2_GCA_003013715.2_ASM301371v2_genomic.gbff</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade3</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade3_GCF_002775015.1_Cand_auris_B11221_V1_genomic.fasta</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade3_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade3</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade3_GCF_002775015.1_Cand_auris_B11221_V1_genomic.gbff</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade4</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade4_GCA_003014415.1_Cand_auris_B11243_genomic.fasta</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade4_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade4</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade4_GCA_003014415.1_Cand_auris_B11243_genomic.gbff</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade5</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade5_GCA_016809505.1_ASM1680950v1_genomic.fasta</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade5_annotated</strong></td>
<td>File</td>
<td>Reference GBFF annotation for <em>C. auris</em> clade5</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade5_GCA_016809505.1_ASM1680950v1_genomic.gbff</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6</strong></td>
<td>File</td>
<td>Reference genome FASTA for <em>C. auris</em> clade6</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade6_GCA_032714025.1_ASM3271402v1_genomic.fasta</td>
<td>TheiaEuk_Illumina_PE, TheiaEuk_ONT</td>
</tr>
<tr>
<td>merlin_magic</td>
<td><strong>cladetyper_ref_clade6</strong></td>
<td>File</td>
<td>The reference assembly for clade 6</td>
<td>gs://theiagen-public-resources-rp/reference_data/eukaryotic/candidozyma/Cauris_Clade6_GCA_032714025.1_ASM3271402v1_genomic.fasta</td>
<td>TheiaProk_FASTA, TheiaProk_Illumina_PE, TheiaProk_Illumina_SE, TheiaProk_ONT</td>
</tr>
<tr>
<td>nextclade_batch</td>
<td><strong>input_ref</strong></td>
<td>File</td>
<td>An optional FASTA file containing reference sequence. This file should contain exactly 1 sequence</td>
<td>Uses the reference fasta associated with the specified nextclade dataset name</td>
<td>Nextclade_Batch</td>
</tr>
<tr>
<td>nextclade_batch</td>
<td><strong>reference_tree_json</strong></td>
<td>File</td>
<td>An optional phylogenetic reference tree file which serves as a target for phylogenetic placement</td>
<td>Uses the reference tree associated with the specified nextclade dataset name</td>
<td>Nextclade_Batch</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>auspice_reference_tree_json</strong></td>
<td>File</td>
<td>An Auspice JSON phylogenetic reference tree which serves as a target for phylogenetic placement.</td>
<td>Inherited from nextclade dataset</td>
<td>TheiaCoV_ClearLabs, TheiaCoV_FASTA, TheiaCoV_FASTA_Batch, TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>input_ref</strong></td>
<td>File</td>
<td>A nucleotide sequence which serves as a reference for the pairwise alignment of all input sequences. This is also the sequence which defines the coordinate system of the genome annotation. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>TheiaCoV_ClearLabs, TheiaCoV_FASTA, TheiaCoV_FASTA_Batch, TheiaCoV_Illumina_PE, TheiaCoV_Illumina_SE, TheiaCoV_ONT</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_gff_file</strong></td>
<td>File</td>
<td>Reference GFF file for the organism being analyzed</td>
<td>Default provided for mpox ("gs://theiagen-public-resources-rp/reference_data/viral/mpox/Mpox-MT903345.1.reference.gff3") and HIV (primer versions 1 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/NC_001802.1.gff3"] and 2 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/AY228557.1.gff3"])</td>
<td>Augur, TheiaCoV_ClearLabs, TheiaCoV_FASTA, TheiaCoV_FASTA_Batch, TheiaCoV_ONT</td>
</tr>
</tbody>
</table>
</div>







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="September 2, 2025 20:28:57 UTC">2025-09-02</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="September 2, 2025 20:28:57 UTC">2025-09-02</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
       2022-2025 <a href="https://www.theiagen.com" target="_blank" rel="noopener">Theiagen Genomics</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/theiagen" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://bsky.app/profile/theiagen.bsky.social" target="_blank" rel="noopener" title="bsky.app" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 7.0.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M407.8 294.7c-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3M288 227.1c-26.1-50.7-97.1-145.2-163.1-191.8C61.6-9.4 37.5-1.7 21.6 5.5 3.3 13.8 0 41.9 0 58.4S9.1 194 15 213.9c19.5 65.7 89.1 87.9 153.2 80.7 3.3-.5 6.6-.9 10-1.4-3.3.5-6.6 1-10 1.4-93.9 14-177.3 48.2-67.9 169.9C220.6 589.1 265.1 437.8 288 361.1c22.9 76.7 49.2 222.5 185.6 103.4 102.4-103.4 28.1-156-65.8-169.9-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3 64.1 7.1 133.6-15.1 153.2-80.7C566.9 194 576 75 576 58.4s-3.3-44.7-21.6-52.9c-15.8-7.1-40-14.9-103.2 29.8C385.1 81.9 314.1 176.4 288 227.1"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.linkedin.com/company/theiagen" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3M135.4 416H69V202.2h66.5V416zM102.2 96a38.5 38.5 0 1 1 0 77 38.5 38.5 0 1 1 0-77m282.1 320h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9z"></path></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "navigation.instant", "navigation.path", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.suggest", "toc.follow"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"alias": true, "default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../../javascripts/tablesort.js"></script>
      
        <script src="../../javascripts/table-search.js"></script>
      
    
  
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>