# MashTree_FASTA

## Quick Facts

{{ render_tsv_table("docs/assets/tables/all_workflows.tsv", sort_by="Name", filters={"Name": "[**MashTree_FASTA**](../workflows/phylogenetic_construction/mashtree_fasta.md)"}, columns=["Workflow Type", "Applicable Kingdom", "Last Known Changes", "Command-line Compatibility","Workflow Level"]) }}

## MashTree_FASTA_PHB

`MashTree_FASTA` creates a phylogenetic tree using Mash distances.

Mash distances are representations of how many kmers two sequences have in common. These distances are generated by transforming all kmers from a sequence into an integer value with hashing and Bloom filters. The hashed kmers are sorted and a "sketch" is created by only using the kmers that appear at the top of the sorted list. These sketches can be compared by counting the number of hashed kmers they have in common. Mashtree uses a neighbor-joining algorithm to cluster these "distances" into phylogenetic trees.

This workflow also features an optional module, `summarize_data`, that creates a presence/absence matrix for the analyzed samples from a list of indicated columns (such as AMR genes, etc.) that can be used in Phandango.

### Inputs

/// html | div[class="searchable-table"]

{{ render_tsv_table("docs/assets/tables/all_inputs.tsv", input_table=True, filters={"Workflow": "MashTree_FASTA"}, columns=["Terra Task Name", "Variable", "Type", "Description", "Default Value", "Terra Status"], sort_by=[("Terra Status", True), "Terra Task Name", "Variable"]) }}

///

### Workflow Actions

??? task "MashTree_FASTA Details"
    `MashTree_FASTA` is run on a set of assembly fastas and creates a phylogenetic tree and matrix.

    !!! techdetails "MashTree_FASTA Technical Details"
        |  | Links |
        | --- | --- |
        | Task | [task_mashtree_fasta.wdl](https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/task_mashtree_fasta.wdl) |
        | Software Source Code | [Mashtree on GitHub](https://github.com/lskatz/mashtree) |
        | Software Documentation | [Mashtree on GitHub](https://github.com/lskatz/mashtree) |
        | Original Publication(s) | [Mashtree: a rapid comparison of whole genome sequence files](https://doi.org/10.21105/joss.01762) |
        
{{ include_md("common_text/data_summary_task.md") }}

### Outputs

/// html | div[class="searchable-table"]

{{ render_tsv_table("docs/assets/tables/all_outputs.tsv", input_table=False, filters={"Workflow": "MashTree"}, columns=["Variable", "Type", "Description"], sort_by=["Variable"]) }}

///

## References

> Katz, L. S., Griswold, T., Morrison, S., Caravas, J., Zhang, S., den Bakker, H.C., Deng, X., and Carleton, H. A., (2019). Mashtree: a rapid comparison of whole genome sequence files. Journal of Open Source Software, 4(44), 1762,Â <https://doi.org/10.21105/joss.01762>
<!-- -->
>Ondov, B. D., Treangen, T. J., Melsted, P., Mallonee, A. B., Bergman, N. H., Koren, S., & Phillippy, A. M. (2016). Mash: Fast genome and metagenome distance estimation using minhash. Genome Biology, 17(1), 132. doi:10.1186/s13059-016-0997-x
