# Snippy_Tree

## Quick Facts

{{ render_tsv_table("docs/assets/tables/all_workflows.tsv", sort_by="Name", filters={"Name": "[**Snippy_Tree**](../workflows/phylogenetic_construction/snippy_tree.md)"}, columns=["Workflow Type", "Applicable Kingdom", "Last Known Changes", "Command-line Compatibility","Workflow Level"]) }}

## Snippy_Tree_PHB

`Snippy_Tree` is a workflow for generating high-quality bacterial phylogenies. It produces a phylogenetic tree and pairwise SNP-distance matrix, with the option to summarize additional metadata to visualize with the tree.

The tree produced by Snippy_Tree will always be a maximum-likelihood phylogeny using a reference-based alignment. There are key options for whether to:

- Generate a core-genome or whole-genome phylogeny (`core_genome`)
- Mask specified regions of the genome with a bed file (e.g. known repetitive regions for TB) (`bed_file`)
- Mask recombination (`use_gubbins`)
- Decide which nucleotide substitution model to use

### Inputs

`Snippy_Tree` is intended to be run after the `Snippy_Variants` workflow. It is a set-level workflow that takes in an array of directories generated by the `Snippy_Variants` workflow, which must be run for each sample that you wish to include in the phylogenetic tree. You should ensure that for all samples included in the phylogeny, `Snippy_Variants` has been run with identical inputs including the same reference genome. When running the `Snippy_Tree` workflow, you will need to provide the same reference genome that you used when running `Snippy_Variants`. `Snippy_Variants` and `Snippy_Tree` can both automatically be run by using the `Snippy_Streamline` workflow.

Sequencing data used in the Snippy_Tree workflow must:

- Be Illumina reads
- Be generated by unbiased whole genome shotgun sequencing
- Pass appropriate QC thresholds for the taxa to ensure that the reads represent reasonably complete genomes that are free of contamination from other taxa or cross-contamination of the same taxa.
- If masking recombination with `Gubbins`, input data should represent whole genomes from the same strain/lineage (e.g. MLST) that share a recent common ancestor.

!!! tip "Guidance for optional inputs"

    Several core and optional tasks can be used to generate the Snippy phylogenetic tree, making it highly flexible and suited to a wide range of datasets. You will need to decide which tasks to use depending on the genomes that you are analyzing. Some guidelines for the optional tasks to use for different genome types are provided below.
    
    ??? toggle "Default settings (suitable for most bacteria)"
    
        The default settings are as follows and are suitable for generating phylogenies for most bacteria
        
        - `core_genome` = true (creates core genome phylogeny)
        - `use_gubbins` = true (recombination masked)
        - nucleotide substitution model will be defined by IQTree's Model Finder
    
    ??? toggle "Phylogenies of _Mycobacterium tuberculosis_ complex"
    
        Phylogenies of MTBC are typically constructed
        
        - Using the H37Rv reference genome
            - `reference_genome_file` = `"gs://theiagen-public-resources-rp/reference_data/bacterial/mycobacterium/MTB-NC_000962.3.fasta"`
        - Masking repetitive regions of the genome (e.g. PE/PPE genes) that are often misaligned
            - `snippy_core_bed` = `"gs://theiagen-public-resources-rp/reference_data/bacterial/mycobacterium/MTB-NC_000962.3.bed"`
        - Without masking recombination because TB can be considered non-recombinant
            - `use_gubbins` = false
        - Using the core genome
            - `core_genome` = true (as default)

/// html | div[class="searchable-table"]

{{ render_tsv_table("docs/assets/tables/all_inputs.tsv", input_table=True, filters={"Workflow": "Snippy_Tree"}, columns=["Terra Task Name", "Variable", "Type", "Description", "Default Value", "Terra Status"], sort_by=[("Terra Status", True), "Terra Task Name", "Variable"]) }}

///

### Workflow Tasks

{{ include_md("common_text/snippy_core_task.md", condition="snippy_tree") }}
{{ include_md("common_text/gubbins_task.md", condition="snippy_tree") }}
{{ include_md("common_text/snp_sites_task.md") }}
{{ include_md("common_text/iqtree2_task.md") }}
{{ include_md("common_text/snp_dists_task.md", condition="snippy") }}
{{ include_md("common_text/data_summary_task.md", condition="snippy") }}
{{ include_md("common_text/shared_variants_task.md", condition="snippy") }}
{{ include_md("common_text/snippy_qc_concatenation_task.md") }}

### Outputs

/// html | div[class="searchable-table"]

{{ render_tsv_table("docs/assets/tables/all_outputs.tsv", input_table=False, filters={"Workflow": "Snippy_Tree"}, columns=["Variable", "Type", "Description"], sort_by=["Variable"]) }}

///

## References

> **Gubbins:** Croucher, Nicholas J., Andrew J. Page, Thomas R. Connor, Aidan J. Delaney, Jacqueline A. Keane, Stephen D. Bentley, Julian Parkhill, and Simon R. Harris. 2015. "Rapid Phylogenetic Analysis of Large Samples of Recombinant Bacterial Whole Genome Sequences Using Gubbins." Nucleic Acids Research 43 (3): e15.
<!-- -->
> **SNP-sites:** Page, Andrew J., Ben Taylor, Aidan J. Delaney, Jorge Soares, Torsten Seemann, Jacqueline A. Keane, and Simon R. Harris. 2016. "SNP-Sites: Rapid Efficient Extraction of SNPs from Multi-FASTA Alignments." Microbial Genomics 2 (4): e000056.
<!-- -->
> **IQTree:** Nguyen, Lam-Tung, Heiko A. Schmidt, Arndt von Haeseler, and Bui Quang Minh. 2015. "IQ-TREE: A Fast and Effective Stochastic Algorithm for Estimating Maximum-Likelihood Phylogenies." Molecular Biology and Evolution 32 (1): 268â€“74.
