<!DOCTYPE html><html lang="en" class="no-js"><head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A collection of bioinformatics workflows for genomic characterization, submission preparation, and genomic epidemiology of pathogens of public health concern.">
      
      
        <meta name="author" content="Theiagen Genomics">
      
      
        <link rel="canonical" href="https://theiagen.github.io/public_health_bioinformatics/main/workflows/genomic_characterization/freyja/">
      
      
        <link rel="prev" href="../../phylogenetic_construction/find_shared_variants/">
      
      
        <link rel="next" href="../../standalone/gambit_query/">
      
      
      <link rel="icon" href="../../../assets/logos/Theiagen-Symbol-Standard-01.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.20">
    
    
      
        <title>Freyja Workflow Series - Public Health Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.e53b48f4.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  <link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#freyja-workflow-series" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version release.
  <a href="../../../.."> 
    <strong>Click here to go to the latest version release.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-header__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Public Health Bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Freyja Workflow Series
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo" aria-label="Switch to Dark Mode" type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to Dark Mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"></path></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo" aria-label="Switch to Light Mode" type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to Light Mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m3.55 19.09 1.41 1.41 1.8-1.79-1.42-1.42M12 6c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6c0-3.32-2.69-6-6-6m8 7h3v-2h-3m-2.76 7.71 1.8 1.79 1.41-1.41-1.79-1.8M20.45 5l-1.41-1.4-1.8 1.79 1.42 1.42M13 1h-2v3h2M6.76 5.39 4.96 3.6 3.55 5l1.79 1.81zM1 13h3v-2H1m12 9h-2v3h2"></path></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../.." class="md-tabs__link">
          
  
  
  Public Health Bioinformatics

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-nav__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    Public Health Bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Public Health Bioinformatics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Public Health Bioinformatics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Getting Started
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            Getting Started
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/commandline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With the Command-Line
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/terra/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With Terra.bio
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/philosophy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Workflow Failure Philosophy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Other Resources
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_3">
            <span class="md-nav__icon md-icon"></span>
            Other Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/sops/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Available SOPs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/versions/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Database and Docker Versions
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit_database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT Database Creation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/phylogenetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Phylogenetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/custom_organisms/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Running Custom Organisms on TheiaCoV
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1">
        
          
          <label class="md-nav__link" for="__nav_1_5_1" id="__nav_1_5_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Type
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Type
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_type/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_2" id="__nav_1_5_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Import
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_2">
            <span class="md-nav__icon md-icon"></span>
            Data Import
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_3" id="__nav_1_5_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Genomic Characterization
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_3">
            <span class="md-nav__icon md-icon"></span>
            Genomic Characterization
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#freyja-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja Overview
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure1" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 1
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja-sequencing-platforms-and-data-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja, Sequencing Platforms and Data Quality
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#freyja-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#freyja_update" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Update_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Update_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_fastq" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_FASTQ_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_FASTQ_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_1" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_1" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_plot" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Plot_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Plot_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_2" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks_1" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_2" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_dashboard" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Dashboard_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Dashboard_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_3" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks_2" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_3" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-freyja-on-other-pathogens" class="md-nav__link">
    <span class="md-ellipsis">
      Running Freyja on other pathogens
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running Freyja on other pathogens">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-the-freyja_pathogen-flag" class="md-nav__link">
    <span class="md-ellipsis">
      Using the freyja_pathogen flag
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Using the freyja_pathogen flag">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure2" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 2
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#providing-the-appropriate-barcode-file" class="md-nav__link">
    <span class="md-ellipsis">
      Providing the appropriate barcode file
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Providing the appropriate barcode file">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure3" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 3
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#figure4" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 4
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      References
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_4" id="__nav_1_5_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Construction
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_4">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Construction
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_5" id="__nav_1_5_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Placement
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_5">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Placement
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_6" id="__nav_1_5_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Public Data Sharing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_6">
            <span class="md-nav__icon md-icon"></span>
            Public Data Sharing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_7">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_7" id="__nav_1_5_1_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exporting Data from Terra
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_7">
            <span class="md-nav__icon md-icon"></span>
            Exporting Data from Terra
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_8">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_8" id="__nav_1_5_1_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Standalone
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_8">
            <span class="md-nav__icon md-icon"></span>
            Standalone
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2" id="__nav_1_5_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Kingdom
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Kingdom
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_kingdom/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_2" id="__nav_1_5_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Any Taxa
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_2">
            <span class="md-nav__icon md-icon"></span>
            Any Taxa
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_3" id="__nav_1_5_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Bacteria
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_3">
            <span class="md-nav__icon md-icon"></span>
            Bacteria
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_4" id="__nav_1_5_2_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Mycotics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_4">
            <span class="md-nav__icon md-icon"></span>
            Mycotics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_5" id="__nav_1_5_2_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Viral
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_5">
            <span class="md-nav__icon md-icon"></span>
            Viral
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#freyja-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja Overview
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure1" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 1
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja-sequencing-platforms-and-data-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja, Sequencing Platforms and Data Quality
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#freyja-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#freyja_update" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Update_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Update_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_fastq" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_FASTQ_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_FASTQ_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_1" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_1" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_plot" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Plot_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Plot_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_2" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks_1" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_2" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_dashboard" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Dashboard_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Dashboard_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_3" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks_2" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_3" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-freyja-on-other-pathogens" class="md-nav__link">
    <span class="md-ellipsis">
      Running Freyja on other pathogens
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running Freyja on other pathogens">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-the-freyja_pathogen-flag" class="md-nav__link">
    <span class="md-ellipsis">
      Using the freyja_pathogen flag
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Using the freyja_pathogen flag">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure2" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 2
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#providing-the-appropriate-barcode-file" class="md-nav__link">
    <span class="md-ellipsis">
      Providing the appropriate barcode file
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Providing the appropriate barcode file">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure3" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 3
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#figure4" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 4
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      References
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_3" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows Alphabetically
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            Workflows Alphabetically
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_alphabetically/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#freyja-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja Overview
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure1" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 1
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja-sequencing-platforms-and-data-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja, Sequencing Platforms and Data Quality
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#freyja-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#freyja_update" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Update_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Update_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_fastq" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_FASTQ_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_FASTQ_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_1" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_1" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_plot" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Plot_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Plot_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_2" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks_1" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_2" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_dashboard" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Dashboard_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Dashboard_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_3" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks_2" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_3" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-freyja-on-other-pathogens" class="md-nav__link">
    <span class="md-ellipsis">
      Running Freyja on other pathogens
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running Freyja on other pathogens">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-the-freyja_pathogen-flag" class="md-nav__link">
    <span class="md-ellipsis">
      Using the freyja_pathogen flag
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Using the freyja_pathogen flag">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure2" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 2
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#providing-the-appropriate-barcode-file" class="md-nav__link">
    <span class="md-ellipsis">
      Providing the appropriate barcode file
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Providing the appropriate barcode file">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure3" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 3
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#figure4" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 4
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      References
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Contributing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            Contributing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/doc_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Documentation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/code_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Code
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#freyja-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja Overview
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure1" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 1
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja-sequencing-platforms-and-data-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja, Sequencing Platforms and Data Quality
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#freyja-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#freyja_update" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Update_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Update_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_fastq" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_FASTQ_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_FASTQ_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_1" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_1" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_plot" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Plot_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Plot_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_2" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks_1" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_2" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja_dashboard" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja_Dashboard_PHB
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Freyja_Dashboard_PHB">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inputs_3" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#analysis-tasks_2" class="md-nav__link">
    <span class="md-ellipsis">
      Analysis Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs_3" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-freyja-on-other-pathogens" class="md-nav__link">
    <span class="md-ellipsis">
      Running Freyja on other pathogens
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running Freyja on other pathogens">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-the-freyja_pathogen-flag" class="md-nav__link">
    <span class="md-ellipsis">
      Using the freyja_pathogen flag
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Using the freyja_pathogen flag">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure2" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 2
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#providing-the-appropriate-barcode-file" class="md-nav__link">
    <span class="md-ellipsis">
      Providing the appropriate barcode file
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Providing the appropriate barcode file">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure3" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 3
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#figure4" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 4
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      References
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="freyja-workflow-series">Freyja Workflow Series<a class="headerlink" href="#freyja-workflow-series" title="Permanent link">¶</a></h1>
<h2 id="quick-facts">Quick Facts<a class="headerlink" href="#quick-facts" title="Permanent link">¶</a></h2>
<table>
<thead>
<tr>
<th><strong>Workflow Type</strong></th>
<th><strong>Applicable Kingdom</strong></th>
<th><strong>Last Known Changes</strong></th>
<th><strong>Command-line Compatibility</strong></th>
<th><strong>Workflow Level</strong></th>
<th><strong>Dockstore</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../../../workflows_overview/workflows_type/#genomic-characterization">Genomic Characterization</a></td>
<td>SARS-CoV-2, <a href="../../../workflows_overview/workflows_kingdom/#viral">Viral</a></td>
<td>v3.1.0</td>
<td>Yes</td>
<td>Sample-level, Set-level</td>
<td><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/Freyja_FASTQ_PHB:main?tab=info">Freyja_FASTQ_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/Freyja_Plot_PHB:main?tab=info">Freyja_Plot_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/Freyja_Dashboard_PHB:main?tab=info">Freyja_Dashboard_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/Freyja_Update_PHB:main?tab=info">Freyja_Update_PHB</a></td>
</tr>
</tbody>
</table>
<h2 id="freyja-overview">Freyja Overview<a class="headerlink" href="#freyja-overview" title="Permanent link">¶</a></h2>
<p><a href="https://github.com/andersen-lab/Freyja">Freyja</a> is a tool for analysing viral mixed sample genomic sequencing data. Developed by Joshua Levy from the <a href="https://andersen-lab.com/">Andersen Lab</a>, it performs two main steps:</p>
<ol>
<li><strong>Variant Frequency Estimation:</strong> Freyja calculates the frequencies of single nucleotide variants (SNVs) in the genomic sequencing data.</li>
<li><strong>Depth-Weighted Demixing:</strong> It separates mixed populations of viral subtypes using a depth-weighted statistical approach, estimating the proportional abundance of each subtype in the sample based on the frequencies of subtype-defining variants.</li>
</ol>
<p>Additional post-processing steps can produce visualizations of aggregated samples.</p>
<div class="admonition dna">
<p class="admonition-title">Wastewater and more</p>
<p>The typical use case of Freyja is to <strong>analyze mixed SARS-CoV-2 samples</strong> from a sequencing dataset, most often <strong>wastewater</strong>, but the tool is not limited to this context. With the appropriate reference genomes and barcode files, Freyja can be adapted for other pathogens, including MPXV, Influenza, RSV, and Measles.</p>
<div class="admonition warning">
<p class="admonition-title">Default Values</p>
<p>The defaults included in the Freyja workflows reflect this use case but <strong>can be adjusted for other pathogens</strong>. See the <a href="./#running-freyja-on-other-pathogens"><strong>Running Freyja on other pathogens</strong></a> section for more information. Please be aware this is an <mark><em>experimental feature</em></mark> and we cannot guarantee complete functionality at this time.</p>
</div>
</div>
<div class="admonition caption">
<p class="admonition-title">Figure 1: Workflow diagram for Freyja Suite of workflows</p>
<h5 id="figure1">Figure 1<a class="headerlink" href="#figure1" title="Permanent link">¶</a></h5>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/Freyja_Suite.png" data-desc-position="bottom"><img alt="**Figure 1: Workflow diagram for Freyja Suite of workflows.**" src="../../../assets/figures/Freyja_Suite.png" width="100%"></a></p>
<p>Depending on the type of data (Illumina or Oxford Nanopore), the Read QC and Filtering steps, as well as the Read Alignment steps use different software. The user can specify if the barcodes and lineages file should be updated with <code>freyja update</code> before running Freyja or if bootstrapping is to be performed with <code>freyja boot</code>.</p>
</div>
<p>Four workflows have been created that perform different parts of Freyja:</p>
<ul>
<li><a href="./#freyja_update"><strong>Freyja_Update_PHB</strong></a></li>
<li><a href="./#freyja_fastq"><strong>Freyja_FASTQ_PHB</strong></a></li>
<li><a href="./#freyja_plot"><strong>Freyja_Plot_PHB</strong></a></li>
<li><a href="./#freyja_dashboard"><strong>Freyja_Dashboard_PHB</strong></a></li>
</ul>
<p>The main workflow is <a href="./#freyja_fastq"><strong>Freyja_FASTQ_PHB</strong></a> (<a href="./#figure1">Figure 1</a>). Depending on the type of input data (Illumina paired-end, Illumina single-end or ONT), it runs various QC modules before aligning the sample with either <a href="https://github.com/lh3/bwa">BWA</a> (Illumina) or <a href="https://github.com/lh3/minimap2">minimap2</a> (ONT) to the provided reference file, followed by iVar for primer trimming. After the preprocessing is completed, <a href="https://github.com/andersen-lab/Freyja">Freyja</a> is run to generate relative lineage abundances (demix) from the sample. Optional bootstrapping may be performed.</p>
<div class="admonition dna">
<p class="admonition-title">Data Compatability</p>
<p>The <strong>Freyja_FASTQ_PHB workflow</strong> is compatible with the following input data types:</p>
<div class="language-text highlight"><pre><span></span><code>- Ilumina Single-End
- Illumina Paired-End
- Oxford Nanopore
</code></pre></div>
</div>
<p><a href="./#freyja_update"><strong>Freyja_Update_PHB</strong></a> will copy the <strong>SARS-CoV-2</strong> reference files that can then be used as input for the <a href="./#freyja_fastq">Freyja_FASTQ_PHB</a> workflow.</p>
<p>Two options are available to visualize the Freyja results: <a href="./#freyja_plot"><strong>Freyja_Plot_PHB</strong></a> and <a href="./#freyja_dashboard"><strong>Freyja_Dashboard_PHB</strong></a>. <a href="./#freyja_plot">Freyja_Plot_PHB</a> aggregates multiple samples using output from <a href="./#freyja_fastq">Freyja_FASTQ_PHB</a> to generate a plot that shows fractional abundance estimates for all samples. including the option to plot sample collection date information. Alternatively, <a href="./#freyja_dashboard"><strong>Freyja_Dashboard_PHB</strong></a> aggregates multiple samples using output from <a href="./#freyja_fastq">Freyja_FASTQ_PHB</a> to generate an interactive visualization. This workflow requires an additional input field called viral load, which is the number of viral copies per liter.</p>
<h3 id="freyja-sequencing-platforms-and-data-quality">Freyja, Sequencing Platforms and Data Quality<a class="headerlink" href="#freyja-sequencing-platforms-and-data-quality" title="Permanent link">¶</a></h3>
<p>The choice of sequencing platform and the quality of the data directly influence Freyja's performance. High-accuracy platforms like Illumina provide reliable SNV detection, enhancing the precision of lineage abundance estimates. In contrast, platforms with higher error rates, such as Nanopore, whilst it has improved greatly in the recent years, may introduce uncertainties in variant calling, affecting the deconvolution process. Sequencing depth requirements will increase as the quality of the sequencing data decreases. A rational target depth is 100X coverage for sequencing data with Q-scores in the range of 25-30.</p>
<p>Additionally, inadequate sequencing depth can hinder Freyja's ability to differentiate between lineages, leading to potential misestimations. Sequencing depth requirements will increase with the complexity of the sample composition and the diversity of lineages present. For samples containing multiple closely related lineages, higher sequencing depth is necessary to resolve subtle differences in genetic variation and accurately estimate lineage abundances. This is particularly important for pathogens with high mutation rates or a large number of cocirculating lineages, such as influenza, where distinguishing between lineages relies on detecting specific single nucleotide variants (SNVs) with high confidence.</p>
<h2 id="freyja-workflows">Freyja Workflows<a class="headerlink" href="#freyja-workflows" title="Permanent link">¶</a></h2>
<h3 id="freyja_update">Freyja_Update_PHB<a class="headerlink" href="#freyja_update" title="Permanent link">¶</a></h3>
<p>This workflow will copy the <strong>SARS-CoV-2 reference files</strong> (<code>curated_lineages.json</code> and <code>usher_barcodes.feather</code>) from <a href="https://github.com/andersen-lab/Freyja/tree/main/freyja/data">the source repository</a> to a user-specific Google Cloud Storage (GCP) location (often a <a href="http://Terra.bio">Terra.bio</a> workspace-associated bucket). These files can then be used as input for the <a href="./#freyja_fastq">Freyja_FASTQ_PHB workflow</a>.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This workflow is compatible only with <strong>SARS-CoV-2 reference files</strong>! To download reference files for other organisms please see the following repository:&nbsp;<a href="https://github.com/gp201/Freyja-barcodes">Freyja Barcodes</a>.</p>
<p>More information is available in the <a href="./#running-freyja-on-other-pathogens"><strong>Running Freyja on other pathogens</strong></a> section.</p>
</div>
<h4 id="inputs">Inputs<a class="headerlink" href="#inputs" title="Permanent link">¶</a></h4>
<p>We recommend running this workflow with <strong>"Run inputs defined by file paths"</strong> selected since no information from a Terra data table is actually being used. We also recommend turning off call caching so new information is retrieved every time.</p>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>freyja_update</td>
<td><strong>gcp_uri</strong></td>
<td>String</td>
<td>The path where you want the Freyja reference files to be stored. Include gs:// at the beginning of the string. Full example with a Terra workspace bucket: "gs://fc-87ddd67a-c674-45a8-9651-f91e3d2f6bb7"</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_update_refs</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_update_refs</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_update_refs</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/freyja:1.5.3</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_update_refs</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>transfer_files</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>transfer_files</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>transfer_files</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Docker image to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/cloudsdktool/google-cloud-cli:427.0.0-alpine</td>
<td>Optional</td>
</tr>
<tr>
<td>transfer_files</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
<h4 id="outputs">Outputs<a class="headerlink" href="#outputs" title="Permanent link">¶</a></h4>
<p>This workflow does not produce any outputs that appear in a Terra data table. The reference files will appear at the location specified with the <code>gcp_uri</code> input variable.</p>
<h3 id="freyja_fastq">Freyja_FASTQ_PHB<a class="headerlink" href="#freyja_fastq" title="Permanent link">¶</a></h3>
<p>Freyja measures SNV frequency and sequencing depth at each position in the genome to return an estimate of the true lineage abundances in the sample. The method uses lineage-defining "barcodes" that, for SARS-CoV-2, are derived from the UShER global phylogenetic tree as a base set for demixing. <strong>Freyja_FASTQ_PHB</strong> returns as output a TSV file that includes the lineages present and their corresponding abundances, along with other values.</p>
<p>The Freyja_FASTQ_PHB workflow is compatible with the multiple input data types: Ilumina Single-End, Illumina Paired-End and Oxford Nanopore. Depending on the type of input data, different input values are used.</p>
<p><strong>Table 1:</strong> Freyja_FASTQ_PHB input configuration for different types of input data.</p>
<table>
<thead>
<tr>
<th>Table Columns</th>
<th>Illumina Paired-End</th>
<th>Illumina Single-End</th>
<th>Oxford Nanopore</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>read1</strong></td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
</tr>
<tr>
<td><strong>read2</strong></td>
<td>✅</td>
<td>❌</td>
<td>❌</td>
</tr>
<tr>
<td><strong>ont</strong></td>
<td><code>false</code></td>
<td><code>false</code></td>
<td><code>true</code></td>
</tr>
</tbody>
</table>
<h4 id="inputs_1">Inputs<a class="headerlink" href="#inputs_1" title="Permanent link">¶</a></h4>
<p>This workflow runs on the sample level.</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:3"><input checked="checked" id="illumina-paired-end-input-data" name="__tabbed_1" type="radio"><input id="illumina-single-end-input-data" name="__tabbed_1" type="radio"><input id="ont-input-data" name="__tabbed_1" type="radio"><div class="tabbed-labels"><label for="illumina-paired-end-input-data">Illumina paired-end input data</label><label for="illumina-single-end-input-data">Illumina single-end input data</label><label for="ont-input-data">ONT input data</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>freyja_fastq</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>FASTQ file containing read1 sequences (Illumina or (ONT)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>reference_genome</strong></td>
<td>File</td>
<td>The reference genome to use; should match the reference used for alignment (Wuhan-Hu-1)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_lineage_metadata</strong></td>
<td>File</td>
<td>File containing the lineage metadata; the "curated_lineages.json" file found <a href="https://github.com/andersen-lab/Freyja/tree/main/freyja/data">https://github.com/andersen-lab/Freyja/tree/main/freyja/data</a> can be used for this variable. Does not need to be provided if update_db is true or if the freyja_pathogen is provided.</td>
<td></td>
<td>Optional, Required</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>6</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>adapt</strong></td>
<td>Float</td>
<td>adaptive lasso penalty parameter</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>bootstrap</strong></td>
<td>Boolean</td>
<td>Perform bootstrapping</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>confirmed_only</strong></td>
<td>Boolean</td>
<td>Include only confirmed SARS-CoV-2 lineages</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/freyja:1.5.3</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>eps</strong></td>
<td>Float</td>
<td>The minimum lineage abundance cut-off value</td>
<td>0.001</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>number_bootstraps</strong></td>
<td>Int</td>
<td>The number of bootstraps to perform (only used if bootstrap = true)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>update_db</strong></td>
<td>Boolean</td>
<td>Updates the Freyja reference files (the usher barcodes and lineage metadata files) but will not save them as output (use Freyja_Update for that purpose). If set to true, the freyja_lineage_metadata and freyja_barcodes files are not required.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>depth_cutoff</strong></td>
<td>Int</td>
<td>The minimum coverage depth with which to exclude sites below this value and group identical barcodes -- THIS MAY NOT WORK FOR NON-SARS-COV-2 ORGANISMS!</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_barcodes</strong></td>
<td>File</td>
<td>Custom barcode file. Does not need to be provided if update_db is true if the freyja_pathogen is provided.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_pathogen</strong></td>
<td>String</td>
<td>Pathogen of interest, used if not providing the barcodes and lineage metadata files. Options: SARS-CoV-2, MPXV, H5NX, H1N1pdm, FLU-B-VIC, MEASLESN450, MEASLES, RSVa, RSVb</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_pathogen</strong></td>
<td>String</td>
<td>Pathogen to be used by Freyja</td>
<td>SARS-CoV-2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>kraken2_target_organism</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td>Severe acute respiratory syndrome coronavirus 2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>ont</strong></td>
<td>Boolean</td>
<td>Indicates if the input data is derived from an ONT instrument.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>primer_bed</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Illumina reverse read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>reference_gff</strong></td>
<td>File</td>
<td>The GFF file for reference; should match the reference used for alignment (Wuhan-Hu-1)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>trimmomatic_min_length</strong></td>
<td>Int</td>
<td>The minimum length cut-off when performing read cleaning</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/seqkit:2.4.0--h9ee0642_0</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>query2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>keep_noprimer_reads</strong></td>
<td>Boolean</td>
<td>Include reads with no primers</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/artic-ncov2019:1.3.0-medaka-1.4.3</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>downsampling_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td>150</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Length of the genome</td>
<td>5000000</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_docker</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-08-28-v4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/nanoq:0.9.0--hec16e2b_1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_max_read_length</strong></td>
<td>Int</td>
<td>Maximum read length to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_max_read_qual</strong></td>
<td>Int</td>
<td>Maximum read quality to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_min_read_length</strong></td>
<td>Int</td>
<td>Minimum read length to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_min_read_qual</strong></td>
<td>Int</td>
<td>Minimum read quality to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_bases</strong></td>
<td>String</td>
<td>Explicitly set the number of bases required e.g., 4.3kb, 7Tb, 9000, 4.1MB. If this option is given, --coverage and --genome-size are ignored</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/rasusa:2.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_fraction_of_reads</strong></td>
<td>Float</td>
<td>Subsample to a fraction of the reads - e.g., 0.5 samples half the reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_number_of_reads</strong></td>
<td>Int</td>
<td>Subsample to a specific number of reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_seed</strong></td>
<td>Int</td>
<td>Random seed to use</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>run_prefix</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>A FASTA file containing adapter sequences</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the Kraken2 task should be called; for non-TheiaCoV workflows, the <code>kraken_db</code> variable must be provided.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the MIDAS task should be called.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>extract_unclassified</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Additional arguments to use with fastp</td>
<td>--detect_adapter_for_pe -g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_complete_only</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_decontaminate_mem</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_is_accession</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_refseq</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>Database to use with MIDAS. Not required as one will be auto-selected when running the MIDAS task.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>The file containing the phix sequence to be used during bbduk task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>Options: "trimmomatic" or "fastp" to indicate which read trimming module to use</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>Allows the user to decide between fastq_scan (default) and fastqc for the evaluation of read quality.</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>taxon_id</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>The minimum quality score to keep during trimming</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trim_quality_trim_score</strong></td>
<td>Int</td>
<td>The minimum quality score to keep during trimming</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>The window size to use during trimming</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Additional command-line arguments to use with trimmomatic</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>--detect_adapter_for_pe -g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>min_qual</strong></td>
<td>Int</td>
<td>Minimum quality score for reads to be included in the analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>freyja_fastq</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>FASTQ file containing read1 sequences (Illumina or (ONT)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>reference_genome</strong></td>
<td>File</td>
<td>The reference genome to use; should match the reference used for alignment (Wuhan-Hu-1)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_lineage_metadata</strong></td>
<td>File</td>
<td>File containing the lineage metadata; the "curated_lineages.json" file found <a href="https://github.com/andersen-lab/Freyja/tree/main/freyja/data">https://github.com/andersen-lab/Freyja/tree/main/freyja/data</a> can be used for this variable. Does not need to be provided if update_db is true or if the freyja_pathogen is provided.</td>
<td></td>
<td>Optional, Required</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>6</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>adapt</strong></td>
<td>Float</td>
<td>adaptive lasso penalty parameter</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>bootstrap</strong></td>
<td>Boolean</td>
<td>Perform bootstrapping</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>confirmed_only</strong></td>
<td>Boolean</td>
<td>Include only confirmed SARS-CoV-2 lineages</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/freyja:1.5.3</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>eps</strong></td>
<td>Float</td>
<td>The minimum lineage abundance cut-off value</td>
<td>0.001</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>number_bootstraps</strong></td>
<td>Int</td>
<td>The number of bootstraps to perform (only used if bootstrap = true)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>update_db</strong></td>
<td>Boolean</td>
<td>Updates the Freyja reference files (the usher barcodes and lineage metadata files) but will not save them as output (use Freyja_Update for that purpose). If set to true, the freyja_lineage_metadata and freyja_barcodes files are not required.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>depth_cutoff</strong></td>
<td>Int</td>
<td>The minimum coverage depth with which to exclude sites below this value and group identical barcodes -- THIS MAY NOT WORK FOR NON-SARS-COV-2 ORGANISMS!</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_barcodes</strong></td>
<td>File</td>
<td>Custom barcode file. Does not need to be provided if update_db is true if the freyja_pathogen is provided.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_pathogen</strong></td>
<td>String</td>
<td>Pathogen of interest, used if not providing the barcodes and lineage metadata files. Options: SARS-CoV-2, MPXV, H5NX, H1N1pdm, FLU-B-VIC, MEASLESN450, MEASLES, RSVa, RSVb</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_pathogen</strong></td>
<td>String</td>
<td>Pathogen to be used by Freyja</td>
<td>SARS-CoV-2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>kraken2_target_organism</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td>Severe acute respiratory syndrome coronavirus 2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>ont</strong></td>
<td>Boolean</td>
<td>Indicates if the input data is derived from an ONT instrument.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>primer_bed</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Illumina reverse read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>reference_gff</strong></td>
<td>File</td>
<td>The GFF file for reference; should match the reference used for alignment (Wuhan-Hu-1)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>trimmomatic_min_length</strong></td>
<td>Int</td>
<td>The minimum length cut-off when performing read cleaning</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/seqkit:2.4.0--h9ee0642_0</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>query2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>keep_noprimer_reads</strong></td>
<td>Boolean</td>
<td>Include reads with no primers</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/artic-ncov2019:1.3.0-medaka-1.4.3</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>downsampling_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td>150</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Length of the genome</td>
<td>5000000</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_docker</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-08-28-v4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/nanoq:0.9.0--hec16e2b_1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_max_read_length</strong></td>
<td>Int</td>
<td>Maximum read length to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_max_read_qual</strong></td>
<td>Int</td>
<td>Maximum read quality to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_min_read_length</strong></td>
<td>Int</td>
<td>Minimum read length to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_min_read_qual</strong></td>
<td>Int</td>
<td>Minimum read quality to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_bases</strong></td>
<td>String</td>
<td>Explicitly set the number of bases required e.g., 4.3kb, 7Tb, 9000, 4.1MB. If this option is given, --coverage and --genome-size are ignored</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/rasusa:2.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_fraction_of_reads</strong></td>
<td>Float</td>
<td>Subsample to a fraction of the reads - e.g., 0.5 samples half the reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_number_of_reads</strong></td>
<td>Int</td>
<td>Subsample to a specific number of reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_seed</strong></td>
<td>Int</td>
<td>Random seed to use</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>run_prefix</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>extract_unclassified</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>--detect_adapter_for_pe -g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_complete_only</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_decontaminate_mem</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_is_accession</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_refseq</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>taxon_id</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>The minimum quality score to keep during trimming</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trim_quality_trim_score</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>A FASTA file containing adapter sequences</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the Kraken2 task should be called; for non-TheiaCoV workflows, the <code>kraken_db</code> variable must be provided.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the MIDAS task should be called.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Additional arguments to use with fastp</td>
<td>--detect_adapter_for_pe -g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>Database to use with MIDAS. Not required as one will be auto-selected when running the MIDAS task.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>The file containing the phix sequence to be used during bbduk task</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>Options: "trimmomatic" or "fastp" to indicate which read trimming module to use</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>Allows the user to decide between fastq_scan (default) and fastqc for the evaluation of read quality.</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>The minimum quality score to keep during trimming</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>The window size to use during trimming</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Additional command-line arguments to use with trimmomatic</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>min_qual</strong></td>
<td>Int</td>
<td>Minimum quality score for reads to be included in the analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>freyja_fastq</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>FASTQ file containing read1 sequences (Illumina or (ONT)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>reference_genome</strong></td>
<td>File</td>
<td>The reference genome to use; should match the reference used for alignment (Wuhan-Hu-1)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_lineage_metadata</strong></td>
<td>File</td>
<td>File containing the lineage metadata; the "curated_lineages.json" file found <a href="https://github.com/andersen-lab/Freyja/tree/main/freyja/data">https://github.com/andersen-lab/Freyja/tree/main/freyja/data</a> can be used for this variable. Does not need to be provided if update_db is true or if the freyja_pathogen is provided.</td>
<td></td>
<td>Optional, Required</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>6</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>bwa</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>adapt</strong></td>
<td>Float</td>
<td>adaptive lasso penalty parameter</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>bootstrap</strong></td>
<td>Boolean</td>
<td>Perform bootstrapping</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>confirmed_only</strong></td>
<td>Boolean</td>
<td>Include only confirmed SARS-CoV-2 lineages</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/freyja:1.5.3</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>eps</strong></td>
<td>Float</td>
<td>The minimum lineage abundance cut-off value</td>
<td>0.001</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>number_bootstraps</strong></td>
<td>Int</td>
<td>The number of bootstraps to perform (only used if bootstrap = true)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja</td>
<td><strong>update_db</strong></td>
<td>Boolean</td>
<td>Updates the Freyja reference files (the usher barcodes and lineage metadata files) but will not save them as output (use Freyja_Update for that purpose). If set to true, the freyja_lineage_metadata and freyja_barcodes files are not required.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>depth_cutoff</strong></td>
<td>Int</td>
<td>The minimum coverage depth with which to exclude sites below this value and group identical barcodes -- THIS MAY NOT WORK FOR NON-SARS-COV-2 ORGANISMS!</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_barcodes</strong></td>
<td>File</td>
<td>Custom barcode file. Does not need to be provided if update_db is true if the freyja_pathogen is provided.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_pathogen</strong></td>
<td>String</td>
<td>Pathogen of interest, used if not providing the barcodes and lineage metadata files. Options: SARS-CoV-2, MPXV, H5NX, H1N1pdm, FLU-B-VIC, MEASLESN450, MEASLES, RSVa, RSVb</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>freyja_pathogen</strong></td>
<td>String</td>
<td>Pathogen to be used by Freyja</td>
<td>SARS-CoV-2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>kraken2_target_organism</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td>Severe acute respiratory syndrome coronavirus 2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>ont</strong></td>
<td>Boolean</td>
<td>Indicates if the input data is derived from an ONT instrument.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>primer_bed</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Illumina reverse read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>reference_gff</strong></td>
<td>File</td>
<td>The GFF file for reference; should match the reference used for alignment (Wuhan-Hu-1)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_fastq</td>
<td><strong>trimmomatic_min_length</strong></td>
<td>Int</td>
<td>The minimum length cut-off when performing read cleaning</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/seqkit:2.4.0--h9ee0642_0</td>
<td>Optional</td>
</tr>
<tr>
<td>get_fasta_genome_size</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>minimap2</td>
<td><strong>query2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>keep_noprimer_reads</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>primer_trim</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/artic-ncov2019:1.3.0-medaka-1.4.3</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>artic_guppyplex_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the Kraken2 task should be called; for non-TheiaCoV workflows, the <code>kraken_db</code> variable must be provided.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>downsampling_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td>150</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Length of the genome</td>
<td>5000000</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_docker</strong></td>
<td>Int</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-08-28-v4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken2_recalculate_abundances_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/nanoq:0.9.0--hec16e2b_1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_max_read_length</strong></td>
<td>Int</td>
<td>Maximum read length to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_max_read_qual</strong></td>
<td>Int</td>
<td>Maximum read quality to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_min_read_length</strong></td>
<td>Int</td>
<td>Minimum read length to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>nanoq_min_read_qual</strong></td>
<td>Int</td>
<td>Minimum read quality to use for filtering</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>ncbi_scrub_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_bases</strong></td>
<td>String</td>
<td>Explicitly set the number of bases required e.g., 4.3kb, 7Tb, 9000, 4.1MB. If this option is given, --coverage and --genome-size are ignored</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/rasusa:2.1.0</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_fraction_of_reads</strong></td>
<td>Float</td>
<td>Subsample to a fraction of the reads - e.g., 0.5 samples half the reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_number_of_reads</strong></td>
<td>Int</td>
<td>Subsample to a specific number of reads</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>rasusa_seed</strong></td>
<td>Int</td>
<td>Random seed to use</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>run_prefix</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_ont</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>This string is searched for in the kraken2 outputs to extract the read percentage</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>extract_unclassified</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>--detect_adapter_for_pe -g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_complete_only</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_decontaminate_mem</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_is_accession</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>host_refseq</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>taxon_id</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>The minimum quality score to keep during trimming</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trim_quality_trim_score</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_pe</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>--detect_adapter_for_pe -g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim_se</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>sam_to_sorted_bam</td>
<td><strong>min_qual</strong></td>
<td>Int</td>
<td>Minimum quality score for reads to be included in the analysis</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<h4 id="analysis-tasks">Analysis Tasks<a class="headerlink" href="#analysis-tasks" title="Permanent link">¶</a></h4>
<div class="tabbed-set tabbed-alternate" data-tabs="2:3"><input checked="checked" id="illumina-paired-end-input-data_1" name="__tabbed_2" type="radio"><input id="illumina-single-end-input-data_1" name="__tabbed_2" type="radio"><input id="ont-input-data_1" name="__tabbed_2" type="radio"><div class="tabbed-labels"><label for="illumina-paired-end-input-data_1">Illumina paired-end input data</label><label for="illumina-single-end-input-data_1">Illumina single-end input data</label><label for="ont-input-data_1">ONT input data</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<details class="task">
<summary><code>read_QC_trim</code>: Read Quality Trimming, Adapter Removal, Quantification, and Identification</summary>
<p><code>read_QC_trim</code> is a sub-workflow that removes low-quality reads, low-quality regions of reads, and sequencing adapters to improve data quality. It uses a number of tasks, described below. The differences between the PE and SE versions of the <code>read_QC_trim</code> sub-workflow lie in the default parameters, the use of two or one input read file(s), and the different output files.</p>
<details class="task">
<summary><code>HRRT</code>: Human Host Sequence Removal</summary>
<p>All reads of human origin <strong>are removed</strong>, including their mates, by using NCBI's <a href="https://github.com/ncbi/sra-human-scrubber"><strong>human read removal tool (HRRT)</strong></a>. </p>
<p>HRRT is based on the <a href="https://doi.org/10.1186/s13059-021-02490-0">SRA Taxonomy Analysis Tool</a> and employs a k-mer database constructed of k-mers from Eukaryota derived from all human RefSeq records with any k-mers found in non-Eukaryota RefSeq records subtracted from the database.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI-Scrub Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ncbi_scrub.wdl">task_ncbi_scrub.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/sra-human-scrubber">HRRT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2023/02/02/scrubbing-human-sequences-sra-submissions/">HRRT on NCBI</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_processing</code> is set to <code>"trimmomatic"</code>. To use <code>fastp</code> instead, set <code>read_processing</code> to <code>"fastp"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>Trimmomatic</code>: Read Trimming (default)</summary>
<p>Read proccessing is available via <code>Trimmomatic</code> by default.</p>
<p>Trimmomatic trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>Trimmomatic</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_trimmomatic.wdl">task_trimmomatic.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/usadellab/Trimmomatic">Trimmomatic on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic Website</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590/">Trimmomatic: a flexible trimmer for Illumina sequence data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>fastp</code>: Read Trimming (alternative)</summary>
<p>To activate this task, set <code>read_processing</code> to <code>"fastp"</code>.</p>
<p><code>fastp</code> trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<p><code>fastp</code> also has additional default parameters and features that are not a part of <code>trimmomatic</code>'s default configuration.</p>
<details class="toggle">
<summary><code>fastp</code> default read-trimming parameters</summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-g</td>
<td>enables polyG tail trimming</td>
</tr>
<tr>
<td>-5 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>-3 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>--detect_adapter_for_pe</td>
<td>enables adapter-trimming <strong>only for paired-end reads</strong></td>
</tr>
</tbody>
</table>
<p>Additional arguments can be passed using the <code>fastp_args</code> optional parameter.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Trimmomatic and fastp Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_fastp.wdl">task_fastp.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/17/i884/5093234?login=false">fastp: an ultra-fast all-in-one FASTQ preprocessor</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>BBDuk</code>: Adapter Trimming and PhiX Removal</summary>
<p>Adapters are manufactured oligonucleotide sequences attached to DNA fragments during the library preparation process. In Illumina sequencing, these adapter sequences are required for attaching reads to flow cells. You can read more about <a href="https://emea.support.illumina.com/bulletins/2020/06/illumina-adapter-portfolio.html">Illumina adapters here</a>. For genome analysis, it's important to remove these sequences since they're not actually from your sample. If you don't remove them, the downstream analysis may be affected.</p>
<p>The <code>bbduk</code> task removes adapters from sequence reads. To do this:</p>
<ul>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/repair-guide/">Repair</a> from the <a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/">BBTools</a> package reorders reads in paired fastq files to ensure the forward and reverse reads of a pair are in the same position in the two fastq files (it <em>re-pairs</em>).</li>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk</a>  (<em>"Bestus Bioinformaticus" Decontamination Using Kmers</em>) is then used to trim the adapters and filter out all reads that have a 31-mer match to <a href="https://emea.illumina.com/products/by-type/sequencing-kits/cluster-gen-sequencing-reagents/phix-control-v3.html">PhiX</a>, which is commonly added to Illumina sequencing runs to monitor and/or improve overall run quality.</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">BBDuk Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_bbduk.wdl">task_bbduk.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://sourceforge.net/projects/bbmap/">BBMap on SourceForge</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk Guide (archived)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_qc</code> is set to <code>"fastq_scan"</code>. To use <code>fastqc</code> instead, set <code>read_qc</code> to <code>"fastqc"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>fastq-scan</code>: Read Quantification (default)</summary>
<p>Read quantification is available via <code>fastq-scan</code> by default.</p>
<p><code>fastq-scan</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>fastq-scan</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastq_scan.wdl">task_fastq_scan.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/fastq-scan/blob/master/README.md">fastq-scan on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>FastQC</code>: Read Quantification (alternative)</summary>
<p>To activate this task, set <code>read_qc</code> to <code>"fastqc"</code>.</p>
<p><code>FastQC</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>This tool also provides a graphical visualization of the read quality.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>FastQC</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastqc.wdl">task_fastqc.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/s-andrews/FastQC">FastQC on Github</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC Website</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_pe.wdl">wf_read_QC_trim_pe.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_se.wdl">wf_read_QC_trim_se.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>bwa</code> Details</summary>
<p>This task aligns the cleaned short reads (Illumina) to the reference genome provided by the user.</p>
<div class="admonition techdetails">
<p class="admonition-title">BWA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_bwa.wdl">task_bwa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/bwa">BWA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bio-bwa.sourceforge.net/">BWA Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and accurate short read alignment with Burrows-Wheeler transform</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>primer_trim</code> Details</summary>
<p>This task trims the primer sequences from the aligned bam file with iVar. The optional input, <code>keep_noprimer_reads</code>, does not have to be modified.</p>
<div class="admonition techdetails">
<p class="admonition-title">Primer Trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ivar_primer_trim.wdl">task_ivar_primer_trim.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td>https://github.com/andersen-lab/ivar</td>
</tr>
<tr>
<td>Software Documentation</td>
<td>https://andersen-lab.github.io/ivar/html/manualpage.html</td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1186/s13059-018-1618-7">An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>read_QC_trim</code>: Read Quality Trimming, Adapter Removal, Quantification, and Identification</summary>
<p><code>read_QC_trim</code> is a sub-workflow that removes low-quality reads, low-quality regions of reads, and sequencing adapters to improve data quality. It uses a number of tasks, described below. The differences between the PE and SE versions of the <code>read_QC_trim</code> sub-workflow lie in the default parameters, the use of two or one input read file(s), and the different output files.</p>
<details class="task">
<summary><code>HRRT</code>: Human Host Sequence Removal</summary>
<p>All reads of human origin <strong>are removed</strong>, including their mates, by using NCBI's <a href="https://github.com/ncbi/sra-human-scrubber"><strong>human read removal tool (HRRT)</strong></a>. </p>
<p>HRRT is based on the <a href="https://doi.org/10.1186/s13059-021-02490-0">SRA Taxonomy Analysis Tool</a> and employs a k-mer database constructed of k-mers from Eukaryota derived from all human RefSeq records with any k-mers found in non-Eukaryota RefSeq records subtracted from the database.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI-Scrub Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ncbi_scrub.wdl">task_ncbi_scrub.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/sra-human-scrubber">HRRT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2023/02/02/scrubbing-human-sequences-sra-submissions/">HRRT on NCBI</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_processing</code> is set to <code>"trimmomatic"</code>. To use <code>fastp</code> instead, set <code>read_processing</code> to <code>"fastp"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>Trimmomatic</code>: Read Trimming (default)</summary>
<p>Read proccessing is available via <code>Trimmomatic</code> by default.</p>
<p>Trimmomatic trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>Trimmomatic</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_trimmomatic.wdl">task_trimmomatic.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/usadellab/Trimmomatic">Trimmomatic on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic Website</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590/">Trimmomatic: a flexible trimmer for Illumina sequence data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>fastp</code>: Read Trimming (alternative)</summary>
<p>To activate this task, set <code>read_processing</code> to <code>"fastp"</code>.</p>
<p><code>fastp</code> trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<p><code>fastp</code> also has additional default parameters and features that are not a part of <code>trimmomatic</code>'s default configuration.</p>
<details class="toggle">
<summary><code>fastp</code> default read-trimming parameters</summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-g</td>
<td>enables polyG tail trimming</td>
</tr>
<tr>
<td>-5 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>-3 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>--detect_adapter_for_pe</td>
<td>enables adapter-trimming <strong>only for paired-end reads</strong></td>
</tr>
</tbody>
</table>
<p>Additional arguments can be passed using the <code>fastp_args</code> optional parameter.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Trimmomatic and fastp Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_fastp.wdl">task_fastp.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/17/i884/5093234?login=false">fastp: an ultra-fast all-in-one FASTQ preprocessor</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>BBDuk</code>: Adapter Trimming and PhiX Removal</summary>
<p>Adapters are manufactured oligonucleotide sequences attached to DNA fragments during the library preparation process. In Illumina sequencing, these adapter sequences are required for attaching reads to flow cells. You can read more about <a href="https://emea.support.illumina.com/bulletins/2020/06/illumina-adapter-portfolio.html">Illumina adapters here</a>. For genome analysis, it's important to remove these sequences since they're not actually from your sample. If you don't remove them, the downstream analysis may be affected.</p>
<p>The <code>bbduk</code> task removes adapters from sequence reads. To do this:</p>
<ul>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/repair-guide/">Repair</a> from the <a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/">BBTools</a> package reorders reads in paired fastq files to ensure the forward and reverse reads of a pair are in the same position in the two fastq files (it <em>re-pairs</em>).</li>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk</a>  (<em>"Bestus Bioinformaticus" Decontamination Using Kmers</em>) is then used to trim the adapters and filter out all reads that have a 31-mer match to <a href="https://emea.illumina.com/products/by-type/sequencing-kits/cluster-gen-sequencing-reagents/phix-control-v3.html">PhiX</a>, which is commonly added to Illumina sequencing runs to monitor and/or improve overall run quality.</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">BBDuk Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_bbduk.wdl">task_bbduk.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://sourceforge.net/projects/bbmap/">BBMap on SourceForge</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk Guide (archived)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_qc</code> is set to <code>"fastq_scan"</code>. To use <code>fastqc</code> instead, set <code>read_qc</code> to <code>"fastqc"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>fastq-scan</code>: Read Quantification (default)</summary>
<p>Read quantification is available via <code>fastq-scan</code> by default.</p>
<p><code>fastq-scan</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>fastq-scan</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastq_scan.wdl">task_fastq_scan.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/fastq-scan/blob/master/README.md">fastq-scan on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>FastQC</code>: Read Quantification (alternative)</summary>
<p>To activate this task, set <code>read_qc</code> to <code>"fastqc"</code>.</p>
<p><code>FastQC</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>This tool also provides a graphical visualization of the read quality.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>FastQC</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastqc.wdl">task_fastqc.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/s-andrews/FastQC">FastQC on Github</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC Website</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_pe.wdl">wf_read_QC_trim_pe.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_se.wdl">wf_read_QC_trim_se.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>bwa</code> Details</summary>
<p>This task aligns the cleaned short reads (Illumina) to the reference genome provided by the user.</p>
<div class="admonition techdetails">
<p class="admonition-title">BWA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_bwa.wdl">task_bwa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/bwa">BWA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bio-bwa.sourceforge.net/">BWA Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and accurate short read alignment with Burrows-Wheeler transform</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>primer_trim</code> Details</summary>
<p>This task trims the primer sequences from the aligned bam file with iVar. The optional input, <code>keep_noprimer_reads</code>, does not have to be modified.</p>
<div class="admonition techdetails">
<p class="admonition-title">Primer Trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ivar_primer_trim.wdl">task_ivar_primer_trim.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td>https://github.com/andersen-lab/ivar</td>
</tr>
<tr>
<td>Software Documentation</td>
<td>https://andersen-lab.github.io/ivar/html/manualpage.html</td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1186/s13059-018-1618-7">An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>read_QC_trim_ont</code>: Read Quality Trimming, Quantification, and Identification</summary>
<p><code>read_QC_trim_ont</code> is a sub-workflow that filters low-quality reads and trims low-quality regions of reads. It uses several tasks, described below.</p>
<details class="task">
<summary><code>HRRT</code>: Human Host Sequence Removal</summary>
<p>All reads of human origin <strong>are removed</strong>, including their mates, by using NCBI's <a href="https://github.com/ncbi/sra-human-scrubber"><strong>human read removal tool (HRRT)</strong></a>. </p>
<p>HRRT is based on the <a href="https://doi.org/10.1186/s13059-021-02490-0">SRA Taxonomy Analysis Tool</a> and employs a k-mer database constructed of k-mers from Eukaryota derived from all human RefSeq records with any k-mers found in non-Eukaryota RefSeq records subtracted from the database.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI-Scrub Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ncbi_scrub.wdl">task_ncbi_scrub.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/sra-human-scrubber">HRRT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2023/02/02/scrubbing-human-sequences-sra-submissions/">HRRT on NCBI</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>artic_guppyplex</code>: Read Filtering</summary>
<p>Reads are filtered by length with <code>artic_guppyplex</code>, which is a part of the <a href="https://artic.network/fieldbioinformatics/fieldbioinformatics-sop.html"><code>ARTIC</code> protocol</a>. Since TheiaCoV was developed primarily for amplicon-based viral sequencing, this task is included to remove chimeric reads that are either too short or too long.</p>
<div class="admonition techdetails">
<p class="admonition-title">artic_guppyplex Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_artic_guppyplex.wdl">task_artic_guppyplex.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/artic-network/fieldbioinformatics/">ARTIC on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://artic.readthedocs.io/en/latest/?badge=latest">ARTIC Documentation</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Kraken2</code>: Read Identification</summary>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<p>Kraken2 is run on both the raw and clean reads.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">Kraken2 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>NanoPlot</code>: Read Quantification</summary>
<p>NanoPlot is used for the determination of mean quality scores, read lengths, and number of reads. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>While this task currently is run <em>outside</em> of the <code>read_QC_trim_ont</code> workflow, it is being included here as it calculates statistics on the read data. This is done so that the actual assembly genome lengths can be used (if an estimated genome length is not provided by the user) to ensure the estimated coverage statistics are accurate.</p>
<div class="admonition techdetails">
<p class="admonition-title">NanoPlot Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_nanoplot.wdl">task_nanoplot.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/wdecoster/NanoPlot">NanoPlot on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/wdecoster/NanoPlot/blob/master/README.md">NanoPlot Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/39/5/btad311/7160911">NanoPack2: population-scale evaluation of long-read sequencing data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim_ont Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_ont.wdl">wf_read_QC_trim_ont.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>minimap2</code>: Read Alignment Details</summary>
<p><code>minimap2</code> is a popular aligner that is used to align reads (or assemblies) to an assembly file. In minimap2, "modes" are a group of preset options.</p>
<p>The mode used in this task is <code>map-ont</code> which is the default mode for long reads and indicates that long reads of ~10% error rates should be aligned to the reference genome. The output file is in SAM format.</p>
<p>For more information regarding modes and the available options for <code>minimap2</code>, please see the <a href="https://lh3.github.io/minimap2/minimap2.html">minimap2 manpage</a></p>
<div class="admonition techdetails">
<p class="admonition-title">minimap2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_minimap2.wdl">task_minimap2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/minimap2">minimap2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://lh3.github.io/minimap2">minimap2</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/18/3094/4994778">Minimap2: pairwise alignment for nucleotide sequences</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
</div>
</div>
<details class="task">
<summary><code>freyja</code> Details</summary>
<p>The Freyja task will call variants and capture sequencing depth information to identify the relative abundance of lineages present. Optionally, if <code>bootstrap</code> is set to true, bootstrapping will be performed. After the optional bootstrapping step, the variants are demixed.</p>
<div class="admonition techdetails">
<p class="admonition-title">Freyja Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/freyja/task_freyja.wdl">task_freyja_one_sample.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/andersen-lab/Freyja">https://github.com/andersen-lab/Freyja</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://andersen-lab.github.io/Freyja/index.html#">https://andersen-lab.github.io/Freyja/index.html#</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h4 id="outputs_1">Outputs<a class="headerlink" href="#outputs_1" title="Permanent link">¶</a></h4>
<p>The main output file used in subsequent Freyja workflows is found under the <code>freyja_demixed</code> column. This TSV file takes on the following format:</p>
<table>
<thead>
<tr>
<th></th>
<th>sample name</th>
</tr>
</thead>
<tbody>
<tr>
<td>summarized</td>
<td>[('Delta', 0.65), ('Other', 0.25), ('Alpha', 0.1')]</td>
</tr>
<tr>
<td>lineages</td>
<td>['B.1.617.2' 'B.1.2' 'AY.6' 'Q.3']</td>
</tr>
<tr>
<td>abundances</td>
<td>"[0.5 0.25 0.15 0.1]"</td>
</tr>
<tr>
<td>resid</td>
<td>3.14159</td>
</tr>
<tr>
<td>coverage</td>
<td>95.8</td>
</tr>
</tbody>
</table>
<ul>
<li>The <code>summarized</code>&nbsp;array denotes a sum of all lineage abundances in a particular WHO designation (i.e. B.1.617.2 and AY.6 abundances are summed in the above example), otherwise they are grouped into "Other".</li>
<li>The <code>lineage</code>&nbsp;array lists the identified lineages in descending order</li>
<li>The <code>abundances</code>&nbsp;array contains the corresponding abundances estimates.</li>
<li>The value of&nbsp;<code>resid</code>&nbsp;corresponds to the residual of the weighted least absolute deviation problem used to estimate lineage abundances.</li>
<li>The&nbsp;<code>coverage</code>&nbsp;value provides the 10x coverage estimate (percent of sites with 10 or greater reads)</li>
</ul>
<div class="admonition tip">
<p class="admonition-title">Click "Ignore empty outputs"</p>
<p>When running the Freyja_FASTQ_PHB workflow, it is recommended to select the "Ignore empty outputs" option in the Terra UI. This will hide the output columns that will not be generated for your input data type.</p>
</div>
<div class="tabbed-set tabbed-alternate" data-tabs="3:3"><input checked="checked" id="illumina-paired-end-output-data" name="__tabbed_3" type="radio"><input id="illumina-single-end-output-data" name="__tabbed_3" type="radio"><input id="ont-output-data" name="__tabbed_3" type="radio"><div class="tabbed-labels"><label for="illumina-paired-end-output-data">Illumina paired-end output data</label><label for="illumina-single-end-output-data">Illumina single-end output data</label><label for="ont-output-data">ONT output data</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>aligned_bai</td>
<td><strong>String</strong></td>
<td>Index companion file to the bam file generated during the consensus assembly process</td>
</tr>
<tr>
<td>aligned_bam</td>
<td><strong>String</strong></td>
<td>Sorted BAM file containing the alignments of reads to the reference genome</td>
</tr>
<tr>
<td>alignment_method</td>
<td><strong>String</strong></td>
<td>The method used to generate the alignment</td>
</tr>
<tr>
<td>bbduk_docker</td>
<td><strong>String</strong></td>
<td>The Docker image for bbduk, which was used to remove the adapters from the sequences</td>
</tr>
<tr>
<td>bwa_version</td>
<td><strong>String</strong></td>
<td>Version of BWA software used</td>
</tr>
<tr>
<td>fastp_html_report</td>
<td><strong>String</strong></td>
<td>The HTML report made with fastp</td>
</tr>
<tr>
<td>fastp_version</td>
<td><strong>String</strong></td>
<td>The version of fastp used</td>
</tr>
<tr>
<td>fastq_scan_clean1_json</td>
<td><strong>String</strong></td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_clean2_json</td>
<td><strong>File</strong></td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean reverse read quality and length</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean1</td>
<td><strong>String</strong></td>
<td>The number of forward reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean2</td>
<td><strong>Int</strong></td>
<td>The number of reverse reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean_pairs</td>
<td><strong>String</strong></td>
<td>The number of read pairs after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw1</td>
<td><strong>String</strong></td>
<td>The number of input forward reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw2</td>
<td><strong>Int</strong></td>
<td>The number of input reserve reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw_pairs</td>
<td><strong>String</strong></td>
<td>The number of input read pairs as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_raw1_json</td>
<td><strong>String</strong></td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_raw2_json</td>
<td><strong>File</strong></td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw reverse read quality and length</td>
</tr>
<tr>
<td>fastq_scan_version</td>
<td><strong>String</strong></td>
<td>The version of fastq_scan</td>
</tr>
<tr>
<td>fastqc_clean1_html</td>
<td><strong>String</strong></td>
<td>An HTML file that provides a graphical visualization of clean forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_clean2_html</td>
<td><strong>File</strong></td>
<td>An HTML file that provides a graphical visualization of clean reverse read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_docker</td>
<td><strong>String</strong></td>
<td>The Docker container used for fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean1</td>
<td><strong>String</strong></td>
<td>The number of forward reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean2</td>
<td><strong>Int</strong></td>
<td>The number of reverse reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean_pairs</td>
<td><strong>String</strong></td>
<td>The number of read pairs after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_raw1</td>
<td><strong>String</strong></td>
<td>The number of input forward reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_num_reads_raw2</td>
<td><strong>Int</strong></td>
<td>The number of input reverse reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_num_reads_raw_pairs</td>
<td><strong>String</strong></td>
<td>The number of input read pairs by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_raw1_html</td>
<td><strong>String</strong></td>
<td>An HTML file that provides a graphical visualization of raw forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_raw2_html</td>
<td><strong>File</strong></td>
<td>An HTML file that provides a graphical visualization of raw reverse read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_version</td>
<td><strong>String</strong></td>
<td>Version of fastqc software used</td>
</tr>
<tr>
<td>freyja_abundances</td>
<td><strong>String</strong></td>
<td>Abundances estimates identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_barcode_file</td>
<td><strong>String</strong></td>
<td>Barcode file used with Freyja</td>
</tr>
<tr>
<td>freyja_barcode_version</td>
<td><strong>String</strong></td>
<td>Name of barcode file used, or the date if update_db is true</td>
</tr>
<tr>
<td>freyja_bootstrap_lineages</td>
<td><strong>String</strong></td>
<td>A CSV that contains the 0.025, 0.05, 0.25, 0.5 (median), 0.75, 0.95, and 0.975 percentiles for each lineage</td>
</tr>
<tr>
<td>freyja_bootstrap_lineages_pdf</td>
<td><strong>String</strong></td>
<td>A boxplot of the bootstrap lineages CSV file</td>
</tr>
<tr>
<td>freyja_bootstrap_summary</td>
<td><strong>String</strong></td>
<td>A CSV that contains the 0.025, 0.05, 0.25, 0.5 (median), 0.75, 0.95, and 0.975 percentiles for each WHO designated VOI/VOC</td>
</tr>
<tr>
<td>freyja_bootstrap_summary_pdf</td>
<td><strong>String</strong></td>
<td>A boxplot of the bootstrap summary CSV file</td>
</tr>
<tr>
<td>freyja_coverage</td>
<td><strong>Float</strong></td>
<td>Coverage identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_demixed</td>
<td><strong>File</strong></td>
<td>The main output TSV; see the section directly above this table for an explanation</td>
</tr>
<tr>
<td>freyja_demixed_parsed</td>
<td><strong>File</strong></td>
<td>Parsed freyja_demixed file, containing the same information, for easy result concatenation</td>
</tr>
<tr>
<td>freyja_depths</td>
<td><strong>File</strong></td>
<td>A TSV listing the depth of every position</td>
</tr>
<tr>
<td>freyja_fastq_wf_analysis_date</td>
<td><strong>String</strong></td>
<td>Date of analysis</td>
</tr>
<tr>
<td>freyja_fastq_wf_version</td>
<td><strong>String</strong></td>
<td>The version of the Public Health Bioinformatics (PHB) repository used</td>
</tr>
<tr>
<td>freyja_lineage_metadata_file</td>
<td><strong>String</strong></td>
<td>Metadata file for lineages identified by Freyja</td>
</tr>
<tr>
<td>freyja_lineages</td>
<td><strong>String</strong></td>
<td>Lineages in descending order identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_metadata_version</td>
<td><strong>String</strong></td>
<td>Name of lineage metadata file used, or the date if update_db is true</td>
</tr>
<tr>
<td>freyja_resid</td>
<td><strong>String</strong></td>
<td>Residual of the weighted least absolute deviation problem used to estimate lineage abundances identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_summarized</td>
<td><strong>String</strong></td>
<td>Sum of all lineage abundances in a particular WHO designation identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_variants</td>
<td><strong>File</strong></td>
<td>The TSV file containing the variants identified by Freyja</td>
</tr>
<tr>
<td>freyja_version</td>
<td><strong>String</strong></td>
<td>version of Freyja used</td>
</tr>
<tr>
<td>ivar_version_primtrim</td>
<td><strong>String</strong></td>
<td>Version of iVar for running the iVar trim command</td>
</tr>
<tr>
<td>kraken_human</td>
<td><strong>Float</strong></td>
<td>Percent of human read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_human_dehosted</td>
<td><strong>Float</strong></td>
<td>Percent of human read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_report</td>
<td><strong>String</strong></td>
<td>Full Kraken report</td>
</tr>
<tr>
<td>kraken_report_dehosted</td>
<td><strong>File</strong></td>
<td>Full Kraken report after host removal</td>
</tr>
<tr>
<td>kraken_sc2</td>
<td><strong>String</strong></td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_sc2_dehosted</td>
<td><strong>String</strong></td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_version</td>
<td><strong>String</strong></td>
<td>Version of Kraken software used</td>
</tr>
<tr>
<td>primer_bed_name</td>
<td><strong>String</strong></td>
<td>Name of the primer bed files used for primer trimming</td>
</tr>
<tr>
<td>primer_trimmed_read_percent</td>
<td><strong>Float</strong></td>
<td>Percentage of read data with primers trimmed as determined by iVar trim</td>
</tr>
<tr>
<td>read1_clean</td>
<td><strong>File</strong></td>
<td>Forward read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read1_dehosted</td>
<td><strong>File</strong></td>
<td>The dehosted forward reads file; suggested read file for SRA submission</td>
</tr>
<tr>
<td>read2_clean</td>
<td><strong>File</strong></td>
<td>Reverse read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read2_dehosted</td>
<td><strong>File</strong></td>
<td>The dehosted reverse reads file; suggested read file for SRA submission</td>
</tr>
<tr>
<td>samtools_version</td>
<td><strong>String</strong></td>
<td>The version of SAMtools used to sort and index the alignment file</td>
</tr>
<tr>
<td>samtools_version_primtrim</td>
<td><strong>String</strong></td>
<td>The version of SAMtools used to create the pileup before running iVar trim</td>
</tr>
<tr>
<td>trimmomatic_docker</td>
<td><strong>String</strong></td>
<td>The docker image used for the trimmomatic module in this workflow</td>
</tr>
<tr>
<td>trimmomatic_version</td>
<td><strong>String</strong></td>
<td>The version of Trimmomatic used</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>aligned_bai</td>
<td><strong>String</strong></td>
<td>Index companion file to the bam file generated during the consensus assembly process</td>
</tr>
<tr>
<td>aligned_bam</td>
<td><strong>String</strong></td>
<td>Sorted BAM file containing the alignments of reads to the reference genome</td>
</tr>
<tr>
<td>alignment_method</td>
<td><strong>String</strong></td>
<td>The method used to generate the alignment</td>
</tr>
<tr>
<td>bbduk_docker</td>
<td><strong>String</strong></td>
<td>The Docker image for bbduk, which was used to remove the adapters from the sequences</td>
</tr>
<tr>
<td>bwa_version</td>
<td><strong>String</strong></td>
<td>Version of BWA software used</td>
</tr>
<tr>
<td>fastp_html_report</td>
<td><strong>String</strong></td>
<td>The HTML report made with fastp</td>
</tr>
<tr>
<td>fastp_version</td>
<td><strong>String</strong></td>
<td>The version of fastp used</td>
</tr>
<tr>
<td>fastq_scan_clean1_json</td>
<td><strong>String</strong></td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean1</td>
<td><strong>String</strong></td>
<td>The number of forward reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw1</td>
<td><strong>String</strong></td>
<td>The number of input forward reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_raw1_json</td>
<td><strong>String</strong></td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_version</td>
<td><strong>String</strong></td>
<td>The version of fastq_scan</td>
</tr>
<tr>
<td>fastqc_clean1_html</td>
<td><strong>String</strong></td>
<td>An HTML file that provides a graphical visualization of clean forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_docker</td>
<td><strong>String</strong></td>
<td>The Docker container used for fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean1</td>
<td><strong>String</strong></td>
<td>The number of forward reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_raw1</td>
<td><strong>String</strong></td>
<td>The number of input forward reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_raw1_html</td>
<td><strong>String</strong></td>
<td>An HTML file that provides a graphical visualization of raw forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_version</td>
<td><strong>String</strong></td>
<td>Version of fastqc software used</td>
</tr>
<tr>
<td>freyja_abundances</td>
<td><strong>String</strong></td>
<td>Abundances estimates identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_barcode_file</td>
<td><strong>String</strong></td>
<td>Barcode file used with Freyja</td>
</tr>
<tr>
<td>freyja_barcode_version</td>
<td><strong>String</strong></td>
<td>Name of barcode file used, or the date if update_db is true</td>
</tr>
<tr>
<td>freyja_bootstrap_lineages</td>
<td><strong>String</strong></td>
<td>A CSV that contains the 0.025, 0.05, 0.25, 0.5 (median), 0.75, 0.95, and 0.975 percentiles for each lineage</td>
</tr>
<tr>
<td>freyja_bootstrap_lineages_pdf</td>
<td><strong>String</strong></td>
<td>A boxplot of the bootstrap lineages CSV file</td>
</tr>
<tr>
<td>freyja_bootstrap_summary</td>
<td><strong>String</strong></td>
<td>A CSV that contains the 0.025, 0.05, 0.25, 0.5 (median), 0.75, 0.95, and 0.975 percentiles for each WHO designated VOI/VOC</td>
</tr>
<tr>
<td>freyja_bootstrap_summary_pdf</td>
<td><strong>String</strong></td>
<td>A boxplot of the bootstrap summary CSV file</td>
</tr>
<tr>
<td>freyja_coverage</td>
<td><strong>Float</strong></td>
<td>Coverage identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_demixed</td>
<td><strong>File</strong></td>
<td>The main output TSV; see the section directly above this table for an explanation</td>
</tr>
<tr>
<td>freyja_demixed_parsed</td>
<td><strong>File</strong></td>
<td>Parsed freyja_demixed file, containing the same information, for easy result concatenation</td>
</tr>
<tr>
<td>freyja_depths</td>
<td><strong>File</strong></td>
<td>A TSV listing the depth of every position</td>
</tr>
<tr>
<td>freyja_fastq_wf_analysis_date</td>
<td><strong>String</strong></td>
<td>Date of analysis</td>
</tr>
<tr>
<td>freyja_fastq_wf_version</td>
<td><strong>String</strong></td>
<td>The version of the Public Health Bioinformatics (PHB) repository used</td>
</tr>
<tr>
<td>freyja_lineage_metadata_file</td>
<td><strong>String</strong></td>
<td>Metadata file for lineages identified by Freyja</td>
</tr>
<tr>
<td>freyja_lineages</td>
<td><strong>String</strong></td>
<td>Lineages in descending order identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_metadata_version</td>
<td><strong>String</strong></td>
<td>Name of lineage metadata file used, or the date if update_db is true</td>
</tr>
<tr>
<td>freyja_resid</td>
<td><strong>String</strong></td>
<td>Residual of the weighted least absolute deviation problem used to estimate lineage abundances identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_summarized</td>
<td><strong>String</strong></td>
<td>Sum of all lineage abundances in a particular WHO designation identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_variants</td>
<td><strong>File</strong></td>
<td>The TSV file containing the variants identified by Freyja</td>
</tr>
<tr>
<td>freyja_version</td>
<td><strong>String</strong></td>
<td>version of Freyja used</td>
</tr>
<tr>
<td>ivar_version_primtrim</td>
<td><strong>String</strong></td>
<td>Version of iVar for running the iVar trim command</td>
</tr>
<tr>
<td>kraken_human</td>
<td><strong>Float</strong></td>
<td>Percent of human read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_human_dehosted</td>
<td><strong>Float</strong></td>
<td>Percent of human read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_report</td>
<td><strong>String</strong></td>
<td>Full Kraken report</td>
</tr>
<tr>
<td>kraken_report_dehosted</td>
<td><strong>File</strong></td>
<td>Full Kraken report after host removal</td>
</tr>
<tr>
<td>kraken_sc2</td>
<td><strong>String</strong></td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_sc2_dehosted</td>
<td><strong>String</strong></td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_version</td>
<td><strong>String</strong></td>
<td>Version of Kraken software used</td>
</tr>
<tr>
<td>primer_bed_name</td>
<td><strong>String</strong></td>
<td>Name of the primer bed files used for primer trimming</td>
</tr>
<tr>
<td>primer_trimmed_read_percent</td>
<td><strong>Float</strong></td>
<td>Percentage of read data with primers trimmed as determined by iVar trim</td>
</tr>
<tr>
<td>samtools_version</td>
<td><strong>String</strong></td>
<td>The version of SAMtools used to sort and index the alignment file</td>
</tr>
<tr>
<td>samtools_version_primtrim</td>
<td><strong>String</strong></td>
<td>The version of SAMtools used to create the pileup before running iVar trim</td>
</tr>
<tr>
<td>trimmomatic_docker</td>
<td><strong>String</strong></td>
<td>The docker image used for the trimmomatic module in this workflow</td>
</tr>
<tr>
<td>trimmomatic_version</td>
<td><strong>String</strong></td>
<td>The version of Trimmomatic used</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>aligned_bai</td>
<td><strong>String</strong></td>
<td>Index companion file to the bam file generated during the consensus assembly process</td>
</tr>
<tr>
<td>aligned_bam</td>
<td><strong>String</strong></td>
<td>Sorted BAM file containing the alignments of reads to the reference genome</td>
</tr>
<tr>
<td>alignment_method</td>
<td><strong>String</strong></td>
<td>The method used to generate the alignment</td>
</tr>
<tr>
<td>freyja_abundances</td>
<td><strong>String</strong></td>
<td>Abundances estimates identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_barcode_file</td>
<td><strong>String</strong></td>
<td>Barcode file used with Freyja</td>
</tr>
<tr>
<td>freyja_barcode_version</td>
<td><strong>String</strong></td>
<td>Name of barcode file used, or the date if update_db is true</td>
</tr>
<tr>
<td>freyja_bootstrap_lineages</td>
<td><strong>String</strong></td>
<td>A CSV that contains the 0.025, 0.05, 0.25, 0.5 (median), 0.75, 0.95, and 0.975 percentiles for each lineage</td>
</tr>
<tr>
<td>freyja_bootstrap_lineages_pdf</td>
<td><strong>String</strong></td>
<td>A boxplot of the bootstrap lineages CSV file</td>
</tr>
<tr>
<td>freyja_bootstrap_summary</td>
<td><strong>String</strong></td>
<td>A CSV that contains the 0.025, 0.05, 0.25, 0.5 (median), 0.75, 0.95, and 0.975 percentiles for each WHO designated VOI/VOC</td>
</tr>
<tr>
<td>freyja_bootstrap_summary_pdf</td>
<td><strong>String</strong></td>
<td>A boxplot of the bootstrap summary CSV file</td>
</tr>
<tr>
<td>freyja_coverage</td>
<td><strong>Float</strong></td>
<td>Coverage identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_demixed</td>
<td><strong>File</strong></td>
<td>The main output TSV; see the section directly above this table for an explanation</td>
</tr>
<tr>
<td>freyja_demixed_parsed</td>
<td><strong>File</strong></td>
<td>Parsed freyja_demixed file, containing the same information, for easy result concatenation</td>
</tr>
<tr>
<td>freyja_depths</td>
<td><strong>File</strong></td>
<td>A TSV listing the depth of every position</td>
</tr>
<tr>
<td>freyja_fastq_wf_analysis_date</td>
<td><strong>String</strong></td>
<td>Date of analysis</td>
</tr>
<tr>
<td>freyja_fastq_wf_version</td>
<td><strong>String</strong></td>
<td>The version of the Public Health Bioinformatics (PHB) repository used</td>
</tr>
<tr>
<td>freyja_lineage_metadata_file</td>
<td><strong>String</strong></td>
<td>Metadata file for lineages identified by Freyja</td>
</tr>
<tr>
<td>freyja_lineages</td>
<td><strong>String</strong></td>
<td>Lineages in descending order identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_metadata_version</td>
<td><strong>String</strong></td>
<td>Name of lineage metadata file used, or the date if update_db is true</td>
</tr>
<tr>
<td>freyja_resid</td>
<td><strong>String</strong></td>
<td>Residual of the weighted least absolute deviation problem used to estimate lineage abundances identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_summarized</td>
<td><strong>String</strong></td>
<td>Sum of all lineage abundances in a particular WHO designation identified by Freyja and parsed from freyja_demixed file</td>
</tr>
<tr>
<td>freyja_variants</td>
<td><strong>File</strong></td>
<td>The TSV file containing the variants identified by Freyja</td>
</tr>
<tr>
<td>freyja_version</td>
<td><strong>String</strong></td>
<td>version of Freyja used</td>
</tr>
<tr>
<td>ivar_version_primtrim</td>
<td><strong>String</strong></td>
<td>Version of iVar for running the iVar trim command</td>
</tr>
<tr>
<td>kraken_human</td>
<td><strong>Float</strong></td>
<td>Percent of human read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_human_dehosted</td>
<td><strong>Float</strong></td>
<td>Percent of human read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_report</td>
<td><strong>String</strong></td>
<td>Full Kraken report</td>
</tr>
<tr>
<td>kraken_report_dehosted</td>
<td><strong>File</strong></td>
<td>Full Kraken report after host removal</td>
</tr>
<tr>
<td>kraken_sc2</td>
<td><strong>String</strong></td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_sc2_dehosted</td>
<td><strong>String</strong></td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_version</td>
<td><strong>String</strong></td>
<td>Version of Kraken software used</td>
</tr>
<tr>
<td>minimap2_docker</td>
<td><strong>String</strong></td>
<td>The Docker image of minimap2</td>
</tr>
<tr>
<td>minimap2_version</td>
<td><strong>String</strong></td>
<td>The version of minimap2</td>
</tr>
<tr>
<td>nanoplot_html_clean</td>
<td><strong>File</strong></td>
<td>An HTML report describing the clean reads</td>
</tr>
<tr>
<td>nanoplot_html_raw</td>
<td><strong>File</strong></td>
<td>An HTML report describing the raw reads</td>
</tr>
<tr>
<td>nanoplot_num_reads_clean1</td>
<td><strong>Int</strong></td>
<td>Number of clean reads</td>
</tr>
<tr>
<td>nanoplot_num_reads_raw1</td>
<td><strong>Int</strong></td>
<td>Number of raw reads</td>
</tr>
<tr>
<td>nanoplot_r1_est_coverage_clean</td>
<td><strong>Float</strong></td>
<td>Estimated coverage on the clean reads by nanoplot</td>
</tr>
<tr>
<td>nanoplot_r1_est_coverage_raw</td>
<td><strong>Float</strong></td>
<td>Estimated coverage on the raw reads by nanoplot</td>
</tr>
<tr>
<td>nanoplot_r1_mean_q_clean</td>
<td><strong>Float</strong></td>
<td>Mean quality score of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_q_raw</td>
<td><strong>Float</strong></td>
<td>Mean quality score of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_readlength_clean</td>
<td><strong>Float</strong></td>
<td>Mean read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_readlength_raw</td>
<td><strong>Float</strong></td>
<td>Mean read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_q_clean</td>
<td><strong>Float</strong></td>
<td>Median quality score of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_q_raw</td>
<td><strong>Float</strong></td>
<td>Median quality score of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_readlength_clean</td>
<td><strong>Float</strong></td>
<td>Median read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_readlength_raw</td>
<td><strong>Float</strong></td>
<td>Median read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_n50_clean</td>
<td><strong>Float</strong></td>
<td>N50 of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_n50_raw</td>
<td><strong>Float</strong></td>
<td>N50 of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_stdev_readlength_clean</td>
<td><strong>Float</strong></td>
<td>Standard deviation read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_stdev_readlength_raw</td>
<td><strong>Float</strong></td>
<td>Standard deviation read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_tsv_clean</td>
<td><strong>File</strong></td>
<td>A TSV report describing the clean reads</td>
</tr>
<tr>
<td>nanoplot_tsv_raw</td>
<td><strong>File</strong></td>
<td>A TSV report describing the raw reads</td>
</tr>
<tr>
<td>nanoq_version</td>
<td><strong>String</strong></td>
<td>Version of nanoq used in analysis</td>
</tr>
<tr>
<td>primer_bed_name</td>
<td><strong>String</strong></td>
<td>Name of the primer bed files used for primer trimming</td>
</tr>
<tr>
<td>primer_trimmed_read_percent</td>
<td><strong>Float</strong></td>
<td>Percentage of read data with primers trimmed as determined by iVar trim</td>
</tr>
<tr>
<td>samtools_version</td>
<td><strong>String</strong></td>
<td>The version of SAMtools used to sort and index the alignment file</td>
</tr>
<tr>
<td>samtools_version_primtrim</td>
<td><strong>String</strong></td>
<td>The version of SAMtools used to create the pileup before running iVar trim</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<h3 id="freyja_plot">Freyja_Plot_PHB<a class="headerlink" href="#freyja_plot" title="Permanent link">¶</a></h3>
<p>This workflow visualizes aggregated freyja_demixed output files produced by <a href="./#freyja_fastq">Freyja_FASTQ_PHB</a> in a single plot (pdf format) which provides fractional abundance estimates for all aggregated samples.</p>
<p>Options exist to provide lineage-specific breakdowns and/or sample collection time information.</p>
<h4 id="inputs_2">Inputs<a class="headerlink" href="#inputs_2" title="Permanent link">¶</a></h4>
<p>This workflow runs on the set level.</p>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>freyja_plot</td>
<td><strong>freyja_demixed</strong></td>
<td>Array[File]</td>
<td>An array containing the output files (freyja_demixed) made by Freyja_FASTQ</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_plot</td>
<td><strong>freyja_plot_name</strong></td>
<td>String</td>
<td>The name of the plot to be produced. Example: "my-freyja-plot"</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_plot</td>
<td><strong>samplename</strong></td>
<td>Array[String]</td>
<td>The names of the samples being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_plot</td>
<td><strong>collection_date</strong></td>
<td>Array[String]</td>
<td>An array containing the collection dates for the sample (YYYY-MM-DD format)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_plot_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_plot_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_plot_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/freyja:1.5.3</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_plot_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_plot_task</td>
<td><strong>mincov</strong></td>
<td>Int</td>
<td>The minimum genome coverage used as a cut-off of data to include in the plot</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_plot_task</td>
<td><strong>plot_day_window</strong></td>
<td>Int</td>
<td>The width of the rolling average window; only used if plot_time_interval is "D"</td>
<td>14</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_plot_task</td>
<td><strong>plot_lineages</strong></td>
<td>Boolean</td>
<td>If true, will plot a lineage-specific breakdown</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_plot_task</td>
<td><strong>plot_time</strong></td>
<td>Boolean</td>
<td>If true, will plot sample collection time information (requires the collection_date input variable)</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_plot_task</td>
<td><strong>plot_time_interval</strong></td>
<td>String</td>
<td>Options: "MS" for month, "D" for day</td>
<td>MS</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
<h4 id="analysis-tasks_1">Analysis Tasks<a class="headerlink" href="#analysis-tasks_1" title="Permanent link">¶</a></h4>
<details class="task">
<summary><code>freyja_plot_task</code> Details</summary>
<p>This task will aggregate multiple samples together, and then creates a plot. Several optional inputs dictate the plot appearance (see each variable's description for more information).</p>
<div class="admonition techdetails">
<p class="admonition-title">Freyja Plot Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/freyja/task_freyja_plot.wdl">wf_freyja_plot.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/andersen-lab/Freyja">https://github.com/andersen-lab/Freyja</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/andersen-lab/Freyja">https://github.com/andersen-lab/Freyja</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h4 id="outputs_2">Outputs<a class="headerlink" href="#outputs_2" title="Permanent link">¶</a></h4>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>freyja_demixed_aggregate</td>
<td>File</td>
<td>A TSV file that summarizes the <code>freyja_demixed</code> outputs for all samples</td>
</tr>
<tr>
<td>freyja_plot</td>
<td>File</td>
<td>A PDF of the plot produced by the workflow</td>
</tr>
<tr>
<td>freyja_plot_metadata</td>
<td>File</td>
<td>The metadata used to create the plot</td>
</tr>
<tr>
<td>freyja_plot_version</td>
<td>String</td>
<td>The version of Freyja used</td>
</tr>
<tr>
<td>freyja_plot_wf_analysis_date</td>
<td>String</td>
<td>The date of analysis</td>
</tr>
<tr>
<td>freyja_plot_wf_version</td>
<td>String</td>
<td>The version of the Public Health Bioinformatics (PHB) repository used</td>
</tr>
</tbody>
</table>
</div>
<h3 id="freyja_dashboard">Freyja_Dashboard_PHB<a class="headerlink" href="#freyja_dashboard" title="Permanent link">¶</a></h3>
<p>This workflow creates a group of interactive visualizations based off of the aggregated freyja_demixed output files produced by <a href="./#freyja_fastq">Freyja_FASTQ_PHB</a> called a "dashboard". Creating this dashboard requires knowing the viral load of your samples (viral copies/litre).</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This dashboard is not "live" — that is, you must rerun the workflow every time you want new data to be included in the visualizations.</p>
</div>
<h4 id="inputs_3">Inputs<a class="headerlink" href="#inputs_3" title="Permanent link">¶</a></h4>
<p>This workflow runs on the set level.</p>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>freyja_dashboard</td>
<td><strong>collection_date</strong></td>
<td>Array[String]</td>
<td>An array containing the collection dates for the sample (YYYY-MM-DD format)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_dashboard</td>
<td><strong>freyja_dashboard_title</strong></td>
<td>String</td>
<td>The name of the dashboard to be produced. Example: "my-freyja-dashboard"</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_dashboard</td>
<td><strong>freyja_demixed</strong></td>
<td>Array[File]</td>
<td>An array containing the output files (freyja_demixed) made by Freyja_FASTQ workflow</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_dashboard</td>
<td><strong>samplename</strong></td>
<td>Array[String]</td>
<td>The names of the samples being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_dashboard</td>
<td><strong>viral_load</strong></td>
<td>Array[String]</td>
<td>An array containing the number of viral copies per liter</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>freyja_dashboard_task</td>
<td><strong>config</strong></td>
<td>File</td>
<td>(found in the optional section, but is required) A yaml file that applies various configurations to the dashboard, such as grouping lineages together, applying colorings, etc. See also <a href="https://github.com/andersen-lab/Freyja/blob/main/freyja/data/plot_config.yml">https://github.com/andersen-lab/Freyja/blob/main/freyja/data/plot_config.yml</a>.</td>
<td></td>
<td>Optional, Required</td>
</tr>
<tr>
<td>freyja_dashboard</td>
<td><strong>dashboard_intro_text</strong></td>
<td>File</td>
<td>A file containing the text to be contained at the top of the dashboard.</td>
<td>SARS-CoV-2 lineage de-convolution performed by the Freyja workflow (<a href="https://github.com/andersen-lab/Freyja">https://github.com/andersen-lab/Freyja</a>).</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_dashboard_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_dashboard_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_dashboard_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/freyja:1.5.3</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_dashboard_task</td>
<td><strong>headerColor</strong></td>
<td>String</td>
<td>A hex color code to change the color of the header</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_dashboard_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_dashboard_task</td>
<td><strong>mincov</strong></td>
<td>Float</td>
<td>The minimum genome coverage used as a cut-off of data to include in the dashboard. Default is set to 60 by the freyja command-line tool (not a WDL task default, per se)</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_dashboard_task</td>
<td><strong>scale_by_viral_load</strong></td>
<td>Boolean</td>
<td>If set to true, averages samples taken the same day while taking viral load into account</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>freyja_dashboard_task</td>
<td><strong>thresh</strong></td>
<td>Float</td>
<td>The minimum lineage abundance cut-off value</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
<h4 id="analysis-tasks_2">Analysis Tasks<a class="headerlink" href="#analysis-tasks_2" title="Permanent link">¶</a></h4>
<details class="task">
<summary><code>freyja_dashboard_task</code> Details</summary>
<p>This task will aggregate multiple samples together, and then create an interactive HTML visualization. Several optional inputs dictate the dashboard appearance (see each variable's description for more information).</p>
<div class="admonition techdetails">
<p class="admonition-title">Freyja Dashboard Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/freyja/task_freyja_dashboard.wdl">wf_freyja_dashboard.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/andersen-lab/Freyja">https://github.com/andersen-lab/Freyja</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/andersen-lab/Freyja">https://github.com/andersen-lab/Freyja</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h4 id="outputs_3">Outputs<a class="headerlink" href="#outputs_3" title="Permanent link">¶</a></h4>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>freyja_dashboard</td>
<td>File</td>
<td>The HTML file of the dashboard created</td>
</tr>
<tr>
<td>freyja_dashboard_metadata</td>
<td>File</td>
<td>The metadata used to create the dashboard</td>
</tr>
<tr>
<td>freyja_dashboard_version</td>
<td>String</td>
<td>The version of Freyja used</td>
</tr>
<tr>
<td>freyja_dashboard_wf_analysis_date</td>
<td>String</td>
<td>The date of analysis</td>
</tr>
<tr>
<td>freyja_dashboard_wf_version</td>
<td>String</td>
<td>The version of the Public Health Bioinformatics (PHB) repository used</td>
</tr>
<tr>
<td>freyja_demixed_aggregate</td>
<td>File</td>
<td>A TSV file that summarizes the <code>freyja_demixed</code> outputs for all samples</td>
</tr>
</tbody>
</table>
</div>
<h2 id="running-freyja-on-other-pathogens">Running Freyja on other pathogens<a class="headerlink" href="#running-freyja-on-other-pathogens" title="Permanent link">¶</a></h2>
<div class="admonition warning">
<p class="admonition-title">Experimental Feature</p>
<p>Please be aware this is an <em>experimental feature</em> and we cannot guarantee complete functionality at this time.</p>
</div>
<p>The main requirement to run Freyja on other pathogens is <strong>the existence of a barcode file for your pathogen of interest</strong>. Currently, barcodes exist for the following organisms:</p>
<ul>
<li>SARS-CoV-2 (default)</li>
<li>FLU-B-VIC</li>
<li>H1N1</li>
<li>H3N2</li>
<li>H5Nx-cattle</li>
<li>H5NX</li>
<li>MEASLESN450</li>
<li>MEASLESgenome</li>
<li>MPX</li>
<li>RSVa</li>
<li>RSVb</li>
</ul>
<div class="admonition dna">
<p class="admonition-title">Freyja barcodes for other pathogens</p>
<p>Data for various pathogens can be found in the following repository:&nbsp;<a href="https://github.com/gp201/Freyja-barcodes">Freyja Barcodes</a></p>
<p>Folders are organized by pathogen, with each subfolder named after the date the barcode was generated, using the format YYYY-MM-DD, as well as a "latest" folder. Barcode files are named <code>barcode.csv</code>, and reference genome files are named <code>reference.fasta</code>.</p>
</div>
<p>There are two ways to run <a href="./#freyja_fastq"><strong>Freyja_FASTQ_PHB</strong></a> for non-SARS-CoV-2 organisms:</p>
<ul>
<li>Using the <code>freyja_pathogen</code> optional input (limited set of allowable organisms)</li>
<li>Providing the appropriate barcode file through the <code>freyja_barcodes</code> optional input (any organism for which barcodes are supplied)</li>
</ul>
<h3 id="using-the-freyja_pathogen-flag">Using the <code>freyja_pathogen</code> flag<a class="headerlink" href="#using-the-freyja_pathogen-flag" title="Permanent link">¶</a></h3>
<p>When using the <code>freyja_pathogen</code> flag, the user must set the optional <code>update_db</code> flag to <em>true</em>, so that the latest version of the barcode file is automatically downloaded by Freyja. </p>
<div class="admonition caption">
<p class="admonition-title">Figure 2:  Optional input for Freyja_FASTQ_PHB to provide the pathogen to be used by Freyja</p>
<h5 id="figure2">Figure 2<a class="headerlink" href="#figure2" title="Permanent link">¶</a></h5>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/Freyja_figure2.png" data-desc-position="bottom"><img alt="**Figure 2:  Optional input for Freyja_FASTQ_PHB to provide the pathogen to be used by Freyja.**" src="../../../assets/figures/Freyja_figure2.png"></a></p>
</div>
<p>Allowed options:</p>
<ul>
<li>SARS-CoV-2 (default)</li>
<li>MPXV</li>
<li>H1N1pdm</li>
<li>H5NX</li>
<li>FLU-B-VIC</li>
<li>MEASLESN450</li>
<li>MEASLES</li>
<li>RSVa</li>
<li>RSVb</li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The <code>freyja_pathogen</code> flag is not used if a barcodes file is provided. This means that this option is ignored if a barcode file is provided through <code>freyja_barcodes</code>.</p>
</div>
<h3 id="providing-the-appropriate-barcode-file">Providing the appropriate barcode file<a class="headerlink" href="#providing-the-appropriate-barcode-file" title="Permanent link">¶</a></h3>
<p>The appropriate barcode file for your organism of interest and reference sequence need to be downloaded and uploaded to your <a href="http://Terra.bio">Terra.bio</a> workspace. When running <a href="./#freyja_fastq"><strong>Freyja_FASTQ_PHB</strong></a>, the appropriate reference and barcodes file need to be passed as inputs. The first is a required input and will show up at the top of the workflows inputs page on <a href="http://Terra.bio">Terra.bio</a> (<a href="./#figure3">Figure 3</a>).</p>
<div class="admonition caption">
<p class="admonition-title">Figure 3:  Required input for Freyja_FASTQ_PHB to provide the reference genome to be used by Freyja</p>
<h5 id="figure3">Figure 3<a class="headerlink" href="#figure3" title="Permanent link">¶</a></h5>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/Freyja_figure3.png" data-desc-position="bottom"><img alt="**Figure 3:  Required input for Freyja_FASTQ_PHB to provide the reference genome to be used by Freyja.**" src="../../../assets/figures/Freyja_figure3.png"></a></p>
</div>
<p>The barcodes file can be passed directly to Freyja by the <code>freyja_barcodes</code> optional input (<a href="./#figure4">Figure 4</a>).</p>
<div class="admonition caption">
<p class="admonition-title">Figure 4: Optional input for Freyja_FASTQ_PHB to provide the barcodes file to be used by Freyja</p>
<h5 id="figure4">Figure 4<a class="headerlink" href="#figure4" title="Permanent link">¶</a></h5>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/Freyja_figure4.png" data-desc-position="bottom"><img alt="**Figure 4: Optional input for Freyja_FASTQ_PHB to provide the barcodes file to be used by Freyja.**" src="../../../assets/figures/Freyja_figure4.png"></a></p>
</div>
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">¶</a></h2>
<p>If you use any of the Freyja workflows, please cite:</p>
<blockquote>
<p>Karthikeyan, S., Levy, J.I., De Hoff, P.&nbsp;<em>et al.</em>&nbsp;Wastewater sequencing reveals early cryptic SARS-CoV-2 variant transmission.&nbsp;<em>Nature</em>&nbsp;<strong>609</strong>, 101–108 (2022). <a href="https://doi.org/10.1038/s41586-022-05049-6">https://doi.org/10.1038/s41586-022-05049-6</a></p>
</blockquote>
<!-- -->
<blockquote>
<p>Freyja source code can be found at <a href="https://github.com/andersen-lab/Freyja">https://github.com/andersen-lab/Freyja</a></p>
</blockquote>
<!-- -->
<blockquote>
<p>Freyja barcodes (non-SARS-CoV-2): <a href="https://github.com/gp201/Freyja-barcodes">https://github.com/gp201/Freyja-barcodes</a></p>
</blockquote>







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="September 10, 2025 17:12:08 UTC">2025-09-10</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="September 11, 2024 17:48:55 UTC">2024-09-11</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      © 2022-2025 <a href="https://www.theiagen.com" target="_blank" rel="noopener">Theiagen Genomics</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/theiagen" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://bsky.app/profile/theiagen.bsky.social" target="_blank" rel="noopener" title="bsky.app" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M407.8 294.7c-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3M288 227.1c-26.1-50.7-97.1-145.2-163.1-191.8C61.6-9.4 37.5-1.7 21.6 5.5 3.3 13.8 0 41.9 0 58.4S9.1 194 15 213.9c19.5 65.7 89.1 87.9 153.2 80.7 3.3-.5 6.6-.9 10-1.4-3.3.5-6.6 1-10 1.4-93.9 14-177.3 48.2-67.9 169.9C220.6 589.1 265.1 437.8 288 361.1c22.9 76.7 49.2 222.5 185.6 103.4 102.4-103.4 28.1-156-65.8-169.9-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3 64.1 7.1 133.6-15.1 153.2-80.7C566.9 194 576 75 576 58.4s-3.3-44.7-21.6-52.9c-15.8-7.1-40-14.9-103.2 29.8C385.1 81.9 314.1 176.4 288 227.1"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.linkedin.com/company/theiagen" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3M135.4 416H69V202.2h66.5V416zM102.2 96a38.5 38.5 0 1 1 0 77 38.5 38.5 0 1 1 0-77m282.1 320h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9z"></path></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "navigation.instant", "navigation.path", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.suggest", "toc.follow"], "search": "../../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"alias": true, "default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../../../javascripts/tablesort.js"></script>
      
        <script src="../../../javascripts/table-search.js"></script>
      
    
  
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>