??? task "`clair3`"

    `Clair3` performs deep learning-based variant detection using a multi-stage approach. The process begins with pileup-based calling for initial variant identification, followed by full-alignment analysis for comprehensive variant detection. Results are merged into a final high-confidence call set.

<!-- if: theiaviral-->
    The variant calling pipeline employs specialized neural networks trained on ONT data to accurately identify:
    - Single nucleotide variants (SNVs)
    - Small insertions and deletions (indels)
    - Structural variants

    ??? dna "`clair3_model` input parameter"
        This parameter specifies the clair3 model to use for variant calling. The default is set to `"r1041_e82_400bps_sup_v500"`, but users may select from other available models that `clair3` was trained on, which may yield better results depending on the basecaller and data type. The following models are available:

        - `"ont"`
        - `"ont_guppy2"`
        - `"ont_guppy5"`
        - `"r941_prom_sup_g5014"`
        - `"r941_prom_hac_g360+g422"`
        - `"r941_prom_hac_g238"`
        - `"r1041_e82_400bps_sup_v500"`
        - `"r1041_e82_400bps_hac_v500"`
        - `"r1041_e82_400bps_sup_v410"`
        - `"r1041_e82_400bps_hac_v410"`

    ???+ warning "Default Parameters and Filtering"
        In this workflow, `clair3` is run with nearly all default parameters. Note that the VCF file produced by the `clair3` task is **unfiltered** and does not represent the final set of variants that will be included in the final consensus genome. A filtered vcf file is generated by the `bcftools_consensus` task. The filtering parameters are as follows:

        - The `min_map_quality` parameter is applied before calling variants.
        - The `min_depth` and `min_allele_freq` parameters are applied after variant calling during consensus genome construction.
<!-- endif -->

    !!! techdetails "Clair3 Technical Details"
        |  | Links |
        | --- | --- |
        | Task | [task_clair3.wdl](https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/variant_calling/task_clair3.wdl) |
        | Software Source Code | [Clair3 on GitHub](https://github.com/HKU-BAL/Clair3) |
        | Software Documentation | [Clair3 Documentation](https://github.com/HKU-BAL/Clair3?tab=readme-ov-file#usage) |
        | Original Publication(s) | [Symphonizing pileup and full-alignment for deep learning-based long-read variant calling](https://doi.org/10.1101/2021.12.29.474431) |