<!DOCTYPE html><html lang="en" class="no-js"><head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A collection of bioinformatics workflows for genomic characterization, submission preparation, and genomic epidemiology of pathogens of public health concern.">
      
      
        <meta name="author" content="Theiagen Genomics">
      
      
        <link rel="canonical" href="https://theiagen.github.io/public_health_bioinformatics/main/guides/gambit/">
      
      
        <link rel="prev" href="../versions/">
      
      
        <link rel="next" href="../gambit_database/">
      
      
      <link rel="icon" href="../../assets/logos/Theiagen-Symbol-Standard-01.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.18">
    
    
      
        <title>Guide to GAMBIT - Public Health Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.7e37652d.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#gambit" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version release.
  <a href="../../.."> 
    <strong>Click here to go to the latest version release.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-header__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Public Health Bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Guide to GAMBIT
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo" aria-label="Switch to Dark Mode" type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to Dark Mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"></path></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo" aria-label="Switch to Light Mode" type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to Light Mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m3.55 19.09 1.41 1.41 1.8-1.79-1.42-1.42M12 6c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6c0-3.32-2.69-6-6-6m8 7h3v-2h-3m-2.76 7.71 1.8 1.79 1.41-1.41-1.79-1.8M20.45 5l-1.41-1.4-1.8 1.79 1.42 1.42M13 1h-2v3h2M6.76 5.39 4.96 3.6 3.55 5l1.79 1.81zM1 13h3v-2H1m12 9h-2v3h2"></path></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../.." class="md-tabs__link">
          
  
  
  Public Health Bioinformatics

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-nav__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    Public Health Bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Public Health Bioinformatics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Public Health Bioinformatics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Getting Started
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            Getting Started
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting_started/commandline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With the Command-Line
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting_started/terra/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With Terra.bio
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting_started/philosophy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Workflow Failure Philosophy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Other Resources
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_3">
            <span class="md-nav__icon md-icon"></span>
            Other Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sops/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Available SOPs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../versions/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Database and Docker Versions
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" checked>
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#gambit-on-terrabio" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT on Terra.bio
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#importing-and-using-gambit-via-the-phb-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      Importing and using GAMBIT via the PHB workflows
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-gambit-on-your-local-machine" class="md-nav__link">
    <span class="md-ellipsis">
      Using GAMBIT on your local machine
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Using GAMBIT on your local machine">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#installation" class="md-nav__link">
    <span class="md-ellipsis">
      Installation
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Installation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#installation-from-bioconda" class="md-nav__link">
    <span class="md-ellipsis">
      Installation from Bioconda
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#installation-with-docker" class="md-nav__link">
    <span class="md-ellipsis">
      Installation with Docker
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#installation-from-source" class="md-nav__link">
    <span class="md-ellipsis">
      Installation from source
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    <span class="md-ellipsis">
      Usage
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Usage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#advanced-usage" class="md-nav__link">
    <span class="md-ellipsis">
      Advanced Usage
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gambit-databases" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Databases
    </span>
  </a>
  
    <nav class="md-nav" aria-label="GAMBIT Databases">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gambit-prokaryotic-databases" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Prokaryotic Databases
    </span>
  </a>
  
    <nav class="md-nav" aria-label="GAMBIT Prokaryotic Databases">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gambit-gtdb-database-v201" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT GTDB Database v2.0.1
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-gtdb-database-v200" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT GTDB Database v2.0.0
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-refseq-curated-database-v130" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT RefSeq Curated Database v1.3.0
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-refseq-curated-database-v120" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT RefSeq Curated Database v1.2.0
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-refseq-curated-database-v110" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT RefSeq Curated Database v1.1.0
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-refseq-curated-database-v100" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT RefSeq Curated Database v1.0.0
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-fungal-databases" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Fungal Databases
    </span>
  </a>
  
    <nav class="md-nav" aria-label="GAMBIT Fungal Databases">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gambit-fungal-database-v100" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Fungal Database v1.0.0
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-fungal-database-v020" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Fungal Database v0.2.0
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#technical-details" class="md-nav__link">
    <span class="md-ellipsis">
      Technical Details
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Technical Details">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#k-mer-based-representation-of-the-genomes" class="md-nav__link">
    <span class="md-ellipsis">
      K-mer-based representation of the genomes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#distance-metric-calculation" class="md-nav__link">
    <span class="md-ellipsis">
      Distance Metric Calculation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#built-in-species-distance-threshold" class="md-nav__link">
    <span class="md-ellipsis">
      Built-in Species Distance Threshold
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Built-in Species Distance Threshold">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure1" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 1
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-distances-correlate-with-sequence-identity" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT distances correlate with sequence identity
    </span>
  </a>
  
    <nav class="md-nav" aria-label="GAMBIT distances correlate with sequence identity">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure2" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 2
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gambit-database-creation-and-curation" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Database Creation and Curation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gambit-faqs" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT FAQs
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#other-resources" class="md-nav__link">
    <span class="md-ellipsis">
      Other Resources
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      References
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../gambit_database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT Database Creation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../phylogenetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Phylogenetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../custom_organisms/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Running Custom Organisms on TheiaCoV
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1">
        
          
          <label class="md-nav__link" for="__nav_1_5_1" id="__nav_1_5_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Type
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Type
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows_overview/workflows_type/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_2" id="__nav_1_5_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Import
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_2">
            <span class="md-nav__icon md-icon"></span>
            Data Import
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_3" id="__nav_1_5_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Genomic Characterization
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_3">
            <span class="md-nav__icon md-icon"></span>
            Genomic Characterization
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_4" id="__nav_1_5_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Construction
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_4">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Construction
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_5" id="__nav_1_5_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Placement
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_5">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Placement
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_6" id="__nav_1_5_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Public Data Sharing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_6">
            <span class="md-nav__icon md-icon"></span>
            Public Data Sharing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_7">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_7" id="__nav_1_5_1_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exporting Data from Terra
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_7">
            <span class="md-nav__icon md-icon"></span>
            Exporting Data from Terra
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_8">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_8" id="__nav_1_5_1_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Standalone
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_8">
            <span class="md-nav__icon md-icon"></span>
            Standalone
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2" id="__nav_1_5_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Kingdom
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Kingdom
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows_overview/workflows_kingdom/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_2" id="__nav_1_5_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Any Taxa
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_2">
            <span class="md-nav__icon md-icon"></span>
            Any Taxa
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_3" id="__nav_1_5_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Bacteria
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_3">
            <span class="md-nav__icon md-icon"></span>
            Bacteria
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_4" id="__nav_1_5_2_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Mycotics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_4">
            <span class="md-nav__icon md-icon"></span>
            Mycotics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_5" id="__nav_1_5_2_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Viral
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_5">
            <span class="md-nav__icon md-icon"></span>
            Viral
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows Alphabetically
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            Workflows Alphabetically
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows_overview/workflows_alphabetically/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiacov/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/genomic_characterization/vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../workflows/data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Contributing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            Contributing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/doc_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Documentation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/code_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Code
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#gambit-on-terrabio" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT on Terra.bio
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#importing-and-using-gambit-via-the-phb-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      Importing and using GAMBIT via the PHB workflows
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-gambit-on-your-local-machine" class="md-nav__link">
    <span class="md-ellipsis">
      Using GAMBIT on your local machine
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Using GAMBIT on your local machine">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#installation" class="md-nav__link">
    <span class="md-ellipsis">
      Installation
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Installation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#installation-from-bioconda" class="md-nav__link">
    <span class="md-ellipsis">
      Installation from Bioconda
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#installation-with-docker" class="md-nav__link">
    <span class="md-ellipsis">
      Installation with Docker
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#installation-from-source" class="md-nav__link">
    <span class="md-ellipsis">
      Installation from source
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    <span class="md-ellipsis">
      Usage
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Usage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#advanced-usage" class="md-nav__link">
    <span class="md-ellipsis">
      Advanced Usage
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gambit-databases" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Databases
    </span>
  </a>
  
    <nav class="md-nav" aria-label="GAMBIT Databases">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gambit-prokaryotic-databases" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Prokaryotic Databases
    </span>
  </a>
  
    <nav class="md-nav" aria-label="GAMBIT Prokaryotic Databases">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gambit-gtdb-database-v201" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT GTDB Database v2.0.1
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-gtdb-database-v200" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT GTDB Database v2.0.0
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-refseq-curated-database-v130" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT RefSeq Curated Database v1.3.0
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-refseq-curated-database-v120" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT RefSeq Curated Database v1.2.0
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-refseq-curated-database-v110" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT RefSeq Curated Database v1.1.0
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-refseq-curated-database-v100" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT RefSeq Curated Database v1.0.0
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-fungal-databases" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Fungal Databases
    </span>
  </a>
  
    <nav class="md-nav" aria-label="GAMBIT Fungal Databases">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gambit-fungal-database-v100" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Fungal Database v1.0.0
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-fungal-database-v020" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Fungal Database v0.2.0
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#technical-details" class="md-nav__link">
    <span class="md-ellipsis">
      Technical Details
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Technical Details">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#k-mer-based-representation-of-the-genomes" class="md-nav__link">
    <span class="md-ellipsis">
      K-mer-based representation of the genomes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#distance-metric-calculation" class="md-nav__link">
    <span class="md-ellipsis">
      Distance Metric Calculation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#built-in-species-distance-threshold" class="md-nav__link">
    <span class="md-ellipsis">
      Built-in Species Distance Threshold
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Built-in Species Distance Threshold">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure1" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 1
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gambit-distances-correlate-with-sequence-identity" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT distances correlate with sequence identity
    </span>
  </a>
  
    <nav class="md-nav" aria-label="GAMBIT distances correlate with sequence identity">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#figure2" class="md-nav__link">
    <span class="md-ellipsis">
      Figure 2
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gambit-database-creation-and-curation" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT Database Creation and Curation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gambit-faqs" class="md-nav__link">
    <span class="md-ellipsis">
      GAMBIT FAQs
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#other-resources" class="md-nav__link">
    <span class="md-ellipsis">
      Other Resources
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    <span class="md-ellipsis">
      References
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="gambit">GAMBIT<a class="headerlink" href="#gambit" title="Permanent link"></a></h1>
<p><strong>GAMBIT (Genomic Approximation Method for Bacterial Identification and Tracking)</strong> determines the taxon of the query genome assembly using a <strong><em>k</em>-mer-based approach</strong> to match the assembly sequence to the closest complete genome in a database.</p>
<div class="admonition dna">
<p class="admonition-title">GAMBIT genomic distance metric correlates with sequence identity!</p>
<p>GAMBIT uses an efficient genomic distance metric along with a curated database to identify genome assemblies in seconds. You can read more about how the distance metric is calculated in the <a href="#technical-details">Technical Details</a> section!</p>
</div>
<p>If the <strong>distance between the query genome assembly and the closest genome in the database is within a built-in species threshold</strong>, GAMBIT will assign the query genome to that species. Species thresholds are determined through a combination of automated and manual curation processes based on the diversity within the taxon.</p>
<div class="admonition tip">
<p class="admonition-title">GAMBIT includes a manually curated, high-quality database!</p>
<p>GAMBIT databases consist of two files:</p>
<ol>
<li>A <strong>signatures file</strong> containing the GAMBIT signatures (compressed representations) of all genomes represented in the database </li>
<li>A <strong>metadata file</strong> relating the represented genomes to their genome accessions, taxonomic identifications, and species thresholds</li>
</ol>
</div>
<div class="grid cards">
<ul>
<li>
<p><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 9h5.5L13 3.5zM6 2h8l6 6v12a2 2 0 0 1-2 2H6a2 2 0 0 1-2-2V4c0-1.11.89-2 2-2m.12 13.5 3.74 3.74 1.42-1.41-2.33-2.33 2.33-2.33-1.42-1.41zm11.16 0-3.74-3.74-1.42 1.41 2.33 2.33-2.33 2.33 1.42 1.41z"></path></svg></span> <strong>Latest GAMBIT Version</strong></p>
<hr>
<p><a href="https://github.com/jlumpe/gambit/releases/tag/v1.0.1">GAMBIT v1.0.1 source code</a></p>
</li>
<li>
<p><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 3C7.58 3 4 4.79 4 7s3.58 4 8 4 8-1.79 8-4-3.58-4-8-4M4 9v3c0 2.21 3.58 4 8 4s8-1.79 8-4V9c0 2.21-3.58 4-8 4s-8-1.79-8-4m0 5v3c0 2.21 3.58 4 8 4s8-1.79 8-4v-3c0 2.21-3.58 4-8 4s-8-1.79-8-4"></path></svg></span> <strong>Latest Database Version</strong></p>
<hr>
<p><a href="#gambit-gtdb-database-v201">GAMBIT Prokaryotic GTDB Database v2.0.1</a></p>
<p><a href="#gambit-fungal-database-v100">GAMBIT Fungal Database v1.0.0</a></p>
</li>
</ul>
</div>
<h2 id="gambit-on-terrabio">GAMBIT on Terra.bio<a class="headerlink" href="#gambit-on-terrabio" title="Permanent link"></a></h2>
<p><a href="https://www.theiagen.com/">Theiagen</a>s <a href="https://github.com/theiagen/public_health_bioinformatics">Public Health Bioinformatics (PHB)</a> is a suite of workflows for characterization, epidemiology and sharing of pathogen genomes. Workflows are available for viruses, bacteria, and fungi.</p>
<h2 id="importing-and-using-gambit-via-the-phb-workflows">Importing and using GAMBIT via the PHB workflows<a class="headerlink" href="#importing-and-using-gambit-via-the-phb-workflows" title="Permanent link"></a></h2>
<p>The <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/Gambit_Query_PHB"><strong>GAMBIT_Query_PHB</strong></a> workflow performs taxon assignment of a genome assembly using the GAMBIT. It can be imported directly to <a href="https://terra.bio">Terra.bio</a> via <a href="https://dockstore.org">Dockstore</a>.</p>
<p>Two inputs are required for the <strong>GAMBIT_Query_PHB</strong> workflow: a genome assembly and a sample name associated with the genome assembly. The default GAMBIT database used for taxonomic identification is the Prokaryotic <a href="#gambit-gtdb-database-v201">GAMBIT Database GTDB v2.0.1</a>, but alternate GAMBIT databases can be provided.</p>
<div class="admonition dna">
<p class="admonition-title">Gambit_Query_PHB</p>
<p>More information on <strong>GAMBIT_Query_PHB</strong> is available <a href="../../workflows/standalone/gambit_query/"><strong>here</strong></a>.</p>
</div>
<div class="admonition tip inline end">
<p class="admonition-title">Import workflows to <a href="https://terra.bio/">Terra.bio</a>:</p>
<p><strong>GAMBIT_Query_PHB</strong></p>
<ul>
<li><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/Gambit_Query_PHB">Gambit_Query_PHB</a></li>
</ul>
<p><strong>TheiaProk Workflow Series</strong></p>
<ul>
<li><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaProk_Illumina_PE_PHB">TheiaProk_Illumina_PE_PHB</a></li>
<li><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaProk_Illumina_SE_PHB">TheiaProk_Illumina_SE_PHB</a></li>
<li><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaProk_ONT_PHB">TheiaProk_ONT_PHB</a></li>
<li><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaProk_FASTA_PHB">TheiaProk_FASTA_PHB</a></li>
</ul>
<p><strong>TheiaEuk Workflow Series</strong></p>
<ul>
<li><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaEuk_Illumina_PE_PHB">TheiaEuk_Illumina_PE_PHB</a></li>
</ul>
</div>
<p>Additionally, GAMBIT is also part of the <strong>TheiaProk</strong> and <strong>TheiaEuk</strong> collection of workflows, the first dedicated to the analysis of prokaryotic data, and the second data to mycotics. The TheiaProk or TheiaEuk most appropriate for your type of input data can be imported from the Dockstore links on the right.</p>
<p>In both, GAMBIT is responsible for performing the taxonomic identification of the assembled sequences, which can trigger taxa-specific submodules for further genomic characterization. For TheiaProk, the default database is the Prokaryotic <a href="#gambit-gtdb-database-v201">GAMBIT Database GTDB v2.0.1</a> and for TheiaEuk, the default database is the <a href="#gambit-fungal-database-v100">Fungal GAMBIT Database v1.0.0</a>.</p>
<div class="admonition dna">
<p class="admonition-title">TheiaProk and TheiaEuk</p>
<p>More information on TheiaProk and TheiaEuk is available on the following pages:</p>
<ul>
<li><a href="../../workflows/genomic_characterization/theiaprok/"><strong>TheiaProk Workflow Series</strong></a></li>
<li><a href="../../workflows/genomic_characterization/theiaeuk/"><strong>TheiaEuk Workflow Series</strong></a></li>
</ul>
</div>
<hr>
<h2 id="using-gambit-on-your-local-machine">Using GAMBIT on your local machine<a class="headerlink" href="#using-gambit-on-your-local-machine" title="Permanent link"></a></h2>
<p>This guide assumes you have prior knowledge of how to install software locally in a Unix command-line environment. The necessary databases will have to be downloaded independently to be used with GAMBIT. They are available in the <a href="#gambit-databases">GAMBIT Databases</a> section of this document and should be placed in a directory of your choice. The directory should not contain any other files with the same extensions.</p>
<h3 id="installation">Installation<a class="headerlink" href="#installation" title="Permanent link"></a></h3>
<h4 id="installation-from-bioconda">Installation from Bioconda<a class="headerlink" href="#installation-from-bioconda" title="Permanent link"></a></h4>
<p>The recommended way to install the tool is through the&nbsp;<a href="https://www.anaconda.com/products/distribution">Conda</a>&nbsp;package manager from the&nbsp;<a href="https://bioconda.github.io/">Bioconda</a>&nbsp;channel. You can simply run the following command to download GAMBITs latest version:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>conda<span class="w"> </span>install<span class="w"> </span>-c<span class="w"> </span>bioconda<span class="w"> </span>gambit
</span></code></pre></div>
<h4 id="installation-with-docker">Installation with Docker<a class="headerlink" href="#installation-with-docker" title="Permanent link"></a></h4>
<p>The latest version of GAMBIT software is available as a Docker container in Theiagens <a href="https://cloud.google.com/artifact-registry">Google Artifact Registry (GAR)</a>. If <a href="https://docs.docker.com/engine/install/">Docker is installed in your system</a> you can simply run the following command to download the container:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>docker<span class="w"> </span>pull<span class="w"> </span>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0
</span></code></pre></div>
<p>You can access the container with the following command (note: with the <code>-v $PWD:/data</code>, your current directory is being mounted to the <code>data/</code> folder inside the container):</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>docker<span class="w"> </span>run<span class="w"> </span>-v<span class="w"> </span><span class="nv">$PWD</span>:/data<span class="w"> </span>-it<span class="w"> </span>us-docker.pkg.dev/general-theiagen/staphb/gambit:1.0.0<span class="w"> </span>bash
</span></code></pre></div>
<h4 id="installation-from-source">Installation from source<a class="headerlink" href="#installation-from-source" title="Permanent link"></a></h4>
<p>These instructions assume that you have <a href="https://git-scm.com/book/en/v2/Getting-Started-Installing-Git">Git</a>, <a href="https://www.python.org/downloads/">Python</a> and <a href="https://pip.pypa.io/en/stable/installation/">Pip</a> installed in your system. Navigate to <a href="https://github.com/jlumpe/gambit">https://github.com/jlumpe/gambit</a> and clone the repository, or use the following command:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>git<span class="w"> </span>clone<span class="w"> </span>https://github.com/jlumpe/gambit.git
</span></code></pre></div>
<p>Installing from source requires the Cython package as well as a C compiler to be installed on your system.  Navigate to the repository and install the package:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>pip<span class="w"> </span>install<span class="w"> </span>.
</span></code></pre></div>
<h3 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link"></a></h3>
<p>Positional arguments are one or more FASTA files containing query genome assemblies. You must provide the path to the directory containing the database files using either the&nbsp;<code>-d</code>&nbsp;option (<em>before</em>&nbsp;the&nbsp;<code>query</code>&nbsp;subcommand) or by setting the&nbsp;<code>GAMBIT_DB_PATH</code>&nbsp;environment variable. The results can be optionally outputted to a file, but by default, they are written to the terminal.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>gambit<span class="w"> </span><span class="o">[</span>-d<span class="w"> </span>&lt;/path/to/database/&gt;<span class="o">]</span><span class="w"> </span>query<span class="w"> </span><span class="o">[</span>-o<span class="w"> </span>results.csv<span class="o">]</span><span class="w"> </span>genome1.fasta<span class="w"> </span>genome2.fasta<span class="w"> </span>...
</span></code></pre></div>
<h4 id="advanced-usage">Advanced Usage<a class="headerlink" href="#advanced-usage" title="Permanent link"></a></h4>
<p>There are many available commands in GAMBIT:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>Usage: gambit [OPTIONS] COMMAND [ARGS]...
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>  Tool for rapid taxonomic identification of microbial pathogens from genomic data.
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>Options:
</span><span id="__span-6-6"><a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a>  -d, --db DIRECTORY  Directory containing GAMBIT database files.
</span><span id="__span-6-7"><a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a>  --version           Show the version and exit.
</span><span id="__span-6-8"><a id="__codelineno-6-8" name="__codelineno-6-8" href="#__codelineno-6-8"></a>  --help              Show this message and exit.
</span><span id="__span-6-9"><a id="__codelineno-6-9" name="__codelineno-6-9" href="#__codelineno-6-9"></a>
</span><span id="__span-6-10"><a id="__codelineno-6-10" name="__codelineno-6-10" href="#__codelineno-6-10"></a>Commands:
</span><span id="__span-6-11"><a id="__codelineno-6-11" name="__codelineno-6-11" href="#__codelineno-6-11"></a>  dist        Calculate the GAMBIT distances between a set of query...
</span><span id="__span-6-12"><a id="__codelineno-6-12" name="__codelineno-6-12" href="#__codelineno-6-12"></a>  query       Predict taxonomy of microbial samples from genome sequences.
</span><span id="__span-6-13"><a id="__codelineno-6-13" name="__codelineno-6-13" href="#__codelineno-6-13"></a>  signatures  Create and inspect GAMBIT signature files.
</span><span id="__span-6-14"><a id="__codelineno-6-14" name="__codelineno-6-14" href="#__codelineno-6-14"></a>  tree        Estimate a relatedness tree for a set of genomes and output...
</span></code></pre></div>
<p>GAMBITs <code>query</code> is the most used as it computes the distance of a query genome to the genomes provided in the database.</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>Usage: gambit query [OPTIONS] GENOMES...
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>  Predict taxonomy of microbial samples from genome sequences.
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>Options:
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a>  -l LISTFILE                     File containing paths to query genomes, one
</span><span id="__span-7-7"><a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a>                                  per line.
</span><span id="__span-7-8"><a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a>  --ldir DIRECTORY                Parent directory of paths in LISTFILE.
</span><span id="__span-7-9"><a id="__codelineno-7-9" name="__codelineno-7-9" href="#__codelineno-7-9"></a>  -o, --output FILENAME           File path to write to. If omitted will write
</span><span id="__span-7-10"><a id="__codelineno-7-10" name="__codelineno-7-10" href="#__codelineno-7-10"></a>                                  to stdout.
</span><span id="__span-7-11"><a id="__codelineno-7-11" name="__codelineno-7-11" href="#__codelineno-7-11"></a>  -f, --outfmt [csv|json|archive]
</span><span id="__span-7-12"><a id="__codelineno-7-12" name="__codelineno-7-12" href="#__codelineno-7-12"></a>                                  Format to output results in.
</span><span id="__span-7-13"><a id="__codelineno-7-13" name="__codelineno-7-13" href="#__codelineno-7-13"></a>  -s, --sigfile FILE              File containing query signatures, to use in
</span><span id="__span-7-14"><a id="__codelineno-7-14" name="__codelineno-7-14" href="#__codelineno-7-14"></a>                                  place of GENOMES.
</span><span id="__span-7-15"><a id="__codelineno-7-15" name="__codelineno-7-15" href="#__codelineno-7-15"></a>  --progress / --no-progress      Show/don't show progress meter.
</span><span id="__span-7-16"><a id="__codelineno-7-16" name="__codelineno-7-16" href="#__codelineno-7-16"></a>  -c, --cores INTEGER RANGE       Number of CPU cores to use.  [x&gt;=1]
</span><span id="__span-7-17"><a id="__codelineno-7-17" name="__codelineno-7-17" href="#__codelineno-7-17"></a>  --help                          Show this message and exit.
</span></code></pre></div>
<hr>
<h2 id="gambit-databases">GAMBIT Databases<a class="headerlink" href="#gambit-databases" title="Permanent link"></a></h2>
<h3 id="gambit-prokaryotic-databases">GAMBIT Prokaryotic Databases<a class="headerlink" href="#gambit-prokaryotic-databases" title="Permanent link"></a></h3>
<h4 id="gambit-gtdb-database-v201">GAMBIT GTDB Database v2.0.1<a class="headerlink" href="#gambit-gtdb-database-v201" title="Permanent link"></a></h4>
<details class="toggle">
<summary>Database Details</summary>
<p>This database is a <strong>patch update</strong> to the v2.0.0 database. This database is identical to the v2.0.0 database <strong>except that the following genomes were removed</strong>. </p>
<div class="language-text highlight"><pre><span></span><code>GCF_003977345.1
GCF_003977375.1
GCF_004025385.1
GCF_002249045.1
GCF_004402105.1
GCF_003977405.1
GCF_003985005.1
GCF_002246205.1
GCF_003977245.1
GCF_007197595.1
GCF_003977285.1
GCF_001063095.1
GCF_002248705.1
</code></pre></div>
<p>These genomes currently named as <em>Shigella</em> species in NCBI, but are actually <em>Escherichia coli</em> according to the best match type strain using ANI. Their removal from the database prevents false assignment of <em>Echerichia coli</em> query genomes to <em>Shigella</em> species.</p>
<p><strong>Database Files</strong></p>
<p>These database files are hosted in a public Google bucket by Theiagen Genomics:</p>
<p><strong>GS URI (for <a href="https://terra.bio">Terra.bio</a> usage):</strong></p>
<ul>
<li><code>gs://gambit-databases-rp/2.0.0/gambit-metadata-2.0.1-20250505.gdb</code></li>
<li><code>gs://gambit-databases-rp/2.0.0/gambit-signatures-2.0.1-20250505.gs</code></li>
</ul>
<p><strong>HTTPS URL (for local download):</strong></p>
<ul>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/2.0.0/gambit-metadata-2.0.1-20250505.gdb">https://storage.cloud.google.com/gambit-databases-rp/2.0.0/gambit-metadata-2.0.1-20250505.gdb</a></li>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/2.0.0/gambit-signatures-2.0.1-20250505.gs">https://storage.cloud.google.com/gambit-databases-rp/2.0.0/gambit-signatures-2.0.1-20250505.gs</a></li>
</ul>
</details>
<h4 id="gambit-gtdb-database-v200">GAMBIT GTDB Database v2.0.0<a class="headerlink" href="#gambit-gtdb-database-v200" title="Permanent link"></a></h4>
<details class="toggle">
<summary>Database Details</summary>
<p>This database is a <strong>major update</strong> to the Curated v1.3.0 database. This iteration of the GAMBIT database relies upon the <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database</a> (GTDB), an initiative to establish a standardised microbial taxonomy based on genome phylogeny. The genomes used to construct the phylogeny are obtained from <a href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a> and <a href="https://www.ncbi.nlm.nih.gov/genbank/">GenBank</a>, independently quality-controlled using <a href="https://github.com/Ecogenomics/CheckM/wiki">CheckM</a> before inclusion in GTDB.</p>
<p>This database was computed from <a href="https://gtdb.ecogenomic.org/stats/r214">GTDB Release 214.1</a> as of April 28th, 2023. </p>
<ul>
<li>
<p><strong>Automated curation efforts</strong></p>
<p>The following curation steps were followed for all species:</p>
<ul>
<li>The candidates for an existing genus were collapsed (e.g genus_A, genus_B becomes genus)</li>
</ul>
<p>The following species were updated:</p>
<ul>
<li><em>Shigella</em> sp<em>.</em><ul>
<li>This genus is not present in GTDB  as it is collapsed under <em>Escherichia coli;</em></li>
<li>All Shigella genomes in RefSeq were added to the database with no clustering using default quality criteria.</li>
</ul>
</li>
<li><em>Mycolicibacterium/Mycolicibacter/Mycolicibacillus/Mycobacteroides/Mycobacterium</em> sp.<ul>
<li>All genomes available were used.</li>
</ul>
</li>
<li><em>Tropheryma whipplei</em><ul>
<li>This species has a low completeness score of 75%;</li>
<li>The CheckM completeness score was lowered to 70% for genomes belonging to this species.</li>
</ul>
</li>
</ul>
</li>
</ul>
<p><strong>Database Files</strong></p>
<p>These database files are hosted in a public Google bucket by Theiagen Genomics:</p>
<p><strong>GS URI (for <a href="https://terra.bio">Terra.bio</a> usage):</strong></p>
<ul>
<li><code>gs://gambit-databases-rp/2.0.0/gambit-metadata-2.0.0-20240628.gdb</code></li>
<li><code>gs://gambit-databases-rp/2.0.0/gambit-signatures-2.0.0-20240628.gs</code></li>
</ul>
<p><strong>HTTPS URL (for local download):</strong></p>
<ul>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/2.0.0/gambit-metadata-2.0.0-20240628.gdb">https://storage.cloud.google.com/gambit-databases-rp/2.0.0/gambit-metadata-2.0.0-20240628.gdb</a></li>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/2.0.0/gambit-signatures-2.0.0-20240628.gs">https://storage.cloud.google.com/gambit-databases-rp/2.0.0/gambit-signatures-2.0.0-20240628.gs</a></li>
</ul>
<p><strong>Taxa included in the GAMBIT database</strong></p>
<p>Summary of species represented in the database with number of genomes representing each species and the species threshold:</p>
<ul>
<li>
<p><a href="https://storage.cloud.google.com/gambit-databases-rp/2.0.0/gambit-taxa-2.0.0-20240628.tsv">https://storage.cloud.google.com/gambit-databases-rp/2.0.0/gambit-taxa-2.0.0-20240628.tsv</a></p>
<p>Note: Species with a threshold of "0" have been sub-speciated. Subspecies are not listed in this table.</p>
</li>
</ul>
</details>
<h4 id="gambit-refseq-curated-database-v130">GAMBIT RefSeq Curated Database v1.3.0<a class="headerlink" href="#gambit-refseq-curated-database-v130" title="Permanent link"></a></h4>
<details class="toggle">
<summary>Database Details</summary>
<p>This database is a <strong>patch update</strong> to the Curated v1.2.0 database. In addition to all of the species included in the v1.2.0 database below, this database replaces all species in the <strong><em>Mycobacterium</em>, <em>Mycolicibacterium</em>, <em>Mycobacteroides</em>, and <em>Mycolicibacter</em></strong> genera with the available genomes in RefSeq as of October 16th, 2023.</p>
<ul>
<li>
<p><strong>Manual curation efforts</strong></p>
<p>The following species were updated:</p>
<ul>
<li><em>Mycolicibacterium/Mycolicibacter/Mycolicibacillus/Mycobacteroides/Mycobacterium</em> sp.<ul>
<li>All genomes available were used</li>
</ul>
</li>
</ul>
</li>
</ul>
<p><strong>Database Files</strong></p>
<p>These database files are hosted in a public Google bucket by Theiagen Genomics:</p>
<p><strong>GS URI (for <a href="https://terra.bio">Terra.bio</a> usage):</strong></p>
<ul>
<li><code>gs://gambit-databases-rp/1.3.0/gambit-metadata-1.3-231016.gdb</code></li>
<li><code>gs://gambit-databases-rp/1.3.0/gambit-signatures-1.3-231016.gs</code></li>
</ul>
<p><strong>HTTPS URL (for local download):</strong></p>
<ul>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/1.3.0/gambit-metadata-1.3-231016.gdb">https://storage.cloud.google.com/gambit-databases-rp/1.3.0/gambit-metadata-1.3-231016.gdb</a></li>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/1.3.0/gambit-signatures-1.3-231016.gs">https://storage.cloud.google.com/gambit-databases-rp/1.3.0/gambit-signatures-1.3-231016.gs</a></li>
</ul>
<p><strong>Taxa included in the GAMBIT database</strong></p>
<p>Summary of species represented in the database with number of genomes representing each species and the species threshold:</p>
<ul>
<li>
<p><a href="https://storage.cloud.google.com/gambit-databases-rp/1.3.0/gambit-1.3-231016-taxa-list.txt">https://storage.cloud.google.com/gambit-databases-rp/1.3.0/gambit-1.3-231016-taxa-list.txt</a></p>
<p>Note: Species with a threshold of "0" have been sub-speciated. Subspecies are not listed in this table.</p>
</li>
</ul>
</details>
<h4 id="gambit-refseq-curated-database-v120">GAMBIT RefSeq Curated Database v1.2.0<a class="headerlink" href="#gambit-refseq-curated-database-v120" title="Permanent link"></a></h4>
<details class="toggle">
<summary>Database Details</summary>
<p>This database is a <strong>patch update</strong> to the RefSeq Curated v1.1.0 database. In addition to all of the species included in the v1.1.0 database below, this database includes new species that were under-represented as of August 11th, 2023.</p>
<ul>
<li>
<p><strong>Automated curation efforts</strong></p>
<p>Automated addition of new species to the database that are genetically distant from all current species in the v1.1.0 GAMBIT database. </p>
<p>Genomes represented in the Genome Taxonomy Database (<a href="https://gtdb.ecogenomic.org/">GTDB</a>) were added. These genomes predominantly originate from the RefSeq and GenBank databases with taxonomic metadata curated by GTDB. Any genomes added to the GAMBIT database from GTDB are added with the metadata from GTDB.</p>
<ul>
<li>
<p>Genomes added from GTDB include the following species:</p>
<p><a href="../../assets/files/gambit-list-of-new-species-db-v1_2_0.txt">gambit-list-of-new-species-db-v1_2_0.txt</a></p>
</li>
</ul>
</li>
<li>
<p><strong>Manual curation efforts</strong></p>
<p>Manually curated updates to several taxa relevant to public health. All genomes representing the taxa below were removed and replaced with the RefSeq genomes representing each species as of August 11th, 2023.</p>
<ul>
<li><em>Citrobacter, Providencia, Hafnia, Neisseria, Proteus, Achromobacter, Aeromonas, Bacillus, Brucella, Afipia, Burkholderia, Paraburkholderia, Corynebacterium, Morganella</em></li>
</ul>
</li>
</ul>
<p><strong>Database Files</strong></p>
<p>These database files are hosted in a public Google bucket by Theiagen Genomics:</p>
<p><strong>GS URI (for <a href="https://terra.bio">Terra.bio</a> usage):</strong></p>
<ul>
<li><code>gs://gambit-databases-rp/1.2.0/gambit-metadata-1.2-231002.gdb</code></li>
<li><code>gs://gambit-databases-rp/1.2.0/gambit-signatures-1.2-231002.gs</code></li>
</ul>
<p><strong>HTTPS URL (for local download):</strong></p>
<ul>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/1.2.0/gambit-metadata-1.2-231002.gdb">https://storage.cloud.google.com/gambit-databases-rp/1.2.0/gambit-metadata-1.2-231002.gdb</a></li>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/1.2.0/gambit-signatures-1.2-231002.gs">https://storage.cloud.google.com/gambit-databases-rp/1.2.0/gambit-signatures-1.2-231002.gs</a></li>
</ul>
<p><strong>Taxa included in the GAMBIT database</strong></p>
<p>Summary of species represented in the database with number of genomes representing each species and the species threshold:</p>
<ul>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/1.2.0/gambit-1.2-231002-taxa-list.txt">https://storage.cloud.google.com/gambit-databases-rp/1.2.0/gambit-1.2-231002-taxa-list.txt</a>
    Note: Species with a threshold of "0" have been sub-speciated. Subspecies are not listed in this table.</li>
</ul>
</details>
<h4 id="gambit-refseq-curated-database-v110">GAMBIT RefSeq Curated Database v1.1.0<a class="headerlink" href="#gambit-refseq-curated-database-v110" title="Permanent link"></a></h4>
<details class="toggle">
<summary>Database Details</summary>
<p>This database is a <strong>patch update</strong> to the RefSeq Curated v1.0.0 database. In addition to all of the species included in the v1.0.0 database below, this database replaces all species in the <strong>Enterobacter, Legionella, and Vibrio genera</strong> with the available genomes in RefSeq as of April 17th, 2023.</p>
<p><strong>Database Files</strong></p>
<p>These database files are hosted in a public Google bucket by Theiagen:</p>
<p><strong>GS URI (for <a href="https://terra.bio">Terra.bio</a> usage):</strong></p>
<ul>
<li><code>gs://gambit-databases-rp/1.1.0/gambit-metadata-1.1-230417.gdb</code></li>
<li><code>gs://gambit-databases-rp/1.1.0/gambit-signatures-1.1-230417.gs</code></li>
</ul>
<p><strong>HTTPS URL (for local download):</strong></p>
<ul>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/1.1.0/gambit-metadata-1.1-230417.gdb">https://storage.cloud.google.com/gambit-databases-rp/1.1.0/gambit-metadata-1.1-230417.gdb</a></li>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/1.1.0/gambit-signatures-1.1-230417.gs">https://storage.cloud.google.com/gambit-databases-rp/1.1.0/gambit-signatures-1.1-230417.gs</a></li>
</ul>
<p><strong>Taxa included in the GAMBIT database</strong></p>
<p>Summary of species represented in the database with number of genomes representing each species and the species threshold:</p>
<ul>
<li>
<p><a href="https://storage.cloud.google.com/gambit-databases-rp/1.1.0/gambit-1.1-230417-taxa-list.txt">https://storage.cloud.google.com/gambit-databases-rp/1.1.0/gambit-1.1-230417-taxa-list.txt</a></p>
<p>Note: Species with a threshold of "0" have been sub-speciated. Subspecies are not listed in this table.</p>
</li>
</ul>
</details>
<h4 id="gambit-refseq-curated-database-v100">GAMBIT RefSeq Curated Database v1.0.0<a class="headerlink" href="#gambit-refseq-curated-database-v100" title="Permanent link"></a></h4>
<details class="toggle">
<summary>Database Details</summary>
<p>The GAMBIT RefSeq Curated v1.0.0 database was used for the analysis described in the <a href="https://doi.org/10.1371/journal.pone.0277575">GAMBIT publication</a>. This database was constructed based on the available genomes in RefSeq as of July 1st, 2016.  Genomes that did not have associated genus and/or species were removed. Additionally, at least two separate sequenced isolates for a given species were required in order to determine the classification threshold.  </p>
<ul>
<li>
<p><strong>Manual curation efforts</strong></p>
<p>Ambiguous genomes were removed if they met any of the following criteria: </p>
<ol>
<li>A genome that did not cluster well with the majority of the other genomes within their species;</li>
<li>A genome that clustered well with some members of their species but also several members of another species in the database&lt;</li>
<li>A genome that did not cluster well with any genomes in the database.</li>
</ol>
</li>
</ul>
<p><strong>Database Files</strong></p>
<p>These database files are hosted in a public Google bucket by Theiagen:</p>
<p><strong>GS URI (for <a href="https://terra.bio">Terra.bio</a> usage):</strong></p>
<ul>
<li><code>gs://gambit-databases-rp/1.0b2/gambit-genomes-1.0b2-rev2-211116.db</code></li>
<li><code>gs://gambit-databases-rp/1.0b2/gambit-signatures-1.0b1-210719.h5</code></li>
</ul>
<p><strong>HTTPS URL (for local download):</strong></p>
<ul>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/1.0b2/gambit-genomes-1.0b2-rev2-211116.db">https://storage.cloud.google.com/gambit-databases-rp/1.0b2/gambit-genomes-1.0b2-rev2-211116.db</a></li>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/1.0b2/gambit-signatures-1.0b1-210719.h5">https://storage.cloud.google.com/gambit-databases-rp/1.0b2/gambit-signatures-1.0b1-210719.h5</a></li>
</ul>
<p><strong>Taxa included in the GAMBIT database</strong></p>
<p>Summary of species represented in the database with number of genomes representing each species and the species threshold:</p>
<ul>
<li>
<p><a href="https://storage.cloud.google.com/gambit-databases-rp/1.0b2/gambit-1.0b2-rev2-211116-taxa-list.txt">https://storage.cloud.google.com/gambit-databases-rp/1.0b2/gambit-1.0b2-rev2-211116-taxa-list.txt</a></p>
<p>Note: Species with a threshold of "0" have been sub-speciated. Subspecies are not listed in this table.</p>
</li>
</ul>
</details>
<hr>
<h3 id="gambit-fungal-databases">GAMBIT Fungal Databases<a class="headerlink" href="#gambit-fungal-databases" title="Permanent link"></a></h3>
<h4 id="gambit-fungal-database-v100">GAMBIT Fungal Database v1.0.0<a class="headerlink" href="#gambit-fungal-database-v100" title="Permanent link"></a></h4>
<details class="toggle">
<summary>Database Details</summary>
<p>The GAMBIT Fungal Database v1.0.0 database was constructed based on the available genomes in RefSeq/GenBank as of December 13th, 2024. For inclusion in the database, species were required to have at least two genomes in GenBank and at least one genome representing the species in RefSeq.</p>
<ol>
<li>Species with a diameter of zero were excluded;</li>
<li>Species with three or fewer genomes and a diameter greater than 0.75 were excluded.</li>
</ol>
<p><strong>Manual curation efforts</strong></p>
<ul>
<li>Species were curated based on GAMBIT diameter:</li>
<li>The database was manually curated to remove highly distant genomes which were likely mislabeled.</li>
<li>Six species were divided into subspecies to ensure non-overlapping species diameters.</li>
<li>Two pairs of species were too closely related to distinguish (<em>Aspergillus flavus/Aspergillus oryzae</em>&nbsp;and&nbsp;<em>Aspergillus niger/Aspergillus welwitschiae</em>), therefore were combined.</li>
</ul>
<p><strong>Database Files</strong></p>
<p>These database files are hosted in a public Google bucket by Theiagen:</p>
<p><strong>GS URI (for <a href="https://terra.bio">Terra.bio</a> usage):</strong></p>
<ul>
<li><code>gs://gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-metadata-1.0.0-20241213.gdb</code></li>
<li><code>gs://gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-signatures-1.0.0-20241213.gs</code></li>
</ul>
<p><strong>HTTPS URL (for local download):</strong></p>
<ul>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-metadata-1.0.0-20241213.gdb">https://storage.cloud.google.com/gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-metadata-1.0.0-20241213.gdb</a></li>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-signatures-1.0.0-20241213.gs">https://storage.cloud.google.com/gambit-databases-rp/fungal-version/1.0.0/gambit-fungal-signatures-1.0.0-20241213.gs</a></li>
</ul>
<p><strong>Taxa included in the GAMBIT database</strong></p>
<p>360 fungal species from 176 genera are represented in the fungal database from a total of 8073 fungal genomes. A table indicating the number of genomes and species diameter for each species represented in the database is indicated below.</p>
<p><a href="../../assets/files/gambit-1.0.0-20241213-taxa-list.txt">gambit-1.0.0-20241213-taxa-list.txt</a></p>
</details>
<h4 id="gambit-fungal-database-v020">GAMBIT Fungal Database v0.2.0<a class="headerlink" href="#gambit-fungal-database-v020" title="Permanent link"></a></h4>
<details class="toggle">
<summary>Database Details</summary>
<p>The GAMBIT Fungal Database v0.2.0 database was used for the analysis described in the <a href="https://doi.org/10.3389/fpubh.2023.1198213">TheiaEuk publication</a>. This database was constructed based on the available genomes in GenBank as of November 30th, 2022. For inclusion in the database, species were required to have at least two genomes in GenBank and at least one genome representing the species in RefSeq.</p>
<ol>
<li>Species with a diameter of zero were excluded;</li>
<li>Species with three or fewer genomes and a diameter greater than 0.75 were excluded.</li>
</ol>
<p><strong>Manual curation efforts</strong></p>
<ul>
<li>Species were curated based on GAMBIT diameter:</li>
<li>The database was manually curated to remove highly distant genomes which were likely mislabeled.</li>
<li>Nine species were divided into subspecies to ensure non-overlapping species diameters.</li>
<li>Two pairs of species were too closely related to distinguish (<em>Aspergillus flavus/Aspergillus oryzae</em>&nbsp;and&nbsp;<em>Aspergillus niger/Aspergillus welwitschiae</em>), therefore were combined.</li>
</ul>
<p><strong>Database Files</strong></p>
<p>These database files are hosted in a public Google bucket by Theiagen:</p>
<p><strong>GS URI (for <a href="https://terra.bio">Terra.bio</a> usage):</strong></p>
<ul>
<li><code>gs://gambit-databases-rp/fungal-version/0.2/221130-theiagen-fungal-v0.2.db</code></li>
<li><code>gs://gambit-databases-rp/fungal-version/0.2/221130-theiagen-fungal-v0.2.h5</code></li>
</ul>
<p><strong>HTTPS URL (for local download):</strong></p>
<ul>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/fungal-version/0.2/221130-theiagen-fungal-v0.2.db">https://storage.cloud.google.com/gambit-databases-rp/fungal-version/0.2/221130-theiagen-fungal-v0.2.db</a></li>
<li><a href="https://storage.cloud.google.com/gambit-databases-rp/fungal-version/0.2/221130-theiagen-fungal-v0.2.h5">https://storage.cloud.google.com/gambit-databases-rp/fungal-version/0.2/221130-theiagen-fungal-v0.2.h5</a></li>
</ul>
<p><strong>Taxa included in the GAMBIT database</strong></p>
<p>245 fungal species from 138 genera are represented in the fungal database from a total of 5,667 fungal genomes. A table indicating the number of genomes and species diameter for each species represented in the database is indicated below.</p>
<p><a href="../../assets/files/gambit-0.2.0-221130-taxa-list.txt">gambit-0.2.0-221130-taxa-list.tsv</a></p>
</details>
<hr>
<h2 id="technical-details">Technical Details<a class="headerlink" href="#technical-details" title="Permanent link"></a></h2>
<h3 id="k-mer-based-representation-of-the-genomes">K-mer-based representation of the genomes<a class="headerlink" href="#k-mer-based-representation-of-the-genomes" title="Permanent link"></a></h3>
<p>A GAMBIT signature is a compressed representation of a genome sequence that supports the efficient calculation of the <strong>GAMBIT genomic distance metric</strong>. It is defined as the set of <em>k</em>-mers present in the genome which occur immediately following a fixed prefix sequence. <strong>GAMBIT finds all 11-mers in a genome assembly that immediately follows the prefix sequence ATGAC</strong>.</p>
<p>This allows not only thousands of genomes to be represented in a relatively small (~3GB) database, but the comparison of the query genome to the set of references provided in the used GAMBIT database to be performed very fast.</p>
<div class="admonition warning">
<p class="admonition-title">Taxonomy information</p>
<p>The <strong>GAMBIT database</strong> used for classification consists of pre-calculated signatures for the reference genomes along with additional genome metadata and a taxonomy tree. As of GAMBIT Prokaryotic database v2.0.0, taxonomy information is derived from the  <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database (GTDB)</a> but restricted to the genus and species ranks and subject to additional curation. The other databases have the taxonomy information derived from the <a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI taxonomy database</a>.</p>
</div>
<h3 id="distance-metric-calculation">Distance Metric Calculation<a class="headerlink" href="#distance-metric-calculation" title="Permanent link"></a></h3>
<div style="float: left; width: 50%; padding: 20px;">
<p>The <strong>Jaccard Index</strong>, also known as the Jaccard Similarity Coefficient, is a statistic used for gauging the similarity and diversity between two sample sets. It ranges from 0 to 1, where if 0 the sets have no elements in common, whereas if 1 the sets are identical. In GAMBIT, the Jaccard Index is used to compare genetic sequences.</p>
<div class="admonition warning">
<p class="admonition-title">Jacard Distance vs Index</p>
<p>The <strong>Jaccard Distance</strong>, equal to one minus the Jaccard index, shares the same properties as the Jaccard index albeit inversely. It ranges from 0 to 1, where  0 the sets are identical and if 1, the sets have no elements in common.</p>
</div>
<p>In GAMBIT, the Jaccard Distance is calculated between two pre-computed k-mer sets in sparse coordinate format, one representing the query genome and another the GAMBIT database.</p>
<h3 id="built-in-species-distance-threshold">Built-in Species Distance Threshold<a class="headerlink" href="#built-in-species-distance-threshold" title="Permanent link"></a></h3>
<p>GAMBIT classifies unknown genomes by finding the distance to the closest reference genome and comparing that distance against the thresholds of the reference genomes species and genus.</p>
<p>For GAMBIT Prokaryotic database v2.0.0 and above, the threshold for a given species corresponds to the maximum intra-species distance ("max intra," or diameter) (<a href="#figure1">Figure 1</a>). Some species are not well separated from their closest sister taxon and, in some cases, even overlap. Such as the case of <em>Escherichia coli</em> and <em>Shigella sonnei</em> in GAMBITs Prokaryotic Database. In these scenarios, the species were divided into subspecies groups based on clustering of their intra-species distances, and then reporting matches to these subgroups and their parent species.</p>
<p>If the query genome distance is greater than the species diameter, GAMBIT attempts to report the genus. Genus diameters are computed and manually curated based on the diversity of the genus.</p>
</div>
<div style="float: right; width: 50%; padding: 20px;"></div>
<div class="admonition caption">
<p class="admonition-title">Distribution of GAMBIT distances</p>
<h5 id="figure1">Figure 1<a class="headerlink" href="#figure1" title="Permanent link"></a></h5>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-distribution-of-distances.png" data-desc-position="bottom"><img alt="**Figure 1: Distribution of GAMBIT distances within a species and to the nearest sister taxon in the GAMBIT reference database.** Three histograms are shown in each panel (each normalized independently). The green histogram represents the distribution of GAMBIT distances from each reference genome in the species to the closest genome also within the same species. The blue histogram represents the distribution of GAMBIT distances for all pairwise comparisons within the species. The red histogram represents the distribution of GAMBIT distances from each genome in the species of interest to the closest genome in the species closest sister taxon. The dashed blue line represents the classification threshold for that species in the GAMBIT database, which in both cases was derived from the maximum intra-species distance. Panel A shows *Klebsiella pneumoniae* and its closest sister taxon *Klebsiella variicola*, panel B shows *Neisseria gonorrhoeae* and its closest sister taxon *Neisseria meningitidis*.  ****https://doi.org/10.1371/journal.pone.0277575.g004" src="../../assets/figures/GAMBIT-distribution-of-distances.png"></a></p>
<p><strong>Figure 1: Distribution of GAMBIT distances within a species and to the nearest sister taxon in the GAMBIT reference database.</strong> Three histograms are shown in each panel (each normalized independently). The green histogram represents the distribution of GAMBIT distances from each reference genome in the species to the closest genome also within the same species. The blue histogram represents the distribution of GAMBIT distances for all pairwise comparisons within the species. The red histogram represents the distribution of GAMBIT distances from each genome in the species of interest to the closest genome in the species closest sister taxon. The dashed blue line represents the classification threshold for that species in the GAMBIT database, which in both cases was derived from the maximum intra-species distance. Panel A shows <em>Klebsiella pneumoniae</em> and its closest sister taxon <em>Klebsiella variicola</em>, panel B shows <em>Neisseria gonorrhoeae</em> and its closest sister taxon <em>Neisseria meningitidis</em>. </p>
<p>Sourced from <a href="https://doi.org/10.1371/journal.pone.0277575.g004">https://doi.org/10.1371/journal.pone.0277575.g004</a>.</p>
</div>
<div style="clear: both;"></div>
<div style="float: left; width: 50%; padding: 20px;">
<h3 id="gambit-distances-correlate-with-sequence-identity">GAMBIT distances correlate with <strong>sequence identity</strong><a class="headerlink" href="#gambit-distances-correlate-with-sequence-identity" title="Permanent link"></a></h3>
<p>Average Nucleotide Identity (ANI) has been the benchmark for nucleic acid comparisons and was used as a baseline measure of genomic similarity to validate the GAMBIT distance metric. ANI is generally used to determine similarity at the species or genus level with thresholds above 0.92 being optimal for species-level calls.</p>
<p>The ANI values were compared against GAMBIT distances for all pairs of genomes in each of the four data sets:</p>
<div class="admonition dna">
<p class="admonition-title">Test sets for GAMBIT distance versus ANI</p>
<table>
<thead>
<tr>
<th>Set</th>
<th>Number of Genomes</th>
<th>Phylogenetic Diversity</th>
<th>Assembly Quality</th>
<th>Reference</th>
</tr>
</thead>
<tbody>
<tr>
<td>Set 1</td>
<td>492</td>
<td>Low (<em>E. coli</em> only)</td>
<td>Medium</td>
<td><a href="https://doi.org/10.1186/s13059-016-0997-x">https://doi.org/10.1186/s13059-016-0997-x</a></td>
</tr>
<tr>
<td>Set 2</td>
<td>70</td>
<td>High (muliple phyla)</td>
<td>High</td>
<td><a href="https://doi.org/10.1073/pnas.0308653100">https://doi.org/10.1073/pnas.0308653100</a></td>
</tr>
<tr>
<td>Set 3</td>
<td>88</td>
<td>High (multiple phyla)</td>
<td>Medium</td>
<td><a href="https://doi.org/10.1371/journal.pone.0277575">https://doi.org/10.1371/journal.pone.0277575</a></td>
</tr>
<tr>
<td>Set 4</td>
<td>604</td>
<td>High (multiple phyla)</td>
<td>Medium</td>
<td><a href="https://doi.org/10.1371/journal.pone.0277575">https://doi.org/10.1371/journal.pone.0277575</a></td>
</tr>
</tbody>
</table>
</div>
<p>Spearman correlation was high in all four data sets (<a href="#figure2">Figure 2</a>) (Set 1 = -0.977; Set 2 = -0.968; Set 3 = -0.969; Set 4 = -0.979) for comparisons in which the ANI was reported by the FastANI tool (100%, 5.59%, 7.42% and 47.4%), revealing a nearly monotonic relationship between GAMBIT distance and ANI.</p>
</div>
<div style="float: right; width: 50%; padding: 20px;"></div>
<div class="admonition caption">
<p class="admonition-title">Relationship between GAMBIT distance and ANI</p>
<h5 id="figure2">Figure 2<a class="headerlink" href="#figure2" title="Permanent link"></a></h5>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-distance-vs-ani.png" data-desc-position="bottom"><img alt="**Figure 2: Relationship between GAMBIT distance and ANI (Average Nucleotide Identity).**  The relationship is nonlinear but very close to monotonic as measured by Spearman correlation (shown in the bottom left corner of each subplot). ANI was calculated using the FastANI tool with default parameter values. GAMBIT distances were calculated for all sets using the same parameter (k = 11, prefix = ATGAC). As FastANI only reports ANI values greater than ~80%, the fraction of total pairwise comparisons shown here were 100%, 5.5%, 7.4% and 47.4% for data sets 14 respectively. https://doi.org/10.1371/journal.pone.0277575.g001" src="../../assets/figures/GAMBIT-distance-vs-ani.png"></a></p>
<p><strong>Figure 2: Relationship between GAMBIT distance and ANI (Average Nucleotide Identity).</strong>  The relationship is nonlinear but very close to monotonic as measured by Spearman correlation (shown in the bottom left corner of each subplot). ANI was calculated using the <a href="https://github.com/ParBLiSS/FastANI">FastANI</a> tool with default parameter values. GAMBIT distances were calculated for all sets using the same parameter (k = 11, prefix = ATGAC). As FastANI only reports ANI values greater than ~80%, the fraction of total pairwise comparisons shown here were 100%, 5.5%, 7.4% and 47.4% for data sets 14 respectively. </p>
<p>Sourced from <a href="https://doi.org/10.1371/journal.pone.0277575.g001">https://doi.org/10.1371/journal.pone.0277575.g001</a>.</p>
</div>
<div style="clear: both;"></div>
<hr>
<h2 id="gambit-database-creation-and-curation">GAMBIT Database Creation and Curation<a class="headerlink" href="#gambit-database-creation-and-curation" title="Permanent link"></a></h2>
<p>As of <a href="#gambit-gtdb-database-v200">v2.0.0</a>, the GAMBIT Database is built iteratively over a <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database (GTDB)</a> release, starting with the species with the most publicly available genomes.</p>
<div class="admonition dna">
<p class="admonition-title">GAMBIT Database Creation can be done by you!</p>
<p><a href="../gambit_database/">GAMBIT Database Creation</a></p>
</div>
<p>The creation of a GAMBIT Database usually follows these steps:</p>
<ul>
<li>From a public repository, the genomes of interest are downloaded. Each taxa should have at least 2 genomes for GAMBIT diameters to be calculated;</li>
<li>The downloaded genomes undergo a round of quality-control to minimize the possibility that the genomes are too fragmented, contaminated or too incomplete. Third-party tools, such as <a href="https://github.com/ablab/quast">QUAST</a> and <a href="https://github.com/Ecogenomics/CheckM">CheckM</a>/<a href="https://busco.ezlab.org/">BUSCO</a> are used in this step;</li>
<li>GAMBIT distances are calculated for all pairs of genomes, removing overlapping sequences and misclassification to increase the accuracy of GAMBIT;</li>
<li>The database undergoes a curation to remove outliers and improve classification in underrepresented taxa and reflect public health usage and biological historical consensus more closely.</li>
</ul>
<p>Because GAMBIT databases have built-in species thresholds, genomes are included in each database version and the thresholds associated with each species are <strong>curated</strong> prior to release. Curation approaches may vary by GAMBIT database release but aim to ensure that <strong>mislabeled genomes are removed</strong> and that species are <strong>non-overlapping</strong>. Detailed information on the curation steps that each GAMBIT Database was subjected to can be found in the <a href="#gambit-databases">GAMBIT Databases</a> section.</p>
<div class="admonition tip">
<p class="admonition-title">Dependent on public data</p>
<p>Please note that GAMBIT databases undergo curation and testing prior to release, but are limited by the <strong>availability and accuracy</strong> of sequencing data in public repositories.</p>
</div>
<hr>
<h2 id="gambit-faqs">GAMBIT FAQs<a class="headerlink" href="#gambit-faqs" title="Permanent link"></a></h2>
<details class="toggle">
<summary>What GAMBIT database should I use?</summary>
<p>GAMBIT Databases are domain-specific. Currently two domains are available: Bacteria and Fungi. Choosing the appropriate type of database for your data is important as it can lead to erroneous/no classification results.</p>
<p>As a rule of thumb, we recommend the latest version of any GAMBIT Database to be used.  Instances where one might prefer to use an older database, versus the most up to date, include:</p>
<ol>
<li>Maintaining use of a database that has been validated previously by your laboratory, or </li>
<li>Utilizing a database that draws the genomes and their annotations from a specific source. For example, bacterial GAMBIT databases v1.0.0 through v1.3.0 draw their genome annotations predominantly from NCBIs RefSeq database, whereas v2.0.0 draws all genome annotations from GTDB. Database v1.0.0 is also inclusive of all bacterial genomes that were available on RefSeq at the time of creation, whereas v2.0.0 excludes genomes that do not expand the diversity of their species.</li>
</ol>
</details>
<details class="toggle">
<summary>How do I list taxa included in a GAMBIT database?</summary>
<p>There are several ways to retrieve the information regarding which taxa were included in a given GAMBIT database release. The easiest way is to download the taxa list file provided <a href="../gambit_database/">on this documentation page</a> for every GAMBIT database release.</p>
<p>Additionally, there are several programmatic ways to retrieve this information directly from the GAMBIT metadata file (which typically ends in ".gdb"). Here we present a few examples: using <a href="https://www.sqlite.org/">SQLite3</a>, <a href="https://dbeaver.io/">DBeaver</a> or the <a href="https://github.com/gambit-suite/gambittools">GAMBITtools</a> software.</p>
<details class="example">
<summary><em>Example 1:</em> SQLite3</summary>
<p>To retrieve the list of taxa directly from the database, the following command can be run (after installing SQLite3 through your favourite installer). Substitute <code>&lt;gambit metadata gdb file&gt;</code> with your metadata file location.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>sqlite3<span class="w"> </span>&lt;gambit<span class="w"> </span>metadata<span class="w"> </span>gdb<span class="w"> </span>file&gt;<span class="w"> </span><span class="s2">"SELECT * FROM taxa;"</span><span class="w"> </span>&gt;<span class="w"> </span>list-of-taxa.tsv
</span></code></pre></div>
<p>To retrieve the list of genomes, the following command can be run. Substitute <code>&lt;gambit metadata gdb file&gt;</code> by your metadata file location.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>sqlite3<span class="w"> </span>&lt;gambit<span class="w"> </span>metadata<span class="w"> </span>gdb<span class="w"> </span>file&gt;<span class="w"> </span><span class="s2">"SELECT * FROM genomes;"</span><span class="w"> </span>&gt;<span class="w"> </span>list-of-genome.tsv
</span></code></pre></div>
</details>
<details class="example">
<summary><em>Example 2:</em> DBeaver</summary>
<p>After downloading and installing <a href="https://dbeaver.io/">DBeaver</a>, open the GAMBIT metadata file by clicking on <code>New Database Connection</code> (or hitting Ctrl+Shift+N) on the top left corner of the window. Under the SQL section, select the <code>SQLite</code> option and open the path to the metadata file. If prompted, install the required drivers by DBeaver. </p>
<div class="admonition caption narrow">
<p class="admonition-title">SQLite</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-dbeaver-open-sqlite.png" data-desc-position="bottom"><img alt="SQLite" src="../../assets/figures/GAMBIT-dbeaver-open-sqlite.png"></a></p>
</div>
<p>Select <code>SQL Editor</code> in the toolbar and then click on <code>New SQL Script</code>. </p>
<div class="admonition caption narrow">
<p class="admonition-title">New SQL Script</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-dbeaver-new-sqlite-script.png" data-desc-position="bottom"><img alt="New SQL Script" src="../../assets/figures/GAMBIT-dbeaver-new-sqlite-script.png"></a></p>
</div>
<p>Type <code>SELECT * FROM taxa;</code> and check that you get the results (press <code>CTRL+Enter</code> or click the orange arrow to execute SQL statements). To save the results click on <code>Export data</code>  on the bottom right corner and select what file format to save the information in (we recommend <strong>CSV format</strong> that can then be loaded onto Excel). </p>
<div class="admonition caption narrow">
<p class="admonition-title">List Taxa</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-dbeaver-list-taxa.png" data-desc-position="bottom"><img alt="List Taxa" src="../../assets/figures/GAMBIT-dbeaver-list-taxa.png"></a></p>
</div>
</details>
<details class="example">
<summary><em>Example 3:</em> GAMBITtools</summary>
<p>The <a href="https://github.com/gambit-suite/gambittools">GAMBITtools</a> suite of scripts are Python tools written for working with GAMBIT. We recommend using <a href="https://www.docker.com/">Docker</a> to interact with GAMBITtools. </p>
<p>After building the gambittools docker image, use the <code>gambit-list-taxa</code> command as demonstrated below to generate a list of taxa included in a GAMBIT database. Substitute <code>&lt;gambit metadata gdb file&gt;</code> with your metadata file location</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>docker<span class="w"> </span>build<span class="w"> </span>-t<span class="w"> </span>gambittools<span class="w"> </span>.
</span></code></pre></div>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>docker<span class="w"> </span>run<span class="w"> </span>-v<span class="w"> </span><span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>:/data<span class="w"> </span>gambittools<span class="w"> </span>gambit-list-taxa<span class="w"> </span>&lt;gambit<span class="w"> </span>metadata<span class="w"> </span>gdb<span class="w"> </span>file&gt;
</span></code></pre></div>
</details>
</details>
<details class="toggle">
<summary>How do I get the number of genomes representing a given species?</summary>
<p>Like retrieving the list of taxa, there are several ways of retrieving the number of species for a given species. </p>
<details class="example">
<summary><em>Example 1:</em> SQLite3</summary>
<p>To retrieve the list of taxa and respective number of genomes directly from the database, the following command can be run (after installing SQLite3 through your favourite installer). Substitute <code>&lt;gambit metadata gdb file&gt;</code> with your metadata file location.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>sqlite3<span class="w"> </span>&lt;gambit<span class="w"> </span>metadata<span class="w"> </span>gdb<span class="w"> </span>file&gt;<span class="w"> </span><span class="s2">"SELECT taxa.name,COUNT(genome_annotations.taxon_id) FROM taxa LEFT JOIN genome_annotations ON genome_annotations.taxon_id = taxa.id WHERE taxa.rank LIKE 'species' GROUP BY taxa.key ORDER BY taxa.name ASC;"</span><span class="w"> </span>&gt;<span class="w"> </span>list-of-taxa-with-number-of-genomes.tsv
</span></code></pre></div>
</details>
<details class="example">
<summary><em>Example 2:</em> DBeaver</summary>
<p>After downloading and installing <a href="https://dbeaver.io/">DBeaver</a>, open the GAMBIT metadata file by clicking on <code>New Database Connection</code> (or hitting Ctrl+Shift+N) on the top left corner of the window. Under the SQL section, select the <code>SQLite</code> option and open the path to the metadata file. If prompted, install the required drivers by DBeaver. </p>
<div class="admonition caption narrow">
<p class="admonition-title">SQLite</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-dbeaver-open-sqlite.png" data-desc-position="bottom"><img alt="SQLite" src="../../assets/figures/GAMBIT-dbeaver-open-sqlite.png"></a></p>
</div>
<p>Select <code>SQL Editor</code> in the toolbar and then click on <code>New SQL Script</code>. </p>
<div class="admonition caption narrow">
<p class="admonition-title">New SQL Script</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-dbeaver-new-sqlite-script.png" data-desc-position="bottom"><img alt="New SQL Script" src="../../assets/figures/GAMBIT-dbeaver-new-sqlite-script.png"></a></p>
</div>
<p>Type <code>SELECT taxa.name,COUNT(genome_annotations.taxon_id) FROM taxa LEFT JOIN genome_annotations ON genome_annotations.taxon_id = taxa.id WHERE taxa.rank LIKE 'species' GROUP BY taxa.key ORDER BY taxa.name ASC;</code> and check that you get the results (press <code>CTRL+Enter</code> or click the orange arrow to execute SQL statements). To save the results click on <code>Export data</code>  on the bottom right corner and select what file format to save the information in (we recommend <strong>CSV format</strong> that can then be loaded onto Excel). </p>
<div class="admonition caption narrow">
<p class="admonition-title">Number of Genomes</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-dbeaver-number-genomes.png" data-desc-position="bottom"><img alt="Number of Genomes" src="../../assets/figures/GAMBIT-dbeaver-number-genomes.png"></a></p>
</div>
</details>
</details>
<details class="toggle">
<summary>How do I get the number of genomes and the distance threshold representing a given species?</summary>
<p>Like retrieving the list of taxa and the number of genomes representing a given species, there are several ways of retrieving the distance threshold for a given species. </p>
<details class="example">
<summary><em>Example 1:</em> SQLite3</summary>
<p>To retrieve the list of taxa and respective number of genomes directly from the database, the following command can be run (after installing SQLite3 through your favourite installer). Substitute <code>&lt;gambit metadata gdb file&gt;</code> with your metadata file location.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>sqlite3<span class="w"> </span>&lt;gambit<span class="w"> </span>metadata<span class="w"> </span>gdb<span class="w"> </span>file&gt;<span class="w"> </span><span class="s2">"SELECT taxa.name,taxa.distance_threshold,COUNT(genome_annotations.taxon_id) FROM taxa LEFT JOIN genome_annotations ON genome_annotations.taxon_id = taxa.id WHERE taxa.rank LIKE 'species' GROUP BY taxa.key ORDER BY taxa.name ASC;"</span><span class="w"> </span>&gt;<span class="w"> </span>list-of-taxa-with-number-of-genomes.tsv
</span></code></pre></div>
</details>
<details class="example">
<summary>Example 2: DBeaver</summary>
<p>After downloading and installing <a href="https://dbeaver.io/">DBeaver</a>, open the GAMBIT metadata file by clicking on <code>New Database Connection</code> (or hitting Ctrl+Shift+N) on the top left corner of the window. Under the SQL section, select the <code>SQLite</code> option and open the path to the metadata file. If prompted, install the required drivers by DBeaver. </p>
<div class="admonition caption narrow">
<p class="admonition-title">SQLite</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-dbeaver-open-sqlite.png" data-desc-position="bottom"><img alt="SQLite" src="../../assets/figures/GAMBIT-dbeaver-open-sqlite.png"></a></p>
</div>
<p>Select <code>SQL Editor</code> in the toolbar and then click on <code>New SQL Script</code>. </p>
<div class="admonition caption narrow">
<p class="admonition-title">New SQL Script</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-dbeaver-new-sqlite-script.png" data-desc-position="bottom"><img alt="New SQL Script" src="../../assets/figures/GAMBIT-dbeaver-new-sqlite-script.png"></a></p>
</div>
<p>Type <code>SELECT taxa.name,taxa.distance_threshold,COUNT(genome_annotations.taxon_id) FROM taxa LEFT JOIN genome_annotations ON genome_annotations.taxon_id = taxa.id WHERE taxa.rank LIKE 'species' GROUP BY taxa.key ORDER BY taxa.name ASC;</code> and check that you get the results (press <code>CTRL+Enter</code> or click the orange arrow to execute SQL statements). To save the results click on <code>Export data</code>  on the bottom right corner and select what file format to save the information in (we recommend <strong>CSV format</strong> that can then be loaded onto Excel). </p>
<div class="admonition caption narrow">
<p class="admonition-title">Number of Genomes and Distance Threshold</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../assets/figures/GAMBIT-dbeaver-number-genomes-and-distance-threshold.png" data-desc-position="bottom"><img alt="Number Genomes and Distance Threshold" src="../../assets/figures/GAMBIT-dbeaver-number-genomes-and-distance-threshold.png"></a></p>
</div>
</details>
</details>
<details class="toggle">
<summary>How do I create a custom GAMBIT database?</summary>
<p>Creating a custom GAMBIT database can be a laborious task.  The easiest way to go about it is to reach out to Theiagen Genomics at <a href="mailto:support@theiagen.com">support@theiagen.com</a> to request assistance.  A guide can be found on <a href="../gambit_database/">GAMBIT Database Creation</a> </p>
</details>
<details class="toggle">
<summary>How well does GAMBIT perform discerning between <em>Escherichia coli</em> and <em>Shigella</em> sp?</summary>
<p>Escherichia coli and Shigella are closely genetically related, to the extent that they would be considered the same species if not for their distinguishing phenotypic characteristics. GAMBIT databases are curated to enable differentiation between the two groups, however, it is worth bearing in mind that these genomes are highly genetically similar thus tools that take a more granular approach to genome comparison may be more reliable. </p>
</details>
<details class="toggle">
<summary>What should I do if a GAMBIT taxonomic assignment does not align with the expected results based on another bioinformatics tool or molecular testing?</summary>
<p>In this instance, please reach out to <a href="mailto:support@theiagen.com">support@theiagen.com</a> and David Hess at the Nevada State Public Health Laboratory <a href="mailto:dhess@med.unr.edu">dhess@med.unr.edu</a>. We will be happy to investigate your sample and improve the GAMBIT database in subsequent versions!</p>
</details>
<hr>
<h2 id="other-resources">Other Resources<a class="headerlink" href="#other-resources" title="Permanent link"></a></h2>
<ul>
<li><a href="https://doi.org/10.1371/journal.pone.0277575">Original GAMBIT publication</a></li>
<li><a href="https://doi.org/10.3389/fpubh.2023.1198213">GAMBIT Fungal Database publication</a></li>
<li><a href="https://github.com/jlumpe/gambit/tree/master">GAMBIT Software GitHub</a> - repository of GAMBIT software</li>
<li><a href="https://github.com/gambit-suite">GAMBIT suite GitHub</a> - tools for working with GAMBIT including GAMBITdb for automated GAMBIT database creation</li>
</ul>
<hr>
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link"></a></h2>
<p>Please cite the article below when using the <strong>GAMBIT software and/or GAMBIT RefSeq Curated Database v1.0.0:</strong></p>
<blockquote>
<p>Lumpe J, Gumbleton L, Gorzalski A, Libuit K, Varghese V, et al. (2023) GAMBIT (Genomic Approximation Method for Bacterial Identification and Tracking): A methodology to rapidly leverage whole genome sequencing of bacterial isolates for clinical identification. PLOS ONE 18(2): e0277575.&nbsp;<a href="https://doi.org/10.1371/journal.pone.0277575">https://doi.org/10.1371/journal.pone.0277575</a></p>
</blockquote>
<p>Please cite the reference below when using the <strong>GAMBIT Fungal Database v0.2.0:</strong></p>
<blockquote>
<p>Ambrosio III, F. J., Scribner, M. R., Wright, S. M., Otieno, J. R., Doughty, E. L., Gorzalski, A., ... &amp; Hess, D. (2023). TheiaEuk: a species-agnostic bioinformatics workflow for fungal genomic characterization.&nbsp;<em>Frontiers in Public Health</em>,&nbsp;<em>11</em>. <a href="https://doi.org/10.3389/fpubh.2023.1198213">https://doi.org/10.3389/fpubh.2023.1198213</a></p>
</blockquote>







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="June 26, 2025 16:35:00 UTC">2025-06-26</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="March 26, 2025 19:00:08 UTC">2025-03-26</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
       2022-2025 <a href="https://www.theiagen.com" target="_blank" rel="noopener">Theiagen Genomics</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/theiagen" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/theiagen" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M459.4 151.7c.3 4.5.3 9.1.3 13.6 0 138.7-105.6 298.6-298.6 298.6-59.5 0-114.7-17.2-161.1-47.1 8.4 1 16.6 1.3 25.3 1.3 49.1 0 94.2-16.6 130.3-44.8-46.1-1-84.8-31.2-98.1-72.8 6.5 1 13 1.6 19.8 1.6 9.4 0 18.8-1.3 27.6-3.6-48.1-9.7-84.1-52-84.1-103v-1.3c14 7.8 30.2 12.7 47.4 13.3-28.3-18.8-46.8-51-46.8-87.4 0-19.5 5.2-37.4 14.3-53C87.4 130.8 165 172.4 252.1 176.9c-1.6-7.8-2.6-15.9-2.6-24C249.5 95.1 296.3 48 354.4 48c30.2 0 57.5 12.7 76.7 33.1 23.7-4.5 46.5-13.3 66.6-25.3-7.8 24.4-24.4 44.8-46.1 57.8 21.1-2.3 41.6-8.1 60.4-16.2-14.3 20.8-32.2 39.3-52.6 54.3"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.linkedin.com/company/theiagen" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3M135.4 416H69V202.2h66.5V416zM102.2 96a38.5 38.5 0 1 1 0 77 38.5 38.5 0 1 1 0-77m282.1 320h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9z"></path></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "navigation.instant", "navigation.path", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.suggest", "toc.follow"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"alias": true, "default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.92b07e13.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../../javascripts/tablesort.js"></script>
      
        <script src="../../javascripts/table-search.js"></script>
      
    
  
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>