<!DOCTYPE html><html lang="en" class="no-js"><head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A collection of bioinformatics workflows for genomic characterization, submission preparation, and genomic epidemiology of pathogens of public health concern.">
      
      
        <meta name="author" content="Theiagen Genomics">
      
      
        <link rel="canonical" href="https://theiagen.github.io/public_health_bioinformatics/main/workflows/genomic_characterization/theiacov/">
      
      
        <link rel="prev" href="../../public_data_sharing/terra_2_ncbi/">
      
      
        <link rel="next" href="../theiaeuk/">
      
      
      <link rel="icon" href="../../../assets/logos/Theiagen-Symbol-Standard-01.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.20">
    
    
      
        <title>TheiaCoV Workflow Series - Public Health Bioinformatics</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.e53b48f4.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  <link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#theiacov-workflow-series" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
          <aside class="md-banner md-banner--warning">
            <div class="md-banner__inner md-grid md-typeset">
              
  You're not viewing the latest version release.
  <a href="../../../.."> 
    <strong>Click here to go to the latest version release.</strong>
  </a>

            </div>
            <script>var el=document.querySelector("[data-md-component=outdated]"),base=new URL("../../.."),outdated=__md_get("__outdated",sessionStorage,base);!0===outdated&&el&&(el.hidden=!1)</script>
          </aside>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-header__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Public Health Bioinformatics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              TheiaCoV Workflow Series
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="light" data-md-color-primary="indigo" data-md-color-accent="indigo" aria-label="Switch to Dark Mode" type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to Dark Mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"></path></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo" aria-label="Switch to Light Mode" type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to Light Mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m3.55 19.09 1.41 1.41 1.8-1.79-1.42-1.42M12 6c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6c0-3.32-2.69-6-6-6m8 7h3v-2h-3m-2.76 7.71 1.8 1.79 1.41-1.41-1.79-1.8M20.45 5l-1.41-1.4-1.8 1.79 1.42 1.42M13 1h-2v3h2M6.76 5.39 4.96 3.6 3.55 5l1.79 1.81zM1 13h3v-2H1m12 9h-2v3h2"></path></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../.." class="md-tabs__link">
          
  
  
  Public Health Bioinformatics

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="https://www.theiagen.com" title="Public Health Bioinformatics" class="md-nav__button md-logo" aria-label="Public Health Bioinformatics" data-md-component="logo">
      
  <img src="../../../assets/logos/Theiagen-Logo-White.png" alt="logo">

    </a>
    Public Health Bioinformatics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/theiagen/public_health_bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Public Health Bioinformatics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Public Health Bioinformatics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Getting Started
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            Getting Started
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/commandline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With the Command-Line
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/terra/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    With Terra.bio
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting_started/philosophy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Workflow Failure Philosophy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Other Resources
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_3">
            <span class="md-nav__icon md-icon"></span>
            Other Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/sops/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Available SOPs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/versions/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Database and Docker Versions
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/gambit_database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to GAMBIT Database Creation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/phylogenetics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Phylogenetics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/custom_organisms/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Guide to Running Custom Organisms on TheiaCoV
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1">
        
          
          <label class="md-nav__link" for="__nav_1_5_1" id="__nav_1_5_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Type
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Type
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_type/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_2" id="__nav_1_5_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Data Import
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_2">
            <span class="md-nav__icon md-icon"></span>
            Data Import
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_3" id="__nav_1_5_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Genomic Characterization
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_3">
            <span class="md-nav__icon md-icon"></span>
            Genomic Characterization
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiacov-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaCoV Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaCoV Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#supported-organisms" class="md-nav__link">
    <span class="md-ellipsis">
      Supported Organisms
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#org-specific" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-Specific Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Core Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Assembly Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#post-assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Post-Assembly Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#org-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-specific Characterization Tasks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_4" id="__nav_1_5_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Construction
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_4">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Construction
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_5" id="__nav_1_5_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Phylogenetic Placement
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_5">
            <span class="md-nav__icon md-icon"></span>
            Phylogenetic Placement
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_6" id="__nav_1_5_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Public Data Sharing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_6">
            <span class="md-nav__icon md-icon"></span>
            Public Data Sharing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_7">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_7" id="__nav_1_5_1_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Exporting Data from Terra
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_7">
            <span class="md-nav__icon md-icon"></span>
            Exporting Data from Terra
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_1_8">
        
          
          <label class="md-nav__link" for="__nav_1_5_1_8" id="__nav_1_5_1_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Standalone
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_1_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_1_8">
            <span class="md-nav__icon md-icon"></span>
            Standalone
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2" id="__nav_1_5_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows by Kingdom
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2">
            <span class="md-nav__icon md-icon"></span>
            Workflows by Kingdom
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_kingdom/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_2">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_2" id="__nav_1_5_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Any Taxa
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_2">
            <span class="md-nav__icon md-icon"></span>
            Any Taxa
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_3">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_3" id="__nav_1_5_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Bacteria
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_3">
            <span class="md-nav__icon md-icon"></span>
            Bacteria
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_4">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_4" id="__nav_1_5_2_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Mycotics
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_4">
            <span class="md-nav__icon md-icon"></span>
            Mycotics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_2_5">
        
          
          <label class="md-nav__link" for="__nav_1_5_2_5" id="__nav_1_5_2_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Viral
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_1_5_2_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_2_5">
            <span class="md-nav__icon md-icon"></span>
            Viral
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiacov-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaCoV Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaCoV Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#supported-organisms" class="md-nav__link">
    <span class="md-ellipsis">
      Supported Organisms
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#org-specific" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-Specific Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Core Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Assembly Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#post-assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Post-Assembly Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#org-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-specific Characterization Tasks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_5_3" checked>
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Workflows Alphabetically
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            Workflows Alphabetically
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../workflows_overview/workflows_alphabetically/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/amr_search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    AMR_Search
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/assembly_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/augur/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Augur
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/basespace_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BaseSpace_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/cauris_cladetyper/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cauris_CladeTyper
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/clair3_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Clair3_Variants_ONT
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/concatenate_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/concatenate_illumina_lanes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Concatenate_Illumina_Lanes
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/core_gene_snp/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Core_Gene_SNP
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/create_terra_table/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Create_Terra_Table
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/czgenepi_prep/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CZGenEpi_Prep
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/dorado_basecalling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Dorado_Basecalling
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/fetch_srr_accession/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Fetch_SRR_Accession
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/find_shared_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Find_Shared_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../freyja/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Freyja Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/gambit_query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    GAMBIT_Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/kraken2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Kraken2
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp3/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP3
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/ksnp4/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    kSNP4
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/lyve_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lyve_SET
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/mashtree_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    MashTree_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/mercury_prep_n_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Mercury_Prep_N_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_amrfinderplus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-AMRFinderPlus
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/ncbi_scrub/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    NCBI-Scrub
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/nextclade_batch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Nextclade_Batch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/ont_barcode_concatenation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ONT_Barcode_Concatenation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pangolin_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pangolin_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/phylocompare/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    PhyloCompare
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rasusa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RASUSA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/rename_fastq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Rename_FASTQ
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_streamline_fasta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Streamline_FASTA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_tree/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Tree
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_construction/snippy_variants/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Snippy_Variants
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_import/sra_fetch/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA_Fetch
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/tbprofiler_tngs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TBProfiler_tNGS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ena/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_ENA
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_gisaid/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_GISAID
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../public_data_sharing/terra_2_ncbi/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Terra_2_NCBI
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    TheiaCoV Workflow Series
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiacov-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaCoV Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaCoV Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#supported-organisms" class="md-nav__link">
    <span class="md-ellipsis">
      Supported Organisms
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#org-specific" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-Specific Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Core Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Assembly Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#post-assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Post-Assembly Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#org-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-specific Characterization Tasks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaeuk/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaEuk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiameta/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaMeta
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaprok/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaProk Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../standalone/theiavalidate/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaValidate
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../theiaviral/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TheiaViral Workflow Series
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/transfer_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transfer_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../phylogenetic_placement/usher/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usher
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vadr_update/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VADR_Update
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data_export/zip_column_content/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Zip_Column_Content
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_6">
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Contributing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            Contributing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/doc_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Documentation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/code_contribution/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Contributing to the Code
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-facts" class="md-nav__link">
    <span class="md-ellipsis">
      Quick Facts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#theiacov-workflows" class="md-nav__link">
    <span class="md-ellipsis">
      TheiaCoV Workflows
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TheiaCoV Workflows">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#supported-organisms" class="md-nav__link">
    <span class="md-ellipsis">
      Supported Organisms
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inputs" class="md-nav__link">
    <span class="md-ellipsis">
      Inputs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#org-specific" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-Specific Parameters
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#core-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Core Tasks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Tasks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Assembly Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#post-assembly-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Post-Assembly Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#org-specific-tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Organism-specific Characterization Tasks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="theiacov-workflow-series">TheiaCoV Workflow Series<a class="headerlink" href="#theiacov-workflow-series" title="Permanent link">¶</a></h1>
<h2 id="quick-facts">Quick Facts<a class="headerlink" href="#quick-facts" title="Permanent link">¶</a></h2>
<table>
<thead>
<tr>
<th><strong>Workflow Type</strong></th>
<th><strong>Applicable Kingdom</strong></th>
<th><strong>Last Known Changes</strong></th>
<th><strong>Command-line Compatibility</strong></th>
<th><strong>Workflow Level</strong></th>
<th><strong>Dockstore</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../../../workflows_overview/workflows_type/#genomic-characterization">Genomic Characterization</a></td>
<td>HIV, Influenza, Monkeypox virus, RSV-A, RSV-B, SARS-CoV-2, <a href="../../../workflows_overview/workflows_kingdom/#viral">Viral</a>, WNV</td>
<td>vX.X.X</td>
<td>Some optional features incompatible, Yes</td>
<td>Sample-level, Set-level</td>
<td><a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaCoV_Illumina_PE_PHB:main?tab=info">TheiaCoV_Illumina_PE_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaCoV_Illumina_SE_PHB:main?tab=info">TheiaCoV_Illumina_SE_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaCoV_ONT_PHB:main?tab=info">TheiaCoV_ONT_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaCoV_ClearLabs_PHB:main?tab=info">TheiaCoV_ClearLabs_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaCoV_FASTA_PHB:main?tab=info">TheiaCoV_FASTA_PHB</a>, <a href="https://dockstore.org/workflows/github.com/theiagen/public_health_bioinformatics/TheiaCoV_FASTA_Batch_PHB:main?tab=info">TheiaCoV_FASTA_Batch_PHB</a></td>
</tr>
</tbody>
</table>
<h2 id="theiacov-workflows">TheiaCoV Workflows<a class="headerlink" href="#theiacov-workflows" title="Permanent link">¶</a></h2>
<p><strong>The TheiaCoV workflows are for the assembly, quality assessment, and characterization of viral genomes.</strong> There are currently five TheiaCoV workflows designed to accommodate different kinds of input data:</p>
<ol>
<li>Illumina paired-end sequencing (<strong>TheiaCoV_Illumina_PE</strong>)</li>
<li>Illumina single-end sequencing (<strong>TheiaCoV_Illumina_SE</strong>)</li>
<li>ONT sequencing (<strong>TheiaCoV_ONT</strong>)</li>
<li>Genome assemblies (<strong>TheiaCoV_FASTA</strong>)</li>
<li>ClearLabs sequencing (<strong>TheiaCoV_ClearLabs</strong>)</li>
</ol>
<p>Additionally, the <strong>TheiaCoV_FASTA_Batch</strong> workflow is available to process several hundred SARS-CoV-2 assemblies at the same time.</p>
<hr>
<div class="admonition dna inline end">
<p class="admonition-title">Key Resources</p>
<p><a href="https://theiagen.notion.site/Docker-Image-and-Reference-Materials-for-SARS-CoV-2-Genomic-Characterization-98328c61f5cb4f77975f512b55d09108?pvs=74"><strong>Reference Materials for SARS-CoV-2</strong></a></p>
<p><a href="https://theiagen.notion.site/Workspace-Reference-Materials-for-MPXV-Genomic-Characterization-a34f355c68c54c0a82e926d4de607bca"><strong>Reference Materials for Mpox</strong></a></p>
<p><a href="../theiaviral/"><strong>Reference Materials for non-default viruses (TheiaViral)</strong></a></p>
<details class="toggle">
<summary>HIV Input JSONs</summary>
<ul>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_Illumina_PE_HIV_v1_2024-04-19.json">TheiaCoV_Illumina_PE_HIV_v1_2024-04-19.json</a></li>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_Illumina_PE_HIV_v2_2024-04-19.json">TheiaCoV_Illumina_PE_HIV_v2_2024-04-19.json</a></li>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_ONT_HIV_v1_2024-04-19.json">TheiaCoV_ONT_HIV_v1_2024-04-19.json</a></li>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_ONT_HIV_v2_2024-04-19.json">TheiaCoV_ONT_HIV_v2_2024-04-19.json</a></li>
</ul>
</details>
<details class="toggle">
<summary>WNV Input JSONs</summary>
<ul>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_Illumina_PE_WNV_2024-04-19.json">TheiaCoV_Illumina_PE_WNV_2024-04-19.json</a></li>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_Illumina_SE_WNV_2024-04-19.json">TheiaCoV_Illumina_SE_WNV_2024-04-19.json</a></li>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_FASTA_WNV_2024-04-19.json">TheiaCoV_FASTA_WNV_2024-04-19.json</a></li>
</ul>
</details>
<details class="toggle">
<summary>Flu Input JSONs</summary>
<ul>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_Illumina_PE_flu_2024-04-19.json">TheiaCoV_Illumina_PE_flu_2024-04-19.json</a></li>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_ONT_flu_2024-04-19.json">TheiaCoV_ONT_flu_2024-04-19.json</a></li>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_FASTA_flu_2024-04-19.json">TheiaCoV_FASTA_flu_2024-04-19.json</a></li>
</ul>
</details>
<details class="toggle">
<summary>RSV-A Input JSONs</summary>
<ul>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_Illumina_PE_RSV-A_2024-04-19.json">TheiaCoV_Illumina_PE_RSV-A_2024-04-19.json</a></li>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_FASTA_RSV-A_2024-04-19.json">TheiaCoV_FASTA_RSV-A_2024-04-19.json</a></li>
</ul>
</details>
<details class="toggle">
<summary>RSV-B Input JSONs</summary>
<ul>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_Illumina_PE_RSV-B_2024-04-19.json">TheiaCoV_Illumina_PE_RSV-B_2024-04-19.json</a></li>
<li><a href="../../../assets/files/input_jsons/TheiaCoV_FASTA_RSV-B_2024-04-19.json">TheiaCoV_FASTA_RSV-B_2024-04-19.json</a></li>
</ul>
</details>
</div>
<div class="admonition caption">
<p class="admonition-title">TheiaCoV Workflow Diagram</p>
<p><a class="glightbox" data-type="image" data-width="auto" data-height="auto" href="../../../assets/figures/TheiaCoV.png" data-desc-position="bottom"><img alt="TheiaCoV Workflow Diagram" src="../../../assets/figures/TheiaCoV.png"></a></p>
</div>
<h3 id="supported-organisms">Supported Organisms<a class="headerlink" href="#supported-organisms" title="Permanent link">¶</a></h3>
<p>These workflows currently support the following organisms. The first option in the list (bolded) is what our workflows use as the <em>standardized</em> organism name:</p>
<ul>
<li><strong>SARS-CoV-2</strong> (<strong><code>"sars-cov-2"</code></strong>, <code>"SARS-CoV-2"</code>) - <mark><em>default organism input</em></mark></li>
<li><strong>Monkeypox virus</strong> (<strong><code>"MPXV"</code></strong>, <code>"mpox"</code>, <code>"monkeypox"</code>, <code>"Monkeypox virus"</code>, <code>"Mpox"</code>)</li>
<li><strong>Human Immunodeficiency Virus</strong> (<strong><code>"HIV"</code></strong>)</li>
<li><strong>West Nile Virus</strong> (<strong><code>"WNV"</code></strong>, <code>"wnv"</code>, <code>"West Nile virus"</code>)</li>
<li><strong>Influenza</strong> (<strong><code>"flu"</code></strong>, <code>"influenza"</code>, <code>"Flu"</code>, <code>"Influenza"</code>)</li>
<li><strong>RSV-A</strong> (<strong><code>"rsv_a"</code></strong>, <code>"rsv-a"</code>, <code>"RSV-A"</code>, <code>"RSV_A"</code>)</li>
<li><strong>RSV-B</strong> (<strong><code>"rsv_b"</code></strong>, <code>"rsv-b"</code>, <code>"RSV-B"</code>, <code>"RSV_B"</code>)</li>
<li><strong>Measles</strong> (<strong><code>"measles"</code></strong>, <code>"Measles"</code>, <code>"mev"</code>, <code>"MeV"</code>, <code>"Morbillivirus"</code>, <code>"morbillivirus"</code>)</li>
<li><strong>Mumps</strong> (<strong><code>"mumps"</code></strong>, <code>"Mumps"</code>, <code>"MuV"</code>, <code>"muv"</code>, <code>"Mumps virus"</code>, <code>"mumps virus"</code>)</li>
<li><strong>Rubella</strong> (<strong><code>"rubella"</code></strong>, <code>"Rubella"</code>, <code>"RuV"</code>, <code>"ruv"</code>, <code>"Rubella virus"</code>, <code>"rubella virus"</code>)</li>
</ul>
<p>The compatibility of each workflow with each pathogen is shown below:</p>
<table>
<thead>
<tr>
<th></th>
<th>SARS-CoV-2</th>
<th>Mpox</th>
<th>HIV</th>
<th>WNV</th>
<th>Influenza</th>
<th>RSV-A</th>
<th>RSV-B</th>
<th>Measles</th>
<th>Mumps</th>
<th>Rubella</th>
</tr>
</thead>
<tbody>
<tr>
<td>Illumina_PE</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
</tr>
<tr>
<td>Illumina_SE</td>
<td>✅</td>
<td>✅</td>
<td>❌</td>
<td>✅</td>
<td>❌</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
</tr>
<tr>
<td>ONT</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>❌</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
</tr>
<tr>
<td>FASTA</td>
<td>✅</td>
<td>✅</td>
<td>❌</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
</tr>
<tr>
<td>ClearLabs</td>
<td>✅</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
</tr>
</tbody>
</table>
<p>We've provided the following information to help you set up the workflow for each organism in the form of input JSONs.</p>
<h3 id="inputs">Inputs<a class="headerlink" href="#inputs" title="Permanent link">¶</a></h3>
<div class="admonition dna">
<p class="admonition-title">Input Data</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:6"><input checked="checked" id="theiacov_illumina_pe" name="__tabbed_1" type="radio"><input id="theiacov_illumina_se" name="__tabbed_1" type="radio"><input id="theiacov_ont" name="__tabbed_1" type="radio"><input id="theiacov_fasta" name="__tabbed_1" type="radio"><input id="theiacov_clearlabs" name="__tabbed_1" type="radio"><input id="theiacov_fasta_batch" name="__tabbed_1" type="radio"><div class="tabbed-labels"><label for="theiacov_illumina_pe">TheiaCoV_Illumina_PE</label><label for="theiacov_illumina_se">TheiaCoV_Illumina_SE</label><label for="theiacov_ont">TheiaCoV_ONT</label><label for="theiacov_fasta">TheiaCoV_FASTA</label><label for="theiacov_clearlabs">TheiaCoV_ClearLabs</label><label for="theiacov_fasta_batch">TheiaCoV_FASTA_Batch</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>The TheiaCoV_Illumina_PE workflow takes in Illumina paired-end read data. Read file names should end with <code>.fastq</code> or <code>.fq</code>, with the optional addition of <code>.gz</code>. When possible, Theiagen recommends zipping files with <a href="https://www.gnu.org/software/gzip/">gzip</a> before Terra uploads to minimize data upload time.</p>
<p>By default, the workflow anticipates&nbsp;<strong>2 x 150bp</strong>&nbsp;reads (i.e. the input reads were generated using a 300-cycle sequencing kit). Modifications to the optional parameter for <code>trim_minlen</code> may be required to accommodate shorter read data, such as the 2 x 75bp reads generated using a 150-cycle sequencing kit.</p>
</div>
<div class="tabbed-block">
<p>TheiaCoV_Illumina_SE takes in Illumina single-end reads. Read file names should end with <code>.fastq</code> or <code>.fq</code>, with the optional addition of <code>.gz</code>. Theiagen highly recommends zipping files with <a href="https://www.gnu.org/software/gzip/">gzip</a> before uploading to Terra to minimize data upload time &amp; save on storage costs.</p>
<p>By default, the workflow anticipates <strong>1 x 35 bp</strong> reads  (i.e. the input reads were generated using a 70-cycle sequencing kit). Modifications to the optional parameter for <code>trim_minlen</code> may be required to accommodate longer read data.</p>
</div>
<div class="tabbed-block">
<p>The TheiaCoV_ONT workflow takes in base-called ONT read data. Read file names should end with <code>.fastq</code> or <code>.fq</code>, with the optional addition of <code>.gz</code>. When possible, Theiagen recommends zipping files with <a href="https://www.gnu.org/software/gzip/">gzip</a> before uploading to Terra to minimize data upload time.</p>
<p><strong>The ONT sequencing kit and base-calling approach can produce substantial variability in the amount and quality of read data. Genome assemblies produced by the TheiaCoV_ONT workflow must be quality assessed before reporting results.</strong></p>
</div>
<div class="tabbed-block">
<p>The TheiaCoV_FASTA workflow takes in assembly files in FASTA format.</p>
<div class="admonition warning">
<p class="admonition-title">Note for TheiaCoV_FASTA users analyzing Influenza:</p>
<p>TheiaCoV_FASTA will use the output of VADR to classify and partition Influenza segments from the input assembly. See <a href="#org-specific-tasks"><code>vadr_flu_segments</code></a> task for more details.</p>
</div>
</div>
<div class="tabbed-block">
<p>The TheiaCoV_ClearLabs workflow takes in read data produced by the Clear Dx platform from ClearLabs. However, many users use the TheiaCoV_FASTA workflow instead of this one due to a few known issues when generating assemblies with this pipeline that are not present when using ClearLabs-generated FASTA files.</p>
</div>
<div class="tabbed-block">
<p>The TheiaCoV_FASTA_Batch workflow takes in a set of assembly files in FASTA format.</p>
</div>
</div>
</div>
</div>
<div class="admonition caption">
<div class="tabbed-set tabbed-alternate" data-tabs="2:6"><input checked="checked" id="theiacov_illumina_pe_1" name="__tabbed_2" type="radio"><input id="theiacov_illumina_se_1" name="__tabbed_2" type="radio"><input id="theiacov_ont_1" name="__tabbed_2" type="radio"><input id="theiacov_fasta_1" name="__tabbed_2" type="radio"><input id="theiacov_clearlabs_1" name="__tabbed_2" type="radio"><input id="theiacov_fasta_batch_1" name="__tabbed_2" type="radio"><div class="tabbed-labels"><label for="theiacov_illumina_pe_1">TheiaCoV_Illumina_PE</label><label for="theiacov_illumina_se_1">TheiaCoV_Illumina_SE</label><label for="theiacov_ont_1">TheiaCoV_ONT</label><label for="theiacov_fasta_1">TheiaCoV_FASTA</label><label for="theiacov_clearlabs_1">TheiaCoV_ClearLabs</label><label for="theiacov_fasta_batch_1">TheiaCoV_FASTA_Batch</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>Illumina forward read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Illumina reverse read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-insaflu-220727</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity</td>
<td>70</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>antiviral_aa_subs</strong></td>
<td>String</td>
<td>Additional list of antiviral resistance associated amino acid substitutions of interest to be searched against those called on the sample segments. They take the format of :, e.g. NA:A26V</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_fasta</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h1_ha_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h1n1_m2_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h3_ha_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h3n2_m2_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_n1_na_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_n2_na_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_pa_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_pb1_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_pb2_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_subtype</strong></td>
<td>String</td>
<td>The influenza subtype being analyzed. Used for picking nextclade datasets. Options: "Yamagata", "Victoria", "H1N1", "H3N2", "H5N1". Only use to override the subtype call from IRMA and ABRicate.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_cross_reference</strong></td>
<td>File</td>
<td>An Excel file to cross-reference BLAST findings; probably useful if novel genotypes are not in the default file used by genoflu.py</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/genoflu:1.06</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_min_percent_identity</strong></td>
<td>Float</td>
<td>Percent identity threshold used for calling matches for each genome segment that make up the final GenoFlu genotype</td>
<td>98</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/irma:1.2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_keep_ref_deletions</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if sites missed (i.e. 0 reads for a site in the reference genome) during read gathering should be deleted by ambiguation by inserting N's or deleting the sequence entirely. False sets this IRMA paramater to "DEL" and true sets it to "NNN"</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_min_ambiguous_threshold</strong></td>
<td>Float</td>
<td>Minimum called Single Nucleotide Variant (SNV) frequency for mixed based calls in the output consensus assembly (AKA amended consensus).</td>
<td>0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_min_avg_consensus_allele_quality</strong></td>
<td>Int</td>
<td>Minimum allele coverage depth to call plurality consensus, otherwise calls "N". Setting this value too high can negatively impact final amended consensus.</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_min_read_length</strong></td>
<td>Int</td>
<td>Minimum read length to include reads in read gathering step in IRMA. This value should not be greater than the typical read length.</td>
<td>75</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_custom_input_dataset</strong></td>
<td>File</td>
<td>For H5N1 flu samples only. A custom Nextclade dataset in JSON format. If provided, this dataset will be used to process any H5N1 flu samples. If not provided, a custom dataset will be selected depending on the GenoFLU Genotype.</td>
<td>Defaults are GenoFLU Genotype specific. Please find these default values here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.14.5</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>vadr_outputs_tgz</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>min_depth</strong></td>
<td>Int</td>
<td>The minimum depth to determine if a position was covered.</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>sc2_s_gene_start</strong></td>
<td>Int</td>
<td>start nucleotide position of the SARS-CoV-2 Spike gene</td>
<td>21563</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>sc2_s_gene_stop</strong></td>
<td>Int</td>
<td>End/Last nucleotide position of the SARS-CoV-2 Spike gene</td>
<td>25384</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_bwa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>6</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_bwa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_bwa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_bwa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_consensus_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_consensus_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_consensus_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_consensus_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_trim_primers_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_trim_primers_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_trim_primers_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_trim_primers_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_variant_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_variant_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_variant_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_variant_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>skip_N</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if regions with depth less than minimum depth should not be added to the consensus sequence</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_primtrim_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_primtrim_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_primtrim_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_primtrim_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>auspice_reference_tree_json</strong></td>
<td>File</td>
<td>An Auspice JSON phylogenetic reference tree which serves as a target for phylogenetic placement.</td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>custom_input_dataset</strong></td>
<td>File</td>
<td>For H5N1 flu samples only. A custom Nextclade dataset in JSON format. If provided, this dataset will be used to process any H5N1 flu samples. If not provided, a custom dataset will be selected depending on the GenoFLU Genotype.</td>
<td>Defaults are GenoFLU Genotype specific. Please find these default values here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.16.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>gene_annotations_gff</strong></td>
<td>File</td>
<td>A genome annotation to specify how to translate the nucleotide sequence to proteins (genome_annotation.gff3). specifying this enables codon-informed alignment and protein alignments. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>input_ref</strong></td>
<td>File</td>
<td>A nucleotide sequence which serves as a reference for the pairwise alignment of all input sequences. This is also the sequence which defines the coordinate system of the genome annotation. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>nextclade_pathogen_json</strong></td>
<td>File</td>
<td>General dataset configuration file. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>verbosity</strong></td>
<td>String</td>
<td>other options are: "off" , "error" , "info" , "debug" , and "trace" (highest level of verbosity)</td>
<td>warn</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>auspice_config</strong></td>
<td>File</td>
<td>Auspice config file for customizing visualizations in the Augur_PHB workflow; takes priority over the other customization values available for augur_export. Defaults are set for various organisms &amp; flu segments. A minimal auspice config file is set in cases where organism is not specified and user does not provide an optional input config file.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>clades_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_genoflu_genotype</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_segment</strong></td>
<td>String</td>
<td>Influenza genome segment being analyzed. Options: "HA" or "NA". Automatically determined. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_subtype</strong></td>
<td>String</td>
<td>The influenza subtype being analyzed. Options: "Yamagata", "Victoria", "H1N1", "H3N2", "H5N1". Automatically determined. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>hiv_primer_version</strong></td>
<td>String</td>
<td>The version of HIV primers used. Options are <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156</a> and <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164</a>. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>v1</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>lat_longs_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_date</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_num_unambig</strong></td>
<td>Int</td>
<td>Minimum number of called bases in genome to pass prefilter</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default value is 0</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>narrow_bandwidth</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>pivot_interval</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>proportion_wide</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_genbank</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>analysis_mode</strong></td>
<td>String</td>
<td>Used to switch between usher and pangolearn analysis modes. Only use usher because pangolearn is no longer supported as of Pangolin v4.3 and higher versions.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>expanded_lineage</strong></td>
<td>Boolean</td>
<td>True/False that determines if a lineage should be expanded without aliases (e.g., BA.1 → B.1.1.529.1)</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>max_ambig</strong></td>
<td>Float</td>
<td>The maximum proportion of Ns allowed for pangolin to attempt an assignment</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum query length allowed for pangolin to attempt an assignment</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>pangolin_arguments</strong></td>
<td>String</td>
<td>Optional arguments for pangolin e.g. ''--skip-scorpio''</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_designation_cache</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if the designation cache should be used</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_scorpio</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if scorpio should be skipped.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent_bases_aligned</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>busco_results</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>gambit_predicted_taxon</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_abundance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>n50_value</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_contigs</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>quast_gc_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>quasitools_illumina_pe</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quasitools_illumina_pe</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>quasitools_illumina_pe</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/quasitools:0.7.0--pyh864c0ab_1</td>
<td>Optional</td>
</tr>
<tr>
<td>quasitools_illumina_pe</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the Kraken2 task should be called; for non-TheiaCoV workflows, the <code>kraken_db</code> variable must be provided.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the MIDAS task should be called.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>extract_unclassified</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Additional arguments to use with fastp</td>
<td>--detect_adapter_for_pe -g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_complete_only</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_decontaminate_mem</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_is_accession</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>host_refseq</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database. Must contain human and viral sequences</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task. Increase this when using large (&gt;30GB kraken2 databases such as the "k2_standard" database)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-files-rp/terra/theiaprok-files/midas/midas_db_v1.2.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>The name of the tool to perform basic read processing; options: "trimmomatic" or "fastp"</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>The tool used for quality control (QC) of reads. Options are "fastq_scan" (default) and "fastqc"</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>taxon_id</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td>0</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Additional arguments to pass to trimmomatic. "-phred33" specifies the Phred Q score encoding which is almost always phred33 with modern sequence data.</td>
<td>-phred33</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>A FASTA file containing adapter sequences</td>
<td>/bbmap/resources/adapters.fa</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>consensus_min_freq</strong></td>
<td>Float</td>
<td>The minimum frequency for a variant to be called a SNP in consensus genome</td>
<td>0.6</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>max_genome_length</strong></td>
<td>Int</td>
<td>Maximum genome length able to pass read screening</td>
<td>2673870</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>min_basepairs</strong></td>
<td>Int</td>
<td>Minimum number of base pairs able to pass read screening</td>
<td>17000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>min_coverage</strong></td>
<td>Int</td>
<td>Minimum genome coverage able to pass read screening</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>min_depth</strong></td>
<td>Int</td>
<td>Minimum depth of reads required to call variants and generate a consensus genome. This value is passed to the iVar software.</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>min_genome_length</strong></td>
<td>Int</td>
<td>Minimum genome length to pass read screening</td>
<td>1700</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>min_proportion</strong></td>
<td>Int</td>
<td>Minimum proportion of total reads in each read file to pass read screening</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>min_reads</strong></td>
<td>Int</td>
<td>Minimum number of reads to pass read screening</td>
<td>57</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>nextclade_dataset_name</strong></td>
<td>String</td>
<td>Nextclade organism dataset names. However, if organism input is set correctly, this input will be automatically assigned the corresponding dataset name. See organism defaults for more information</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>nextclade_dataset_tag</strong></td>
<td>String</td>
<td>Nextclade dataset tag. Used for pulling up-to-date reference genomes and associated information specific to nextclade datasets (QC thresholds, organism-specific information like SARS-CoV-2 clade &amp; lineage information, etc.) that is required for running the Nextclade tool.</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>organism</strong></td>
<td>String</td>
<td>The organism that is being analyzed. Options: "sars-cov-2", "MPXV", "WNV", "HIV", "flu", "rsv_a", "rsv_b". However, "flu" is not available for TheiaCoV_Illumina_SE</td>
<td>sars-cov-2</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>pangolin_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>File that contains the phix used</td>
<td>/bbmap/resources/phix174_ill.ref.fa.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>primer_bed</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>qc_check_table</strong></td>
<td>File</td>
<td>TSV value with taxons for rows and QC values for columns; internal cells represent user-determined QC thresholds; if provided, turns on the QC Check task. See below for an example QC Check table.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>reference_gene_locations_bed</strong></td>
<td>File</td>
<td>Use to provide locations of interest where average coverage will be calculated</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>reference_genome</strong></td>
<td>File</td>
<td>An optional reference genome used for consensus assembly and QC</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>reference_gff</strong></td>
<td>File</td>
<td>The general feature format (gff) of the reference genome.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>seq_method</strong></td>
<td>String</td>
<td>The sequencing methodology used to generate the input read data; for TheiaProk workflows, this input will be used in the "seq_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td>ILLUMINA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>skip_screen</strong></td>
<td>Boolean</td>
<td>Set to True to skip the read screening prior to analysis</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>trim_min_length</strong></td>
<td>Int</td>
<td>Specifies minimum length of each read after trimming to be kept</td>
<td>75</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>trim_primers</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if primers should be trimmed.</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>Specifies the minimum average quality of bases in a sliding window to be kept</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>Specifies window size for trimming (the number of bases to average the quality across)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>vadr_max_length</strong></td>
<td>Int</td>
<td>Maximum length of contig allowed to run VADR</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>vadr_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32 (RSV-A and RSV-B) and 8 (all other TheiaCoV organisms)</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>vadr_model_file</strong></td>
<td>File</td>
<td>Path to the a tar + gzipped VADR model file</td>
<td>Defaults are organism-specific. Please find default values for all organisms here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>.</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>vadr_options</strong></td>
<td>String</td>
<td>Additional options to provide to VADR</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>vadr_skip_length</strong></td>
<td>Int</td>
<td>Minimum assembly length (unambiguous) to run VADR</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_pe</td>
<td><strong>variant_min_freq</strong></td>
<td>Float</td>
<td>Minimum frequency for a variant to be reported in ivar outputs</td>
<td>0.6</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/vadr:1.6.4</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum length subsequence to possibly replace Ns for the fasta-trim-terminal-ambigs.pl VADR script</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>Illumina forward read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>min_depth</strong></td>
<td>Int</td>
<td>The minimum depth to determine if a position was covered.</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>sc2_s_gene_start</strong></td>
<td>Int</td>
<td>start nucleotide position of the SARS-CoV-2 Spike gene</td>
<td>21563</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>sc2_s_gene_stop</strong></td>
<td>Int</td>
<td>End/Last nucleotide position of the SARS-CoV-2 Spike gene</td>
<td>25384</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_bwa_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>6</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_bwa_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_bwa_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_bwa_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_consensus_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_consensus_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_consensus_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_consensus_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_trim_primers_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_trim_primers_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_trim_primers_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_trim_primers_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_variant_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_variant_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_variant_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/ivar:1.3.1-titan</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>ivar_variant_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>skip_N</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if regions with depth less than minimum depth should not be added to the consensus sequence</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_primtrim_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_primtrim_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_primtrim_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>ivar_consensus</td>
<td><strong>stats_n_coverage_primtrim_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>auspice_reference_tree_json</strong></td>
<td>File</td>
<td>An Auspice JSON phylogenetic reference tree which serves as a target for phylogenetic placement.</td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>custom_input_dataset</strong></td>
<td>File</td>
<td>For H5N1 flu samples only. A custom Nextclade dataset in JSON format. If provided, this dataset will be used to process any H5N1 flu samples. If not provided, a custom dataset will be selected depending on the GenoFLU Genotype.</td>
<td>Defaults are GenoFLU Genotype specific. Please find these default values here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.16.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>gene_annotations_gff</strong></td>
<td>File</td>
<td>A genome annotation to specify how to translate the nucleotide sequence to proteins (genome_annotation.gff3). specifying this enables codon-informed alignment and protein alignments. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>input_ref</strong></td>
<td>File</td>
<td>A nucleotide sequence which serves as a reference for the pairwise alignment of all input sequences. This is also the sequence which defines the coordinate system of the genome annotation. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>nextclade_pathogen_json</strong></td>
<td>File</td>
<td>General dataset configuration file. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>verbosity</strong></td>
<td>String</td>
<td>other options are: "off" , "error" , "info" , "debug" , and "trace" (highest level of verbosity)</td>
<td>warn</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>auspice_config</strong></td>
<td>File</td>
<td>Auspice config file for customizing visualizations in the Augur_PHB workflow; takes priority over the other customization values available for augur_export. Defaults are set for various organisms &amp; flu segments. A minimal auspice config file is set in cases where organism is not specified and user does not provide an optional input config file.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>clades_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_genoflu_genotype</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_segment</strong></td>
<td>String</td>
<td>Influenza genome segment being analyzed. Options: "HA" or "NA". Automatically determined. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_subtype</strong></td>
<td>String</td>
<td>The influenza subtype being analyzed. Options: "Yamagata", "Victoria", "H1N1", "H3N2", "H5N1". Automatically determined. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>hiv_primer_version</strong></td>
<td>String</td>
<td>The version of HIV primers used. Options are <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156</a> and <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164</a>. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>v1</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>kraken_target_organism_input</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td>Default provided for mpox (Monkeypox virus), WNV (West Nile virus), and HIV (Human immunodeficiency virus 1)</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>lat_longs_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_date</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_num_unambig</strong></td>
<td>Int</td>
<td>Minimum number of called bases in genome to pass prefilter</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default value is 0</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>narrow_bandwidth</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>pivot_interval</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>proportion_wide</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_genbank</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>analysis_mode</strong></td>
<td>String</td>
<td>Used to switch between usher and pangolearn analysis modes. Only use usher because pangolearn is no longer supported as of Pangolin v4.3 and higher versions.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>expanded_lineage</strong></td>
<td>Boolean</td>
<td>True/False that determines if a lineage should be expanded without aliases (e.g., BA.1 → B.1.1.529.1)</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>max_ambig</strong></td>
<td>Float</td>
<td>The maximum proportion of Ns allowed for pangolin to attempt an assignment</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum query length allowed for pangolin to attempt an assignment</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>pangolin_arguments</strong></td>
<td>String</td>
<td>Optional arguments for pangolin e.g. ''--skip-scorpio''</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_designation_cache</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if the designation cache should be used</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_scorpio</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if scorpio should be skipped.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent_bases_aligned</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>busco_results</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>gambit_predicted_taxon</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_abundance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>n50_value</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_clean2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_raw2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_contigs</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>quast_gc_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>bbduk_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the Kraken2 task should be called; for non-TheiaCoV workflows, the <code>kraken_db</code> variable must be provided.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_midas</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the MIDAS task should be called.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>fastp_args</strong></td>
<td>String</td>
<td>Additional arguments to use with fastp</td>
<td>-g -5 20 -3 20</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database. Must contain human and viral sequences</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task. Increase this when using large (&gt;30GB kraken2 databases such as the "k2_standard" database)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>midas_db</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td>gs://theiagen-public-files-rp/terra/theiaprok-files/midas/midas_db_v1.2.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_processing</strong></td>
<td>String</td>
<td>The name of the tool to perform basic read processing; options: "trimmomatic" or "fastp"</td>
<td>trimmomatic</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>read_qc</strong></td>
<td>String</td>
<td>The tool used for quality control (QC) of reads. Options are "fastq_scan" (default) and "fastqc"</td>
<td>fastq_scan</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>trimmomatic_args</strong></td>
<td>String</td>
<td>Additional arguments to pass to trimmomatic. "-phred33" specifies the Phred Q score encoding which is almost always phred33 with modern sequence data.</td>
<td>-phred33</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>adapters</strong></td>
<td>File</td>
<td>A FASTA file containing adapter sequences</td>
<td>/bbmap/resources/adapters.fa</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>consensus_min_freq</strong></td>
<td>Float</td>
<td>The minimum frequency for a variant to be called a SNP in consensus genome</td>
<td>0.6</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>max_genome_length</strong></td>
<td>Int</td>
<td>Maximum genome length able to pass read screening</td>
<td>2673870</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>min_basepairs</strong></td>
<td>Int</td>
<td>Minimum number of base pairs able to pass read screening</td>
<td>17000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>min_coverage</strong></td>
<td>Int</td>
<td>Minimum genome coverage able to pass read screening</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>min_depth</strong></td>
<td>Int</td>
<td>Minimum depth of reads required to call variants and generate a consensus genome. This value is passed to the iVar software.</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>min_genome_length</strong></td>
<td>Int</td>
<td>Minimum genome length to pass read screening</td>
<td>1700</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>min_reads</strong></td>
<td>Int</td>
<td>Minimum number of reads to pass read screening</td>
<td>57</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>nextclade_dataset_name</strong></td>
<td>String</td>
<td>Nextclade organism dataset names. However, if organism input is set correctly, this input will be automatically assigned the corresponding dataset name. See organism defaults for more information</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>nextclade_dataset_tag</strong></td>
<td>String</td>
<td>Nextclade dataset tag. Used for pulling up-to-date reference genomes and associated information specific to nextclade datasets (QC thresholds, organism-specific information like SARS-CoV-2 clade &amp; lineage information, etc.) that is required for running the Nextclade tool.</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>organism</strong></td>
<td>String</td>
<td>The organism that is being analyzed. Options: "sars-cov-2", "MPXV", "WNV", "HIV", "flu", "rsv_a", "rsv_b". However, "flu" is not available for TheiaCoV_Illumina_SE</td>
<td>sars-cov-2</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>pangolin_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>phix</strong></td>
<td>File</td>
<td>File that contains the phix used</td>
<td>/bbmap/resources/phix174_ill.ref.fa.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>primer_bed</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>qc_check_table</strong></td>
<td>File</td>
<td>TSV value with taxons for rows and QC values for columns; internal cells represent user-determined QC thresholds; if provided, turns on the QC Check task. See below for an example QC Check table.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>reference_gene_locations_bed</strong></td>
<td>File</td>
<td>Use to provide locations of interest where average coverage will be calculated</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>reference_genome</strong></td>
<td>File</td>
<td>An optional reference genome used for consensus assembly and QC</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>reference_gff</strong></td>
<td>File</td>
<td>The general feature format (gff) of the reference genome.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>seq_method</strong></td>
<td>String</td>
<td>The sequencing methodology used to generate the input read data; for TheiaProk workflows, this input will be used in the "seq_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td>ILLUMINA</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>skip_mash</strong></td>
<td>Boolean</td>
<td>If true, skips estimation of genome size and coverage using mash in read screening steps. As a result, providing true also prevents screening using these parameters.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>skip_screen</strong></td>
<td>Boolean</td>
<td>Set to True to skip the read screening prior to analysis</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>trim_min_length</strong></td>
<td>Int</td>
<td>Specifies minimum length of each read after trimming to be kept</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>trim_primers</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if primers should be trimmed.</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>trim_quality_min_score</strong></td>
<td>Int</td>
<td>Specifies the minimum average quality of bases in a sliding window to be kept</td>
<td>30</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>trim_window_size</strong></td>
<td>Int</td>
<td>Specifies window size for trimming (the number of bases to average the quality across)</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>vadr_max_length</strong></td>
<td>Int</td>
<td>Maximum length of contig allowed to run VADR</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>vadr_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32 (RSV-A and RSV-B) and 8 (all other TheiaCoV organisms)</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>vadr_model_file</strong></td>
<td>File</td>
<td>Path to the a tar + gzipped VADR model file</td>
<td>Defaults are organism-specific. Please find default values for all organisms here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>.</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>vadr_options</strong></td>
<td>String</td>
<td>Additional options to provide to VADR</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>vadr_skip_length</strong></td>
<td>Int</td>
<td>Minimum assembly length (unambiguous) to run VADR</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_illumina_se</td>
<td><strong>variant_min_freq</strong></td>
<td>Float</td>
<td>Minimum frequency for a variant to be reported in ivar outputs</td>
<td>0.6</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/vadr:1.6.4</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum length subsequence to possibly replace Ns for the fasta-trim-terminal-ambigs.pl VADR script</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiacov_ont</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>ONT read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>clean_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/artic-ncov2019-epi2me</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>medaka_model</strong></td>
<td>String</td>
<td>In order to obtain the best results, the appropriate model must be set to match the sequencer's basecaller model; this string takes the format of {pore}{device}{caller variant}_{caller_version}. See also <a href="https://github.com/nanoporetech/medaka?tab=readme-ov-file#models">https://github.com/nanoporetech/medaka?tab=readme-ov-file#models</a>.</td>
<td>r941_min_high_g360</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-insaflu-220727</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity</td>
<td>70</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>antiviral_aa_subs</strong></td>
<td>String</td>
<td>Additional list of antiviral resistance associated amino acid substitutions of interest to be searched against those called on the sample segments. They take the format of :, e.g. NA:A26V</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_fasta</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h1_ha_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h1n1_m2_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h3_ha_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h3n2_m2_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_n1_na_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_n2_na_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_pa_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_pb1_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_pb2_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_subtype</strong></td>
<td>String</td>
<td>The influenza subtype being analyzed. Used for picking nextclade datasets. Options: "Yamagata", "Victoria", "H1N1", "H3N2", "H5N1". Only use to override the subtype call from IRMA and ABRicate.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_cross_reference</strong></td>
<td>File</td>
<td>An Excel file to cross-reference BLAST findings; probably useful if novel genotypes are not in the default file used by genoflu.py</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/genoflu:1.06</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_min_percent_identity</strong></td>
<td>Float</td>
<td>Percent identity threshold used for calling matches for each genome segment that make up the final GenoFlu genotype</td>
<td>98</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/irma:1.2.0</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_keep_ref_deletions</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if sites missed (i.e. 0 reads for a site in the reference genome) during read gathering should be deleted by ambiguation by inserting N's or deleting the sequence entirely. False sets this IRMA paramater to "DEL" and true sets it to "NNN"</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_min_ambiguous_threshold</strong></td>
<td>Float</td>
<td>Minimum called Single Nucleotide Variant (SNV) frequency for mixed based calls in the output consensus assembly (AKA amended consensus).</td>
<td>0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_min_avg_consensus_allele_quality</strong></td>
<td>Int</td>
<td>Minimum allele coverage depth to call plurality consensus, otherwise calls "N". Setting this value too high can negatively impact final amended consensus.</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_min_read_length</strong></td>
<td>Int</td>
<td>Minimum read length to include reads in read gathering step in IRMA. This value should not be greater than the typical read length.</td>
<td>75</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_custom_input_dataset</strong></td>
<td>File</td>
<td>For H5N1 flu samples only. A custom Nextclade dataset in JSON format. If provided, this dataset will be used to process any H5N1 flu samples. If not provided, a custom dataset will be selected depending on the GenoFLU Genotype.</td>
<td>Defaults are GenoFLU Genotype specific. Please find these default values here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.14.5</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>vadr_outputs_tgz</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>min_depth</strong></td>
<td>Int</td>
<td>The minimum depth to determine if a position was covered.</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>sc2_s_gene_start</strong></td>
<td>Int</td>
<td>start nucleotide position of the SARS-CoV-2 Spike gene</td>
<td>21563</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>sc2_s_gene_stop</strong></td>
<td>Int</td>
<td>End/Last nucleotide position of the SARS-CoV-2 Spike gene</td>
<td>25384</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_clean</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/nanoplot:1.40.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>The maximum length of clean reads, for which reads longer than the length specified will be hidden.</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>nanoplot_raw</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>auspice_reference_tree_json</strong></td>
<td>File</td>
<td>An Auspice JSON phylogenetic reference tree which serves as a target for phylogenetic placement.</td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>custom_input_dataset</strong></td>
<td>File</td>
<td>For H5N1 flu samples only. A custom Nextclade dataset in JSON format. If provided, this dataset will be used to process any H5N1 flu samples. If not provided, a custom dataset will be selected depending on the GenoFLU Genotype.</td>
<td>Defaults are GenoFLU Genotype specific. Please find these default values here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.16.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>gene_annotations_gff</strong></td>
<td>File</td>
<td>A genome annotation to specify how to translate the nucleotide sequence to proteins (genome_annotation.gff3). specifying this enables codon-informed alignment and protein alignments. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>input_ref</strong></td>
<td>File</td>
<td>A nucleotide sequence which serves as a reference for the pairwise alignment of all input sequences. This is also the sequence which defines the coordinate system of the genome annotation. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>nextclade_pathogen_json</strong></td>
<td>File</td>
<td>General dataset configuration file. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>verbosity</strong></td>
<td>String</td>
<td>other options are: "off" , "error" , "info" , "debug" , and "trace" (highest level of verbosity)</td>
<td>warn</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>auspice_config</strong></td>
<td>File</td>
<td>Auspice config file for customizing visualizations in the Augur_PHB workflow; takes priority over the other customization values available for augur_export. Defaults are set for various organisms &amp; flu segments. A minimal auspice config file is set in cases where organism is not specified and user does not provide an optional input config file.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>clades_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_genoflu_genotype</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_segment</strong></td>
<td>String</td>
<td>Influenza genome segment being analyzed. Options: "HA" or "NA". Automatically determined. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_subtype</strong></td>
<td>String</td>
<td>The influenza subtype being analyzed. Options: "Yamagata", "Victoria", "H1N1", "H3N2", "H5N1". Automatically determined. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>hiv_primer_version</strong></td>
<td>String</td>
<td>The version of HIV primers used. Options are <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156</a> and <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164</a>. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>v1</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>lat_longs_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_date</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_num_unambig</strong></td>
<td>Int</td>
<td>Minimum number of called bases in genome to pass prefilter</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default value is 0</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>narrow_bandwidth</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>pivot_interval</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>proportion_wide</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_genbank</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_gff_file</strong></td>
<td>File</td>
<td>Reference GFF file for the organism being analyzed</td>
<td>Default provided for mpox ("gs://theiagen-public-resources-rp/reference_data/viral/mpox/Mpox-MT903345.1.reference.gff3") and HIV (primer versions 1 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/NC_001802.1.gff3"] and 2 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/AY228557.1.gff3"])</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>analysis_mode</strong></td>
<td>String</td>
<td>Used to switch between usher and pangolearn analysis modes. Only use usher because pangolearn is no longer supported as of Pangolin v4.3 and higher versions.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>expanded_lineage</strong></td>
<td>Boolean</td>
<td>True/False that determines if a lineage should be expanded without aliases (e.g., BA.1 → B.1.1.529.1)</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>max_ambig</strong></td>
<td>Float</td>
<td>The maximum proportion of Ns allowed for pangolin to attempt an assignment</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum query length allowed for pangolin to attempt an assignment</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>pangolin_arguments</strong></td>
<td>String</td>
<td>Optional arguments for pangolin e.g. ''--skip-scorpio''</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_designation_cache</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if the designation cache should be used</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_scorpio</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if scorpio should be skipped.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent_bases_aligned</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>busco_results</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>gambit_predicted_taxon</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_abundance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>n50_value</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_clean2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_raw2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_contigs</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>quast_gc_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>quasitools_ont</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>quasitools_ont</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>quasitools_ont</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/quasitools:0.7.0--pyh864c0ab_1</td>
<td>Optional</td>
</tr>
<tr>
<td>quasitools_ont</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>quasitools_ont</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/bactopia/gather_samples:2.0.2</td>
<td>Optional</td>
</tr>
<tr>
<td>raw_check_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/artic-ncov2019:1.3.0-medaka-1.4.3</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>artic_guppyplex_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>call_kraken</strong></td>
<td>Boolean</td>
<td>True/False variable that determines if the Kraken2 task should be called; for non-TheiaCoV workflows, the <code>kraken_db</code> variable must be provided.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>downsampling_coverage</strong></td>
<td>Float</td>
<td>The desired coverage to sub-sample the reads to with RASUSA</td>
<td>150</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-08-28-v4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken2_recalculate_abundances_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_db</strong></td>
<td>File</td>
<td>A kraken2 database to use with the kraken2 optional task. The file must be a .tar.gz kraken2 database. Must contain human and viral sequences</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task. Increase this when using large (&gt;30GB kraken2 databases such as the "k2_standard" database)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>kraken_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/nanoq:0.9.0--hec16e2b_1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_max_read_length</strong></td>
<td>Int</td>
<td>The maximum read length to keep after trimming</td>
<td>100000</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_max_read_qual</strong></td>
<td>Int</td>
<td>The maximum read quality to keep after trimming</td>
<td>40</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_min_read_length</strong></td>
<td>Int</td>
<td>The minimum read length to keep after trimming</td>
<td>500</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>nanoq_min_read_qual</strong></td>
<td>Int</td>
<td>The minimum read quality to keep after trimming</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>ncbi_scrub_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_bases</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_docker</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_fraction_of_reads</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_number_of_reads</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>read_QC_trim</td>
<td><strong>rasusa_seed</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage_primtrim</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage_primtrim</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage_primtrim</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage_primtrim</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>irma_min_consensus_support</strong></td>
<td>Int</td>
<td>Minimum consensus support threshold used by IRMA with ONT data.</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>max_genome_length</strong></td>
<td>Int</td>
<td>Maximum genome length able to pass read screening</td>
<td>2673870</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>max_length</strong></td>
<td>Int</td>
<td>Maximum length for a read based on the SARS-CoV-2 primer scheme</td>
<td>700</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>min_basepairs</strong></td>
<td>Int</td>
<td>Minimum number of base pairs able to pass read screening</td>
<td>17000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>min_coverage</strong></td>
<td>Int</td>
<td>Minimum genome coverage able to pass read screening</td>
<td>10</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>min_genome_length</strong></td>
<td>Int</td>
<td>Minimum genome length to pass read screening</td>
<td>1700</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum length of a read based on the SARS-CoV-2 primer scheme</td>
<td>400</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>min_reads</strong></td>
<td>Int</td>
<td>Minimum number of reads to pass read screening</td>
<td>57</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>nextclade_dataset_name</strong></td>
<td>String</td>
<td>Nextclade organism dataset names. However, if organism input is set correctly, this input will be automatically assigned the corresponding dataset name. See organism defaults for more information</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>nextclade_dataset_tag</strong></td>
<td>String</td>
<td>Nextclade dataset tag. Used for pulling up-to-date reference genomes and associated information specific to nextclade datasets (QC thresholds, organism-specific information like SARS-CoV-2 clade &amp; lineage information, etc.) that is required for running the Nextclade tool.</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>normalise</strong></td>
<td>Int</td>
<td>Used to normalize the amount of reads to the indicated level before variant calling</td>
<td>200</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>organism</strong></td>
<td>String</td>
<td>The organism that is being analyzed. Options: "sars-cov-2", "MPXV", "WNV", "HIV", "flu", "rsv_a", "rsv_b". However, "flu" is not available for TheiaCoV_Illumina_SE</td>
<td>sars-cov-2</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>pangolin_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>primer_bed</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>qc_check_table</strong></td>
<td>File</td>
<td>TSV value with taxons for rows and QC values for columns; internal cells represent user-determined QC thresholds; if provided, turns on the QC Check task. See below for an example QC Check table.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>reference_gene_locations_bed</strong></td>
<td>File</td>
<td>Use to provide locations of interest where average coverage will be calculated</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>reference_genome</strong></td>
<td>File</td>
<td>An optional reference genome used for consensus assembly and QC</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>seq_method</strong></td>
<td>String</td>
<td>The sequencing methodology used to generate the input read data; for TheiaProk workflows, this input will be used in the "seq_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td>OXFORD_NANOPORE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>skip_mash</strong></td>
<td>Boolean</td>
<td>If true, skips estimation of genome size and coverage using mash in read screening steps. As a result, providing true also prevents screening using these parameters.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>skip_screen</strong></td>
<td>Boolean</td>
<td>Set to True to skip the read screening prior to analysis</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>vadr_max_length</strong></td>
<td>Int</td>
<td>Maximum length of contig allowed to run VADR</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>vadr_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32 (RSV-A and RSV-B) and 8 (all other TheiaCoV organisms)</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>vadr_model_file</strong></td>
<td>File</td>
<td>Path to the a tar + gzipped VADR model file</td>
<td>Defaults are organism-specific. Please find default values for all organisms here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>.</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>vadr_options</strong></td>
<td>String</td>
<td>Additional options to provide to VADR</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_ont</td>
<td><strong>vadr_skip_length</strong></td>
<td>Int</td>
<td>Minimum assembly length (unambiguous) to run VADR</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/vadr:1.6.4</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum length subsequence to possibly replace Ns for the fasta-trim-terminal-ambigs.pl VADR script</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiacov_fasta</td>
<td><strong>assembly_fasta</strong></td>
<td>File</td>
<td>Input assembly FASTA file. Must contain either all 8 influenza genome segments or a single segment provided in multi-FASTA format.</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>input_assembly_method</strong></td>
<td>String</td>
<td>Method used to generate the assembly file</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>seq_method</strong></td>
<td>String</td>
<td>The sequencing methodology used to generate the input read data</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>flu_segment</strong></td>
<td>String</td>
<td>Influenza genome segment being analyzed. Options: "HA" or "NA". Only required if input assembly is a singular flu segment.</td>
<td></td>
<td>Optional, Required</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/abricate:1.0.1-insaflu-220727</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_min_percent_coverage</strong></td>
<td>Int</td>
<td>Minimum DNA percent coverage</td>
<td>60</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>abricate_flu_min_percent_identity</strong></td>
<td>Int</td>
<td>Minimum DNA percent identity</td>
<td>70</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>antiviral_aa_subs</strong></td>
<td>String</td>
<td>Additional list of antiviral resistance associated amino acid substitutions of interest to be searched against those called on the sample segments. They take the format of :, e.g. NA:A26V</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_docker</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>assembly_metrics_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h1_ha_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h1n1_m2_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h3_ha_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_h3n2_m2_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_n1_na_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_n2_na_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_pa_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_pb1_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>flu_pb2_ref</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_cross_reference</strong></td>
<td>File</td>
<td>An Excel file to cross-reference BLAST findings; probably useful if novel genotypes are not in the default file used by genoflu.py</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>25</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/genoflu:1.06</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>genoflu_min_percent_identity</strong></td>
<td>Float</td>
<td>Percent identity threshold used for calling matches for each genome segment that make up the final GenoFlu genotype</td>
<td>98</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_cpu</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_disk_size</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_docker_image</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_keep_ref_deletions</strong></td>
<td>Boolean</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_memory</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_min_ambiguous_threshold</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_min_avg_consensus_allele_quality</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_min_consensus_support</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>irma_min_read_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_custom_input_dataset</strong></td>
<td>File</td>
<td>For H5N1 flu samples only. A custom Nextclade dataset in JSON format. If provided, this dataset will be used to process any H5N1 flu samples. If not provided, a custom dataset will be selected depending on the GenoFLU Genotype.</td>
<td>Defaults are GenoFLU Genotype specific. Please find these default values here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.14.5</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>nextclade_output_parser_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>flu_track</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>auspice_reference_tree_json</strong></td>
<td>File</td>
<td>An Auspice JSON phylogenetic reference tree which serves as a target for phylogenetic placement.</td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>custom_input_dataset</strong></td>
<td>File</td>
<td>For H5N1 flu samples only. A custom Nextclade dataset in JSON format. If provided, this dataset will be used to process any H5N1 flu samples. If not provided, a custom dataset will be selected depending on the GenoFLU Genotype.</td>
<td>Defaults are GenoFLU Genotype specific. Please find these default values here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.16.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>gene_annotations_gff</strong></td>
<td>File</td>
<td>A genome annotation to specify how to translate the nucleotide sequence to proteins (genome_annotation.gff3). specifying this enables codon-informed alignment and protein alignments. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>input_ref</strong></td>
<td>File</td>
<td>A nucleotide sequence which serves as a reference for the pairwise alignment of all input sequences. This is also the sequence which defines the coordinate system of the genome annotation. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>nextclade_pathogen_json</strong></td>
<td>File</td>
<td>General dataset configuration file. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>verbosity</strong></td>
<td>String</td>
<td>other options are: "off" , "error" , "info" , "debug" , and "trace" (highest level of verbosity)</td>
<td>warn</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>auspice_config</strong></td>
<td>File</td>
<td>Auspice config file for customizing visualizations in the Augur_PHB workflow; takes priority over the other customization values available for augur_export. Defaults are set for various organisms &amp; flu segments. A minimal auspice config file is set in cases where organism is not specified and user does not provide an optional input config file.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>clades_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_genoflu_genotype</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>gene_locations_bed_file</strong></td>
<td>File</td>
<td>Use to provide locations of interest where average coverage will be calculated</td>
<td>Default provided for SARS-CoV-2 ("gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/sc2_gene_locations.bed") and mpox ("gs://theiagen-public-resources-rp/reference_data/viral/mpox/mpox_gene_locations.bed")</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>hiv_primer_version</strong></td>
<td>String</td>
<td>The version of HIV primers used. Options are <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156</a> and <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164</a>. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>v1</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>kraken_target_organism_input</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td>Default provided for mpox (Monkeypox virus), WNV (West Nile virus), and HIV (Human immunodeficiency virus 1)</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>lat_longs_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_date</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_num_unambig</strong></td>
<td>Int</td>
<td>Minimum number of called bases in genome to pass prefilter</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default value is 0</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>narrow_bandwidth</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>pangolin_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>pivot_interval</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>primer_bed_file</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td>REQUIRED FOR SARS-CoV-2, MPOX, WNV, RSV-A &amp; RSV-B. Provided by default only for HIV primer versions 1 ("gs://theiagen-public-resources-rp/reference_data/viral/hiv/HIV-1_v1.0.primer.hyphen.bed" and 2 ("gs://theiagen-public-resources-rp/reference_data/viral/hiv/HIV-1_v2.0.primer.hyphen400.1.bed")</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>proportion_wide</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_genbank</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_gff_file</strong></td>
<td>File</td>
<td>Reference GFF file for the organism being analyzed</td>
<td>Default provided for mpox ("gs://theiagen-public-resources-rp/reference_data/viral/mpox/Mpox-MT903345.1.reference.gff3") and HIV (primer versions 1 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/NC_001802.1.gff3"] and 2 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/AY228557.1.gff3"])</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>analysis_mode</strong></td>
<td>String</td>
<td>Used to switch between usher and pangolearn analysis modes. Only use usher because pangolearn is no longer supported as of Pangolin v4.3 and higher versions.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>expanded_lineage</strong></td>
<td>Boolean</td>
<td>True/False that determines if a lineage should be expanded without aliases (e.g., BA.1 → B.1.1.529.1)</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>max_ambig</strong></td>
<td>Float</td>
<td>The maximum proportion of Ns allowed for pangolin to attempt an assignment</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum query length allowed for pangolin to attempt an assignment</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>pangolin_arguments</strong></td>
<td>String</td>
<td>Optional arguments for pangolin e.g. ''--skip-scorpio''</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_designation_cache</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if the designation cache should be used</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_scorpio</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if scorpio should be skipped.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent_bases_aligned</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>busco_results</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>gambit_predicted_taxon</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_human_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>meanbaseq_trim</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_abundance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>n50_value</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_clean1</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_clean2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_raw1</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_raw2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_contigs</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>quast_gc_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>flu_subtype</strong></td>
<td>String</td>
<td>The influenza subtype being analyzed. Options: "Yamagata", "Victoria", "H1N1", "H3N2", "H5N1". Automatically determined.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>User-specified expected genome length to be used in genome statistics calculations</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>nextclade_dataset_name</strong></td>
<td>String</td>
<td>Nextclade organism dataset names. However, if organism input is set correctly, this input will be automatically assigned the corresponding dataset name. See organism defaults for more information</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>nextclade_dataset_tag</strong></td>
<td>String</td>
<td>Nextclade dataset tag. Used for pulling up-to-date reference genomes and associated information specific to nextclade datasets (QC thresholds, organism-specific information like SARS-CoV-2 clade &amp; lineage information, etc.) that is required for running the Nextclade tool.</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>organism</strong></td>
<td>String</td>
<td>The organism that is being analyzed. Options: "sars-cov-2", "MPXV", "WNV", "HIV", "flu", "rsv_a", "rsv_b". However, "flu" is not available for TheiaCoV_Illumina_SE</td>
<td>sars-cov-2</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>qc_check_table</strong></td>
<td>File</td>
<td>TSV value with taxons for rows and QC values for columns; internal cells represent user-determined QC thresholds; if provided, turns on the QC Check task. See below for an example QC Check table.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>reference_genome</strong></td>
<td>File</td>
<td>An optional reference genome used for consensus assembly and QC</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>vadr_max_length</strong></td>
<td>Int</td>
<td>Maximum length of contig allowed to run VADR</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>vadr_memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32 (RSV-A and RSV-B) and 8 (all other TheiaCoV organisms)</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>vadr_model_file</strong></td>
<td>File</td>
<td>Path to the a tar + gzipped VADR model file</td>
<td>Defaults are organism-specific. Please find default values for all organisms here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>.</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>vadr_opts</strong></td>
<td>String</td>
<td>Additional options to provide to VADR</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta</td>
<td><strong>vadr_skip_length</strong></td>
<td>Int</td>
<td>Minimum assembly length (unambiguous) to run VADR</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/vadr:1.6.4</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum length subsequence to possibly replace Ns for the fasta-trim-terminal-ambigs.pl VADR script</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>primer_bed</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>read1</strong></td>
<td>File</td>
<td>Clear Dx-produced read file in FASTQ file format (compression optional)</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>samplename</strong></td>
<td>String</td>
<td>The name of the sample being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>medaka_model</strong></td>
<td>String</td>
<td>In order to obtain the best results, the appropriate model must be set to match the sequencer's basecaller model; this string takes the format of {pore}{device}{caller variant}_{caller_version}. See also <a href="https://github.com/nanoporetech/medaka?tab=readme-ov-file#models">https://github.com/nanoporetech/medaka?tab=readme-ov-file#models</a>.</td>
<td>r941_min_high_g360</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>16</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>genome_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>consensus_qc</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>fastq_scan_clean_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>fastq_scan_clean_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>fastq_scan_clean_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/fastq-scan:1.0.1--h4ac6f70_3</td>
<td>Optional</td>
</tr>
<tr>
<td>fastq_scan_clean_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>fastq_scan_clean_reads</td>
<td><strong>read1_name</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>fastq_scan_raw_reads</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>1</td>
<td>Optional</td>
</tr>
<tr>
<td>fastq_scan_raw_reads</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>fastq_scan_raw_reads</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/biocontainers/fastq-scan:1.0.1--h4ac6f70_3</td>
<td>Optional</td>
</tr>
<tr>
<td>fastq_scan_raw_reads</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>fastq_scan_raw_reads</td>
<td><strong>read1_name</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>sc2_s_gene_start</strong></td>
<td>Int</td>
<td>start nucleotide position of the SARS-CoV-2 Spike gene</td>
<td>21563</td>
<td>Optional</td>
</tr>
<tr>
<td>gene_coverage</td>
<td><strong>sc2_s_gene_stop</strong></td>
<td>Int</td>
<td>End/Last nucleotide position of the SARS-CoV-2 Spike gene</td>
<td>25384</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_dehosted</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_dehosted</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task. Increase this when using large (&gt;30GB kraken2 databases such as the "k2_standard" database)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_dehosted</td>
<td><strong>docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_dehosted</td>
<td><strong>kraken2_db</strong></td>
<td>File</td>
<td>The database used to run Kraken2. Must contain viral and human sequences.</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_dehosted</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_dehosted</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_raw</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_raw</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task. Increase this when using large (&gt;30GB kraken2 databases such as the "k2_standard" database)</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_raw</td>
<td><strong>docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_raw</td>
<td><strong>kraken2_db</strong></td>
<td>File</td>
<td>The database used to run Kraken2. Must contain viral and human sequences.</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_raw</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>kraken2_raw</td>
<td><strong>read2</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_scrub_se</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_scrub_se</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_scrub_se</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:2.2.1</td>
<td>Optional</td>
</tr>
<tr>
<td>ncbi_scrub_se</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/python/python:3.8.18-slim</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_output_parser</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>auspice_reference_tree_json</strong></td>
<td>File</td>
<td>An Auspice JSON phylogenetic reference tree which serves as a target for phylogenetic placement.</td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>custom_input_dataset</strong></td>
<td>File</td>
<td>For H5N1 flu samples only. A custom Nextclade dataset in JSON format. If provided, this dataset will be used to process any H5N1 flu samples. If not provided, a custom dataset will be selected depending on the GenoFLU Genotype.</td>
<td>Defaults are GenoFLU Genotype specific. Please find these default values here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.16.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>gene_annotations_gff</strong></td>
<td>File</td>
<td>A genome annotation to specify how to translate the nucleotide sequence to proteins (genome_annotation.gff3). specifying this enables codon-informed alignment and protein alignments. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>input_ref</strong></td>
<td>File</td>
<td>A nucleotide sequence which serves as a reference for the pairwise alignment of all input sequences. This is also the sequence which defines the coordinate system of the genome annotation. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>nextclade_pathogen_json</strong></td>
<td>File</td>
<td>General dataset configuration file. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>verbosity</strong></td>
<td>String</td>
<td>other options are: "off" , "error" , "info" , "debug" , and "trace" (highest level of verbosity)</td>
<td>warn</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>auspice_config</strong></td>
<td>File</td>
<td>Auspice config file for customizing visualizations in the Augur_PHB workflow; takes priority over the other customization values available for augur_export. Defaults are set for various organisms &amp; flu segments. A minimal auspice config file is set in cases where organism is not specified and user does not provide an optional input config file.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>clades_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_genoflu_genotype</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_segment</strong></td>
<td>String</td>
<td>Influenza genome segment being analyzed. Options: "HA" or "NA". Automatically determined. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_subtype</strong></td>
<td>String</td>
<td>The influenza subtype being analyzed. Options: "Yamagata", "Victoria", "H1N1", "H3N2", "H5N1". Automatically determined. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>gene_locations_bed_file</strong></td>
<td>File</td>
<td>Use to provide locations of interest where average coverage will be calculated</td>
<td>Default provided for SARS-CoV-2 ("gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/sc2_gene_locations.bed") and mpox ("gs://theiagen-public-resources-rp/reference_data/viral/mpox/mpox_gene_locations.bed")</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>genome_length_input</strong></td>
<td>Int</td>
<td>Use to specify the expected genome length; provided by default for all supported organisms</td>
<td>Default provided for SARS-CoV-2 (29903), mpox (197200), WNV (11000), flu (13000), RSV-A (16000), RSV-B (16000), HIV (primer versions 1 [9181] and 2 [9840])</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>hiv_primer_version</strong></td>
<td>String</td>
<td>The version of HIV primers used. Options are <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156</a> and <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164</a>. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>v1</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>lat_longs_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_date</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_num_unambig</strong></td>
<td>Int</td>
<td>Minimum number of called bases in genome to pass prefilter</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default value is 0</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>narrow_bandwidth</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>pangolin_docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>pivot_interval</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>primer_bed_file</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td>REQUIRED FOR SARS-CoV-2, MPOX, WNV, RSV-A &amp; RSV-B. Provided by default only for HIV primer versions 1 ("gs://theiagen-public-resources-rp/reference_data/viral/hiv/HIV-1_v1.0.primer.hyphen.bed" and 2 ("gs://theiagen-public-resources-rp/reference_data/viral/hiv/HIV-1_v2.0.primer.hyphen400.1.bed")</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>proportion_wide</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_genbank</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_gff_file</strong></td>
<td>File</td>
<td>Reference GFF file for the organism being analyzed</td>
<td>Default provided for mpox ("gs://theiagen-public-resources-rp/reference_data/viral/mpox/Mpox-MT903345.1.reference.gff3") and HIV (primer versions 1 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/NC_001802.1.gff3"] and 2 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/AY228557.1.gff3"])</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_max_length</strong></td>
<td>Int</td>
<td>Maximum length for the fasta-trim-terminal-ambigs.pl VADR script</td>
<td>Default provided for SARS-CoV-2 (30000), mpox (210000), WNV (11000), flu (0), RSV-A (15500) and RSV-B (15500).</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_mem</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32 (RSV-A, RSV-B, WNV) and 16 (all other TheiaCoV organisms)</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_model</strong></td>
<td>File</td>
<td>Path to the a tar + gzipped VADR model file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-sarscov2-1.3-2.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_options</strong></td>
<td>String</td>
<td>Options for the v-annotate.pl VADR script</td>
<td>--mkey sarscov2 --glsearch -s -r --nomisc --lowsim5seq 6 --lowsim3seq 6 --alt_fail lowscore,insertnn,deletinn --noseqnamemax --out_allfasta</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_skip_length</strong></td>
<td>Int</td>
<td>Minimum assembly length (unambiguous) to run VADR</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>analysis_mode</strong></td>
<td>String</td>
<td>Used to switch between usher and pangolearn analysis modes. Only use usher because pangolearn is no longer supported as of Pangolin v4.3 and higher versions.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>expanded_lineage</strong></td>
<td>Boolean</td>
<td>True/False that determines if a lineage should be expanded without aliases (e.g., BA.1 → B.1.1.529.1)</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>max_ambig</strong></td>
<td>Float</td>
<td>The maximum proportion of Ns allowed for pangolin to attempt an assignment</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum query length allowed for pangolin to attempt an assignment</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>pangolin_arguments</strong></td>
<td>String</td>
<td>Optional arguments for pangolin e.g. ''--skip-scorpio''</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_designation_cache</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if the designation cache should be used</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_scorpio</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if scorpio should be skipped.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>ani_highest_percent_bases_aligned</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>assembly_length</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>busco_results</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>combined_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-03-16</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>est_coverage_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>gambit_predicted_taxon</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_sc2_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>kraken_target_organism_dehosted</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_abundance</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>midas_secondary_genus_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>n50_value</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_clean2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>num_reads_raw2</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>number_contigs</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>quast_gc_percent</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r1_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_q_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_clean</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>r2_mean_readlength_raw</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_mean_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>qc_check_task</td>
<td><strong>sc2_s_gene_percent_coverage</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage_primtrim</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage_primtrim</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage_primtrim</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15</td>
<td>Optional</td>
</tr>
<tr>
<td>stats_n_coverage_primtrim</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>medaka_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/artic-ncov2019:1.3.0-medaka-1.4.3</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>nextclade_dataset_name</strong></td>
<td>String</td>
<td>Nextclade organism dataset names. However, if organism input is set correctly, this input will be automatically assigned the corresponding dataset name. See organism defaults for more information</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>nextclade_dataset_tag</strong></td>
<td>String</td>
<td>Nextclade dataset tag. Used for pulling up-to-date reference genomes and associated information specific to nextclade datasets (QC thresholds, organism-specific information like SARS-CoV-2 clade &amp; lineage information, etc.) that is required for running the Nextclade tool.</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>normalise</strong></td>
<td>Int</td>
<td>Used to normalize the amount of reads to the indicated level before variant calling</td>
<td>20000</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>organism</strong></td>
<td>String</td>
<td>The organism that is being analyzed. Options: "sars-cov-2", "MPXV", "WNV", "HIV", "flu", "rsv_a", "rsv_b". However, "flu" is not available for TheiaCoV_Illumina_SE</td>
<td>sars-cov-2</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>qc_check_table</strong></td>
<td>File</td>
<td>TSV value with taxons for rows and QC values for columns; internal cells represent user-determined QC thresholds; if provided, turns on the QC Check task. See below for an example QC Check table.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>reference_genome</strong></td>
<td>File</td>
<td>An optional reference genome used for consensus assembly and QC</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>seq_method</strong></td>
<td>String</td>
<td>The sequencing methodology used to generate the input read data; for TheiaProk workflows, this input will be used in the "seq_id" column in any taxon-specific tables created in the Export Taxon Tables task</td>
<td>OXFORD_NANOPORE</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_clearlabs</td>
<td><strong>target_organism</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/vadr:1.6.4</td>
<td>Optional</td>
</tr>
<tr>
<td>vadr</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum length subsequence to possibly replace Ns for the fasta-trim-terminal-ambigs.pl VADR script</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Terra Task Name</strong></th>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Terra Status</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>assembly_fastas</strong></td>
<td>Array[File]</td>
<td>The assembly files for your samples in FASTA format</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>bucket_name</strong></td>
<td>String</td>
<td>The GCP bucket for the workspace where the TheiaCoV_FASTA_Batch output files are saved. We recommend using a unique GSURI for the bucket associated with your Terra workspace. The root GSURI is accessible in the Dashboard page of your workspace in the "Cloud Information" section.Do not include the prefix gs:// in the stringExample: ""fc-c526190d-4332-409b-8086-be7e1af9a0b6/theiacov_fasta_batch-2024-04-15-seq-run-1/</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>project_name</strong></td>
<td>String</td>
<td>The name of the Terra project where the data can be found. Example: "my-terra-project"</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>samplenames</strong></td>
<td>Array[String]</td>
<td>The names of the samples being analyzed</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>table_name</strong></td>
<td>String</td>
<td>The name of the Terra table where the data can be found. Example: "sars-cov-2-sample"</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>workspace_name</strong></td>
<td>String</td>
<td>The name of the Terra workspace where the data can be found. Example "my-terra-workspace"</td>
<td></td>
<td>Required</td>
</tr>
<tr>
<td>cat_files_fasta</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>cat_files_fasta</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>cat_files_fasta</td>
<td><strong>docker_image</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/utility:1.1</td>
<td>Optional</td>
</tr>
<tr>
<td>cat_files_fasta</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>auspice_reference_tree_json</strong></td>
<td>File</td>
<td>An Auspice JSON phylogenetic reference tree which serves as a target for phylogenetic placement.</td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>2</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>custom_input_dataset</strong></td>
<td>File</td>
<td>For H5N1 flu samples only. A custom Nextclade dataset in JSON format. If provided, this dataset will be used to process any H5N1 flu samples. If not provided, a custom dataset will be selected depending on the GenoFLU Genotype.</td>
<td>Defaults are GenoFLU Genotype specific. Please find these default values here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a></td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>50</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/nextstrain/nextclade:3.16.0</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>gene_annotations_gff</strong></td>
<td>File</td>
<td>A genome annotation to specify how to translate the nucleotide sequence to proteins (genome_annotation.gff3). specifying this enables codon-informed alignment and protein alignments. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/03-genome-annotation.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>input_ref</strong></td>
<td>File</td>
<td>A nucleotide sequence which serves as a reference for the pairwise alignment of all input sequences. This is also the sequence which defines the coordinate system of the genome annotation. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/02-reference-sequence.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>nextclade_pathogen_json</strong></td>
<td>File</td>
<td>General dataset configuration file. See here for more info: <a href="https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html">https://docs.nextstrain.org/projects/nextclade/en/latest/user/input-files/05-pathogen-config.html</a></td>
<td>Inherited from nextclade dataset</td>
<td>Optional</td>
</tr>
<tr>
<td>nextclade_v3</td>
<td><strong>verbosity</strong></td>
<td>String</td>
<td>other options are: "off" , "error" , "info" , "debug" , and "trace" (highest level of verbosity)</td>
<td>warn</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>auspice_config</strong></td>
<td>File</td>
<td>Auspice config file for customizing visualizations in the Augur_PHB workflow; takes priority over the other customization values available for augur_export. Defaults are set for various organisms &amp; flu segments. A minimal auspice config file is set in cases where organism is not specified and user does not provide an optional input config file.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>clades_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_genoflu_genotype</strong></td>
<td>String</td>
<td>Internal component, do not modify</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_segment</strong></td>
<td>String</td>
<td>Influenza genome segment being analyzed. Options: "HA" or "NA". Automatically determined. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>flu_subtype</strong></td>
<td>String</td>
<td>The influenza subtype being analyzed. Options: "Yamagata", "Victoria", "H1N1", "H3N2", "H5N1". Automatically determined. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>N/A</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>gene_locations_bed_file</strong></td>
<td>File</td>
<td>Use to provide locations of interest where average coverage will be calculated</td>
<td>Default provided for SARS-CoV-2 ("gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/sc2_gene_locations.bed") and mpox ("gs://theiagen-public-resources-rp/reference_data/viral/mpox/mpox_gene_locations.bed")</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>genome_length_input</strong></td>
<td>Int</td>
<td>Use to specify the expected genome length; provided by default for all supported organisms</td>
<td>Default provided for SARS-CoV-2 (29903), mpox (197200), WNV (11000), flu (13000), RSV-A (16000), RSV-B (16000), HIV (primer versions 1 [9181] and 2 [9840])</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>hiv_primer_version</strong></td>
<td>String</td>
<td>The version of HIV primers used. Options are <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L156</a> and <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl#L164</a>. This input is ignored if provided for TheiaCoV_Illumina_SE and TheiaCoV_ClearLabs</td>
<td>v1</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>kraken_target_organism_input</strong></td>
<td>String</td>
<td>The organism whose abundance the user wants to check in their reads. This should be a proper taxonomic name recognized by the Kraken database.</td>
<td>Default provided for mpox (Monkeypox virus), WNV (West Nile virus), and HIV (Human immunodeficiency virus 1)</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>lat_longs_tsv</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_date</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>min_num_unambig</strong></td>
<td>Int</td>
<td>Minimum number of called bases in genome to pass prefilter</td>
<td>Defaults are organism-specific. Please find default values for all organisms (and for Flu - their respective genome segments and subtypes) here: <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl">https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_organism_parameters.wdl</a>. For an organism without set defaults, the default value is 0</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>narrow_bandwidth</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>pivot_interval</strong></td>
<td>Int</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>primer_bed_file</strong></td>
<td>File</td>
<td>The bed file containing the primers used when sequencing was performed</td>
<td>REQUIRED FOR SARS-CoV-2, MPOX, WNV, RSV-A &amp; RSV-B. Provided by default only for HIV primer versions 1 ("gs://theiagen-public-resources-rp/reference_data/viral/hiv/HIV-1_v1.0.primer.hyphen.bed" and 2 ("gs://theiagen-public-resources-rp/reference_data/viral/hiv/HIV-1_v2.0.primer.hyphen400.1.bed")</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>proportion_wide</strong></td>
<td>Float</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_genbank</strong></td>
<td>File</td>
<td>Internal component, do not modify</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_genome</strong></td>
<td>File</td>
<td>An optional reference genome used for consensus assembly and QC</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>reference_gff_file</strong></td>
<td>File</td>
<td>Reference GFF file for the organism being analyzed</td>
<td>Default provided for mpox ("gs://theiagen-public-resources-rp/reference_data/viral/mpox/Mpox-MT903345.1.reference.gff3") and HIV (primer versions 1 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/NC_001802.1.gff3"] and 2 ["gs://theiagen-public-resources-rp/reference_data/viral/hiv/AY228557.1.gff3"])</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_max_length</strong></td>
<td>Int</td>
<td>Maximum length for the fasta-trim-terminal-ambigs.pl VADR script</td>
<td>Default provided for SARS-CoV-2 (30000), mpox (210000), WNV (11000), flu (0), RSV-A (15500) and RSV-B (15500).</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_mem</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>32 (RSV-A, RSV-B, WNV) and 16 (all other TheiaCoV organisms)</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_model</strong></td>
<td>File</td>
<td>Path to the a tar + gzipped VADR model file</td>
<td>gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-sarscov2-1.3-2.tar.gz</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_options</strong></td>
<td>String</td>
<td>Options for the v-annotate.pl VADR script</td>
<td>--mkey sarscov2 --glsearch -s -r --nomisc --lowsim5seq 6 --lowsim3seq 6 --alt_fail lowscore,insertnn,deletinn --noseqnamemax --out_allfasta</td>
<td>Optional</td>
</tr>
<tr>
<td>organism_parameters</td>
<td><strong>vadr_skip_length</strong></td>
<td>Int</td>
<td>Minimum assembly length (unambiguous) to run VADR</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>analysis_mode</strong></td>
<td>String</td>
<td>Used to switch between usher and pangolearn analysis modes. Only use usher because pangolearn is no longer supported as of Pangolin v4.3 and higher versions.</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>expanded_lineage</strong></td>
<td>Boolean</td>
<td>True/False that determines if a lineage should be expanded without aliases (e.g., BA.1 → B.1.1.529.1)</td>
<td>TRUE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>max_ambig</strong></td>
<td>Float</td>
<td>The maximum proportion of Ns allowed for pangolin to attempt an assignment</td>
<td>0.5</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>min_length</strong></td>
<td>Int</td>
<td>Minimum query length allowed for pangolin to attempt an assignment</td>
<td>10000</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>pangolin_arguments</strong></td>
<td>String</td>
<td>Optional arguments for pangolin e.g. ''--skip-scorpio''</td>
<td></td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_designation_cache</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if the designation cache should be used</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>pangolin4</td>
<td><strong>skip_scorpio</strong></td>
<td>Boolean</td>
<td>A True/False option that determines if scorpio should be skipped.</td>
<td>FALSE</td>
<td>Optional</td>
</tr>
<tr>
<td>sm_theiacov_fasta_wrangling</td>
<td><strong>cpu</strong></td>
<td>Int</td>
<td>Number of CPUs to allocate to the task</td>
<td>8</td>
<td>Optional</td>
</tr>
<tr>
<td>sm_theiacov_fasta_wrangling</td>
<td><strong>disk_size</strong></td>
<td>Int</td>
<td>Amount of storage (in GB) to allocate to the task</td>
<td>100</td>
<td>Optional</td>
</tr>
<tr>
<td>sm_theiacov_fasta_wrangling</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/terra-tools:2023-08-28-v4</td>
<td>Optional</td>
</tr>
<tr>
<td>sm_theiacov_fasta_wrangling</td>
<td><strong>memory</strong></td>
<td>Int</td>
<td>Amount of memory/RAM (in GB) to allocate to the task</td>
<td>4</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>nextclade_dataset_name</strong></td>
<td>String</td>
<td>Nextclade organism dataset name. Options: "nextstrain/sars-cov-2/wuhan-hu-1/orfs" However, if organism input is set correctly, this input will be automatically assigned the corresponding dataset name.</td>
<td>sars-cov-2</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>nextclade_dataset_tag</strong></td>
<td>String</td>
<td>Nextclade dataset tag. Used for pulling up-to-date reference genomes and associated information specific to nextclade datasets (QC thresholds, organism-specific information like SARS-CoV-2 clade &amp; lineage information, etc.) that is required for running the Nextclade tool.</td>
<td>2024-06-13--23-42-47Z</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>organism</strong></td>
<td>String</td>
<td>The organism that is being analyzed. Options: "sars-cov-2"</td>
<td>sars-cov-2</td>
<td>Optional</td>
</tr>
<tr>
<td>theiacov_fasta_batch</td>
<td><strong>pangolin_docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>docker</strong></td>
<td>String</td>
<td>The Docker container to use for the task</td>
<td>us-docker.pkg.dev/general-theiagen/theiagen/alpine-plus-bash:3.20.0</td>
<td>Optional</td>
</tr>
<tr>
<td>version_capture</td>
<td><strong>timezone</strong></td>
<td>String</td>
<td>Set the time zone to get an accurate date of analysis (uses UTC by default)</td>
<td></td>
<td>Optional</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<h3 id="org-specific">Organism-Specific Parameters<a class="headerlink" href="#org-specific" title="Permanent link">¶</a></h3>
<p><strong>The <code>organism_parameters</code> sub-workflow is the first step in all TheiaCoV workflows</strong>. This step automatically sets the different parameters needed for each downstream tool to the appropriate value for the user-designated organism (by default, <code>"sars-cov-2"</code> is the default organism).</p>
<div class="admonition dna">
<p>The following tables include the relevant organism-specific parameters; <strong>all of these default values can be overwritten by providing a value for the "Overwrite Variable Name" field</strong>.</p>
<div class="tabbed-set tabbed-alternate" data-tabs="3:10"><input checked="checked" id="sars-cov-2" name="__tabbed_3" type="radio"><input id="mpox" name="__tabbed_3" type="radio"><input id="west-nile-virus" name="__tabbed_3" type="radio"><input id="influenza" name="__tabbed_3" type="radio"><input id="rsv-a" name="__tabbed_3" type="radio"><input id="rsv-b" name="__tabbed_3" type="radio"><input id="hiv" name="__tabbed_3" type="radio"><input id="measles" name="__tabbed_3" type="radio"><input id="mumps" name="__tabbed_3" type="radio"><input id="rubella" name="__tabbed_3" type="radio"><div class="tabbed-labels"><label for="sars-cov-2">SARS-CoV-2</label><label for="mpox">Mpox</label><label for="west-nile-virus">West Nile Virus</label><label for="influenza">Influenza</label><label for="rsv-a">RSV-A</label><label for="rsv-b">RSV-B</label><label for="hiv">HIV</label><label for="measles">Measles</label><label for="mumps">Mumps</label><label for="rubella">Rubella</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table" markdown="block">
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>Organism</strong></th>
<th><strong>Default Value</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>gene_locations_bed_file</td>
<td>sars-cov-2</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/sc2_gene_locations.bed"</code></td>
</tr>
<tr>
<td>genome_length_input</td>
<td>sars-cov-2</td>
<td><code>29903</code></td>
</tr>
<tr>
<td>kraken_target_organism_input</td>
<td>sars-cov-2</td>
<td><code>"Severe acute respiratory syndrome coronavirus 2"</code></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>sars-cov-2</td>
<td><code>"nextstrain/sars-cov-2/wuhan-hu-1/orfs"</code></td>
</tr>
<tr>
<td>pangolin_docker_image</td>
<td>sars-cov-2</td>
<td><code>"us-docker.pkg.dev/general-theiagen/staphb/pangolin:4.3.1-pdata-1.34 "</code></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>sars-cov-2</td>
<td><code>"2025-08-02--08-55-17Z"</code></td>
</tr>
<tr>
<td>reference_genome</td>
<td>sars-cov-2</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/sars-cov-2/MN908947.fasta"</code></td>
</tr>
<tr>
<td>vadr_max_length</td>
<td>sars-cov-2</td>
<td><code>30000</code></td>
</tr>
<tr>
<td>vadr_skip_length</td>
<td>sars-cov-2</td>
<td><code>10000</code></td>
</tr>
<tr>
<td>vadr_mem</td>
<td>sars-cov-2</td>
<td><code>8</code></td>
</tr>
<tr>
<td>vadr_options</td>
<td>sars-cov-2</td>
<td><code>"--mkey sarscov2 --glsearch -s -r --nomisc --lowsim5seq 6 --lowsim3seq 6 --alt_fail lowscore,insertnn,deletinn --noseqnamemax --out_allfasta"</code></td>
</tr>
<tr>
<td>vadr_model_file</td>
<td>sars-cov-2</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-sarscov2-1.3-2.tar.gz"</code></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table" markdown="block">
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>Organism</strong></th>
<th><strong>Default Value</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>gene_locations_bed_file</td>
<td>MPXV</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/mpox/mpox_gene_locations.bed"</code></td>
</tr>
<tr>
<td>genome_length_input</td>
<td>MPXV</td>
<td><code>197200</code></td>
</tr>
<tr>
<td>kraken_target_organism_input</td>
<td>MPXV</td>
<td><code>"Monkeypox virus"</code></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>MPXV</td>
<td><code>"nextstrain/mpox/lineage-b.1"</code></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>MPXV</td>
<td><code>"2025-04-25--12-24-24Z"</code></td>
</tr>
<tr>
<td>primer_bed_file</td>
<td>MPXV</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/mpox/MPXV.primer.bed"</code></td>
</tr>
<tr>
<td>reference_genome</td>
<td>MPXV</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/mpox/MPXV.MT903345.reference.fasta"</code></td>
</tr>
<tr>
<td>reference_gff_file</td>
<td>MPXV</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/mpox/Mpox-MT903345.1.reference.gff3"</code></td>
</tr>
<tr>
<td>vadr_max_length</td>
<td>MPXV</td>
<td><code>210000</code></td>
</tr>
<tr>
<td>vadr_skip_length</td>
<td>MPXV</td>
<td><code>65480</code></td>
</tr>
<tr>
<td>vadr_mem</td>
<td>MPXV</td>
<td><code>8</code></td>
</tr>
<tr>
<td>vadr_options</td>
<td>MPXV</td>
<td><code>"--mkey mpxv --glsearch --minimap2 -s -r --nomisc --r_lowsimok --r_lowsimxd 100 --r_lowsimxl 2000 --alt_pass discontn,dupregin --s_overhang 150 --out_allfasta"</code></td>
</tr>
<tr>
<td>vadr_model_file</td>
<td>MPXV</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-mpxv-1.4.2-1.tar.gz"</code></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table" markdown="block">
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>Organism</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Notes</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>genome_length_input</td>
<td>WNV</td>
<td><code>11000</code></td>
<td></td>
</tr>
<tr>
<td>kraken_target_organism_input</td>
<td>WNV</td>
<td><code>"West Nile virus</code>"</td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>WNV</td>
<td><code>"NA"</code></td>
<td>TheiaCoV's Nextclade currently does not support WNV</td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>WNV</td>
<td><code>"NA"</code></td>
<td>TheiaCoV's Nextclade currently does not support WNV</td>
</tr>
<tr>
<td>primer_bed_file</td>
<td>WNV</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/wnv/al/wnv/WNV-L1_primer.bed"</code></td>
<td></td>
</tr>
<tr>
<td>reference_genome</td>
<td>WNV</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/wnv/NC_009942.1_wnv_L1.fasta"</code></td>
<td></td>
</tr>
<tr>
<td>vadr_max_length</td>
<td>WNV</td>
<td><code>11000</code></td>
<td></td>
</tr>
<tr>
<td>vadr_skip_length</td>
<td>WNV</td>
<td><code>3000</code></td>
<td></td>
</tr>
<tr>
<td>vadr_mem</td>
<td>WNV</td>
<td><code>8</code></td>
<td></td>
</tr>
<tr>
<td>vadr_options</td>
<td>WNV</td>
<td><code>"--mkey flavi --nomisc --noprotid --out_allfasta"</code></td>
<td></td>
</tr>
<tr>
<td>vadr_model_file</td>
<td>WNV</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-flavi-1.2-1.tar.gz"</code></td>
<td></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table" markdown="block">
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>Organism</strong></th>
<th><strong>Flu Segment</strong></th>
<th><strong>Flu Subtype</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Notes</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>flu_segment</td>
<td>flu</td>
<td>all</td>
<td>all</td>
<td>N/A</td>
<td>TheiaCoV will attempt to automatically assign a flu segment</td>
</tr>
<tr>
<td>flu_subtype</td>
<td>flu</td>
<td>all</td>
<td>all</td>
<td>N/A</td>
<td>TheiaCoV will attempt to automatically assign a flu subtype</td>
</tr>
<tr>
<td>genome_length_input</td>
<td>flu</td>
<td>all</td>
<td>all</td>
<td><code>13500</code></td>
<td></td>
</tr>
<tr>
<td>vadr_max_length</td>
<td>flu</td>
<td>all</td>
<td>all</td>
<td><code>13500</code></td>
<td></td>
</tr>
<tr>
<td>vadr_skip_length</td>
<td>flu</td>
<td>all</td>
<td>all</td>
<td><code>500</code></td>
<td></td>
</tr>
<tr>
<td>vadr_mem</td>
<td>flu</td>
<td>all</td>
<td>all</td>
<td><code>8</code></td>
<td></td>
</tr>
<tr>
<td>vadr_options</td>
<td>flu</td>
<td>all</td>
<td>all</td>
<td><code>"--mkey flu --atgonly --xnocomp --nomisc --alt_fail extrant5,extrant3"</code></td>
<td></td>
</tr>
<tr>
<td>vadr_model_file</td>
<td>flu</td>
<td>all</td>
<td>all</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-flu-1.6.3-2.tar.gz"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>flu</td>
<td>ha</td>
<td>h1n1</td>
<td><code>"nextstrain/flu/h1n1pdm/ha/MW626062"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>flu</td>
<td>ha</td>
<td>h1n1</td>
<td><code>"2025-08-07--09-22-32Z"</code></td>
<td></td>
</tr>
<tr>
<td>reference_genome</td>
<td>flu</td>
<td>ha</td>
<td>h1n1</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_h1n1pdm_ha.fasta"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>flu</td>
<td>ha</td>
<td>h3n2</td>
<td><code>"nextstrain/flu/h3n2/ha/EPI1857216"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>flu</td>
<td>ha</td>
<td>h3n2</td>
<td><code>"2025-08-07--09-22-32Z"</code></td>
<td></td>
</tr>
<tr>
<td>reference_genome</td>
<td>flu</td>
<td>ha</td>
<td>h3n2</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_h3n2_ha.fasta"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>flu</td>
<td>ha</td>
<td>victoria</td>
<td><code>"nextstrain/flu/vic/ha/KX058884"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>flu</td>
<td>ha</td>
<td>victoria</td>
<td><code>"2025-08-07--09-22-32Z"</code></td>
<td></td>
</tr>
<tr>
<td>reference_genome</td>
<td>flu</td>
<td>ha</td>
<td>victoria</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_vic_ha.fasta"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>flu</td>
<td>ha</td>
<td>yamagata</td>
<td><code>"nextstrain/flu/yam/ha/JN993010"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>flu</td>
<td>ha</td>
<td>yamagata</td>
<td><code>"2024-01-30--16-34-55Z"</code></td>
<td></td>
</tr>
<tr>
<td>reference_genome</td>
<td>flu</td>
<td>ha</td>
<td>yamagata</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_yam_ha.fasta"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>flu</td>
<td>ha</td>
<td>h5n1</td>
<td><code>"community/moncla-lab/iav-h5/ha/all-clades"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>flu</td>
<td>ha</td>
<td>h5n1</td>
<td><code>"2025-08-07--09-22-32Z"</code></td>
<td></td>
</tr>
<tr>
<td>reference_genome</td>
<td>flu</td>
<td>ha</td>
<td>h5n1</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_h5n1_ha.fasta"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>flu</td>
<td>na</td>
<td>h1n1</td>
<td><code>"nextstrain/flu/h1n1pdm/na/MW626056"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>flu</td>
<td>na</td>
<td>h1n1</td>
<td><code>"2025-08-07--09-22-32Z"</code></td>
<td></td>
</tr>
<tr>
<td>reference_genome</td>
<td>flu</td>
<td>na</td>
<td>h1n1</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_h1n1pdm_na.fasta"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>flu</td>
<td>na</td>
<td>h3n2</td>
<td><code>"nextstrain/flu/h3n2/na/EPI1857215"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>flu</td>
<td>na</td>
<td>h3n2</td>
<td><code>"2025-08-07--09-22-32Z"</code></td>
<td></td>
</tr>
<tr>
<td>reference_genome</td>
<td>flu</td>
<td>na</td>
<td>h3n2</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_h3n2_na.fasta"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>flu</td>
<td>na</td>
<td>victoria</td>
<td><code>"nextstrain/flu/vic/na/CY073894"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>flu</td>
<td>na</td>
<td>victoria</td>
<td><code>"2025-08-07--09-22-32Z"</code></td>
<td></td>
</tr>
<tr>
<td>reference_genome</td>
<td>flu</td>
<td>na</td>
<td>victoria</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_vic_na.fasta"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>flu</td>
<td>na</td>
<td>yamagata</td>
<td><code>"NA"</code></td>
<td></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>flu</td>
<td>na</td>
<td>yamagata</td>
<td><code>"NA"</code></td>
<td></td>
</tr>
<tr>
<td>reference_genome</td>
<td>flu</td>
<td>na</td>
<td>yamagata</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/reference_yam_na.fasta"</code></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div class="admonition tip">
<p class="admonition-title">H5N1 Additional Defaults</p>
<p>If the sample is designated as H5N1 by either ABRicate or IRMA, an H5N1-specific Nextclade task will run with the following datasets depending on the GenoFLU genotype. </p>
<p>Alternatively, if a <code>nextclade_custom_input_dataset</code> variable is provided (available under the <code>flu_track</code> task name), the workflow will run that custom dataset on all H5N1 samples, regardless of the GenoFLU genotype.</p>
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>GenoFLU Genotype</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Notes</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>nextclade_custom_input_dataset</td>
<td>B3.13</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/nextclade_avian-flu_h5n1-cattle-outbreak_h5n1-b3.13_2025-06-24.json"</code></td>
<td>Extracted from <a href="https://nextstrain.org/avian-flu/h5n1-cattle-outbreak/genome">nextclade/avian-flu/h5n1-cattle-outbreak</a> on 2025-06-24</td>
</tr>
<tr>
<td>nextclade_custom_input_dataset</td>
<td>D1.1</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/flu/nextclade_avian-flu_h5n1-d1.1_2025-06-24.json"</code></td>
<td>Extracted from <a href="https://nextstrain.org/avian-flu/h5n1-d1.1/genome">nextclade/avian-flu/h5n1-d1.1</a> on 2025-06-24</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table" markdown="block">
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>Organism</strong></th>
<th><strong>Default Value</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>genome_length_input</td>
<td>rsv_a</td>
<td><code>16000</code></td>
</tr>
<tr>
<td>kraken_target_organism</td>
<td>rsv_a</td>
<td><code>"Human respiratory syncytial virus A"</code></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>rsv_a</td>
<td><code>nextstrain/rsv/a/EPI_ISL_412866</code></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>rsv_a</td>
<td><code>"2024-11-27--02-51-00Z"</code></td>
</tr>
<tr>
<td>reference_genome</td>
<td>rsv_a</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/rsv/reference_rsv_a.EPI_ISL_412866.fasta"</code></td>
</tr>
<tr>
<td>vadr_max_length</td>
<td>rsv_a</td>
<td><code>15500</code></td>
</tr>
<tr>
<td>vadr_skip_length</td>
<td>rsv_a</td>
<td><code>5000</code></td>
</tr>
<tr>
<td>vadr_mem</td>
<td>rsv_a</td>
<td><code>32</code></td>
</tr>
<tr>
<td>vadr_options</td>
<td>rsv_a</td>
<td><code>"--mkey rsv --xnocomp -r"</code></td>
</tr>
<tr>
<td>vadr_model_file</td>
<td>rsv_a</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-rsv-1.5-2.tar.gz"</code></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table" markdown="block">
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>Organism</strong></th>
<th><strong>Default Value</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>genome_length_input</td>
<td>rsv_b</td>
<td><code>16000</code></td>
</tr>
<tr>
<td>kraken_target_organism</td>
<td>rsv_b</td>
<td><code>"human respiratory syncytial virus"</code></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>rsv_b</td>
<td><code>nextstrain/rsv/b/EPI_ISL_1653999</code></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>rsv_b</td>
<td><code>"2025-03-04--17-31-25Z"</code></td>
</tr>
<tr>
<td>reference_genome</td>
<td>rsv_b</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/rsv/reference_rsv_b.EPI_ISL_1653999.fasta"</code></td>
</tr>
<tr>
<td>vadr_max_length</td>
<td>rsv_b</td>
<td><code>15500</code></td>
</tr>
<tr>
<td>vadr_mem</td>
<td>rsv_b</td>
<td><code>32</code></td>
</tr>
<tr>
<td>vadr_options</td>
<td>rsv_b</td>
<td><code>"--mkey rsv --xnocomp -r"</code></td>
</tr>
<tr>
<td>vadr_model_file</td>
<td>rsv_b</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-rsv-1.5-2.tar.gz"</code></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table" markdown="block">
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>Organism</strong></th>
<th><strong>Default Value</strong></th>
<th><strong>Notes</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>kraken_target_organism_input</td>
<td>HIV</td>
<td><code>"Human immunodeficiency virus 1"</code></td>
<td></td>
</tr>
<tr>
<td>genome_length_input</td>
<td>HIV-v1</td>
<td><code>9181</code></td>
<td>This version of HIV originates from Oregon</td>
</tr>
<tr>
<td>primer_bed_file</td>
<td>HIV-v1</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/hiv/HIV-1_v1.0.primer.hyphen.bed"</code></td>
<td>This version of HIV originates from Oregon</td>
</tr>
<tr>
<td>reference_genome</td>
<td>HIV-v1</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/hiv/NC_001802.1.fasta"</code></td>
<td>This version of HIV originates from Oregon</td>
</tr>
<tr>
<td>reference_gff_file</td>
<td>HIV-v1</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/hiv/NC_001802.1.gff3"</code></td>
<td>This version of HIV originates from Oregon</td>
</tr>
<tr>
<td>genome_length_input</td>
<td>HIV-v2</td>
<td><code>9840</code></td>
<td>This version of HIV originates from Southern Africa</td>
</tr>
<tr>
<td>primer_bed_file</td>
<td>HIV-v2</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/hiv/HIV-1_v2.0.primer.hyphen400.1.bed"</code></td>
<td>This version of HIV originates from Southern Africa</td>
</tr>
<tr>
<td>reference_genome</td>
<td>HIV-v2</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/hiv/AY228557.1.headerchanged.fasta"</code></td>
<td>This version of HIV originates from Southern Africa</td>
</tr>
<tr>
<td>reference_gff_file</td>
<td>HIV-v2</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/hiv/AY228557.1.gff3"</code></td>
<td>This version of HIV originates from Southern Africa</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table" markdown="block">
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>Organism</strong></th>
<th><strong>Default Value</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>kraken_target_organism_input</td>
<td>measles</td>
<td><code>"Measles morbillivirus"</code></td>
</tr>
<tr>
<td>genome_length_input</td>
<td>measles</td>
<td><code>16000</code></td>
</tr>
<tr>
<td>nextclade_dataset_name_input</td>
<td>measles</td>
<td><code>"nextstrain/measles/genome/WHO-2012"</code></td>
</tr>
<tr>
<td>nextclade_dataset_tag_input</td>
<td>measles</td>
<td><code>"2025-08-11--19-06-01Z"</code></td>
</tr>
<tr>
<td>reference_genome</td>
<td>measles</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/measles/NC_001498.1_measles_reference.fasta"</code></td>
</tr>
<tr>
<td>vadr_max_length</td>
<td>measles</td>
<td><code>18000</code></td>
</tr>
<tr>
<td>vadr_skip_length</td>
<td>measles</td>
<td><code>0</code></td>
</tr>
<tr>
<td>vadr_mem</td>
<td>measles</td>
<td><code>24</code></td>
</tr>
<tr>
<td>vadr_options</td>
<td>measles</td>
<td><code>"--mkey mev -r --indefclass 0.01"</code></td>
</tr>
<tr>
<td>vadr_model_file</td>
<td>measles</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-mev-1.02.tar.gz"</code></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table" markdown="block">
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>Organism</strong></th>
<th><strong>Default Value</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>reference_genome</td>
<td>mumps</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/mumps/NC_002200.1_mumps_reference.fasta"</code></td>
</tr>
<tr>
<td>genome_length_input</td>
<td>mumps</td>
<td><code>15300</code></td>
</tr>
<tr>
<td>vadr_options</td>
<td>mumps</td>
<td><code>"--mkey muv -r --indefclass 0.025"</code></td>
</tr>
<tr>
<td>vadr_max_length</td>
<td>mumps</td>
<td><code>18000</code></td>
</tr>
<tr>
<td>vadr_skip_length</td>
<td>mumps</td>
<td><code>0</code></td>
</tr>
<tr>
<td>vadr_mem</td>
<td>mumps</td>
<td><code>16</code></td>
</tr>
<tr>
<td>vadr_model_file</td>
<td>mumps</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-muv-1.01.tar.gz"</code></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table" markdown="block">
<table>
<thead>
<tr>
<th><strong>Overwrite Variable Name</strong></th>
<th><strong>Organism</strong></th>
<th><strong>Default Value</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>reference_genome</td>
<td>rubella</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/viral/rubella/NC_001545.2_rubella_reference.fasta"</code></td>
</tr>
<tr>
<td>genome_length_input</td>
<td>rubella</td>
<td><code>9800</code></td>
</tr>
<tr>
<td>vadr_options</td>
<td>rubella</td>
<td><code>"--mkey ruv -r"</code></td>
</tr>
<tr>
<td>vadr_max_length</td>
<td>rubella</td>
<td><code>10000</code></td>
</tr>
<tr>
<td>vadr_skip_length</td>
<td>rubella</td>
<td><code>0</code></td>
</tr>
<tr>
<td>vadr_mem</td>
<td>rubella</td>
<td><code>16</code></td>
</tr>
<tr>
<td>vadr_model_file</td>
<td>rubella</td>
<td><code>"gs://theiagen-public-resources-rp/reference_data/databases/vadr_models/vadr-models-ruv-1.01.tar.gz"</code></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<h3 id="core-tasks">Core Tasks<a class="headerlink" href="#core-tasks" title="Permanent link">¶</a></h3>
<div class="admonition dna">
<p>These tasks are performed for all organisms. They include tasks that are performed regardless of and specific for the input data type. They perform read trimming and assembly appropriate to the input data type.</p>
</div>
<details class="task">
<summary><code>versioning</code>: Version Capture</summary>
<p>The <code>versioning</code> task captures the workflow version from the GitHub (code repository) version.</p>
<div class="admonition techdetails">
<p class="admonition-title">Version Capture Technical details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/task_versioning.wdl">task_versioning.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h4 id="assembly-tasks">Assembly Tasks<a class="headerlink" href="#assembly-tasks" title="Permanent link">¶</a></h4>
<div class="admonition caption">
<div class="tabbed-set tabbed-alternate" data-tabs="4:5"><input checked="checked" id="theiacov_illumina_pe_2" name="__tabbed_4" type="radio"><input id="theiacov_illumina_se_2" name="__tabbed_4" type="radio"><input id="theiacov_ont_2" name="__tabbed_4" type="radio"><input id="theiacov_clearlabs_2" name="__tabbed_4" type="radio"><input id="theiacov_fasta_2" name="__tabbed_4" type="radio"><div class="tabbed-labels"><label for="theiacov_illumina_pe_2">TheiaCoV_Illumina_PE</label><label for="theiacov_illumina_se_2">TheiaCoV_Illumina_SE</label><label for="theiacov_ont_2">TheiaCoV_ONT</label><label for="theiacov_clearlabs_2">TheiaCoV_ClearLabs</label><label for="theiacov_fasta_2">TheiaCoV_FASTA</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<details class="task">
<summary><code>read_QC_trim</code>: Read Quality Trimming, Adapter Removal, Quantification, and Identification</summary>
<p><code>read_QC_trim</code> is a sub-workflow that removes low-quality reads, low-quality regions of reads, and sequencing adapters to improve data quality. It uses a number of tasks, described below. The differences between the PE and SE versions of the <code>read_QC_trim</code> sub-workflow lie in the default parameters, the use of two or one input read file(s), and the different output files.</p>
<details class="task">
<summary><code>HRRT</code>: Human Host Sequence Removal</summary>
<p>All reads of human origin <strong>are removed</strong>, including their mates, by using NCBI's <a href="https://github.com/ncbi/sra-human-scrubber"><strong>human read removal tool (HRRT)</strong></a>. </p>
<p>HRRT is based on the <a href="https://doi.org/10.1186/s13059-021-02490-0">SRA Taxonomy Analysis Tool</a> and employs a k-mer database constructed of k-mers from Eukaryota derived from all human RefSeq records with any k-mers found in non-Eukaryota RefSeq records subtracted from the database.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI-Scrub Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ncbi_scrub.wdl">task_ncbi_scrub.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/sra-human-scrubber">HRRT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2023/02/02/scrubbing-human-sequences-sra-submissions/">HRRT on NCBI</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_processing</code> is set to <code>"trimmomatic"</code>. To use <code>fastp</code> instead, set <code>read_processing</code> to <code>"fastp"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>Trimmomatic</code>: Read Trimming (default)</summary>
<p>Read proccessing is available via <code>Trimmomatic</code> by default.</p>
<p>Trimmomatic trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>Trimmomatic</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_trimmomatic.wdl">task_trimmomatic.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/usadellab/Trimmomatic">Trimmomatic on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic Website</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590/">Trimmomatic: a flexible trimmer for Illumina sequence data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>fastp</code>: Read Trimming (alternative)</summary>
<p>To activate this task, set <code>read_processing</code> to <code>"fastp"</code>.</p>
<p><code>fastp</code> trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<p><code>fastp</code> also has additional default parameters and features that are not a part of <code>trimmomatic</code>'s default configuration.</p>
<details class="toggle">
<summary><code>fastp</code> default read-trimming parameters</summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-g</td>
<td>enables polyG tail trimming</td>
</tr>
<tr>
<td>-5 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>-3 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>--detect_adapter_for_pe</td>
<td>enables adapter-trimming <strong>only for paired-end reads</strong></td>
</tr>
</tbody>
</table>
<p>Additional arguments can be passed using the <code>fastp_args</code> optional parameter.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Trimmomatic and fastp Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_fastp.wdl">task_fastp.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/17/i884/5093234?login=false">fastp: an ultra-fast all-in-one FASTQ preprocessor</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>BBDuk</code>: Adapter Trimming and PhiX Removal</summary>
<p>Adapters are manufactured oligonucleotide sequences attached to DNA fragments during the library preparation process. In Illumina sequencing, these adapter sequences are required for attaching reads to flow cells. You can read more about <a href="https://emea.support.illumina.com/bulletins/2020/06/illumina-adapter-portfolio.html">Illumina adapters here</a>. For genome analysis, it's important to remove these sequences since they're not actually from your sample. If you don't remove them, the downstream analysis may be affected.</p>
<p>The <code>bbduk</code> task removes adapters from sequence reads. To do this:</p>
<ul>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/repair-guide/">Repair</a> from the <a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/">BBTools</a> package reorders reads in paired fastq files to ensure the forward and reverse reads of a pair are in the same position in the two fastq files (it <em>re-pairs</em>).</li>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk</a>  (<em>"Bestus Bioinformaticus" Decontamination Using Kmers</em>) is then used to trim the adapters and filter out all reads that have a 31-mer match to <a href="https://emea.illumina.com/products/by-type/sequencing-kits/cluster-gen-sequencing-reagents/phix-control-v3.html">PhiX</a>, which is commonly added to Illumina sequencing runs to monitor and/or improve overall run quality.</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">BBDuk Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_bbduk.wdl">task_bbduk.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://sourceforge.net/projects/bbmap/">BBMap on SourceForge</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk Guide (archived)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_qc</code> is set to <code>"fastq_scan"</code>. To use <code>fastqc</code> instead, set <code>read_qc</code> to <code>"fastqc"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>fastq-scan</code>: Read Quantification (default)</summary>
<p>Read quantification is available via <code>fastq-scan</code> by default.</p>
<p><code>fastq-scan</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>fastq-scan</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastq_scan.wdl">task_fastq_scan.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/fastq-scan/blob/master/README.md">fastq-scan on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>FastQC</code>: Read Quantification (alternative)</summary>
<p>To activate this task, set <code>read_qc</code> to <code>"fastqc"</code>.</p>
<p><code>FastQC</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>This tool also provides a graphical visualization of the read quality.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>FastQC</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastqc.wdl">task_fastqc.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/s-andrews/FastQC">FastQC on Github</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC Website</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>Kraken2</code>: Read Identification</summary>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<p>Kraken2 is run on both the raw and clean reads.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">Kraken2 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_pe.wdl">wf_read_QC_trim_pe.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_se.wdl">wf_read_QC_trim_se.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<p><strong><em>If non-influenza</em></strong></p>
<details class="task">
<summary><code>ivar_consensus</code>: Alignment, Consensus, Variant Detection, and Assembly Statistics</summary>
<p><code>iVar Consensus</code> is a sub-workflow within TheiaCoV that performs reference-based consensus assembly using the <a href="https://andersen-lab.github.io/ivar/html/index.html">iVar</a> tool by Nathan Grubaugh from the Andersen lab.</p>
<details class="task">
<summary><code>bwa</code>: Read Alignment to the Reference</summary>
<p>BWA (Burrow-Wheeler Aligner) is used to align the cleaned read files to a reference genome, either  determined by the user or provided by the <a href="./#org-specific"><em>organism-specific parameters</em> section</a> (see above). The resulting BAM file is used for primer trimming, variant calling, and consensus generation in downstream tasks.</p>
<div class="admonition techdetails">
<p class="admonition-title">BWA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_bwa.wdl">task_bwa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/bwa">BWA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bio-bwa.sourceforge.net/">BWA Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and accurate short read alignment with Burrows-Wheeler transform</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ivar_trim</code>: Primer Trimming (optional)</summary>
<p>To deactivate this task, set <code>trim_primers</code> to <code>false</code>.</p>
<p>Using the user-provided (or, more rarely, a <a href="./#org-specific"><em>organism-specific parameters</em>-determined</a>) <code>primer_bed</code> file, iVar soft-clips primer sequences from an aligned and sorted BAM file and then trims the reads based on a quality threshold of 20 using a sliding window approach. If the resulting read is greater than 30 bp, the read is written to a a new BAM file consisting of only trimmed reads (or reads that did not have a primer identified).</p>
<div class="admonition techdetails">
<p class="admonition-title">iVar Trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ivar_primer_trim.wdl">task_ivar_primer_trim.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://andersen-lab.github.io/ivar/html/">iVar on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://andersen-lab.github.io/ivar/html/">iVar on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="http://dx.doi.org/10.1186/s13059-018-1618-7">An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>assembly_metrics</code>: Mapping Statistics</summary>
<p>The <code>assembly_metrics</code> task generates mapping statistics from a BAM file. It uses samtools to generate a summary of the mapping statistics, which includes coverage, depth, average base quality, average mapping quality, and other relevant metrics.</p>
<p>This task is run twice: once on the untrimmed reads and, if primer trimming is enabled, once on the primer-trimmed reads. This allows for a comparison of mapping statistics before and after primer trimming, which can be useful for assessing the impact of primer trimming on the quality of the alignment and subsequent analyses.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>assembly_metrics</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ivar_consensus</code>: Consensus Assembly</summary>
<p>iVar's <code>consensus</code> tool generates a reference-based consensus assembly. Several parameters can be set that determine the stringency of the consensus assembly, including minimum quality, minimum allele frequency, and minimum depth.</p>
<p>For TheiaCoV, the following default parameters are used:</p>
<ul>
<li>minimum quality: 20</li>
<li>minimum depth: 100</li>
<li>minimum allele frequency: 0.6</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">iVar Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_ivar_consensus.wdl">task_ivar_consensus.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://andersen-lab.github.io/ivar/html/">Ivar on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://andersen-lab.github.io/ivar/html/manualpage.html">Ivar Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="http://dx.doi.org/10.1186/s13059-018-1618-7">An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ivar_variants</code>: Variant Calling</summary>
<p>iVar uses the outputs of <code>samtools mpileup</code> to call single nucleotide variants (SNVs) and insertions/deletions (indels). Several key parameters can be set to determine the stringency of variant calling, including minimum quality, minimum allele frequency, and minimum depth.</p>
<p>This task returns a VCF file containing all called variants, the number of detected variants, and the proportion of those variants with allele frequencies between 0.6 and 0.9 (also known as <em>intermediate</em> variants).</p>
<p>For TheiaCoV, the following default parameters are used:</p>
<ul>
<li>minimum quality: 20</li>
<li>minimum depth: 100</li>
<li>minimum allele frequency: 0.06</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">iVar Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_ivar_variant_call.wdl">task_ivar_variant_call.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://andersen-lab.github.io/ivar/html/">Ivar on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://andersen-lab.github.io/ivar/html/manualpage.html">Ivar Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="http://dx.doi.org/10.1186/s13059-018-1618-7">An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">iVar Consensus Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_ivar_consensus.wdl">wf_ivar_consensus.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<p><strong><em>If influenza</em></strong></p>
<details class="task">
<summary><code>irma</code>: Assembly and Characterization</summary>
<p>Cleaned reads are assembled using <code>irma</code> which stands for Iterative Refinement Meta-Assembler. IRMA first sorts reads to Flu genome segments using LABEL, then iteratively maps read to collection of reference sequences (in this case for Influenza virus) and iteratively edits the references to account for high population diversity and mutational rates that are characteristic of Influenza genomes. Assemblies produced by <code>irma</code> will be ordered from largest to smallest assembled flu segment. <code>irma</code> also performs typing and subtyping as part of the assembly process. Note: IRMA does not differentiate between Flu B Victoria and Yamagata lineages. For determining this information, please review the <code>abricate</code> task outputs which will provide this information.</p>
<p>Due to the segmented nature of the Influenza genome and the various downstream bioinformatics tools that require the genome assembly, the IRMA task &amp; TheiaCoV workflows output various genome assembly files. Briefly they are:</p>
<ul>
<li><code>assembly_fasta</code> - The full genome assembly in FASTA format, with 1 FASTA entry per genome segment. There should be 8 segments in total, but depending on the quality and depth of sequence data, some segments may not be assembled and nor present in this output file.</li>
<li><code>irma_assembly_fasta_concatenated</code> - The full genome assembly in FASTA format, but with all segments concatenated into a single FASTA entry. This is not your typical FASTA file and is purposely created to be used with a custom Nextclade dataset for the H5N1 B3.13 genotype that is based on a concatenated reference genome.</li>
<li><code>irma_&lt;segment-abbreviation&gt;_segment_fasta</code> - Individual FASTA files that only contain the sequence for 1 segment, for example the HA segment. There are 8 of these in total.</li>
</ul>
<p>General statistics about the assembly are generated with the <code>consensus_qc</code> task (<a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a>).</p>
<div class="admonition techdetails">
<p class="admonition-title">IRMA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_irma.wdl">task_irma.wdl</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://wonder.cdc.gov/amd/flu/irma/">IRMA website</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-3030-6">Viral deep sequencing needs an adaptive approach: IRMA, the iterative refinement meta-assembler</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>read_QC_trim</code>: Read Quality Trimming, Adapter Removal, Quantification, and Identification</summary>
<p><code>read_QC_trim</code> is a sub-workflow that removes low-quality reads, low-quality regions of reads, and sequencing adapters to improve data quality. It uses a number of tasks, described below. The differences between the PE and SE versions of the <code>read_QC_trim</code> sub-workflow lie in the default parameters, the use of two or one input read file(s), and the different output files.</p>
<details class="task">
<summary><code>HRRT</code>: Human Host Sequence Removal</summary>
<p>All reads of human origin <strong>are removed</strong>, including their mates, by using NCBI's <a href="https://github.com/ncbi/sra-human-scrubber"><strong>human read removal tool (HRRT)</strong></a>. </p>
<p>HRRT is based on the <a href="https://doi.org/10.1186/s13059-021-02490-0">SRA Taxonomy Analysis Tool</a> and employs a k-mer database constructed of k-mers from Eukaryota derived from all human RefSeq records with any k-mers found in non-Eukaryota RefSeq records subtracted from the database.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI-Scrub Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ncbi_scrub.wdl">task_ncbi_scrub.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/sra-human-scrubber">HRRT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2023/02/02/scrubbing-human-sequences-sra-submissions/">HRRT on NCBI</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_processing</code> is set to <code>"trimmomatic"</code>. To use <code>fastp</code> instead, set <code>read_processing</code> to <code>"fastp"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>Trimmomatic</code>: Read Trimming (default)</summary>
<p>Read proccessing is available via <code>Trimmomatic</code> by default.</p>
<p>Trimmomatic trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>Trimmomatic</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_trimmomatic.wdl">task_trimmomatic.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/usadellab/Trimmomatic">Trimmomatic on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic Website</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590/">Trimmomatic: a flexible trimmer for Illumina sequence data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>fastp</code>: Read Trimming (alternative)</summary>
<p>To activate this task, set <code>read_processing</code> to <code>"fastp"</code>.</p>
<p><code>fastp</code> trims low-quality regions of Illumina paired-end or single-end reads with a sliding window (with a default window size of 4, specified with <code>trim_window_size</code>), cutting once the average quality within the window falls below the <code>trim_quality_trim_score</code> (default of 20 for paired-end, 30 for single-end). The read is discarded if it is trimmed below <code>trim_minlen</code> (default of 75 for paired-end, 25 for single-end).</p>
<p><code>fastp</code> also has additional default parameters and features that are not a part of <code>trimmomatic</code>'s default configuration.</p>
<details class="toggle">
<summary><code>fastp</code> default read-trimming parameters</summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-g</td>
<td>enables polyG tail trimming</td>
</tr>
<tr>
<td>-5 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>-3 20</td>
<td>enables read end-trimming</td>
</tr>
<tr>
<td>--detect_adapter_for_pe</td>
<td>enables adapter-trimming <strong>only for paired-end reads</strong></td>
</tr>
</tbody>
</table>
<p>Additional arguments can be passed using the <code>fastp_args</code> optional parameter.</p>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Trimmomatic and fastp Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_fastp.wdl">task_fastp.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/OpenGene/fastp">fastp on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/34/17/i884/5093234?login=false">fastp: an ultra-fast all-in-one FASTQ preprocessor</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>BBDuk</code>: Adapter Trimming and PhiX Removal</summary>
<p>Adapters are manufactured oligonucleotide sequences attached to DNA fragments during the library preparation process. In Illumina sequencing, these adapter sequences are required for attaching reads to flow cells. You can read more about <a href="https://emea.support.illumina.com/bulletins/2020/06/illumina-adapter-portfolio.html">Illumina adapters here</a>. For genome analysis, it's important to remove these sequences since they're not actually from your sample. If you don't remove them, the downstream analysis may be affected.</p>
<p>The <code>bbduk</code> task removes adapters from sequence reads. To do this:</p>
<ul>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/repair-guide/">Repair</a> from the <a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/">BBTools</a> package reorders reads in paired fastq files to ensure the forward and reverse reads of a pair are in the same position in the two fastq files (it <em>re-pairs</em>).</li>
<li><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk</a>  (<em>"Bestus Bioinformaticus" Decontamination Using Kmers</em>) is then used to trim the adapters and filter out all reads that have a 31-mer match to <a href="https://emea.illumina.com/products/by-type/sequencing-kits/cluster-gen-sequencing-reagents/phix-control-v3.html">PhiX</a>, which is commonly added to Illumina sequencing runs to monitor and/or improve overall run quality.</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">BBDuk Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_bbduk.wdl">task_bbduk.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://sourceforge.net/projects/bbmap/">BBMap on SourceForge</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://archive.jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/">BBDuk Guide (archived)</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition dna">
<p>By default, <code>read_qc</code> is set to <code>"fastq_scan"</code>. To use <code>fastqc</code> instead, set <code>read_qc</code> to <code>"fastqc"</code>. These tasks are mutually exclusive.</p>
<details class="task">
<summary><code>fastq-scan</code>: Read Quantification (default)</summary>
<p>Read quantification is available via <code>fastq-scan</code> by default.</p>
<p><code>fastq-scan</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>fastq-scan</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastq_scan.wdl">task_fastq_scan.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/fastq-scan/blob/master/README.md">fastq-scan on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>FastQC</code>: Read Quantification (alternative)</summary>
<p>To activate this task, set <code>read_qc</code> to <code>"fastqc"</code>.</p>
<p><code>FastQC</code> quantifies the forward and reverse reads in FASTQ files. For paired-end data, it also provide the total number of read pairs. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>This tool also provides a graphical visualization of the read quality.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>FastQC</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastqc.wdl">task_fastqc.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/s-andrews/FastQC">FastQC on Github</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC Website</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<details class="task">
<summary><code>Kraken2</code>: Read Identification</summary>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<p>Kraken2 is run on both the raw and clean reads.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">Kraken2 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_pe.wdl">wf_read_QC_trim_pe.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_se.wdl">wf_read_QC_trim_se.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ivar_consensus</code>: Alignment, Consensus, Variant Detection, and Assembly Statistics</summary>
<p><code>iVar Consensus</code> is a sub-workflow within TheiaCoV that performs reference-based consensus assembly using the <a href="https://andersen-lab.github.io/ivar/html/index.html">iVar</a> tool by Nathan Grubaugh from the Andersen lab.</p>
<details class="task">
<summary><code>bwa</code>: Read Alignment to the Reference</summary>
<p>BWA (Burrow-Wheeler Aligner) is used to align the cleaned read files to a reference genome, either  determined by the user or provided by the <a href="./#org-specific"><em>organism-specific parameters</em> section</a> (see above). The resulting BAM file is used for primer trimming, variant calling, and consensus generation in downstream tasks.</p>
<div class="admonition techdetails">
<p class="admonition-title">BWA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_bwa.wdl">task_bwa.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/lh3/bwa">BWA on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://bio-bwa.sourceforge.net/">BWA Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and accurate short read alignment with Burrows-Wheeler transform</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ivar_trim</code>: Primer Trimming (optional)</summary>
<p>To deactivate this task, set <code>trim_primers</code> to <code>false</code>.</p>
<p>Using the user-provided (or, more rarely, a <a href="./#org-specific"><em>organism-specific parameters</em>-determined</a>) <code>primer_bed</code> file, iVar soft-clips primer sequences from an aligned and sorted BAM file and then trims the reads based on a quality threshold of 20 using a sliding window approach. If the resulting read is greater than 30 bp, the read is written to a a new BAM file consisting of only trimmed reads (or reads that did not have a primer identified).</p>
<div class="admonition techdetails">
<p class="admonition-title">iVar Trim Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ivar_primer_trim.wdl">task_ivar_primer_trim.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://andersen-lab.github.io/ivar/html/">iVar on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://andersen-lab.github.io/ivar/html/">iVar on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="http://dx.doi.org/10.1186/s13059-018-1618-7">An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>assembly_metrics</code>: Mapping Statistics</summary>
<p>The <code>assembly_metrics</code> task generates mapping statistics from a BAM file. It uses samtools to generate a summary of the mapping statistics, which includes coverage, depth, average base quality, average mapping quality, and other relevant metrics.</p>
<p>This task is run twice: once on the untrimmed reads and, if primer trimming is enabled, once on the primer-trimmed reads. This allows for a comparison of mapping statistics before and after primer trimming, which can be useful for assessing the impact of primer trimming on the quality of the alignment and subsequent analyses.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>assembly_metrics</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ivar_consensus</code>: Consensus Assembly</summary>
<p>iVar's <code>consensus</code> tool generates a reference-based consensus assembly. Several parameters can be set that determine the stringency of the consensus assembly, including minimum quality, minimum allele frequency, and minimum depth.</p>
<p>For TheiaCoV, the following default parameters are used:</p>
<ul>
<li>minimum quality: 20</li>
<li>minimum depth: 100</li>
<li>minimum allele frequency: 0.6</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">iVar Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_ivar_consensus.wdl">task_ivar_consensus.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://andersen-lab.github.io/ivar/html/">Ivar on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://andersen-lab.github.io/ivar/html/manualpage.html">Ivar Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="http://dx.doi.org/10.1186/s13059-018-1618-7">An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>ivar_variants</code>: Variant Calling</summary>
<p>iVar uses the outputs of <code>samtools mpileup</code> to call single nucleotide variants (SNVs) and insertions/deletions (indels). Several key parameters can be set to determine the stringency of variant calling, including minimum quality, minimum allele frequency, and minimum depth.</p>
<p>This task returns a VCF file containing all called variants, the number of detected variants, and the proportion of those variants with allele frequencies between 0.6 and 0.9 (also known as <em>intermediate</em> variants).</p>
<p>For TheiaCoV, the following default parameters are used:</p>
<ul>
<li>minimum quality: 20</li>
<li>minimum depth: 100</li>
<li>minimum allele frequency: 0.06</li>
</ul>
<div class="admonition techdetails">
<p class="admonition-title">iVar Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/variant_detection/task_ivar_variant_call.wdl">task_ivar_variant_call.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://andersen-lab.github.io/ivar/html/">Ivar on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://andersen-lab.github.io/ivar/html/manualpage.html">Ivar Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="http://dx.doi.org/10.1186/s13059-018-1618-7">An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">iVar Consensus Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_ivar_consensus.wdl">wf_ivar_consensus.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>read_QC_trim_ont</code>: Read Quality Trimming, Quantification, and Identification</summary>
<p><code>read_QC_trim_ont</code> is a sub-workflow that filters low-quality reads and trims low-quality regions of reads. It uses several tasks, described below.</p>
<details class="task">
<summary><code>HRRT</code>: Human Host Sequence Removal</summary>
<p>All reads of human origin <strong>are removed</strong>, including their mates, by using NCBI's <a href="https://github.com/ncbi/sra-human-scrubber"><strong>human read removal tool (HRRT)</strong></a>. </p>
<p>HRRT is based on the <a href="https://doi.org/10.1186/s13059-021-02490-0">SRA Taxonomy Analysis Tool</a> and employs a k-mer database constructed of k-mers from Eukaryota derived from all human RefSeq records with any k-mers found in non-Eukaryota RefSeq records subtracted from the database.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI-Scrub Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ncbi_scrub.wdl">task_ncbi_scrub.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/sra-human-scrubber">HRRT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2023/02/02/scrubbing-human-sequences-sra-submissions/">HRRT on NCBI</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>artic_guppyplex</code>: Read Filtering</summary>
<p>Reads are filtered by length with <code>artic_guppyplex</code>, which is a part of the <a href="https://artic.network/fieldbioinformatics/fieldbioinformatics-sop.html"><code>ARTIC</code> protocol</a>. Since TheiaCoV was developed primarily for amplicon-based viral sequencing, this task is included to remove chimeric reads that are either too short or too long.</p>
<div class="admonition techdetails">
<p class="admonition-title">artic_guppyplex Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_artic_guppyplex.wdl">task_artic_guppyplex.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/artic-network/fieldbioinformatics/">ARTIC on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://artic.readthedocs.io/en/latest/?badge=latest">ARTIC Documentation</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Kraken2</code>: Read Identification</summary>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<p>Kraken2 is run on both the raw and clean reads.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">Kraken2 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>NanoPlot</code>: Read Quantification</summary>
<p>NanoPlot is used for the determination of mean quality scores, read lengths, and number of reads. This task is run once with raw reads as input and once with clean reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> clean reads than raw reads.</p>
<p>While this task currently is run <em>outside</em> of the <code>read_QC_trim_ont</code> workflow, it is being included here as it calculates statistics on the read data. This is done so that the actual assembly genome lengths can be used (if an estimated genome length is not provided by the user) to ensure the estimated coverage statistics are accurate.</p>
<div class="admonition techdetails">
<p class="admonition-title">NanoPlot Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_nanoplot.wdl">task_nanoplot.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/wdecoster/NanoPlot">NanoPlot on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/wdecoster/NanoPlot/blob/master/README.md">NanoPlot Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://academic.oup.com/bioinformatics/article/39/5/btad311/7160911">NanoPack2: population-scale evaluation of long-read sequencing data</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">read_QC_trim_ont Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Subworkflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_read_QC_trim_ont.wdl">wf_read_QC_trim_ont.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<p><strong><em>If non-influenza</em></strong></p>
<details class="task">
<summary><code>artic_consensus</code>: Alignment, Primer Trimming, Variant Detection, and Consensus</summary>
<p>This task runs the <code>Artic minion</code> command which is a pipeline with a number of stages, <a href="https://artic.readthedocs.io/en/latest/minion/#0">described in detail in the ARTIC documentation</a>. Briefly, these stages are as follows:</p>
<p>Input reads are aligned to the appropriate reference and only mapped reads are retained. Alignment post-processing occurs, where primers are removed and various trimming steps are undertaken. Variants are detected, and a consensus assembly file is generated.</p>
<p>Please note that the Medaka model is set by default to <code>"r941_min_high_g360"</code> which may not be suitable for your sequencing data. Please be sure to change this parameter if needed.</p>
<div class="admonition info">
<p>Read-trimming is performed on raw read data generated on the ClearLabs instrument and thus not a required step in the TheiaCoV_ClearLabs workflow.</p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Artic Consensus Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_artic_consensus.wdl">task_artic_consensus.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/artic-network/fieldbioinformatics/">ARTIC on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://artic.readthedocs.io/en/latest/?badge=latest">ARTIC Documentation</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>assembly_metrics</code>: Mapping Statistics</summary>
<p>The <code>assembly_metrics</code> task generates mapping statistics from a BAM file. It uses samtools to generate a summary of the mapping statistics, which includes coverage, depth, average base quality, average mapping quality, and other relevant metrics.</p>
<p>This task is run twice: once on the untrimmed reads and, if primer trimming is enabled, once on the primer-trimmed reads. This allows for a comparison of mapping statistics before and after primer trimming, which can be useful for assessing the impact of primer trimming on the quality of the alignment and subsequent analyses.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>assembly_metrics</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<p><strong><em>If influenza</em></strong></p>
<details class="task">
<summary><code>irma</code>: Assembly and Characterization</summary>
<p>Cleaned reads are assembled using <code>irma</code> which stands for Iterative Refinement Meta-Assembler. IRMA first sorts reads to Flu genome segments using LABEL, then iteratively maps read to collection of reference sequences (in this case for Influenza virus) and iteratively edits the references to account for high population diversity and mutational rates that are characteristic of Influenza genomes. Assemblies produced by <code>irma</code> will be ordered from largest to smallest assembled flu segment. <code>irma</code> also performs typing and subtyping as part of the assembly process. Note: IRMA does not differentiate between Flu B Victoria and Yamagata lineages. For determining this information, please review the <code>abricate</code> task outputs which will provide this information.</p>
<p>Due to the segmented nature of the Influenza genome and the various downstream bioinformatics tools that require the genome assembly, the IRMA task &amp; TheiaCoV workflows output various genome assembly files. Briefly they are:</p>
<ul>
<li><code>assembly_fasta</code> - The full genome assembly in FASTA format, with 1 FASTA entry per genome segment. There should be 8 segments in total, but depending on the quality and depth of sequence data, some segments may not be assembled and nor present in this output file.</li>
<li><code>irma_assembly_fasta_concatenated</code> - The full genome assembly in FASTA format, but with all segments concatenated into a single FASTA entry. This is not your typical FASTA file and is purposely created to be used with a custom Nextclade dataset for the H5N1 B3.13 genotype that is based on a concatenated reference genome.</li>
<li><code>irma_&lt;segment-abbreviation&gt;_segment_fasta</code> - Individual FASTA files that only contain the sequence for 1 segment, for example the HA segment. There are 8 of these in total.</li>
</ul>
<p>General statistics about the assembly are generated with the <code>consensus_qc</code> task (<a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a>).</p>
<div class="admonition techdetails">
<p class="admonition-title">IRMA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_irma.wdl">task_irma.wdl</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://wonder.cdc.gov/amd/flu/irma/">IRMA website</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-3030-6">Viral deep sequencing needs an adaptive approach: IRMA, the iterative refinement meta-assembler</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>assembly_metrics</code>: Mapping Statistics</summary>
<p>The <code>assembly_metrics</code> task generates mapping statistics from a BAM file. It uses samtools to generate a summary of the mapping statistics, which includes coverage, depth, average base quality, average mapping quality, and other relevant metrics.</p>
<p>This task is run twice: once on the untrimmed reads and, if primer trimming is enabled, once on the primer-trimmed reads. This allows for a comparison of mapping statistics before and after primer trimming, which can be useful for assessing the impact of primer trimming on the quality of the alignment and subsequent analyses.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>assembly_metrics</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>HRRT</code>: Human Host Sequence Removal</summary>
<p>All reads of human origin <strong>are removed</strong>, including their mates, by using NCBI's <a href="https://github.com/ncbi/sra-human-scrubber"><strong>human read removal tool (HRRT)</strong></a>. </p>
<p>HRRT is based on the <a href="https://doi.org/10.1186/s13059-021-02490-0">SRA Taxonomy Analysis Tool</a> and employs a k-mer database constructed of k-mers from Eukaryota derived from all human RefSeq records with any k-mers found in non-Eukaryota RefSeq records subtracted from the database.</p>
<div class="admonition techdetails">
<p class="admonition-title">NCBI-Scrub Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/read_filtering/task_ncbi_scrub.wdl">task_ncbi_scrub.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/sra-human-scrubber">HRRT on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2023/02/02/scrubbing-human-sequences-sra-submissions/">HRRT on NCBI</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>fastq-scan</code>: Read Quantification</summary>
<p><code>fastq-scan</code> quantifies the reads in the FASTQ files. This task is run once with raw reads as input and once with dehosted reads as input. If QC has been performed correctly, you should expect <strong>fewer</strong> dehosted (or "clean") reads than raw reads.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>fastq-scan</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_fastq_scan.wdl">task_fastq_scan.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/rpetit3/fastq-scan">fastq-scan on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/rpetit3/fastq-scan/blob/master/README.md">fastq-scan on GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>Kraken2</code>: Read Identification</summary>
<p><code>Kraken2</code> is a bioinformatics tool originally designed for metagenomic applications. It has additionally proven valuable for validating taxonomic assignments and checking contamination of single-species (e.g. bacterial isolate, eukaryotic isolate, viral isolate, etc.) whole genome sequence data.</p>
<p>Kraken2 is run on both the raw and clean reads.</p>
<div class="admonition info">
<p class="admonition-title">Database-dependent</p>
<p>This workflow automatically uses a viral-specific Kraken2 database. This database was generated in-house from RefSeq's viral sequence collection and human genome GRCh38. It's available at <code>gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz</code>.</p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Kraken2 Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl">task_kraken2.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/DerrickWood/kraken2/">Kraken2 on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown">Kraken2 Documentation</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://link.springer.com/article/10.1186/s13059-019-1891-0">Improved metagenomic analysis with Kraken 2</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>artic_consensus</code>: Alignment, Primer Trimming, Variant Detection, and Consensus</summary>
<p>This task runs the <code>Artic minion</code> command which is a pipeline with a number of stages, <a href="https://artic.readthedocs.io/en/latest/minion/#0">described in detail in the ARTIC documentation</a>. Briefly, these stages are as follows:</p>
<p>Input reads are aligned to the appropriate reference and only mapped reads are retained. Alignment post-processing occurs, where primers are removed and various trimming steps are undertaken. Variants are detected, and a consensus assembly file is generated.</p>
<p>Please note that the Medaka model is set by default to <code>"r941_min_high_g360"</code> which may not be suitable for your sequencing data. Please be sure to change this parameter if needed.</p>
<div class="admonition info">
<p>Read-trimming is performed on raw read data generated on the ClearLabs instrument and thus not a required step in the TheiaCoV_ClearLabs workflow.</p>
</div>
<div class="admonition techdetails">
<p class="admonition-title">Artic Consensus Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_artic_consensus.wdl">task_artic_consensus.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/artic-network/fieldbioinformatics/">ARTIC on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://artic.readthedocs.io/en/latest/?badge=latest">ARTIC Documentation</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>assembly_metrics</code>: Mapping Statistics</summary>
<p>The <code>assembly_metrics</code> task generates mapping statistics from a BAM file. It uses samtools to generate a summary of the mapping statistics, which includes coverage, depth, average base quality, average mapping quality, and other relevant metrics.</p>
<p>This task is run twice: once on the untrimmed reads and, if primer trimming is enabled, once on the primer-trimmed reads. This allows for a comparison of mapping statistics before and after primer trimming, which can be useful for assessing the impact of primer trimming on the quality of the alignment and subsequent analyses.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>assembly_metrics</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_assembly_metrics.wdl">task_assembly_metrics.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">samtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">samtools</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/bioinformatics/btp352">The Sequence Alignment/Map format and SAMtools</a><br><a href="https://doi.org/10.1093/gigascience/giab008">Twelve Years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<div class="admonition dna">
<p>Since this workflow requires FASTA files as input, no assembly or read trimming is performed, and the workflow proceeds directly to the "post-assembly tasks" section below.</p>
</div>
</div>
</div>
</div>
</div>
<h4 id="post-assembly-tasks">Post-Assembly Tasks<a class="headerlink" href="#post-assembly-tasks" title="Permanent link">¶</a></h4>
<div class="admonition dna">
<p>These tasks are performed for all organisms after assembly (or directly after input for TheiaCoV_FASTA).</p>
</div>
<details class="task">
<summary><code>consensus_qc</code>: Assembly Statistics</summary>
<p>The consensus_qc task generates a summary of genomic statistics from a consensus genome. This includes the total number of bases, "N" bases, degenerate bases, and an estimate of the percent coverage to the reference genome.</p>
<div class="admonition techdetails">
<p class="admonition-title"><code>consensus_qc</code> Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_consensus_qc.wdl">task_consensus_qc.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>qc_check</code>: Check QC Metrics Against User-Defined Thresholds (optional)</summary>
<p>To activate this task, provide a <code>qc_check_table</code> as input.</p>
<p>The <code>qc_check</code> task compares generated QC metrics against user-defined thresholds for each metric. This task will run if the user provides a <code>qc_check_table</code> TSV file. If all QC metrics meet the threshold, the <code>qc_check</code> output variable will read <code>QC_PASS</code>. Otherwise, the output will read <code>QC_NA</code> if the task could not proceed or <code>QC_ALERT</code> followed by a string indicating what metric failed.</p>
<p>The <code>qc_check</code> task applies quality thresholds according to the specified organism, which should match the <em>standardized</em> <code>organism</code> input in the TheiaCoV workflows.</p>
<details class="toggle">
<summary>Formatting the <em>qc_check_table.tsv</em></summary>
<ul>
<li>The first column of the qc_check_table lists the <code>organism</code> that the task will assess and the header of this column must be "<strong>taxon</strong>".</li>
<li>Each subsequent column indicates a QC metric and lists a threshold for each organism that will be checked. <strong>The column names must exactly match expected values, so we highly recommend copy and pasting the header from the template file below as a starting place.</strong></li>
</ul>
</details>
<details class="toggle">
<summary>Template <em>qc_check_table.tsv</em> files</summary>
<ul>
<li>TheiaCoV_Illumina_PE: <a href="../../../assets/files/TheiaCoV_Illumina_PE_qc_check_template.tsv">TheiaCoV_Illumina_PE_qc_check_template.tsv</a></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Example Purposes Only</p>
<p>The QC threshold values shown in the file above are for example purposes only and should not be presumed to be sufficient for every dataset.</p>
</div>
</details>
<div class="admonition techdetails">
<p class="admonition-title">qc_check Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/comparisons/task_qc_check_phb.wdl">task_qc_check_phb.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<h4 id="org-specific-tasks">Organism-specific Characterization Tasks<a class="headerlink" href="#org-specific-tasks" title="Permanent link">¶</a></h4>
<div class="admonition dna">
<p>The following tasks are organism-specific. The following table summarizes the characterization tools that are run for the indicated organism.</p>
<table>
<thead>
<tr>
<th></th>
<th><strong>SARS-CoV-2</strong></th>
<th><strong>Mpox</strong></th>
<th><strong>West Nile Virus</strong></th>
<th><strong>Influenza</strong></th>
<th><strong>RSV-A</strong></th>
<th><strong>RSV-B</strong></th>
<th><strong>HIV</strong></th>
<th><strong>Measles</strong></th>
<th><strong>Mumps</strong></th>
<th><strong>Rubella</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Pangolin</strong></td>
<td>✅</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
</tr>
<tr>
<td><strong>Nextclade</strong></td>
<td>✅</td>
<td>✅</td>
<td>❌</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>❌</td>
<td>✅</td>
<td>❌</td>
<td>❌</td>
</tr>
<tr>
<td><strong>VADR</strong></td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
<td>❌</td>
<td>✅</td>
<td>✅</td>
<td>✅</td>
</tr>
<tr>
<td><strong>VADR Flu Segments</strong></td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>✅</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
</tr>
<tr>
<td><strong>Quasitools HyDRA</strong></td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>✅</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
</tr>
<tr>
<td><strong>IRMA</strong></td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>✅</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
</tr>
<tr>
<td><strong>Abricate</strong></td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>✅</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
</tr>
<tr>
<td><strong>% Gene Coverage</strong></td>
<td>✅</td>
<td>✅</td>
<td>➕</td>
<td>➕</td>
<td>➕</td>
<td>➕</td>
<td>➕</td>
<td>➕</td>
<td>➕</td>
<td>➕</td>
</tr>
<tr>
<td><strong>Antiviral Detection</strong></td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>✅</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
</tr>
<tr>
<td><strong>GenoFLU</strong></td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>✅</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
<td>❌</td>
</tr>
</tbody>
</table>
<p>✅ This task runs automatically for these organisms<br>
➕ This task can run for these organisms if optional parameter(s) are provided; see task description for details.<br>
❌ This task will not run for these organisms</p>
</div>
<div class="tabbed-set tabbed-alternate" data-tabs="5:10"><input checked="checked" id="sars-cov-2_1" name="__tabbed_5" type="radio"><input id="mpox_1" name="__tabbed_5" type="radio"><input id="west-nile-virus_1" name="__tabbed_5" type="radio"><input id="influenza_1" name="__tabbed_5" type="radio"><input id="rsv-a_1" name="__tabbed_5" type="radio"><input id="rsv-b_1" name="__tabbed_5" type="radio"><input id="hiv_1" name="__tabbed_5" type="radio"><input id="measles_1" name="__tabbed_5" type="radio"><input id="mumps_1" name="__tabbed_5" type="radio"><input id="rubella_1" name="__tabbed_5" type="radio"><div class="tabbed-labels"><label for="sars-cov-2_1">SARS-CoV-2</label><label for="mpox_1">Mpox</label><label for="west-nile-virus_1">West Nile Virus</label><label for="influenza_1">Influenza</label><label for="rsv-a_1">RSV-A</label><label for="rsv-b_1">RSV-B</label><label for="hiv_1">HIV</label><label for="measles_1">Measles</label><label for="mumps_1">Mumps</label><label for="rubella_1">Rubella</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<details class="task">
<summary><code>pangolin</code></summary>
<p>Pangolin (Phylogenetic Assignment of Named Global Outbreak Lineages) was developed to implement a dynamic nomenclature for designating SARS-CoV-2 lineage assignments and is used by researchers and public health agencies worldwide to track the spread and transmission of SARS-CoV-2.</p>
<p>Pangolin aligns input sequences against an early SARS-CoV-2 reference and generates a unique hash for the alignment. The hash is checked against a designation cache to see if it matches any previously identified lineages, and is checked via <a href="https://github.com/cov-lineages/scorpio">scorpio</a> (<em>Serious Constellations of Reoccuring Phylogenetically-Independent Origin</em>) to determine if the hash matches any variant of concern (VOC) <a href="https://github.com/cov-lineages/constellations/tree/main"><em>constellations</em></a>, which are groups of functionally meaningful mutations that can independently evolve. Following a QC check, an inference pipeline is run: either <a href="https://cov-lineages.org/resources/pangolin/pangolearn.html">pangoLEARN</a> or <a href="https://github.com/yatisht/usher">UShER</a> (which is the default inference model). The final lineage report is then generated.</p>
<div class="admonition techdetails">
<p class="admonition-title">Pangolin Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/betacoronavirus/task_pangolin.wdl">task_pangolin.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/cov-lineages/pangolin">Pangolin on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://cov-lineages.org/resources/pangolin.html">Pangolin on cov-lineages.org</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1038/s41564-020-0770-5">A dynamic nomenclature proposal for SARS-CoV-2 lineages to assist genomic epidemiology</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>nextclade</code></summary>
<p>Nextclade is an open-source project used to analyze viral genomes, particularly for clade assignment and mutation calling. Simply, Nextclade works by aligning viral genomes to a reference genome, calling variants between the two sequences, and then assigning clades based on those identified mutations. </p>
<p>Clade assignment is performed via <em>phylogenetic placement</em>. Phylogenetic placement compares the mutations of the provided sequence to the mutations of each node found in a reference tree, where the root of that tree is the reference genome. The node that is most similar to the sample is used to both assign a clade designation and calculate where the sample should be placed in the phylogenetic tree.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nextclade Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_nextclade.wdl">task_nextclade.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/nextclade">https://github.com/nextstrain/nextclade</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/">Nextclade</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03773">Nextclade: clade assignment, mutation calling and quality control for viral genomes.</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>vadr</code></summary>
<p>VADR (Viral Annotation DefineR) annotates and validates completed assembly files. For details on VADR default models/parameters, see the <a href="./#org-specific"><em>organism-specific parameters and logic</em> section</a>. It was primarily developed to test viral sequences to confirm they would be accepted to NCBI's GenBank data repository, but has found wide usage in general sequence validation and annotation.</p>
<p>As part of the analysis of the assemblies, more than 70 types of unexpected characteristics, also known as <em>alerts</em>, can be reported. Any identified alerts can be found in the <code>vadr_alerts_list</code> output. Fatal alerts indicate that the sample is unlikely to be accepted to GenBank; non-fatal alerts are designated as passing sequences, but may still require further investigation. A full description of the potential alerts can be found on the <a href="https://github.com/ncbi/vadr/blob/master/documentation/alerts.md">VADR README here</a>, including details on how to allow sequencecs to pass despite having fatal alerts.</p>
<div class="admonition techdetails">
<p class="admonition-title">VADR Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_vadr.wdl">task_vadr.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/vadr">https://github.com/ncbi/vadr</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/vadr/wiki">https://github.com/ncbi/vadr/wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td>For SARS-CoV-2: <em><a href="https://doi.org/10.1093/nargab/lqad002">Faster SARS-CoV-2 sequence validation and annotation for GenBank using VADR</a></em><br> For non-SARS_CoV-2: <a href="https://doi.org/10.1186/s12859-020-3537-3"><em>VADR: validation and annotation of virus sequence submissions to GenBank</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>gene_coverage</code></summary>
<p>This task calculates the percent of a region (typically genes) covered above a minimum depth using <code>samtools</code> and basic arithmetic. By default, this task runs for SARS-CoV-2 and Mpox, but if a BED file is provided with regions of interest, this task can run for other organisms as well.</p>
<div class="admonition techdetails">
<p class="admonition-title">Gene Coverage Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_gene_coverage.wdl">task_gene_coverage.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">SAMtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">SAMTools Manual</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/gigascience/giab008">Twelve years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>nextclade</code></summary>
<p>Nextclade is an open-source project used to analyze viral genomes, particularly for clade assignment and mutation calling. Simply, Nextclade works by aligning viral genomes to a reference genome, calling variants between the two sequences, and then assigning clades based on those identified mutations. </p>
<p>Clade assignment is performed via <em>phylogenetic placement</em>. Phylogenetic placement compares the mutations of the provided sequence to the mutations of each node found in a reference tree, where the root of that tree is the reference genome. The node that is most similar to the sample is used to both assign a clade designation and calculate where the sample should be placed in the phylogenetic tree.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nextclade Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_nextclade.wdl">task_nextclade.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/nextclade">https://github.com/nextstrain/nextclade</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/">Nextclade</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03773">Nextclade: clade assignment, mutation calling and quality control for viral genomes.</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>vadr</code></summary>
<p>VADR (Viral Annotation DefineR) annotates and validates completed assembly files. For details on VADR default models/parameters, see the <a href="./#org-specific"><em>organism-specific parameters and logic</em> section</a>. It was primarily developed to test viral sequences to confirm they would be accepted to NCBI's GenBank data repository, but has found wide usage in general sequence validation and annotation.</p>
<p>As part of the analysis of the assemblies, more than 70 types of unexpected characteristics, also known as <em>alerts</em>, can be reported. Any identified alerts can be found in the <code>vadr_alerts_list</code> output. Fatal alerts indicate that the sample is unlikely to be accepted to GenBank; non-fatal alerts are designated as passing sequences, but may still require further investigation. A full description of the potential alerts can be found on the <a href="https://github.com/ncbi/vadr/blob/master/documentation/alerts.md">VADR README here</a>, including details on how to allow sequencecs to pass despite having fatal alerts.</p>
<div class="admonition techdetails">
<p class="admonition-title">VADR Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_vadr.wdl">task_vadr.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/vadr">https://github.com/ncbi/vadr</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/vadr/wiki">https://github.com/ncbi/vadr/wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td>For SARS-CoV-2: <em><a href="https://doi.org/10.1093/nargab/lqad002">Faster SARS-CoV-2 sequence validation and annotation for GenBank using VADR</a></em><br> For non-SARS_CoV-2: <a href="https://doi.org/10.1186/s12859-020-3537-3"><em>VADR: validation and annotation of virus sequence submissions to GenBank</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>gene_coverage</code></summary>
<p>This task calculates the percent of a region (typically genes) covered above a minimum depth using <code>samtools</code> and basic arithmetic. By default, this task runs for SARS-CoV-2 and Mpox, but if a BED file is provided with regions of interest, this task can run for other organisms as well.</p>
<div class="admonition techdetails">
<p class="admonition-title">Gene Coverage Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/basic_statistics/task_gene_coverage.wdl">task_gene_coverage.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/samtools/samtools">SAMtools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://www.htslib.org/doc/samtools.html">SAMTools Manual</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/gigascience/giab008">Twelve years of SAMtools and BCFtools</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>vadr</code></summary>
<p>VADR (Viral Annotation DefineR) annotates and validates completed assembly files. For details on VADR default models/parameters, see the <a href="./#org-specific"><em>organism-specific parameters and logic</em> section</a>. It was primarily developed to test viral sequences to confirm they would be accepted to NCBI's GenBank data repository, but has found wide usage in general sequence validation and annotation.</p>
<p>As part of the analysis of the assemblies, more than 70 types of unexpected characteristics, also known as <em>alerts</em>, can be reported. Any identified alerts can be found in the <code>vadr_alerts_list</code> output. Fatal alerts indicate that the sample is unlikely to be accepted to GenBank; non-fatal alerts are designated as passing sequences, but may still require further investigation. A full description of the potential alerts can be found on the <a href="https://github.com/ncbi/vadr/blob/master/documentation/alerts.md">VADR README here</a>, including details on how to allow sequencecs to pass despite having fatal alerts.</p>
<div class="admonition techdetails">
<p class="admonition-title">VADR Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_vadr.wdl">task_vadr.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/vadr">https://github.com/ncbi/vadr</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/vadr/wiki">https://github.com/ncbi/vadr/wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td>For SARS-CoV-2: <em><a href="https://doi.org/10.1093/nargab/lqad002">Faster SARS-CoV-2 sequence validation and annotation for GenBank using VADR</a></em><br> For non-SARS_CoV-2: <a href="https://doi.org/10.1186/s12859-020-3537-3"><em>VADR: validation and annotation of virus sequence submissions to GenBank</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>nextclade</code></summary>
<p>Nextclade is an open-source project used to analyze viral genomes, particularly for clade assignment and mutation calling. Simply, Nextclade works by aligning viral genomes to a reference genome, calling variants between the two sequences, and then assigning clades based on those identified mutations. </p>
<p>Clade assignment is performed via <em>phylogenetic placement</em>. Phylogenetic placement compares the mutations of the provided sequence to the mutations of each node found in a reference tree, where the root of that tree is the reference genome. The node that is most similar to the sample is used to both assign a clade designation and calculate where the sample should be placed in the phylogenetic tree.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nextclade Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_nextclade.wdl">task_nextclade.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/nextclade">https://github.com/nextstrain/nextclade</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/">Nextclade</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03773">Nextclade: clade assignment, mutation calling and quality control for viral genomes.</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>vadr</code></summary>
<p>VADR (Viral Annotation DefineR) annotates and validates completed assembly files. For details on VADR default models/parameters, see the <a href="./#org-specific"><em>organism-specific parameters and logic</em> section</a>. It was primarily developed to test viral sequences to confirm they would be accepted to NCBI's GenBank data repository, but has found wide usage in general sequence validation and annotation.</p>
<p>As part of the analysis of the assemblies, more than 70 types of unexpected characteristics, also known as <em>alerts</em>, can be reported. Any identified alerts can be found in the <code>vadr_alerts_list</code> output. Fatal alerts indicate that the sample is unlikely to be accepted to GenBank; non-fatal alerts are designated as passing sequences, but may still require further investigation. A full description of the potential alerts can be found on the <a href="https://github.com/ncbi/vadr/blob/master/documentation/alerts.md">VADR README here</a>, including details on how to allow sequencecs to pass despite having fatal alerts.</p>
<div class="admonition techdetails">
<p class="admonition-title">VADR Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_vadr.wdl">task_vadr.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/vadr">https://github.com/ncbi/vadr</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/vadr/wiki">https://github.com/ncbi/vadr/wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td>For SARS-CoV-2: <em><a href="https://doi.org/10.1093/nargab/lqad002">Faster SARS-CoV-2 sequence validation and annotation for GenBank using VADR</a></em><br> For non-SARS_CoV-2: <a href="https://doi.org/10.1186/s12859-020-3537-3"><em>VADR: validation and annotation of virus sequence submissions to GenBank</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>vadr_flu_segments</code></summary>
<p>This task processes a full or partial influenza genome assembly in multifasta format, along with the output <code>.tar.gz</code> file from a VADR run. It extracts each segment into its own fasta file and also generates a concatenated fasta containing all segments combined into a single sequence. Segment names are assigned based on the specified flu type (A or B) and the segment classification found in the <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#sqc">VADR .sqc file</a>.</p>
<p>Note: Results may be unreliable if segment lengths deviate from those expected for Influenza A or B. For best results, the input assembly should contain all 8 segments as separate contigs. If the assembly is partial, the task will still extract available segments but may not produce a complete concatenated sequence. Empty fasta files will be created for missing segments.</p>
<div class="admonition techdetails">
<p class="admonition-title">VADR Flu Segments Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/influenza/task_vadr_flu_segments.wdl">task_vadr_flu_segments.wdl</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>irma</code></summary>
<p>Cleaned reads are assembled using <code>irma</code> which stands for Iterative Refinement Meta-Assembler. IRMA first sorts reads to Flu genome segments using LABEL, then iteratively maps read to collection of reference sequences (in this case for Influenza virus) and iteratively edits the references to account for high population diversity and mutational rates that are characteristic of Influenza genomes. Assemblies produced by <code>irma</code> will be ordered from largest to smallest assembled flu segment. <code>irma</code> also performs typing and subtyping as part of the assembly process. Note: IRMA does not differentiate between Flu B Victoria and Yamagata lineages. For determining this information, please review the <code>abricate</code> task outputs which will provide this information.</p>
<div class="admonition techdetails">
<p class="admonition-title">IRMA Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/assembly/task_irma.wdl">task_irma.wdl</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://wonder.cdc.gov/amd/flu/irma/">IRMA website</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-3030-6">Viral deep sequencing needs an adaptive approach: IRMA, the iterative refinement meta-assembler</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>abricate</code></summary>
<p>ABRicate assigns types and subtype/lineages for flu samples using a version of the <a href="https://github.com/INSaFLU/INSaFLU">INSaFLU</a> ("INSide the FLU") database <a href="https://github.com/epi2me-labs/wf-flu/tree/master/data/primer_schemes/V1/blastdb/insaflu">described here</a>. </p>
<p>ABRicate typically works by screening contigs for the presence of acquired resistance genes, but when using the INSaFLU database, the algorithm works by assigning contigs to the most closely corresponding viral segment in the INSaFLU database, which is used to call the flu type and subtype.</p>
<div class="admonition techdetails">
<p class="admonition-title">ABRicate Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/2dff853defc6ea540a058873f6fe6a78cc2350c7/tasks/gene_typing/drug_resistance/task_abricate.wdl#L59">task_abricate.wdl (abricate_flu subtask)</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/tseemann/abricate">ABRicate on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/tseemann/abricate">ABRicate on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><em>INSaFLU database</em>: <a href="https://doi.org/10.1186/s13073-018-0555-0">INSaFLU: an automated open web-based bioinformatics suite "from-reads" for influenza whole-genome-sequencing-based surveillance</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>flu_antiviral_substitutions</code></summary>
<p>This subworkflow determines if any antiviral mutations are present in the HA, NA, and MP segments of H1N1 or H3N2 flu sample, or any in non-subtype-specific PA, PB1, and PB2 segments.</p>
<p>These mutations are identified by generating a multiple sequence alignment (MSA) between each individual flu segment and the respective reference genome using MAFFT. Amino acid mutations are then called from the MSA. The resulting mutations are compared against <a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/drug_resistance/task_flu_antiviral_subs.wdl">a list of known amino-acid substitutions associated with antiviral resistance</a> and any matches are reported. </p>
<p>This list of amino-acid substitutions includes both substitutions reported in the scientific literature and those inferred to potentially cause antiviral resistance based on analogous antiviral mutations in other flu subtypes. <a href="../../../assets/files/antiviral_resistance_flu_aa_substitutions_explanations.xlsx"><strong>A table with the explanation for each amino-acid substitution in the antiviral resistance task is available here</strong></a>.</p>
<p>The list of known amino-acid substitutions associated with resistance can be expanded via optional user input <code>antiviral_aa_subs</code> in the format "<code>NA:V95A,HA:I97V</code>", i.e. <code>Protein:AAPositionAA</code>. </p>
<details class="toggle">
<summary>Currently, the default mutations considered confer resistance to the following antivirals</summary>
<ul>
<li>A_315675</li>
<li>Amantadine</li>
<li>Compound_367</li>
<li>Favipiravir</li>
<li>Fludase</li>
<li>L_742_001</li>
<li>Laninamivir</li>
<li>Oseltamivir (tamiflu)</li>
<li>Peramivir</li>
<li>Pimodivir</li>
<li>Rimantadine</li>
<li>Xofluza</li>
<li>Zanamivir</li>
</ul>
</details>
<div class="admonition techdetails">
<p class="admonition-title">Antiviral Substitutions Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sub-workflow</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/utilities/wf_influenza_antiviral_substitutions.wdl">wf_influenza_antiviral_substitutions.wdl</a></td>
</tr>
<tr>
<td>Tasks</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/alignment/task_mafft.wdl">task_mafft.wdl</a><br><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/gene_typing/drug_resistance/task_flu_antiviral_subs.wdl">task_flu_antiviral_subs.wdl</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1093/molbev/mst010">MAFFT Multiple Sequence Alignment Software Version 7: Improvements in Performance and Usability</a><br><a href="https://doi.org/10.1016/j.tibtech.2019.09.009">Next-Generation Sequencing: An Eye-Opener for the Surveillance of Antiviral Resistance in Influenza</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>genoflu</code></summary>
<p>This task determines the whole-genome genotype of a H5N1 (<mark>currently only for the 2.3.4.4b clade of H5N1</mark>) flu sample by comparing each segment of the sample against a curated database of H5N1 references. Each segment is assigned a type, and the whole-genome genotype is assigned based on the combination of segment types, according to the <a href="https://github.com/USDA-VS/GenoFLU/blob/main/docs/Genotyping_reference_for_US_H5_2.3.4.4b.pdf">GenoFLU reference table</a>.</p>
<div class="admonition techdetails">
<p class="admonition-title">GenoFLU Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/orthomyxoviridae/task_genoflu.wdl">task_genoflu.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/USDA-VS/GenoFLU">GenoFLU on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/USDA-VS/GenoFLU">GenoFLU on GitHub</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.1016/j.virol.2023.109860">H5N1 highly pathogenic avian influenza clade 2.3.4.4b in wild and domestic birds: Introductions into the United States and reassortments, December 2021-April 2022</a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>nextclade</code></summary>
<p>Nextclade is an open-source project used to analyze viral genomes, particularly for clade assignment and mutation calling. Simply, Nextclade works by aligning viral genomes to a reference genome, calling variants between the two sequences, and then assigning clades based on those identified mutations. </p>
<p>Clade assignment is performed via <em>phylogenetic placement</em>. Phylogenetic placement compares the mutations of the provided sequence to the mutations of each node found in a reference tree, where the root of that tree is the reference genome. The node that is most similar to the sample is used to both assign a clade designation and calculate where the sample should be placed in the phylogenetic tree.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nextclade Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_nextclade.wdl">task_nextclade.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/nextclade">https://github.com/nextstrain/nextclade</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/">Nextclade</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03773">Nextclade: clade assignment, mutation calling and quality control for viral genomes.</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>vadr</code></summary>
<p>VADR (Viral Annotation DefineR) annotates and validates completed assembly files. For details on VADR default models/parameters, see the <a href="./#org-specific"><em>organism-specific parameters and logic</em> section</a>. It was primarily developed to test viral sequences to confirm they would be accepted to NCBI's GenBank data repository, but has found wide usage in general sequence validation and annotation.</p>
<p>As part of the analysis of the assemblies, more than 70 types of unexpected characteristics, also known as <em>alerts</em>, can be reported. Any identified alerts can be found in the <code>vadr_alerts_list</code> output. Fatal alerts indicate that the sample is unlikely to be accepted to GenBank; non-fatal alerts are designated as passing sequences, but may still require further investigation. A full description of the potential alerts can be found on the <a href="https://github.com/ncbi/vadr/blob/master/documentation/alerts.md">VADR README here</a>, including details on how to allow sequencecs to pass despite having fatal alerts.</p>
<div class="admonition techdetails">
<p class="admonition-title">VADR Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_vadr.wdl">task_vadr.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/vadr">https://github.com/ncbi/vadr</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/vadr/wiki">https://github.com/ncbi/vadr/wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td>For SARS-CoV-2: <em><a href="https://doi.org/10.1093/nargab/lqad002">Faster SARS-CoV-2 sequence validation and annotation for GenBank using VADR</a></em><br> For non-SARS_CoV-2: <a href="https://doi.org/10.1186/s12859-020-3537-3"><em>VADR: validation and annotation of virus sequence submissions to GenBank</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>nextclade</code></summary>
<p>Nextclade is an open-source project used to analyze viral genomes, particularly for clade assignment and mutation calling. Simply, Nextclade works by aligning viral genomes to a reference genome, calling variants between the two sequences, and then assigning clades based on those identified mutations. </p>
<p>Clade assignment is performed via <em>phylogenetic placement</em>. Phylogenetic placement compares the mutations of the provided sequence to the mutations of each node found in a reference tree, where the root of that tree is the reference genome. The node that is most similar to the sample is used to both assign a clade designation and calculate where the sample should be placed in the phylogenetic tree.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nextclade Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_nextclade.wdl">task_nextclade.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/nextclade">https://github.com/nextstrain/nextclade</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/">Nextclade</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03773">Nextclade: clade assignment, mutation calling and quality control for viral genomes.</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>vadr</code></summary>
<p>VADR (Viral Annotation DefineR) annotates and validates completed assembly files. For details on VADR default models/parameters, see the <a href="./#org-specific"><em>organism-specific parameters and logic</em> section</a>. It was primarily developed to test viral sequences to confirm they would be accepted to NCBI's GenBank data repository, but has found wide usage in general sequence validation and annotation.</p>
<p>As part of the analysis of the assemblies, more than 70 types of unexpected characteristics, also known as <em>alerts</em>, can be reported. Any identified alerts can be found in the <code>vadr_alerts_list</code> output. Fatal alerts indicate that the sample is unlikely to be accepted to GenBank; non-fatal alerts are designated as passing sequences, but may still require further investigation. A full description of the potential alerts can be found on the <a href="https://github.com/ncbi/vadr/blob/master/documentation/alerts.md">VADR README here</a>, including details on how to allow sequencecs to pass despite having fatal alerts.</p>
<div class="admonition techdetails">
<p class="admonition-title">VADR Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_vadr.wdl">task_vadr.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/vadr">https://github.com/ncbi/vadr</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/vadr/wiki">https://github.com/ncbi/vadr/wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td>For SARS-CoV-2: <em><a href="https://doi.org/10.1093/nargab/lqad002">Faster SARS-CoV-2 sequence validation and annotation for GenBank using VADR</a></em><br> For non-SARS_CoV-2: <a href="https://doi.org/10.1186/s12859-020-3537-3"><em>VADR: validation and annotation of virus sequence submissions to GenBank</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>quasitools</code></summary>
<p><code>quasitools</code> performs genomic characterization for HIV by using the HyDRA module for identifying drug resistance mutations in HIV-1 samples based on the <a href="https://hivdb.stanford.edu/">Stanford HIV Drug Resistance Database</a> and the <a href="https://hivdb.stanford.edu/page/who-sdrm-list/">2009 WHO list for Surveillance of Transmitted HIVDR</a>; see also the papers linked below.</p>
<p>The HyDRA module in quasitools maps the sample sequence against an annotated HIV-1 reference and performs variant calling. Those variants are compared to the databases described above, and any matches are reported, along with the complete list of variants. </p>
<div class="admonition techdetails">
<p class="admonition-title">quasitools Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/species_typing/lentivirus/task_quasitools.wdl">task_quasitools.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/phac-nml/quasitools/">quasitools on GitHub</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://phac-nml.github.io/quasitools/hydra/">quasitools HyDRA README</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td>quasitools preprint: <em><a href="https://doi.org/10.1101/733238">quasitools: A Collection of Tools for Viral Quasispecies Analysis</a></em><br>WHO 2009 Database: <em><a href="https://doi.org/10.1371/journal.pone.0004724">Drug resistance mutations for surveillance of transmitted HIV-1 drug-resistance: 2009 update</a></em><br>Stanford Database: <em><a href="https://doi.org/10.1093/nar/gkg100">Human immunodeficiency virus reverse transcriptase and protease sequence database</a></em></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>nextclade</code></summary>
<p>Nextclade is an open-source project used to analyze viral genomes, particularly for clade assignment and mutation calling. Simply, Nextclade works by aligning viral genomes to a reference genome, calling variants between the two sequences, and then assigning clades based on those identified mutations. </p>
<p>Clade assignment is performed via <em>phylogenetic placement</em>. Phylogenetic placement compares the mutations of the provided sequence to the mutations of each node found in a reference tree, where the root of that tree is the reference genome. The node that is most similar to the sample is used to both assign a clade designation and calculate where the sample should be placed in the phylogenetic tree.</p>
<div class="admonition techdetails">
<p class="admonition-title">Nextclade Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_nextclade.wdl">task_nextclade.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/nextstrain/nextclade">https://github.com/nextstrain/nextclade</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://docs.nextstrain.org/projects/nextclade/en/stable/">Nextclade</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td><a href="https://doi.org/10.21105/joss.03773">Nextclade: clade assignment, mutation calling and quality control for viral genomes.</a></td>
</tr>
</tbody>
</table>
</div>
</details>
<details class="task">
<summary><code>vadr</code></summary>
<p>VADR (Viral Annotation DefineR) annotates and validates completed assembly files. For details on VADR default models/parameters, see the <a href="./#org-specific"><em>organism-specific parameters and logic</em> section</a>. It was primarily developed to test viral sequences to confirm they would be accepted to NCBI's GenBank data repository, but has found wide usage in general sequence validation and annotation.</p>
<p>As part of the analysis of the assemblies, more than 70 types of unexpected characteristics, also known as <em>alerts</em>, can be reported. Any identified alerts can be found in the <code>vadr_alerts_list</code> output. Fatal alerts indicate that the sample is unlikely to be accepted to GenBank; non-fatal alerts are designated as passing sequences, but may still require further investigation. A full description of the potential alerts can be found on the <a href="https://github.com/ncbi/vadr/blob/master/documentation/alerts.md">VADR README here</a>, including details on how to allow sequencecs to pass despite having fatal alerts.</p>
<div class="admonition techdetails">
<p class="admonition-title">VADR Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_vadr.wdl">task_vadr.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/vadr">https://github.com/ncbi/vadr</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/vadr/wiki">https://github.com/ncbi/vadr/wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td>For SARS-CoV-2: <em><a href="https://doi.org/10.1093/nargab/lqad002">Faster SARS-CoV-2 sequence validation and annotation for GenBank using VADR</a></em><br> For non-SARS_CoV-2: <a href="https://doi.org/10.1186/s12859-020-3537-3"><em>VADR: validation and annotation of virus sequence submissions to GenBank</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>vadr</code></summary>
<p>VADR (Viral Annotation DefineR) annotates and validates completed assembly files. For details on VADR default models/parameters, see the <a href="./#org-specific"><em>organism-specific parameters and logic</em> section</a>. It was primarily developed to test viral sequences to confirm they would be accepted to NCBI's GenBank data repository, but has found wide usage in general sequence validation and annotation.</p>
<p>As part of the analysis of the assemblies, more than 70 types of unexpected characteristics, also known as <em>alerts</em>, can be reported. Any identified alerts can be found in the <code>vadr_alerts_list</code> output. Fatal alerts indicate that the sample is unlikely to be accepted to GenBank; non-fatal alerts are designated as passing sequences, but may still require further investigation. A full description of the potential alerts can be found on the <a href="https://github.com/ncbi/vadr/blob/master/documentation/alerts.md">VADR README here</a>, including details on how to allow sequencecs to pass despite having fatal alerts.</p>
<div class="admonition techdetails">
<p class="admonition-title">VADR Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_vadr.wdl">task_vadr.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/vadr">https://github.com/ncbi/vadr</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/vadr/wiki">https://github.com/ncbi/vadr/wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td>For SARS-CoV-2: <em><a href="https://doi.org/10.1093/nargab/lqad002">Faster SARS-CoV-2 sequence validation and annotation for GenBank using VADR</a></em><br> For non-SARS_CoV-2: <a href="https://doi.org/10.1186/s12859-020-3537-3"><em>VADR: validation and annotation of virus sequence submissions to GenBank</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
<div class="tabbed-block">
<details class="task">
<summary><code>vadr</code></summary>
<p>VADR (Viral Annotation DefineR) annotates and validates completed assembly files. For details on VADR default models/parameters, see the <a href="./#org-specific"><em>organism-specific parameters and logic</em> section</a>. It was primarily developed to test viral sequences to confirm they would be accepted to NCBI's GenBank data repository, but has found wide usage in general sequence validation and annotation.</p>
<p>As part of the analysis of the assemblies, more than 70 types of unexpected characteristics, also known as <em>alerts</em>, can be reported. Any identified alerts can be found in the <code>vadr_alerts_list</code> output. Fatal alerts indicate that the sample is unlikely to be accepted to GenBank; non-fatal alerts are designated as passing sequences, but may still require further investigation. A full description of the potential alerts can be found on the <a href="https://github.com/ncbi/vadr/blob/master/documentation/alerts.md">VADR README here</a>, including details on how to allow sequencecs to pass despite having fatal alerts.</p>
<div class="admonition techdetails">
<p class="admonition-title">VADR Technical Details</p>
<table>
<thead>
<tr>
<th></th>
<th>Links</th>
</tr>
</thead>
<tbody>
<tr>
<td>Task</td>
<td><a href="https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/quality_control/advanced_metrics/task_vadr.wdl">task_vadr.wdl</a></td>
</tr>
<tr>
<td>Software Source Code</td>
<td><a href="https://github.com/ncbi/vadr">https://github.com/ncbi/vadr</a></td>
</tr>
<tr>
<td>Software Documentation</td>
<td><a href="https://github.com/ncbi/vadr/wiki">https://github.com/ncbi/vadr/wiki</a></td>
</tr>
<tr>
<td>Original Publication(s)</td>
<td>For SARS-CoV-2: <em><a href="https://doi.org/10.1093/nargab/lqad002">Faster SARS-CoV-2 sequence validation and annotation for GenBank using VADR</a></em><br> For non-SARS_CoV-2: <a href="https://doi.org/10.1186/s12859-020-3537-3"><em>VADR: validation and annotation of virus sequence submissions to GenBank</em></a></td>
</tr>
</tbody>
</table>
</div>
</details>
</div>
</div>
</div>
<h3 id="outputs">Outputs<a class="headerlink" href="#outputs" title="Permanent link">¶</a></h3>
<div class="tabbed-set tabbed-alternate" data-tabs="6:6"><input checked="checked" id="theiacov_illumina_pe_3" name="__tabbed_6" type="radio"><input id="theiacov_illumina_se_3" name="__tabbed_6" type="radio"><input id="theiacov_ont_3" name="__tabbed_6" type="radio"><input id="theiacov_fasta_3" name="__tabbed_6" type="radio"><input id="theiacov_clearlabs_3" name="__tabbed_6" type="radio"><input id="theiacov_fasta_batch_2" name="__tabbed_6" type="radio"><div class="tabbed-labels"><label for="theiacov_illumina_pe_3">TheiaCoV_Illumina_PE</label><label for="theiacov_illumina_se_3">TheiaCoV_Illumina_SE</label><label for="theiacov_ont_3">TheiaCoV_ONT</label><label for="theiacov_fasta_3">TheiaCoV_FASTA</label><label for="theiacov_clearlabs_3">TheiaCoV_ClearLabs</label><label for="theiacov_fasta_batch_2">TheiaCoV_FASTA_Batch</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>abricate_flu_database</td>
<td>String</td>
<td>ABRicate database used for analysis</td>
</tr>
<tr>
<td>abricate_flu_results</td>
<td>File</td>
<td>File containing all results from ABRicate</td>
</tr>
<tr>
<td>abricate_flu_subtype</td>
<td>String</td>
<td>Flu subtype as determined by ABRicate</td>
</tr>
<tr>
<td>abricate_flu_type</td>
<td>String</td>
<td>Flu type as determined by ABRicate</td>
</tr>
<tr>
<td>abricate_flu_version</td>
<td>String</td>
<td>Version of ABRicate</td>
</tr>
<tr>
<td>aligned_bai</td>
<td>String</td>
<td>Index companion file to the bam file generated during the consensus assembly process</td>
</tr>
<tr>
<td>aligned_bam</td>
<td>String</td>
<td>Sorted BAM file containing the alignments of reads to the reference genome</td>
</tr>
<tr>
<td>assembly_fasta</td>
<td>String</td>
<td>Consensus genome assembly; for lower quality flu samples, the output may state "Assembly could not be generated" when there is too little and/or too low quality data for IRMA to produce an assembly. Contigs will be ordered from largest to smallest when IRMA is used.</td>
</tr>
<tr>
<td>assembly_length_unambiguous</td>
<td>Int</td>
<td>Number of unambiguous basecalls within the consensus assembly</td>
</tr>
<tr>
<td>assembly_mean_coverage</td>
<td>String</td>
<td>Mean sequencing depth throughout the consensus assembly. Generated after performing primer trimming and calculated using the SAMtools coverage command</td>
</tr>
<tr>
<td>assembly_method</td>
<td>String</td>
<td>Method employed to generate consensus assembly</td>
</tr>
<tr>
<td>auspice_json</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_h5n1</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza H5N1 whole genome that includes the samples included in the "avian-flu/h5n1-cattle-outbreak" nextstrain build that is focused on B3.13 genotype and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_ha</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza HA segment that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_na</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza NA segment that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>bbduk_docker</td>
<td>String</td>
<td>The Docker image for bbduk, which was used to remove the adapters from the sequences</td>
</tr>
<tr>
<td>bwa_version</td>
<td>String</td>
<td>Version of BWA software used</td>
</tr>
<tr>
<td>consensus_flagstat</td>
<td>File</td>
<td>Output from the SAMtools flagstat command to assess quality of the alignment file (BAM)</td>
</tr>
<tr>
<td>consensus_n_variant_min_depth</td>
<td>Int</td>
<td>Minimum read depth to call variants for iVar consensus and iVar variants. Also represents the minimum consensus support threshold used by IRMA with Illumina Influenza data.</td>
</tr>
<tr>
<td>consensus_stats</td>
<td>File</td>
<td>Output from the SAMtools stats command to assess quality of the alignment file (BAM)</td>
</tr>
<tr>
<td>est_percent_gene_coverage_tsv</td>
<td>File</td>
<td>Percent coverage for each gene in the organism being analyzed (depending on the organism input)</td>
</tr>
<tr>
<td>fastp_html_report</td>
<td>File</td>
<td>The HTML report made with fastp</td>
</tr>
<tr>
<td>fastp_version</td>
<td>String</td>
<td>The version of fastp used</td>
</tr>
<tr>
<td>fastq_scan_clean1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_clean2_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean reverse read quality and length</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean1</td>
<td>Int</td>
<td>The number of forward reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean2</td>
<td>Int</td>
<td>The number of reverse reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean_pairs</td>
<td>String</td>
<td>The number of read pairs after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw1</td>
<td>Int</td>
<td>The number of input forward reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw2</td>
<td>Int</td>
<td>The number of input reserve reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw_pairs</td>
<td>String</td>
<td>The number of input read pairs as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_raw1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_raw2_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw reverse read quality and length</td>
</tr>
<tr>
<td>fastq_scan_version</td>
<td>String</td>
<td>The version of fastq_scan</td>
</tr>
<tr>
<td>fastqc_clean1_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of clean forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_clean2_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of clean reverse read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_docker</td>
<td>String</td>
<td>The Docker container used for fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean1</td>
<td>Int</td>
<td>The number of forward reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean2</td>
<td>Int</td>
<td>The number of reverse reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean_pairs</td>
<td>String</td>
<td>The number of read pairs after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_raw1</td>
<td>Int</td>
<td>The number of input forward reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_num_reads_raw2</td>
<td>Int</td>
<td>The number of input reverse reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_num_reads_raw_pairs</td>
<td>String</td>
<td>The number of input read pairs by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_raw1_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of raw forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_raw2_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of raw reverse read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_version</td>
<td>String</td>
<td>Version of fastqc software used</td>
</tr>
<tr>
<td>flu_A_315675_resistance</td>
<td>String</td>
<td>resistance mutations to A_315675</td>
</tr>
<tr>
<td>flu_L_742_001_resistance</td>
<td>String</td>
<td>resistance mutations to L_742_001</td>
</tr>
<tr>
<td>flu_amantadine_resistance</td>
<td>String</td>
<td>resistance mutations to amantadine</td>
</tr>
<tr>
<td>flu_compound_367_resistance</td>
<td>String</td>
<td>resistance mutations to compound_367</td>
</tr>
<tr>
<td>flu_favipiravir_resistance</td>
<td>String</td>
<td>resistance mutations to favipiravir</td>
</tr>
<tr>
<td>flu_fludase_resistance</td>
<td>String</td>
<td>resistance mutations to fludase</td>
</tr>
<tr>
<td>flu_laninamivir_resistance</td>
<td>String</td>
<td>resistance mutations to laninamivir</td>
</tr>
<tr>
<td>flu_oseltamivir_resistance</td>
<td>String</td>
<td>resistance mutations to oseltamivir (Tamiflu®)</td>
</tr>
<tr>
<td>flu_peramivir_resistance</td>
<td>String</td>
<td>resistance mutations to peramivir (Rapivab®)</td>
</tr>
<tr>
<td>flu_pimodivir_resistance</td>
<td>String</td>
<td>resistance mutations to pimodivir</td>
</tr>
<tr>
<td>flu_rimantadine_resistance</td>
<td>String</td>
<td>resistance mutations to rimantadine</td>
</tr>
<tr>
<td>flu_xofluza_resistance</td>
<td>String</td>
<td>resistance mutations to xofluza (Baloxavir marboxil)</td>
</tr>
<tr>
<td>flu_zanamivir_resistance</td>
<td>String</td>
<td>resistance mutations to zanamivir (Relenza®)</td>
</tr>
<tr>
<td>genoflu_all_segments</td>
<td>String</td>
<td>The genotypes for each individual flu segment</td>
</tr>
<tr>
<td>genoflu_genotype</td>
<td>String</td>
<td>The genotype of the whole genome, based off of the individual segments types</td>
</tr>
<tr>
<td>genoflu_output_tsv</td>
<td>File</td>
<td>The output file from GenoFLU</td>
</tr>
<tr>
<td>genoflu_version</td>
<td>String</td>
<td>The version of GenoFLU used</td>
</tr>
<tr>
<td>irma_assembly_fasta_concatenated</td>
<td>File</td>
<td>Assembly FASTA file of all Influenza genome segments concatenated into one sequence/FASTA entry</td>
</tr>
<tr>
<td>irma_docker</td>
<td>String</td>
<td>Docker image used to run IRMA</td>
</tr>
<tr>
<td>irma_ha_segment_fasta</td>
<td>File</td>
<td>HA (Haemagglutinin) assembly fasta file</td>
</tr>
<tr>
<td>irma_mp_segment_fasta</td>
<td>File</td>
<td>MP (Matrix Protein) assembly fasta file</td>
</tr>
<tr>
<td>irma_na_segment_fasta</td>
<td>File</td>
<td>NA (Neuraminidase) assembly fasta file</td>
</tr>
<tr>
<td>irma_np_segment_fasta</td>
<td>File</td>
<td>NP (Nucleoprotein) assembly fasta file</td>
</tr>
<tr>
<td>irma_ns_segment_fasta</td>
<td>File</td>
<td>NS (Nonstructural) assembly fasta file</td>
</tr>
<tr>
<td>irma_pa_segment_fasta</td>
<td>File</td>
<td>PA (Polymerase acidic) assembly fasta file</td>
</tr>
<tr>
<td>irma_pb1_segment_fasta</td>
<td>File</td>
<td>PB1 (Polymerase basic 1) assembly fasta file</td>
</tr>
<tr>
<td>irma_pb2_segment_fasta</td>
<td>File</td>
<td>PB2 (Polymerase basic 2) assembly fasta file</td>
</tr>
<tr>
<td>irma_subtype</td>
<td>String</td>
<td>Flu subtype as determined by IRMA</td>
</tr>
<tr>
<td>irma_subtype_notes</td>
<td>String</td>
<td>Helpful note to user about Flu B subtypes. Output will be blank for Flu A samples. For Flu B samples it will state: "IRMA does not differentiate Victoria and Yamagata Flu B lineages. See abricate_flu_subtype output column"</td>
</tr>
<tr>
<td>irma_type</td>
<td>String</td>
<td>Flu type as determined by IRMA</td>
</tr>
<tr>
<td>irma_version</td>
<td>String</td>
<td>Version of IRMA used</td>
</tr>
<tr>
<td>ivar_tsv</td>
<td>File</td>
<td>Variant descriptor file generated by iVar variants</td>
</tr>
<tr>
<td>ivar_variant_proportion_intermediate</td>
<td>String</td>
<td>The proportion of variants of intermediate frequency</td>
</tr>
<tr>
<td>ivar_variant_version</td>
<td>String</td>
<td>Version of iVar for running the iVar variants command</td>
</tr>
<tr>
<td>ivar_vcf</td>
<td>File</td>
<td>iVar tsv output converted to VCF format</td>
</tr>
<tr>
<td>ivar_version_consensus</td>
<td>String</td>
<td>Version of iVar for running the iVar consensus command</td>
</tr>
<tr>
<td>ivar_version_primtrim</td>
<td>String</td>
<td>Version of iVar for running the iVar trim command</td>
</tr>
<tr>
<td>kraken_human</td>
<td>Float</td>
<td>Percent of human read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_human_dehosted</td>
<td>Float</td>
<td>Percent of human read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_report</td>
<td>File</td>
<td>Full Kraken report</td>
</tr>
<tr>
<td>kraken_report_dehosted</td>
<td>File</td>
<td>Full Kraken report after host removal</td>
</tr>
<tr>
<td>kraken_sc2</td>
<td>String</td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_sc2_dehosted</td>
<td>String</td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_target_organism</td>
<td>String</td>
<td>Percent of target organism read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_target_organism_dehosted</td>
<td>String</td>
<td>Percent of target organism read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_target_organism_name</td>
<td>String</td>
<td>The name of the target organism; e.g., "Monkeypox" or "Human immunodeficiency virus"</td>
</tr>
<tr>
<td>kraken_version</td>
<td>String</td>
<td>Version of Kraken software used</td>
</tr>
<tr>
<td>meanbaseq_trim</td>
<td>String</td>
<td>Mean quality of the nucleotide basecalls aligned to the reference genome after primer trimming</td>
</tr>
<tr>
<td>meanmapq_trim</td>
<td>String</td>
<td>Mean quality of the mapped reads to the reference genome after primer trimming</td>
</tr>
<tr>
<td>nextclade_aa_dels</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_h5n1</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_h5n1</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_ha</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_ha</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_na</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to Flu; it includes deletions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_na</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to Flu; it includes deletions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_h5n1</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_h5n1</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_ha</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_ha</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_na</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_na</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for NA segment</td>
</tr>
<tr>
<td>nextclade_clade</td>
<td>String</td>
<td>Nextclade clade designation, will be blank for Flu.</td>
</tr>
<tr>
<td>nextclade_clade_flu_h5n1</td>
<td>String</td>
<td>Nextclade clade designation, specific to Flu 5N1 whole genome. NOTE: Output will be blank or <code>NA</code> since this nextclade dataset does assign clades</td>
</tr>
<tr>
<td>nextclade_clade_flu_ha</td>
<td>String</td>
<td>Nextclade clade designation, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_clade_flu_na</td>
<td>String</td>
<td>Nextclade clade designation, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_docker</td>
<td>String</td>
<td>Docker image used to run Nextclade</td>
</tr>
<tr>
<td>nextclade_ds_tag</td>
<td>String</td>
<td>Dataset tag used to run Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_ds_tag_flu_ha</td>
<td>String</td>
<td>Dataset tag used to run Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_ds_tag_flu_na</td>
<td>String</td>
<td>Dataset tag used to run Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_json</td>
<td>File</td>
<td>Nextclade output in JSON file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_json_flu_h5n1</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_json_flu_ha</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_json_flu_na</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_lineage</td>
<td>String</td>
<td>Nextclade lineage designation</td>
</tr>
<tr>
<td>nextclade_qc</td>
<td>String</td>
<td>QC metric as determined by Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_qc_flu_h5n1</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_qc_flu_h5n1</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_qc_flu_ha</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_ha</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_na</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_na</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_tsv</td>
<td>File</td>
<td>Nextclade output in TSV file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_tsv_flu_h5n1</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_tsv_flu_ha</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_tsv_flu_na</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_version</td>
<td>String</td>
<td>The version of Nextclade software used</td>
</tr>
<tr>
<td>number_Degenerate</td>
<td>Int</td>
<td>Number of degenerate basecalls within the consensus assembly</td>
</tr>
<tr>
<td>number_N</td>
<td>Int</td>
<td>Number of fully ambiguous basecalls within the consensus assembly</td>
</tr>
<tr>
<td>number_Total</td>
<td>Int</td>
<td>Total number of nucleotides within the consensus assembly</td>
</tr>
<tr>
<td>pango_lineage</td>
<td>String</td>
<td>Pango lineage as determined by Pangolin</td>
</tr>
<tr>
<td>pango_lineage_expanded</td>
<td>String</td>
<td>Pango lineage without use of aliases; e.g., "BA.1" → "B.1.1.529.1"</td>
</tr>
<tr>
<td>pango_lineage_report</td>
<td>File</td>
<td>Full Pango lineage report generated by Pangolin</td>
</tr>
<tr>
<td>pangolin_assignment_version</td>
<td>String</td>
<td>The version of the pangolin software (e.g. PANGO or PUSHER) used for lineage assignment</td>
</tr>
<tr>
<td>pangolin_conflicts</td>
<td>String</td>
<td>Number of lineage conflicts as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_docker</td>
<td>String</td>
<td>Docker image used to run Pangolin</td>
</tr>
<tr>
<td>pangolin_notes</td>
<td>String</td>
<td>Lineage notes as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_versions</td>
<td>String</td>
<td>All Pangolin software and database versions</td>
</tr>
<tr>
<td>percent_reference_coverage</td>
<td>Float</td>
<td>Percent coverage of the reference genome after performing primer trimming; calculated as assembly_length_unambiguous / length of the reference genome (SC2: 29903) x 100</td>
</tr>
<tr>
<td>percentage_mapped_reads</td>
<td>String</td>
<td>Percentage of reads that successfully aligned to the reference genome. This value is calculated by number of mapped reads / total number of reads x 100.</td>
</tr>
<tr>
<td>primer_bed_name</td>
<td>String</td>
<td>Name of the primer bed files used for primer trimming</td>
</tr>
<tr>
<td>primer_trimmed_read_percent</td>
<td>Float</td>
<td>Percentage of read data with primers trimmed as determined by iVar trim</td>
</tr>
<tr>
<td>qc_check</td>
<td>String</td>
<td>A string that indicates whether or not the sample passes a set of pre-determined and user-provided QC thresholds</td>
</tr>
<tr>
<td>qc_standard</td>
<td>File</td>
<td>The file used in the QC Check task containing the QC thresholds.</td>
</tr>
<tr>
<td>quasitools_coverage_file</td>
<td>File</td>
<td>The coverage report created by Quasitools HyDRA</td>
</tr>
<tr>
<td>quasitools_date</td>
<td>String</td>
<td>Date of Quasitools analysis</td>
</tr>
<tr>
<td>quasitools_dr_report</td>
<td>File</td>
<td>Drug resistance report created by Quasitools HyDRA</td>
</tr>
<tr>
<td>quasitools_hydra_vcf</td>
<td>File</td>
<td>The VCF created by Quasitools HyDRA</td>
</tr>
<tr>
<td>quasitools_mutations_report</td>
<td>File</td>
<td>The mutation report created by Quasitools HyDRA</td>
</tr>
<tr>
<td>quasitools_version</td>
<td>String</td>
<td>Version of Quasitools used</td>
</tr>
<tr>
<td>read1_aligned</td>
<td>File</td>
<td>Forward read file of only aligned reads</td>
</tr>
<tr>
<td>read1_clean</td>
<td>File</td>
<td>Forward read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read1_dehosted</td>
<td>File</td>
<td>The dehosted forward reads file; suggested read file for SRA submission</td>
</tr>
<tr>
<td>read1_unaligned</td>
<td>File</td>
<td>Forward read file of unaligned reads</td>
</tr>
<tr>
<td>read2_aligned</td>
<td>File</td>
<td>Reverse read file of only aligned reads</td>
</tr>
<tr>
<td>read2_clean</td>
<td>File</td>
<td>Reverse read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read2_dehosted</td>
<td>File</td>
<td>The dehosted reverse reads file; suggested read file for SRA submission</td>
</tr>
<tr>
<td>read2_unaligned</td>
<td>File</td>
<td>Reverse read file of unaligned reads</td>
</tr>
<tr>
<td>read_screen_clean</td>
<td>String</td>
<td>PASS or FAIL result from clean read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_clean_tsv</td>
<td>File</td>
<td>Clean read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>read_screen_raw</td>
<td>String</td>
<td>PASS or FAIL result from raw read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_raw_tsv</td>
<td>File</td>
<td>Raw read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>samtools_version</td>
<td>String</td>
<td>The version of SAMtools used to sort and index the alignment file</td>
</tr>
<tr>
<td>samtools_version_consensus</td>
<td>String</td>
<td>The version of SAMtools used to create the pileup before running iVar consensus</td>
</tr>
<tr>
<td>samtools_version_primtrim</td>
<td>String</td>
<td>The version of SAMtools used to create the pileup before running iVar trim</td>
</tr>
<tr>
<td>samtools_version_stats</td>
<td>String</td>
<td>The version of SAMtools used to assess the quality of read mapping</td>
</tr>
<tr>
<td>sc2_s_gene_mean_coverage</td>
<td>Float</td>
<td>Mean read depth for the S gene in SARS-CoV-2</td>
</tr>
<tr>
<td>sc2_s_gene_percent_coverage</td>
<td>Float</td>
<td>Percent coverage of the S gene in SARS-CoV-2</td>
</tr>
<tr>
<td>seq_platform</td>
<td>String</td>
<td>Description of the sequencing methodology used to generate the input read data</td>
</tr>
<tr>
<td>sorted_bam_unaligned</td>
<td>File</td>
<td>A BAM file that only contains reads that did not align to the reference</td>
</tr>
<tr>
<td>sorted_bam_unaligned_bai</td>
<td>File</td>
<td>Index companion file to a BAM file that only contains reads that did not align to the reference</td>
</tr>
<tr>
<td>theiacov_illumina_pe_analysis_date</td>
<td>String</td>
<td>Date of analysis</td>
</tr>
<tr>
<td>theiacov_illumina_pe_version</td>
<td>String</td>
<td>Version of PHB used for running the workflow</td>
</tr>
<tr>
<td>trimmomatic_docker</td>
<td>String</td>
<td>The docker image used for the trimmomatic module in this workflow</td>
</tr>
<tr>
<td>trimmomatic_version</td>
<td>String</td>
<td>The version of Trimmomatic used</td>
</tr>
<tr>
<td>vadr_alerts_list</td>
<td>File</td>
<td>A file containing all of the fatal alerts as determined by VADR</td>
</tr>
<tr>
<td>vadr_all_outputs_tar_gz</td>
<td>File</td>
<td>A .tar.gz file (gzip-compressed tar archive file) containing all outputs from the VADR command v-annotate.pl. This file must be uncompressed &amp; extracted to see the many files within. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description of all files present within the archive. Useful when deeply investigating a sample's genome &amp; annotations.</td>
</tr>
<tr>
<td>vadr_classification_summary_file</td>
<td>File</td>
<td>Per-sequence tabular classification file. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#explanation-of-sqc-suffixed-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#explanation-of-sqc-suffixed-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_docker</td>
<td>String</td>
<td>Docker image used to run VADR</td>
</tr>
<tr>
<td>vadr_fastas_zip_archive</td>
<td>File</td>
<td>Zip archive containing all fasta files created during VADR analysis</td>
</tr>
<tr>
<td>vadr_feature_tbl_fail</td>
<td>File</td>
<td>5 column feature table output for failing sequences. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_feature_tbl_pass</td>
<td>File</td>
<td>5 column feature table output for passing sequences. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_num_alerts</td>
<td>String</td>
<td>Number of fatal alerts as determined by VADR</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>aligned_bai</td>
<td>File</td>
<td>Index companion file to the bam file generated during the consensus assembly process</td>
</tr>
<tr>
<td>aligned_bam</td>
<td>File</td>
<td>Sorted BAM file containing the alignments of reads to the reference genome</td>
</tr>
<tr>
<td>assembly_fasta</td>
<td>File</td>
<td>Consensus genome assembly; for lower quality flu samples, the output may state "Assembly could not be generated" when there is too little and/or too low quality data for IRMA to produce an assembly. Contigs will be ordered from largest to smallest when IRMA is used.</td>
</tr>
<tr>
<td>assembly_length_unambiguous</td>
<td>Int</td>
<td>Number of unambiguous basecalls within the consensus assembly</td>
</tr>
<tr>
<td>assembly_mean_coverage</td>
<td>Float</td>
<td>Mean sequencing depth throughout the consensus assembly. Generated after performing primer trimming and calculated using the SAMtools coverage command</td>
</tr>
<tr>
<td>assembly_method</td>
<td>String</td>
<td>Method employed to generate consensus assembly</td>
</tr>
<tr>
<td>auspice_json</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>bbduk_docker</td>
<td>String</td>
<td>The Docker image for bbduk, which was used to remove the adapters from the sequences</td>
</tr>
<tr>
<td>bwa_version</td>
<td>String</td>
<td>Version of BWA software used</td>
</tr>
<tr>
<td>consensus_flagstat</td>
<td>File</td>
<td>Output from the SAMtools flagstat command to assess quality of the alignment file (BAM)</td>
</tr>
<tr>
<td>consensus_n_variant_min_depth</td>
<td>Int</td>
<td>Minimum read depth to call variants for iVar consensus and iVar variants. Also represents the minimum consensus support threshold used by IRMA with Illumina Influenza data.</td>
</tr>
<tr>
<td>consensus_stats</td>
<td>File</td>
<td>Output from the SAMtools stats command to assess quality of the alignment file (BAM)</td>
</tr>
<tr>
<td>est_percent_gene_coverage_tsv</td>
<td>File</td>
<td>Percent coverage for each gene in the organism being analyzed (depending on the organism input)</td>
</tr>
<tr>
<td>fastp_html_report</td>
<td>File</td>
<td>The HTML report made with fastp</td>
</tr>
<tr>
<td>fastp_version</td>
<td>String</td>
<td>The version of fastp used</td>
</tr>
<tr>
<td>fastq_scan_clean1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean1</td>
<td>Int</td>
<td>The number of forward reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw1</td>
<td>Int</td>
<td>The number of input forward reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_raw1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_version</td>
<td>String</td>
<td>The version of fastq_scan</td>
</tr>
<tr>
<td>fastqc_clean1_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of clean forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_docker</td>
<td>String</td>
<td>The Docker container used for fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_clean1</td>
<td>Int</td>
<td>The number of forward reads after cleaning by fastqc</td>
</tr>
<tr>
<td>fastqc_num_reads_raw1</td>
<td>Int</td>
<td>The number of input forward reads by fastqc before cleaning</td>
</tr>
<tr>
<td>fastqc_raw1_html</td>
<td>File</td>
<td>An HTML file that provides a graphical visualization of raw forward read quality from fastqc to open in an internet browser</td>
</tr>
<tr>
<td>fastqc_version</td>
<td>String</td>
<td>Version of fastqc software used</td>
</tr>
<tr>
<td>ivar_tsv</td>
<td>File</td>
<td>Variant descriptor file generated by iVar variants</td>
</tr>
<tr>
<td>ivar_variant_proportion_intermediate</td>
<td>String</td>
<td>The proportion of variants of intermediate frequency</td>
</tr>
<tr>
<td>ivar_variant_version</td>
<td>String</td>
<td>Version of iVar for running the iVar variants command</td>
</tr>
<tr>
<td>ivar_vcf</td>
<td>File</td>
<td>iVar tsv output converted to VCF format</td>
</tr>
<tr>
<td>ivar_version_consensus</td>
<td>String</td>
<td>Version of iVar for running the iVar consensus command</td>
</tr>
<tr>
<td>ivar_version_primtrim</td>
<td>String</td>
<td>Version of iVar for running the iVar trim command</td>
</tr>
<tr>
<td>kraken_human</td>
<td>Float</td>
<td>Percent of human read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_human_dehosted</td>
<td>Float</td>
<td>Percent of human read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_report</td>
<td>File</td>
<td>Full Kraken report</td>
</tr>
<tr>
<td>kraken_report_dehosted</td>
<td>File</td>
<td>Full Kraken report after host removal</td>
</tr>
<tr>
<td>kraken_sc2</td>
<td>String</td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_sc2_dehosted</td>
<td>String</td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_target_organism</td>
<td>String</td>
<td>Percent of target organism read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_target_organism_dehosted</td>
<td>String</td>
<td>Percent of target organism read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_target_organism_name</td>
<td>String</td>
<td>The name of the target organism; e.g., "Monkeypox" or "Human immunodeficiency virus"</td>
</tr>
<tr>
<td>kraken_version</td>
<td>String</td>
<td>Version of Kraken software used</td>
</tr>
<tr>
<td>meanbaseq_trim</td>
<td>Float</td>
<td>Mean quality of the nucleotide basecalls aligned to the reference genome after primer trimming</td>
</tr>
<tr>
<td>meanmapq_trim</td>
<td>Float</td>
<td>Mean quality of the mapped reads to the reference genome after primer trimming</td>
</tr>
<tr>
<td>nextclade_aa_dels</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_aa_subs</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_clade</td>
<td>String</td>
<td>Nextclade clade designation, will be blank for Flu.</td>
</tr>
<tr>
<td>nextclade_docker</td>
<td>String</td>
<td>Docker image used to run Nextclade</td>
</tr>
<tr>
<td>nextclade_ds_tag</td>
<td>String</td>
<td>Dataset tag used to run Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_json</td>
<td>File</td>
<td>Nextclade output in JSON file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_lineage</td>
<td>String</td>
<td>Nextclade lineage designation</td>
</tr>
<tr>
<td>nextclade_qc</td>
<td>String</td>
<td>QC metric as determined by Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_tsv</td>
<td>File</td>
<td>Nextclade output in TSV file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_version</td>
<td>String</td>
<td>The version of Nextclade software used</td>
</tr>
<tr>
<td>number_Degenerate</td>
<td>Int</td>
<td>Number of degenerate basecalls within the consensus assembly</td>
</tr>
<tr>
<td>number_N</td>
<td>Int</td>
<td>Number of fully ambiguous basecalls within the consensus assembly</td>
</tr>
<tr>
<td>number_Total</td>
<td>Int</td>
<td>Total number of nucleotides within the consensus assembly</td>
</tr>
<tr>
<td>pango_lineage</td>
<td>String</td>
<td>Pango lineage as determined by Pangolin</td>
</tr>
<tr>
<td>pango_lineage_expanded</td>
<td>String</td>
<td>Pango lineage without use of aliases; e.g., "BA.1" → "B.1.1.529.1"</td>
</tr>
<tr>
<td>pango_lineage_report</td>
<td>File</td>
<td>Full Pango lineage report generated by Pangolin</td>
</tr>
<tr>
<td>pangolin_assignment_version</td>
<td>String</td>
<td>The version of the pangolin software (e.g. PANGO or PUSHER) used for lineage assignment</td>
</tr>
<tr>
<td>pangolin_conflicts</td>
<td>String</td>
<td>Number of lineage conflicts as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_docker</td>
<td>String</td>
<td>Docker image used to run Pangolin</td>
</tr>
<tr>
<td>pangolin_notes</td>
<td>String</td>
<td>Lineage notes as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_versions</td>
<td>String</td>
<td>All Pangolin software and database versions</td>
</tr>
<tr>
<td>percent_reference_coverage</td>
<td>Float</td>
<td>Percent coverage of the reference genome after performing primer trimming; calculated as assembly_length_unambiguous / length of the reference genome (SC2: 29903) x 100</td>
</tr>
<tr>
<td>percentage_mapped_reads</td>
<td>String</td>
<td>Percentage of reads that successfully aligned to the reference genome. This value is calculated by number of mapped reads / total number of reads x 100.</td>
</tr>
<tr>
<td>primer_bed_name</td>
<td>String</td>
<td>Name of the primer bed files used for primer trimming</td>
</tr>
<tr>
<td>primer_trimmed_read_percent</td>
<td>Float</td>
<td>Percentage of read data with primers trimmed as determined by iVar trim</td>
</tr>
<tr>
<td>qc_check</td>
<td>String</td>
<td>A string that indicates whether or not the sample passes a set of pre-determined and user-provided QC thresholds</td>
</tr>
<tr>
<td>qc_standard</td>
<td>File</td>
<td>The file used in the QC Check task containing the QC thresholds.</td>
</tr>
<tr>
<td>read1_aligned</td>
<td>File</td>
<td>Forward read file of only aligned reads</td>
</tr>
<tr>
<td>read1_clean</td>
<td>File</td>
<td>Forward read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read1_unaligned</td>
<td>File</td>
<td>Forward read file of unaligned reads</td>
</tr>
<tr>
<td>read_screen_clean</td>
<td>String</td>
<td>PASS or FAIL result from clean read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_clean_tsv</td>
<td>File</td>
<td>Clean read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>read_screen_raw</td>
<td>String</td>
<td>PASS or FAIL result from raw read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_raw_tsv</td>
<td>File</td>
<td>Raw read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>samtools_version</td>
<td>String</td>
<td>The version of SAMtools used to sort and index the alignment file</td>
</tr>
<tr>
<td>samtools_version_consensus</td>
<td>String</td>
<td>The version of SAMtools used to create the pileup before running iVar consensus</td>
</tr>
<tr>
<td>samtools_version_primtrim</td>
<td>String</td>
<td>The version of SAMtools used to create the pileup before running iVar trim</td>
</tr>
<tr>
<td>samtools_version_stats</td>
<td>String</td>
<td>The version of SAMtools used to assess the quality of read mapping</td>
</tr>
<tr>
<td>sc2_s_gene_mean_coverage</td>
<td>Float</td>
<td>Mean read depth for the S gene in SARS-CoV-2</td>
</tr>
<tr>
<td>sc2_s_gene_percent_coverage</td>
<td>Float</td>
<td>Percent coverage of the S gene in SARS-CoV-2</td>
</tr>
<tr>
<td>seq_platform</td>
<td>String</td>
<td>Description of the sequencing methodology used to generate the input read data</td>
</tr>
<tr>
<td>sorted_bam_unaligned</td>
<td>File</td>
<td>A BAM file that only contains reads that did not align to the reference</td>
</tr>
<tr>
<td>sorted_bam_unaligned_bai</td>
<td>File</td>
<td>Index companion file to a BAM file that only contains reads that did not align to the reference</td>
</tr>
<tr>
<td>theiacov_illumina_se_analysis_date</td>
<td>String</td>
<td>Date of analysis</td>
</tr>
<tr>
<td>theiacov_illumina_se_version</td>
<td>String</td>
<td>Version of PHB used for running the workflow</td>
</tr>
<tr>
<td>trimmomatic_docker</td>
<td>String</td>
<td>The docker image used for the trimmomatic module in this workflow</td>
</tr>
<tr>
<td>trimmomatic_version</td>
<td>String</td>
<td>The version of Trimmomatic used</td>
</tr>
<tr>
<td>vadr_alerts_list</td>
<td>File</td>
<td>A file containing all of the fatal alerts as determined by VADR</td>
</tr>
<tr>
<td>vadr_all_outputs_tar_gz</td>
<td>File</td>
<td>A .tar.gz file (gzip-compressed tar archive file) containing all outputs from the VADR command v-annotate.pl. This file must be uncompressed &amp; extracted to see the many files within. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description of all files present within the archive. Useful when deeply investigating a sample's genome &amp; annotations.</td>
</tr>
<tr>
<td>vadr_classification_summary_file</td>
<td>File</td>
<td>Per-sequence tabular classification file. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#explanation-of-sqc-suffixed-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#explanation-of-sqc-suffixed-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_docker</td>
<td>String</td>
<td>Docker image used to run VADR</td>
</tr>
<tr>
<td>vadr_fastas_zip_archive</td>
<td>File</td>
<td>Zip archive containing all fasta files created during VADR analysis</td>
</tr>
<tr>
<td>vadr_feature_tbl_fail</td>
<td>File</td>
<td>5 column feature table output for failing sequences. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_feature_tbl_pass</td>
<td>File</td>
<td>5 column feature table output for passing sequences. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_num_alerts</td>
<td>String</td>
<td>Number of fatal alerts as determined by VADR</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>abricate_flu_database</td>
<td>String</td>
<td>ABRicate database used for analysis</td>
</tr>
<tr>
<td>abricate_flu_results</td>
<td>File</td>
<td>File containing all results from ABRicate</td>
</tr>
<tr>
<td>abricate_flu_subtype</td>
<td>String</td>
<td>Flu subtype as determined by ABRicate</td>
</tr>
<tr>
<td>abricate_flu_type</td>
<td>String</td>
<td>Flu type as determined by ABRicate</td>
</tr>
<tr>
<td>abricate_flu_version</td>
<td>String</td>
<td>Version of ABRicate</td>
</tr>
<tr>
<td>aligned_bai</td>
<td>File</td>
<td>Index companion file to the bam file generated during the consensus assembly process</td>
</tr>
<tr>
<td>aligned_bam</td>
<td>File</td>
<td>Sorted BAM file containing the alignments of reads to the reference genome</td>
</tr>
<tr>
<td>artic_docker</td>
<td>String</td>
<td>Docker image utilized for read trimming and consensus genome assembly</td>
</tr>
<tr>
<td>artic_version</td>
<td>String</td>
<td>Version of the Artic software utilized for read trimming and conesnsus genome assembly</td>
</tr>
<tr>
<td>assembly_fasta</td>
<td>String</td>
<td>Consensus genome assembly; for lower quality flu samples, the output may state "Assembly could not be generated" when there is too little and/or too low quality data for IRMA to produce an assembly. Contigs will be ordered from largest to smallest when IRMA is used.</td>
</tr>
<tr>
<td>assembly_length_unambiguous</td>
<td>Int</td>
<td>Number of unambiguous basecalls within the consensus assembly</td>
</tr>
<tr>
<td>assembly_mean_coverage</td>
<td>String</td>
<td>Mean sequencing depth throughout the consensus assembly. Generated after performing primer trimming and calculated using the SAMtools coverage command</td>
</tr>
<tr>
<td>assembly_method</td>
<td>String</td>
<td>Method employed to generate consensus assembly</td>
</tr>
<tr>
<td>auspice_json</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_h5n1</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza H5N1 whole genome that includes the samples included in the "avian-flu/h5n1-cattle-outbreak" nextstrain build that is focused on B3.13 genotype and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_ha</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza HA segment that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_na</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza NA segment that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>consensus_flagstat</td>
<td>File</td>
<td>Output from the SAMtools flagstat command to assess quality of the alignment file (BAM)</td>
</tr>
<tr>
<td>consensus_stats</td>
<td>File</td>
<td>Output from the SAMtools stats command to assess quality of the alignment file (BAM)</td>
</tr>
<tr>
<td>est_coverage_clean</td>
<td>Float</td>
<td>Estimated coverage calculated from clean reads and genome length</td>
</tr>
<tr>
<td>est_coverage_raw</td>
<td>Float</td>
<td>Estimated coverage calculated from raw reads and genome length</td>
</tr>
<tr>
<td>est_percent_gene_coverage_tsv</td>
<td>File</td>
<td>Percent coverage for each gene in the organism being analyzed (depending on the organism input)</td>
</tr>
<tr>
<td>flu_A_315675_resistance</td>
<td>String</td>
<td>resistance mutations to A_315675</td>
</tr>
<tr>
<td>flu_L_742_001_resistance</td>
<td>String</td>
<td>resistance mutations to L_742_001</td>
</tr>
<tr>
<td>flu_amantadine_resistance</td>
<td>String</td>
<td>resistance mutations to amantadine</td>
</tr>
<tr>
<td>flu_compound_367_resistance</td>
<td>String</td>
<td>resistance mutations to compound_367</td>
</tr>
<tr>
<td>flu_favipiravir_resistance</td>
<td>String</td>
<td>resistance mutations to favipiravir</td>
</tr>
<tr>
<td>flu_fludase_resistance</td>
<td>String</td>
<td>resistance mutations to fludase</td>
</tr>
<tr>
<td>flu_laninamivir_resistance</td>
<td>String</td>
<td>resistance mutations to laninamivir</td>
</tr>
<tr>
<td>flu_oseltamivir_resistance</td>
<td>String</td>
<td>resistance mutations to oseltamivir (Tamiflu®)</td>
</tr>
<tr>
<td>flu_peramivir_resistance</td>
<td>String</td>
<td>resistance mutations to peramivir (Rapivab®)</td>
</tr>
<tr>
<td>flu_pimodivir_resistance</td>
<td>String</td>
<td>resistance mutations to pimodivir</td>
</tr>
<tr>
<td>flu_rimantadine_resistance</td>
<td>String</td>
<td>resistance mutations to rimantadine</td>
</tr>
<tr>
<td>flu_xofluza_resistance</td>
<td>String</td>
<td>resistance mutations to xofluza (Baloxavir marboxil)</td>
</tr>
<tr>
<td>flu_zanamivir_resistance</td>
<td>String</td>
<td>resistance mutations to zanamivir (Relenza®)</td>
</tr>
<tr>
<td>genoflu_all_segments</td>
<td>String</td>
<td>The genotypes for each individual flu segment</td>
</tr>
<tr>
<td>genoflu_genotype</td>
<td>String</td>
<td>The genotype of the whole genome, based off of the individual segments types</td>
</tr>
<tr>
<td>genoflu_output_tsv</td>
<td>File</td>
<td>The output file from GenoFLU</td>
</tr>
<tr>
<td>genoflu_version</td>
<td>String</td>
<td>The version of GenoFLU used</td>
</tr>
<tr>
<td>irma_assembly_fasta_concatenated</td>
<td>File</td>
<td>Assembly FASTA file of all Influenza genome segments concatenated into one sequence/FASTA entry</td>
</tr>
<tr>
<td>irma_docker</td>
<td>String</td>
<td>Docker image used to run IRMA</td>
</tr>
<tr>
<td>irma_ha_segment_fasta</td>
<td>File</td>
<td>HA (Haemagglutinin) assembly fasta file</td>
</tr>
<tr>
<td>irma_min_consensus_support_threshold</td>
<td>Int</td>
<td>Minimum consensus support threshold used by IRMA with ONT data. For illumina data, see output called <code>consensus_n_variant_min_depth</code> for this value</td>
</tr>
<tr>
<td>irma_mp_segment_fasta</td>
<td>File</td>
<td>MP (Matrix Protein) assembly fasta file</td>
</tr>
<tr>
<td>irma_na_segment_fasta</td>
<td>File</td>
<td>NA (Neuraminidase) assembly fasta file</td>
</tr>
<tr>
<td>irma_np_segment_fasta</td>
<td>File</td>
<td>NP (Nucleoprotein) assembly fasta file</td>
</tr>
<tr>
<td>irma_ns_segment_fasta</td>
<td>File</td>
<td>NS (Nonstructural) assembly fasta file</td>
</tr>
<tr>
<td>irma_pa_segment_fasta</td>
<td>File</td>
<td>PA (Polymerase acidic) assembly fasta file</td>
</tr>
<tr>
<td>irma_pb1_segment_fasta</td>
<td>File</td>
<td>PB1 (Polymerase basic 1) assembly fasta file</td>
</tr>
<tr>
<td>irma_pb2_segment_fasta</td>
<td>File</td>
<td>PB2 (Polymerase basic 2) assembly fasta file</td>
</tr>
<tr>
<td>irma_subtype</td>
<td>String</td>
<td>Flu subtype as determined by IRMA</td>
</tr>
<tr>
<td>irma_subtype_notes</td>
<td>String</td>
<td>Helpful note to user about Flu B subtypes. Output will be blank for Flu A samples. For Flu B samples it will state: "IRMA does not differentiate Victoria and Yamagata Flu B lineages. See abricate_flu_subtype output column"</td>
</tr>
<tr>
<td>irma_type</td>
<td>String</td>
<td>Flu type as determined by IRMA</td>
</tr>
<tr>
<td>irma_version</td>
<td>String</td>
<td>Version of IRMA used</td>
</tr>
<tr>
<td>kraken_human</td>
<td>Float</td>
<td>Percent of human read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_human_dehosted</td>
<td>Float</td>
<td>Percent of human read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_report</td>
<td>File</td>
<td>Full Kraken report</td>
</tr>
<tr>
<td>kraken_report_dehosted</td>
<td>File</td>
<td>Full Kraken report after host removal</td>
</tr>
<tr>
<td>kraken_sc2</td>
<td>String</td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_sc2_dehosted</td>
<td>String</td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_target_organism</td>
<td>String</td>
<td>Percent of target organism read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_target_organism_dehosted</td>
<td>String</td>
<td>Percent of target organism read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_target_organism_name</td>
<td>String</td>
<td>The name of the target organism; e.g., "Monkeypox" or "Human immunodeficiency virus"</td>
</tr>
<tr>
<td>kraken_version</td>
<td>String</td>
<td>Version of Kraken software used</td>
</tr>
<tr>
<td>meanbaseq_trim</td>
<td>Float</td>
<td>Mean quality of the nucleotide basecalls aligned to the reference genome after primer trimming</td>
</tr>
<tr>
<td>meanmapq_trim</td>
<td>Float</td>
<td>Mean quality of the mapped reads to the reference genome after primer trimming</td>
</tr>
<tr>
<td>medaka_reference</td>
<td>String</td>
<td>Reference sequence used in medaka task</td>
</tr>
<tr>
<td>medaka_vcf</td>
<td>File</td>
<td>A VCF file containing the identified variants</td>
</tr>
<tr>
<td>nanoplot_docker</td>
<td>String</td>
<td>Docker image for nanoplot</td>
</tr>
<tr>
<td>nanoplot_html_clean</td>
<td>File</td>
<td>An HTML report describing the clean reads</td>
</tr>
<tr>
<td>nanoplot_html_raw</td>
<td>File</td>
<td>An HTML report describing the raw reads</td>
</tr>
<tr>
<td>nanoplot_num_reads_clean1</td>
<td>Int</td>
<td>Number of clean reads</td>
</tr>
<tr>
<td>nanoplot_num_reads_raw1</td>
<td>Int</td>
<td>Number of raw reads</td>
</tr>
<tr>
<td>nanoplot_r1_est_coverage_clean</td>
<td>Float</td>
<td>Estimated coverage on the clean reads by nanoplot</td>
</tr>
<tr>
<td>nanoplot_r1_est_coverage_raw</td>
<td>Float</td>
<td>Estimated coverage on the raw reads by nanoplot</td>
</tr>
<tr>
<td>nanoplot_r1_mean_q_clean</td>
<td>Float</td>
<td>Mean quality score of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_q_raw</td>
<td>Float</td>
<td>Mean quality score of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_readlength_clean</td>
<td>Float</td>
<td>Mean read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_mean_readlength_raw</td>
<td>Float</td>
<td>Mean read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_q_clean</td>
<td>Float</td>
<td>Median quality score of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_q_raw</td>
<td>Float</td>
<td>Median quality score of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_readlength_clean</td>
<td>Float</td>
<td>Median read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_median_readlength_raw</td>
<td>Float</td>
<td>Median read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_n50_clean</td>
<td>Float</td>
<td>N50 of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_n50_raw</td>
<td>Float</td>
<td>N50 of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_stdev_readlength_clean</td>
<td>Float</td>
<td>Standard deviation read length of clean forward reads</td>
</tr>
<tr>
<td>nanoplot_r1_stdev_readlength_raw</td>
<td>Float</td>
<td>Standard deviation read length of raw forward reads</td>
</tr>
<tr>
<td>nanoplot_tsv_clean</td>
<td>File</td>
<td>A TSV report describing the clean reads</td>
</tr>
<tr>
<td>nanoplot_tsv_raw</td>
<td>File</td>
<td>A TSV report describing the raw reads</td>
</tr>
<tr>
<td>nanoplot_version</td>
<td>String</td>
<td>Version of nanoplot used for analysis</td>
</tr>
<tr>
<td>nextclade_aa_dels</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_h5n1</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_h5n1</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_ha</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_ha</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_na</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to Flu; it includes deletions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_na</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to Flu; it includes deletions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_h5n1</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_h5n1</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_ha</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_ha</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_na</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_na</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for NA segment</td>
</tr>
<tr>
<td>nextclade_clade</td>
<td>String</td>
<td>Nextclade clade designation, will be blank for Flu.</td>
</tr>
<tr>
<td>nextclade_clade_flu_h5n1</td>
<td>String</td>
<td>Nextclade clade designation, specific to Flu 5N1 whole genome. NOTE: Output will be blank or <code>NA</code> since this nextclade dataset does assign clades</td>
</tr>
<tr>
<td>nextclade_clade_flu_ha</td>
<td>String</td>
<td>Nextclade clade designation, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_clade_flu_na</td>
<td>String</td>
<td>Nextclade clade designation, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_docker</td>
<td>String</td>
<td>Docker image used to run Nextclade</td>
</tr>
<tr>
<td>nextclade_ds_tag</td>
<td>String</td>
<td>Dataset tag used to run Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_ds_tag_flu_ha</td>
<td>String</td>
<td>Dataset tag used to run Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_ds_tag_flu_na</td>
<td>String</td>
<td>Dataset tag used to run Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_json</td>
<td>File</td>
<td>Nextclade output in JSON file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_json_flu_h5n1</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_json_flu_ha</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_json_flu_na</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_lineage</td>
<td>String</td>
<td>Nextclade lineage designation</td>
</tr>
<tr>
<td>nextclade_qc</td>
<td>String</td>
<td>QC metric as determined by Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_qc_flu_h5n1</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_qc_flu_h5n1</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_qc_flu_ha</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_ha</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_na</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_na</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_tsv</td>
<td>File</td>
<td>Nextclade output in TSV file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_tsv_flu_h5n1</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_tsv_flu_ha</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_tsv_flu_na</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_version</td>
<td>String</td>
<td>The version of Nextclade software used</td>
</tr>
<tr>
<td>number_Degenerate</td>
<td>Int</td>
<td>Number of degenerate basecalls within the consensus assembly</td>
</tr>
<tr>
<td>number_N</td>
<td>Int</td>
<td>Number of fully ambiguous basecalls within the consensus assembly</td>
</tr>
<tr>
<td>number_Total</td>
<td>Int</td>
<td>Total number of nucleotides within the consensus assembly</td>
</tr>
<tr>
<td>pango_lineage</td>
<td>String</td>
<td>Pango lineage as determined by Pangolin</td>
</tr>
<tr>
<td>pango_lineage_expanded</td>
<td>String</td>
<td>Pango lineage without use of aliases; e.g., "BA.1" → "B.1.1.529.1"</td>
</tr>
<tr>
<td>pango_lineage_report</td>
<td>File</td>
<td>Full Pango lineage report generated by Pangolin</td>
</tr>
<tr>
<td>pangolin_assignment_version</td>
<td>String</td>
<td>The version of the pangolin software (e.g. PANGO or PUSHER) used for lineage assignment</td>
</tr>
<tr>
<td>pangolin_conflicts</td>
<td>String</td>
<td>Number of lineage conflicts as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_docker</td>
<td>String</td>
<td>Docker image used to run Pangolin</td>
</tr>
<tr>
<td>pangolin_notes</td>
<td>String</td>
<td>Lineage notes as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_versions</td>
<td>String</td>
<td>All Pangolin software and database versions</td>
</tr>
<tr>
<td>percent_reference_coverage</td>
<td>Float</td>
<td>Percent coverage of the reference genome after performing primer trimming; calculated as assembly_length_unambiguous / length of the reference genome (SC2: 29903) x 100</td>
</tr>
<tr>
<td>percentage_mapped_reads</td>
<td>String</td>
<td>Percentage of reads that successfully aligned to the reference genome. This value is calculated by number of mapped reads / total number of reads x 100.</td>
</tr>
<tr>
<td>primer_bed_name</td>
<td>String</td>
<td>Name of the primer bed files used for primer trimming</td>
</tr>
<tr>
<td>qc_check</td>
<td>String</td>
<td>A string that indicates whether or not the sample passes a set of pre-determined and user-provided QC thresholds</td>
</tr>
<tr>
<td>qc_standard</td>
<td>File</td>
<td>The file used in the QC Check task containing the QC thresholds.</td>
</tr>
<tr>
<td>quasitools_coverage_file</td>
<td>File</td>
<td>The coverage report created by Quasitools HyDRA</td>
</tr>
<tr>
<td>quasitools_date</td>
<td>String</td>
<td>Date of Quasitools analysis</td>
</tr>
<tr>
<td>quasitools_dr_report</td>
<td>File</td>
<td>Drug resistance report created by Quasitools HyDRA</td>
</tr>
<tr>
<td>quasitools_hydra_vcf</td>
<td>File</td>
<td>The VCF created by Quasitools HyDRA</td>
</tr>
<tr>
<td>quasitools_mutations_report</td>
<td>File</td>
<td>The mutation report created by Quasitools HyDRA</td>
</tr>
<tr>
<td>quasitools_version</td>
<td>String</td>
<td>Version of Quasitools used</td>
</tr>
<tr>
<td>read1_aligned</td>
<td>File</td>
<td>Forward read file of only aligned reads</td>
</tr>
<tr>
<td>read1_dehosted</td>
<td>File</td>
<td>The dehosted forward reads file; suggested read file for SRA submission</td>
</tr>
<tr>
<td>read1_trimmed</td>
<td>File</td>
<td>Forward read file after quality trimming and adapter removal</td>
</tr>
<tr>
<td>read_screen_clean</td>
<td>String</td>
<td>PASS or FAIL result from clean read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_clean_tsv</td>
<td>File</td>
<td>Clean read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>read_screen_raw</td>
<td>String</td>
<td>PASS or FAIL result from raw read screening; FAIL accompanied by the reason(s) for failure</td>
</tr>
<tr>
<td>read_screen_raw_tsv</td>
<td>File</td>
<td>Raw read screening report TSV depicting read counts, total read base pairs, and estimated genome length</td>
</tr>
<tr>
<td>samtools_version</td>
<td>String</td>
<td>The version of SAMtools used to sort and index the alignment file</td>
</tr>
<tr>
<td>sc2_s_gene_mean_coverage</td>
<td>Float</td>
<td>Mean read depth for the S gene in SARS-CoV-2</td>
</tr>
<tr>
<td>sc2_s_gene_percent_coverage</td>
<td>Float</td>
<td>Percent coverage of the S gene in SARS-CoV-2</td>
</tr>
<tr>
<td>seq_platform</td>
<td>String</td>
<td>Description of the sequencing methodology used to generate the input read data</td>
</tr>
<tr>
<td>theiacov_ont_analysis_date</td>
<td>String</td>
<td>Date of analysis</td>
</tr>
<tr>
<td>theiacov_ont_version</td>
<td>String</td>
<td>Version of PHB used for running the workflow</td>
</tr>
<tr>
<td>vadr_alerts_list</td>
<td>File</td>
<td>A file containing all of the fatal alerts as determined by VADR</td>
</tr>
<tr>
<td>vadr_all_outputs_tar_gz</td>
<td>File</td>
<td>A .tar.gz file (gzip-compressed tar archive file) containing all outputs from the VADR command v-annotate.pl. This file must be uncompressed &amp; extracted to see the many files within. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description of all files present within the archive. Useful when deeply investigating a sample's genome &amp; annotations.</td>
</tr>
<tr>
<td>vadr_classification_summary_file</td>
<td>File</td>
<td>Per-sequence tabular classification file. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#explanation-of-sqc-suffixed-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#explanation-of-sqc-suffixed-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_docker</td>
<td>String</td>
<td>Docker image used to run VADR</td>
</tr>
<tr>
<td>vadr_fastas_zip_archive</td>
<td>File</td>
<td>Zip archive containing all fasta files created during VADR analysis</td>
</tr>
<tr>
<td>vadr_feature_tbl_fail</td>
<td>File</td>
<td>5 column feature table output for failing sequences. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_feature_tbl_pass</td>
<td>File</td>
<td>5 column feature table output for passing sequences. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_num_alerts</td>
<td>String</td>
<td>Number of fatal alerts as determined by VADR</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>abricate_flu_database</td>
<td>String</td>
<td>ABRicate database used for analysis</td>
</tr>
<tr>
<td>abricate_flu_results</td>
<td>File</td>
<td>File containing all results from ABRicate</td>
</tr>
<tr>
<td>abricate_flu_subtype</td>
<td>String</td>
<td>Flu subtype as determined by ABRicate</td>
</tr>
<tr>
<td>abricate_flu_type</td>
<td>String</td>
<td>Flu type as determined by ABRicate</td>
</tr>
<tr>
<td>abricate_flu_version</td>
<td>String</td>
<td>Version of ABRicate</td>
</tr>
<tr>
<td>assembly_length_unambiguous</td>
<td>Int</td>
<td>Number of unambiguous basecalls within the consensus assembly</td>
</tr>
<tr>
<td>assembly_method</td>
<td>String</td>
<td>Method employed to generate consensus assembly</td>
</tr>
<tr>
<td>auspice_json</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_h5n1</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza H5N1 whole genome that includes the samples included in the "avian-flu/h5n1-cattle-outbreak" nextstrain build that is focused on B3.13 genotype and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_ha</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza HA segment that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>auspice_json_flu_na</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis on Influenza NA segment that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>flu_A_315675_resistance</td>
<td>String</td>
<td>resistance mutations to A_315675</td>
</tr>
<tr>
<td>flu_L_742_001_resistance</td>
<td>String</td>
<td>resistance mutations to L_742_001</td>
</tr>
<tr>
<td>flu_amantadine_resistance</td>
<td>String</td>
<td>resistance mutations to amantadine</td>
</tr>
<tr>
<td>flu_compound_367_resistance</td>
<td>String</td>
<td>resistance mutations to compound_367</td>
</tr>
<tr>
<td>flu_favipiravir_resistance</td>
<td>String</td>
<td>resistance mutations to favipiravir</td>
</tr>
<tr>
<td>flu_fludase_resistance</td>
<td>String</td>
<td>resistance mutations to fludase</td>
</tr>
<tr>
<td>flu_laninamivir_resistance</td>
<td>String</td>
<td>resistance mutations to laninamivir</td>
</tr>
<tr>
<td>flu_oseltamivir_resistance</td>
<td>String</td>
<td>resistance mutations to oseltamivir (Tamiflu®)</td>
</tr>
<tr>
<td>flu_peramivir_resistance</td>
<td>String</td>
<td>resistance mutations to peramivir (Rapivab®)</td>
</tr>
<tr>
<td>flu_pimodivir_resistance</td>
<td>String</td>
<td>resistance mutations to pimodivir</td>
</tr>
<tr>
<td>flu_rimantadine_resistance</td>
<td>String</td>
<td>resistance mutations to rimantadine</td>
</tr>
<tr>
<td>flu_xofluza_resistance</td>
<td>String</td>
<td>resistance mutations to xofluza (Baloxavir marboxil)</td>
</tr>
<tr>
<td>flu_zanamivir_resistance</td>
<td>String</td>
<td>resistance mutations to zanamivir (Relenza®)</td>
</tr>
<tr>
<td>genoflu_all_segments</td>
<td>String</td>
<td>The genotypes for each individual flu segment</td>
</tr>
<tr>
<td>genoflu_genotype</td>
<td>String</td>
<td>The genotype of the whole genome, based off of the individual segments types</td>
</tr>
<tr>
<td>genoflu_output_tsv</td>
<td>File</td>
<td>The output file from GenoFLU</td>
</tr>
<tr>
<td>genoflu_version</td>
<td>String</td>
<td>The version of GenoFLU used</td>
</tr>
<tr>
<td>nextclade_aa_dels</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_h5n1</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_h5n1</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_ha</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_ha</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to flu; it includes deletions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_na</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to Flu; it includes deletions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_dels_flu_na</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Specific to Flu; it includes deletions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_h5n1</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_h5n1</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_ha</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_ha</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for HA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_na</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for NA segment</td>
</tr>
<tr>
<td>nextclade_aa_subs_flu_na</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Specific to Flu; it includes substitutions for NA segment</td>
</tr>
<tr>
<td>nextclade_clade</td>
<td>String</td>
<td>Nextclade clade designation, will be blank for Flu.</td>
</tr>
<tr>
<td>nextclade_clade_flu_h5n1</td>
<td>String</td>
<td>Nextclade clade designation, specific to Flu 5N1 whole genome. NOTE: Output will be blank or <code>NA</code> since this nextclade dataset does assign clades</td>
</tr>
<tr>
<td>nextclade_clade_flu_ha</td>
<td>String</td>
<td>Nextclade clade designation, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_clade_flu_na</td>
<td>String</td>
<td>Nextclade clade designation, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_docker</td>
<td>String</td>
<td>Docker image used to run Nextclade</td>
</tr>
<tr>
<td>nextclade_ds_tag</td>
<td>String</td>
<td>Dataset tag used to run Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_ds_tag_flu_ha</td>
<td>String</td>
<td>Dataset tag used to run Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_ds_tag_flu_na</td>
<td>String</td>
<td>Dataset tag used to run Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_json</td>
<td>File</td>
<td>Nextclade output in JSON file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_json_flu_h5n1</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_json_flu_ha</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_json_flu_na</td>
<td>File</td>
<td>Nextclade output in JSON file format, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_lineage</td>
<td>String</td>
<td>Nextclade lineage designation</td>
</tr>
<tr>
<td>nextclade_qc</td>
<td>String</td>
<td>QC metric as determined by Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_qc_flu_h5n1</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_qc_flu_h5n1</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_qc_flu_ha</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_ha</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_na</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_qc_flu_na</td>
<td>String</td>
<td>QC metric as determined by Nextclade, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_tsv</td>
<td>File</td>
<td>Nextclade output in TSV file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_tsv_flu_h5n1</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu H5N1 whole genome</td>
</tr>
<tr>
<td>nextclade_tsv_flu_ha</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu HA segment</td>
</tr>
<tr>
<td>nextclade_tsv_flu_na</td>
<td>File</td>
<td>Nextclade output in TSV file format, specific to Flu NA segment</td>
</tr>
<tr>
<td>nextclade_version</td>
<td>String</td>
<td>The version of Nextclade software used</td>
</tr>
<tr>
<td>number_Degenerate</td>
<td>Int</td>
<td>Number of degenerate basecalls within the consensus assembly</td>
</tr>
<tr>
<td>number_N</td>
<td>Int</td>
<td>Number of fully ambiguous basecalls within the consensus assembly</td>
</tr>
<tr>
<td>number_Total</td>
<td>Int</td>
<td>Total number of nucleotides within the consensus assembly</td>
</tr>
<tr>
<td>pango_lineage</td>
<td>String</td>
<td>Pango lineage as determined by Pangolin</td>
</tr>
<tr>
<td>pango_lineage_expanded</td>
<td>String</td>
<td>Pango lineage without use of aliases; e.g., "BA.1" → "B.1.1.529.1"</td>
</tr>
<tr>
<td>pango_lineage_report</td>
<td>File</td>
<td>Full Pango lineage report generated by Pangolin</td>
</tr>
<tr>
<td>pangolin_assignment_version</td>
<td>String</td>
<td>The version of the pangolin software (e.g. PANGO or PUSHER) used for lineage assignment</td>
</tr>
<tr>
<td>pangolin_conflicts</td>
<td>String</td>
<td>Number of lineage conflicts as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_docker</td>
<td>String</td>
<td>Docker image used to run Pangolin</td>
</tr>
<tr>
<td>pangolin_notes</td>
<td>String</td>
<td>Lineage notes as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_versions</td>
<td>String</td>
<td>All Pangolin software and database versions</td>
</tr>
<tr>
<td>percent_reference_coverage</td>
<td>Float</td>
<td>Percent coverage of the reference genome after performing primer trimming; calculated as assembly_length_unambiguous / length of the reference genome (SC2: 29903) x 100</td>
</tr>
<tr>
<td>qc_check</td>
<td>String</td>
<td>A string that indicates whether or not the sample passes a set of pre-determined and user-provided QC thresholds</td>
</tr>
<tr>
<td>qc_standard</td>
<td>File</td>
<td>The file used in the QC Check task containing the QC thresholds.</td>
</tr>
<tr>
<td>seq_platform</td>
<td>String</td>
<td>Description of the sequencing methodology used to generate the input read data</td>
</tr>
<tr>
<td>theiacov_fasta_analysis_date</td>
<td>String</td>
<td>Date of analysis</td>
</tr>
<tr>
<td>theiacov_fasta_version</td>
<td>String</td>
<td>Version of PHB used for running the workflow</td>
</tr>
<tr>
<td>vadr_alerts_list</td>
<td>File</td>
<td>A file containing all of the fatal alerts as determined by VADR</td>
</tr>
<tr>
<td>vadr_all_outputs_tar_gz</td>
<td>File</td>
<td>A .tar.gz file (gzip-compressed tar archive file) containing all outputs from the VADR command v-annotate.pl. This file must be uncompressed &amp; extracted to see the many files within. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description of all files present within the archive. Useful when deeply investigating a sample's genome &amp; annotations.</td>
</tr>
<tr>
<td>vadr_classification_summary_file</td>
<td>File</td>
<td>Per-sequence tabular classification file. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#explanation-of-sqc-suffixed-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#explanation-of-sqc-suffixed-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_docker</td>
<td>String</td>
<td>Docker image used to run VADR</td>
</tr>
<tr>
<td>vadr_fastas_zip_archive</td>
<td>File</td>
<td>Zip archive containing all fasta files created during VADR analysis</td>
</tr>
<tr>
<td>vadr_feature_tbl_fail</td>
<td>File</td>
<td>5 column feature table output for failing sequences. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_feature_tbl_pass</td>
<td>File</td>
<td>5 column feature table output for passing sequences. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_flu_ha_segment_fasta</td>
<td>File</td>
<td>HA (Haemagglutinin) assembly fasta file</td>
</tr>
<tr>
<td>vadr_flu_mp_segment_fasta</td>
<td>File</td>
<td>MP (Matrix Protein) assembly fasta file</td>
</tr>
<tr>
<td>vadr_flu_na_segment_fasta</td>
<td>File</td>
<td>NA (Neuraminidase) assembly fasta file</td>
</tr>
<tr>
<td>vadr_flu_np_segment_fasta</td>
<td>File</td>
<td>NP (Nucleoprotein) assembly fasta file</td>
</tr>
<tr>
<td>vadr_flu_ns_segment_fasta</td>
<td>File</td>
<td>NS (Nonstructural) assembly fasta file</td>
</tr>
<tr>
<td>vadr_flu_pa_segment_fasta</td>
<td>File</td>
<td>PA (Polymerase acidic) assembly fasta file</td>
</tr>
<tr>
<td>vadr_flu_pb1_segment_fasta</td>
<td>File</td>
<td>PB1 (Polymerase basic 1) assembly fasta file</td>
</tr>
<tr>
<td>vadr_flu_pb2_segment_fasta</td>
<td>File</td>
<td>PB2 (Polymerase basic 2) assembly fasta file</td>
</tr>
<tr>
<td>vadr_flu_segment_concatenated_fasta</td>
<td>File</td>
<td>Assembly FASTA file of all Influenza genome segments concatenated into one sequence/FASTA entry</td>
</tr>
<tr>
<td>vadr_num_alerts</td>
<td>String</td>
<td>Number of fatal alerts as determined by VADR</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>aligned_bai</td>
<td>File</td>
<td>Index companion file to the bam file generated during the consensus assembly process</td>
</tr>
<tr>
<td>aligned_bam</td>
<td>File</td>
<td>Sorted BAM file containing the alignments of reads to the reference genome</td>
</tr>
<tr>
<td>artic_docker</td>
<td>String</td>
<td>Docker image utilized for read trimming and consensus genome assembly</td>
</tr>
<tr>
<td>artic_version</td>
<td>String</td>
<td>Version of the Artic software utilized for read trimming and conesnsus genome assembly</td>
</tr>
<tr>
<td>assembly_fasta</td>
<td>File</td>
<td>Consensus genome assembly; for lower quality flu samples, the output may state "Assembly could not be generated" when there is too little and/or too low quality data for IRMA to produce an assembly. Contigs will be ordered from largest to smallest when IRMA is used.</td>
</tr>
<tr>
<td>assembly_length_unambiguous</td>
<td>Int</td>
<td>Number of unambiguous basecalls within the consensus assembly</td>
</tr>
<tr>
<td>assembly_mean_coverage</td>
<td>Float</td>
<td>Mean sequencing depth throughout the consensus assembly. Generated after performing primer trimming and calculated using the SAMtools coverage command</td>
</tr>
<tr>
<td>assembly_method</td>
<td>String</td>
<td>Method employed to generate consensus assembly</td>
</tr>
<tr>
<td>auspice_json</td>
<td>File</td>
<td>Auspice-compatable JSON output generated from Nextclade analysis that includes the Nextclade default samples for clade-typing and the single sample placed on this tree</td>
</tr>
<tr>
<td>consensus_flagstat</td>
<td>File</td>
<td>Output from the SAMtools flagstat command to assess quality of the alignment file (BAM)</td>
</tr>
<tr>
<td>consensus_stats</td>
<td>File</td>
<td>Output from the SAMtools stats command to assess quality of the alignment file (BAM)</td>
</tr>
<tr>
<td>est_percent_gene_coverage_tsv</td>
<td>File</td>
<td>Percent coverage for each gene in the organism being analyzed (depending on the organism input)</td>
</tr>
<tr>
<td>fastq_scan_clean1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about clean forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_num_reads_clean1</td>
<td>Int</td>
<td>The number of forward reads after cleaning as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_num_reads_raw1</td>
<td>Int</td>
<td>The number of input forward reads as calculated by fastq_scan</td>
</tr>
<tr>
<td>fastq_scan_raw1_json</td>
<td>File</td>
<td>The JSON file output from <code>fastq-scan</code> containing summary stats about raw forward read quality and length</td>
</tr>
<tr>
<td>fastq_scan_version</td>
<td>String</td>
<td>The version of fastq_scan</td>
</tr>
<tr>
<td>kraken_human</td>
<td>Float</td>
<td>Percent of human read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_human_dehosted</td>
<td>Float</td>
<td>Percent of human read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_report</td>
<td>File</td>
<td>Full Kraken report</td>
</tr>
<tr>
<td>kraken_report_dehosted</td>
<td>File</td>
<td>Full Kraken report after host removal</td>
</tr>
<tr>
<td>kraken_sc2</td>
<td>String</td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_sc2_dehosted</td>
<td>String</td>
<td>Percent of SARS-CoV-2 read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_target_organism</td>
<td>String</td>
<td>Percent of target organism read data detected using the Kraken2 software</td>
</tr>
<tr>
<td>kraken_target_organism_dehosted</td>
<td>String</td>
<td>Percent of target organism read data detected using the Kraken2 software after host removal</td>
</tr>
<tr>
<td>kraken_target_organism_name</td>
<td>String</td>
<td>The name of the target organism; e.g., "Monkeypox" or "Human immunodeficiency virus"</td>
</tr>
<tr>
<td>kraken_version</td>
<td>String</td>
<td>Version of Kraken software used</td>
</tr>
<tr>
<td>meanbaseq_trim</td>
<td>Float</td>
<td>Mean quality of the nucleotide basecalls aligned to the reference genome after primer trimming</td>
</tr>
<tr>
<td>meanmapq_trim</td>
<td>Float</td>
<td>Mean quality of the mapped reads to the reference genome after primer trimming</td>
</tr>
<tr>
<td>medaka_reference</td>
<td>String</td>
<td>Reference sequence used in medaka task</td>
</tr>
<tr>
<td>nextclade_aa_dels</td>
<td>String</td>
<td>Amino-acid deletions as detected by NextClade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_aa_subs</td>
<td>String</td>
<td>Amino-acid substitutions as detected by Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_clade</td>
<td>String</td>
<td>Nextclade clade designation, will be blank for Flu.</td>
</tr>
<tr>
<td>nextclade_docker</td>
<td>String</td>
<td>Docker image used to run Nextclade</td>
</tr>
<tr>
<td>nextclade_ds_tag</td>
<td>String</td>
<td>Dataset tag used to run Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_json</td>
<td>File</td>
<td>Nextclade output in JSON file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_lineage</td>
<td>String</td>
<td>Nextclade lineage designation</td>
</tr>
<tr>
<td>nextclade_qc</td>
<td>String</td>
<td>QC metric as determined by Nextclade. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_tsv</td>
<td>File</td>
<td>Nextclade output in TSV file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_version</td>
<td>String</td>
<td>The version of Nextclade software used</td>
</tr>
<tr>
<td>number_Degenerate</td>
<td>Int</td>
<td>Number of degenerate basecalls within the consensus assembly</td>
</tr>
<tr>
<td>number_N</td>
<td>Int</td>
<td>Number of fully ambiguous basecalls within the consensus assembly</td>
</tr>
<tr>
<td>number_Total</td>
<td>Int</td>
<td>Total number of nucleotides within the consensus assembly</td>
</tr>
<tr>
<td>pango_lineage</td>
<td>String</td>
<td>Pango lineage as determined by Pangolin</td>
</tr>
<tr>
<td>pango_lineage_expanded</td>
<td>String</td>
<td>Pango lineage without use of aliases; e.g., "BA.1" → "B.1.1.529.1"</td>
</tr>
<tr>
<td>pango_lineage_report</td>
<td>File</td>
<td>Full Pango lineage report generated by Pangolin</td>
</tr>
<tr>
<td>pangolin_assignment_version</td>
<td>String</td>
<td>The version of the pangolin software (e.g. PANGO or PUSHER) used for lineage assignment</td>
</tr>
<tr>
<td>pangolin_conflicts</td>
<td>String</td>
<td>Number of lineage conflicts as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_docker</td>
<td>String</td>
<td>Docker image used to run Pangolin</td>
</tr>
<tr>
<td>pangolin_notes</td>
<td>String</td>
<td>Lineage notes as determined by Pangolin</td>
</tr>
<tr>
<td>pangolin_versions</td>
<td>String</td>
<td>All Pangolin software and database versions</td>
</tr>
<tr>
<td>percent_reference_coverage</td>
<td>Float</td>
<td>Percent coverage of the reference genome after performing primer trimming; calculated as assembly_length_unambiguous / length of the reference genome (SC2: 29903) x 100</td>
</tr>
<tr>
<td>percentage_mapped_reads</td>
<td>Float</td>
<td>Percentage of reads that successfully aligned to the reference genome. This value is calculated by number of mapped reads / total number of reads x 100.</td>
</tr>
<tr>
<td>primer_bed_name</td>
<td>String</td>
<td>Name of the primer bed files used for primer trimming</td>
</tr>
<tr>
<td>qc_check</td>
<td>String</td>
<td>A string that indicates whether or not the sample passes a set of pre-determined and user-provided QC thresholds</td>
</tr>
<tr>
<td>qc_standard</td>
<td>File</td>
<td>The file used in the QC Check task containing the QC thresholds.</td>
</tr>
<tr>
<td>read1_aligned</td>
<td>File</td>
<td>Forward read file of only aligned reads</td>
</tr>
<tr>
<td>read1_dehosted</td>
<td>File</td>
<td>The dehosted forward reads file; suggested read file for SRA submission</td>
</tr>
<tr>
<td>samtools_version_stats</td>
<td>String</td>
<td>The version of SAMtools used to assess the quality of read mapping</td>
</tr>
<tr>
<td>sc2_s_gene_mean_coverage</td>
<td>Float</td>
<td>Mean read depth for the S gene in SARS-CoV-2</td>
</tr>
<tr>
<td>sc2_s_gene_percent_coverage</td>
<td>Float</td>
<td>Percent coverage of the S gene in SARS-CoV-2</td>
</tr>
<tr>
<td>seq_platform</td>
<td>String</td>
<td>Description of the sequencing methodology used to generate the input read data</td>
</tr>
<tr>
<td>theiacov_clearlabs_analysis_date</td>
<td>String</td>
<td>Date of analysis</td>
</tr>
<tr>
<td>theiacov_clearlabs_version</td>
<td>String</td>
<td>Version of PHB used for running the workflow</td>
</tr>
<tr>
<td>vadr_alerts_list</td>
<td>File</td>
<td>A file containing all of the fatal alerts as determined by VADR</td>
</tr>
<tr>
<td>vadr_all_outputs_tar_gz</td>
<td>File</td>
<td>A .tar.gz file (gzip-compressed tar archive file) containing all outputs from the VADR command v-annotate.pl. This file must be uncompressed &amp; extracted to see the many files within. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description of all files present within the archive. Useful when deeply investigating a sample's genome &amp; annotations.</td>
</tr>
<tr>
<td>vadr_classification_summary_file</td>
<td>File</td>
<td>Per-sequence tabular classification file. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#explanation-of-sqc-suffixed-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#explanation-of-sqc-suffixed-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_docker</td>
<td>String</td>
<td>Docker image used to run VADR</td>
</tr>
<tr>
<td>vadr_fastas_zip_archive</td>
<td>File</td>
<td>Zip archive containing all fasta files created during VADR analysis</td>
</tr>
<tr>
<td>vadr_feature_tbl_fail</td>
<td>File</td>
<td>5 column feature table output for failing sequences. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_feature_tbl_pass</td>
<td>File</td>
<td>5 column feature table output for passing sequences. See <a href="https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files">https://github.com/ncbi/vadr/blob/master/documentation/formats.md#format-of-v-annotatepl-output-files</a> for more complete description.</td>
</tr>
<tr>
<td>vadr_num_alerts</td>
<td>String</td>
<td>Number of fatal alerts as determined by VADR</td>
</tr>
<tr>
<td>variants_from_ref_vcf</td>
<td>File</td>
<td>Number of variants relative to the reference genome</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="tabbed-block">
<div class="admonition warning">
<p class="admonition-title">Overwrite Warning</p>
<p><strong>TheiaCoV_FASTA_Batch_PHB</strong> workflow will <strong>output results to the set-level data table in addition to overwriting the Pangolin &amp; Nextclade output columns in the sample-level data table</strong>. Users can view the set-level workflow output TSV file called <code>"Datatable"</code> to view exactly which columns were overwritten in the sample-level data table.</p>
</div>
<div class="searchable-table">
<table>
<thead>
<tr>
<th><strong>Variable</strong></th>
<th><strong>Type</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>datatable</td>
<td>File</td>
<td>Sample-level data table TSV file that was used to update the original sample-level data table in the last step of the TheiaCoV_FASTA_Batch workflow.</td>
</tr>
<tr>
<td>nextclade_json</td>
<td>File</td>
<td>Nextclade output in JSON file format. Will be blank for Flu</td>
</tr>
<tr>
<td>nextclade_tsv</td>
<td>File</td>
<td>Nextclade output in TSV file format. Will be blank for Flu</td>
</tr>
<tr>
<td>pango_lineage_report</td>
<td>File</td>
<td>Full Pango lineage report generated by Pangolin</td>
</tr>
<tr>
<td>theiacov_fasta_batch_analysis_date</td>
<td>String</td>
<td>Date that the workflow was run.</td>
</tr>
<tr>
<td>theiacov_fasta_batch_version</td>
<td>String</td>
<td>Version of the workflow that was used.</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="September 10, 2025 17:12:08 UTC">2025-09-10</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="September 11, 2024 17:48:55 UTC">2024-09-11</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      © 2022-2025 <a href="https://www.theiagen.com" target="_blank" rel="noopener">Theiagen Genomics</a>

    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/theiagen" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://bsky.app/profile/theiagen.bsky.social" target="_blank" rel="noopener" title="bsky.app" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M407.8 294.7c-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3M288 227.1c-26.1-50.7-97.1-145.2-163.1-191.8C61.6-9.4 37.5-1.7 21.6 5.5 3.3 13.8 0 41.9 0 58.4S9.1 194 15 213.9c19.5 65.7 89.1 87.9 153.2 80.7 3.3-.5 6.6-.9 10-1.4-3.3.5-6.6 1-10 1.4-93.9 14-177.3 48.2-67.9 169.9C220.6 589.1 265.1 437.8 288 361.1c22.9 76.7 49.2 222.5 185.6 103.4 102.4-103.4 28.1-156-65.8-169.9-3.3-.4-6.7-.8-10-1.3 3.4.4 6.7.9 10 1.3 64.1 7.1 133.6-15.1 153.2-80.7C566.9 194 576 75 576 58.4s-3.3-44.7-21.6-52.9c-15.8-7.1-40-14.9-103.2 29.8C385.1 81.9 314.1 176.4 288 227.1"></path></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://www.linkedin.com/company/theiagen" target="_blank" rel="noopener" title="www.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3M135.4 416H69V202.2h66.5V416zM102.2 96a38.5 38.5 0 1 1 0 77 38.5 38.5 0 1 1 0-77m282.1 320h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9z"></path></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "navigation.instant", "navigation.path", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.suggest", "toc.follow"], "search": "../../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"alias": true, "default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../../../javascripts/tablesort.js"></script>
      
        <script src="../../../javascripts/table-search.js"></script>
      
    
  
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>