# Reference_Genome_Consensus

## Quick Facts

| **Workflow Type** | **Applicable Kingdom** | **Last Known Changes** | **Command-line Compatibility** | **Workflow Level** |
|---|---|---|---|---|
###update this table ###################################
| [Standalone](../../workflows_overview/workflows_type.md/#link-to-workflow-type) | [Virus](../../workflows_overview/workflows_kingdom.md/#link-to-applicable-kingdom) | PHB vX.X.X | yes| Sample-level |

## Workflow_Name_On_Terra

A workflow for reference-guided genome assembly, read trimming, alignment, coverage filtering, and consensus sequence generation from ONT basecalled FASTQ files.

### Inputs

Input should be ordered as they appear on Terra

| **Terra Task Name** | **Variable** | **Type** | **Description** | **Default Value** | **Terra Status** |
|---|---|---|---|---|---|
| fastqc_se | **fastq_file** | File | Input ONT basecalled FASTQ file | N/A | Required |
| seqtk_trim | **left_trim** | Int | Number of bases to trim from the left (optional) | N/A | Optional |
| seqtk_trim | **right_trim** | Int | Number of bases to trim from the right (optional) | N/A | Optional |
| seqtk_trim | **min_length** | Int | Minimum read length to retain (optional) | N/A | Optional |
| minimap2_align_ont | **reference_file** | File | Reference genome file for alignment | N/A | Required |
| minimap2_align_ont | **sample_name** | String | Sample name for output file naming | N/A | Required |
| filter_coverage | **min_coverage** | Int | Minimum coverage for filtering regions | 10 | Optional |
| medaka_consensus | **medaka_model_override** | String | Override for Medaka model selection (optional) | N/A | Optional |

### Workflow Tasks

This workflow includes the following tasks:

??? task "`fastqc`: Run FastQC"
    Quality control check of the input FASTQ file.

    !!! techdetails "FastQC Technical Details"
        |  | Links |
        | --- | --- |
        | Task | [fastqc_se](../../tasks/quality_control/basic_statistics/task_fastqc.wdl) |
        | Software Source Code | [FastQC](https://github.com/s-andrews/FastQC) |
        | Software Documentation | [FastQC Docs](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) |

??? task "`seqtk`: Trimming and Filtering Reads"
    Trims and filters reads based on user-defined parameters such as left trim, right trim, and minimum read length.

    !!! techdetails "Seqtk Technical Details"
        |  | Links |
        | --- | --- |
        | Task | [seqtk_trim](../../tasks/quality_control/read_filtering/task_seqtk_trim.wdl) |
        | Software Source Code | [Seqtk](https://github.com/lh3/seqtk) |

??? task "`minimap2`: Align Reads to Reference"
    Aligns filtered reads to the reference genome.

    !!! techdetails "Minimap2 Technical Details"
        |  | Links |
        | --- | --- |
        | Task | [minimap2_align_ont](../../tasks/alignment/task_minimap2.wdl) |
        | Software Source Code | [Minimap2](https://github.com/lh3/minimap2) |
        | Original Publication(s) | [Li 2018](https://doi.org/10.1093/bioinformatics/bty191) |

??? task "`samtools`: Process BAM File and Calculate Coverage"
    Converts SAM to BAM, sorts and indexes the BAM file, and calculates coverage.

    !!! techdetails "Samtools Technical Details"
        |  | Links |
        | --- | --- |
        | Task | [samtools_process](../../tasks/utilities/file_handling/task_samtools_process.wdl) |
        | Software Source Code | [Samtools](http://www.htslib.org/) |

??? task "`filter_coverage`: Extract Regions with Sufficient Coverage"
    Filters aligned reads to retain regions meeting the minimum coverage threshold.

    !!! techdetails "Filter Coverage Technical Details"
        |  | Links |
        | --- | --- |
        | Task | [filter_coverage](../../tasks/utilities/file_handling/task_filter_coverage.wdl) |
        | Software Source Code | [Bedtools](https://bedtools.readthedocs.io/) |

??? task "`medaka`: Generate Polished Consensus Sequence"
    Generates a consensus sequence using Medaka.

    !!! techdetails "Medaka Technical Details"
        |  | Links |
        | --- | --- |
        | Task | [medaka_consensus](../../tasks/polishing/task_medaka.wdl) |
        | Software Source Code | [Medaka](https://github.com/nanoporetech/medaka) |
        | Software Documentation | [Medaka Docs](https://nanoporetech.github.io/medaka/) |

### Outputs

| **Variable** | **Type** | **Description** |
|---|---|---|
| fastqc_report | File | HTML report from FastQC |
| read_length_plot | File | Plot of read length distribution |
| coverage_file | File | Coverage file generated by Samtools |
| medaka_consensus_fasta | File | Polished consensus sequence in FASTA format |
| medaka_version | String | Version of Medaka used |
| medaka_model | String | Medaka model used for consensus generation |
| samtools_version | String | Version of Samtools used |

## References

> Li, H. (2018). Minimap2: pairwise alignment for nucleotide sequences. Bioinformatics, 34(18), 3094â€“3100. https://doi.org/10.1093/bioinformatics/bty191
> Medaka: Nanoporetech's tool for consensus sequence generation. https://github.com/nanoporetech/medaka